--- a/llvm/unittests/TableGen/CMakeLists.txt
+++ b/llvm/unittests/TableGen/CMakeLists.txt
@@ -1,7 +1,7 @@
-set(LLVM_LINK_COMPONENTS
-  TableGen
-  Support
-  )
+#set(LLVM_LINK_COMPONENTS
+#  TableGen
+#  Support
+#  )
 
 set(LLVM_TARGET_DEFINITIONS Automata.td)
 
@@ -15,4 +15,4 @@
   ParserEntryPointTest.cpp
   )
 
-target_link_libraries(TableGenTests PRIVATE LLVMTableGenCommon LLVMTableGen)
+target_link_libraries(TableGenTests PRIVATE LLVMTableGenCommon LLVMTableGen LLVMSupport)
