--- a/llvm/include/llvm/ExecutionEngine/JITLink/i386.h	2025-04-05 15:44:02.075557700 -0700
+++ b/llvm/include/llvm/ExecutionEngine/JITLink/i386.h	2025-04-05 15:44:32.085589500 -0700
@@ -16,6 +16,8 @@
 #include "llvm/ExecutionEngine/JITLink/JITLink.h"
 #include "llvm/ExecutionEngine/JITLink/TableManager.h"
 
+#undef i386
+
 namespace llvm::jitlink::i386 {
 /// Represets i386 fixups
 enum EdgeKind_i386 : Edge::Kind {
