--- a/clang/tools/clang-shlib/CMakeLists.txt	2025-04-07 11:32:30.213586600 -0700
+++ b/clang/tools/clang-shlib/CMakeLists.txt	2025-04-07 11:33:05.428487700 -0700
@@ -57,7 +57,7 @@
 
 # Optimize function calls for default visibility definitions to avoid PLT and
 # reduce dynamic relocations.
-if (NOT APPLE AND NOT MINGW AND NOT LLVM_LINKER_IS_SOLARISLD_ILLUMOS)
+if (NOT APPLE AND NOT MINGW AND NOT CYGWIN AND NOT LLVM_LINKER_IS_SOLARISLD_ILLUMOS)
   target_link_options(clang-cpp PRIVATE LINKER:-Bsymbolic-functions)
 endif()
 if (MINGW OR CYGWIN)
