From 7e9b0669061b7d3588fdfe18dc31a780e5ada69d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mateusz=20Miku=C5=82a?= <oss@mateuszmikula.dev>
Date: Fri, 7 Feb 2025 22:33:39 +0100
Subject: [PATCH 2/2] hack: add system search paths

---
 lld/MinGW/Driver.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lld/MinGW/Driver.cpp b/lld/MinGW/Driver.cpp
index 131bacdf1468..94acd68abd12 100644
--- a/lld/MinGW/Driver.cpp
+++ b/lld/MinGW/Driver.cpp
@@ -546,6 +546,10 @@ bool link(ArrayRef<const char *> argsArr, llvm::raw_ostream &stdoutOS,
     add("-exclude-symbols:" + StringRef(a->getValue()));
 
   std::vector<StringRef> searchPaths;
+  #ifdef __CYGWIN__
+    searchPaths.push_back("/usr/lib");
+    searchPaths.push_back("/usr/lib/w32api");
+  #endif
   for (auto *a : args.filtered(OPT_L)) {
     searchPaths.push_back(a->getValue());
     add("-libpath:" + StringRef(a->getValue()));
-- 
2.47.1.windows.2

