From 85fb3c828a59cdbe2aefef762c10a8bc0237c53a Mon Sep 17 00:00:00 2001
From: Mihail Konev <k.mvc@ya.ru>
Date: Sun, 5 Mar 2017 22:58:09 +0500
Subject: [PATCH 3/3] fix current dir inheritance for alt-f2 on msys2

---
 src/child.c   |  2 ++
 src/winmain.c | 13 +++++++++++++
 2 files changed, 15 insertions(+)

diff --git a/src/child.c b/src/child.c
index 11de61a..ba0f6e1 100644
--- a/src/child.c
+++ b/src/child.c
@@ -808,6 +808,8 @@ child_fork(int argc, char *argv[], int moni)
     // propagate shortcut-inherited icon
     if (icon_is_from_shortcut)
       setenv("MINTTY_ICON", cs__wcstoutf(cfg.icon), true);
+    // tell alt-f2 mintty instance that it is such one
+    setenv("MINTTY_IS_CHILD", "1", 1);
 
 #if CYGWIN_VERSION_DLL_MAJOR >= 1005
     execv("/proc/self/exe", argv);
diff --git a/src/winmain.c b/src/winmain.c
index 0ccca9a..789b09e 100644
--- a/src/winmain.c
+++ b/src/winmain.c
@@ -2185,6 +2185,12 @@ opts[] = {
   {0, 0, 0, 0}
 };
 
+static int getenvi(const char *varname) {
+  char *s = getenv(varname);
+  if (s == NULL) { return 0; }
+  return atoi(s);
+}
+
 int
 main(int argc, char *argv[])
 {
@@ -2788,6 +2794,13 @@ main(int argc, char *argv[])
       launcher_setup_env();
       launcher_setup_argv();
     }
+    // Ask /etc/post-install/05-home-dir.post not to "cd ~/" in an Alt-F2 window
+    {
+      if (getenvi("MINTTY_IS_CHILD")) {
+        putenv("CHERE_INVOKING=1");
+      }
+      unsetenv("MINTTY_IS_CHILD");
+    }
 
     // Create child process.
     child_create(
-- 
2.11.0

