--- mintty-3.8.3.orig/src/Makefile
+++ mintty-3.8.3/src/Makefile
@@ -130,8 +130,8 @@ scriptlib=-lusp10
 #downldlib="${SYSTEMROOT}/System32/urlmon.dll"
 downldlib=
 ifneq ($(platform), msys)
-  emojilib = -lGdiPlus
-  scriptlib += -lDWrite
+  emojilib = -lgdiplus
+  scriptlib += -ldwrite
 endif
 extralibs=$(scriptlib) $(downldlib) $(emojilib) -lshlwapi
 LDFLAGS := -L$(shell $(CC) -print-file-name=w32api) -static-libgcc
