--- mintty-2.7.3/src/termout.c.orig	2017-01-25 08:57:56.073058100 +0300
+++ mintty-2.7.3/src/termout.c	2017-01-25 08:58:34.766729800 +0300
@@ -1608,11 +1608,12 @@
         }
         else if (c >= '0' && c <= '9') {
           uint i = term.csi_argc - 1;
-          if (i < lengthof(term.csi_argv))
+          if (i < lengthof(term.csi_argv)) {
             term.csi_argv[i] = 10 * term.csi_argv[i] + c - '0';
             if ((int)term.csi_argv[i] < 0)
               term.csi_argv[i] = INT_MAX;  // capture overflow
             term.csi_argv_defined[i] = 1;
+          }
         }
         else if (c < 0x40)
           term.esc_mod = term.esc_mod ? 0xFF : c;
