diff -urN -x .build -x .inst -x .sinst automake-1.9.6-orig/tests/Makefile.in automake-1.9.6/tests/Makefile.in
--- automake-1.9.6-orig/tests/Makefile.in	2005-07-10 14:14:53.000000000 -0400
+++ automake-1.9.6/tests/Makefile.in	2005-08-28 23:28:38.453556800 -0400
@@ -34,6 +34,7 @@
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
+LIBOBJDIR =
 subdir = tests
 DIST_COMMON = README $(srcdir)/Makefile.am $(srcdir)/Makefile.in \
 	$(srcdir)/aclocal.in $(srcdir)/automake.in $(srcdir)/defs.in
@@ -66,7 +67,6 @@
 ECHO_T = @ECHO_T@
 EGREP = @EGREP@
 FGREP = @FGREP@
-GREP = @GREP@
 INSTALL_DATA = @INSTALL_DATA@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
@@ -102,27 +102,20 @@
 build_os = @build_os@
 build_vendor = @build_vendor@
 datadir = @datadir@
-datarootdir = @datarootdir@
-docdir = @docdir@
-dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host_alias = @host_alias@
-htmldir = @htmldir@
 includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
 libdir = @libdir@
 libexecdir = @libexecdir@
-localedir = @localedir@
 localstatedir = @localstatedir@
 mandir = @mandir@
 mkdir_p = @mkdir_p@
 oldincludedir = @oldincludedir@
-pdfdir = @pdfdir@
 pkgvdatadir = @pkgvdatadir@
 prefix = @prefix@
 program_transform_name = @program_transform_name@
-psdir = @psdir@
 sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
@@ -538,6 +531,7 @@
 pr300-ltlib.test \
 pr300-prog.test \
 pr307.test \
+pr401.test \
 prefix.test \
 primary.test \
 primary2.test \
