--- origsrc/automake-1.9.6/tests/subst.test	2005-05-14 16:21:27.000000000 -0400
+++ src/automake-1.9.6/tests/subst.test	2008-10-27 19:28:56.286828800 -0400
@@ -24,8 +24,14 @@
 . ./defs || exit 1
 
 cat >> configure.in << 'END'
-AC_SUBST($1)
-AC_SUBST([$]$1)  dnl this is the actual invocation that was used
+dnl This test used to have the following lines, which cannot have
+dnl worked sensibly with Autoconf for years, however:
+dnl AC_SUBST($1)
+dnl AC_SUBST([$]$1)  dnl this is the actual invocation that was used
+dnl
+AC_DEFUN([FOO],
+[AC_SUBST([$1])])
+FOO([BAR])
 END
 
 : > Makefile.am
