unchanged:
--- plotutils-2.6/libplotter/Makefile-orig.am	2017-11-07 12:37:12.864600600 -0500
+++ plotutils-2.6/libplotter/Makefile.am	2017-11-07 12:37:15.106167000 -0500
@@ -2,7 +2,7 @@
 
 lib_LTLIBRARIES = libplotter.la
 
-libplotter_la_LDFLAGS = -version-info 4:4:2
+libplotter_la_LDFLAGS = -no-undefined -version-info 4:4:2
 
 INCLUDES = $(X_CFLAGS) -I$(srcdir)/../include -DLIBPLOT -DLIBPLOTTER
 
only in patch2:
unchanged:
--- plotutils-2.6/libplot/Makefile-orig.am	2017-11-07 12:36:48.597127000 -0500
+++ plotutils-2.6/libplot/Makefile.am	2017-11-07 12:36:50.054730000 -0500
@@ -4,7 +4,7 @@
 
 EXTRA_DIST = DEDICATION HUMOR README-cgm README-gif README-hpgl README-tek VERSION
 
-libplot_la_LDFLAGS = -version-info 4:4:2
+libplot_la_LDFLAGS = -no-undefined -version-info 4:4:2
 
 INCLUDES = $(X_CFLAGS) -I$(srcdir)/../include -DLIBPLOT
 
