diff -Naur catgets-1.1-orig/Makefile catgets-1.1/Makefile
--- catgets-1.1-orig/Makefile	2013-05-01 19:04:15.486328100 +0000
+++ catgets-1.1/Makefile	2013-05-01 19:03:04.533203100 +0000
@@ -8,6 +8,10 @@
   EXEEXT=.exe
   ICONV_LIBOPT = -liconv
   INTL_LIBOPT = -lintl
+else ifeq ($(OSTYPE),msys)
+  EXEEXT=.exe
+  ICONV_LIBOPT = -liconv
+  INTL_LIBOPT = -lintl
 else
   EXEEXT=
   ICONV_LIBOPT =
@@ -17,6 +21,8 @@
 .PHONY: all
 ifeq ($(OSTYPE),cygwin)
 all: libcatgets.a cygcatgets1.dll gencat$(EXEEXT)
+else ifeq ($(OSTYPE),msys)
+all: libcatgets.a msys-catgets1.dll gencat$(EXEEXT)
 else
 all: libcatgets.a libcatgets.so.1.0 gencat$(EXEEXT)
 endif
@@ -41,6 +47,9 @@
 ifeq ($(OSTYPE),cygwin)
 cygcatgets1.dll: $(CATGETS_OBJS)
 	$(CCLD) -shared $(CATGETS_OBJS) -o $@ -Wl,--enable-auto-image-base -Xlinker --out-implib -Xlinker libcatgets.dll.a
+else ifeq ($(OSTYPE),msys)
+msys-catgets1.dll: $(CATGETS_OBJS)
+	$(CCLD) -shared $(CATGETS_OBJS) -o $@ -Wl,--enable-auto-image-base -Xlinker --out-implib -Xlinker libcatgets.dll.a
 else
 libcatgets.so.1.0: $(CATGETS_OBJS)
 	$(CCLD) -shared -o $@ -Wl,-soname,libcatgets.so.1 -lc $(CATGETS_OBJS)
@@ -73,6 +82,12 @@
 	install -s -m 755 cygcatgets1.dll  $(DESTDIR)/usr/bin/
 	mkdir -p $(DESTDIR)/usr/lib
 	install -m 644 libcatgets.dll.a $(DESTDIR)/usr/lib/
+else ifeq ($(OSTYPE),msys)
+install-so:
+	mkdir -p $(DESTDIR)/usr/bin
+	install -s -m 755 msys-catgets1.dll  $(DESTDIR)/usr/bin/
+	mkdir -p $(DESTDIR)/usr/lib
+	install -m 644 libcatgets.dll.a $(DESTDIR)/usr/lib/
 else
 install-so:
 	mkdir -p $(DESTDIR)/usr/lib
