From 3fbfc30d0be48ae120c11ce67fedbda8e1d60a27 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Sun, 18 Jan 2026 10:43:26 +0800
Subject: [PATCH] Fixes detecting of HAVE_SIGSETJMP in cygwin

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 740ded3..f67e7c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -172,7 +172,7 @@ AC_CHECK_TYPE(sighandler_t,[],[],[
 #endif])
 dnl This test was ripped from gnuplot's configure.in:
 AC_MSG_CHECKING(for sigsetjmp)
-AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <setjmp.h>]], [[jmp_buf env; sigsetjmp(env, 1);]])],[AC_MSG_RESULT(yes)
+AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <setjmp.h>]], [[sigjmp_buf env; sigsetjmp(env, 1);]])],[AC_MSG_RESULT(yes)
    AC_DEFINE(HAVE_SIGSETJMP,1,
              [ Define if we have sigsetjmp(). ])],[AC_MSG_RESULT(no)])
 
-- 
2.52.0.windows.1

