diff --git a/test/lit.cfg.py b/test/lit.cfg.py
index dacda6894a0..dd35dc83b8c 100644
--- a/test/lit.cfg.py
+++ b/test/lit.cfg.py
@@ -154,7 +154,7 @@ if not re.match(r'^x86_64.*-(windows-msvc|windows-gnu)$', config.target_triple):
     config.available_features.add('LP64')
 
 # [PR12920] "clang-driver" -- set if gcc driver is not used.
-if not re.match(r'.*-(cygwin)$', config.target_triple):
+if not re.match(r'.*-(cygwin|msys)$', config.target_triple):
     config.available_features.add('clang-driver')
 
 # Tests that are specific to the Apple Silicon macOS.
