--- libpsl-0.16.1/src/psl-make-dafsa	2016-12-12 04:11:05.000000000 -0600
+++ libpsl-0.16.1/src/psl-make-dafsa	2016-12-15 16:05:03.145989800 -0600
@@ -224,6 +224,7 @@ import sys
 import os.path
 import time
 import hashlib
+import locale
 
 class InputError(Exception):
   """Exception raised for errors in the input file."""
@@ -633,6 +634,8 @@ def main():
   if len(sys.argv) < 3:
     usage()
 
+  locale.setlocale(locale.LC_CTYPE, 'C.UTF-8')
+
   converter = words_to_cxx
   parser = parse_psl
   utf_mode = True
