--- origsrc/libpsl-0.16.0/src/Makefile.am	2016-12-06 13:00:23.000000000 -0600
+++ src/libpsl-0.16.0/src/Makefile.am	2016-12-15 04:11:54.570845600 -0600
@@ -9,15 +9,15 @@ lib_LTLIBRARIES = libpsl.la
 libpsl_la_SOURCES = psl.c lookup_string_in_fixed_set.c
 libpsl_la_CPPFLAGS = -I$(top_srcdir)/include -DPSL_DISTFILE=\"$(PSL_DISTFILE)\"
 # include ABI version information
-libpsl_la_LDFLAGS = -version-info $(LIBPSL_SO_VERSION)
+libpsl_la_LDFLAGS = -version-info $(LIBPSL_SO_VERSION) -no-undefined
 if WITH_LIBICU
   libpsl_la_LDFLAGS += $(LIBICU_LIBS)
 endif
 if WITH_LIBIDN2
-  libpsl_la_LDFLAGS += -lidn2 -lunistring
+  libpsl_la_LDFLAGS += -lidn2 -lunistring $(LTLIBICONV)
 endif
 if WITH_LIBIDN
-  libpsl_la_LDFLAGS += -lidn -lunistring
+  libpsl_la_LDFLAGS += -lidn -lunistring $(LTLIBICONV)
 endif
 
 # Build rule for suffix_dafsa.c
