--- libpsl-0.21.0/tests/Makefile.am.orig	2019-06-10 09:09:03.218431200 +0300
+++ libpsl-0.21.0/tests/Makefile.am	2019-06-10 09:09:56.178505300 +0300
@@ -33,9 +33,9 @@
 # dafsa.psl and dafsa_ascii.psl must be created before any test is executed
 # check-local target works in parallel to the tests, so the test suite will likely fail
 BUILT_SOURCES = psl.dafsa psl_ascii.dafsa
-psl.dafsa: $(top_srcdir)/list/public_suffix_list.dat
+psl.dafsa: $(PSL_FILE)
 	$(top_srcdir)/src/psl-make-dafsa --output-format=binary "$(PSL_FILE)" psl.dafsa
-psl_ascii.dafsa: $(top_srcdir)/list/public_suffix_list.dat
+psl_ascii.dafsa: $(PSL_FILE)
 	$(top_srcdir)/src/psl-make-dafsa --output-format=binary --encoding=ascii "$(PSL_FILE)" psl_ascii.dafsa
 
 clean-local:
