diff --git a/src/lib/libast/port/astconf.c b/src/lib/libast/port/astconf.c
index b6b58027..396cd46d 100644
--- a/src/lib/libast/port/astconf.c
+++ b/src/lib/libast/port/astconf.c
@@ -1355,6 +1355,24 @@ astgetconf(const char* name, const char* path, const char* value, int flags, Err
	Lookup_t	look;
	Sfio_t*		tmp;

+#ifdef __CYGWIN__
+	/*
+	 * hack for Cygwin to make Cygwin UNC paths (with leading
+	 * two slashes) work.
+	 *
+	 * Testcase:
+	 * cd into UNC path, print "pwd -P", result must start with
+	 * "//<hostname>/<path>"
+	 * ---- snip ----
+	 * $ bash -c 'cd "//derfw1fs10.global.loc/rfw" ; path_to_kshdir/ksh.exe -c "pwd -P ; true" ; true'
+	 * //derfw1fs10.global.loc/rfw
+	 * ---- snip ----
+	 *
+	 */
+	if (name && (!strcmp(name, "PATH_LEADING_SLASHES"))) {
+		return "1";
+	}
+#endif /* __CYGWIN__ */
	if (!name)
	{
		if (path)
