--- ksh-1.0.10/src/cmd/ksh93/tests/shtests.orig	2025-03-13 11:48:28.444460100 -0700
+++ ksh-1.0.10/src/cmd/ksh93/tests/shtests	2025-03-13 11:48:30.253240200 -0700
@@ -284,7 +284,7 @@
 
 tmp=$(
 	d=${TMPDIR:-/tmp}/ksh93.shtests.$$.${RANDOM:-0}
-	mkdir -m700 -- "$d" && CDPATH= cd -P -- "$d" && pwd
+	mkdir -- "$d" && CDPATH= cd -P -- "$d" && pwd
 ) || {
 	echo 'mkdir failed' >&2
 	exit 1
@@ -372,7 +372,7 @@
 		for lang in $locales
 		do	o=$u
 			tmp_s=$tmp/$u.$lang
-			mkdir -m700 "$tmp_s" || exit
+			mkdir "$tmp_s" || exit
 			cd "$tmp_s" || exit
 			if	[[ $lang == C ]]
 			then	lang=
@@ -415,7 +415,7 @@
 		o="$u(shcomp)"
 		echo test $o begins ${time:+"at $(date +%Y-%m-%d+%H:%M:%S)"}
 		tmp_s=$tmp/$u.shcomp
-		mkdir -m700 "$tmp_s" || exit
+		mkdir "$tmp_s" || exit
 		# Must define aliases from _common in script itself so they are expanded at compile time.
 		# Replace the first few lines with those to avoid affecting $LINENO; they are comments anyway.
 		num_alias=$(grep -c '^alias [[:alnum:]_]\{2,\}=' _common)
