--- ksh-1.0.10/bin/package.orig	2025-03-13 11:30:37.844745700 -0700
+++ ksh-1.0.10/bin/package	2025-03-13 11:33:55.678158000 -0700
@@ -1,3 +1,4 @@
+#!/bin/sh
 ########################################################################
 #                                                                      #
 #               This software is part of the ast package               #
@@ -1792,6 +1793,12 @@
 	esac
 fi
 
+if test /bin -ef /usr/bin; then
+	# /bin and /usr/bin are the same - but MSYS2 gcc dies badly if invoked
+	# as /bin/cc, so make sure /usr/bin is first
+	DEFPATH="/usr/bin:$DEFPATH"
+fi
+
 export DEFPATH  # for iffe, etc.
 
 PATH=$(sanitize_PATH "/opt/ast/bin:$DEFPATH:$PATH")
