From 1ddc4cd699de9df111a2223b017f8f3a1952b7f8 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 21 Aug 2015 09:52:47 +0100
Subject: [PATCH 11/N] Add debugging for strace make_command_line

---
 winsup/utils/mingw/strace.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/winsup/utils/mingw/strace.cc b/winsup/utils/mingw/strace.cc
index cdfc3d6..e7ec38e 100644
--- a/winsup/utils/mingw/strace.cc
+++ b/winsup/utils/mingw/strace.cc
@@ -352,6 +352,7 @@ create_child (char **argv)
     flags |= CREATE_NEW_CONSOLE | CREATE_NEW_PROCESS_GROUP;
 
   make_command_line (one_line, argv);
+  printf ("create_child: %s\n", one_line.buf);
 
   SetConsoleCtrlHandler (NULL, 0);
 /* Commit message for this code was:
