From dfabd11a56f0ab060cd6789b1f5674a032060923 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Sat, 10 Jan 2026 05:08:03 +0800
Subject: [PATCH] fixes for building with gcc 15.

---
 src/libs/libgroff/getopt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libs/libgroff/getopt.c b/src/libs/libgroff/getopt.c
index 6efa529..015284d 100644
--- a/src/libs/libgroff/getopt.c
+++ b/src/libs/libgroff/getopt.c
@@ -122,7 +122,7 @@ static struct _getopt_data getopt_data;
    whose names are inconsistent.  */
 
 #ifndef getenv
-extern char *getenv ();
+extern char *getenv (const char *name);
 #endif
 
 #endif /* not __GNU_LIBRARY__ */
-- 
2.52.0.windows.1

