diff -Naur groff-1.22.2/config.guess groff-1.22.2m/config.guess
--- groff-1.22.2/config.guess	2013-02-07 12:06:06.000000000 +0000
+++ groff-1.22.2m/config.guess	2013-03-18 13:13:24.000000000 +0000
@@ -851,6 +851,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur groff-1.22.2/config.rpath groff-1.22.2m/config.rpath
--- groff-1.22.2/config.rpath	2013-02-07 12:06:06.000000000 +0000
+++ groff-1.22.2m/config.rpath	2013-03-18 13:14:27.000000000 +0000
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur groff-1.22.2/src/libs/gnulib/build-aux/compile groff-1.22.2m/src/libs/gnulib/build-aux/compile
--- groff-1.22.2/src/libs/gnulib/build-aux/compile	2013-02-07 12:06:12.000000000 +0000
+++ groff-1.22.2m/src/libs/gnulib/build-aux/compile	2013-03-18 13:17:07.000000000 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur groff-1.22.2/src/libs/gnulib/build-aux/config.guess groff-1.22.2m/src/libs/gnulib/build-aux/config.guess
--- groff-1.22.2/src/libs/gnulib/build-aux/config.guess	2013-02-07 12:06:12.000000000 +0000
+++ groff-1.22.2m/src/libs/gnulib/build-aux/config.guess	2013-03-18 13:17:49.000000000 +0000
@@ -851,6 +851,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur groff-1.22.2/src/libs/gnulib/lib/Makefile.am groff-1.22.2m/src/libs/gnulib/lib/Makefile.am
--- groff-1.22.2/src/libs/gnulib/lib/Makefile.am	2013-02-07 12:06:12.000000000 +0000
+++ groff-1.22.2m/src/libs/gnulib/lib/Makefile.am	2013-03-18 13:18:46.000000000 +0000
@@ -113,7 +113,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -Naur groff-1.22.2/src/libs/gnulib/lib/Makefile.in groff-1.22.2m/src/libs/gnulib/lib/Makefile.in
--- groff-1.22.2/src/libs/gnulib/lib/Makefile.in	2013-02-07 12:06:12.000000000 +0000
+++ groff-1.22.2m/src/libs/gnulib/lib/Makefile.in	2013-03-18 13:19:11.000000000 +0000
@@ -942,7 +942,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
