--- origsrc/man-1.6g/configure	2010-12-31 21:28:46.000000000 +0100
+++ src/man-1.6g/configure	2013-04-25 11:47:03.558665800 +0200
@@ -26,7 +26,7 @@ LANGOPT="+lang {$LANGUAGES|all|none}"
 # (Indeed, -r may cause the terminal to get into funny states.
 # Very inconvenient. For viewing pages in strange locales, set LC_*.)
 #
-DEFAULTLESSOPT="-is"
+DEFAULTLESSOPT="-isrR"
 #
 # Note that not creating any cat directories (/var/cache/man or so)
 # and not making man suid or sgid is recommended.
@@ -366,7 +366,7 @@ rm -f conftest
 #   for i in /bin:/usr/bin:$PATH
 # does not work everywhere.)
 # Some people might prefer having $PATH first in PREPATH
-PREPATH=/bin:/usr/bin:/usr/ucb:/usr/local/bin:$PATH
+PREPATH=/usr/bin:/bin:/usr/ucb:/usr/local/bin:$PATH
 IFS0="$IFS"
 IFS=":$IFS"
 for i in $PREPATH
@@ -480,18 +480,18 @@ else
   then
     if test $Fnroff = "missing"
     then
-      nroff="nroff -Tlatin1 -mandoc"
+      nroff="nroff -c -mandoc 2>/dev/null"
     else
-      nroff="$Fnroff -Tlatin1 -mandoc"
+      nroff="$Fnroff -c -mandoc 2>/dev/null"
     fi
     troff="troff -mandoc"
     echo "Warning: could not find groff"
   else
     if test $Fnroff = "missing"
     then
-      nroff="$Fgroff -Tlatin1 -mandoc"
+      nroff="$Fgroff -c -mandoc 2>/dev/null"
     else
-      nroff="$Fnroff -Tlatin1 -mandoc"
+      nroff="$Fnroff -c -mandoc 2>/dev/null"
     fi
     troff="$Fgroff -Tps -mandoc"
     jnroff="$Fgroff -Tnippon -mandocj"
@@ -964,7 +964,7 @@ fi
 
 # What sections do we anticipate?
 
-tmpsections="1 1p 8 2 3 3p 4 5 6 7 9 0p tcl n l p o"
+tmpsections="1 1p 8 2 3 3p 4 5 6 7 9 0p tcl n l p o m 8c mingw"
 
 if [ x$default = x ]; then
   echo ""
--- origsrc/man-1.6g/man/bg/makewhatis.man	2006-01-12 02:28:18.000000000 +0100
+++ src/man-1.6g/man/bg/makewhatis.man	2013-04-25 11:47:03.559665900 +0200
@@ -65,7 +65,7 @@ makewhatis \- създава база от данни на w
 от пътя до
 .IR "оригиналите на страниците " или " готовите страници от ръководството" .
 Ако не е зададена стойност, се използва
-.I  \'1 2 3 4 5 6 7 8 9 n l\'
+.I  \'1 2 3 4 5 6 7 8 9 n l m 8c mingw\'
 .TP 
 .BI -c " път_до_готовите_страници"
 Прочитат се готовите страници в
--- origsrc/man-1.6g/man/el/makewhatis.man	2005-08-21 01:26:06.000000000 +0200
+++ src/man-1.6g/man/el/makewhatis.man	2013-04-25 11:47:03.560666000 +0200
@@ -72,7 +72,7 @@ makewhatis \- ДзмйпхсгЯб фзт вЬуещт дедп
 .IR catpath " . "
 Бн брпхуйЬжей з ерйлпгЮ, 
 з фймЮ фзт иещсеЯфбй ьфй еЯнбй
-.I  \'1 2 3 4 5 6 7 8 9 n l\' 
+.I  \'1 2 3 4 5 6 7 8 9 n l m 8c mingw\'
 .TP 
 .BI -c " catpath"
 Убсюнпнфбй пй рспдйбмпсцщмЭнет уелЯдет рпх всЯукпнфбй уфз
--- origsrc/man-1.6g/man/fr/makewhatis.man	2006-01-26 00:17:36.000000000 +0100
+++ src/man-1.6g/man/fr/makewhatis.man	2013-04-25 11:47:03.561666000 +0200
@@ -72,7 +72,7 @@ Chercher dans les sections
 de
 .IR manpath " ou " catpath .
 Si cette option est absente, sa valeur est
-.IR  " " \' "1 2 3 4 5 6 7 8 9 n l"\'
+.IR  " " \' "1 2 3 4 5 6 7 8 9 n l m 8c mingw"\'
 .TP 
 .BI -c " catpath"
 Les pages prйformatйes de manuel situйes dans
--- origsrc/man-1.6g/man/ro/makewhatis.man	2005-08-21 01:26:06.000000000 +0200
+++ src/man-1.6g/man/ro/makewhatis.man	2013-04-25 11:47:03.563666100 +0200
@@ -64,7 +64,7 @@ Cautг оn
 ale
 .IR cale_man " sau " cale_cat .
 Dacг opюiunea este absentг, valoarea sa este consideratг a fi
-.I  \'1 2 3 4 5 6 7 8 9 n l\'
+.I  \'1 2 3 4 5 6 7 8 9 n l m 8c mingw\'
 .TP 
 .BI -c " cale_cat"
 Paginile preformatate de manual aflate оn
--- origsrc/man-1.6g/src/apropos.sh	2005-08-21 01:26:06.000000000 +0200
+++ src/man-1.6g/src/apropos.sh	2013-04-25 11:47:03.564666200 +0200
@@ -19,9 +19,9 @@ program=`basename $0`
 
 # When man pages in your favorite locale look to grep like binary files
 # (and you use GNU grep) you may want to add the 'a' option to *grepopt1.
-aproposgrepopt1='i'
+aproposgrepopt1='ai'
 aproposgrepopt2=''
-whatisgrepopt1='iw'
+whatisgrepopt1='aiw'
 whatisgrepopt2='^'
 grepopt1=$%apropos_or_whatis%grepopt1
 grepopt2=$%apropos_or_whatis%grepopt2
--- origsrc/man-1.6g/src/man.conf.in	2010-10-26 02:22:24.000000000 +0200
+++ src/man-1.6g/src/man.conf.in	2013-04-25 11:47:03.565666200 +0200
@@ -32,15 +32,18 @@
 # This file is also read by man in order to find how to call nroff, less, etc.,
 # and to determine the correspondence between extensions and decompressors.
 #
-# MANBIN		/usr/local/bin/man
+# MANBIN /usr/bin/man
 #
 # Every automatically generated MANPATH includes these fields
 #
-MANPATH	/usr/man
 MANPATH	/usr/share/man
-MANPATH	/usr/local/man
-MANPATH	/usr/local/share/man
+MANPATH	/usr/ssl/man
+MANPATH	/usr/man
+MANPATH	/usr/X11R6/share/man
 MANPATH	/usr/X11R6/man
+MANPATH	/usr/local/share/man
+MANPATH	/usr/local/man
+MANPATH	/opt/gnome/man
 #
 # Uncomment if you want to include one of these by default
 #
@@ -64,9 +67,8 @@ MANPATH_MAP	/usr/bin		/usr/share/man
 MANPATH_MAP	/usr/sbin		/usr/share/man
 MANPATH_MAP	/usr/local/bin		/usr/local/share/man
 MANPATH_MAP	/usr/local/sbin		/usr/local/share/man
-MANPATH_MAP	/usr/X11R6/bin		/usr/X11R6/man
-MANPATH_MAP	/usr/bin/X11		/usr/X11R6/man
-MANPATH_MAP	/usr/bin/mh		/usr/share/man
+MANPATH_MAP	/usr/X11R6/bin		/usr/X11R6/share/man
+MANPATH_MAP	/opt/gnome/bin		/opt/gnome/man
 #
 # NOAUTOPATH keeps man from automatically adding directories that look like
 # manual page directories to the path.
