--- a/meson.options
+++ b/meson.options
@@ -1 +1,2 @@
 option('docs', type : 'boolean', value : false)
+option('static-libalpm', type : 'boolean', value : false, description : 'Link libalpm statically')

--- a/meson.build
+++ b/meson.build
@@ -13,7 +13,7 @@
 # Dependencies
 python = import('python').find_installation('python3', pure: false)
 python_dep = python.dependency()
-libalpm_dep = dependency('libalpm')
+libalpm_dep = dependency('libalpm', static: get_option('static-libalpm'))
 
 common_c_args = [
     '-DVERSION="@0@"'.format(meson.project_version()),
