--- origsrc/ruby-1.8.7-p352/ext/tk/extconf.rb	2011-06-29 22:53:30.000000000 -0500
+++ src/ruby-1.8.7-p352/ext/tk/extconf.rb	2011-10-26 19:07:48.530860500 -0500
@@ -106,7 +106,7 @@ end
 # fuctions
 ##############################################################
 def is_win32?
-  /mswin|mingw|cygwin|bccwin/ =~ RUBY_PLATFORM
+  /mswin|mingw|bccwin/ =~ RUBY_PLATFORM
 end
 
 def is_macosx?
@@ -553,6 +553,7 @@ def get_ext_list()
   exts = [CONFIG['DLEXT']]
   exts.concat %w(dll) if is_win32?
   exts.concat %w(bundle dylib) if is_macosx?
+  exts.concat %w(dll.a) if /cygwin/ =~ RUBY_PLATFORM
 
   if TkLib_Config["tcltk-stubs"] || TkLib_Config['enable-shared'] == false
     exts.unshift "lib" if is_win32?
@@ -602,7 +602,7 @@ def libcheck_for_tclConfig(tcldir, tkdir
         #puts "check #{file} #{$1} #{tclfunc} #{tcldir}"
         #find_library($1, tclfunc, tcldir)
         if (tclconf && tclconf["TCL_SHARED_BUILD"] == "0") ||
-            (ext != CONFIG['DLEXT'] && ext == CONFIG['LIBEXT']) || ext == "a"
+            (!tclconf && ((ext != CONFIG['DLEXT'] && ext == CONFIG['LIBEXT']) || ext == "a"))
           # static link
           tcllibs = $libs + " -DSTATIC_BUILD " + file.quote
 
@@ -645,7 +645,7 @@ def libcheck_for_tclConfig(tcldir, tkdir
         #puts "check #{file} #{$1} #{tkfunc} #{tkdir}"
         # find_library($1, tkfunc, tkdir)
         if (tkconf && tkconf["TCL_SHARED_BUILD"] == "0") ||
-            (ext != CONFIG['DLEXT'] && ext == CONFIG['LIBEXT']) || ext == "a"
+            (!tkconf && ((ext != CONFIG['DLEXT'] && ext == CONFIG['LIBEXT']) || ext == "a"))
           # static link
           tklibs = " -DSTATIC_BUILD " + file.quote
 
@@ -1762,7 +1762,6 @@ unless is_win32?
   print("\n") # progress
 end
 $CPPFLAGS ||= ""
-$CPPFLAGS += ' -D_WIN32' if /cygwin/ =~ RUBY_PLATFORM
 
 # Does ruby have nativethread ?
 TkLib_Config["ruby_with_thread"] =
--- origsrc/ruby-1.9.3-p194/ext/tk/stubs.c	2011-05-15 06:55:52.000000000 -0500
+++ src/ruby-1.9.3-p194/ext/tk/stubs.c	2012-08-19 13:54:34.862671000 -0500
@@ -71,7 +71,7 @@ _nativethread_consistency_check(ip)
 
 #if defined USE_TCL_STUBS && defined USE_TK_STUBS
 
-#if defined _WIN32 || defined __CYGWIN__
+#if defined _WIN32
 #  ifdef HAVE_RUBY_RUBY_H
 #    include "ruby/util.h"
 #  else
@@ -100,6 +100,10 @@ _nativethread_consistency_check(ip)
 #  undef DLEXT
 #  define DLEXT ".dylib"
 # endif
+# if defined(__CYGWIN__)
+#  undef DLEXT
+#  define DLEXT ".dll"
+# endif
 #endif
 
 static DL_HANDLE tcl_dll = (DL_HANDLE)0;
