#fix for upstream bug #12071
diff -ruN ruby-2.3.0.orig/lib/resolv.rb ruby-2.3.0/lib/resolv.rb
--- ruby-2.3.0.orig/lib/resolv.rb	2016-03-30 18:19:22.528842900 +0200
+++ ruby-2.3.0/lib/resolv.rb	2016-03-30 18:21:53.232329900 +0200
@@ -168,7 +168,7 @@
 
   class Hosts
     begin
-      raise LoadError unless /mswin|mingw|cygwin|msys/ =~ RUBY_PLATFORM
+      raise LoadError unless /mswin|mingw/ =~ RUBY_PLATFORM
       require 'win32/resolv'
       DefaultFileName = Win32::Resolv.get_hosts_path
     rescue LoadError
@@ -967,7 +967,7 @@
         if File.exist? filename
           config_hash = Config.parse_resolv_conf(filename)
         else
-          if /mswin|cygwin|msys|mingw|bccwin/ =~ RUBY_PLATFORM
+          if /mswin|mingw|bccwin/ =~ RUBY_PLATFORM
             require 'win32/resolv'
             search, nameserver = Win32::Resolv.get_resolv_info
             config_hash = {}
