--- src/ruby-2.4.0/lib/rubygems.rb.orig	2016-10-30 02:32:48.000000000 -0400
+++ src/ruby-2.4.0/lib/rubygems.rb	2017-03-07 23:37:48.944394100 -0500
@@ -121,7 +121,6 @@
 
   WIN_PATTERNS = [
     /bccwin/i,
-    /cygwin/i,
     /djgpp/i,
     /mingw/i,
     /mswin/i,
--- src/ruby-3.4.1.orig/.bundle/gems/syslog-0.2.0/ext/syslog/extconf.rb	2024-12-24 23:43:22.000000000 -0800
+++ src/ruby-3.4.1/.bundle/gems/syslog-0.2.0/ext/syslog/extconf.rb	2025-01-16 13:37:44.696083900 -0800
@@ -11,7 +11,7 @@
 end
 
 def windows?
-  RbConfig::CONFIG["host_os"] =~ /mswin|mingw|cygwin/
+  RbConfig::CONFIG["host_os"] =~ /mswin|mingw/
 end
 
 if windows?
