--- src/ruby-2.5.0/configure.ac.orig	2016-12-17 02:48:58.000000000 -0500
+++ src/ruby-2.5.0/configure.ac	2017-03-07 22:19:28.115668500 -0500
@@ -3607,7 +3607,7 @@ else
     [darwin*], [
 	RUBY_APPEND_OPTION(XLDFLAGS, [-Wl,-u,_objc_msgSend])
 	DLEXT=bundle],
-    [cygwin*|mingw*|*djgpp*], [
+    [mingw*|*djgpp*], [
 	LOAD_RELATIVE=1
 	DLEXT=so],
     [
@@ -3918,7 +3918,7 @@ AS_CASE("$enable_shared", [yes], [
     [interix*], [
 	LIBRUBYARG_SHARED='-L. -L${libdir} -l$(RUBY_SO_NAME)'
 	],
-    [mingw*|cygwin*|mswin*], [
+    [mingw*|mswin*], [
 	LIBRUBY_RELATIVE=yes
 	])
 ], [
@@ -4336,7 +4336,7 @@ test "$program_suffix" != NONE &&
 
 RUBY_INSTALL_NAME="${ri_prefix}"'$(RUBY_BASE_NAME)'"${ri_suffix}"
 AS_CASE(["$target_os"],
-  [cygwin*|mingw*], [
+  [mingw*], [
     RUBYW_INSTALL_NAME="${ri_prefix}"'$(RUBYW_BASE_NAME)'"${ri_suffix}"
     rubyw_install_name='$(RUBYW_INSTALL_NAME)'
     ])
