--- ruby-2.5.0/template/ruby.pc.in.orig	2018-02-08 21:31:08.867483500 -0500
+++ ruby-2.5.0/template/ruby.pc.in	2018-02-08 21:35:41.229623700 -0500
@@ -8,7 +8,7 @@ includedir=@includedir@
 MAJOR=@MAJOR@
 MINOR=@MINOR@
 TEENY=@TEENY@
-ruby_version=@ruby_version@
+ruby_version=@RUBY_PROGRAM_VERSION@
 RUBY_API_VERSION=@RUBY_API_VERSION@
 RUBY_PROGRAM_VERSION=@RUBY_PROGRAM_VERSION@
 RUBY_BASE_NAME=@RUBY_BASE_NAME@
