--- ruby-2.7.7/configure.ac.orig	2022-11-24 13:42:08.000000000 +0100
+++ ruby-2.7.7/configure.ac	2022-12-03 16:05:33.824160000 +0100
@@ -3522,7 +3522,7 @@
 	AS_CASE(["$target_os"],
 	[cygwin*], [
 	    AS_IF([test x"$enable_shared" = xyes], [
-		LIBRUBY_SO='cyg$(RUBY_SO_NAME)'.dll
+		LIBRUBY_SO='msys-$(RUBY_SO_NAME)'.dll
 		LIBRUBY_DLDFLAGS="${LIBRUBY_DLDFLAGS}"' $(RUBYDEF)'
 	    ])
 	    ],
--- ruby-2.7.7/test/fiddle/helper.rb.orig	2022-11-24 13:42:08.000000000 +0100
+++ ruby-2.7.7/test/fiddle/helper.rb	2022-12-03 16:07:14.857663800 +0100
@@ -8,8 +8,8 @@
 
 case RUBY_PLATFORM
 when /cygwin/
-  libc_so = "cygwin1.dll"
-  libm_so = "cygwin1.dll"
+  libc_so = "msys-2.0.dll"
+  libm_so = "msys-2.0.dll"
 when /android/
   libdir = '/system/lib'
   if [0].pack('L!').size == 8
