--- ruby-3.4.1.orig/ext/win32/resolv/extconf.rb	2024-12-24 23:43:20.000000000 -0800
+++ ruby-3.4.1/ext/win32/resolv/extconf.rb	2025-01-16 21:55:13.492757500 -0800
@@ -1,5 +1,5 @@
 require 'mkmf'
-if RUBY_ENGINE == "ruby" and have_library('iphlpapi', 'GetNetworkParams')
+if RUBY_ENGINE == "ruby" and have_library('iphlpapi', 'GetNetworkParams', ['windows.h', 'iphlpapi.h'])
   create_makefile('win32/resolv')
 else
   File.write('Makefile', "all clean install:\n\t@echo Done: $(@)\n")
