From d637d257d25e507667a358797690e6577e37f9a4 Mon Sep 17 00:00:00 2001
From: Gennady Feldman <gena01@gmail.com>
Date: Wed, 13 Dec 2017 07:23:15 -0500
Subject: [PATCH 2/3] configure: handle MSYS2

---
 configure.ac | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure.ac b/configure.ac
index 028d5559..94f99385 100644
--- a/configure.ac
+++ b/configure.ac
@@ -585,6 +585,10 @@ case "$host_os" in
 		AC_MSG_RESULT(cygwin)
 		PLATFORM=cygwin
 		;;
+	*msys*)
+		AC_MSG_RESULT(msys)
+		PLATFORM=msys
+		;;
 	*)
 		AC_MSG_RESULT(unknown)
 		PLATFORM=unknown
-- 
2.29.1

