From 2361ec975b010d3735bbfed17e4c210f80d3c87f Mon Sep 17 00:00:00 2001
From: Gennady Feldman <gena01@gmail.com>
Date: Wed, 13 Dec 2017 07:23:15 -0500
Subject: [PATCH 3/3] do not check directory permissions

---
 tmux.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tmux.c b/tmux.c
index 53b984d7..9c44b8d1 100644
--- a/tmux.c
+++ b/tmux.c
@@ -150,7 +150,11 @@ make_label(const char *label)
 		errno = ENOTDIR;
 		goto fail;
 	}
+#ifdef __MSYS__
+	if (sb.st_uid != uid) {
+#else /* __MSYS__ */
 	if (sb.st_uid != uid || (sb.st_mode & S_IRWXO) != 0) {
+#endif /* __MSYS__ */
 		errno = EACCES;
 		goto fail;
 	}
-- 
2.29.1

