From 0446961d33abb2889c2baa3bbadae34c3ff773d9 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 27 Oct 2020 12:43:35 +0100
Subject: [PATCH] Avoid unsupported GCC option

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 3da954ae..32835e1e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -44,7 +44,7 @@ endif
 CPPFLAGS += -iquote.
 endif
 
-CFLAGS += -Wno-unused-parameter -Wno-unused-variable -Wno-null-pointer-arithmetic
+CFLAGS += -Wno-unused-parameter -Wno-unused-variable
 CFLAGS += -Wno-deprecated-declarations -Wno-format-nonliteral
 
 # Set flags for Solaris.
-- 
2.29.1

