diff --unified --recursive --text --new-file tmate-2.4.0.orig/configure.ac tmate-2.4.0/configure.ac
--- tmate-2.4.0.orig/configure.ac	2024-09-14 16:52:14.116386988 +0200
+++ tmate-2.4.0/configure.ac	2024-09-14 16:52:46.076480650 +0200
@@ -201,7 +201,7 @@
 
 PKG_CHECK_MODULES(
   MSGPACK,
-  msgpack >= 1.1.0,
+  msgpack-c >= 1.1.0,
   [
     CPPFLAGS="$MSGPACK_CFLAGS $CPPFLAGS"
     LIBS="$MSGPACK_LIBS $LIBS"
@@ -210,7 +210,7 @@
   found_msgpack=no
 )
 if test "x$found_msgpack" = xno; then
-  AC_MSG_ERROR("msgpack >= 1.1.0 not found")
+  AC_MSG_ERROR("msgpack-c >= 1.1.0 not found")
 fi
 
 PKG_CHECK_MODULES(
