From c8d2dced8ce29c2bfcbc07bd302b71b43d3eafb9 Mon Sep 17 00:00:00 2001
From: "albfan" <albertofanjul@gmail.com>
Date: Sun, 6 Aug 2017 17:56:42 +0200
Subject: [PATCH] Compile for Msys2

---
 Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Makefile b/Makefile
index e4d31d9..03e9305 100644
--- a/Makefile
+++ b/Makefile
@@ -73,6 +73,11 @@ ifeq ($(UNAME_O),Cygwin)
 	LIBS += -liconv
 	EXT = .exe
 endif
+ifeq ($(UNAME_O),Msys)
+	CFLAGS += -I/mingw$(ARCH)/lib/libzip/include
+	LIBS += -liconv -llibzip -lzip -lz -L/mingw$(ARCH)/lib
+	EXT = .exe
+endif
 ifneq ($(MINGW32),)
 	CFLAGS += -I$(REGEX_DIR) -I$(ZLIB_DIR) -I$(ICONV_DIR)/include/ -I$(LIBZIP_DIR)/lib/
 	LIBS = $(REGEX_DIR)/regex.o
-- 
2.10.1

