For backwards compatibility

--- origsrc/tcl8.5.10/tcl.m4	2011-10-24 20:47:16.878248200 -0500
+++ src/tcl8.5.10/tcl.m4	2011-10-24 20:50:38.013444700 -0500
@@ -3106,6 +3106,7 @@ AC_DEFUN([TEA_SETUP_COMPILER_CC], [
 
     AC_PROG_CC
     AC_PROG_CPP
+    AC_PROG_INSTALL
 
     #--------------------------------------------------------------------
     # Checks to see if the make program sets the $MAKE variable.
