diff -Naur inetutils-1.9.2-orig/configure.ac inetutils-1.9.2/configure.ac
--- inetutils-1.9.2-orig/configure.ac	2014-12-12 11:55:10.606800000 +0300
+++ inetutils-1.9.2/configure.ac	2014-12-12 11:55:10.653600000 +0300
@@ -560,7 +560,13 @@
   AC_MSG_CHECKING(hosts_ctl in -lwrap);
   save_LIBS=$LIBS
   LIBS="$save_LIBS -lwrap"
-  AC_TRY_LINK([int allow_severity = 1; int deny_severity = 1;],
+  AC_TRY_LINK([#ifdef __CYGWIN__
+	       extern int allow_severity = 1;
+	       extern int deny_severity = 1;
+	       #else
+	       int allow_severity = 1;
+	       int deny_severity = 1;
+	       #endif],
               hosts_ctl(), eval "ac_cv_lib_wrap_hosts_ctl=yes",
               eval "ac_cv_lib_wrap_hosts_ctl=no")
   LIBS=$save_LIBS
@@ -596,7 +602,7 @@
 		  sys/proc.h sys/select.h sys/time.h sys/wait.h \
                   sys/resource.h \
 		  stropts.h tcpd.h utmp.h utmpx.h unistd.h \
-                  vis.h osockaddr.h], [], [], [
+                  vis.h osockaddr.h crypt.h], [], [], [
 #include <sys/types.h>
 #ifdef HAVE_STDIO_H
 # include <stdio.h>
@@ -860,7 +866,11 @@
 
 AC_DECL_SYS_SIGLIST
 
-AC_CHECK_DECLS(crypt, , , [#include <unistd.h>])
+AC_CHECK_DECLS(crypt, , ,
+      [IU_FLUSHLEFT([#include <unistd.h>
+                     #ifdef HAVE_CRYPT_H
+                     # include <crypt.h>
+                     #endif])])
 
 # EWOULDBLOCK is more or less the BSD version of posix EAGAIN.
 IU_CHECK_MACRO(EWOULDBLOCK, [#include <errno.h>], , ,
@@ -887,6 +897,8 @@
       iu_syslog_includes), , iu_syslog_includes),)
 undefine([iu_syslog_includes])
 
+IU_CHECK_MACRO(TIOCNOTTY, [#include <sys/ioctl.h>],,,)
+
 # Some systems don't declare common functions (especially if they
 # return int), at least in the expected header file.  Check.
 AC_CHECK_DECLS([fclose, pclose], , , [#include <stdio.h>])
@@ -999,8 +1011,6 @@
     [Define to 1 for a system using streams for ptys])
   ;;
 *cygwin*)
-  CFLAGS="${CFLAGS} -DNCURSES_STATIC"
-  LDFLAGS="${LDFLAGS} -Wl,-static"
   LIBS="/usr/lib/textmode.o ${LIBS}"
   AC_DEFINE(UTMPX, 1, [FIXME])
   ;;
diff -Naur inetutils-1.9.2-orig/man/Makefile.am inetutils-1.9.2/man/Makefile.am
--- inetutils-1.9.2-orig/man/Makefile.am	2013-09-26 12:36:38.000000000 +0300
+++ inetutils-1.9.2/man/Makefile.am	2014-12-12 11:55:10.653600000 +0300
@@ -216,4 +216,4 @@
 		--include=$(srcdir)/$(TOOL).h2m		\
 		--output=$(TOOL).$(SECTION)		\
 		--section $(SECTION)			\
-		$(mapped_name)
+		$(mapped_name)$(EXEEXT)
