diff -durNp inetutils-1.9.4.orig/man/syslogd.8 inetutils-1.9.4/man/syslogd.8
--- inetutils-1.9.4.orig/man/syslogd.8	2015-05-12 20:17:12.000000000 +0800
+++ inetutils-1.9.4/man/syslogd.8	2017-05-14 21:25:31.093750000 +0800
@@ -30,6 +30,14 @@ print debug information (implies \fB\-\-
 override configuration directory (default:
 \fI\,/usr/local/etc/syslog.d\/\fP)
 .TP
+\fB\-D\fR, \fB\-\-no\-daemonize\fR
+This is a synonym for \fB\-\-no\-detach\fR, and is provided for
+backwards compatibility with previous Cygwin ports.  This flag
+(or its synonym) is necessary when installing syslogd as service
+started from cygrunsrv.  The usual way to +install syslogd as
+service is to run the syslogd-config script.  See
+/usr/share/doc/Cygwin/inetutils.README for more information.
+.TP
 \fB\-f\fR, \fB\-\-rcfile\fR=\fI\,FILE\/\fR
 override configuration file (default:
 \fI\,/usr/local/etc/syslog.conf\/\fP)
diff -durNp inetutils-1.9.4.orig/src/syslogd.c inetutils-1.9.4/src/syslogd.c
--- inetutils-1.9.4.orig/src/syslogd.c	2015-05-12 20:13:25.000000000 +0800
+++ inetutils-1.9.4/src/syslogd.c	2017-05-14 21:27:45.562500000 +0800
@@ -348,6 +348,7 @@ static struct argp_option argp_options[]
   {"mark", 'm', "INTVL", 0, "specify timestamp interval in minutes"
    " (0 for no timestamping)", GRP+1},
   {"no-detach", 'n', NULL, 0, "do not enter daemon mode", GRP+1},
+  {"no-daemonize", 'D', NULL, 0, "Synonym for -n", GRP+1},
   {"no-forward", OPT_NO_FORWARD, NULL, 0, "do not forward any messages "
    "(overrides --hop)", GRP+1},
 #ifdef PATH_KLOG
@@ -432,6 +433,7 @@ parse_opt (int key, char *arg, struct ar
       break;
 
     case 'n':
+	case 'D':
       NoDetach = 1;
       break;
 
@@ -1108,7 +1110,11 @@ printsys (const char *msg)
   char *lp, *q, line[MAXLINE + 1];
   const char *p;
 
+#ifdef __CYGWIN__
+  strcpy (line, "kernel: ");
+#else
   strcpy (line, "vmunix: ");
+#endif
   lp = line + strlen (line);
   for (p = msg; *p != '\0';)
     {
