diff -Naur inetutils-1.9.2-orig/man/syslogd.8 inetutils-1.9.2/man/syslogd.8
--- inetutils-1.9.2-orig/man/syslogd.8	2013-12-23 15:26:57.000000000 +0300
+++ inetutils-1.9.2/man/syslogd.8	2014-12-12 11:55:13.134000000 +0300
@@ -30,6 +30,14 @@
 override configuration directory (default:
 \fI/usr/local/etc/syslog.d\fP)
 .TP
+\fB\-D\fR, \fB\-\-no\-daemonize\fR
+This is a synonym for \fB\-\-no\-detach\fR, and is provided for
+backwards compatibility with previous Cygwin ports.  This flag
+(or its synonym) is necessary when installing syslogd as service
+started from cygrunsrv.  The usual way to +install syslogd as
+service is to run the syslogd-config script.  See
+/usr/share/doc/Cygwin/inetutils.README for more information.
+.TP
 \fB\-f\fR, \fB\-\-rcfile\fR=\fIFILE\fR
 override configuration file (default:
 \fI/usr/local/etc/syslog.conf\fP)
diff -Naur inetutils-1.9.2-orig/src/syslogd.c inetutils-1.9.2/src/syslogd.c
--- inetutils-1.9.2-orig/src/syslogd.c	2013-10-29 13:38:56.000000000 +0300
+++ inetutils-1.9.2/src/syslogd.c	2014-12-12 11:55:13.149600000 +0300
@@ -348,6 +348,7 @@
   {"mark", 'm', "INTVL", 0, "specify timestamp interval in minutes"
    " (0 for no timestamping)", GRP+1},
   {"no-detach", 'n', NULL, 0, "do not enter daemon mode", GRP+1},
+  {"no-daemonize", 'D', NULL, 0, "Synonym for -n", GRP+1},
   {"no-forward", OPT_NO_FORWARD, NULL, 0, "do not forward any messages "
    "(overrides --hop)", GRP+1},
 #ifdef PATH_KLOG
@@ -431,6 +432,7 @@
       break;
 
     case 'n':
+    case 'D':
       NoDetach = 1;
       break;
 
@@ -1097,7 +1099,11 @@
   char *lp, *q, line[MAXLINE + 1];
   const char *p;
 
+#ifdef __CYGWIN__
+  strcpy (line, "kernel: ");
+#else
   strcpy (line, "vmunix: ");
+#endif
   lp = line + strlen (line);
   for (p = msg; *p != '\0';)
     {
