diff -durN inetutils-1.9.4.orig/.version inetutils-1.9.4/.version
--- inetutils-1.9.4.orig/.version	1970-01-01 08:00:00.000000000 +0800
+++ inetutils-1.9.4/.version	2017-05-14 21:33:59.421875000 +0800
@@ -0,0 +1 @@
+1.9.4
diff -durN inetutils-1.9.4.orig/configure.ac inetutils-1.9.4/configure.ac
--- inetutils-1.9.4.orig/configure.ac	2017-05-14 21:30:40.875000000 +0800
+++ inetutils-1.9.4/configure.ac	2017-05-14 21:36:54.406250000 +0800
@@ -230,10 +230,15 @@
 AM_CONDITIONAL([ENABLE_libls], [test "$enable_libls" = yes])
 
 # At least OpenSolaris is missing <protocols/talkd.h>.
-AC_CHECK_HEADER(protocols/talkd.h, , ,
+AC_CHECK_HEADER(protocols/osockaddr.h, , ,
   [IU_FLUSHLEFT([#include <sys/types.h>
 		 #include <sys/socket.h>])])
 
+AC_CHECK_HEADER(protocols/talkd.h, , ,
+  [IU_FLUSHLEFT([#include <sys/types.h>
+		 #include <sys/socket.h>
+		 #include <protocols/osockaddr.h>])])
+		 
 if test "$ac_cv_header_protocols_talkd_h" = no; then
   AC_MSG_WARN([protocols/talkd.h is not available, not building talk, nor talkd])
   IU_DISABLE_TARGET(talk)
@@ -611,7 +616,7 @@
 		  sys/proc.h sys/select.h sys/time.h sys/wait.h \
                   sys/resource.h \
 		  stropts.h tcpd.h utmp.h utmpx.h unistd.h \
-                  vis.h osockaddr.h crypt.h], [], [], [
+                  vis.h crypt.h], [], [], [
 #include <sys/types.h>
 #ifdef HAVE_STDIO_H
 # include <stdio.h>
diff -durN inetutils-1.9.4.orig/libinetutils/daemon.c inetutils-1.9.4/libinetutils/daemon.c
--- inetutils-1.9.4.orig/libinetutils/daemon.c	2017-05-14 21:30:39.781250000 +0800
+++ inetutils-1.9.4/libinetutils/daemon.c	2017-05-14 21:37:41.578125000 +0800
@@ -56,6 +56,7 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <string.h>
 #include <unused-parameter.h>
 
 /*
