diff -Naur inetutils-1.9.2-orig/.version inetutils-1.9.2/.version
--- inetutils-1.9.2-orig/.version	1970-01-01 03:00:00.000000000 +0300
+++ inetutils-1.9.2/.version	2014-12-12 11:55:14.241600000 +0300
@@ -0,0 +1 @@
+1.9.2
diff -Naur inetutils-1.9.2-orig/configure.ac inetutils-1.9.2/configure.ac
--- inetutils-1.9.2-orig/configure.ac	2014-12-12 11:55:10.996800000 +0300
+++ inetutils-1.9.2/configure.ac	2014-12-12 11:55:14.241600000 +0300
@@ -226,10 +226,15 @@
 AC_SUBST(libls_BUILD)
 
 # At least OpenSolaris is missing <protocols/talkd.h>.
-AC_CHECK_HEADER(protocols/talkd.h, , ,
+AC_CHECK_HEADER(protocols/osockaddr.h, , ,
   [IU_FLUSHLEFT([#include <sys/types.h>
 		 #include <sys/socket.h>])])
 
+AC_CHECK_HEADER(protocols/talkd.h, , ,
+  [IU_FLUSHLEFT([#include <sys/types.h>
+		 #include <sys/socket.h>
+		 #include <protocols/osockaddr.h>])])
+
 if test "$ac_cv_header_protocols_talkd_h" = no; then
   AC_MSG_WARN([protocols/talkd.h is not available, not building talk, nor talkd])
   IU_DISABLE_TARGET(talk)
@@ -602,7 +607,7 @@
 		  sys/proc.h sys/select.h sys/time.h sys/wait.h \
                   sys/resource.h \
 		  stropts.h tcpd.h utmp.h utmpx.h unistd.h \
-                  vis.h osockaddr.h crypt.h], [], [], [
+                  vis.h crypt.h], [], [], [
 #include <sys/types.h>
 #ifdef HAVE_STDIO_H
 # include <stdio.h>
diff -Naur inetutils-1.9.2-orig/libinetutils/daemon.c inetutils-1.9.2/libinetutils/daemon.c
--- inetutils-1.9.2-orig/libinetutils/daemon.c	2014-12-12 11:55:11.355600000 +0300
+++ inetutils-1.9.2/libinetutils/daemon.c	2014-12-12 11:55:14.241600000 +0300
@@ -56,6 +56,7 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <string.h>
 #include <unused-parameter.h>
 
 /*
diff -Naur inetutils-1.9.2-orig/Makefile.am inetutils-1.9.2/Makefile.am
--- inetutils-1.9.2-orig/Makefile.am	2013-09-26 12:36:38.000000000 +0300
+++ inetutils-1.9.2/Makefile.am	2014-12-12 11:55:14.241600000 +0300
@@ -18,6 +18,8 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see `http://www.gnu.org/licenses/'.
 
+ACLOCAL_AMFLAGS = -I m4 -I am
+
 EXTRA_DIST = paths ChangeLog.0 summary.sh.in CHECKLIST
 
 SUBDIRS = lib \
