diff -durN gc-7.6.0.orig/config.guess gc-7.6.0/config.guess
--- gc-7.6.0.orig/config.guess	2016-08-03 04:23:34.000000000 +0800
+++ gc-7.6.0/config.guess	2017-05-13 19:58:35.656250000 +0800
@@ -867,6 +867,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -durN gc-7.6.0.orig/configure.ac gc-7.6.0/configure.ac
--- gc-7.6.0.orig/configure.ac	2016-08-03 04:21:36.000000000 +0800
+++ gc-7.6.0/configure.ac	2017-05-13 20:15:39.718750000 +0800
@@ -263,7 +263,7 @@
      *-*-irix*)
         AC_DEFINE(GC_IRIX_THREADS)
         ;;
-     *-*-cygwin*)
+     *-*-cygwin* | *-*-msys*)
         AC_DEFINE(GC_WIN32_THREADS)
         if test "${enable_parallel_mark}" != no; then
           AC_DEFINE(PARALLEL_MARK)
@@ -458,7 +458,7 @@
 # We never want libdl on darwin. It is a fake libdl that just ends up making
 # dyld calls anyway.  The same applies to Cygwin.
   *-*-darwin*) ;;
-  *-*-cygwin*) ;;
+  *-*-cygwin* | *-*-msys*) ;;
   *)
     AC_CHECK_LIB(dl, dlopen, THREADDLLIBS="$THREADDLLIBS -ldl")
     ;;
@@ -521,7 +521,7 @@
 if test "$GCC" = yes; then
   if test "${enable_cplusplus}" = yes; then
     case "$host" in
-      *-*-cygwin* | *-*-mingw*)
+      *-*-cygwin* | *-*-msys* | *-*-mingw*)
         AC_MSG_CHECKING([whether libsupc++ required])
         SUPC="`$CXX -print-file-name=libsupc++.a 2>/dev/null`"
         if test -n "$SUPC" -a "$SUPC" != "libsupc++.a"; then
