From 1b09e8244b027a40db30ceaaf77b932975c38e39 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 13 Jun 2014 00:12:33 +0100
Subject: [PATCH 1/2] more debugging info

---
 lib/libalpm/util.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/libalpm/util.c b/lib/libalpm/util.c
index 79495be..be12fb7 100644
--- a/lib/libalpm/util.c
+++ b/lib/libalpm/util.c
@@ -562,7 +562,7 @@ int _alpm_run_chroot(alpm_handle_t *handle, const char *cmd, char *const argv[])
 		umask(0022);
 		execv(cmd, argv);
 		/* execv only returns if there was an error */
-		fprintf(stderr, _("call to execv failed (%s)\n"), strerror(errno));
+		fprintf(stderr, _("call to execv (%s) failed (%s)\n"), cmd, strerror(errno));
 		exit(1);
 	} else {
 		/* this code runs for the parent only (wait on the child) */
@@ -606,7 +606,7 @@ int _alpm_run_chroot(alpm_handle_t *handle, const char *cmd, char *const argv[])
 		if(WIFEXITED(status)) {
 			_alpm_log(handle, ALPM_LOG_DEBUG, "call to waitpid succeeded\n");
 			if(WEXITSTATUS(status) != 0) {
-				_alpm_log(handle, ALPM_LOG_ERROR, _("command failed to execute correctly\n"));
+				_alpm_log(handle, ALPM_LOG_ERROR, _("command (%s) failed to execute correctly\n"), cmd);
 				retval = 1;
 			}
 		} else if(WIFSIGNALED(status) != 0) {
-- 
2.0.0

