From 05b56288358aa9001f6c160f6fd7431551e8a2fd Mon Sep 17 00:00:00 2001
From: Renato Silva <br.renatosilva@gmail.com>
Date: Wed, 15 Oct 2014 16:01:04 -0300
Subject: [PATCH 3/3] Remove useless, unreliable and buggy code from Bazaar
 support.

---
 scripts/makepkg.sh.in | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/scripts/makepkg.sh.in b/scripts/makepkg.sh.in
index 0ef77ca..8d0ea66 100644
--- a/scripts/makepkg.sh.in
+++ b/scripts/makepkg.sh.in
@@ -444,23 +444,6 @@ download_bzr() {
 			exit 1
 		fi
 	elif (( ! HOLDVER )); then
-		# Make sure we are fetching the right repo
-		local distant_url="$(bzr info $url 2> /dev/null | sed -n '/branch root/{s/  branch root: //p;q;}')"
-		local local_url="$(bzr config parent_location -d $dir)"
-		if [[ -n $distant_url ]]; then
-			if [[ $distant_url != "$local_url" ]]; then
-				error "$(gettext "%s is not a branch of %s")" "$dir" "$url"
-				plain "$(gettext "Aborting...")"
-				exit 1
-			fi
-		else
-			if [[ $url != "$local_url" ]] ; then
-				error "$(gettext "%s is not a branch of %s")" "$dir" "$url"
-				error "$(gettext "The local URL is %s")" "$local_url"
-				plain "$(gettext "Aborting...")"
-				exit 1
-			fi
-		fi
 		msg2 "$(gettext "Pulling %s ...")" "${displaylocation}"
 		cd_safe "$dir"
 		if ! bzr pull "$url" --overwrite; then
-- 
2.1.2

