From 2948b4550a1de0390c72f043237ed97e9d717865 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Sun, 19 Oct 2014 16:45:35 +0100
Subject: [PATCH 3/3] use busybox for msys2 post installs

---
 src/pacman/pacman.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/pacman/pacman.c b/src/pacman/pacman.c
index df1ea83..c55135c 100644
--- a/src/pacman/pacman.c
+++ b/src/pacman/pacman.c
@@ -1024,6 +1024,21 @@ static void cl_to_log(int argc, char *argv[])
  */
 int main(int argc, char *argv[])
 {
+
+#ifdef __MSYS__
+	char *path = NULL;
+   char *oldpath = getenv("PATH");
+   char *newpath = "/usr/busybox:";
+	int pathsize = 0;
+	if (oldpath)
+		pathsize = strlen(newpath) + strlen(oldpath);
+	if (pathsize) 
+		path = alloca(pathsize);
+	if (path)
+		if(snprintf(path, pathsize, "%s%s", newpath, oldpath))
+			setenv("PATH", path, 1);
+#endif
+
 	int ret = 0;
 	size_t i;
 	struct sigaction new_action, old_action;
-- 
2.1.2