From 0d6fcafb95f85d7c40b0127840a1a6736fd14d9e Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Tue, 28 Feb 2023 20:30:50 +0100
Subject: [PATCH 04/N] test fixes

---
 test/pacman/pmtest.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/test/pacman/pmtest.py b/test/pacman/pmtest.py
index f5f033f..da2041b 100644
--- a/test/pacman/pmtest.py
+++ b/test/pacman/pmtest.py
@@ -297,8 +297,10 @@ def run(self, pacman):
         # Change to the tmp dir before running pacman, so that local package
         # archives are made available more easily.
         time_start = time.time()
-        self.retcode = subprocess.call(cmd, stdout=output, stderr=output,
-                cwd=os.path.join(self.root, util.TMPDIR), env={'LC_ALL': 'C'})
+        self.retcode = subprocess.call(cmd,
+                cwd=os.path.join(self.root, util.TMPDIR),
+                env={'LC_ALL': 'C', 'PATH': os.environ['PATH']},
+                stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
         time_end = time.time()
         vprint("\ttime elapsed: %.2fs" % (time_end - time_start))
 
