diff -Naur pacman-5.0.1-orig/lib/libalpm/Makefile.am pacman-5.0.1/lib/libalpm/Makefile.am
--- pacman-5.0.1-orig/lib/libalpm/Makefile.am	2015-10-18 08:38:28.000000000 +0300
+++ pacman-5.0.1/lib/libalpm/Makefile.am	2016-07-15 09:42:23.014037400 +0300
@@ -64,7 +64,7 @@
 	sha2.h sha2.c
 endif
 
-libalpm_la_LDFLAGS = -no-undefined -version-info $(LIB_VERSION_INFO)
+libalpm_la_LDFLAGS = -no-undefined -version-info $(LIB_VERSION_INFO) -Wc,--static -Wc,-static-libgcc
 
 libalpm_la_CFLAGS = \
 	$(AM_CFLAGS) \
@@ -69,9 +69,9 @@
 	$(NETTLE_CFLAGS)
 
 libalpm_la_LIBADD = \
-	$(LTLIBINTL) \
 	$(GPGME_LIBS) \
 	$(LIBARCHIVE_LIBS) \
 	$(LIBCURL_LIBS) \
 	$(LIBSSL_LIBS) \
-	$(NETTLE_LIBS)
+	$(NETTLE_LIBS) \
+	$(LTLIBINTL)
diff -Naur pacman-5.0.1-orig/src/pacman/Makefile.am pacman-5.0.1/src/pacman/Makefile.am
--- pacman-5.0.1-orig/src/pacman/Makefile.am	2015-12-05 15:40:17.000000000 +0300
+++ pacman-5.0.1/src/pacman/Makefile.am	2016-07-15 09:42:23.016037400 +0300
@@ -24,6 +24,8 @@
 AM_CFLAGS = -pedantic -D_GNU_SOURCE $(WARNING_CFLAGS) \
 	$(LIBARCHIVE_CFLAGS)
 
+AM_LDFLAGS = -Wc,--static -Wc,-static-libgcc
+
 if USE_GIT_VERSION
 GIT_VERSION := $(shell sh -c 'git describe --abbrev=4 --dirty | sed s/^v//')
 AM_CPPFLAGS += -DGIT_VERSION=\"$(GIT_VERSION)\"
diff -Naur pacman-5.0.1-orig/src/util/Makefile.am pacman-5.0.1/src/util/Makefile.am
--- pacman-5.0.1-orig/src/util/Makefile.am	2015-05-12 07:00:54.000000000 +0300
+++ pacman-5.0.1/src/util/Makefile.am	2016-07-15 09:42:23.019037400 +0300
@@ -17,6 +17,7 @@
 
 AM_CFLAGS = -pedantic -D_GNU_SOURCE $(WARNING_CFLAGS) \
 	$(LIBARCHIVE_CFLAGS)
+AM_LDFLAGS = -Wc,--static -Wc,-static-libgcc
 
 cleanupdelta_SOURCES = cleanupdelta.c
 cleanupdelta_LDADD = $(top_builddir)/lib/libalpm/.libs/libalpm.la
