diff -Naur pacman-5.1.0-orig/lib/libalpm/version.c pacman-5.1.0/lib/libalpm/version.c
--- pacman-5.1.0-orig/lib/libalpm/version.c	2018-05-14 03:02:18.000000000 +0300
+++ pacman-5.1.0/lib/libalpm/version.c	2018-06-21 12:53:58.862056200 +0300
@@ -50,7 +50,11 @@
 	/* se points to version terminator */
 	se = strrchr(s, '-');

+#ifdef __MSYS__
+	if(*s == '~') {
+#else
 	if(*s == ':') {
+#endif
 		epoch = evr;
 		*s++ = '\0';
 		version = s;
@@ -250,7 +254,7 @@
 		return 0;
 	}

-	/* Parse both versions into [epoch:]version[-release] triplets. We probably
+	/* Parse both versions into [epoch:|~]version[-release] triplets. We probably
 	 * don't need epoch and release to support all the same magic, but it is
 	 * easier to just run it all through the same code. */
 	full1 = strdup(a);
diff -Naur pacman-5.1.0-orig/scripts/libmakepkg/util/pkgbuild.sh.in pacman-5.1.0/scripts/libmakepkg/util/pkgbuild.sh.in
--- pacman-5.1.0-orig/scripts/libmakepkg/util/pkgbuild.sh.in	2018-05-12 16:15:04.000000000 +0300
+++ pacman-5.1.0/scripts/libmakepkg/util/pkgbuild.sh.in	2018-06-21 12:53:58.872056200 +0300
@@ -149,7 +149,11 @@
 ##
 get_full_version() {
 	if (( epoch > 0 )); then
-		printf "%s\n" "$epoch:$pkgver-$pkgrel"
+		if [[ -n $MSYSTEM ]]; then
+			printf "%s\n" "${epoch}~${pkgver}-${pkgrel}"
+		else
+			printf "%s\n" "$epoch:$pkgver-$pkgrel"
+		fi
 	else
 		printf "%s\n" "$pkgver-$pkgrel"
 	fi