diff -Naur pacman-5.0.1-orig/lib/libalpm/libalpm.pc.in pacman-5.0.1/lib/libalpm/libalpm.pc.in
--- pacman-5.0.1-orig/lib/libalpm/libalpm.pc.in	2015-12-05 15:40:17.000000000 +0300
+++ pacman-5.0.1/lib/libalpm/libalpm.pc.in	2016-07-15 09:42:23.555037400 +0300
@@ -9,4 +9,4 @@
 Requires.private: libarchive @pc_crypto@ @pc_libcurl@ @pc_gpgme@
 Cflags: -I${includedir} @LFS_CFLAGS@
 Libs: -L${libdir} -lalpm
-Libs.private: @LIBS@ @pc_gpgme_libs@
+Libs.private: @LIBS@ @pc_gpgme_libs@ -lintl -liconv
