diff -ur pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/arch_specific.sh.in pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/arch_specific.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/arch_specific.sh.in	2021-01-08 13:26:12.922712100 -0800
+++ pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/arch_specific.sh.in	2021-01-08 14:58:00.405221100 -0800
@@ -61,7 +61,7 @@
 		for a in ${arch[@]}; do
 			if [[ $a == "any" ]]; then
 				for i in ${pkgbuild_schema_arch_arrays[@]}; do
-					if exists_function_variable "package_$pkg" "${i}_${a}"; then
+					if exists_function_variable "$(get_package_func_name "$pkg")" "${i}_${a}"; then
 						error "$(gettext "Can not provide architecture specific variables for the '%s' architecture: %s")" "any" "${i}_${a}"
 						ret=1
 					fi
@@ -70,7 +70,7 @@
 
 			for i in ${pkgbuild_schema_arrays[@]} ${pkgbuild_schema_strings[@]}; do
 				in_array "$i" ${pkgbuild_schema_arch_arrays[@]} && continue
-				if exists_function_variable "package_$pkg" "${i}_${a}"; then
+				if exists_function_variable "$(get_package_func_name "$pkg")" "${i}_${a}"; then
 					error "$(gettext "%s can not be architecture specific: %s")" "$i" "${i}_${a}"
 					ret=1
 				fi
diff -ur pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/backup.sh.in pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/backup.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/backup.sh.in	2021-01-08 13:26:12.922712100 -0800
+++ pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/backup.sh.in	2021-01-09 12:57:22.838387200 -0800
@@ -35,7 +35,7 @@
 
 	backup_list=("${backup[@]}")
 	for name in "${pkgname[@]}"; do
-		if extract_function_variable "package_$name" backup 1 list; then
+		if extract_function_variable "$(get_package_func_name "$name")" backup 1 list; then
 			backup_list+=("${list[@]}")
 		fi
 	done
diff -ur pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/changelog.sh.in pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/changelog.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/changelog.sh.in	2021-01-08 13:26:12.922712100 -0800
+++ pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/changelog.sh.in	2021-01-09 12:58:55.648412200 -0800
@@ -40,7 +40,7 @@
 	local pkgname_backup=(${pkgname[@]})
 	local pkgname
 	for pkgname in "${pkgname_backup[@]}"; do
-		if extract_function_variable "package_$pkgname" changelog 0 file; then
+		if extract_function_variable "$(get_package_func_name "$pkgname")" changelog 0 file; then
 			changelog_list+=("$file")
 		fi
 	done
diff -ur pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/install.sh.in pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/install.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/install.sh.in	2021-01-08 13:26:12.938338100 -0800
+++ pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/install.sh.in	2021-01-09 12:59:06.846194300 -0800
@@ -40,7 +40,7 @@
 	local pkgname_backup=(${pkgname[@]})
 	local pkgname
 	for pkgname in "${pkgname_backup[@]}"; do
-		extract_function_variable "package_$pkgname" install 0 file
+		extract_function_variable "$(get_package_func_name "$pkgname")" install 0 file
 		install_list+=("$file")
 	done
 
diff -ur pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/options.sh.in pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/options.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/options.sh.in	2021-01-08 13:26:12.922712100 -0800
+++ pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/options.sh.in	2021-01-09 13:00:04.816250400 -0800
@@ -35,7 +35,7 @@
 
 	options_list=("${options[@]}")
 	for name in "${pkgname[@]}"; do
-		if extract_function_variable "package_$name" options 1 list; then
+		if extract_function_variable "$(get_package_func_name "$name")" options 1 list; then
 			options_list+=("${list[@]}")
 		fi
 	done
diff -ur pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/package_function.sh.in pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/package_function.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/package_function.sh.in	2021-01-08 13:26:12.922712100 -0800
+++ pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/package_function.sh.in	2021-01-09 13:02:28.684662900 -0800
@@ -34,10 +34,10 @@
 	local i ret=0
 
 	if (( ${#pkgname[@]} == 1 )); then
-		if have_function 'package' && have_function "package_$pkgname"; then
-			error "$(gettext "Conflicting %s and %s functions in %s")" "package()" "package_$pkgname()" "$BUILDFILE"
+		if have_function 'package' && have_function "$(get_package_func_name "$pkgname")"; then
+			error "$(gettext "Conflicting %s and %s functions in %s")" "package()" "$(get_package_func_name "$pkgname")" "$BUILDFILE"
 			ret=1
-		elif have_function 'build' && ! { have_function 'package' || have_function "package_$pkgname"; }; then
+		elif have_function 'build' && ! { have_function 'package' || have_function "$(get_package_func_name "$pkgname")"; }; then
 			error "$(gettext "Missing %s function in %s")" "package()" "$BUILDFILE"
 			ret=1
 		fi
@@ -47,8 +47,8 @@
 			ret=1
 		fi
 		for i in "${pkgname[@]}"; do
-			if ! have_function "package_$i"; then
-				error "$(gettext "Missing %s function for split package '%s'")" "package_$i()" "$i"
+			if ! have_function "$(get_package_func_name "$i")"; then
+				error "$(gettext "Missing %s function for split package '%s'")" "$(get_package_func_name "$i")()" "$i"
 				ret=1
 			fi
 		done
diff -ur pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/package_function_variable.sh.in pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/package_function_variable.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/package_function_variable.sh.in	2021-01-08 13:26:12.922712100 -0800
+++ pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/package_function_variable.sh.in	2021-01-09 13:03:27.882695700 -0800
@@ -42,7 +42,7 @@
 
 			for i in ${pkgbuild_schema_arrays[@]} ${pkgbuild_schema_strings[@]}; do
 				in_array "$i" ${pkgbuild_schema_package_overrides[@]} && continue
-				if exists_function_variable "package_$pkg" "${i}_${a}"; then
+				if exists_function_variable "$(get_package_func_name "$pkg")" "${i}_${a}"; then
 					error "$(gettext "%s can not be set inside a package function")" "${i}_${a}"
 					ret=1
 				fi
@@ -51,7 +51,7 @@
 
 		for i in ${pkgbuild_schema_arrays[@]} ${pkgbuild_schema_strings[@]}; do
 			in_array "$i" ${pkgbuild_schema_package_overrides[@]} && continue
-			if exists_function_variable "package_$pkg" "$i"; then
+			if exists_function_variable "$(get_package_func_name "$pkg")" "$i"; then
 				error "$(gettext "%s can not be set inside a package function")" "$i"
 				ret=1
 			fi
diff -ur pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/variable.sh.in pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/variable.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/lint_pkgbuild/variable.sh.in	2021-01-08 13:26:12.953961000 -0800
+++ pacman-5.2.2/scripts/libmakepkg/lint_pkgbuild/variable.sh.in	2021-01-09 13:04:05.843238800 -0800
@@ -68,7 +68,7 @@
 	# package function variables
 	for pkg in ${pkgname[@]}; do
 		for i in ${pkgbuild_schema_arrays[@]}; do
-			if extract_function_variable "package_$pkg" $i 0 out; then
+			if extract_function_variable "$(get_package_func_name "$pkg")" $i 0 out; then
 				error "$(gettext "%s should be an array")" "$i"
 				ret=1
 			fi
@@ -78,7 +78,7 @@
 			[[ $a == "any" ]] && continue
 
 			for i in ${pkgbuild_schema_arch_arrays[@]}; do
-				if extract_function_variable "package_$pkg" "${i}_${a}" 0 out; then
+				if extract_function_variable "$(get_package_func_name "$pkg")" "${i}_${a}" 0 out; then
 					error "$(gettext "%s should be an array")" "${i}_${a}"
 					ret=1
 				fi
@@ -86,7 +86,7 @@
 		done
 
 		for i in ${pkgbuild_schema_strings[@]}; do
-			if extract_function_variable "package_$pkg" $i 1 out; then
+			if extract_function_variable "$(get_package_func_name "$pkg")" $i 1 out; then
 				error "$(gettext "%s should not be an array")" "$i"
 				ret=1
 			fi
diff -ur pacman-5.2.2.orig/scripts/libmakepkg/util/pkgbuild.sh.in pacman-5.2.2/scripts/libmakepkg/util/pkgbuild.sh.in
--- pacman-5.2.2.orig/scripts/libmakepkg/util/pkgbuild.sh.in	2021-01-08 13:26:12.813338800 -0800
+++ pacman-5.2.2/scripts/libmakepkg/util/pkgbuild.sh.in	2021-01-09 14:10:30.931872500 -0800
@@ -28,6 +28,15 @@
 	declare -f "$1" >/dev/null
 }
 
+get_package_func_name() {
+	local pkgname=$1
+	if ! have_function "package_${pkgname}" && [ -n "${MINGW_PACKAGE_PREFIX}" ] && have_function "package_${pkgname#${MINGW_PACKAGE_PREFIX}-}"; then
+		echo "package_${pkgname#${MINGW_PACKAGE_PREFIX}-}"
+	else
+		echo "package_${pkgname}"
+	fi
+}
+
 grep_function() {
 	{ declare -f "$1" || declare -f package; } 2>/dev/null | grep -E "$2"
 }
@@ -125,7 +134,7 @@
 
 	if [[ $pkgname ]]; then
 		extract_global_variable "$attrname" "$isarray" "$outputvar"
-		extract_function_variable "package_$pkgname" "$attrname" "$isarray" "$outputvar"
+		extract_function_variable "$(get_package_func_name "$pkgname")" "$attrname" "$isarray" "$outputvar"
 	else
 		extract_global_variable "$attrname" "$isarray" "$outputvar"
 	fi
@@ -144,12 +152,12 @@
 	done
 
 	for name in "${pkgname[@]}"; do
-		if extract_function_variable "package_$name" "$attrname" 1 list; then
+		if extract_function_variable "$(get_package_func_name "$name")" "$attrname" 1 list; then
 			all_list+=("${list[@]}")
 		fi
 
 		for a in "${arch[@]}"; do
-			if extract_function_variable "package_$name" "${attrname}_$a" 1 list; then
+			if extract_function_variable "$(get_package_func_name "$name")" "${attrname}_$a" 1 list; then
 				all_list+=("${list[@]}")
 			fi
 		done
diff -ur pacman-5.2.2.orig/scripts/makepkg.sh.in pacman-5.2.2/scripts/makepkg.sh.in
--- pacman-5.2.2.orig/scripts/makepkg.sh.in	2021-01-08 13:26:13.172719400 -0800
+++ pacman-5.2.2/scripts/makepkg.sh.in	2021-01-09 14:53:30.713121700 -0800
@@ -435,7 +435,11 @@
 }
 
 run_package() {
-	run_function_safe "package${1:+_$1}"
+	local pkgfunc="package"
+	if [[ -n "$1" ]]; then
+		pkgfunc="$(get_package_func_name "$1")"
+	fi
+	run_function_safe "$pkgfunc"
 }
 
 find_libdepends() {
@@ -769,7 +773,7 @@
 
 		[[ ${!i} ]] && files+=("${!i}")
 		for pkgname in "${pkgname_backup[@]}"; do
-			if extract_function_variable "package_$pkgname" "$i" 0 file; then
+			if extract_function_variable "$(get_package_func_name "$pkgname")" "$i" 0 file; then
 				files+=("$file")
 			fi
 		done
@@ -1230,7 +1230,7 @@
 # check we have the software required to process the PKGBUILD
 check_software || exit $E_MISSING_MAKEPKG_DEPS
 
-if (( ${#pkgname[@]} > 1 )) || have_function package_${pkgname}; then
+if (( ${#pkgname[@]} > 1 )) || have_function "$(get_package_func_name "$pkgname")"; then
 	SPLITPKG=1
 fi
 
