--- pacman-5.2.2/scripts/makepkg.sh.in.orig	2021-01-26 18:02:05.720517800 -0800
+++ pacman-5.2.2/scripts/makepkg.sh.in	2021-01-26 18:57:45.117782300 -0800
@@ -357,6 +357,16 @@
 
 source_buildfile() {
 	source_safe "$@"
+	if [[ -n "${MINGW_PACKAGE_PREFIX}" ]] && ! have_function "package"; then
+		local _name
+		for _name in "${pkgname[@]}"; do
+			local _short="package_${_name#${MINGW_PACKAGE_PREFIX}-}"
+			if ! have_function "package_${_name}" && have_function "${_short}"; then
+				local _func="$(declare -f "${_short}")"
+				eval "${_func/#${_short}/package_${_name}}"
+			fi
+		done
+	fi
 }
 
 run_function_safe() {
