--- pacman-5.2.2/scripts/libmakepkg/buildenv.sh.in.orig	2021-05-15 07:20:51.021291800 +0100
+++ pacman-5.2.2/scripts/libmakepkg/buildenv.sh.in	2021-05-15 07:52:13.014867100 +0100
@@ -36,5 +36,5 @@
 	done
 
 	# ensure all necessary build variables are exported
-	export CPPFLAGS CFLAGS CXXFLAGS LDFLAGS RUSTFLAGS MAKEFLAGS CHOST
+	export CC CXX CPPFLAGS CFLAGS CXXFLAGS LDFLAGS RUSTFLAGS MAKEFLAGS CMAKE_BUILD_PARALLEL_LEVEL CHOST
 }
--- pacman-5.2.2/scripts/libmakepkg/buildenv/buildflags.sh.in.orig	2021-05-15 07:55:29.592543800 +0100
+++ pacman-5.2.2/scripts/libmakepkg/buildenv/buildflags.sh.in	2021-05-15 07:55:42.773495200 +0100
@@ -30,6 +30,6 @@
 
 buildenv_buildflags() {
 	if check_option "buildflags" "n"; then
-		unset CPPFLAGS CFLAGS DEBUG_CFLAGS CXXFLAGS DEBUG_CXXFLAGS LDFLAGS RUSTFLAGS DEBUG_RUSTFLAGS
+		unset CC CXX CPPFLAGS CFLAGS DEBUG_CFLAGS CXXFLAGS DEBUG_CXXFLAGS LDFLAGS RUSTFLAGS DEBUG_RUSTFLAGS
 	fi
 }
--- pacman-5.2.2/scripts/libmakepkg/buildenv/makeflags.sh.in.orig	2021-05-20 20:36:40.970218800 -0700
+++ pacman-5.2.2/scripts/libmakepkg/buildenv/makeflags.sh.in	2021-05-20 20:37:18.690028900 -0700
@@ -30,6 +30,6 @@
 
 buildenv_makeflags() {
 	if check_option "makeflags" "n"; then
-		unset MAKEFLAGS
+		unset MAKEFLAGS CMAKE_BUILD_PARALLEL_LEVEL
 	fi
 }
