From fb822167d46322f4d1f00093f8a1e3ff1e2380a1 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 28 Apr 2024 13:20:59 +0200
Subject: [PATCH 33/N] fixup! change shebang

---
 scripts/pacman-db-upgrade.sh.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/pacman-db-upgrade.sh.in b/scripts/pacman-db-upgrade.sh.in
index c1cb200..216dc25 100644
--- a/scripts/pacman-db-upgrade.sh.in
+++ b/scripts/pacman-db-upgrade.sh.in
@@ -18,6 +18,8 @@
 #   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
+set -e
+
 # Avoid creating world-unreadable files
 umask 022
 
