From bfe70fa6232a6ada9ff23130a9020e36d718344c Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Tue, 30 Apr 2024 07:45:40 +0200
Subject: [PATCH 36/N] fixup! Core update

Fixes #39
---
 lib/libalpm/package.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/libalpm/package.c b/lib/libalpm/package.c
index 8f592ad..4feeaf1 100644
--- a/lib/libalpm/package.c
+++ b/lib/libalpm/package.c
@@ -815,6 +815,8 @@ int SYMEXPORT alpm_pkg_is_core_package(const alpm_pkg_t *pkg)
 		strcmp(pkg->name, "mintty")              == 0 ||
 		strcmp(pkg->name, "msys2-runtime")       == 0 ||
 		strcmp(pkg->name, "msys2-runtime-devel") == 0 ||
+		strncmp(pkg->name, "msys2-runtime-",
+				strlen("msys2-runtime-"))        == 0 ||
 		strcmp(pkg->name, "pacman")              == 0 ||
 		strcmp(pkg->name, "pacman-mirrors")      == 0;
 }
