--- pacman-5.1.1/scripts/Makefile.am.orig	2018-07-30 09:17:05.342738900 +0300
+++ pacman-5.1.1/scripts/Makefile.am	2018-07-30 09:17:33.625588500 +0300
@@ -227,11 +227,11 @@
 	$(srcdir)/library/output_format.sh \
 	$(srcdir)/library/utils_fixed_path.sh
 
-repo-remove: $(srcdir)/repo-add.sh.in
+repo-remove: repo-add
 	$(AM_V_at)$(RM) repo-remove
 	$(AM_V_at)$(LN_S) repo-add repo-remove
 
-repo-elephant: $(srcdir)/repo-add.sh.in
+repo-elephant: repo-add
 	$(AM_V_at)$(RM) repo-elephant
 	$(AM_V_at)$(LN_S) repo-add repo-elephant
 
