diff -Naur pacman-contrib-1.1.0/src/Makefile.am.orig pacman-contrib-1.1.0/src/Makefile.am
--- pacman-contrib-1.1.0/src/Makefile.am.orig
+++ pacman-contrib-1.1.0/src/Makefile.am
@@ -7,6 +7,7 @@
 # paths set at make time
 conffile  = ${sysconfdir}/pacman.conf
 dbpath    = ${localstatedir}/lib/pacman/
+gpgdir    = ${sysconfdir}/pacman.d/gnupg/
 
 bin_SCRIPTS = \
 	$(OURSCRIPTS)
@@ -82,7 +83,8 @@
 AM_CPPFLAGS = \
 	-DLOCALEDIR=\"@localedir@\" \
 	-DCONFFILE=\"$(conffile)\" \
-	-DDBPATH=\"$(dbpath)\"
+	-DDBPATH=\"$(dbpath)\" \
+	-DGPGDIR=\"$(gpgdir)\"
 
 AM_CFLAGS = \
 	-pedantic \
diff -Naur pacman-contrib-1.1.0/src/pactree.c.orig pacman-contrib-1.1.0/src/pactree.c
--- pacman-contrib-1.1.0/src/pactree.c.orig
+++ pacman-contrib-1.1.0/src/pactree.c
@@ -117,6 +117,7 @@ static int unique = 0;
 static int searchsyncs = 0;
 static const char *dbpath = DBPATH;
 static const char *configfile = CONFFILE;
+static const char *gpgdir = GPGDIR;

 /* Trim whitespace and newlines from a string
  */
@@ -500,6 +501,8 @@
 		cleanup(1);
 	}
 
+	alpm_option_set_gpgdir(handle, gpgdir);
+
 	if(searchsyncs) {
 		if(register_syncs() != 0) {
 			cleanup(1);
