From 5f859ca81f00d649dffb2b6831f8683f0da1703b Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Mon, 12 Jan 2026 01:36:53 +0800
Subject: [PATCH] cscope: Fixes detecting of HAVE_SIGSETJMP in cygwin

Signed-off-by: Yonggang Luo <luoyonggang@gmail.com>
---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 323f119..076f96a 100644
--- a/configure.in
+++ b/configure.in
@@ -393,7 +393,7 @@ AC_CHECK_TYPE(sighandler_t,[],[],[
 #endif])
 dnl This test was ripped from gnuplot's configure.in:
 AC_MSG_CHECKING(for sigsetjmp)
-AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <setjmp.h>]], [[jmp_buf env; sigsetjmp(env, 1);]])],[AC_MSG_RESULT(yes)
+AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <setjmp.h>]], [[sigjmp_buf env; sigsetjmp(env, 1);]])],[AC_MSG_RESULT(yes)
    AC_DEFINE(HAVE_SIGSETJMP,1,
              [ Define if we have sigsetjmp(). ])],[AC_MSG_RESULT(no)])
 
-- 
2.52.0.windows.1

