diff -Naur icu-orig/source/tools/genrb/derb.c icu-cygwin/source/tools/genrb/derb.c
--- icu-orig/source/tools/genrb/derb.c	2013-05-24 00:02:16.000000000 +0400
+++ icu-cygwin/source/tools/genrb/derb.c	2014-02-03 10:14:41.742200000 +0400
@@ -35,9 +35,11 @@
 #include <fcntl.h>
 #define USE_FILENO_BINARY_MODE 1
 /* Windows likes to rename Unix-like functions */
+#if U_PLATFORM != U_PF_CYGWIN
 #ifndef fileno
 #define fileno _fileno
 #endif
+#endif
 #ifndef setmode
 #define setmode _setmode
 #endif
diff -Naur icu-orig/source/tools/toolutil/udbgutil.cpp icu-cygwin/source/tools/toolutil/udbgutil.cpp
--- icu-orig/source/tools/toolutil/udbgutil.cpp	2013-05-24 00:02:24.000000000 +0400
+++ icu-cygwin/source/tools/toolutil/udbgutil.cpp	2014-02-03 10:14:41.742200000 +0400
@@ -339,8 +339,10 @@
  */
 U_CAPI const char *udbg_getPlatform(void)
 {
-#if U_PLATFORM_HAS_WIN32_API
+#if U_PLATFORM_USES_ONLY_WIN32_API
     return "Windows";
+#elif U_PLATFORM == U_PF_CYGWIN
+    return "Cygwin";
 #elif U_PLATFORM == U_PF_UNKNOWN
     return "unknown";
 #elif U_PLATFORM == U_PF_DARWIN
