--- icu/source/tools/toolutil/pkg_genc.c	2014-10-03 12:11:34.000000000 -0400
+++ icu/source/tools/toolutil/pkg_genc.c	2015-02-16 13:49:08.700942400 -0500
@@ -275,7 +275,7 @@ writeAssemblyCode(const char *filename,
         uprv_strcpy(outFilePath, bufferStr);
     }
 
-#ifdef WINDOWS_WITH_GNUC
+#if defined WINDOWS_WITH_GNUC && U_PLATFORM != U_PF_CYGWIN
     /* Need to fix the file seperator character when using MinGW. */
     swapFileSepChar(outFilePath, U_FILE_SEP_CHAR, '/');
 #endif
