--- patch/lib/stddef.in.h.orig	2026-09-04 23:09:41.469628500 +0000
+++ patch/lib/stddef.in.h	2026-09-04 23:08:38.070630600 +0000
@@ -33,7 +33,7 @@
     /* Avoid warning triggered by "gcc -std=gnu23 -Wsystem-headers" \
        in Fedora 40 with gcc 14.0.1.                                \
        <https://gcc.gnu.org/bugzilla/show_bug.cgi?id=114870>.  */   \
-    && !@STDDEF_NOT_IDEMPOTENT@
+    && !0
 /* Special invocation convention inside gcc header files.  In
    particular, <stddef.h> in some ancient versions of GCC blindly
    redefined NULL when __need_wint_t was defined, even though wint_t
