--- patch/src/patch.c.orig	2026-09-04 21:05:13.144816800 +0000
+++ patch/src/patch.c	2026-09-04 21:07:29.979820200 +0000
@@ -1562,9 +1562,15 @@
 static FILE *
 create_output_file (struct outfile *out, int open_flags)
 {
+#ifdef __CYGWIN__
+  int fd = create_file (out, O_WRONLY | O_BINARY | open_flags,
+			instat.st_mode, true);
+  FILE *f = fdopen (fd, "wb");
+#else
   int fd = create_file (out, O_WRONLY | binary_transput | open_flags,
 			instat.st_mode, true);
   FILE *f = fdopen (fd, binary_transput ? "wb" : "w");
+#endif
   if (! f)
     pfatal ("Can't create file %s", quotearg (out->name));
   return f;
--- patch/src/pch.c.orig	2012-09-22 21:44:33 +0400
+++ patch/src/pch.c	2013-02-24 13:54:00 +0400
@@ -135,8 +135,10 @@
 #if HAVE_SETMODE_DOS
     if (binary_transput)
       {
+#ifndef __CYGWIN__
 	if (isatty (pfd))
 	  fatal ("cannot read binary data from tty on this platform");
+#endif
 	setmode (pfd, O_BINARY);
       }
 #endif
