--- src/patch-2.7.1/src/Makefile.in.orig	2013-07-26 18:29:38.000000000 +0400
+++ src/patch-2.7.1/src/Makefile.in	2013-07-27 08:49:08.000000000 +0400
@@ -915,6 +915,7 @@
 WERROR_CFLAGS = @WERROR_CFLAGS@
 WINDOWS_64_BIT_OFF_T = @WINDOWS_64_BIT_OFF_T@
 WINDOWS_64_BIT_ST_SIZE = @WINDOWS_64_BIT_ST_SIZE@
+WINDRES = windres
 WINT_T_SUFFIX = @WINT_T_SUFFIX@
 YACC = @YACC@
 YFLAGS = @YFLAGS@
@@ -1057,9 +1058,12 @@
 
 clean-binPROGRAMS:
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
-patch$(EXEEXT): $(patch_OBJECTS) $(patch_DEPENDENCIES) $(EXTRA_patch_DEPENDENCIES) 
+patch.res: patch.rc patch.manifest
+	$(WINDRES) --input $< --output $@ --output-format=coff
+
+patch$(EXEEXT): $(patch_OBJECTS) patch.res $(patch_DEPENDENCIES) $(EXTRA_patch_DEPENDENCIES) 
 	@rm -f patch$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(patch_OBJECTS) $(patch_LDADD) $(LIBS)
+	$(AM_V_CCLD)$(LINK) $(patch_OBJECTS) patch.res $(patch_LDADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
--- /dev/null	2013-07-12 17:17:55.297805948 +0400
+++ src/patch-2.7.1/src/patch.manifest	2013-07-27 08:49:36.000000000 +0400
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+  <assemblyIdentity version="1.0.0.0"
+     processorArchitecture="x86"
+     name="GNU.patch.patch"
+     type="win32"/>
+
+  <!-- Identify the application security requirements. -->
+  <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
+    <security>
+      <requestedPrivileges>
+        <requestedExecutionLevel level="asInvoker" uiAccess="false"/>
+      </requestedPrivileges>
+    </security>
+  </trustInfo>
+</assembly>
--- /dev/null	2013-07-12 17:17:55.297805948 +0400
+++ src/patch-2.7.1/src/patch.rc	2013-07-27 08:50:20.000000000 +0400
@@ -0,0 +1 @@
+1 24 patch.manifest
--- src/patch-2.7.1/src/Makefile.am.orig	2012-09-14 13:15:41.000000000 +0400
+++ src/patch-2.7.1/src/Makefile.am	2013-07-27 08:58:34.000000000 +0400
@@ -17,6 +17,7 @@
 ## Process this file with automake to produce Makefile.in.
 
 AM_CFLAGS = $(WARN_CFLAGS) $(WERROR_CFLAGS)
+WINDRES = windres
 
 bin_PROGRAMS = patch
 patch_SOURCES = \
@@ -34,9 +35,14 @@
 
 AM_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib
 patch_LDADD = $(LDADD) $(top_builddir)/lib/libpatch.a $(LIB_CLOCK_GETTIME) \
-	      $(LIB_XATTR)
+	      $(LIB_XATTR) patch.res
 
 if ENABLE_MERGE
   patch_SOURCES += merge.c
   AM_CPPFLAGS += -DENABLE_MERGE
 endif
+
+patch.res: patch.rc patch.manifest
+	$(WINDRES) --input $< --output $@ --output-format=coff
+
+patch_DEPENDENCIES = patch.res
