--- patch/src/patch.c.orig
+++ patch/src/patch.c
@@ -398,6 +398,6 @@
 	    if (S_ISREG (file_type) && instat.st_size != 0)
 	      {
-		int oflags = (O_RDONLY | binary_transput
+		int oflags = (O_RDONLY | (binary_transput ?: O_TEXT)
 			      | (follow_symlinks ? 0 : O_NOFOLLOW));
 		ifd = safe_open (inname, oflags, 0);
 		if (ifd < 0)
--- patch/src/pch.c.orig
+++ patch/src/pch.c
@@ -135,8 +135,10 @@
 #if HAVE_SETMODE_DOS
     if (binary_transput)
       {
+#ifndef __CYGWIN__
 	if (isatty (pfd))
 	  fatal ("cannot read binary data from tty on this platform");
+#endif
 	setmode (pfd, O_BINARY);
       }
 #endif
