From 7af7722467c3ce3a09f9e130ee038756baf54471 Mon Sep 17 00:00:00 2001
From: Ben Wijen <ben@wijen.net>
Date: Tue, 16 Dec 2025 14:50:35 +0100
Subject: [PATCH 3/3] module/ffi/mosquitto-nyacc.ffi: add include helper

---
 module/ffi/mosquitto-nyacc.ffi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/module/ffi/mosquitto-nyacc.ffi b/module/ffi/mosquitto-nyacc.ffi
index b8d5b8a..5b320bb 100644
--- a/module/ffi/mosquitto-nyacc.ffi
+++ b/module/ffi/mosquitto-nyacc.ffi
@@ -10,4 +10,10 @@
 (define-ffi-module (ffi mosquitto)
   #:pkg-config "libmosquitto"
   #:cpp-defs '("bool=char" "true=1" "false=0")
+  #:inc-help '(
+    ("stdint.h"
+      "int8_t" "uint8_t" "int16_t" "uint16_t" "int32_t" "uint32_t"
+      "int64_t" "uint64_t" "uintptr_t" "intptr_t" "intmax_t" "uintmax_t"
+      "int_least8_t" "uint_least8_t" "int_least16_t" "uint_least16_t"
+      "int_least32_t" "uint_least32_t" "int_least64_t" "uint_least64_t"))
   #:include '("mosquitto.h"))
-- 
2.49.0.windows.1

