diff -Naur a/nanorc.nanorc b/nanorc.nanorc
--- a/nanorc.nanorc	2020-10-10 16:25:18.000000000 -0300
+++ b/nanorc.nanorc	2024-10-16 13:16:04.816910360 -0300
@@ -23,7 +23,7 @@
 icolor brightmagenta " brightmagenta"
 icolor brightcyan " brightcyan"
 icolor brightwhite " brightwhite"
-icolor brightnormal " brightnormal"
+# icolor brightnormal " brightnormal"
 icolor ,black ",black "
 icolor ,red ",red "
 icolor ,green ",green "
