--- autoconf-archive-2019.01.06/m4/ax_check_gl.m4.orig	2019-01-05 18:00:00.000000000 -0500
+++ autoconf-archive-2019.01.06/m4/ax_check_gl.m4	2019-01-08 04:03:58.689522600 -0500
@@ -262,7 +262,7 @@ AC_DEFUN([AX_CHECK_GL],
          [*-darwin*],[_AX_CHECK_DARWIN_GL],
          dnl some windows may support X11 opengl, and should be able to linked
          dnl by -lGL. However I have no machine to test it.
-         [*-cygwin*|*-mingw*],[
+         [*-cygwin*|*-msys*|*-mingw*],[
           _AX_CHECK_GL_MANUAL_LIBS_GENERIC([opengl32 GL gl])
           AC_CHECK_HEADERS([windows.h])
           ],
--- autoconf-archive-2019.01.06/m4/ax_check_glu.m4.orig	2019-01-05 18:00:00.000000000 -0500
+++ autoconf-archive-2019.01.06/m4/ax_check_glu.m4	2019-01-08 04:03:58.693535900 -0500
@@ -211,7 +211,7 @@ AC_DEFUN([AX_CHECK_GLU],[
   dnl Setup GLU_CFLAGS and GLU_LIBS
   AS_CASE([${host}],
           [*-darwin*],[_AX_CHECK_DARWIN_GLU],
-          [*-cygwin*],[_AX_CHECK_GLU_SEARCH_LIBS([GLU glu MesaGLU glu32])
+          [*-cygwin*|*-msys*],[_AX_CHECK_GLU_SEARCH_LIBS([GLU glu MesaGLU glu32])
                        AC_CHECK_HEADERS([windows.h])],
           # try first native
  	  [*-mingw*],[_AX_CHECK_GLU_SEARCH_LIBS([glu32 GLU glu MesaGLU])
--- autoconf-archive-2019.01.06/m4/ax_jni_include_dir.m4.orig	2019-01-05 18:00:00.000000000 -0500
+++ autoconf-archive-2019.01.06/m4/ax_jni_include_dir.m4	2019-01-08 04:03:58.696529100 -0500
@@ -115,7 +115,7 @@ linux*)         _JNI_INC_SUBDIRS="linux
 osf*)           _JNI_INC_SUBDIRS="alpha";;
 solaris*)       _JNI_INC_SUBDIRS="solaris";;
 mingw*)		_JNI_INC_SUBDIRS="win32";;
-cygwin*)	_JNI_INC_SUBDIRS="win32";;
+cygwin*|msys*)	_JNI_INC_SUBDIRS="win32";;
 *)              _JNI_INC_SUBDIRS="genunix";;
 esac
 
--- autoconf-archive-2019.01.06/m4/ax_prog_javah.m4.orig	2019-01-05 18:00:00.000000000 -0500
+++ autoconf-archive-2019.01.06/m4/ax_prog_javah.m4	2019-01-08 04:03:58.700517300 -0500
@@ -36,7 +36,7 @@ AS_IF([test -n "$ac_cv_path_JAVAH"],
         ax_prog_javah_bin_dir=`AS_DIRNAME([$_ACJAVAH_FOLLOWED])`
         ac_dir="`AS_DIRNAME([$ax_prog_javah_bin_dir])`/include"
         AS_CASE([$build_os],
-                [cygwin*|mingw*],
+                [cygwin*|msys*|mingw*],
                 [ac_machdep=win32],
                 [ac_machdep=`AS_ECHO($build_os) | sed 's,[[-0-9]].*,,'`])
         CPPFLAGS="$ac_save_CPPFLAGS -I$ac_dir -I$ac_dir/$ac_machdep"
--- autoconf-archive-2019.01.06/m4/ax_set_default_paths_system.m4.orig	2019-01-05 18:00:00.000000000 -0500
+++ autoconf-archive-2019.01.06/m4/ax_set_default_paths_system.m4	2019-01-08 04:03:58.703509600 -0500
@@ -35,7 +35,7 @@ case "$prefix:$ac_default_prefix" in
     result=""
     AC_MSG_CHECKING(default prefix path)
     case "${target_os}" in
-      *cygwin* | *mingw* | *uwin* | *djgpp | *emx* )
+      *cygwin* | *msys* | *mingw* | *uwin* | *djgpp | *emx* )
 	if test "${host_os}" = "${target_os}" ; then
            ac_default_prefix="/programs"
            result="(win/dos target)"
