diff -up cvs-1.11.23/src/login.c.cvspass cvs-1.11.23/src/login.c
--- cvs-1.11.23/src/login.c.cvspass	2008-05-07 11:57:34.000000000 -0400
+++ cvs-1.11.23/src/login.c	2008-07-15 15:34:34.000000000 -0400
@@ -315,6 +315,15 @@ internal error: can only call password_e
     fp = CVS_FOPEN (passfile, "r");
     if (fp == NULL)
     {
+       fp = CVS_FOPEN (passfile, "w");
+       if (fp != NULL)
+       {
+           fclose (fp);
+           fp = CVS_FOPEN (passfile, "r");
+       }
+    }
+    if (fp == NULL)
+    {
 	error (0, errno, "warning: failed to open %s for reading", passfile);
 	goto process;
     }
diff -up cvs-1.11.23/src/logmsg.c.cvspass cvs-1.11.23/src/logmsg.c
--- cvs-1.11.23/src/logmsg.c.cvspass	2008-05-07 11:57:34.000000000 -0400
+++ cvs-1.11.23/src/logmsg.c	2008-07-15 15:34:34.000000000 -0400
@@ -269,7 +269,7 @@ do_editor (dir, messagep, repository, ch
     }
 
     (void) fprintf (fp,
-  "%s----------------------------------------------------------------------\n",
+  "\n%s----------------------------------------------------------------------\n",
 		    CVSEDITPREFIX);
     (void) fprintf (fp,
   "%sEnter Log.  Lines beginning with `%.*s' are removed automatically\n%s\n",
diff -up cvs-1.11.23/src/sanity.sh.cvspass cvs-1.11.23/src/sanity.sh
--- cvs-1.11.23/src/sanity.sh.cvspass	2008-05-06 21:16:00.000000000 -0400
+++ cvs-1.11.23/src/sanity.sh	2008-07-15 15:37:43.000000000 -0400
@@ -15590,6 +15590,7 @@ description:
 revision 1\.1
 date: [0-9/]* [0-9:]*;  author: ${username};  state: Exp;
 branches:  1\.1\.2;
+x
 xCVS: ----------------------------------------------------------------------
 xCVS: Enter Log.  Lines beginning with .CVS:. are removed automatically
 xCVS:
@@ -15601,6 +15602,7 @@ xCVS: ----------------------------------
 ----------------------------
 revision 1\.1\.2\.1
 date: [0-9/]* [0-9:]*;  author: ${username};  state: Exp;  lines: ${PLUS}1 -0
+x
 xCVS: ----------------------------------------------------------------------
 xCVS: Enter Log.  Lines beginning with .CVS:. are removed automatically
 xCVS:
@@ -15622,6 +15624,7 @@ description:
 revision 1\.1
 date: [0-9/]* [0-9:]*;  author: ${username};  state: Exp;
 branches:  1\.1\.2;
+x
 xCVS: ----------------------------------------------------------------------
 xCVS: Enter Log.  Lines beginning with .CVS:. are removed automatically
 xCVS:
@@ -15633,6 +15636,7 @@ xCVS: ----------------------------------
 ----------------------------
 revision 1\.1\.2\.1
 date: [0-9/]* [0-9:]*;  author: ${username};  state: Exp;  lines: ${PLUS}1 -0
+x
 xCVS: ----------------------------------------------------------------------
 xCVS: Enter Log.  Lines beginning with .CVS:. are removed automatically
 xCVS:
