--- origsrc/git-1.7.9/git-gui/Makefile	2012-01-27 13:51:04.000000000 -0700
+++ src/git-1.7.9/git-gui/Makefile	2012-02-11 12:07:50.890625000 -0700
@@ -136,7 +136,7 @@ GITGUI_SCRIPT   := $$0
 GITGUI_RELATIVE :=
 GITGUI_MACOSXAPP :=
 
-ifeq ($(uname_O),Cygwin)
+ifeq ($(uname_O),noThanks)
 	GITGUI_SCRIPT := `cygpath --windows --absolute "$(GITGUI_SCRIPT)"`
 
 	# Is this a Cygwin Tcl/Tk binary?  If so it knows how to do
