Index: git-2.10.0/git-sh-setup.sh
===================================================================
--- git-2.10.0.orig/git-sh-setup.sh	2016-09-03 03:59:09.000000000 +0200
+++ git-2.10.0/git-sh-setup.sh	2016-09-07 17:24:58.817642893 +0200
@@ -3,7 +3,7 @@
 # a few helper shell functions.
 
 # Source git-sh-i18n for gettext support.
-. git-sh-i18n
+. "$(git --exec-path)/git-sh-i18n"
 
 # Having this variable in your environment would break scripts because
 # you would cause "cd" to be taken to unexpected places.  If you
