--- git-2.35.1/git-gui/git-gui.sh.orig	2022-01-29 01:48:42.000000000 +0100
+++ git-2.35.1/git-gui/git-gui.sh	2022-03-12 18:51:13.610007600 +0100
@@ -51,7 +51,10 @@
 if { [info exists ::env(GIT_GUI_LIB_DIR) ] } {
 	set oguilib $::env(GIT_GUI_LIB_DIR)
 } else {
-	set oguilib {@@GITGUI_LIBDIR@@}
+	set oguilib [exec cygpath \
+			--windows \
+			--absolute \
+			{@@GITGUI_LIBDIR@@}]
 }
 set oguirel {@@GITGUI_RELATIVE@@}
 if {$oguirel eq {1}} {
@@ -274,10 +277,6 @@
 				set _iscygwin 0
 			} else {
 				set _iscygwin 1
-				# Handle MSys2 which is only cygwin when MSYSTEM is MSYS.
-				if {[info exists ::env(MSYSTEM)] && $::env(MSYSTEM) ne "MSYS"} {
-					set _iscygwin 0
-				}
 			}
 		} else {
 			set _iscygwin 0
