--- a/git-sh-setup.sh
+++ b/git-sh-setup.sh
@@ -315,28 +315,17 @@ case $(uname -s) in
 	find () {
 		/usr/bin/find "$@"
 	}
-	# git sees Windows-style pwd
-	pwd () {
-		builtin pwd -W
-	}
-	is_absolute_path () {
-		case "$1" in
-		[/\\]* | [A-Za-z]:*)
-			return 0 ;;
-		esac
-		return 1
-	}
 	;;
-*)
-	is_absolute_path () {
-		case "$1" in
-		/*)
-			return 0 ;;
-		esac
-		return 1
-	}
 esac
 
+is_absolute_path () {
+	case "$1" in
+	/*)
+		return 0 ;;
+	esac
+	return 1
+}
+
 # Make sure we are in a valid repository of a vintage we understand,
 # if we require to be in a git repository.
 git_dir_init () {
