--- src/git-2.3.5/git-sh-setup.sh.old	2015-04-01 00:14:52.000000000 +0200
+++ src/git-2.3.5/git-sh-setup.sh	2015-04-18 21:47:15.753865900 +0200
@@ -306,28 +305,17 @@ case $(uname -s) in
 	find () {
 		/usr/bin/find "$@"
 	}
-	# git sees Windows-style pwd
-	pwd () {
-		builtin pwd -W
-	}
-	is_absolute_path () {
-		case "$1" in
-		[/\\]* | [A-Za-z]:*)
-			return 0 ;;
-		esac
-		return 1
-	}
 	;;
-*)
-	is_absolute_path () {
-		case "$1" in
-		/*)
-			return 0 ;;
-		esac
-		return 1
-	}
 esac

+is_absolute_path () {
+	case "$1" in
+	/*)
+		return 0 ;;
+	esac
+	return 1
+}
+
 # Make sure we are in a valid repository of a vintage we understand,
 # if we require to be in a git repository.
 git_dir_init () {
