diff --color -uprN mtrnew/Makefile.am mtr-0.94/Makefile.am
--- mtrnew/Makefile.am	2020-07-10 12:31:11.000000000 +0800
+++ mtr-0.94/Makefile.am	2021-06-02 16:54:10.365561700 +0800
@@ -123,7 +123,7 @@ if CYGWIN
 mtr_packet_SOURCES += \
 	packet/probe_cygwin.c packet/probe_cygwin.h \
 	packet/wait_cygwin.c
-mtr_packet_LDADD += -lcygwin -liphlpapi -lws2_32
+mtr_packet_LDADD += -lmsys-2.0 -liphlpapi -lws2_32
 
 dist_windows_aux = \
 	$(srcdir)/build-aux/mtr.bat \
diff --color -uprN mtrnew/configure mtr-0.94/configure
--- mtrnew/configure	2020-09-24 18:32:12.000000000 +0800
+++ mtr-0.94/configure	2021-06-02 16:53:45.412294600 +0800
@@ -5455,7 +5455,7 @@ $as_echo "no" >&6; }
 	fi
 fi
 
- if test "$host_os" = cygwin; then
+ if test "$host_os" = msys; then
   CYGWIN_TRUE=
   CYGWIN_FALSE='#'
 else
diff --color -uprN mtrnew/packet/probe_cygwin.h mtr-0.94/packet/probe_cygwin.h
--- mtrnew/packet/probe_cygwin.h	2019-03-11 02:50:36.000000000 +0800
+++ mtr-0.94/packet/probe_cygwin.h	2021-06-02 16:53:13.258420100 +0800
@@ -19,6 +19,8 @@
 #ifndef PROBE_CYGWIN_H
 #define PROBE_CYGWIN_H
 
+typedef struct in6_addr IN6_ADDR, *PIN6_ADDR, *LPIN6_ADDR; 
+
 #include <arpa/inet.h>
 #include <windows.h>
 #include <iphlpapi.h>
