diff --git a/configure.ac b/configure.ac
index 830f1bf..a009590 100644
--- a/configure.ac
+++ b/configure.ac
@@ -79,7 +79,7 @@ dnl should only exist on windows, I hope
 AC_CHECK_STDCALL_FUNC([OpenClipboard],[void *])
 AC_CHECK_DECLS([cygwin_conv_path], [],[
 case "$host" in
-*cygwin* ) AC_MSG_ERROR([At least cygwin-1.7 is required]) ;;
+*cygwin* | *msys* ) AC_MSG_ERROR([At least cygwin-1.7 is required]) ;;
 esac],dnl
   [[#include <sys/cygwin.h>]])
 
@@ -88,7 +88,7 @@ AM_CONDITIONAL(WITH_WINDOWS_PROGRAMS, test "$ac_cv_func_OpenClipboard" = yes)
 AM_CONDITIONAL(WITH_NATIVE_IPC_PROGRAMS, test "$HAVE_INTRINSIC_IPC" = yes)
 host_is_cygwin=no
 case "$host" in
-*cygwin* ) host_is_cygwin=yes ;;
+*cygwin* | *msys* ) host_is_cygwin=yes ;;
 esac
 AM_CONDITIONAL(WITH_CYGWIN_SPECIFIC_PROGRAMS, test "$host_is_cygwin" = yes)
 
