--- nettle-3.7.3/configure.ac.orig	2021-06-06 22:06:29.000000000 +0200
+++ nettle-3.7.3/configure.ac	2022-04-18 21:14:30.819409900 +0200
@@ -666,8 +666,8 @@
         LIBHOGWEED_FORLINK='libhogweed-$(LIBHOGWEED_MAJOR).dll'
         ;;
       cygwin*)
-        LIBNETTLE_FORLINK='cygnettle-$(LIBNETTLE_MAJOR).dll'
-        LIBHOGWEED_FORLINK='cyghogweed-$(LIBHOGWEED_MAJOR).dll'
+        LIBNETTLE_FORLINK='msys-nettle-$(LIBNETTLE_MAJOR).dll'
+        LIBHOGWEED_FORLINK='msys-hogweed-$(LIBHOGWEED_MAJOR).dll'
         ;;
     esac
     if test "x$cross_compiling" = xyes ; then
diff -aur nettle-3.2/testsuite/dlopen-test.c.orig nettle-3.2/testsuite/dlopen-test.c > ../3.2-dlopen-test-msys-test.patch
--- nettle-3.2/testsuite/dlopen-test.c.orig	2016-05-17 07:17:36.777183000 -0400
+++ nettle-3.2/testsuite/dlopen-test.c	2016-05-17 07:18:23.983817400 -0400
@@ -9,7 +9,11 @@
 main (int argc UNUSED, char **argv UNUSED)
 {
 #if HAVE_LIBDL
+  #ifdef __MSYS__
+  void *handle = dlopen ("../msys-nettle-8.dll", RTLD_NOW);
+  #else
   void *handle = dlopen ("../libnettle." SO_EXT, RTLD_NOW);
+  #endif
   int (*get_version)(void);
   if (!handle)
     {
