diff -Naur nettle-3.3-orig/aclocal.m4 nettle-3.3/aclocal.m4
--- nettle-3.3-orig/aclocal.m4	2013-01-14 21:44:12.000000000 +0000
+++ nettle-3.3/aclocal.m4	2013-04-24 18:30:00.456054700 +0000
@@ -35,6 +35,7 @@
 	# Could also use -fpic, depending on the number of symbol references
 	solaris*)	CCPIC="-fPIC" ;;
 	cygwin*)	CCPIC="" ;;
+	msys*)	CCPIC="" ;;
 	mingw32*)	CCPIC="" ;;
 	*)		CCPIC="-fpic" ;;
       esac
diff -Naur nettle-3.3-orig/configure.ac nettle-3.3/configure.ac
--- nettle-3.3-orig/configure.ac	2013-01-14 21:44:12.000000000 +0000
+++ nettle-3.3/configure.ac	2013-04-24 18:33:10.252929700 +0000
@@ -569,7 +569,7 @@
 W64_ABI=no
 
 case "$host_os" in
-  mingw32*|cygwin*)
+  mingw32*|cygwin*|msys*)
     # The actual DLLs, e.g. libnettle-$major-$minor.dll, are normally
     # installed into the bin dir (or more exactly $libdir/../bin, for
     # automake), while libnettle.dll.a, which is a stub file for
@@ -583,6 +583,10 @@
         LIBNETTLE_FORLINK='cygnettle-$(LIBNETTLE_MAJOR).dll'
         LIBHOGWEED_FORLINK='cyghogweed-$(LIBHOGWEED_MAJOR).dll'
         ;;
+      msys*)
+        LIBNETTLE_FORLINK='msys-nettle-$(LIBNETTLE_MAJOR).dll'
+        LIBHOGWEED_FORLINK='msys-hogweed-$(LIBHOGWEED_MAJOR).dll'
+        ;;
     esac
     if test "x$cross_compiling" = xyes ; then
 	case "$ABI" in
