diff -Naur nettle-2.6-orig/aclocal.m4 nettle-2.6/aclocal.m4
--- nettle-2.6-orig/aclocal.m4	2013-01-14 21:44:12.000000000 +0000
+++ nettle-2.6/aclocal.m4	2013-04-24 18:30:00.456054700 +0000
@@ -35,6 +35,7 @@
 	# Could also use -fpic, depending on the number of symbol references
 	solaris*)	CCPIC="-fPIC" ;;
 	cygwin*)	CCPIC="" ;;
+	msys*)	CCPIC="" ;;
 	mingw32*)	CCPIC="" ;;
 	*)		CCPIC="-fpic" ;;
       esac
diff -Naur nettle-2.6-orig/config.guess nettle-2.6/config.guess
--- nettle-2.6-orig/config.guess	2013-01-14 21:44:12.000000000 +0000
+++ nettle-2.6/config.guess	2013-04-24 18:31:26.346679700 +0000
@@ -803,6 +803,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
@@ -843,6 +846,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur nettle-2.6-orig/config.sub nettle-2.6/config.sub
--- nettle-2.6-orig/config.sub	2013-01-14 21:44:12.000000000 +0000
+++ nettle-2.6/config.sub	2013-04-24 18:31:48.502929700 +0000
@@ -1340,7 +1340,7 @@
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
-	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
+	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-android* \
 	      | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
diff -Naur nettle-2.6-orig/configure nettle-2.6/configure
--- nettle-2.6-orig/configure	2013-01-14 21:44:12.000000000 +0000
+++ nettle-2.6/configure	2013-04-24 18:33:55.096679700 +0000
@@ -4627,6 +4627,7 @@
 	# Could also use -fpic, depending on the number of symbol references
 	solaris*)	CCPIC="-fPIC" ;;
 	cygwin*)	CCPIC="" ;;
+	msys*)		CCPIC="" ;;
 	mingw32*)	CCPIC="" ;;
 	*)		CCPIC="-fpic" ;;
       esac
@@ -4680,7 +4681,7 @@
 W64_ABI=no
 
 case "$host_os" in
-  mingw32*|cygwin*)
+  mingw32*|cygwin*|msys*)
     # The actual DLLs, e.g. libnettle-$major-$minor.dll, are normally
     # installed into the bin dir (or more exactly $libdir/../bin, for
     # automake), while libnettle.dll.a, which is a stub file for
@@ -4694,6 +4695,10 @@
         LIBNETTLE_FORLINK='cygnettle-$(LIBNETTLE_MAJOR).dll'
         LIBHOGWEED_FORLINK='cyghogweed-$(LIBHOGWEED_MAJOR).dll'
         ;;
+      msys*)
+        LIBNETTLE_FORLINK='msys-nettle-$(LIBNETTLE_MAJOR).dll'
+        LIBHOGWEED_FORLINK='msys-hogweed-$(LIBHOGWEED_MAJOR).dll'
+        ;;
     esac
     if test "x$cross_compiling" = xyes ; then
 	case "$ABI" in
@@ -4713,13 +4713,13 @@
     if test "x$ABI" = x64 ; then
         W64_ABI=yes
     fi
-    LIBNETTLE_SONAME=''
+    LIBNETTLE_SONAME='msys-nettle-$(LIBNETTLE_MAJOR).dll'
     LIBNETTLE_FILE='libnettle.dll.a'
     LIBNETTLE_FILE_SRC='$(LIBNETTLE_FILE)'
     LIBNETTLE_LINK='$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,--out-implib=$(LIBNETTLE_FILE) -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--whole-archive'
     LIBNETTLE_LIBS='-Wl,--no-whole-archive $(LIBS)'
 
-    LIBHOGWEED_SONAME=''
+    LIBHOGWEED_SONAME='msys-hogweed-$(LIBHOGWEED_MAJOR).dll'
     LIBHOGWEED_FILE='libhogweed.dll.a'
     LIBHOGWEED_FILE_SRC='$(LIBHOGWEED_FILE)'
     LIBHOGWEED_LINK='$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,--out-implib=$(LIBHOGWEED_FILE) -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--whole-archive'    
diff -Naur nettle-2.6-orig/configure.ac nettle-2.6/configure.ac
--- nettle-2.6-orig/configure.ac	2013-01-14 21:44:12.000000000 +0000
+++ nettle-2.6/configure.ac	2013-04-24 18:33:10.252929700 +0000
@@ -271,7 +271,7 @@
 W64_ABI=no
 
 case "$host_os" in
-  mingw32*|cygwin*)
+  mingw32*|cygwin*|msys*)
     # The actual DLLs, e.g. libnettle-$major-$minor.dll, are normally
     # installed into the bin dir (or more exactly $libdir/../bin, for
     # automake), while libnettle.dll.a, which is a stub file for
@@ -285,6 +285,10 @@
         LIBNETTLE_FORLINK='cygnettle-$(LIBNETTLE_MAJOR)-$(LIBNETTLE_MINOR).dll'
         LIBHOGWEED_FORLINK='cyghogweed-$(LIBHOGWEED_MAJOR)-$(LIBHOGWEED_MINOR).dll'
         ;;
+      msys*)
+        LIBNETTLE_FORLINK='msys-nettle-$(LIBNETTLE_MAJOR).dll'
+        LIBHOGWEED_FORLINK='msys-hogweed-$(LIBHOGWEED_MAJOR).dll'
+        ;;
     esac
     if test "x$cross_compiling" = xyes ; then
 	case "$ABI" in
