diff -Naur Alien-Build-1.76-orig/corpus/cmake-libpalindrome/libpalindrome/CMakeLists.txt Alien-Build-1.76/corpus/cmake-libpalindrome/libpalindrome/CMakeLists.txt
--- Alien-Build-1.76-orig/corpus/cmake-libpalindrome/libpalindrome/CMakeLists.txt	2019-06-23 18:01:22.000000000 +0300
+++ Alien-Build-1.76/corpus/cmake-libpalindrome/libpalindrome/CMakeLists.txt	2019-07-01 23:54:09.202424300 +0300
@@ -3,12 +3,12 @@
 target_include_directories (palindrome PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories (palindromeStatic PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
-IF (WIN32 OR CYGWIN OR MINGW)
+IF (WIN32 OR CYGWIN OR MSYS OR MINGW)
 ELSE()
   set_target_properties(palindromeStatic PROPERTIES OUTPUT_NAME palindrome)
 ENDIF()
 
-IF (WIN32 OR CYGWIN OR MINGW)
+IF (WIN32 OR CYGWIN OR MSYS OR MINGW)
   install(TARGETS palindrome RUNTIME DESTINATION bin)
 ELSE()
   install(TARGETS palindrome LIBRARY DESTINATION lib)
diff -Naur Alien-Build-1.76-orig/t/alien_build_plugin_build_cmake.t Alien-Build-1.76/t/alien_build_plugin_build_cmake.t
--- Alien-Build-1.76-orig/t/alien_build_plugin_build_cmake.t	2019-06-23 18:01:22.000000000 +0300
+++ Alien-Build-1.76/t/alien_build_plugin_build_cmake.t	2019-07-01 23:55:59.605739000 +0300
@@ -44,7 +44,7 @@
           }
           else
           {
-            my $lib = $^O =~ /^(cygwin|MSWin32)$/ ? '-lpalindromeStatic' : '-lpalindrome';
+            my $lib = $^O =~ /^(cygwin|msys|MSWin32)$/ ? '-lpalindromeStatic' : '-lpalindrome';
             $build->runtime_prop->{$_} = "-L$prefix/lib $lib" for qw( libs libs_static );
           }
         };
