diff -durN cloog-0.18.4.orig/autoconf/config.guess cloog-0.18.4/autoconf/config.guess
--- cloog-0.18.4.orig/autoconf/config.guess	2014-10-02 22:14:47.000000000 +0800
+++ cloog-0.18.4/autoconf/config.guess	2017-05-09 15:47:33.312500000 +0800
@@ -866,6 +866,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -durN cloog-0.18.4.orig/osl/autoconf/config.guess cloog-0.18.4/osl/autoconf/config.guess
--- cloog-0.18.4.orig/osl/autoconf/config.guess	2014-10-02 22:14:44.000000000 +0800
+++ cloog-0.18.4/osl/autoconf/config.guess	2017-05-09 15:48:21.156250000 +0800
@@ -866,6 +866,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
