diff -durN cloog-0.18.4.orig/isl/configure.ac cloog-0.18.4/isl/configure.ac
--- cloog-0.18.4.orig/isl/configure.ac	2017-05-09 15:24:43.125000000 +0800
+++ cloog-0.18.4/isl/configure.ac	2017-05-09 15:55:57.921875000 +0800
@@ -222,9 +222,6 @@
 AC_CONFIG_HEADERS(isl_config.h)
 AC_CONFIG_FILES(Makefile)
 AC_CONFIG_FILES(doc/Makefile)
-if test $with_clang = system; then
-	AC_CONFIG_FILES(interface/Makefile)
-fi
 AC_CONFIG_FILES([bound_test.sh], [chmod +x bound_test.sh])
 AC_CONFIG_FILES([codegen_test.sh], [chmod +x codegen_test.sh])
 AC_CONFIG_FILES([pip_test.sh], [chmod +x pip_test.sh])
diff -durN cloog-0.18.4.orig/isl/Makefile.am cloog-0.18.4/isl/Makefile.am
--- cloog-0.18.4.orig/isl/Makefile.am	2017-05-09 15:24:41.765625000 +0800
+++ cloog-0.18.4/isl/Makefile.am	2017-05-09 15:55:12.468750000 +0800
@@ -172,7 +172,7 @@
 	isl_vertices.c \
 	isl_yaml.h
 libisl_la_LIBADD = @MP_LIBS@
-libisl_la_LDFLAGS = -version-info @versioninfo@ \
+libisl_la_LDFLAGS = -version-info @versioninfo@ -no-undefined \
 	@MP_LDFLAGS@
 
 isl_test_LDFLAGS = @MP_LDFLAGS@
diff -durN cloog-0.18.4.orig/Makefile.am cloog-0.18.4/Makefile.am
--- cloog-0.18.4.orig/Makefile.am	2017-05-09 15:24:36.093750000 +0800
+++ cloog-0.18.4/Makefile.am	2017-05-09 15:54:16.468750000 +0800
@@ -92,7 +92,7 @@
 AM_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include
 AM_CFLAGS = $(CFLAGS_WARN)
 libcloog_isl_la_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include @ISL_CPPFLAGS@ @OSL_CPPFLAGS@
-libcloog_isl_la_LDFLAGS = -version-info @versioninfo@ \
+libcloog_isl_la_LDFLAGS = -version-info @versioninfo@ -no-undefined \
 	-rpath $(libdir) @ISL_LDFLAGS@ @OSL_LDFLAGS@
 libcloog_isl_la_LIBADD = @ISL_LIBS@ @OSL_LIBS@ $(ISL_LA) $(OSL_LA)
 libcloog_isl_la_SOURCES = \
