--- cloog-0.18.3/Makefile.am.orig	2013-10-16 21:08:32.616800000 +0400
+++ cloog-0.18.3/Makefile.am	2013-10-11 11:27:03.000000000 +0400
@@ -90,7 +90,7 @@
 AM_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include
 AM_CFLAGS = $(CFLAGS_WARN)
 libcloog_isl_la_CPPFLAGS = -I. -I$(top_builddir)/include -I$(top_srcdir)/include @ISL_CPPFLAGS@ @OSL_CPPFLAGS@
-libcloog_isl_la_LDFLAGS = -version-info @versioninfo@ \
+libcloog_isl_la_LDFLAGS = -version-info @versioninfo@ -no-undefined \
 	-rpath $(libdir) @ISL_LDFLAGS@ @OSL_LDFLAGS@
 libcloog_isl_la_LIBADD = @ISL_LIBS@ @OSL_LIBS@ $(ISL_LA) $(OSL_LA)
 libcloog_isl_la_SOURCES = \
--- cloog-0.18.3/isl/Makefile.am.orig	2013-10-16 21:09:33.160400000 +0400
+++ cloog-0.18.3/isl/Makefile.am	2013-10-11 11:27:21.000000000 +0400
@@ -154,7 +154,7 @@
 	isl_vertices_private.h \
 	isl_vertices.c
 libisl_la_LIBADD = @MP_LIBS@
-libisl_la_LDFLAGS = -version-info @versioninfo@ \
+libisl_la_LDFLAGS = -version-info @versioninfo@ -no-undefined \
 	@MP_LDFLAGS@
 
 isl_test_LDFLAGS = @MP_LDFLAGS@
--- cloog-0.18.3/isl/configure.ac.orig	2013-10-16 21:09:33.160400000 +0400
+++ cloog-0.18.3/isl/configure.ac	2013-10-11 11:27:21.000000000 +0400
@@ -222,9 +222,6 @@
 AC_CONFIG_HEADERS(isl_config.h)
 AC_CONFIG_FILES(Makefile)
 AC_CONFIG_FILES(doc/Makefile)
-if test $with_clang = system; then
-	AC_CONFIG_FILES(interface/Makefile)
-fi
 AC_CONFIG_FILES([bound_test.sh], [chmod +x bound_test.sh])
 AC_CONFIG_FILES([codegen_test.sh], [chmod +x codegen_test.sh])
 AC_CONFIG_FILES([pip_test.sh], [chmod +x pip_test.sh])
