--- a/Makefile.am.orig	2018-07-15 16:11:53.971108000 +0200
+++ a/Makefile.am	2018-07-15 17:20:52.885833600 +0200
@@ -181,19 +181,6 @@
 	@echo "              *-----------------------------------------------*/"
 	doxygen ./autoconf/Doxyfile
 
-#/*****************************************************************************
-# *                                   Doc                                     *
-# *****************************************************************************/
-
-if HAVE_TEXI2DVI
-pdf_DATA = doc/cloog.pdf
-dist_pdf_DATA = doc/cloog.pdf
-doc/cloog.pdf: doc/cloog.texi doc/gitversion.texi
-	$(TEXI2DVI) -I $(top_builddir)/doc --pdf $< -o $@
-endif
-
-doc/gitversion.texi: @GIT_INDEX@
-	echo '@set VERSION '`$(top_builddir)/genversion.sh`'' > $@
 
 #/*****************************************************************************
 # *                                   Tests                                    *
