From 71c1392f4562f7abc4f15ac6acc124e0d2d09de1 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Tue, 1 Dec 2015 00:28:32 -0800
Subject: [PATCH] crt: fix _MINGW_UDERSCORE_PREFIX for arm


diff --git a/mingw-w64-headers/crt/_mingw_mac.h b/mingw-w64-headers/crt/_mingw_mac.h
index 65d0e70..843aebd 100644
--- a/mingw-w64-headers/crt/_mingw_mac.h
+++ b/mingw-w64-headers/crt/_mingw_mac.h
@@ -36,7 +36,49 @@
 #define __MINGW32_MAJOR_VERSION 3
 #define __MINGW32_MINOR_VERSION 11
 
-#ifdef _WIN64
+/* Set VC specific compiler target macros.  */
+#if defined(__x86_64) && defined(_X86_)
+#  undef _X86_  /* _X86_ is not for __x86_64 */
+#endif
+
+#if defined(_X86_) && !defined(_M_IX86) && !defined(_M_IA64) \
+   && !defined(_M_AMD64) && !defined(__x86_64)
+#  if defined(__i486__)
+#    define _M_IX86 400
+#  elif defined(__i586__)
+#    define _M_IX86 500
+#  else
+     /* This gives wrong (600 instead of 300) value if -march=i386 is specified
+      but we cannot check for__i386__ as it is defined for all 32-bit CPUs. */
+#    define _M_IX86 600
+#  endif
+#endif /* if defined(_X86_) && !defined(_M_IX86) && !defined(_M_IA64) ... */
+
+#if defined(__x86_64) && !defined(_M_IX86) && !defined(_M_IA64) \
+   && !defined(_M_AMD64)
+#  define _M_AMD64 100
+#  define _M_X64 100
+#endif
+
+#if defined(__ia64__) && !defined(_M_IX86) && !defined(_M_IA64) \
+   && !defined(_M_AMD64) && !defined(_X86_) && !defined(__x86_64)
+#  define _M_IA64 100
+#endif
+
+#if defined(__arm__) && !defined(_M_ARM) && !defined(_M_ARMT) \
+   && !defined(_M_THUMB)
+#  define _M_ARM 100
+#  define _M_ARMT 100
+#  define _M_THUMB 100
+#  ifndef _ARM_
+#    define _ARM_ 1
+#  endif
+#  ifndef _M_ARM_NT
+#    define _M_ARM_NT 1
+#  endif
+#endif
+
+#ifndef _X86_
    /* MS does not prefix symbols by underscores for 64-bit.  */
 #  ifndef __MINGW_USE_UNDERSCORE_PREFIX
      /* As we have to support older gcc version, which are using underscores
@@ -57,11 +99,11 @@
 #      define __MINGW_USE_UNDERSCORE_PREFIX 0
 #    endif /* __USER_LABEL_PREFIX__ */
 #  endif
-#else /* ! ifdef _WIN64 */
-   /* For 32-bits we have always to prefix by underscore.  */
+#else /* ! ifndef _X86_ */
+   /* For x86 we have always to prefix by underscore.  */
 #  undef __MINGW_USE_UNDERSCORE_PREFIX
 #  define __MINGW_USE_UNDERSCORE_PREFIX 1
-#endif /* ifdef _WIN64 */
+#endif /* ifndef _X86_ */
 
 #if __MINGW_USE_UNDERSCORE_PREFIX == 0
 #  define __MINGW_IMP_SYMBOL(sym) __imp_##sym
@@ -75,42 +117,6 @@
 #  define __MINGW_LSYMBOL(sym) sym
 #endif /* if __MINGW_USE_UNDERSCORE_PREFIX == 0 */
 
-/* Set VC specific compiler target macros.  */
-#if defined(__x86_64) && defined(_X86_)
-#  undef _X86_	/* _X86_ is not for __x86_64 */
-#endif
-
-#if defined(_X86_) && !defined(_M_IX86) && !defined(_M_IA64) \
-   && !defined(_M_AMD64) && !defined(__x86_64)
-#  if defined(__i486__)
-#    define _M_IX86 400
-#  elif defined(__i586__)
-#    define _M_IX86 500
-#  else
-     /* This gives wrong (600 instead of 300) value if -march=i386 is specified
-      but we cannot check for__i386__ as it is defined for all 32-bit CPUs. */
-#    define _M_IX86 600
-#  endif
-#endif /* if defined(_X86_) && !defined(_M_IX86) && !defined(_M_IA64) ... */
-
-#if defined(__x86_64) && !defined(_M_IX86) && !defined(_M_IA64) \
-   && !defined(_M_AMD64)
-#  define _M_AMD64 100
-#  define _M_X64 100
-#endif
-
-#if defined(__ia64__) && !defined(_M_IX86) && !defined(_M_IA64) \
-   && !defined(_M_AMD64) && !defined(_X86_) && !defined(__x86_64)
-#  define _M_IA64 100
-#endif
-
-#if defined(__arm__) && !defined(_M_ARM)
-#  define _M_ARM 100
-#  ifndef _ARM_
-#    define _ARM_ 1
-#  endif
-#endif
-
 #ifndef __PTRDIFF_TYPE__
 #  ifdef _WIN64
 #    define __PTRDIFF_TYPE__ long long int
-- 
2.6.3

