diff --git a/configure.ac b/configure.ac
index 505ebea..ab6c41b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -36,6 +36,7 @@ AC_TYPE_UINT8_T
 AC_CHECK_FUNCS([memset])
 
 dnl AM_PROG_AR 
+AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 dnl LT_INIT
 
@@ -79,6 +80,7 @@ elif test $enable_native = "yes"; then
 fi
 
 case $host_os in
+	*msys*) LDFLAGS="${LDFLAGS} -no-undefined" ;;
 	*mingw*) LDFLAGS="${LDFLAGS} -no-undefined" ;;
 esac
 
 