diff -up libgcrypt-1.7.3/tests/benchmark.c.eccfix libgcrypt-1.7.3/tests/benchmark.c
--- libgcrypt-1.7.3/tests/benchmark.c.eccfix	2016-07-14 11:19:17.000000000 +0200
+++ libgcrypt-1.7.3/tests/benchmark.c	2016-11-22 16:21:00.109004197 +0100
@@ -1412,7 +1412,7 @@ ecc_bench (int iterations, int print_hea
 {
 #if USE_ECC
   gpg_error_t err;
-  const char *p_sizes[] = { "192", "224", "256", "384", "521", "Ed25519",
+  const char *p_sizes[] = { "224", "256", "384", "521", "Ed25519",
               "gost256", "gost512" };
   int testno;
 
diff -up libgcrypt-1.7.3/tests/dsa-rfc6979.c.eccfix libgcrypt-1.7.3/tests/dsa-rfc6979.c
--- libgcrypt-1.7.3/tests/dsa-rfc6979.c.eccfix	2016-02-18 09:38:03.000000000 +0100
+++ libgcrypt-1.7.3/tests/dsa-rfc6979.c	2016-11-22 16:22:11.804674008 +0100
@@ -210,16 +210,6 @@ check_dsa_rfc6979 (void)
       " ))"
     },
     {
-      "ECDSA, 192 bits (prime field)",
-      "(private-key"
-      " (ecdsa"
-      " (curve \"NIST P-192\")"
-      " (q #04AC2C77F529F91689FEA0EA5EFEC7F210D8EEA0B9E047ED56"
-      "       3BC723E57670BD4887EBC732C523063D0A7C957BC97C1C43#)"
-      " (d #6FAB034934E4C0FC9AE67F5B5659A9D7D1FEFD187EE09FD4#)"
-      " ))"
-    },
-    {
       "ECDSA, 224 bits (prime field)",
       "(private-key"
       " (ecdsa"
@@ -443,89 +433,6 @@ check_dsa_rfc6979 (void)
       "C9F0BDABCC0D880BB137A994CC7F3980CE91CC10FAF529FC46565B15CEA854E1"
     },
     {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-1, message = \"sample\"",
-      "sha1", "sample",
-      "37D7CA00D2C7B0E5E412AC03BD44BA837FDD5B28CD3B0021",
-      "98C6BD12B23EAF5E2A2045132086BE3EB8EBD62ABF6698FF",
-      "57A22B07DEA9530F8DE9471B1DC6624472E8E2844BC25B64"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-224, message = \"sample\"",
-      "sha224", "sample",
-      "4381526B3FC1E7128F202E194505592F01D5FF4C5AF015D8",
-      "A1F00DAD97AEEC91C95585F36200C65F3C01812AA60378F5",
-      "E07EC1304C7C6C9DEBBE980B9692668F81D4DE7922A0F97A"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-256, message = \"sample\"",
-      "sha256", "sample",
-      "32B1B6D7D42A05CB449065727A84804FB1A3E34D8F261496",
-      "4B0B8CE98A92866A2820E20AA6B75B56382E0F9BFD5ECB55",
-      "CCDB006926EA9565CBADC840829D8C384E06DE1F1E381B85"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-384, message = \"sample\"",
-      "sha384", "sample",
-      "4730005C4FCB01834C063A7B6760096DBE284B8252EF4311",
-      "DA63BF0B9ABCF948FBB1E9167F136145F7A20426DCC287D5",
-      "C3AA2C960972BD7A2003A57E1C4C77F0578F8AE95E31EC5E"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-512, message = \"sample\"",
-      "sha512", "sample",
-      "A2AC7AB055E4F20692D49209544C203A7D1F2C0BFBC75DB1",
-      "4D60C5AB1996BD848343B31C00850205E2EA6922DAC2E4B8",
-      "3F6E837448F027A1BF4B34E796E32A811CBB4050908D8F67"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-1, message = \"test\"",
-      "sha1", "test",
-      "D9CF9C3D3297D3260773A1DA7418DB5537AB8DD93DE7FA25",
-      "0F2141A0EBBC44D2E1AF90A50EBCFCE5E197B3B7D4DE036D",
-      "EB18BC9E1F3D7387500CB99CF5F7C157070A8961E38700B7"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-224, message = \"test\"",
-      "sha224", "test",
-      "F5DC805F76EF851800700CCE82E7B98D8911B7D510059FBE",
-      "6945A1C1D1B2206B8145548F633BB61CEF04891BAF26ED34",
-      "B7FB7FDFC339C0B9BD61A9F5A8EAF9BE58FC5CBA2CB15293"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-256, message = \"test\"",
-      "sha256", "test",
-      "5C4CE89CF56D9E7C77C8585339B006B97B5F0680B4306C6C",
-      "3A718BD8B4926C3B52EE6BBE67EF79B18CB6EB62B1AD97AE",
-      "5662E6848A4A19B1F1AE2F72ACD4B8BBE50F1EAC65D9124F"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-384, message = \"test\"",
-      "sha384", "test",
-      "5AFEFB5D3393261B828DB6C91FBC68C230727B030C975693",
-      "B234B60B4DB75A733E19280A7A6034BD6B1EE88AF5332367",
-      "7994090B2D59BB782BE57E74A44C9A1C700413F8ABEFE77A"
-    },
-    {
-      "ECDSA, 192 bits (prime field)",
-      "With SHA-512, message = \"test\"",
-      "sha512", "test",
-      "0758753A5254759C7CFBAD2E2D9B0792EEE44136C9480527",
-      "FE4F4AE86A58B6507946715934FE2D8FF9D95B6B098FE739",
-      "74CF5605C98FBA0E1EF34D4B5A1577A7DCF59457CAE52290"
-    },
-
-
-
-    {
       "ECDSA, 224 bits (prime field)",
       "With SHA-1, message = \"sample\"",
       "sha1", "sample",
