diff -up libgcrypt-1.7.3/cipher/bufhelp.h.aliasing libgcrypt-1.7.3/cipher/bufhelp.h
--- libgcrypt-1.7.3/cipher/bufhelp.h.aliasing	2016-04-07 17:30:08.000000000 +0200
+++ libgcrypt-1.7.3/cipher/bufhelp.h	2016-11-22 17:00:13.065692916 +0100
@@ -35,6 +35,11 @@
 # define BUFHELP_FAST_UNALIGNED_ACCESS 1
 #endif
 
+#if _GCRY_GCC_VERSION >= 40400
+# pragma GCC push_options
+# pragma GCC optimize ("no-strict-aliasing")
+#endif
+
 
 #ifdef BUFHELP_FAST_UNALIGNED_ACCESS
 /* Define type with one-byte alignment on architectures with fast unaligned
@@ -429,4 +434,8 @@ static inline void buf_put_le64(void *_b
 
 #endif /*BUFHELP_FAST_UNALIGNED_ACCESS*/
 
+#if _GCRY_GCC_VERSION >= 40400
+# pragma GCC pop_options
+#endif
+
 #endif /*GCRYPT_BUFHELP_H*/
