Index: src/tftp-hpa-5.2/README.security
===================================================================
--- src.orig/tftp-hpa-5.2/README.security
+++ src/tftp-hpa-5.2/README.security
@@ -22,7 +22,7 @@ probably the following:
 	       =======================================
 
 Starting in version 0.17, tftp-hpa operates in genuine "wait" mode,
-which means that an in.tftpd process hangs around for some time after
+which means that a tftpd process hangs around for some time after
 the last service request has arrived.  This speeds up servicing a
 subsequent request, which apparently has been a problem in the past,
 resulting in "request storms" as the client keeps retrying, resulting
@@ -36,12 +36,12 @@ directly.  Thus, if your /etc/inetd.conf
 line):
 
 tftp	dgram	udp	wait	root	/usr/sbin/tcpd
-/usr/sbin/in.tftpd -s /tftpboot -r blksize
+/usr/sbin/tftpd -s /tftpboot -r blksize
 
 ... it's better to change to ...
 
-tftp	dgram	udp	wait	root	/usr/sbin/in.tftpd
-in.tftpd -s /tftpboot -r blksize
+tftp	dgram	udp	wait	root	/usr/sbin/tftpd
+tftpd -s /tftpboot -r blksize
 
 You should make sure that you are using "wait" option in tftpd; you
 also need to have tftpd spawned as root in order for chroot (-s) to
Index: src/tftp-hpa-5.2/tftp-xinetd
===================================================================
--- src.orig/tftp-hpa-5.2/tftp-xinetd
+++ src/tftp-hpa-5.2/tftp-xinetd
@@ -9,7 +9,7 @@ service tftp
 	protocol		= udp
 	wait			= yes
 	user			= root
-	server			= /usr/sbin/in.tftpd
+	server			= /usr/sbin/tftpd
 	server_args		= -s /tftpboot
 	disable			= yes
 	per_source		= 11
Index: src/tftp-hpa-5.2/tftpd/Makefile
===================================================================
--- src.orig/tftp-hpa-5.2/tftpd/Makefile
+++ src/tftp-hpa-5.2/tftpd/Makefile
@@ -18,9 +18,9 @@ tftpd.8: tftpd.8.in ../version
 
 install: all
 	mkdir -p $(INSTALLROOT)$(SBINDIR) $(INSTALLROOT)$(MANDIR)/man8
-	$(INSTALL_PROGRAM) tftpd$(X) $(INSTALLROOT)$(SBINDIR)/in.tftpd
-	$(INSTALL_DATA)    tftpd.8 $(INSTALLROOT)$(MANDIR)/man8/in.tftpd.8
-	cd $(INSTALLROOT)$(MANDIR)/man8 && $(LN_S) -f in.tftpd.8 tftpd.8
+	$(INSTALL_PROGRAM) tftpd$(X) $(INSTALLROOT)$(SBINDIR)/tftpd$(X)
+	$(INSTALL_DATA)    tftpd.8 $(INSTALLROOT)$(MANDIR)/man8/tftpd.8
+	#cd $(INSTALLROOT)$(MANDIR)/man8 && $(LN_S) -f in.tftpd.8 tftpd.8
 
 clean:
 	rm -f *.o *.obj *.exe tftpd tftpsubs.c tftpsubs.h tftpd.8
Index: src/tftp-hpa-5.2/tftpd/tftpd.8.in
===================================================================
--- src.orig/tftp-hpa-5.2/tftpd/tftpd.8.in
+++ src/tftp-hpa-5.2/tftpd/tftpd.8.in
@@ -35,7 +35,7 @@
 .B tftpd
 \- Trivial File Transfer Protocol server
 .SH SYNOPSIS
-.B in.tftpd
+.B tftpd
 .RI [ options... ]
 .I directory...
 .SH DESCRIPTION
@@ -160,7 +160,7 @@ Specify the use of filename remapping.
 .I remap-file
 is a file containing the remapping rules.  See the section on filename
 remapping below.  This option may not be compiled in, see the output of
-.B "in.tftpd \-V"
+.B "tftpd \-V"
 to verify whether or not it is available.
 .TP
 \fB\-\-verbose\fP, \fB\-v\fP
@@ -363,7 +363,7 @@ network; this may not be appropriate on
 implications should be considered before enabling TFTP service.
 Typically, some kind of firewall or packet-filter solution should be
 employed.  If appropriately compiled (see the output of
-.BR  "in.tftpd \-\-version" )
+.BR  "tftpd \-\-version" )
 .B tftpd
 will query the
 .BR hosts_access (5)
