Index: src/tftp-hpa-5.2/tftp/main.c
===================================================================
--- src.orig/tftp-hpa-5.2/tftp/main.c
+++ src/tftp-hpa-5.2/tftp/main.c
@@ -876,13 +876,13 @@ static void makeargv(void)
 
     margc = 0;
     for (cp = line; *cp;) {
-        while (isspace(*cp))
+        while (isspace((int)*cp))
             cp++;
         if (*cp == '\0')
             break;
         *argp++ = cp;
         margc += 1;
-        while (*cp != '\0' && !isspace(*cp))
+        while (*cp != '\0' && !isspace((int)*cp))
             cp++;
         if (*cp == '\0')
             break;
Index: src/tftp-hpa-5.2/tftpd/recvfrom.c
===================================================================
--- src.orig/tftp-hpa-5.2/tftpd/recvfrom.c
+++ src/tftp-hpa-5.2/tftpd/recvfrom.c
@@ -183,7 +183,7 @@ myrecvfrom(int s, void *buf, int len, un
         bzero(myaddr, sizeof(*myaddr));
         myaddr->sa.sa_family = from->sa_family;
 
-        if (msg.msg_controllen < sizeof(struct cmsghdr) ||
+        if ((unsigned int)msg.msg_controllen < sizeof(struct cmsghdr) ||
             (msg.msg_flags & MSG_CTRUNC))
             return n;           /* No information available */
 
Index: src/tftp-hpa-5.2/tftpd/tftpd.c
===================================================================
--- src.orig/tftp-hpa-5.2/tftpd/tftpd.c
+++ src/tftp-hpa-5.2/tftpd/tftpd.c
@@ -1103,7 +1103,7 @@ int tftp(struct tftphdr *tp, int size)
             mode = ++cp;
         } else if (argn == 2) {
             for (cp = mode; *cp; cp++)
-                *cp = tolower(*cp);
+                *cp = tolower((int)*cp);
             for (pf = formats; pf->f_mode; pf++) {
                 if (!strcmp(pf->f_mode, mode))
                     break;
Index: src/tftp-hpa-5.2/tftpd/remap.c
===================================================================
--- src.orig/tftp-hpa-5.2/tftpd/remap.c
+++ src/tftp-hpa-5.2/tftpd/remap.c
@@ -159,7 +159,7 @@ static int readescstring(char *buf, char
     char *p = *str;
     int wasbs = 0, len = 0;
 
-    while (*p && isspace(*p))
+    while (*p && isspace((int)*p))
         p++;
 
     if (!*p) {
@@ -169,7 +169,7 @@ static int readescstring(char *buf, char
     }
 
     while (*p) {
-        if (!wasbs && (isspace(*p) || *p == '#')) {
+        if (!wasbs && (isspace((int)*p) || *p == '#')) {
             *buf = '\0';
             *str = p;
             return len;
