Index: src/tftp-hpa-5.2/tftpd/tftpd.c
===================================================================
--- src.orig/tftp-hpa-5.2/tftpd/tftpd.c
+++ src/tftp-hpa-5.2/tftpd/tftpd.c
@@ -578,11 +578,12 @@ int main(int argc, char **argv)
                     syslog(LOG_ERR,
                            "cannot open IPv6 socket, disable IPv6: %m");
                 }
+            } else {
+                set_socket_nonblock(fd6, 1);
+                memset(&bindaddr6, 0, sizeof bindaddr6);
+                bindaddr6.sin6_family = AF_INET6;
+                bindaddr6.sin6_port = htons(IPPORT_TFTP);
             }
-            set_socket_nonblock(fd6, 1);
-            memset(&bindaddr6, 0, sizeof bindaddr6);
-            bindaddr6.sin6_family = AF_INET6;
-            bindaddr6.sin6_port = htons(IPPORT_TFTP);
         }
 #endif
         if (address) {
