--- tftp-hpa-5.2-orig/configure.in	2014-12-12 13:15:56.678600000 +0300
+++ tftp-hpa-5.2/configure.in	2014-12-12 13:17:06.879400000 +0300
@@ -168,6 +168,13 @@
       *-liberty* )
         LIBS=`echo $LIBS | sed -e 's@-liberty@-lcygwin -liberty@g'` ;;
     esac
+    ;;
+  *-*-msys*)
+    case "$LIBS" in
+      *-liberty* )
+        LIBS=`echo $LIBS | sed -e 's@-liberty@-lmsys-2.0 -liberty@g'` ;;
+    esac
+    ;;
 esac
 
 PA_SEARCH_LIBS_AND_ADD(bsd_signal, bsd, bsdsignal)
