diff -Naur diffstat-1.62/config.guess diffstat-1.62-msys/config.guess
--- diffstat-1.62/config.guess	2013-02-10 14:28:37.000000000 +0000
+++ diffstat-1.62-msys/config.guess	2013-04-10 11:28:19.249023400 +0000
@@ -883,6 +883,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
