--- gyp/pylib/gyp/generator/make.py.orig	2021-05-13 12:43:49.416056500 +0200
+++ gyp/pylib/gyp/generator/make.py	2021-05-13 12:44:38.698110700 +0200
@@ -1375,7 +1375,10 @@
       if self.flavor == 'aix':
         target_ext = '.a'
       else:
-        target_ext = '.so'
+        if self.flavor == 'win':
+          target_ext = '.dll'
+        else:
+          target_ext = '.so'
     elif self.type == 'none':
       target = '%s.stamp' % target
     elif self.type != 'executable':
