diff -Naur libssh2-1.4.3-orig/config.guess libssh2-1.4.3/config.guess
--- libssh2-1.4.3-orig/config.guess	2012-06-04 18:47:14.000000000 +0000
+++ libssh2-1.4.3/config.guess	2013-04-25 06:38:52.212890600 +0000
@@ -883,6 +883,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
diff -Naur libssh2-1.4.3-orig/config.rpath libssh2-1.4.3/config.rpath
--- libssh2-1.4.3-orig/config.rpath	2012-06-11 12:51:24.000000000 +0000
+++ libssh2-1.4.3/config.rpath	2013-04-25 06:39:21.208984300 +0000
@@ -129,7 +129,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -179,7 +179,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -323,7 +323,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -510,7 +510,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur libssh2-1.4.3-orig/configure.ac libssh2-1.4.3/configure.ac
--- libssh2-1.4.3-orig/configure.ac	2012-08-13 21:34:35.000000000 +0000
+++ libssh2-1.4.3/configure.ac	2013-04-25 06:44:03.552734300 +0000
@@ -244,7 +244,7 @@
 AM_CONDITIONAL([HAVE_SYS_UN_H], test "x$have_sys_un_h" = xyes)
 
 case $host in
-  *-*-cygwin* | *-*-cegcc*)
+  *-*-cygwin* | *-*-msys* | *-*-cegcc*)
     # These are POSIX-like systems using BSD-like sockets API.
     ;;
   *)
