From 393316144122b076f32e4ad9cb26cd8af780afdc Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Sun, 11 Jan 2026 09:39:51 +0800
Subject: [PATCH] Remove bool that conflict with c23

---
 yodl/root/root.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/yodl/root/root.h b/yodl/root/root.h
index 5542ec8..127526b 100644
--- a/yodl/root/root.h
+++ b/yodl/root/root.h
@@ -15,13 +15,6 @@
 
 #define MAX_LINE_LENGTH (1000)
 
-typedef enum
-{
-    false = 0,
-    true  = 1
-}
-bool;
-
 typedef enum
 {
     SUCCESS = 0,
-- 
2.52.0.windows.1

