From 08c2316f2f7791bc3a9243aabe8df84aaaf8b205 Mon Sep 17 00:00:00 2001
From: David Macek <david.macek.0@gmail.com>
Date: Sat, 14 Nov 2015 11:27:43 +0100
Subject: [PATCH] Apply POSIX-to-Win conversion on all arguments on MSYS

---
 src/unix-adapter/main.cc | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/unix-adapter/main.cc b/unix-adapter/main.cc
index da19735..b8092a0 100644
--- a/src/unix-adapter/main.cc
+++ b/src/unix-adapter/main.cc
@@ -412,6 +412,16 @@ int main(int argc, char *argv[])
         exit(1);
     }
 
+#if defined(__MSYS__)
+    {
+        int argc = args.childArgv.size();
+        int argi = 1;
+        for (; argi < argc; ++argi) {
+            args.childArgv[argi] = convertPosixPathToWin(args.childArgv[argi]);
+        }
+    }
+#endif
+
     {
         // Start the child process under the console.
         args.childArgv[0] = convertPosixPathToWin(args.childArgv[0]);
-- 
2.6.2.windows.1
