--- winpty-0.4.3/src/unix-adapter/main.cc.orig	2023-12-17 14:19:19.673630100 +0100
+++ winpty-0.4.3/src/unix-adapter/main.cc	2023-12-17 14:20:01.022627000 +0100
@@ -198,6 +199,10 @@
     sigaction(SIGWINCH, &resizeSigAct, NULL);
 }
 
+#ifndef CYGWIN_VERSION_CYGWIN_CONV
+    #define CYGWIN_VERSION_CYGWIN_CONV 181
+#endif
+
 // Convert the path to a Win32 path if it is a POSIX path, and convert slashes
 // to backslashes.
 static std::string convertPosixPathToWin(const std::string &path)
