--- winpty-0.4.0/src/unix-adapter/subdir.mk.orig	2016-10-07 03:12:21.901940000 -0400
+++ winpty-0.4.0/src/unix-adapter/subdir.mk	2016-10-07 03:16:52.075885600 -0400
@@ -28,6 +28,7 @@
 	build/unix-adapter/unix-adapter/Util.o \
 	build/unix-adapter/unix-adapter/WakeupFd.o \
 	build/unix-adapter/unix-adapter/main.o \
+        build/unix-adapter/unix-adapter/path_conv.o \
 	build/unix-adapter/shared/DebugClient.o \
 	build/unix-adapter/shared/WinptyAssert.o \
 	build/unix-adapter/shared/WinptyVersion.o
