diff -durN man-db-2.7.6.orig/build-aux/config.rpath man-db-2.7.6/build-aux/config.rpath
--- man-db-2.7.6.orig/build-aux/config.rpath	2017-05-17 21:23:59.734375000 +0800
+++ man-db-2.7.6/build-aux/config.rpath	2017-05-17 21:42:36.765625000 +0800
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -durN man-db-2.7.6.orig/configure man-db-2.7.6/configure
--- man-db-2.7.6.orig/configure	2017-05-17 21:31:43.609375000 +0800
+++ man-db-2.7.6/configure	2017-05-17 21:48:07.906250000 +0800
@@ -15760,7 +15760,7 @@
      else
        case "$host_os" in
                                                                osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
@@ -19386,7 +19386,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -21230,6 +21230,8 @@
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
             cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+                     # Guess yes on msys.
+            msys*)   gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
@@ -24554,7 +24556,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
@@ -29589,6 +29591,8 @@
            case "$host_os" in
                       # Guess no on Cygwin.
              cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+                      # Guess no on msys.
+             msys*)   gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
@@ -30671,7 +30675,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -30814,6 +30818,8 @@
              gl_cv_func_dup2_works="guessing no" ;;
            cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
+           msys*) # on MSYS, dup2(1,1) returns 0
+             gl_cv_func_dup2_works="guessing no" ;;
            aix* | freebsd*)
                    # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
                    # not EBADF.
@@ -31170,7 +31176,7 @@
 else
   if test "$cross_compiling" = yes; then :
   case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac
@@ -33258,7 +33264,7 @@
 else
   if test "$cross_compiling" = yes; then :
   case "$host_os" in
-          cygwin*) # on cygwin 1.5.25, getdtablesize() automatically grows
+          cygwin* | msys*) # on cygwin 1.5.25, getdtablesize() automatically grows
             gl_cv_func_getdtablesize_works="guessing no" ;;
           *) gl_cv_func_getdtablesize_works="guessing yes" ;;
          esac
@@ -35223,7 +35229,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -durN man-db-2.7.6.orig/gnulib/lib/Makefile.am man-db-2.7.6/gnulib/lib/Makefile.am
--- man-db-2.7.6.orig/gnulib/lib/Makefile.am	2017-05-17 21:28:42.859375000 +0800
+++ man-db-2.7.6/gnulib/lib/Makefile.am	2017-05-17 21:50:32.281250000 +0800
@@ -917,7 +917,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -durN man-db-2.7.6.orig/gnulib/lib/Makefile.in man-db-2.7.6/gnulib/lib/Makefile.in
--- man-db-2.7.6.orig/gnulib/lib/Makefile.in	2017-05-17 21:32:00.296875000 +0800
+++ man-db-2.7.6/gnulib/lib/Makefile.in	2017-05-17 21:51:00.531250000 +0800
@@ -2478,7 +2478,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -durN man-db-2.7.6.orig/gnulib/m4/btowc.m4 man-db-2.7.6/gnulib/m4/btowc.m4
--- man-db-2.7.6.orig/gnulib/m4/btowc.m4	2017-05-17 21:25:29.703125000 +0800
+++ man-db-2.7.6/gnulib/m4/btowc.m4	2017-05-17 21:51:43.921875000 +0800
@@ -50,6 +50,8 @@
            case "$host_os" in
                       # Guess no on Cygwin.
              cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+                      # Guess no on msys.
+             msys*)   gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
diff -durN man-db-2.7.6.orig/gnulib/m4/double-slash-root.m4 man-db-2.7.6/gnulib/m4/double-slash-root.m4
--- man-db-2.7.6.orig/gnulib/m4/double-slash-root.m4	2017-05-17 21:25:35.000000000 +0800
+++ man-db-2.7.6/gnulib/m4/double-slash-root.m4	2017-05-17 21:52:24.734375000 +0800
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -durN man-db-2.7.6.orig/gnulib/m4/dup2.m4 man-db-2.7.6/gnulib/m4/dup2.m4
--- man-db-2.7.6.orig/gnulib/m4/dup2.m4	2017-05-17 21:25:35.875000000 +0800
+++ man-db-2.7.6/gnulib/m4/dup2.m4	2017-05-17 21:52:59.703125000 +0800
@@ -80,6 +80,8 @@
              gl_cv_func_dup2_works="guessing no" ;;
            cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
+           msys*) # on MSYS, dup2(1,1) returns 0
+             gl_cv_func_dup2_works="guessing no" ;;
            aix* | freebsd*)
                    # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
                    # not EBADF.
diff -durN man-db-2.7.6.orig/gnulib/m4/fcntl.m4 man-db-2.7.6/gnulib/m4/fcntl.m4
--- man-db-2.7.6.orig/gnulib/m4/fcntl.m4	2017-05-17 21:25:40.890625000 +0800
+++ man-db-2.7.6/gnulib/m4/fcntl.m4	2017-05-17 21:53:34.500000000 +0800
@@ -69,7 +69,7 @@
          [gl_cv_func_fcntl_f_dupfd_works=yes],
          [gl_cv_func_fcntl_f_dupfd_works=no],
          [case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac])])
diff -durN man-db-2.7.6.orig/gnulib/m4/getcwd.m4 man-db-2.7.6/gnulib/m4/getcwd.m4
--- man-db-2.7.6.orig/gnulib/m4/getcwd.m4	2017-05-17 21:25:45.078125000 +0800
+++ man-db-2.7.6/gnulib/m4/getcwd.m4	2017-05-17 21:56:50.484375000 +0800
@@ -52,6 +52,8 @@
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
             cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+                     # Guess yes on msys.
+            msys*)   gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
diff -durN man-db-2.7.6.orig/gnulib/m4/getdtablesize.m4 man-db-2.7.6/gnulib/m4/getdtablesize.m4
--- man-db-2.7.6.orig/gnulib/m4/getdtablesize.m4	2017-05-17 21:25:45.500000000 +0800
+++ man-db-2.7.6/gnulib/m4/getdtablesize.m4	2017-05-17 21:57:23.140625000 +0800
@@ -28,7 +28,7 @@
         [gl_cv_func_getdtablesize_works=yes],
         [gl_cv_func_getdtablesize_works=no],
         [case "$host_os" in
-          cygwin*) # on cygwin 1.5.25, getdtablesize() automatically grows
+          cygwin* | msys*) # on cygwin 1.5.25, getdtablesize() automatically grows
             gl_cv_func_getdtablesize_works="guessing no" ;;
           *) gl_cv_func_getdtablesize_works="guessing yes" ;;
          esac])
diff -durN man-db-2.7.6.orig/gnulib/m4/intl.m4 man-db-2.7.6/gnulib/m4/intl.m4
--- man-db-2.7.6.orig/gnulib/m4/intl.m4	2017-05-17 21:25:50.843750000 +0800
+++ man-db-2.7.6/gnulib/m4/intl.m4	2017-05-17 21:58:02.843750000 +0800
@@ -113,7 +113,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -125,7 +125,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -durN man-db-2.7.6.orig/gnulib/m4/malloc.m4 man-db-2.7.6/gnulib/m4/malloc.m4
--- man-db-2.7.6.orig/gnulib/m4/malloc.m4	2017-05-17 21:25:57.250000000 +0800
+++ man-db-2.7.6/gnulib/m4/malloc.m4	2017-05-17 21:59:34.296875000 +0800
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -durN man-db-2.7.6.orig/gnulib/m4/printf.m4 man-db-2.7.6/gnulib/m4/printf.m4
--- man-db-2.7.6.orig/gnulib/m4/printf.m4	2017-05-17 21:26:06.562500000 +0800
+++ man-db-2.7.6/gnulib/m4/printf.m4	2017-05-17 22:00:50.093750000 +0800
@@ -730,6 +730,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+		   msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -807,6 +808,7 @@
 changequote(,)dnl
          case "$host_os" in
            cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+		   msys*)        gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -1459,6 +1461,8 @@
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
            cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+                                 # Guess yes on msys.
+           msys*)                gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -durN man-db-2.7.6.orig/gnulib/m4/realloc.m4 man-db-2.7.6/gnulib/m4/realloc.m4
--- man-db-2.7.6.orig/gnulib/m4/realloc.m4	2017-05-17 21:26:08.156250000 +0800
+++ man-db-2.7.6/gnulib/m4/realloc.m4	2017-05-17 22:01:17.046875000 +0800
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
diff -durN man-db-2.7.6.orig/gnulib/m4/threadlib.m4 man-db-2.7.6/gnulib/m4/threadlib.m4
--- man-db-2.7.6.orig/gnulib/m4/threadlib.m4	2017-05-17 21:26:21.734375000 +0800
+++ man-db-2.7.6/gnulib/m4/threadlib.m4	2017-05-17 22:03:18.218750000 +0800
@@ -68,7 +68,7 @@
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
