--- man-db-2.6.7.1-orig/libdb/Makefile.am	2013-12-09 04:43:01.000000000 +0400
+++ man-db-2.6.7.1/libdb/Makefile.am	2014-06-14 12:21:58.440600000 +0400
@@ -43,4 +43,4 @@
 
 libmandb_la_LIBADD = ../lib/libman.la $(DBLIBS)
 
-libmandb_la_LDFLAGS = -avoid-version -release $(VERSION) -rpath $(pkglibdir)
+libmandb_la_LDFLAGS = -avoid-version -no-undefined -release $(VERSION) -rpath $(pkglibdir)
--- man-db-2.6.7.1-orig/lib/Makefile.am	2013-12-09 04:43:01.000000000 +0400
+++ man-db-2.6.7.1/lib/Makefile.am	2014-06-14 12:24:38.687800000 +0400
@@ -61,5 +61,5 @@
 	$(LIBCOMPRESS) @LTLIBINTL@
 
 libman_la_LDFLAGS = \
-	-avoid-version -release $(VERSION) -rpath $(pkglibdir) \
+	-avoid-version -no-undefined -release $(VERSION) -rpath $(pkglibdir) \
 	$(libpipeline_LIBS)
