--- Test-Script-1.07/inc/Module/Install/Can.pm.orig	2009-11-24 05:47:12.000000000 +0300
+++ Test-Script-1.07/inc/Module/Install/Can.pm	2013-11-13 21:57:47.044600000 +0400
@@ -59,7 +59,7 @@
 }
 
 # Fix Cygwin bug on maybe_command();
-if ( $^O eq 'cygwin' ) {
+if ( $^O eq 'cygwin' or $^O eq 'msys' ) {
 	require ExtUtils::MM_Cygwin;
 	require ExtUtils::MM_Win32;
 	if ( ! defined(&ExtUtils::MM_Cygwin::maybe_command) ) {
