diff -aurN 000/colordiff.pl 001/colordiff.pl
--- 000/colordiff.pl
+++ 001/colordiff.pl
@@ -503,7 +503,13 @@
         }
     }
     elsif ($diff_type eq 'diffu') {
-        if (/^-/) {
+        if (/^--- /) {
+            print "$colour{darkyellow}";
+        }
+        elsif (/^\+\+\+ /) {
+            print "$colour{darkyellow}";
+        }
+        elsif (/^-/) {
             print "$file_old";
         }
         elsif (/^\+/) {
