Force noinst test modules to build shared.

--- origsrc/heimdal-1.2.1/lib/kadm5/Makefile.am	2008-07-28 07:04:41.000000000 -0500
+++ src/heimdal-1.2.1/lib/kadm5/Makefile.am	2009-07-07 19:46:23.224814700 -0500
@@ -19,7 +19,8 @@ noinst_PROGRAMS = test_pw_quality
 noinst_LTLIBRARIES = sample_passwd_check.la
 
 sample_passwd_check_la_SOURCES = sample_passwd_check.c
-sample_passwd_check_la_LDFLAGS = -module
+sample_passwd_check_la_LDFLAGS = -module -avoid-version -rpath $(libdir)
+sample_passwd_check_la_LIBADD = ../krb5/libkrb5.la
 
 libkadm5srv_la_LIBADD = \
 	$(LIB_com_err) ../krb5/libkrb5.la \
--- origsrc/heimdal-1.2.1/tests/plugin/Makefile.am	2008-07-28 07:04:34.000000000 -0500
+++ src/heimdal-1.2.1/tests/plugin/Makefile.am	2009-07-07 19:46:37.817537600 -0500
@@ -25,10 +25,11 @@ krb5.conf: krb5.conf.in Makefile
 	$(do_subst) < $(srcdir)/krb5.conf.in > krb5.conf.tmp
 	mv krb5.conf.tmp krb5.conf
 
-lib_LTLIBRARIES = windc.la
+noinst_LTLIBRARIES = windc.la
 
 windc_la_SOURCES = windc.c
-windc_la_LDFLAGS = -module
+windc_la_LDFLAGS = -module -avoid-version -rpath $(libdir)
+windc_la_LIBADD = $(LDADD)
 
 CLEANFILES= \
 	$(TESTS) \
