--- heimdal-7.1.0/configure.ac.orig	2016-01-12 12:08:18.406147300 +0300
+++ heimdal-7.1.0/configure.ac	2016-01-12 12:10:33.177960000 +0300
@@ -299,14 +299,14 @@
 
 rk_TEST_PACKAGE(libedit,
 [#include <stdio.h>
-#if defined(HAVE_READLINE_READLINE_H)
-#include <readline/readline.h>
+#if defined(HAVE_EDITLINE_READLINE_H)
+#include <editline/readline.h>
 #elif defined(HAVE_READLINE_H)
 #include <readline.h>
 #elif defined(HAVE_EDITLINE_READLINE_H)
 #include <editline/readline.h>
 #endif
-],-ledit,,, READLINE,, [readline.h readline/readline.h editline/readline.h])
+],-ledit,,, READLINE,, [readline.h editline/readline.h])
 
 AC_CONFIG_SUBDIRS([lib/libedit])
 
