--- origsrc/heimdal-1.5.2/cf/install-catman.sh	2012-01-10 15:53:51.000000000 -0600
+++ src/heimdal-1.5.2/cf/install-catman.sh	2012-01-25 20:22:35.502695800 -0600
@@ -38,8 +38,9 @@ for f in "$@"; do
 		fi
 		if test "$cmd" = install ; then
 			target="$mandir/$link.$section"
-			for lncmd in "ln -f $mandir/$base.$section $target" \
+			for lncmd in \
 				   "ln -s $base.$section $target" \
+				   "ln -f $mandir/$base.$section $target" \
 				   "cp -f $mandir/$base.$section $target"
 			do
 				if eval "$lncmd"; then
