--- heimdal-heimdal-7.5.0/lib/hcrypto/Makefile.am.orig	2018-06-27 08:49:04.243933000 +0300
+++ heimdal-heimdal-7.5.0/lib/hcrypto/Makefile.am	2018-06-27 08:51:02.999799700 +0300
@@ -9,6 +9,7 @@
 endif
 
 AM_CPPFLAGS += -I$(top_srcdir)/lib/hx509 \
+	       -I$(top_srcdir)/lib \
 	       -I$(srcdir)/libtommath -DUSE_HCRYPTO_LTM=1
 
 lib_LTLIBRARIES = libhcrypto.la
@@ -296,9 +297,6 @@
 	libtommath/bn_mp_to_signed_bin_n.c \
 	libtommath/bn_mp_to_unsigned_bin_n.c
 
-
-$(libhcrypto_la_OBJECTS): hcrypto-link
-
 libhcrypto_la_CPPFLAGS = -DBUILD_HCRYPTO_LIB $(AM_CPPFLAGS)
 
 if versionscript
@@ -306,11 +304,6 @@
 endif
 $(libhcrypto_la_OBJECTS): $(srcdir)/version-script.map
 
-
-hcrypto-link:
-	$(LN_S) $(srcdir)/../hcrypto hcrypto
-	touch hcrypto-link
-
 do_subst = sed -e 's,[@]srcdir[@],$(srcdir),g' -e 's,[@]exeext[@],$(exeext),g'
 
 test_crypto: test_crypto.in Makefile
@@ -323,7 +316,6 @@
 	crypto-test2 \
 	error \
 	hcrypto \
-	hcrypto-link \
 	test.file \
 	test_crypto \
 	test-out* \
