diff -Naur heimdal-heimdal-7.5.0-orig/lib/hcrypto/camellia.c heimdal-heimdal-7.5.0/lib/hcrypto/camellia.c
--- heimdal-heimdal-7.5.0-orig/lib/hcrypto/camellia.c	2017-12-08 10:48:25.000000000 +0300
+++ heimdal-heimdal-7.5.0/lib/hcrypto/camellia.c	2018-07-06 15:55:40.543752700 +0300
@@ -46,7 +46,7 @@
 		 const int bits, CAMELLIA_KEY *key)
 {
     key->bits = bits;
-    Camellia_Ekeygen(bits, userkey, key->key);
+    Camellia_Ekeygen_Heimdal(bits, userkey, key->key);
     return 1;
 }
 
@@ -54,7 +54,7 @@
 CAMELLIA_encrypt(const unsigned char *in, unsigned char *out,
 		 const CAMELLIA_KEY *key)
 {
-    Camellia_EncryptBlock(key->bits, in, key->key, out);
+    Camellia_EncryptBlock_Heimdal(key->bits, in, key->key, out);
 
 }
 
@@ -62,7 +62,7 @@
 CAMELLIA_decrypt(const unsigned char *in, unsigned char *out,
 		 const CAMELLIA_KEY *key)
 {
-    Camellia_DecryptBlock(key->bits, in, key->key, out);
+    Camellia_DecryptBlock_Heimdal(key->bits, in, key->key, out);
 }
 
 void
diff -Naur heimdal-heimdal-7.5.0-orig/lib/hcrypto/camellia-ntt.c heimdal-heimdal-7.5.0/lib/hcrypto/camellia-ntt.c
--- heimdal-heimdal-7.5.0-orig/lib/hcrypto/camellia-ntt.c	2017-12-08 10:48:25.000000000 +0300
+++ heimdal-heimdal-7.5.0/lib/hcrypto/camellia-ntt.c	2018-07-06 15:55:36.097744900 +0300
@@ -1383,7 +1383,7 @@
  * API for compatibility
  */
 
-void Camellia_Ekeygen(const int keyBitLength,
+void Camellia_Ekeygen_Heimdal(const int keyBitLength,
 		      const unsigned char *rawKey,
 		      KEY_TABLE_TYPE keyTable)
 {
@@ -1403,7 +1403,7 @@
 }
 
 
-void Camellia_EncryptBlock(const int keyBitLength,
+void Camellia_EncryptBlock_Heimdal(const int keyBitLength,
 			   const unsigned char *plaintext,
 			   const KEY_TABLE_TYPE keyTable,
 			   unsigned char *ciphertext)
@@ -1434,7 +1434,7 @@
     PUTU32(ciphertext + 12, tmp[3]);
 }
 
-void Camellia_DecryptBlock(const int keyBitLength,
+void Camellia_DecryptBlock_Heimdal(const int keyBitLength,
 			   const unsigned char *ciphertext,
 			   const KEY_TABLE_TYPE keyTable,
 			   unsigned char *plaintext)
diff -Naur heimdal-heimdal-7.5.0-orig/lib/hcrypto/camellia-ntt.h heimdal-heimdal-7.5.0/lib/hcrypto/camellia-ntt.h
--- heimdal-heimdal-7.5.0-orig/lib/hcrypto/camellia-ntt.h	2017-12-08 10:48:25.000000000 +0300
+++ heimdal-heimdal-7.5.0/lib/hcrypto/camellia-ntt.h	2018-07-06 15:55:39.389350700 +0300
@@ -43,16 +43,16 @@
 typedef u32 KEY_TABLE_TYPE[CAMELLIA_TABLE_WORD_LEN];
 
 
-void Camellia_Ekeygen(const int keyBitLength,
+void Camellia_Ekeygen_Heimdal(const int keyBitLength,
 		      const unsigned char *rawKey,
 		      KEY_TABLE_TYPE keyTable);
 
-void Camellia_EncryptBlock(const int keyBitLength,
+void Camellia_EncryptBlock_Heimdal(const int keyBitLength,
 			   const unsigned char *plaintext,
 			   const KEY_TABLE_TYPE keyTable,
 			   unsigned char *cipherText);
 
-void Camellia_DecryptBlock(const int keyBitLength,
+void Camellia_DecryptBlock_Heimdal(const int keyBitLength,
 			   const unsigned char *cipherText,
 			   const KEY_TABLE_TYPE keyTable,
 			   unsigned char *plaintext);
