--- heimdal-heimdal-7.7.1/lib/hcrypto/Makefile.am.orig	2022-11-15 16:56:44.000000000 +0100
+++ heimdal-heimdal-7.7.1/lib/hcrypto/Makefile.am	2022-12-04 19:10:32.624493900 +0100
@@ -9,6 +9,7 @@
 endif
 
 AM_CPPFLAGS += -I$(top_srcdir)/lib/hx509 \
+	       -I$(top_srcdir)/lib \
 	       -I$(srcdir)/libtommath -DUSE_HCRYPTO_LTM=1
 
 lib_LTLIBRARIES = libhcrypto.la
@@ -294,8 +295,6 @@
 	libtommath/bn_mp_to_unsigned_bin_n.c
 
 
-$(libhcrypto_la_OBJECTS) $(test_rand_OBJECTS): hcrypto-link
-
 libhcrypto_la_CPPFLAGS = -DBUILD_HCRYPTO_LIB $(AM_CPPFLAGS)
 
 if versionscript
@@ -303,11 +302,6 @@
 endif
 $(libhcrypto_la_OBJECTS): $(srcdir)/version-script.map
 
-
-hcrypto-link:
-	$(LN_S) $(srcdir)/../hcrypto hcrypto
-	touch hcrypto-link
-
 do_subst = sed -e 's,[@]srcdir[@],$(srcdir),g' -e 's,[@]exeext[@],$(exeext),g'
 
 test_crypto: test_crypto.in Makefile
@@ -320,7 +314,6 @@
 	crypto-test2 \
 	error \
 	hcrypto \
-	hcrypto-link \
 	test.file \
 	test_crypto \
 	test-out* \
