--- a/cf/roken-frag.m4
+++ b/cf/roken-frag.m4
@@ -538,6 +538,12 @@
 #endif],
 inet_aton)
 
-AC_FIND_FUNC_NO_LIBS(crypt, crypt)dnl
+AC_FIND_FUNC_NO_LIBS(crypt, crypt,
+[#ifdef HAVE_CRYPT_H
+#include <crypt.h>
+#elif defined(HAVE_UNISTD_H)
+#include <unistd.h>
+#endif],
+["", ""])dnl
 
 AC_REQUIRE([rk_BROKEN_REALLOC])dnl
--- a/cf/check-getpwnam_r-posix.m4
+++ b/cf/check-getpwnam_r-posix.m4
@@ -3,5 +3,5 @@
 dnl check for getpwnam_r, and if it's posix or not
 
 AC_DEFUN([AC_CHECK_GETPWNAM_R_POSIX],[
-AC_FIND_FUNC_NO_LIBS(getpwnam_r,c_r)
+AC_FIND_FUNC_NO_LIBS(getpwnam_r,c_r,[#include <pwd.h>],["", 0, 0, 0, 0])
 if test "$ac_cv_func_getpwnam_r" = yes; then
