--- a/lib/krb5/kuserok.c
+++ b/lib/krb5/kuserok.c
@@ -120,7 +120,7 @@
 		struct stat *dirlstat,
 		const char *owner)
 {
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
     /*
      * XXX Implement this!
      *
@@ -193,7 +193,7 @@
 		 const char *filename,
 		 FILE *file, const char *owner)
 {
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
     /*
      * XXX Implement this!
      *
