--- a/lib/hdb/Makefile.am
+++ b/lib/hdb/Makefile.am
@@ -105,3 +105,4 @@ libhdb_la_LIBADD =
 libhdb_la_LIBADD = \
+	$(LIB_heimbase) \
 	$(LIB_com_err) \
 	../krb5/libkrb5.la \
--- a/lib/ntlm/Makefile.am
+++ b/lib/ntlm/Makefile.am
@@ -20,5 +20,6 @@
 $(libheimntlm_la_OBJECTS): $(srcdir)/version-script.map
 
 libheimntlm_la_LIBADD = \
+	$(LIB_com_err) \
 	../krb5/libkrb5.la \
 	$(top_builddir)/lib/wind/libwind.la \
--- a/lib/kafs/Makefile.am
+++ b/lib/kafs/Makefile.am
@@ -38,6 +38,6 @@
 AIX_SRC =
 endif # AIX
 
-libkafs_la_LIBADD = $(DEPLIB_krb5) $(LIBADD_roken)
+libkafs_la_LIBADD = ../asn1/libasn1.la $(DEPLIB_krb5) $(LIBADD_roken)
 
 lib_LTLIBRARIES = libkafs.la
--- a/lib/gssapi/Makefile.am
+++ b/lib/gssapi/Makefile.am
@@ -220,5 +220,6 @@
 endif
 
 libgssapi_la_LIBADD = \
+	$(top_builddir)/lib/base/libheimbase.la \
 	$(top_builddir)/lib/ntlm/libheimntlm.la \
 	$(top_builddir)/lib/krb5/libkrb5.la \
--- a/lib/kadm5/Makefile.am
+++ b/lib/kadm5/Makefile.am
@@ -25,5 +25,6 @@
 sample_passwd_check_la_LIBADD = ../krb5/libkrb5.la
 
 libkadm5srv_la_LIBADD = \
+	../base/libheimbase.la \
 	$(LIB_com_err) ../krb5/libkrb5.la \
 	../hdb/libhdb.la $(LIBADD_roken)
--- a/kdc/Makefile.am
+++ b/kdc/Makefile.am
@@ -103,5 +103,6 @@
 endif
 
 libkdc_la_LIBADD = \
+	$(top_builddir)/lib/base/libheimbase.la \
 	$(LIB_pkinit) \
 	$(top_builddir)/lib/hdb/libhdb.la \
