From 0c42653da8ccaeb14b7c28ba9670eeac0460037e Mon Sep 17 00:00:00 2001
From: Elliott Sales de Andrade <quantum.analyst@gmail.com>
Date: Sat, 16 Sep 2017 05:34:29 -0400
Subject: [PATCH 9/9] Fix minor typo.

---
 src/deps_log.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/deps_log.h b/src/deps_log.h
index b1aa361..3812a28 100644
--- a/src/deps_log.h
+++ b/src/deps_log.h
@@ -58,7 +58,7 @@ struct State;
 ///      concurrent writes of multiple ninja processes to the log).
 ///    dependency records are an array of 4-byte integers
 ///      [output path id,
-///       output path mtime (lower 4 bytes), output path mtime (upper 8 bytes),
+///       output path mtime (lower 4 bytes), output path mtime (upper 4 bytes),
 ///       input path id, input path id...]
 ///      (The mtime is compared against the on-disk output path mtime
 ///      to verify the stored data is up-to-date.)
-- 
2.14.1

