--- BLAKE3-1.6.1/c/CMakeLists.txt.orig	2025-03-15 08:06:05.461612600 +0100
+++ BLAKE3-1.6.1/c/CMakeLists.txt	2025-03-15 08:06:15.407136700 +0100
@@ -45,7 +45,7 @@
   set(BLAKE3_CFLAGS_AVX2 "-mavx2" CACHE STRING "the compiler flags to enable AVX2")
   set(BLAKE3_CFLAGS_AVX512 "-mavx512f -mavx512vl" CACHE STRING "the compiler flags to enable AVX512")
 
-  if (WIN32)
+  if (WIN32 OR CYGWIN)
     set(BLAKE3_AMD64_ASM_SOURCES
       blake3_avx2_x86-64_windows_gnu.S
       blake3_avx512_x86-64_windows_gnu.S
