--- origsrc/rarian-0.8.0/librarian/Makefile.am	2008-03-09 13:46:22.000000000 -0500
+++ src/rarian-0.8.0/librarian/Makefile.am	2008-05-25 23:46:31.406250000 -0500
@@ -30,4 +30,4 @@
 endif
 
 librarian_la_CFLAGS = -DDATADIR=\""$(datadir)"\" -DSYSCONFDIR=\""$(sysconfdir)"\" -DI_KNOW_RARIAN_0_8_IS_UNSTABLE
-librarian_la_LDFLAGS = -export-symbols-regex "^rrn_.*"
\ No newline at end of file
+librarian_la_LDFLAGS = -export-symbols-regex "^rrn_.*" -no-undefined
