Make sure we don't end up with //UNC paths.

--- origsrc/rarian-0.8.1/librarian/rarian-reg-full.c	2008-09-01 13:40:21.000000000 -0500
+++ src/rarian-0.8.1/librarian/rarian-reg-full.c	2009-11-12 01:38:10.092197400 -0600
@@ -703,8 +703,8 @@ process_section_path_full (RrnListEntry 
 	  }
 	  if (entry->text[0] == '/') {
 	    /* Absolute path */
-	    new_uri = malloc (sizeof (char) * strlen(entry->text)+6);
-	    sprintf (new_uri, "file:/%s", entry->text);
+	    new_uri = malloc (sizeof (char) * strlen(entry->text)+7);
+	    sprintf (new_uri, "file://%s", entry->text);
 	    free (entry->text);
 	    entry->text = new_uri;
 	    goto done;
--- origsrc/rarian-0.8.1/librarian/rarian-reg-utils.c	2008-09-01 13:40:21.000000000 -0500
+++ src/rarian-0.8.1/librarian/rarian-reg-utils.c	2009-11-12 01:39:44.987162900 -0600
@@ -723,7 +723,7 @@ process_path (RrnReg *reg)
 {
 	char *prefix = NULL;
 	RrnSect *child = reg->children;
-	if (!strncmp ("file://", reg->uri, 7)) {
+	if (!strncmp ("file:///", reg->uri, 8)) {
 		/* No processing needs done.  The URI is already in the file: scheme */
 		return;
 	}
@@ -782,8 +782,8 @@ process_section_path (char *owner_path, 
 	}
 	if (section->uri[0] == '/') {
 		/* Absolute path */
-		new_uri = malloc (sizeof (char) * strlen(section->uri)+6);
-		sprintf (new_uri, "file:/%s", section->uri);
+		new_uri = malloc (sizeof (char) * strlen(section->uri)+7);
+		sprintf (new_uri, "file://%s", section->uri);
 		free (section->uri);
 		section->uri = new_uri;
 		goto done;
--- origsrc/rarian-0.8.1/util/rarian-sk-preinstall.cpp	2008-09-01 13:40:21.000000000 -0500
+++ src/rarian-0.8.1/util/rarian-sk-preinstall.cpp	2009-11-12 01:30:19.236576500 -0600
@@ -97,7 +97,7 @@ process_new_url (char *input)
     if (i == 1) {
       /* Normal path.  Add file:/ to the start */
       new_url = (char *) malloc (sizeof(char) * (strlen (input) + 7));
-      sprintf (new_url, "file:/%s", input);
+      sprintf (new_url, "file://%s", input);
     } else {
       /* Don't know what to do.  Just copy and append file: to it */
       new_url = (char *) malloc (sizeof(char) * (strlen(input) + 6));
