--- dos2unix-6.0.3/Makefile.orig	2012-09-15 18:14:06.000000000 +0000
+++ dos2unix-6.0.3/Makefile	2013-07-16 22:24:56.124166500 +0000
@@ -39,7 +39,7 @@
 CC		= gcc
 CPP		= cpp
 CPP_FLAGS_POD	= ALL
-STRIP		= strip
+STRIP		= true
 
 PACKAGE		= dos2unix
 UNIX2DOS	= unix2dos
@@ -50,10 +50,10 @@
 ENABLE_NLS	= 1
 # Large File Support (LFS)
 LFS             = 1
-DEBUG = 0
+DEBUG = 1
 UCS = 1
 
-EXE=
+EXE=.exe
 
 BIN		= $(PACKAGE)$(EXE)
 UNIX2DOS_BIN	= $(UNIX2DOS)$(EXE)
@@ -117,7 +117,7 @@
 
 # ......................................................... OS flags ...
 
-D2U_OS =
+D2U_OS = 
 
 ifndef D2U_OS
 ifeq (Linux, $(shell uname -s))
@@ -153,10 +153,9 @@
 	D2U_OS = msys
 	EXE = .exe
 	VERSIONSUFFIX	= -msys
-	EO_XNOTATION=1
-	UCS =
+	docsubdir = $(PACKAGE)/$(DOS2UNIX_VERSION)
 ifdef ENABLE_NLS
-	LDFLAGS_EXTRA = -lintl -liconv
+	LDFLAGS_EXTRA = -lintl -liconv -Wl,--enable-auto-import
 endif
 endif
 endif
