diff -ru dos2unix-7.3.4.orig/common.c dos2unix-7.3.4/common.c
--- dos2unix-7.3.4.orig/common.c	2016-05-24 22:11:15.000000000 +0200
+++ dos2unix-7.3.4/common.c	2016-05-25 21:05:33.862572800 +0200
@@ -800,6 +800,7 @@
   return path;
 }
 
+#ifdef NO_MKSTEMP
 char *basename(char *path)
 {
   char *ptr;
@@ -821,6 +822,7 @@
    return ptr ;
 }
 #endif
+#endif
 
 /* Standard mktemp() is not safe to use (See mktemp(3)).
  * On Windows it is recommended to use GetTempFileName() (See MSDN).
