diff -Naur expat-2.1.1/conftools/config.guess expat-2.1.1-p/conftools/config.guess
--- expat-2.1.1/conftools/config.guess	2012-03-24 23:12:39.000000000 +0400
+++ expat-2.1.1-p/conftools/config.guess	2013-02-13 00:22:35.406250000 +0400
@@ -844,6 +847,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
