diff -Naur nano-8.3-orig/doc/sample.nanorc.in nano-8.3/doc/sample.nanorc.in
--- nano-8.3-orig/doc/sample.nanorc.in	2024-12-20 07:38:10.000000000 -0300
+++ nano-8.3/doc/sample.nanorc.in	2025-02-21 11:33:55.337769515 -0300
@@ -219,17 +219,17 @@
 
 ## Paint the interface elements of nano.  These are examples; there are
 ## no colors by default, except for errorcolor and spotlightcolor.
-# set titlecolor bold,white,blue
-# set promptcolor lightwhite,grey
-# set statuscolor bold,white,green
-# set errorcolor bold,white,red
-# set spotlightcolor black,lightyellow
-# set selectedcolor lightwhite,magenta
-# set stripecolor ,#444
-# set scrollercolor cyan
-# set numbercolor cyan
-# set keycolor cyan
-# set functioncolor green
+set titlecolor bold,white,magenta
+set promptcolor lightwhite,grey
+set statuscolor bold,white,#770
+set errorcolor bold,white,red
+set spotlightcolor black,lightgreen
+set selectedcolor lightwhite,blue
+set stripecolor ,#444
+set scrollercolor orange
+set numbercolor orange
+set keycolor orange
+set functioncolor yellow
 
 ## In root's .nanorc you might want to use:
 # set titlecolor bold,white,magenta
