--- origsrc/nano-2.4.0/src/text.c	2015-03-22 22:45:12.000000000 -0500
+++ src/nano-2.4.0/src/text.c	2015-03-23 14:43:04.155338700 -0500
@@ -2664,7 +2664,7 @@ const char *do_alt_speller(char *tempfil
     ssize_t current_y_save = openfile->current_y;
     ssize_t lineno_save = openfile->current->lineno;
     struct stat spellfileinfo;
-    __time_t timestamp;
+    time_t timestamp;
     pid_t pid_spell;
     char *ptr;
     static int arglen = 3;
