diff -Naur nano-2.3.2-orig/config.guess nano-2.3.2/config.guess
--- nano-2.3.2-orig/config.guess	2013-03-22 19:45:23.000000000 +0400
+++ nano-2.3.2/config.guess	2013-08-22 23:25:51.911132800 +0400
@@ -801,6 +801,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
@@ -844,6 +847,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur nano-2.3.2-orig/config.rpath nano-2.3.2/config.rpath
--- nano-2.3.2-orig/config.rpath	2013-01-01 06:53:40.000000000 +0400
+++ nano-2.3.2/config.rpath	2013-08-22 23:25:51.911132800 +0400
@@ -93,7 +93,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -131,7 +131,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -225,7 +225,7 @@
       # see comment about different semantics on the GNU ld section
       ld_shlibs=no
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -370,9 +370,9 @@
     sys_lib_search_path_spec="/shlib /usr/lib /usr/X11/lib /usr/contrib/lib /lib /usr/local/lib"
     sys_lib_dlsearch_path_spec="/shlib /usr/lib /usr/local/lib"
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     case $GCC,$host_os in
-      yes,cygwin*)
+      yes,cygwin* | yes,msys*)
         shlibext=dll.a
         ;;
       yes,mingw*)
diff -Naur nano-2.3.2-orig/doc/syntax/cmake.nanorc nano-2.3.2/doc/syntax/cmake.nanorc
--- nano-2.3.2-orig/doc/syntax/cmake.nanorc	2013-01-01 06:53:33.000000000 +0400
+++ nano-2.3.2/doc/syntax/cmake.nanorc	2013-08-22 23:25:51.926757800 +0400
@@ -12,7 +12,7 @@
 icolor brightred "^[[:space:]]*\<((end)?(function|macro)|return)"
 
 icolor cyan start="\$(\{|ENV\{)" end="\}"
-color magenta "\<(APPLE|UNIX|WIN32|CYGWIN|BORLAND|MINGW|MSVC(_IDE|60|71|80|90)?)\>"
+color magenta "\<(APPLE|UNIX|WIN32|CYGWIN|MSYS|BORLAND|MINGW|MSVC(_IDE|60|71|80|90)?)\>"
 
 icolor brightblue "^([[:space:]]*)?#.*"
 icolor brightblue "[[:space:]]#.*"
