--- nano-2.3.2-orig/configure.ac	2013-03-22 18:16:07.000000000 +0400
+++ nano-2.3.2/configure.ac	2013-08-22 23:42:41.250000000 +0400
@@ -438,11 +438,11 @@
 dnl Checks for libraries.
 
 if eval "test x$CURSES_LIB_NAME = x"; then
-    AC_CHECK_HEADERS(ncurses.h)
+    AC_CHECK_HEADERS(ncursesw/ncurses.h)
 
     if test x$enable_utf8 != xno; then
 	OLDLIBS="$LIBS"
-	AC_CHECK_TOOL(NCURSESW_CONFIG, ncursesw5-config, no)
+	AC_CHECK_TOOL(NCURSESW_CONFIG, ncursesw6-config, no)
 	if test "x$NCURSESW_CONFIG" != xno; then
 	    CURSES_LIB=`$NCURSESW_CONFIG --libs`
 	    LIBS="$CURSES_LIB $LIBS"
