--- nano-2.3.2/src/winio.c.orig	2013-08-22 23:34:34.337890700 +0400
+++ nano-2.3.2/src/winio.c	2013-08-22 23:35:37.481445300 +0400
@@ -1746,7 +1746,7 @@
 	     * wheel is equivalent to moving down three lines. */
 	    for (i = 0; i < 3; i++)
 		unget_kbinput((mevent.bstate & BUTTON4_PRESSED) ?
-			 sc_seq_or(up_void, 0) : sc_seq_or(DO_DOWN_VOID, 0), FALSE,
+			 sc_seq_or(do_up_void, 0) : sc_seq_or(do_down_void, 0), FALSE,
 			FALSE);
 
 	    return 1;
