diff --unified '--color=auto' -urN screen-5.0.1.orig/etc/etcscreenrc screen-5.0.1/etc/etcscreenrc
--- screen-5.0.1.orig/etc/etcscreenrc	2025-05-15 11:31:27.000000000 -0400
+++ screen-5.0.1/etc/etcscreenrc	2025-09-25 07:53:06.039806300 -0400
@@ -1,20 +1,14 @@
-#
-# This is an example for the global screenrc file.
-# You may want to install this file as /usr/local/etc/screenrc.
-# Check config.h for the exact location.
-#
-# Flaws of termcap and standard settings are done here.
-#
 
 #startup_message off
 
 #defflow on # will force screen to process ^S/^Q
+defflow off # leave it off, so we can use ^S to save in *emacs
 
-deflogin on
+#deflogin on
 #autodetach off
 
-vbell on
-vbell_msg "   Wuff  ----  Wuff!!  "
+#vbell on
+#vbell_msg "   Wuff  ----  Wuff!!  "
 
 # all termcap entries are now duplicated as terminfo entries.
 # only difference should be the slightly modified syntax, and check for
@@ -65,6 +59,9 @@
 termcap xterm 'is=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;4;6l'
 terminfo xterm 'is=\E[r\E[m\E[2J\E[H\E[?7h\E[?1;4;6l'
 
+# Set the hardstatus prop on gui terms to set the titlebar/icon title
+termcapinfo xterm*|rxvt*|kterm*|Eterm*|cygwin hs:ts=\E]0;:fs=\007:ds=\E]0;\007
+
 #
 # Do not use xterms alternate window buffer. 
 # This one would not add lines to the scrollback buffer.
@@ -81,14 +78,19 @@
 termcap wy75-42 nx:xo:Z0=\E[?3h\E[31h:Z1=\E[?3l\E[31h
 terminfo wy75-42 nx:xo:Z0=\E[?3h\E[31h:Z1=\E[?3l\E[31h
 
-#remove some stupid / dangerous key bindings
+# Enable true (24-bit) color
+truecolor on
+
+# Remove some unwanted key bindings
 bind ^k
 #bind L
 bind ^\
-#make them better
-bind \\ quit
-bind K kill
-bind I login on
-bind O login off
-bind } history
+
+# Adjust key bindings
+bind \\ quit --confirm
+bind k  kill --confirm
+bind K  kill
+bind I  login on
+bind O  login off
+bind }  history
 
