diff '--color=auto' -urN screen-5.0.0.m1/socket.c screen-5.0.0/socket.c
--- screen-5.0.0.m1/socket.c	2024-08-28 15:55:03.000000000 -0400
+++ screen-5.0.0/socket.c	2024-09-18 11:01:23.921380700 -0400
@@ -361,6 +361,9 @@
 	a.sun_path[ARRAY_SIZE(a.sun_path) - 1] = 0;
 	xseteuid(real_uid);
 	xsetegid(real_gid);
+    #ifdef __CYGWIN__
+        setsockopt(s, SOL_SOCKET, SO_PEERCRED, NULL, 0);
+    #endif
 	if (connect(s, (struct sockaddr *)&a, strlen(SocketPath) + 2) != -1) {
 		if (quietflag) {
 			Kill(D_userpid, SIG_BYE);
@@ -390,6 +393,9 @@
 	chmod(SocketPath, SOCKMODE);
 	if (chown(SocketPath, real_uid, real_gid))
 		Panic(errno, "chown");
+    #ifdef __CYGWIN__
+        setsockopt(s, SOL_SOCKET, SO_PEERCRED, NULL, 0);
+    #endif
 	if (listen(s, 5) == -1)
 		Panic(errno, "listen");
 #ifdef F_SETOWN
@@ -412,6 +418,9 @@
 	a.sun_path[ARRAY_SIZE(a.sun_path) - 1] = 0;
 	xseteuid(real_uid);
 	xsetegid(real_gid);
+    #ifdef __CYGWIN__
+        setsockopt(s, SOL_SOCKET, SO_PEERCRED, NULL, 0);
+    #endif
 	if (connect(s, (struct sockaddr *)&a, strlen(SocketPath) + 2) == -1) {
 		if (err)
 			Msg(errno, "%s: connect", SocketPath);
