diff --unified '--color=auto' -urN screen-5.0.1.orig/doc/screen.texinfo screen-5.0.1/doc/screen.texinfo
--- screen-5.0.1.orig/doc/screen.texinfo	2025-05-15 11:39:40.000000000 -0400
+++ screen-5.0.1/doc/screen.texinfo	2025-09-25 07:28:15.906312000 -0400
@@ -5940,7 +5940,7 @@
 hardware and operating system, the compiler used to compile, a
 description of the bug behavior, and the conditions that triggered the
 bug. In case of security issue please send an mail to
-@w{@samp{alexander_naumov@opensuse.org}}.
+@w{@samp{alexander_naumov@@opensuse.org}}.
 
 @node Availability,  , Reporting Bugs, Bugs
 @section Availability
