--- origsrc/screen-5.0.0/tty.c	2024-08-29 04:55:03.000000000 +0900
+++ src/screen-5.0.0/tty.c	2025-04-06 00:57:11.489640800 +0900
@@ -802,6 +802,7 @@ static void consredir_readev_fn(Event *
 
 int TtyGrabConsole(int fd, bool on, char *rc_name)
 {
+#ifndef __CYGWIN__
 	Display *d;
 #ifdef SRIOCSREDIR
 	int cfd;
@@ -880,6 +881,7 @@ int TtyGrabConsole(int fd, bool on, char
 	consredir_ev.type = EV_READ;
 	consredir_ev.handler = consredir_readev_fn;
 	evenq(&consredir_ev);
+#endif
 	return 0;
 }
 
