--- origsrc/gettext-0.18.1.1/gettext-tools/configure.ac	2010-06-06 15:04:04.000000000 -0500
+++ src/gettext-0.18.1.1/gettext-tools/configure.ac	2013-03-07 22:13:24.525723500 -0600
@@ -108,7 +108,7 @@ AC_SUBST([LTNOUNDEF])
 dnl On mingw and Cygwin, we can activate special Makefile rules which add
 dnl version information to the shared libraries and executables.
 case "$host_os" in
-  mingw* | cygwin*) is_woe32=yes ;;
+  mingw*) is_woe32=yes ;;
   *) is_woe32=no ;;
 esac
 AM_CONDITIONAL([WOE32], [test $is_woe32 = yes])
@@ -211,7 +211,7 @@ dnl    must arrange to define the auxili
 dnl    exported variables _also_ in the static library.
 if test "$enable_shared" = yes; then
   case "$host_os" in
-    mingw* | cygwin*) is_woe32dll=yes ;;
+    mingw*) is_woe32dll=yes ;;
     *) is_woe32dll=no ;;
   esac
 else
--- origsrc/gettext-0.18.1.1/m4/woe32-dll.m4	2010-06-06 07:49:58.000000000 -0500
+++ src/gettext-0.18.1.1/m4/woe32-dll.m4	2013-03-07 23:33:34.183820200 -0600
@@ -29,7 +29,7 @@ AC_DEFUN([gl_WOE32_DLL],
 [
   AC_REQUIRE([AC_CANONICAL_HOST])
   case "$host_os" in
-    mingw* | cygwin*)
+    mingw*)
       AC_MSG_CHECKING([for auto-import of symbols])
       AC_CACHE_VAL([gl_cv_ld_autoimport], [
         gl_save_LDFLAGS="$LDFLAGS"
