--- origsrc/gettext-0.18.1.1/gettext-tools/misc/autopoint.in	2010-06-06 08:49:57.000000000 -0400
+++ src/gettext-0.18.1.1/gettext-tools/misc/autopoint.in	2011-10-16 00:57:42.789133600 -0400
@@ -161,9 +161,9 @@ Options:
       --version        print version information and exit
   -f, --force          force overwriting of files that already exist
   -n, --dry-run        print modifications but don't perform them"
-#  echo "\
-#  -V version           copy the infrastructure of the specified gettext version
-#                         (dangerous)"
+  echo "\
+  -V version           copy the infrastructure of the specified gettext version
+                         (dangerous)"
   echo "
 Report bugs to <bug-gnu-gettext@gnu.org>."
 }
@@ -210,17 +210,17 @@ func_fatal_error ()
         force=yes ;;
       --help | --hel | --he | --h )
         func_usage; exit 0 ;;
-#      -V ) # Some people put a space between -V and the version number.
-#        shift
-#        if test $# = 0; then
-#          func_usage 1>&2
-#          exit 1
-#        fi
-#        ver=$1;
-#        shift ;;
-#      -V*) # Some people omit the space between -V and the version number.
-#        ver=`echo "X$1" | sed -e 's/^X-V//'`
-#        shift ;;
+      -V ) # Some people put a space between -V and the version number.
+        shift
+        if test $# = 0; then
+          func_usage 1>&2
+          exit 1
+        fi
+        ver=$1;
+        shift ;;
+      -V*) # Some people omit the space between -V and the version number.
+        ver=`echo "X$1" | sed -e 's/^X-V//'`
+        shift ;;
       --version | --versio | --versi | --vers | --ver | --ve | --v )
         func_version
         exit 0 ;;
@@ -276,7 +276,7 @@ fi
 if test -n "$xver"; then
   if test -n "$ver"; then
     if test "X$ver" != "X$xver"; then
-      func_fatal_error "Version mismatch: specified -V $ver but the package uses gettext version $xver"
+      echo "autopoint: warning: Version mismatch: specified -V $ver but the package uses gettext version $xver. Forcibly upgrading to $ver" 1>&2
     fi
   else
     ver="$xver"

