--- origsrc/gettext-0.19.3/gettext-tools/tests/format-c-3-prg.c	2013-05-31 00:01:54.000000000 -0500
+++ src/gettext-0.19.3/gettext-tools/tests/format-c-3-prg.c	2014-11-17 23:26:21.975557100 -0600
@@ -34,7 +34,7 @@
 /* Disable the override of setlocale that libgnuintl.h activates on MacOS X
    and Windows.  This test relies on the fake setlocale function in
    setlocale.c.  */
-#undef setlocale
+/* #undef setlocale */
 
 #define _(string) gettext (string)
 
--- origsrc/gettext-0.19.3/gettext-tools/tests/format-c-4-prg.c	2013-05-31 00:01:54.000000000 -0500
+++ src/gettext-0.19.3/gettext-tools/tests/format-c-4-prg.c	2014-11-17 23:26:21.979557600 -0600
@@ -34,7 +34,7 @@
 /* Disable the override of setlocale that libgnuintl.h activates on MacOS X
    and Windows.  This test relies on the fake setlocale function in
    setlocale.c.  */
-#undef setlocale
+/* #undef setlocale */
 
 #define _(string) gettext (string)
 
--- origsrc/gettext-0.19.3/gettext-tools/tests/plural-1	2014-04-14 20:52:11.000000000 -0500
+++ src/gettext-0.19.3/gettext-tools/tests/plural-1	2014-11-17 23:26:21.983058100 -0600
@@ -50,15 +50,15 @@ ${DIFF} plural-1-fr.po.strip plural-1-fr
 
 : ${DIFF=diff}
 echo 'un morceau de gateau' > plural-1-cake.ok
-LANGUAGE= ../cake fr 1 > plural-1-cake.tmp || exit 1
+LANGUAGE= ../cake fr_FR 1 > plural-1-cake.tmp || exit 1
 LC_ALL=C tr -d '\r' < plural-1-cake.tmp > cake.out || exit 1
 ${DIFF} plural-1-cake.ok cake.out || exit 1
 echo '2 morceaux de gateau' > plural-1-cake.ok
-LANGUAGE= ../cake fr 2 > plural-1-cake.tmp || exit 1
+LANGUAGE= ../cake fr_FR 2 > plural-1-cake.tmp || exit 1
 LC_ALL=C tr -d '\r' < plural-1-cake.tmp > cake.out || exit 1
 ${DIFF} plural-1-cake.ok cake.out || exit 1
 echo '10 morceaux de gateau' > plural-1-cake.ok
-LANGUAGE= ../cake fr 10 > plural-1-cake.tmp || exit 1
+LANGUAGE= ../cake fr_FR 10 > plural-1-cake.tmp || exit 1
 LC_ALL=C tr -d '\r' < plural-1-cake.tmp > cake.out || exit 1
 ${DIFF} plural-1-cake.ok cake.out || exit 1
 
--- origsrc/gettext-0.19.3/gettext-tools/tests/plural-1-prg.c	2013-05-31 00:01:54.000000000 -0500
+++ src/gettext-0.19.3/gettext-tools/tests/plural-1-prg.c	2014-11-17 23:26:21.989058800 -0600
@@ -30,7 +30,7 @@
 /* Disable the override of setlocale that libgnuintl.h activates on MacOS X
    and Windows.  This test relies on the fake setlocale function in
    setlocale.c.  */
-#undef setlocale
+/* #undef setlocale */
 
 int
 main (int argc, char *argv[])
--- origsrc/gettext-0.19.3/gettext-tools/tests/plural-2	2014-04-14 20:52:11.000000000 -0500
+++ src/gettext-0.19.3/gettext-tools/tests/plural-2	2014-11-17 23:25:50.637077600 -0600
@@ -3,8 +3,8 @@
 
 : ${MSGFMT=msgfmt}
 test -d plural-2-dir || mkdir plural-2-dir
-test -d plural-2-dir/ll || mkdir plural-2-dir/ll
-test -d plural-2-dir/ll/LC_MESSAGES || mkdir plural-2-dir/ll/LC_MESSAGES
+test -d plural-2-dir/es || mkdir plural-2-dir/es
+test -d plural-2-dir/es/LC_MESSAGES || mkdir plural-2-dir/es/LC_MESSAGES
 
 cat <<EOF > plural-2.data
 ja
@@ -44,7 +44,7 @@ EOF
 while read lang; do
   read formula
   read dataok
-  cat > plural-2-ll.po <<EOF
+  cat > plural-2-es.po <<EOF
 msgid ""
 msgstr ""
 "MIME-Version: 1.0\n"
@@ -65,10 +65,10 @@ msgstr[7] "7"
 msgstr[8] "8"
 msgstr[9] "9"
 EOF
-  ${MSGFMT} -o plural-2-dir/ll/LC_MESSAGES/plural.mo plural-2-ll.po || exit 1
+  ${MSGFMT} -o plural-2-dir/es/LC_MESSAGES/plural.mo plural-2-es.po || exit 1
   (for i in '' 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19; do
      LANGUAGE= TEXTDOMAIN=plural TEXTDOMAINDIR=plural-2-dir \
-       $NGETTEXT --env LC_ALL=ll X Y ${i}0 ${i}1 ${i}2 ${i}3 ${i}4 ${i}5 ${i}6 ${i}7 ${i}8 ${i}9
+       $NGETTEXT --env LC_ALL=es_ES X Y ${i}0 ${i}1 ${i}2 ${i}3 ${i}4 ${i}5 ${i}6 ${i}7 ${i}8 ${i}9
    done) > dataout
   test "$dataok" = `cat dataout` || {
     echo "Formula evaluation error for language $lang" 1>&2
