diff --git origsrc/gettext-0.19.8.1.orig/gettext-tools/misc/autopoint.in src/gettext-0.19.8.1/gettext-tools/misc/autopoint.in
index 6d0746f..5b7c0bb 100755
--- origsrc/gettext-0.19.8.1.orig/gettext-tools/misc/autopoint.in
+++ src/gettext-0.19.8.1/gettext-tools/misc/autopoint.in
@@ -193,9 +193,9 @@ Options:
       --version        print version information and exit
   -f, --force          force overwriting of files that already exist
   -n, --dry-run        print modifications but don't perform them"
-#  echo "\
-#  -V version           copy the infrastructure of the specified gettext version
-#                         (dangerous)"
+  echo "\
+  -V version           copy the infrastructure of the specified gettext version
+                         (dangerous)"
   echo "
 Report bugs to <bug-gnu-gettext@gnu.org>."
 }
@@ -249,17 +249,17 @@ unset CLICOLOR_FORCE GREP_OPTIONS
         force=yes ;;
       --help | --hel | --he | --h )
         func_usage; exit 0 ;;
-#      -V ) # Some people put a space between -V and the version number.
-#        shift
-#        if test $# = 0; then
-#          func_usage 1>&2
-#          exit 1
-#        fi
-#        ver=$1;
-#        shift ;;
-#      -V*) # Some people omit the space between -V and the version number.
-#        ver=`echo "X$1" | sed -e 's/^X-V//'`
-#        shift ;;
+      -V ) # Some people put a space between -V and the version number.
+        shift
+        if test $# = 0; then
+          func_usage 1>&2
+          exit 1
+        fi
+        ver=$1;
+        shift ;;
+      -V*) # Some people omit the space between -V and the version number.
+        ver=`echo "X$1" | sed -e 's/^X-V//'`
+        shift ;;
       --version | --versio | --versi | --vers | --ver | --ve | --v )
         func_version
         exit 0 ;;
@@ -355,7 +355,7 @@ else
   if test -n "$xver"; then
     if test -n "$ver"; then
       if test "X$ver" != "X$xver"; then
-        func_fatal_error "Version mismatch: specified -V $ver but the package uses gettext version $xver"
+        echo "autopoint: warning: Version mismatch: specified -V $ver but the package uses gettext version $xver. Forcibly upgrading to $ver" 1>&2
       fi
     else
       ver="$xver"
