--- origsrc/m4/woe32-dll.m4	2020-10-04 17:58:09.000000000 -0600
+++ src/m4/woe32-dll.m4	2023-06-17 10:04:48.858239300 -0600
@@ -30,7 +30,7 @@ AC_DEFUN([gl_WOE32_DLL],
 [
   AC_REQUIRE([AC_CANONICAL_HOST])
   case "$host_os" in
-    mingw* | cygwin*)
+    mingw*)
       AC_CACHE_CHECK([for auto-import of symbols],
         [gl_cv_ld_autoimport],
         [dnl --disable-auto-import is unsupported in MSVC and in MSVC/clang.
