--- origsrc/gettext-runtime/intl/gnulib-lib/localename.h	2024-02-21 03:43:51.000000000 -0700
+++ src/gettext-runtime/intl/gnulib-lib/localename.h	2024-02-26 07:40:54.664851300 -0700
@@ -91,7 +91,7 @@ extern const char * gl_locale_name_envir
 
    The result must not be freed; it is statically allocated.  */
 extern const char * gl_locale_name_default (void)
-#if !(HAVE_CFPREFERENCESCOPYAPPVALUE || defined _WIN32 || defined __CYGWIN__)
+#if !(HAVE_CFPREFERENCESCOPYAPPVALUE || defined _WIN32)
   _GL_ATTRIBUTE_CONST
 #endif
   ;
