diff --git gettext-0.19.8.1.orig/gettext-runtime/m4/intl.m4 gettext-0.19.8.1/gettext-runtime/m4/intl.m4
index 42fac95..9ab0952 100644
--- gettext-0.19.8.1.orig/gettext-runtime/m4/intl.m4
+++ gettext-0.19.8.1/gettext-runtime/m4/intl.m4
@@ -113,7 +113,7 @@ AC_DEFUN([AM_INTL_SUBDIR],
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -125,7 +125,7 @@ AC_DEFUN([AM_INTL_SUBDIR],
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff --git gettext-0.19.8.1.orig/gettext-runtime/m4/threadlib.m4 gettext-0.19.8.1/gettext-runtime/m4/threadlib.m4
index b43534e..bc7a2a8 100644
--- gettext-0.19.8.1.orig/gettext-runtime/m4/threadlib.m4
+++ gettext-0.19.8.1/gettext-runtime/m4/threadlib.m4
@@ -68,7 +68,7 @@ changequote(,)dnl
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
diff --git gettext-0.19.8.1.orig/build-aux/config.rpath gettext-0.19.8.1/build-aux/config.rpath
index 98183ff..c5d797d 100755
--- gettext-0.19.8.1.orig/build-aux/config.rpath
+++ gettext-0.19.8.1/build-aux/config.rpath
@@ -57,7 +57,7 @@ else
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@ hardcode_direct=no
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@ if test "$with_gnu_ld" = yes; then
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@ else
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@ case "$host_os" in
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;