diff --git gettext-0.19.8.1.orig/build-aux/compile gettext-0.19.8.1/build-aux/compile
index a85b723..3cc6cf3 100755
--- gettext-0.19.8.1.orig/build-aux/compile
+++ gettext-0.19.8.1/build-aux/compile
@@ -53,7 +53,7 @@ func_file_conv ()
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@ func_file_conv ()
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff --git gettext-0.19.8.1.orig/build-aux/config.guess gettext-0.19.8.1/build-aux/config.guess
index c4bd827..f03849d 100755
--- gettext-0.19.8.1.orig/build-aux/config.guess
+++ gettext-0.19.8.1/build-aux/config.guess
@@ -891,6 +891,9 @@ EOF
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff --git gettext-0.19.8.1.orig/build-aux/config.libpath gettext-0.19.8.1/build-aux/config.libpath
index 849fa7b..9a67cdc 100755
--- gettext-0.19.8.1.orig/build-aux/config.libpath
+++ gettext-0.19.8.1/build-aux/config.libpath
@@ -56,7 +56,7 @@ case $host_os in
   bsdi[45]*)
     shlibpath_var=LD_LIBRARY_PATH
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: first we should search . and the directory the executable is in
     shlibpath_var=PATH
     ;;
diff --git gettext-0.19.8.1.orig/build-aux/config.rpath gettext-0.19.8.1/build-aux/config.rpath
index 98183ff..c5d797d 100755
--- gettext-0.19.8.1.orig/build-aux/config.rpath
+++ gettext-0.19.8.1/build-aux/config.rpath
@@ -57,7 +57,7 @@ else
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@ hardcode_direct=no
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@ if test "$with_gnu_ld" = yes; then
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@ else
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@ case "$host_os" in
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff --git gettext-0.19.8.1.orig/gettext-runtime/configure.ac gettext-0.19.8.1/gettext-runtime/configure.ac
index 71adec3..b223f26 100644
--- gettext-0.19.8.1.orig/gettext-runtime/configure.ac
+++ gettext-0.19.8.1/gettext-runtime/configure.ac
@@ -76,7 +76,7 @@ LT_LANG([Windows Resource])
 dnl On mingw and Cygwin, we can activate special Makefile rules which add
 dnl version information to the shared libraries and executables.
 case "$host_os" in
-  mingw* | cygwin*) is_woe32=yes ;;
+  mingw* | cygwin* | msys*) is_woe32=yes ;;
   *) is_woe32=no ;;
 esac
 AM_CONDITIONAL([WOE32], [test $is_woe32 = yes])
@@ -116,7 +116,7 @@ dnl    must arrange to define the auxiliary pointer variables for the
 dnl    exported variables _also_ in the static library.
 if test "$enable_shared" = yes; then
   case "$host_os" in
-    mingw* | cygwin*) is_woe32dll=yes ;;
+    mingw* | cygwin* | msys*) is_woe32dll=yes ;;
     *) is_woe32dll=no ;;
   esac
 else
diff --git gettext-0.19.8.1.orig/gettext-runtime/gnulib-lib/Makefile.am gettext-0.19.8.1/gettext-runtime/gnulib-lib/Makefile.am
index 2fec716..c117381 100644
--- gettext-0.19.8.1.orig/gettext-runtime/gnulib-lib/Makefile.am
+++ gettext-0.19.8.1/gettext-runtime/gnulib-lib/Makefile.am
@@ -478,7 +478,7 @@ install-exec-localcharset: all-local
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff --git gettext-0.19.8.1.orig/gettext-runtime/gnulib-m4/double-slash-root.m4 gettext-0.19.8.1/gettext-runtime/gnulib-m4/double-slash-root.m4
index dfd3921..ab67944 100644
--- gettext-0.19.8.1.orig/gettext-runtime/gnulib-m4/double-slash-root.m4
+++ gettext-0.19.8.1/gettext-runtime/gnulib-m4/double-slash-root.m4
@@ -16,7 +16,7 @@ AC_DEFUN([gl_DOUBLE_SLASH_ROOT],
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff --git gettext-0.19.8.1.orig/gettext-runtime/gnulib-m4/setlocale.m4 gettext-0.19.8.1/gettext-runtime/gnulib-m4/setlocale.m4
index a359069..9fffac3 100644
--- gettext-0.19.8.1.orig/gettext-runtime/gnulib-m4/setlocale.m4
+++ gettext-0.19.8.1/gettext-runtime/gnulib-m4/setlocale.m4
@@ -14,7 +14,7 @@ AC_DEFUN([gl_FUNC_SETLOCALE],
     mingw*) REPLACE_SETLOCALE=1 ;;
     dnl On Cygwin 1.5.x, setlocale always succeeds but setlocale(LC_CTYPE,NULL)
     dnl is then still "C".
-    cygwin*)
+    cygwin* | msys*)
       case `uname -r` in
         1.5.*) REPLACE_SETLOCALE=1 ;;
       esac
diff --git gettext-0.19.8.1.orig/gettext-runtime/libasprintf/configure.ac gettext-0.19.8.1/gettext-runtime/libasprintf/configure.ac
index 736bf48..6264895 100644
--- gettext-0.19.8.1.orig/gettext-runtime/libasprintf/configure.ac
+++ gettext-0.19.8.1/gettext-runtime/libasprintf/configure.ac
@@ -44,7 +44,7 @@ LT_INIT([win32-dll])
 case "$host_os" in
   # On Cygwin, without -no-undefined, a warning is emitted and only a static
   # library is built.
-  beos* | mingw* | cygwin* | os2*) LTNOUNDEF='-no-undefined' ;;
+  beos* | mingw* | cygwin* | msys* | os2*) LTNOUNDEF='-no-undefined' ;;
   *) LTNOUNDEF='' ;;
 esac
 AC_SUBST([LTNOUNDEF])
diff --git gettext-0.19.8.1.orig/gettext-runtime/m4/intl.m4 gettext-0.19.8.1/gettext-runtime/m4/intl.m4
index 42fac95..9ab0952 100644
--- gettext-0.19.8.1.orig/gettext-runtime/m4/intl.m4
+++ gettext-0.19.8.1/gettext-runtime/m4/intl.m4
@@ -113,7 +113,7 @@ AC_DEFUN([AM_INTL_SUBDIR],
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -125,7 +125,7 @@ AC_DEFUN([AM_INTL_SUBDIR],
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff --git gettext-0.19.8.1.orig/gettext-runtime/m4/threadlib.m4 gettext-0.19.8.1/gettext-runtime/m4/threadlib.m4
index b43534e..bc7a2a8 100644
--- gettext-0.19.8.1.orig/gettext-runtime/m4/threadlib.m4
+++ gettext-0.19.8.1/gettext-runtime/m4/threadlib.m4
@@ -68,7 +68,7 @@ changequote(,)dnl
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
diff --git gettext-0.19.8.1.orig/gettext-tools/configure.ac gettext-0.19.8.1/gettext-tools/configure.ac
index 3b132ed..a4fa470 100644
--- gettext-0.19.8.1.orig/gettext-tools/configure.ac
+++ gettext-0.19.8.1/gettext-tools/configure.ac
@@ -111,7 +111,7 @@ dnl On Cygwin 1.7.10-1 or later, error_* variables are defined in the
 dnl Cygwin DLL.  To avoid any conflict with gettext DLLs, always link
 dnl to gnulib's error.c.  See GETTEXTLIB_EXPORTS_FLAGS setting below.
 case "$host_os" in
-  cygwin*) ac_cv_lib_error_at_line=no ;;
+  cygwin* | msys*) ac_cv_lib_error_at_line=no ;;
 esac
 
 dnl Checks for libraries.
diff --git gettext-0.19.8.1.orig/gettext-tools/examples/hello-c++-kde/admin/config.guess gettext-0.19.8.1/gettext-tools/examples/hello-c++-kde/admin/config.guess
index 58dc0e2..ea86d52 100755
--- gettext-0.19.8.1.orig/gettext-tools/examples/hello-c++-kde/admin/config.guess
+++ gettext-0.19.8.1/gettext-tools/examples/hello-c++-kde/admin/config.guess
@@ -761,6 +761,9 @@ EOF
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit 0 ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit 0 ;;
     i*:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit 0 ;;
diff --git gettext-0.19.8.1.orig/gettext-tools/examples/hello-c++-kde/admin/config.sub gettext-0.19.8.1/gettext-tools/examples/hello-c++-kde/admin/config.sub
index 2684853..cf78800 100755
--- gettext-0.19.8.1.orig/gettext-tools/examples/hello-c++-kde/admin/config.sub
+++ gettext-0.19.8.1/gettext-tools/examples/hello-c++-kde/admin/config.sub
@@ -1110,7 +1110,7 @@ case $os in
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* \
-	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
+	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-lib/Makefile.gnulib gettext-0.19.8.1/gettext-tools/gnulib-lib/Makefile.gnulib
index 438bf9d..693bd00 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-lib/Makefile.gnulib
+++ gettext-0.19.8.1/gettext-tools/gnulib-lib/Makefile.gnulib
@@ -1810,7 +1810,7 @@ install-exec-localcharset: all-local
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/btowc.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/btowc.m4
index 47e8fd8..9e1dbf7 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/btowc.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/btowc.m4
@@ -49,7 +49,7 @@ int main ()
 changequote(,)dnl
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/csharpexec.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/csharpexec.m4
index 39aceef..e5517b3 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/csharpexec.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/csharpexec.m4
@@ -53,7 +53,7 @@ AC_DEFUN([gt_CSHARPEXEC],
            ifelse([$1], , , [&& clix $2/$1 >/dev/null 2>/dev/null]); then
           HAVE_CLIX=1
           case $host_os in
-            cygwin* | mingw* | pw32*)
+            cygwin* | msys* | mingw* | pw32*)
               CLIX_PATH_VAR=PATH
               ;;
             darwin* | rhapsody*)
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/double-slash-root.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/double-slash-root.m4
index dfd3921..ab67944 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/double-slash-root.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/double-slash-root.m4
@@ -16,7 +16,7 @@ AC_DEFUN([gl_DOUBLE_SLASH_ROOT],
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/dup2.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/dup2.m4
index 5b68312..0ed448e 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/dup2.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/dup2.m4
@@ -78,7 +78,7 @@ AC_DEFUN([gl_FUNC_DUP2],
         [case "$host_os" in
            mingw*) # on this platform, dup2 always returns 0 for success
              gl_cv_func_dup2_works="guessing no" ;;
-           cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+           cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
            aix* | freebsd*)
                    # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/getcwd.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/getcwd.m4
index 566c3c1..e7cefcb 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/getcwd.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/getcwd.m4
@@ -51,7 +51,7 @@ AC_DEFUN([gl_FUNC_GETCWD_NULL],
                      # Guess yes on glibc systems.
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
-            cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+            cygwin* | msys*) gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/malloc.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/malloc.m4
index c393690..675278e 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/malloc.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/malloc.m4
@@ -33,7 +33,7 @@ AC_DEFUN([_AC_FUNC_MALLOC_IF],
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/printf.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/printf.m4
index e495e0c..01be733 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/printf.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/printf.m4
@@ -720,7 +720,7 @@ changequote(,)dnl
            openbsd*)        gl_cv_func_printf_directive_ls="guessing no";;
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
-           cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           cygwin* | msys*)         gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -797,7 +797,7 @@ int main ()
         [
 changequote(,)dnl
          case "$host_os" in
-           cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           cygwin* | msys*)      gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -1449,7 +1449,7 @@ changequote(,)dnl
            darwin[1-6].*)        gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
-           cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           cygwin* | msys*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/realloc.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/realloc.m4
index fc22825..ce1b2fa 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/realloc.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/realloc.m4
@@ -33,7 +33,7 @@ AC_DEFUN([_AC_FUNC_REALLOC_IF],
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/setlocale.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/setlocale.m4
index a359069..9fffac3 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/setlocale.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/setlocale.m4
@@ -14,7 +14,7 @@ AC_DEFUN([gl_FUNC_SETLOCALE],
     mingw*) REPLACE_SETLOCALE=1 ;;
     dnl On Cygwin 1.5.x, setlocale always succeeds but setlocale(LC_CTYPE,NULL)
     dnl is then still "C".
-    cygwin*)
+    cygwin* | msys*)
       case `uname -r` in
         1.5.*) REPLACE_SETLOCALE=1 ;;
       esac
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/wctob.m4 gettext-0.19.8.1/gettext-tools/gnulib-m4/wctob.m4
index bdb5299..7a70017 100644
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-m4/wctob.m4
+++ gettext-0.19.8.1/gettext-tools/gnulib-m4/wctob.m4
@@ -29,14 +29,14 @@ AC_DEFUN([gl_FUNC_WCTOB],
 changequote(,)dnl
         case "$host_os" in
             # Guess no on Solaris <= 9 and Cygwin.
-          solaris2.[1-9] | solaris2.[1-9].* | cygwin*)
+          solaris2.[1-9] | solaris2.[1-9].* | cygwin* | msys*)
             gl_cv_func_wctob_works="guessing no" ;;
             # Guess yes otherwise.
           *) gl_cv_func_wctob_works="guessing yes" ;;
         esac
 changequote([,])dnl
         case "$host_os" in
-          cygwin*)
+          cygwin* | msys*)
             AC_RUN_IFELSE(
               [AC_LANG_SOURCE([[
 #include <locale.h>
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-tests/test-copy-acl.sh gettext-0.19.8.1/gettext-tools/gnulib-tests/test-copy-acl.sh
index 147bf56..351b121 100755
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-tests/test-copy-acl.sh
+++ gettext-0.19.8.1/gettext-tools/gnulib-tests/test-copy-acl.sh
@@ -128,7 +128,7 @@ cd "$builddir" ||
   # the programs.
   # func_test_same_acls file1 file2
   case $acl_flavor in
-    linux | cygwin | freebsd | solaris)
+    linux | cygwin | msys | freebsd | solaris)
       func_test_same_acls ()
       {
         getfacl "$1" | sed -e "s/$1/FILENAME/g" > tmpaclout1
@@ -302,7 +302,7 @@ cd "$builddir" ||
 
         ;;
 
-      cygwin)
+      cygwin | msys)
 
         # Set an ACL for a group.
         setfacl -m group:0:1 tmpfile0
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-tests/test-copy-file.sh gettext-0.19.8.1/gettext-tools/gnulib-tests/test-copy-file.sh
index 98da280..db955b6 100755
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-tests/test-copy-file.sh
+++ gettext-0.19.8.1/gettext-tools/gnulib-tests/test-copy-file.sh
@@ -122,7 +122,7 @@ cd "$builddir" ||
   # the programs.
   # func_test_same_acls file1 file2
   case $acl_flavor in
-    linux | cygwin | freebsd | solaris)
+    linux | cygwin | msys | freebsd | solaris)
       func_test_same_acls ()
       {
         getfacl "$1" | sed -e "s/$1/FILENAME/g" > tmpaclout1
@@ -296,7 +296,7 @@ cd "$builddir" ||
 
         ;;
 
-      cygwin)
+      cygwin | msys)
 
         # Set an ACL for a group.
         setfacl -m group:0:1 tmpfile0
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-tests/test-file-has-acl.sh gettext-0.19.8.1/gettext-tools/gnulib-tests/test-file-has-acl.sh
index 26610c1..e160de1 100755
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-tests/test-file-has-acl.sh
+++ gettext-0.19.8.1/gettext-tools/gnulib-tests/test-file-has-acl.sh
@@ -247,7 +247,7 @@ cd "$builddir" ||
         fi
         ;;
 
-      cygwin)
+      cygwin | msys)
 
         # Set an ACL for a group.
         if setfacl -m group:0:1 tmpfile0; then
diff --git gettext-0.19.8.1.orig/gettext-tools/gnulib-tests/test-set-mode-acl.sh gettext-0.19.8.1/gettext-tools/gnulib-tests/test-set-mode-acl.sh
index f0b1641..053adfe 100755
--- gettext-0.19.8.1.orig/gettext-tools/gnulib-tests/test-set-mode-acl.sh
+++ gettext-0.19.8.1/gettext-tools/gnulib-tests/test-set-mode-acl.sh
@@ -177,7 +177,7 @@ cd "$builddir" ||
           linux | freebsd | solaris)
             setfacl -m user:$auid:1 tmpfile0
             ;;
-          cygwin)
+          cygwin | msys)
             setfacl -m group:0:1 tmpfile0
             ;;
           hpux)
diff --git gettext-0.19.8.1.orig/gettext-tools/libgettextpo/Makefile.gnulib gettext-0.19.8.1/gettext-tools/libgettextpo/Makefile.gnulib
index c68e6f5..b36e9cc 100644
--- gettext-0.19.8.1.orig/gettext-tools/libgettextpo/Makefile.gnulib
+++ gettext-0.19.8.1/gettext-tools/libgettextpo/Makefile.gnulib
@@ -561,7 +561,7 @@ install-exec-localcharset: all-local
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
