diff -Naur archive-orig/gettext-0.11/config.rpath archive/gettext-0.11/config.rpath
--- archive-orig/gettext-0.11/config.rpath	2002-05-01 21:56:22.000000000 +0400
+++ archive/gettext-0.11/config.rpath	2015-03-03 11:43:26.420800000 +0300
@@ -95,7 +95,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* )
+  cygwin* | msys* | mingw* | pw32* )
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -130,7 +130,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -220,7 +220,7 @@
       # see comment about different semantics on the GNU ld section
       ld_shlibs=no
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -354,9 +354,9 @@
     sys_lib_search_path_spec="/shlib /usr/lib /usr/X11/lib /usr/contrib/lib /lib /usr/local/lib"
     sys_lib_dlsearch_path_spec="/shlib /usr/lib /usr/local/lib"
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     case $GCC,$host_os in
-      yes,cygwin*)
+      yes,cygwin* | yes,msys*)
         shlibext=dll.a
         ;;
       yes,mingw*)
diff -Naur archive-orig/gettext-0.11.1/config.rpath archive/gettext-0.11.1/config.rpath
--- archive-orig/gettext-0.11.1/config.rpath	2002-05-01 22:02:35.000000000 +0400
+++ archive/gettext-0.11.1/config.rpath	2015-03-03 11:43:59.602000000 +0300
@@ -93,7 +93,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -131,7 +131,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -225,7 +225,7 @@
       # see comment about different semantics on the GNU ld section
       ld_shlibs=no
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -370,9 +370,9 @@
     sys_lib_search_path_spec="/shlib /usr/lib /usr/X11/lib /usr/contrib/lib /lib /usr/local/lib"
     sys_lib_dlsearch_path_spec="/shlib /usr/lib /usr/local/lib"
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     case $GCC,$host_os in
-      yes,cygwin*)
+      yes,cygwin* | yes,msys*)
         shlibext=dll.a
         ;;
       yes,mingw*)
diff -Naur archive-orig/gettext-0.11.2/config.rpath archive/gettext-0.11.2/config.rpath
--- archive-orig/gettext-0.11.2/config.rpath	2002-05-01 22:02:35.000000000 +0400
+++ archive/gettext-0.11.2/config.rpath	2015-03-03 11:44:29.179600000 +0300
@@ -93,7 +93,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -131,7 +131,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -225,7 +225,7 @@
       # see comment about different semantics on the GNU ld section
       ld_shlibs=no
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -370,9 +370,9 @@
     sys_lib_search_path_spec="/shlib /usr/lib /usr/X11/lib /usr/contrib/lib /lib /usr/local/lib"
     sys_lib_dlsearch_path_spec="/shlib /usr/lib /usr/local/lib"
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     case $GCC,$host_os in
-      yes,cygwin*)
+      yes,cygwin* | yes,msys*)
         shlibext=dll.a
         ;;
       yes,mingw*)
diff -Naur archive-orig/gettext-0.11.3/config.rpath archive/gettext-0.11.3/config.rpath
--- archive-orig/gettext-0.11.3/config.rpath	2002-05-01 22:02:35.000000000 +0400
+++ archive/gettext-0.11.3/config.rpath	2015-03-03 11:45:11.299600000 +0300
@@ -93,7 +93,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -131,7 +131,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -225,7 +225,7 @@
       # see comment about different semantics on the GNU ld section
       ld_shlibs=no
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -370,9 +370,9 @@
     sys_lib_search_path_spec="/shlib /usr/lib /usr/X11/lib /usr/contrib/lib /lib /usr/local/lib"
     sys_lib_dlsearch_path_spec="/shlib /usr/lib /usr/local/lib"
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     case $GCC,$host_os in
-      yes,cygwin*)
+      yes,cygwin* | yes,msys*)
         shlibext=dll.a
         ;;
       yes,mingw*)
diff -Naur archive-orig/gettext-0.11.4/config.rpath archive/gettext-0.11.4/config.rpath
--- archive-orig/gettext-0.11.4/config.rpath	2002-05-01 22:02:35.000000000 +0400
+++ archive/gettext-0.11.4/config.rpath	2015-03-03 11:45:37.694800000 +0300
@@ -93,7 +93,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -131,7 +131,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -225,7 +225,7 @@
       # see comment about different semantics on the GNU ld section
       ld_shlibs=no
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -370,9 +370,9 @@
     sys_lib_search_path_spec="/shlib /usr/lib /usr/X11/lib /usr/contrib/lib /lib /usr/local/lib"
     sys_lib_dlsearch_path_spec="/shlib /usr/lib /usr/local/lib"
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     case $GCC,$host_os in
-      yes,cygwin*)
+      yes,cygwin* | yes,msys*)
         shlibext=dll.a
         ;;
       yes,mingw*)
diff -Naur archive-orig/gettext-0.11.5/config.rpath archive/gettext-0.11.5/config.rpath
--- archive-orig/gettext-0.11.5/config.rpath	2002-05-01 22:02:35.000000000 +0400
+++ archive/gettext-0.11.5/config.rpath	2015-03-03 11:46:03.809200000 +0300
@@ -93,7 +93,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -131,7 +131,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -225,7 +225,7 @@
       # see comment about different semantics on the GNU ld section
       ld_shlibs=no
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -370,9 +370,9 @@
     sys_lib_search_path_spec="/shlib /usr/lib /usr/X11/lib /usr/contrib/lib /lib /usr/local/lib"
     sys_lib_dlsearch_path_spec="/shlib /usr/lib /usr/local/lib"
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     case $GCC,$host_os in
-      yes,cygwin*)
+      yes,cygwin* | yes,msys*)
         shlibext=dll.a
         ;;
       yes,mingw*)
diff -Naur archive-orig/gettext-0.12/config.rpath archive/gettext-0.12/config.rpath
--- archive-orig/gettext-0.12/config.rpath	2003-05-17 19:34:29.000000000 +0400
+++ archive/gettext-0.12/config.rpath	2015-03-03 11:46:52.403200000 +0300
@@ -109,7 +109,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -149,7 +149,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -268,7 +268,7 @@
       ;;
     bsdi4*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -440,7 +440,7 @@
     ;;
   bsdi4*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.12.1/config.rpath archive/gettext-0.12.1/config.rpath
--- archive-orig/gettext-0.12.1/config.rpath	2003-05-17 19:34:29.000000000 +0400
+++ archive/gettext-0.12.1/config.rpath	2015-03-03 11:47:15.194800000 +0300
@@ -109,7 +109,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -149,7 +149,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -268,7 +268,7 @@
       ;;
     bsdi4*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -440,7 +440,7 @@
     ;;
   bsdi4*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.13/config.rpath archive/gettext-0.13/config.rpath
--- archive-orig/gettext-0.13/config.rpath	2003-11-30 17:31:21.000000000 +0300
+++ archive/gettext-0.13/config.rpath	2015-03-03 11:47:39.000400000 +0300
@@ -109,7 +109,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -149,7 +149,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -268,7 +268,7 @@
       ;;
     bsdi4*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -440,7 +440,7 @@
     ;;
   bsdi4*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.13.1/config.rpath archive/gettext-0.13.1/config.rpath
--- archive-orig/gettext-0.13.1/config.rpath	2003-11-30 17:31:21.000000000 +0300
+++ archive/gettext-0.13.1/config.rpath	2015-03-03 11:48:04.116400000 +0300
@@ -109,7 +109,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -149,7 +149,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -268,7 +268,7 @@
       ;;
     bsdi4*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -440,7 +440,7 @@
     ;;
   bsdi4*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.14/config.rpath archive/gettext-0.14/config.rpath
--- archive-orig/gettext-0.14/config.rpath	2003-11-30 17:31:21.000000000 +0300
+++ archive/gettext-0.14/config.rpath	2015-03-03 11:48:33.694000000 +0300
@@ -109,7 +109,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -149,7 +149,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -268,7 +268,7 @@
       ;;
     bsdi4*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -440,7 +440,7 @@
     ;;
   bsdi4*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.14.1/config.rpath archive/gettext-0.14.1/config.rpath
--- archive-orig/gettext-0.14.1/config.rpath	2003-11-30 17:31:21.000000000 +0300
+++ archive/gettext-0.14.1/config.rpath	2015-03-03 11:48:54.863200000 +0300
@@ -109,7 +109,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -149,7 +149,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -268,7 +268,7 @@
       ;;
     bsdi4*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -440,7 +440,7 @@
     ;;
   bsdi4*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.14.2/config.rpath archive/gettext-0.14.2/config.rpath
--- archive-orig/gettext-0.14.2/config.rpath	2005-02-24 17:21:47.000000000 +0300
+++ archive/gettext-0.14.2/config.rpath	2015-03-03 11:49:18.918400000 +0300
@@ -112,7 +112,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -152,7 +152,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -278,7 +278,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -459,7 +459,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.14.3/config.rpath archive/gettext-0.14.3/config.rpath
--- archive-orig/gettext-0.14.3/config.rpath	2005-02-24 17:21:47.000000000 +0300
+++ archive/gettext-0.14.3/config.rpath	2015-03-03 11:49:58.277200000 +0300
@@ -112,7 +112,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -152,7 +152,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -278,7 +278,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -459,7 +459,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.14.4/config.rpath archive/gettext-0.14.4/config.rpath
--- archive-orig/gettext-0.14.4/config.rpath	2005-02-24 17:21:47.000000000 +0300
+++ archive/gettext-0.14.4/config.rpath	2015-03-03 11:50:31.302400000 +0300
@@ -112,7 +112,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -152,7 +152,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -278,7 +278,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -459,7 +459,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.14.5/config.rpath archive/gettext-0.14.5/config.rpath
--- archive-orig/gettext-0.14.5/config.rpath	2005-02-24 17:21:47.000000000 +0300
+++ archive/gettext-0.14.5/config.rpath	2015-03-03 11:50:59.070400000 +0300
@@ -112,7 +112,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -152,7 +152,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -278,7 +278,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -459,7 +459,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.14.6/config.rpath archive/gettext-0.14.6/config.rpath
--- archive-orig/gettext-0.14.6/config.rpath	2005-02-24 17:21:47.000000000 +0300
+++ archive/gettext-0.14.6/config.rpath	2015-03-03 11:51:25.496800000 +0300
@@ -112,7 +112,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -152,7 +152,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -278,7 +278,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -459,7 +459,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.15/config.rpath archive/gettext-0.15/config.rpath
--- archive-orig/gettext-0.15/config.rpath	2006-07-21 17:36:21.000000000 +0400
+++ archive/gettext-0.15/config.rpath	2015-03-03 11:51:55.386400000 +0300
@@ -129,7 +129,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -179,7 +179,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -323,7 +323,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -500,7 +500,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.16/config.rpath archive/gettext-0.16/config.rpath
--- archive-orig/gettext-0.16/config.rpath	2006-07-21 17:36:21.000000000 +0400
+++ archive/gettext-0.16/config.rpath	2015-03-03 11:52:25.556800000 +0300
@@ -129,7 +129,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -179,7 +179,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -323,7 +323,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -500,7 +500,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.16/m4/intl.m4 archive/gettext-0.16/m4/intl.m4
--- archive-orig/gettext-0.16/m4/intl.m4	2006-10-26 17:24:50.000000000 +0400
+++ archive/gettext-0.16/m4/intl.m4	2015-03-03 11:52:59.939200000 +0300
@@ -98,7 +98,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      cygwin*) is_woe32dll=yes ;;
+      cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
diff -Naur archive-orig/gettext-0.16.1/config.rpath archive/gettext-0.16.1/config.rpath
--- archive-orig/gettext-0.16.1/config.rpath	2006-07-21 17:36:21.000000000 +0400
+++ archive/gettext-0.16.1/config.rpath	2015-03-03 11:53:30.749200000 +0300
@@ -129,7 +129,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -179,7 +179,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -323,7 +323,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -500,7 +500,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur archive-orig/gettext-0.16.1/m4/intl.m4 archive/gettext-0.16.1/m4/intl.m4
--- archive-orig/gettext-0.16.1/m4/intl.m4	2006-10-26 17:24:50.000000000 +0400
+++ archive/gettext-0.16.1/m4/intl.m4	2015-03-03 11:53:44.414800000 +0300
@@ -98,7 +98,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      cygwin*) is_woe32dll=yes ;;
+      cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
diff -Naur archive-orig/gettext-0.17/config.rpath archive/gettext-0.17/config.rpath
--- archive-orig/gettext-0.17/config.rpath	2007-11-07 05:50:28.000000000 +0300
+++ archive/gettext-0.17/config.rpath	2015-03-03 11:56:35.422000000 +0300
@@ -64,7 +64,7 @@
           ;;
       esac
       ;;
-    mingw* | cygwin* | pw32* | os2*)
+    mingw* | cygwin* | msys* | pw32* | os2*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -132,7 +132,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -182,7 +182,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -326,7 +326,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -517,7 +517,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.17/m4/intl.m4 archive/gettext-0.17/m4/intl.m4
--- archive-orig/gettext-0.17/m4/intl.m4	2007-11-07 05:50:28.000000000 +0300
+++ archive/gettext-0.17/m4/intl.m4	2015-03-03 11:56:57.683200000 +0300
@@ -97,7 +97,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -109,7 +109,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.18/config.rpath archive/gettext-0.18/config.rpath
--- archive-orig/gettext-0.18/config.rpath	2010-05-09 23:42:50.000000000 +0400
+++ archive/gettext-0.18/config.rpath	2015-03-03 11:57:25.825600000 +0300
@@ -64,7 +64,7 @@
           ;;
       esac
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -138,7 +138,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -188,7 +188,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -332,7 +332,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -523,7 +523,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.18/intl/Makefile.in archive/gettext-0.18/intl/Makefile.in
--- archive-orig/gettext-0.18/intl/Makefile.in	2010-05-09 23:42:55.000000000 +0400
+++ archive/gettext-0.18/intl/Makefile.in	2015-03-03 11:57:47.150800000 +0300
@@ -414,7 +414,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -500,7 +500,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.18/m4/intl.m4 archive/gettext-0.18/m4/intl.m4
--- archive-orig/gettext-0.18/m4/intl.m4	2010-05-09 23:42:58.000000000 +0400
+++ archive/gettext-0.18/m4/intl.m4	2015-03-03 11:58:10.550800000 +0300
@@ -113,7 +113,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -125,7 +125,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.18.1/config.rpath archive/gettext-0.18.1/config.rpath
--- archive-orig/gettext-0.18.1/config.rpath	2010-06-04 04:07:46.000000000 +0400
+++ archive/gettext-0.18.1/config.rpath	2015-03-03 12:00:46.722400000 +0300
@@ -64,7 +64,7 @@
           ;;
       esac
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -138,7 +138,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -188,7 +188,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -332,7 +332,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -523,7 +523,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.18.1/intl/Makefile.in archive/gettext-0.18.1/intl/Makefile.in
--- archive-orig/gettext-0.18.1/intl/Makefile.in	2010-06-04 04:07:28.000000000 +0400
+++ archive/gettext-0.18.1/intl/Makefile.in	2015-03-03 12:01:14.864800000 +0300
@@ -414,7 +414,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -500,7 +500,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.18.1/m4/intl.m4 archive/gettext-0.18.1/m4/intl.m4
--- archive-orig/gettext-0.18.1/m4/intl.m4	2010-06-04 04:07:30.000000000 +0400
+++ archive/gettext-0.18.1/m4/intl.m4	2015-03-03 12:01:29.138800000 +0300
@@ -113,7 +113,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -125,7 +125,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.18.2/config.rpath archive/gettext-0.18.2/config.rpath
--- archive-orig/gettext-0.18.2/config.rpath	2013-03-05 13:08:30.000000000 +0400
+++ archive/gettext-0.18.2/config.rpath	2015-03-03 12:01:54.348400000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.18.2/intl/Makefile.in archive/gettext-0.18.2/intl/Makefile.in
--- archive-orig/gettext-0.18.2/intl/Makefile.in	2013-03-05 13:08:24.000000000 +0400
+++ archive/gettext-0.18.2/intl/Makefile.in	2015-03-03 12:02:54.205600000 +0300
@@ -418,7 +418,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -504,7 +504,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.18.2/m4/intl.m4 archive/gettext-0.18.2/m4/intl.m4
--- archive-orig/gettext-0.18.2/m4/intl.m4	2013-03-05 13:08:25.000000000 +0400
+++ archive/gettext-0.18.2/m4/intl.m4	2015-03-03 12:03:09.244000000 +0300
@@ -113,7 +113,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -125,7 +125,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.18.3/config.rpath archive/gettext-0.18.3/config.rpath
--- archive-orig/gettext-0.18.3/config.rpath	2014-01-07 10:32:23.000000000 +0400
+++ archive/gettext-0.18.3/config.rpath	2015-03-03 12:04:13.375600000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.18.3/intl/Makefile.in archive/gettext-0.18.3/intl/Makefile.in
--- archive-orig/gettext-0.18.3/intl/Makefile.in	2014-01-07 10:32:14.000000000 +0400
+++ archive/gettext-0.18.3/intl/Makefile.in	2015-03-03 12:03:44.687200000 +0300
@@ -418,7 +418,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -504,7 +504,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.18.3/m4/intl.m4 archive/gettext-0.18.3/m4/intl.m4
--- archive-orig/gettext-0.18.3/m4/intl.m4	2014-01-07 10:32:15.000000000 +0400
+++ archive/gettext-0.18.3/m4/intl.m4	2015-03-03 12:06:42.714400000 +0300
@@ -112,7 +112,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -124,7 +124,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.19/config.rpath archive/gettext-0.19/config.rpath
--- archive-orig/gettext-0.19/config.rpath	2014-06-02 03:26:46.000000000 +0400
+++ archive/gettext-0.19/config.rpath	2015-03-03 12:07:10.950400000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.19/intl/Makefile.in archive/gettext-0.19/intl/Makefile.in
--- archive-orig/gettext-0.19/intl/Makefile.in	2014-06-02 03:26:33.000000000 +0400
+++ archive/gettext-0.19/intl/Makefile.in	2015-03-03 12:07:33.164800000 +0300
@@ -418,7 +418,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -504,7 +504,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.19/m4/intl.m4 archive/gettext-0.19/m4/intl.m4
--- archive-orig/gettext-0.19/m4/intl.m4	2014-06-02 03:26:35.000000000 +0400
+++ archive/gettext-0.19/m4/intl.m4	2015-03-03 12:07:51.198400000 +0300
@@ -112,7 +112,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -124,7 +124,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.19.1/config.rpath archive/gettext-0.19.1/config.rpath
--- archive-orig/gettext-0.19.1/config.rpath	2014-06-10 09:55:08.000000000 +0400
+++ archive/gettext-0.19.1/config.rpath	2015-03-03 12:08:33.458800000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.19.1/intl/Makefile.in archive/gettext-0.19.1/intl/Makefile.in
--- archive-orig/gettext-0.19.1/intl/Makefile.in	2014-06-10 09:55:00.000000000 +0400
+++ archive/gettext-0.19.1/intl/Makefile.in	2015-03-03 12:08:49.916800000 +0300
@@ -418,7 +418,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -504,7 +504,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.19.1/m4/intl.m4 archive/gettext-0.19.1/m4/intl.m4
--- archive-orig/gettext-0.19.1/m4/intl.m4	2014-06-10 09:55:01.000000000 +0400
+++ archive/gettext-0.19.1/m4/intl.m4	2015-03-03 12:09:41.693200000 +0300
@@ -112,7 +112,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -124,7 +124,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.19.2/config.rpath archive/gettext-0.19.2/config.rpath
--- archive-orig/gettext-0.19.2/config.rpath	2014-07-14 11:51:32.000000000 +0400
+++ archive/gettext-0.19.2/config.rpath	2015-03-03 12:10:08.010400000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.19.2/intl/Makefile.in archive/gettext-0.19.2/intl/Makefile.in
--- archive-orig/gettext-0.19.2/intl/Makefile.in	2014-07-14 11:51:23.000000000 +0400
+++ archive/gettext-0.19.2/intl/Makefile.in	2015-03-03 12:11:17.711200000 +0300
@@ -418,7 +418,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -503,7 +503,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.19.2/m4/intl.m4 archive/gettext-0.19.2/m4/intl.m4
--- archive-orig/gettext-0.19.2/m4/intl.m4	2014-07-14 11:51:24.000000000 +0400
+++ archive/gettext-0.19.2/m4/intl.m4	2015-03-03 12:11:35.448400000 +0300
@@ -112,7 +112,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin | msys* is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -124,7 +124,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.19.3/config.rpath archive/gettext-0.19.3/config.rpath
--- archive-orig/gettext-0.19.3/config.rpath	2014-10-16 10:09:16.000000000 +0400
+++ archive/gettext-0.19.3/config.rpath	2015-03-03 12:12:10.720000000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.19.3/intl/Makefile.in archive/gettext-0.19.3/intl/Makefile.in
--- archive-orig/gettext-0.19.3/intl/Makefile.in	2014-10-16 10:09:07.000000000 +0400
+++ archive/gettext-0.19.3/intl/Makefile.in	2015-03-03 12:12:30.064000000 +0300
@@ -418,7 +418,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -503,7 +503,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.19.3/m4/intl.m4 archive/gettext-0.19.3/m4/intl.m4
--- archive-orig/gettext-0.19.3/m4/intl.m4	2014-10-16 10:09:09.000000000 +0400
+++ archive/gettext-0.19.3/m4/intl.m4	2015-03-03 12:12:52.980400000 +0300
@@ -113,7 +113,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -125,7 +125,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur archive-orig/gettext-0.19.4/config.rpath archive/gettext-0.19.4/config.rpath
--- archive-orig/gettext-0.19.4/config.rpath	2014-12-24 10:34:53.000000000 +0300
+++ archive/gettext-0.19.4/config.rpath	2015-03-03 12:13:20.405200000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur archive-orig/gettext-0.19.4/intl/Makefile.in archive/gettext-0.19.4/intl/Makefile.in
--- archive-orig/gettext-0.19.4/intl/Makefile.in	2014-12-24 10:34:43.000000000 +0300
+++ archive/gettext-0.19.4/intl/Makefile.in	2015-03-03 12:13:53.695600000 +0300
@@ -429,7 +429,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -514,7 +514,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur archive-orig/gettext-0.19.4/m4/intl.m4 archive/gettext-0.19.4/m4/intl.m4
--- archive-orig/gettext-0.19.4/m4/intl.m4	2014-12-24 10:34:46.000000000 +0300
+++ archive/gettext-0.19.4/m4/intl.m4	2015-03-03 12:14:11.432800000 +0300
@@ -113,7 +113,7 @@
   dnl    exported variables _also_ in the static library.
   if test "$enable_shared" = yes; then
     case "$host_os" in
-      mingw* | cygwin*) is_woe32dll=yes ;;
+      mingw* | cygwin* | msys*) is_woe32dll=yes ;;
       *) is_woe32dll=no ;;
     esac
   else
@@ -125,7 +125,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
