--- origsrc/gettext-tools/configure.ac	2023-06-17 04:58:50.000000000 -0600
+++ src/gettext-tools/configure.ac	2023-06-17 10:01:43.448648800 -0600
@@ -103,10 +103,10 @@ dnl Prepares the libtool configuration f
 dnl sets the RC variable to a program that compiles Windows resource files.
 LT_LANG([Windows Resource])
 
-dnl On mingw and Cygwin, we can activate special Makefile rules which add
+dnl On mingw, we can activate special Makefile rules which add
 dnl version information to the shared libraries and executables.
 case "$host_os" in
-  mingw* | cygwin*) is_woe32=yes ;;
+  mingw*) is_woe32=yes ;;
   *) is_woe32=no ;;
 esac
 AM_CONDITIONAL([WOE32], [test $is_woe32 = yes])
@@ -263,7 +263,7 @@ if test "$gl_cv_libxml_use_included" = y
     [Define if libxml is being used as a static library.])
 fi
 
-dnl Compilation on mingw and Cygwin needs special Makefile rules, because
+dnl Compilation on mingw needs special Makefile rules, because
 dnl 1. when we install a shared library, we must arrange to export
 dnl    auxiliary pointer variables for every exported variable,
 dnl 2. when we install a shared library and a static library simultaneously,
@@ -272,7 +272,7 @@ dnl    must arrange to define the auxili
 dnl    exported variables _also_ in the static library.
 if test "$enable_shared" = yes; then
   case "$host_os" in
-    mingw* | cygwin*) is_woe32dll=yes ;;
+    mingw*) is_woe32dll=yes ;;
     *) is_woe32dll=no ;;
   esac
 else
@@ -281,7 +281,7 @@ fi
 AM_CONDITIONAL([WOE32DLL], [test $is_woe32dll = yes])
 if test $is_woe32dll = yes; then
   AC_DEFINE([WOE32DLL], [1],
-    [Define when --enable-shared is used on mingw or Cygwin.])
+    [Define when --enable-shared is used on mingw.])
 fi
 
 GETTEXTLIB_EXPORTS_FLAGS=
