--- origsrc/gettext-tools/tests/plural-2	2019-05-11 05:29:00.000000000 -0600
+++ src/gettext-tools/tests/plural-2	2021-07-26 23:26:38.655478800 -0600
@@ -3,8 +3,8 @@
 
 : ${MSGFMT=msgfmt}
 test -d plural-2-dir || mkdir plural-2-dir
-test -d plural-2-dir/ll || mkdir plural-2-dir/ll
-test -d plural-2-dir/ll/LC_MESSAGES || mkdir plural-2-dir/ll/LC_MESSAGES
+test -d plural-2-dir/es || mkdir plural-2-dir/es
+test -d plural-2-dir/es/LC_MESSAGES || mkdir plural-2-dir/es/LC_MESSAGES
 
 cat <<EOF > plural-2.data
 ja
@@ -44,7 +44,7 @@ EOF
 while read lang; do
   read formula
   read dataok
-  cat > plural-2-ll.po <<EOF
+  cat > plural-2-es.po <<EOF
 msgid ""
 msgstr ""
 "MIME-Version: 1.0\n"
@@ -65,10 +65,10 @@ msgstr[7] "7"
 msgstr[8] "8"
 msgstr[9] "9"
 EOF
-  ${MSGFMT} -o plural-2-dir/ll/LC_MESSAGES/plural.mo plural-2-ll.po || Exit 1
+  ${MSGFMT} -o plural-2-dir/es/LC_MESSAGES/plural.mo plural-2-es.po || Exit 1
   (for i in '' 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19; do
      LANGUAGE= TEXTDOMAIN=plural TEXTDOMAINDIR=plural-2-dir \
-       $NGETTEXT --env LC_ALL=ll X Y ${i}0 ${i}1 ${i}2 ${i}3 ${i}4 ${i}5 ${i}6 ${i}7 ${i}8 ${i}9
+       $NGETTEXT --env LC_ALL=es_ES X Y ${i}0 ${i}1 ${i}2 ${i}3 ${i}4 ${i}5 ${i}6 ${i}7 ${i}8 ${i}9
    done) > dataout
   test "$dataok" = `cat dataout` || {
     echo "Formula evaluation error for language $lang" 1>&2
