--- origsrc/gettext-0.19.8.1/gettext-runtime/intl/libgnuintl.in.h	2016-06-09 17:56:00.000000000 -0500
+++ src/gettext-0.19.8.1/gettext-runtime/intl/libgnuintl.in.h	2016-10-20 11:03:43.187813700 -0500
@@ -435,7 +435,7 @@ extern int vswprintf (wchar_t *, size_t,
 extern char *setlocale (int, const char *);
 #endif
 
-#if @HAVE_NEWLOCALE@
+#if @HAVE_NEWLOCALE@ && __POSIX_VISIBLE >= 200809
 
 #undef newlocale
 #define newlocale libintl_newlocale
