From 29b2a606db737df5284de2c97ad361e2e571c0ac Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Thu, 13 Aug 2015 13:05:31 +0100
Subject: [PATCH] diable some useless arm libs


diff --git a/mingw-w64-crt/Makefile.in b/mingw-w64-crt/Makefile.in
index 162879c..23539b0 100644
--- a/mingw-w64-crt/Makefile.in
+++ b/mingw-w64-crt/Makefile.in
@@ -6899,7 +6899,6 @@ winrtlibs = \
 @LIBARM32_TRUE@	libarm32/libportabledeviceapi.a \
 @LIBARM32_TRUE@	libarm32/libportabledeviceclassextension.a \
 @LIBARM32_TRUE@	libarm32/libpots.a \
-@LIBARM32_TRUE@	libarm32/libpowerwmiprovider.a \
 @LIBARM32_TRUE@	libarm32/libpowrprof.a \
 @LIBARM32_TRUE@	libarm32/libprintdialogs.a \
 @LIBARM32_TRUE@	libarm32/libprintfilterpipelineprxy.a \
@@ -7095,9 +7094,6 @@ winrtlibs = \
 @LIBARM32_TRUE@	libarm32/libw32time.a libarm32/libw32topl.a \
 @LIBARM32_TRUE@	libarm32/libwabsyncprovider.a \
 @LIBARM32_TRUE@	libarm32/libwbemcomn.a libarm32/libwbiosrvc.a \
-@LIBARM32_TRUE@	libarm32/libwcl.a libarm32/libwcletw.a \
-@LIBARM32_TRUE@	libarm32/libwclpowrprof.a libarm32/libwclsqm.a \
-@LIBARM32_TRUE@	libarm32/libwclunicode.a libarm32/libwclwdi.a \
 @LIBARM32_TRUE@	libarm32/libwcmapi.a libarm32/libwcmcsp.a \
 @LIBARM32_TRUE@	libarm32/libwcmsvc.a libarm32/libwcncsvc.a \
 @LIBARM32_TRUE@	libarm32/libwcneapauthproxy.a \
@@ -7116,7 +7112,6 @@ winrtlibs = \
 @LIBARM32_TRUE@	libarm32/libwersvc.a libarm32/libwerui.a \
 @LIBARM32_TRUE@	libarm32/libwevtapi.a libarm32/libwevtfwd.a \
 @LIBARM32_TRUE@	libarm32/libwevtsvc.a libarm32/libwfdprov.a \
-@LIBARM32_TRUE@	libarm32/libwhealogr.a libarm32/libwhhelper.a \
 @LIBARM32_TRUE@	libarm32/libwiarpc.a libarm32/libwiaservc.a \
 @LIBARM32_TRUE@	libarm32/libwiashext.a libarm32/libwiatrace.a \
 @LIBARM32_TRUE@	libarm32/libwifidisplay.a libarm32/libwimgapi.a \
-- 
2.4.5

