diff --git conemu.orig/src/ConEmu/ConEmuStart.cpp conemu/src/ConEmu/ConEmuStart.cpp
index 7a27727..934ba96 100644
--- conemu.orig/src/ConEmu/ConEmuStart.cpp
+++ conemu/src/ConEmu/ConEmuStart.cpp
@@ -977,7 +977,7 @@ bool CConEmuStart::ParseCommandLine(LPCWSTR pszCmdLine, int& iResult)
 
 					if (!gpConEmu->opt.SizeVal.Exists)
 					{
-						gpConEmu->opt.SizeVal.SetInt(szNext);
+						gpConEmu->opt.SizeVal.SetInt(szNext, 10);
 					}
 				}
 				// ADD fontname; by Mors
@@ -1199,7 +1199,7 @@ bool CConEmuStart::ParseCommandLine(LPCWSTR pszCmdLine, int& iResult)
 					NeedNextArg();
 
 					if (!gpConEmu->opt.FrameWidth.Exists)
-						gpConEmu->opt.FrameWidth.SetInt(szNext);
+						gpConEmu->opt.FrameWidth.SetInt(szNext, 10);
 				}
 				else if (szArg.OneOfSwitches(L"-ShowHide", L"-ShowHideTSA"))
 				{
@@ -1273,7 +1273,7 @@ bool CConEmuStart::ParseCommandLine(LPCWSTR pszCmdLine, int& iResult)
 
 					if (!gpConEmu->opt.BufferHeightVal.Exists)
 					{
-						gpConEmu->opt.BufferHeightVal.SetInt(szNext);
+						gpConEmu->opt.BufferHeightVal.SetInt(szNext, 10);
 
 						if (gpConEmu->opt.BufferHeightVal.GetInt() < 0)
 						{
