diff --git conemu.orig/src/common/Memory.h conemu/src/common/Memory.h
index f1abd1b..c802e1f 100644
--- conemu.orig/src/common/Memory.h
+++ conemu/src/common/Memory.h
@@ -44,10 +44,10 @@ extern HANDLE ghHeap;
 bool HeapInitialize();
 void HeapDeinitialize();
 
-void * __cdecl operator new(size_t size);
-void * __cdecl operator new[](size_t size);
-void __cdecl operator delete(void *block);
-void __cdecl operator delete[](void *ptr);
+// void * __cdecl operator new(size_t size);
+// void * __cdecl operator new[](size_t size);
+// void __cdecl operator delete(void *block);
+// void __cdecl operator delete[](void *ptr);
 
 #ifdef TRACK_MEMORY_ALLOCATIONS
 	struct xf_mem_block
