From 912ea1a9c0f910f993b9acb77e5a6368ee50f59d Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 10 Nov 2014 19:10:32 +0000
Subject: [PATCH 02/18] Dialog Flags added to Mingw-w64

---
 src/ConEmu/ConfirmDlg.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ConEmu/ConfirmDlg.h b/src/ConEmu/ConfirmDlg.h
index 4c4a591..a3568a9 100644
--- a/src/ConEmu/ConfirmDlg.h
+++ b/src/ConEmu/ConfirmDlg.h
@@ -49,7 +49,7 @@ struct ConfirmCloseParam
 };
 int ConfirmCloseConsoles(const ConfirmCloseParam& Parm);
 
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__MINGW64_VERSION_MAJOR)
 
 enum _TASKDIALOG_FLAGS
 {
-- 
2.1.3

