From 395338a039dfa97ea462eae7b74104e89152a9b5 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 10 Nov 2014 19:50:49 +0000
Subject: [PATCH 03/18] SecureZeroMemory support already exists in Mingw-w64

---
 src/common/RConStartArgs.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/RConStartArgs.cpp b/src/common/RConStartArgs.cpp
index b1548d5..8efecb4 100644
--- a/src/common/RConStartArgs.cpp
+++ b/src/common/RConStartArgs.cpp
@@ -43,7 +43,7 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define MCHKHEAP PRAGMA_ERROR("Restricted in ConEmuHk")
 #endif
 
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__MINGW64_VERSION_MAJOR)
 #define SecureZeroMemory(p,s) memset(p,0,s)
 #endif
 
-- 
2.1.3

