diff --git conemu.orig/src/common/defines.h conemu/src/common/defines.h
index 0a47cf8..c32f6d8 100644
--- conemu.orig/src/common/defines.h
+++ conemu/src/common/defines.h
@@ -51,12 +51,12 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define WARNING(s)
 #define PRAGMA_ERROR(s)
 
-#ifndef max
-#define max(a,b)            (((a) > (b)) ? (a) : (b))
+#ifndef MAX
+#define MAX(a,b)            (((a) > (b)) ? (a) : (b))
 #endif
 
-#ifndef min
-#define min(a,b)            (((a) < (b)) ? (a) : (b))
+#ifndef MIN
+#define MIN(a,b)            (((a) < (b)) ? (a) : (b))
 #endif
 
 //#define _ASSERT(f)
