diff --git conemu.orig/src/common/CEStr.h conemu/src/common/CEStr.h
index d723522..05f769f 100644
--- conemu.orig/src/common/CEStr.h
+++ conemu/src/common/CEStr.h
@@ -63,7 +63,7 @@ public:
 	// *** Not copyable, not implemented, use explicit Set method ***
 	#if defined(__GNUC__)
 	public:
-	CEStr(const CEStr&) = delete;
+	// CEStr(const CEStr&) = delete;
 	CEStr& operator=(const CEStr &) = delete;
 	#else
 	private:
