diff --git conemu.orig/src/ConEmuC/ConEmuC.cpp conemu/src/ConEmuC/ConEmuC.cpp
index 19d6a56..a4d5e1f 100644
--- conemu.orig/src/ConEmuC/ConEmuC.cpp
+++ conemu/src/ConEmuC/ConEmuC.cpp
@@ -215,7 +215,7 @@ int DoParseArgs(LPCWSTR asCmdLine)
 	#define HL(fore) Highlighter hl(hOut, csbi.wAttributes, fore)
 
 	#if defined(__GNUC__)
-	lstrcpyn(szCLVer, "GNUC");
+	lstrcpyA(szCLVer, "GNUC");
 	#elif defined(_MSC_VER)
 	_wsprintfA(szCLVer, SKIPCOUNT(szCLVer) "VC %u.%u", (int)(_MSC_VER / 100), (int)(_MSC_VER % 100));
 	#else
