From cde6c617ea13f55fce2bf12e0f5cf512fb184f4e Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 10 Nov 2014 20:48:13 +0000
Subject: [PATCH 06/18] IID_IDragSourceHelper2 is already defined in mingw-w64

---
 src/ConEmu/DragDropData.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ConEmu/DragDropData.cpp b/src/ConEmu/DragDropData.cpp
index 23fde66..bafd69e 100644
--- a/src/ConEmu/DragDropData.cpp
+++ b/src/ConEmu/DragDropData.cpp
@@ -77,7 +77,7 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif
 
 // COM TaskBarList interface support
-#if defined(__GNUC__) && defined(USE_DRAG_HELPER)
+#if defined(__GNUC__) && defined(USE_DRAG_HELPER) && !defined(__MINGW64_VERSION_MAJOR)
 const IID IID_IDragSourceHelper2 = {0x83E07D0D, 0x0C5F, 0x4163, {0xBF, 0x1A, 0x60, 0xB2, 0x74, 0x05, 0x1E, 0x40}};
 #endif
 
-- 
2.1.3

