diff -durN conemu.orig/src/ConEmuHK/makefile_gcc conemu/src/ConEmuHK/makefile_gcc
--- conemu.orig/src/ConEmuHK/makefile_gcc	2017-05-28 21:08:05.281250000 +0800
+++ conemu/src/ConEmuHK/makefile_gcc	2017-05-28 22:46:11.828125000 +0800
@@ -57,6 +57,7 @@
 		../common/WObjects.cpp       \
 		../common/WThreads.cpp       \
 		../common/WUser.cpp          \
+		../common/WSession.cpp		 \
 		../modules/minhook/src/buffer.c     \
 		../modules/minhook/src/hook.c       \
 		../modules/minhook/src/trampoline.c \
