From 54d68aa85aba5372fc1cbc3cad25ca4514c92ef6 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 10 Nov 2014 20:49:15 +0000
Subject: [PATCH 07/18] wmemmove_s is available in mingw-w64

---
 src/ConEmu/Header.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ConEmu/Header.h b/src/ConEmu/Header.h
index 8dab248..7683612 100644
--- a/src/ConEmu/Header.h
+++ b/src/ConEmu/Header.h
@@ -52,7 +52,7 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "../common/Memory.h"
 #include "../common/MAssert.h"
 
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__MINGW64_VERSION_MAJOR)
 #define wmemmove_s(d,ds,s,ss) wmemmove(d,s,ss)
 #define SecureZeroMemory(p,s) memset(p,0,s)
 #endif
-- 
2.1.3

