From 78d260d023005113a8c3df27de9fb6a337b7bad5 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 10 Nov 2014 20:50:15 +0000
Subject: [PATCH 08/18] use _printf because printf can be undefined in this
 case

---
 src/ConEmuCD/ConsoleMain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ConEmuCD/ConsoleMain.cpp b/src/ConEmuCD/ConsoleMain.cpp
index eae3afa..769bd96 100644
--- a/src/ConEmuCD/ConsoleMain.cpp
+++ b/src/ConEmuCD/ConsoleMain.cpp
@@ -3384,7 +3384,7 @@ int DoParseArgs(LPCWSTR asCmdLine)
 	_printf("Standard shell splitter\n");
 	for (int j = 0; j < iShellCount; j++)
 	{
-		printf("%u: `", j);
+		_printf("%u: `", j);
 		_wprintf(ppszShl[j]);
 		_printf("`\n");
 	}
-- 
2.1.3

