From 6d09ddb72d41ba46c119a60b5060f6f7eaae3ee5 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 10 Nov 2014 20:51:04 +0000
Subject: [PATCH 09/18] Internet structs and defines are already defined in
 mingw-w64

---
 src/ConEmuCD/Downloader.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ConEmuCD/Downloader.cpp b/src/ConEmuCD/Downloader.cpp
index b20794d..ec883ff 100644
--- a/src/ConEmuCD/Downloader.cpp
+++ b/src/ConEmuCD/Downloader.cpp
@@ -34,7 +34,7 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "../common/MSectionSimple.h"
 #include "Downloader.h"
 
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__MINGW64_VERSION_MAJOR)
 typedef struct {
     DWORD dwMajorVersion;
     DWORD dwMinorVersion;
-- 
2.1.3

