From 55a4a591c72047aea8b674bf2ae849850d8d7141 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 10 Nov 2014 20:52:01 +0000
Subject: [PATCH 10/18] __forceinline is already defined in mingw-w64

---
 src/ConEmu/kl_parts.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ConEmu/kl_parts.h b/src/ConEmu/kl_parts.h
index c302109..3fd7709 100644
--- a/src/ConEmu/kl_parts.h
+++ b/src/ConEmu/kl_parts.h
@@ -24,7 +24,7 @@ typedef unsigned __int16 u16;
 typedef unsigned __int64 u64;
 typedef DWORD u32;
 
-#if defined(__GNUC__)
+#if defined(__GNUC__) && !defined(__MINGW64_VERSION_MAJOR)
 #define __forceinline __inline__
 #endif
 template <class T>__forceinline const T& klMin(const T &a, const T &b) {return a < b ? a : b;}
-- 
2.1.3

