From cdd911925374a45f16decf0a03ff8ad442584490 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Tue, 11 Nov 2014 04:17:02 +0000
Subject: [PATCH 12/18] AssertMessageBox is not supported in mingw-w64

---
 src/ConEmu/ConEmuApp.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ConEmu/ConEmuApp.cpp b/src/ConEmu/ConEmuApp.cpp
index e275795..d0ce118 100644
--- a/src/ConEmu/ConEmuApp.cpp
+++ b/src/ConEmu/ConEmuApp.cpp
@@ -3336,7 +3336,11 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine
 	gnOsVer = ((gOSVer.dwMajorVersion & 0xFF) << 8) | (gOSVer.dwMinorVersion & 0xFF);
 	HeapInitialize();
 	RemoveOldComSpecC();
+
+#if !defined(__MINGW64_VERSION_MAJOR)
 	AssertMsgBox = MsgBox;
+#endif
+
 	gn_MainThreadId = GetCurrentThreadId();
 
 	// On Vista and higher ensure our process will be
-- 
2.1.3

