From e1260cf291a7c9336edb199ce8f24b9378953ab0 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Tue, 11 Nov 2014 04:24:03 +0000
Subject: [PATCH 15/18] fix writeprocessd export for gcc

---
 src/ConEmuHk/Ansi.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/ConEmuHk/Ansi.h b/src/ConEmuHk/Ansi.h
index 3e52bb2..f55395f 100644
--- a/src/ConEmuHk/Ansi.h
+++ b/src/ConEmuHk/Ansi.h
@@ -45,6 +45,14 @@ typedef BOOL (WINAPI* OnWriteConsoleW_t)(HANDLE hConsoleOutput, const VOID *lpBu
 
 struct MSectionSimple;
 
+#if defined(__GNUC__)
+extern "C" {
+#endif
+	BOOL WINAPI WriteProcessed(LPCWSTR lpBuffer, DWORD nNumberOfCharsToWrite, LPDWORD lpNumberOfCharsWritten);
+#if defined(__GNUC__)
+};
+#endif
+
 struct CEAnsi
 {
 //private:
-- 
2.1.3

