From 4c4e3d4b561e9a78825eea30bb0af72665945514 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Tue, 11 Nov 2014 04:24:57 +0000
Subject: [PATCH 16/18] fix ConEmuHK linking under gcc

---
 src/ConEmuHk/makefile_gcc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/ConEmuHk/makefile_gcc b/src/ConEmuHk/makefile_gcc
index e2de4df..a5b2081 100644
--- a/src/ConEmuHk/makefile_gcc
+++ b/src/ConEmuHk/makefile_gcc
@@ -9,12 +9,14 @@ SRCS = \
 		../common/InQueue.cpp        \
 		../common/MAssert.cpp        \
 		../common/Memory.cpp         \
+		../common/Monitors.cpp       \
 		../common/MSectionSimple.cpp \
 		../common/MSection.cpp       \
 		../common/MSecurity.cpp      \
 		../common/MStrSafe.cpp       \
 		../common/RConStartArgs.cpp  \
 		../common/WConsole.cpp     \
+		../common/WFiles.cpp       \
 		../common/WObjects.cpp     \
 		Ansi.cpp \
 		ConEmuHooks.cpp \
@@ -112,7 +114,7 @@ AR = $(PREFIX)ar
 RANLIB = $(PREFIX)ranlib
 CXXFLAGS = $(ENCFLAGS) -I $(COMMON) -I $(COMINC) -Wall -Os -funsigned-char -fomit-frame-pointer -fstrict-aliasing -fno-rtti -fno-exceptions $(CXXWIDE) $(USERCPP) $(WINVER)
 CCFLAGS = -I $(COMMON) -I $(COMINC) -Wall -Os -funsigned-char -fomit-frame-pointer -fstrict-aliasing -fno-exceptions $(CCWIDE) $(USERCPP) $(WINVER)
-LNKFLAGS = -mdll $(BASEADDRESS) -s -lgcc $(USERLIBSFIRST) $(COMMONLIB) -luser32 -lkernel32 -ladvapi32 -lshell32 $(USERLIBS)
+LNKFLAGS = -mdll $(BASEADDRESS) -s -lgcc $(USERLIBSFIRST) $(COMMONLIB) -luser32 -lkernel32 -ladvapi32 -lshell32 -luuid -lgdi32 $(USERLIBS)
 RCFLAGS = -I $(COMINC) $(RCWIDE) $(USERRC)
 
 OBJS = $(patsubst %.cpp,$(OBJDIR)/%.o,$(filter %.cpp,$(SRCS))) $(patsubst %.c,$(OBJDIR)/%.o,$(filter %.c,$(SRCS))) $(RES)
-- 
2.1.3

