From 00d6a150802ac1a989abe9ec8b0efe6577cd1354 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Tue, 11 Nov 2014 04:26:37 +0000
Subject: [PATCH 17/18] Add guard to vista transparency for mingw-w64

---
 src/common/WConsole.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/common/WConsole.cpp b/src/common/WConsole.cpp
index 2f6cef1..19fa119 100644
--- a/src/common/WConsole.cpp
+++ b/src/common/WConsole.cpp
@@ -424,6 +424,8 @@ void GetConsoleSizeInfo(CONSOLE_INFO *pci)
 #undef ENABLE_AUTO_POSITION
 #endif
 
+#if !defined(__MINGW64_VERSION_MAJOR)
+
 //VISTA support:
 #ifndef ENABLE_AUTO_POSITION
 typedef struct _CONSOLE_FONT_INFOEX
@@ -437,6 +439,7 @@ typedef struct _CONSOLE_FONT_INFOEX
 } CONSOLE_FONT_INFOEX, *PCONSOLE_FONT_INFOEX;
 #endif
 
+#endif
 
 // Vista+ only
 BOOL apiGetConsoleScreenBufferInfoEx(HANDLE hConsoleOutput, MY_CONSOLE_SCREEN_BUFFER_INFOEX* lpConsoleScreenBufferInfoEx)
-- 
2.1.3

