From 7504bd26518f24511418bab5c69b695122547b75 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Wed, 26 Nov 2014 02:58:50 +0000
Subject: [PATCH] update minimal windows version to 0x0600

---
 src/ConEmu/makefile_gcc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ConEmu/makefile_gcc b/src/ConEmu/makefile_gcc
index 757ba2a..76cc4d1 100644
--- a/src/ConEmu/makefile_gcc
+++ b/src/ConEmu/makefile_gcc
@@ -151,8 +151,8 @@ else
   BITNAME = 
 endif
 RCWIDE = -D UNICODE -D _UNICODE $(BITDEF)
-CXXWIDE = -D UNICODE -D _UNICODE -D _WIN32_WINNT=0x0500 -D WINVER=0x0500 -D _WIN32_IE=0x0500 $(BITDEF)
-CCWIDE = -D UNICODE -D _UNICODE -D _WIN32_WINNT=0x0500 -D WINVER=0x0500 -D _WIN32_IE=0x0500 $(BITDEF)
+CXXWIDE = -D UNICODE -D _UNICODE -D _WIN32_WINNT=0x0601 -D WINVER=0x0601 -D _WIN32_IE=0x0601 $(BITDEF)
+CCWIDE = -D UNICODE -D _UNICODE -D _WIN32_WINNT=0x0601 -D WINVER=0x0601 -D _WIN32_IE=0x0601 $(BITDEF)
 
 OBJDIR = ../../gcc/conemu$(DIRBIT)
 REOBJDIR = \.\.\/\.\.\/gcc\/conemu$(DIRBIT)
-- 
2.1.3

