From ac82a7980ccb291b3747bfe07a41af6bc8404201 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Wed, 26 Nov 2014 02:47:24 +0000
Subject: [PATCH] removed random call to pszText

---
 src/ConEmu/ConEmuApp.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/ConEmu/ConEmuApp.cpp b/src/ConEmu/ConEmuApp.cpp
index f93b544..9fc1b64 100644
--- a/src/ConEmu/ConEmuApp.cpp
+++ b/src/ConEmu/ConEmuApp.cpp
@@ -601,8 +601,6 @@ int MyGetDlgItemText(HWND hDlg, WORD nID, wchar_t (&rszText)[size])
 	CEStr szText;
 	size_t cchMax = 0;
 	int nLen = MyGetDlgItemText(hDlg, nID, cchMax, szText.ms_Arg);
-	if (!pszText)
-		return false;
 	if (lstrcmp(rszText, szText.ms_Arg) == 0)
 		return false;
 	lstrcpyn(rszText, szText.ms_Arg, size);
-- 
2.1.3

