From dcd1586a87c7a506907efe81af336829ed849347 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Wed, 26 Nov 2014 03:23:50 +0000
Subject: [PATCH] on msys2 ignore recommended files warning

---
 src/ConEmu/ConEmu.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ConEmu/ConEmu.cpp b/src/ConEmu/ConEmu.cpp
index 8176e84..cf1088a 100644
--- a/src/ConEmu/ConEmu.cpp
+++ b/src/ConEmu/ConEmu.cpp
@@ -964,7 +964,7 @@ bool CConEmuMain::CheckRequiredFiles()
 		}
 	}
 
-	if (*szRequired || *szRecommended)
+	if (*szRequired)
 	{
 		size_t cchMax = _tcslen(szRequired) + _tcslen(szRecommended) + _tcslen(ms_ConEmuExe) + 255;
 		wchar_t* pszMsg = (wchar_t*)calloc(cchMax, sizeof(*pszMsg));
-- 
2.1.3

