diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e9142d59..08e09e1b4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -219,8 +219,6 @@ CHECK_INCLUDE_FILES (sys/uio.h             HAVE_SYS_UIO_H)
 CHECK_INCLUDE_FILES (time.h                HAVE_TIME_H)
 CHECK_INCLUDE_FILES (dlfcn.h               HAVE_DLFCN_H)
 CHECK_INCLUDE_FILES (unistd.h              HAVE_UNISTD_H)
-CHECK_INCLUDE_FILES (iphlpapi.h            HAVE_IPHLPAPI_H)
-
 # On OpenBSD, you must include sys/types.h before netinet/tcp.h
 IF (HAVE_SYS_TYPES_H)
   CHECK_INCLUDE_FILES ("sys/types.h;netinet/tcp.h"         HAVE_NETINET_TCP_H)
@@ -234,6 +232,8 @@ ENDIF ()
 IF (WIN32)
 CHECK_INCLUDE_FILES ("winsock2.h;windows.h"            HAVE_WINSOCK2_H)
 CHECK_INCLUDE_FILES ("winsock2.h;ws2tcpip.h;windows.h" HAVE_WS2TCPIP_H)
+CHECK_INCLUDE_FILES ("winsock2.h;iphlpapi.h;windows.h" HAVE_IPHLPAPI_H)
+CHECK_INCLUDE_FILES ("winsock2.h;netioapi.h;windows.h" HAVE_NETIOAPI_H)
 CHECK_INCLUDE_FILES ("winsock.h;windows.h"             HAVE_WINSOCK_H)
 CHECK_INCLUDE_FILES (windows.h                         HAVE_WINDOWS_H)
 ENDIF ()
