--- a/Makefile-libbz2_so
+++ b/Makefile-libbz2_so
@@ -35,10 +35,8 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.7 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.7
-	rm -f libbz2.so.1.0
-	ln -s libbz2.so.1.0.7 libbz2.so.1.0
+	$(CC) $(CFLAGS) -shared -Wl,--out-implib -Wl,libbz2.dll.a -o msys-bz2-1.dll $(OBJS)
+	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.dll.a
 
 clean: 
 	rm -f $(OBJS) bzip2.o libbz2.so.1.0.7 libbz2.so.1.0 bzip2-shared
