diff -Naur XML-NamespaceSupport-1.11/inc/Module/Install/Can.pm XML-NamespaceSupport-1.11/inc/Module/Install/Can.pm
--- XML-NamespaceSupport-1.11/inc/Module/Install/Can.pm	2010-03-16 19:18:52.000000000 +0000
+++ XML-NamespaceSupport-1.11/inc/Module/Install/Can.pm	2013-06-16 16:03:38.364812500 +0000
@@ -59,7 +59,7 @@
 }
 
 # Fix Cygwin bug on maybe_command();
-if ( $^O eq 'cygwin' ) {
+if ( $^O eq 'cygwin' or $^O eq 'msys' ) {
 	require ExtUtils::MM_Cygwin;
 	require ExtUtils::MM_Win32;
 	if ( ! defined(&ExtUtils::MM_Cygwin::maybe_command) ) {
