diff --git a/Makefile.PL b/Makefile.PL
index fcc3a7d..937789a 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -8,8 +8,14 @@ use ExtUtils::MakeMaker;
 use File::Spec::Functions qw( catfile rel2abs );
 use Getopt::Long qw( GetOptionsFromArray );
 
-unless( require( catfile qw(inc IO Interactive Tiny.pm) ) ) {
-    die 'Your distribution is incomplete: Failed to load bundled IO::Interactive::Tiny';
+BEGIN {
+  # Must use a bundled version
+  # "./" prefix required to subvert @INC traversal
+  # "catfile" not useful here as it eats "." and may produces wrong \
+  #  under windows ( require always takes / )
+  local $@;
+  eval { require "./inc/IO/Interactive/Tiny.pm" } or
+    die "Your distribution is incomplete: Failed to load bundled IO::Interactive::Tiny\n$@";
 }
 
 run(\@ARGV, [qw{ssl crypto ssl32 ssleay32 eay32 libeay32 z}]);
-- 
2.13.1
