diff --git a/src/configure.ac b/src/configure.ac-patched
index ff1f482..b99c7e2 100644
--- a/src/configure.ac
+++ b/src/configure.ac-patched
@@ -618,6 +618,8 @@
       fi
       if test -f "${vi_cv_path_lua_pfx}/bin/cyglua-${vi_cv_version_lua}.dll"; then
 	vi_cv_dll_name_lua="cyglua-${vi_cv_version_lua}.dll"
+      elif test -f "${vi_cv_path_lua_pfx}/bin/msys-lua-${vi_cv_version_lua}.dll"; then
+	vi_cv_dll_name_lua="msys-lua-${vi_cv_version_lua}.dll"
       else
 	if test "x$MACOSX" = "xyes"; then
 	  ext="dylib"
