diff --git a/src/configure.in b/src/configure.in-patched
index ff1f482..b99c7e2 100644
--- a/src/configure.in
+++ b/src/configure.in-patched
@@ -2755,7 +2755,7 @@ dnl ---------------------------------------------------------------------------
 dnl Check for Cygwin, which needs an extra source file if not using X11
 AC_MSG_CHECKING(for CYGWIN environment)
 case `uname` in
-    CYGWIN*)    CYGWIN=yes; AC_MSG_RESULT(yes)
+    CYGWIN*|MSYS*)    CYGWIN=yes; AC_MSG_RESULT(yes)
                 AC_MSG_CHECKING(for CYGWIN clipboard support)
                 if test "x$with_x" = "xno" ; then
                   OS_EXTRA_SRC=winclip.c; OS_EXTRA_OBJ=objects/winclip.o
