diff -Naur ExtUtils-Depends-0.304/lib/ExtUtils/Depends.pm ExtUtils-Depends-0.304/lib/ExtUtils/Depends.pm
--- ExtUtils-Depends-0.304/lib/ExtUtils/Depends.pm	2011-01-26 10:55:19.000000000 +0300
+++ ExtUtils-Depends-0.304/lib/ExtUtils/Depends.pm	2013-07-17 08:44:02.824218700 +0400
@@ -312,6 +312,7 @@
 	my %mappers = (
 		MSWin32 => sub { $_[0] . '\.(?:lib|a)' },
 		cygwin	=> sub { $_[0] . '\.dll'},
+		msys	=> sub { $_[0] . '\.dll'},
 	);
 	my $mapper = $mappers{$^O};
 	return () unless defined $mapper;
diff -Naur ExtUtils-Depends-0.304/t/04_extra_libs.t ExtUtils-Depends-0.304/t/04_extra_libs.t
--- ExtUtils-Depends-0.304/t/04_extra_libs.t	2010-11-23 06:27:56.000000000 +0300
+++ ExtUtils-Depends-0.304/t/04_extra_libs.t	2013-07-17 08:44:31.199218700 +0400
@@ -12,9 +12,9 @@
 
 my $tmp_inc = temp_inc;
 
-plan (($^O eq 'MSWin32' || $^O eq 'cygwin') ?
+plan (($^O eq 'MSWin32' || $^O eq 'cygwin' || $^O eq 'msys') ?
         (tests => 1) :
-        (skip_all => 'test only applicable to MSWin32 and cygwin'));
+        (skip_all => 'test only applicable to MSWin32 and cygwin/MSYS'));
 
 my $dep_info = ExtUtils::Depends->new ('DepTest');
 $dep_info->save_config (catfile $tmp_inc, qw(DepTest Install Files.pm));
