diff -Naur ExtUtils-Depends-0.404/lib/ExtUtils/Depends.pm ExtUtils-Depends-0.404/lib/ExtUtils/Depends.pm
--- ExtUtils-Depends-0.404/lib/ExtUtils/Depends.pm	2011-01-26 10:55:19.000000000 +0300
+++ ExtUtils-Depends-0.404/lib/ExtUtils/Depends.pm	2013-07-17 08:44:02.824218700 +0400
@@ -341,6 +341,7 @@
 	my %mappers = (
 		MSWin32 => sub { $_[0] . '\.(?:lib|a)' },
 		cygwin	=> sub { $_[0] . '\.dll'},
+		msys	=> sub { $_[0] . '\.dll'},
 		android => sub { $_[0] . '\.' . $Config{dlext} },
 	);
 	my $mapper = $mappers{$^O};
diff -Naur ExtUtils-Depends-0.404/t/04_extra_libs.t ExtUtils-Depends-0.404/t/04_extra_libs.t
--- ExtUtils-Depends-0.404/t/04_extra_libs.t	2010-11-23 06:27:56.000000000 +0300
+++ ExtUtils-Depends-0.404/t/04_extra_libs.t	2013-07-17 08:44:31.199218700 +0400
@@ -12,10 +12,10 @@
 
 my $tmp_inc = temp_inc;
 
-plan (($^O eq 'MSWin32' || $^O eq 'cygwin' || $^O eq 'android') ?
+plan (($^O eq 'MSWin32' || $^O eq 'cygwin' || $^O eq 'msys' || $^O eq 'android') ?
         (tests => 1) :
         (skip_all
-            => "test only for 'MSWin32', 'cygwin', and 'android'"));
+            => "test only for 'MSWin32', 'cygwin', 'msys', and 'android'"));
 
 my $dep_info = ExtUtils::Depends->new ('DepTest');
 $dep_info->save_config (catfile $tmp_inc, qw(DepTest Install Files.pm));
