diff -Naur binutils-2.26/binutils/bucomm.c binutils-2.26.0002/binutils/bucomm.c
--- binutils-2.26/binutils/bucomm.c	2015-11-13 09:27:40.000000000 +0100
+++ binutils-2.26.0002/binutils/bucomm.c	2016-03-10 16:56:14.238768551 +0100
@@ -574,6 +574,15 @@
 get_file_size (const char * file_name)
 {
   struct stat statbuf;
+  int f, t;
+  t = -1;
+  f = open (file_name, O_RDONLY | O_BINARY);
+
+  if (f != 0)
+    {
+      t = isatty (f);
+      close (f);
+    }
 
   if (stat (file_name, &statbuf) < 0)
     {
@@ -583,8 +592,15 @@
 	non_fatal (_("Warning: could not locate '%s'.  reason: %s"),
 		   file_name, strerror (errno));
     }
-  else if (! S_ISREG (statbuf.st_mode))
-    non_fatal (_("Warning: '%s' is not an ordinary file"), file_name);
+  else if (! S_ISREG (statbuf.st_mode) || t > 0)
+    {
+#ifdef _WIN32
+      /* libtool passes /dev/null and checks for /dev/null in the output */
+      if (stricmp (file_name, "nul") == 0)
+        file_name = "/dev/null";
+#endif
+      non_fatal (_("Warning: '%s' is not an ordinary file"), file_name);
+    }
   else if (statbuf.st_size < 0)
     non_fatal (_("Warning: '%s' has negative size, probably it is too large"),
                file_name);
diff -Naur binutils-2.26/binutils/elfedit.c binutils-2.26.0002/binutils/elfedit.c
--- binutils-2.26/binutils/elfedit.c	2015-11-13 09:27:40.000000000 +0100
+++ binutils-2.26.0002/binutils/elfedit.c	2016-03-10 16:49:47.531467910 +0100
@@ -483,6 +483,14 @@
 check_file (const char *file_name, struct stat *statbuf_p)
 {
   struct stat statbuf;
+  int f, t;
+  t = -1;
+  f = open (file_name, O_RDONLY | O_BINARY);
+  if (f != 0)
+    {
+      t = isatty (f);
+      close (f);
+    }
 
   if (statbuf_p == NULL)
     statbuf_p = &statbuf;
@@ -497,8 +505,13 @@
       return 1;
     }
 
-  if (! S_ISREG (statbuf_p->st_mode))
+  if (! S_ISREG (statbuf_p->st_mode) || t > 0)
     {
+#ifdef _WIN32
+      /* libtool passes /dev/null and checks for /dev/null in the output */
+      if (stricmp (file_name, "nul") == 0)
+        file_name = "/dev/null";
+#endif
       error (_("'%s' is not an ordinary file\n"), file_name);
       return 1;
     }
diff -Naur binutils-2.26/binutils/readelf.c binutils-2.26.0002/binutils/readelf.c
--- binutils-2.26/binutils/readelf.c	2015-11-13 09:27:41.000000000 +0100
+++ binutils-2.26.0002/binutils/readelf.c	2016-03-10 16:49:47.551467426 +0100
@@ -16493,6 +16493,14 @@
   struct stat statbuf;
   char armag[SARMAG];
   int ret;
+  int f, t;
+  t = -1;
+  f = open (file_name, O_RDONLY | O_BINARY);
+  if (f != 0)
+    {
+      t = isatty (f);
+      close (f);
+    }
 
   if (stat (file_name, &statbuf) < 0)
     {
@@ -16504,8 +16512,13 @@
       return 1;
     }
 
-  if (! S_ISREG (statbuf.st_mode))
+  if (! S_ISREG (statbuf.st_mode) || t > 0)
     {
+#ifdef _WIN32
+      /* libtool passes /dev/null and checks for /dev/null in the output */
+      if (stricmp (file_name, "nul") == 0)
+        file_name = "/dev/null";
+#endif
       error (_("'%s' is not an ordinary file\n"), file_name);
       return 1;
     }
