From 4259fef8f6afa3d68c285484d860a3558aed26c3 Mon Sep 17 00:00:00 2001
From: Wang Jinghao <zheng.xianyuwang@gmail.com>
Date: Wed, 24 Dec 2025 19:45:05 +0800
Subject: [PATCH 1/2] Add -lregex to unit target

---
 Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.am b/Makefile.am
index 4c0bb60..208e925 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -228,6 +228,7 @@ RUNTEST = ${top_srcdir}/runtest
 AM_CXXFLAGS = -I$(top_srcdir) -g
 check_PROGRAMS = unit
 unit_SOURCES = testsuite/libdejagnu/unit.cc
+unit_LDADD = $(LDADD) -lregex
 
 # Documentation.
 
-- 
2.52.0

