From 5441b671c064ba0acf88376b1f5edb3f67908c8f Mon Sep 17 00:00:00 2001
From: Wang Jinghao <zheng.xianyuwang@gmail.com>
Date: Wed, 24 Dec 2025 21:43:38 +0800
Subject: [PATCH 2/2] Remove redundant ESC escape characters

---
 lib/target.exp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/target.exp b/lib/target.exp
index 36ae639..7292a5c 100644
--- a/lib/target.exp
+++ b/lib/target.exp
@@ -283,6 +283,9 @@ proc prune_warnings { text } {
     # Ignore linker warning when searching 64bit libraries in /lib.
     regsub -all "(^|\n)\[^\n\]*: skipping incompatible \[^\n\]* when searching for \[^\n\]*" $text "" text
 
+    # Remove ESC escape sequences
+    regsub -all -- "\033\\\[\[0-9;?\]*\[a-zA-Z\]" $text "" text
+
     # It might be tempting to get carried away and delete blank lines, etc.
     # Just delete *exactly* what we're ask to, and that's it.
     return $text
-- 
2.52.0

