--- elinks-git/configure.in.orig	2014-09-07 19:55:58.321000000 +0400
+++ elinks-git/configure.in	2014-09-07 20:09:12.185600000 +0400
@@ -288,7 +288,7 @@
 AC_CHECK_FUNCS(gettimeofday clock_gettime)
 AC_CHECK_FUNCS(setitimer, HAVE_SETITIMER=yes)
 
-AC_CHECK_FUNCS([cygwin_conv_to_full_win32_path])
+AC_CHECK_FUNCS([cygwin_conv_path])
 
 AC_CHECK_FUNCS(setenv putenv, HAVE_SETENV_OR_PUTENV=yes)
 AC_CHECK_FUNCS(unsetenv)
--- elinks-git/src/session/download.c.orig	2014-09-07 20:25:12.165100000 +0400
+++ elinks-git/src/session/download.c	2014-09-07 20:27:21.058300000 +0400
@@ -1047,14 +1047,10 @@
 
 		if (*prog == '%') {
 			input = 0;
-#if defined(HAVE_CYGWIN_CONV_TO_FULL_WIN32_PATH)
-#ifdef MAX_PATH
-			unsigned char new_path[MAX_PATH];
-#else
-			unsigned char new_path[1024];
-#endif
+#if defined(HAVE_CYGWIN_CONV_PATH
+			unsigned char new_path[PATH_MAX];
 
-			cygwin_conv_to_full_win32_path(file, new_path);
+			cygwin_conv_path(CCP_POSIX_TO_WIN_A|CCP_ABSOLUTE, file, new_path, PATH_MAX+1);
 			add_to_string(&name, new_path);
 #else
 			add_shell_quoted_to_string(&name, file, strlen(file));
