diff -urN -x .build -x .inst -x .sinst automake-1.6.3-orig/m4/Makefile.in automake-1.6.3/m4/Makefile.in
--- automake-1.6.3-orig/m4/Makefile.in	2002-07-28 13:18:29.000000000 -0400
+++ automake-1.6.3/m4/Makefile.in	2005-06-26 02:48:36.140625000 -0400
@@ -89,6 +89,7 @@
 
 nodist_m4data_DATA = amversion.m4
 DISTCLEANFILES = amversion.m4
+EXTRA_DIST = dirlist
 subdir = m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/lib/mkinstalldirs
 CONFIG_CLEAN_FILES = amversion.m4
diff -urN -x .build -x .inst -x .sinst automake-1.6.3-orig/tests/Makefile.in automake-1.6.3/tests/Makefile.in
--- automake-1.6.3-orig/tests/Makefile.in	2002-07-28 13:18:30.000000000 -0400
+++ automake-1.6.3/tests/Makefile.in	2005-06-26 02:48:36.296875000 -0400
@@ -204,6 +204,7 @@
 depend3.test \
 depend4.test \
 dirforbid.test \
+dirlist.test \
 dirname.test \
 discover.test \
 distcommon.test \
