--- origsrc/automake-1.6.3/tests/depcomp2.test	2001-12-26 03:22:05.000000000 -0500
+++ src/automake-1.6.3/tests/depcomp2.test	2008-10-25 17:03:51.255889600 -0400
@@ -37,4 +37,6 @@
 $AUTOMAKE --add-missing || exit 1
 $AUTOCONF || exit 1
 CC='gcc' ./configure 2>error.log
-test -z "`cat error.log`"
+-test -z "`cat error.log`"
+# accomodate newer autoconf
+test -z "`cat error.log | sed -e '/seems to ignore the --datarootdir setting/d'`"
--- origsrc/automake-1.6.3/tests/man2.test	2002-04-26 11:58:38.000000000 -0400
+++ src/automake-1.6.3/tests/man2.test	2008-10-25 17:03:51.275918400 -0400
@@ -26,6 +26,7 @@
 ./configure --prefix=''
 DESTDIR="`pwd`/=inst" $MAKE -e SHELL=/bin/sh install
 
-test -f ./=inst/man/man2/foo.2
-test -f ./=inst/man/man4/foo.4
-test -f ./=inst/man/man4/bar.4
+# accomodate newer autoconf
+test -f ./=inst/man/man2/foo.2 || test -f ./=inst/share/man/man2/foo.2
+test -f ./=inst/man/man4/foo.4 || test -f ./=inst/share/man/man4/foo.4
+test -f ./=inst/man/man4/bar.4 || test -f ./=inst/share/man/man4/bar.4
