--- brotli-1.0.5/CMakeLists.txt.orig	2018-07-06 21:46:37.732217200 +0300
+++ brotli-1.0.5/CMakeLists.txt	2018-07-06 21:57:09.791368900 +0300
@@ -142,8 +142,11 @@
 add_library(brotlienc SHARED ${BROTLI_ENC_C})
 
 add_library(brotlicommon-static STATIC ${BROTLI_COMMON_C})
+set_target_properties(brotlicommon-static PROPERTIES OUTPUT_NAME brotlicommon)
 add_library(brotlidec-static STATIC ${BROTLI_DEC_C})
+set_target_properties(brotlidec-static PROPERTIES OUTPUT_NAME brotlidec)
 add_library(brotlienc-static STATIC ${BROTLI_ENC_C})
+set_target_properties(brotlienc-static PROPERTIES OUTPUT_NAME brotlienc)
 
 # Older CMake versions does not understand INCLUDE_DIRECTORIES property.
 include_directories(${BROTLI_INCLUDE_DIRS})
