diff -Naur swig-4.0.0-orig/Lib/swiglabels.swg swig-4.0.0/Lib/swiglabels.swg
--- swig-4.0.0-orig/Lib/swiglabels.swg	2019-04-26 21:48:33.000000000 +0300
+++ swig-4.0.0/Lib/swiglabels.swg	2019-05-27 08:49:41.190538400 +0300
@@ -91,7 +91,7 @@
 
 /* calling conventions for Windows */
 #ifndef SWIGSTDCALL
-# if defined(_WIN32) || defined(__WIN32__) || defined(__CYGWIN__)
+# if defined(_WIN32) || defined(__WIN32__)
 #   define SWIGSTDCALL __stdcall
 # else
 #   define SWIGSTDCALL
