diff -Naur swig-2.0.10-orig/configure swig-2.0.10/configure
--- swig-2.0.10-orig/configure	2013-05-27 22:50:41.000000000 +0400
+++ swig-2.0.10/configure	2013-07-23 13:48:07.083984400 +0400
@@ -6373,7 +6373,7 @@
 	case $host in
 	*-*-hp*) SO=.sl;;
 	*-*-darwin*) SO=.bundle;;
-	*-*-cygwin* | *-*-mingw*) SO=.dll;;
+	*-*-cygwin* | *-*-msys* | *-*-mingw*) SO=.dll;;
 	*) SO=.so;;
 	esac
 fi
@@ -6390,7 +6390,7 @@
 then
 	case $host in
 	*-*-aix*) LDSHARED="\$(srcdir)/ld_so_aix \$(CC)";;
-	*-*-cygwin* | *-*-mingw*)
+	*-*-cygwin* | *-*-msys* | *-*-mingw*)
             if test "$GCC" = yes; then
                 LDSHARED="$CC -shared"
             else
@@ -6460,7 +6460,7 @@
              fi;;
         *-*-hp*) TRYLINKINGWITHCXX="CXXSHARED= $CXX +z ";;
         *-*-darwin*) TRYLINKINGWITHCXX="CXXSHARED= $CXX -bundle -undefined suppress -flat_namespace";;
-        *-*-cygwin* | *-*-mingw*)
+        *-*-cygwin* | *-*-msys* | *-*-mingw*)
             if test "$GCC" = yes; then
                 TRYLINKINGWITHCXX="CXXSHARED= $CXX -shared "
             else
@@ -7495,7 +7495,7 @@
 
 # Cygwin (Windows) needs the library for dynamic linking
 case $host in
-*-*-cygwin* | *-*-mingw*) TCLDYNAMICLINKING="$TCLLIB";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) TCLDYNAMICLINKING="$TCLLIB";;
 *)TCLDYNAMICLINKING="";;
 esac
 
@@ -7663,7 +7663,7 @@
 
   # Cygwin (Windows) needs the library for dynamic linking
   case $host in
-  *-*-cygwin* | *-*-mingw*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw*)
     PYTHONDYNAMICLINKING="-L$PYLIB $PYLINK"
     DEFS="-DUSE_DL_IMPORT $DEFS"
     ;;
@@ -7861,7 +7861,7 @@
 
   # Cygwin (Windows) needs the library for dynamic linking
   case $host in
-  *-*-cygwin* | *-*-mingw*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw*)
     PYTHON3DYNAMICLINKING="-L$PYLIB $PY3LINK"
     DEFS="-DUSE_DL_IMPORT $DEFS"
     ;;
@@ -8032,7 +8032,7 @@
 
 # Cygwin (Windows) needs the library for dynamic linking
 case $host in
-*-*-cygwin* | *-*-mingw*) PERL5DYNAMICLINKING="-L$PERL5EXT -l$PERL5LIB";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) PERL5DYNAMICLINKING="-L$PERL5EXT -l$PERL5LIB";;
 *)PERL5DYNAMICLINKING="";;
 esac
 fi
@@ -8303,7 +8303,7 @@
 
   # Add in default installation directory on Windows for Cygwin
   case $host in
-  *-*-cygwin* | *-*-mingw*) JAVAINCDIR="c:/Program*Files/Java/jdk*/include d:/Program*Files/Java/jdk*/include c:/j2sdk*/include d:/j2sdk*/include c:/jdk*/include d:/jdk*/include $JAVAINCDIR";;
+  *-*-cygwin* | *-*-msys* | *-*-mingw*) JAVAINCDIR="c:/Program*Files/Java/jdk*/include d:/Program*Files/Java/jdk*/include c:/j2sdk*/include d:/j2sdk*/include c:/jdk*/include d:/jdk*/include $JAVAINCDIR";;
   *-*-darwin*) JAVAINCDIR="/System/Library/Frameworks/JavaVM.framework/Headers $JAVAINCDIR";;
   *);;
   esac
@@ -8341,7 +8341,7 @@
 
 # java.exe on Cygwin requires the Windows standard (Pascal) calling convention as it is a normal Windows executable and not a Cygwin built executable
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
     if test "$GCC" = yes; then
         JAVADYNAMICLINKING=" -mno-cygwin -mthreads -Wl,--add-stdcall-alias"
         JAVACFLAGS="-mno-cygwin -mthreads"
@@ -8361,7 +8361,7 @@
 
 # Java on Windows platforms including Cygwin doesn't use libname.dll, rather name.dll when loading dlls
 case $host in
-*-*-cygwin* | *-*-mingw*) JAVALIBRARYPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) JAVALIBRARYPREFIX="";;
 *)JAVALIBRARYPREFIX="lib";;
 esac
 
@@ -9241,7 +9241,7 @@
 fi
 
 case $host in
-*-*-cygwin* | *-*-mingw*)	RUBYDYNAMICLINKING="-L$RUBYLIB $RUBYLINK";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*)	RUBYDYNAMICLINKING="-L$RUBYLIB $RUBYLINK";;
 *)		RUBYDYNAMICLINKING="";;
 esac
 fi
@@ -10198,7 +10198,7 @@
 
 if test -z "$CSHARPCOMPILERBIN" ; then
   case $host in
-  *-*-cygwin* | *-*-mingw*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw*)
     # prefer Mono gmcs (.NET 2.0) over mcs (.NET 1.1) - note mcs-1.2.3 has major pinvoke bug
     for ac_prog in csc mono-csc gmcs mcs cscc
 do
@@ -10465,7 +10465,7 @@
 
 # Cygwin requires the Windows standard (Pascal) calling convention as it is a Windows executable and not a Cygwin built executable
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
     if test "$GCC" = yes; then
         CSHARPDYNAMICLINKING=" -mno-cygwin -mthreads -Wl,--add-stdcall-alias"
         CSHARPCFLAGS="-mno-cygwin -mthreads"
@@ -10481,7 +10481,7 @@
 
 # CSharp on Windows platforms including Cygwin doesn't use libname.dll, rather name.dll when loading dlls
 case $host in
-*-*-cygwin* | *-*-mingw*) CSHARPLIBRARYPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) CSHARPLIBRARYPREFIX="";;
 *)CSHARPLIBRARYPREFIX="lib";;
 esac
 
@@ -11413,7 +11413,7 @@
 
 # Do not prefix library file names with "lib" on Windows.
 case $host in
-*-*-cygwin* | *-*-mingw*) DLIBPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) DLIBPREFIX="";;
 *)DLIBPREFIX="lib";;
 esac
 
@@ -11609,7 +11609,7 @@
 # Translate path for native Windows compilers for use with 'make check'
 ROOT_DIR=`pwd`
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
   if (cygpath --mixed $ROOT_DIR) >/dev/null 2>/dev/null; then
     ROOT_DIR=`cygpath --mixed $ROOT_DIR`
   fi
@@ -11654,7 +11654,7 @@
 case $build in
         # Windows does not understand unix directories. Convert into a windows directory with drive letter.
         *-*-mingw*) SWIG_LIB_WIN_UNIX=`cmd //c echo $SWIG_LIB | sed -e "s/ *$//"`;; # This echo converts unix to mixed paths. Then zap unexpected trailing space.
-        *-*-cygwin*) SWIG_LIB_WIN_UNIX=`cygpath --mixed "$SWIG_LIB"`;;
+        *-*-cygwin* | *-*-msys*) SWIG_LIB_WIN_UNIX=`cygpath --mixed "$SWIG_LIB"`;;
         *) SWIG_LIB_WIN_UNIX="";;
 esac
 
diff -Naur swig-2.0.10-orig/configure.ac swig-2.0.10/configure.ac
--- swig-2.0.10-orig/configure.ac	2013-05-27 22:24:04.000000000 +0400
+++ swig-2.0.10/configure.ac	2013-07-23 13:49:28.552734400 +0400
@@ -168,7 +168,7 @@
 	case $host in
 	*-*-hp*) SO=.sl;;
 	*-*-darwin*) SO=.bundle;;
-	*-*-cygwin* | *-*-mingw*) SO=.dll;;
+	*-*-cygwin* | *-*-msys* | *-*-mingw*) SO=.dll;;
 	*) SO=.so;;
 	esac
 fi
@@ -183,7 +183,7 @@
 then
 	case $host in
 	*-*-aix*) LDSHARED="\$(srcdir)/ld_so_aix \$(CC)";;
-	*-*-cygwin* | *-*-mingw*)
+	*-*-cygwin* | *-*-msys* | *-*-mingw*)
             if test "$GCC" = yes; then
                 LDSHARED="$CC -shared"
             else
@@ -249,7 +249,7 @@
              fi;;
         *-*-hp*) TRYLINKINGWITHCXX="CXXSHARED= $CXX +z ";;
         *-*-darwin*) TRYLINKINGWITHCXX="CXXSHARED= $CXX -bundle -undefined suppress -flat_namespace";;
-        *-*-cygwin* | *-*-mingw*)
+        *-*-cygwin* | *-*-msys* | *-*-mingw*)
             if test "$GCC" = yes; then
                 TRYLINKINGWITHCXX="CXXSHARED= $CXX -shared "
             else
@@ -601,7 +601,7 @@
 
 # Cygwin (Windows) needs the library for dynamic linking
 case $host in
-*-*-cygwin* | *-*-mingw*) TCLDYNAMICLINKING="$TCLLIB";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) TCLDYNAMICLINKING="$TCLLIB";;
 *)TCLDYNAMICLINKING="";;
 esac
 
@@ -708,7 +708,7 @@
 
   # Cygwin (Windows) needs the library for dynamic linking
   case $host in
-  *-*-cygwin* | *-*-mingw*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw*)
     PYTHONDYNAMICLINKING="-L$PYLIB $PYLINK"
     DEFS="-DUSE_DL_IMPORT $DEFS"
     ;;
@@ -907,7 +907,7 @@
 
 # Cygwin (Windows) needs the library for dynamic linking
 case $host in
-*-*-cygwin* | *-*-mingw*) PERL5DYNAMICLINKING="-L$PERL5EXT -l$PERL5LIB";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) PERL5DYNAMICLINKING="-L$PERL5EXT -l$PERL5LIB";;
 *)PERL5DYNAMICLINKING="";;
 esac
 fi
@@ -1013,7 +1013,7 @@
 
   # Add in default installation directory on Windows for Cygwin
   case $host in
-  *-*-cygwin* | *-*-mingw*) JAVAINCDIR="c:/Program*Files/Java/jdk*/include d:/Program*Files/Java/jdk*/include c:/j2sdk*/include d:/j2sdk*/include c:/jdk*/include d:/jdk*/include $JAVAINCDIR";;
+  *-*-cygwin* | *-*-msys* | *-*-mingw*) JAVAINCDIR="c:/Program*Files/Java/jdk*/include d:/Program*Files/Java/jdk*/include c:/j2sdk*/include d:/j2sdk*/include c:/jdk*/include d:/jdk*/include $JAVAINCDIR";;
   *-*-darwin*) JAVAINCDIR="/System/Library/Frameworks/JavaVM.framework/Headers $JAVAINCDIR";;
   *);;
   esac
@@ -1046,7 +1046,7 @@
 
 # java.exe on Cygwin requires the Windows standard (Pascal) calling convention as it is a normal Windows executable and not a Cygwin built executable
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
     if test "$GCC" = yes; then
         JAVADYNAMICLINKING=" -mno-cygwin -mthreads -Wl,--add-stdcall-alias"
         JAVACFLAGS="-mno-cygwin -mthreads"
@@ -1066,7 +1066,7 @@
 
 # Java on Windows platforms including Cygwin doesn't use libname.dll, rather name.dll when loading dlls
 case $host in
-*-*-cygwin* | *-*-mingw*) JAVALIBRARYPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) JAVALIBRARYPREFIX="";;
 *)JAVALIBRARYPREFIX="lib";;
 esac
 
@@ -1395,7 +1395,7 @@
 fi
 
 case $host in
-*-*-cygwin* | *-*-mingw*)	RUBYDYNAMICLINKING="-L$RUBYLIB $RUBYLINK";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*)	RUBYDYNAMICLINKING="-L$RUBYLIB $RUBYLINK";;
 *)		RUBYDYNAMICLINKING="";;
 esac
 fi
@@ -1723,7 +1723,7 @@
 
 if test -z "$CSHARPCOMPILERBIN" ; then
   case $host in
-  *-*-cygwin* | *-*-mingw*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw*)
     # prefer Mono gmcs (.NET 2.0) over mcs (.NET 1.1) - note mcs-1.2.3 has major pinvoke bug
     AC_CHECK_PROGS(CSHARPCOMPILER, csc mono-csc gmcs mcs cscc)
     if test -n "$CSHARPCOMPILER" && test "$CSHARPCOMPILER" = "csc" ; then
@@ -1782,7 +1782,7 @@
 
 # Cygwin requires the Windows standard (Pascal) calling convention as it is a Windows executable and not a Cygwin built executable
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
     if test "$GCC" = yes; then
         CSHARPDYNAMICLINKING=" -mno-cygwin -mthreads -Wl,--add-stdcall-alias"
         CSHARPCFLAGS="-mno-cygwin -mthreads"
@@ -1798,7 +1798,7 @@
 
 # CSharp on Windows platforms including Cygwin doesn't use libname.dll, rather name.dll when loading dlls
 case $host in
-*-*-cygwin* | *-*-mingw*) CSHARPLIBRARYPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) CSHARPLIBRARYPREFIX="";;
 *)CSHARPLIBRARYPREFIX="lib";;
 esac
 
@@ -2177,7 +2177,7 @@
 
 # Do not prefix library file names with "lib" on Windows.
 case $host in
-*-*-cygwin* | *-*-mingw*) DLIBPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) DLIBPREFIX="";;
 *)DLIBPREFIX="lib";;
 esac
 
@@ -2373,7 +2373,7 @@
 # Translate path for native Windows compilers for use with 'make check'
 ROOT_DIR=`pwd`
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
   if (cygpath --mixed $ROOT_DIR) >/dev/null 2>/dev/null; then
     ROOT_DIR=`cygpath --mixed $ROOT_DIR`
   fi
@@ -2396,7 +2396,7 @@
 case $build in
         # Windows does not understand unix directories. Convert into a windows directory with drive letter.
         *-*-mingw*) SWIG_LIB_WIN_UNIX=`cmd //c echo $SWIG_LIB | sed -e "s/[ ]*$//"`;; # This echo converts unix to mixed paths. Then zap unexpected trailing space.
-        *-*-cygwin*) SWIG_LIB_WIN_UNIX=`cygpath --mixed "$SWIG_LIB"`;;
+        *-*-cygwin* | *-*-msys*) SWIG_LIB_WIN_UNIX=`cygpath --mixed "$SWIG_LIB"`;;
         *) SWIG_LIB_WIN_UNIX="";;
 esac
 AC_DEFINE_UNQUOTED(SWIG_LIB_WIN_UNIX, ["$SWIG_LIB_WIN_UNIX"], [Directory for SWIG system-independent libraries (Unix install on native Windows)])
diff -Naur swig-2.0.10-orig/Lib/swiglabels.swg swig-2.0.10/Lib/swiglabels.swg
--- swig-2.0.10-orig/Lib/swiglabels.swg	2013-05-27 22:24:04.000000000 +0400
+++ swig-2.0.10/Lib/swiglabels.swg	2013-07-23 13:42:09.338867200 +0400
@@ -89,7 +89,7 @@
 
 /* calling conventions for Windows */
 #ifndef SWIGSTDCALL
-# if defined(_WIN32) || defined(__WIN32__) || defined(__CYGWIN__)
+# if defined(_WIN32) || defined(__WIN32__)
 #   define SWIGSTDCALL __stdcall
 # else
 #   define SWIGSTDCALL
diff -Naur swig-2.0.10-orig/Tools/config/compile swig-2.0.10/Tools/config/compile
--- swig-2.0.10-orig/Tools/config/compile	2013-05-27 22:50:41.000000000 +0400
+++ swig-2.0.10/Tools/config/compile	2013-07-23 13:53:48.043945300 +0400
@@ -54,7 +54,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -68,7 +68,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur swig-2.0.10-orig/Tools/config/config.guess swig-2.0.10/Tools/config/config.guess
--- swig-2.0.10-orig/Tools/config/config.guess	2013-05-27 22:50:41.000000000 +0400
+++ swig-2.0.10/Tools/config/config.guess	2013-07-23 13:50:30.319336000 +0400
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur swig-2.0.10-orig/Tools/mkwindows.sh swig-2.0.10/Tools/mkwindows.sh
--- swig-2.0.10-orig/Tools/mkwindows.sh	2013-05-27 22:24:04.000000000 +0400
+++ swig-2.0.10/Tools/mkwindows.sh	2013-07-23 13:42:37.279296900 +0400
@@ -32,6 +32,7 @@
 mingw=`echo "$uname" | grep -i mingw`
 linux=`echo "$uname" | grep -i linux`
 cygwin=`echo "$uname" | grep -i cygwin`
+msys=`echo "$uname" | grep -i msys`
 if test "$mingw"; then
   echo "Building native Windows executable on MinGW";
   if test x$zip = x; then
@@ -51,9 +52,17 @@
         zip=zip
       fi
       compileflags="-O2 -mno-cygwin"
-    else
-      echo "Unknown platform. Requires either Linux or MinGW."
-      exit 1;
+	else
+      if test "$msys"; then
+        echo "Building native Windows executable on MSYS"
+        if test x$zip = x; then
+          zip=zip
+        fi
+        compileflags="-O2 -mno-cygwin"
+      else
+        echo "Unknown platform. Requires either Linux or MinGW."
+        exit 1;
+	  fi
     fi
   fi
 fi
