--- swig-3.0.7/configure.ac.orig	2015-08-03 00:51:59.000000000 +0300
+++ swig-3.0.7/configure.ac	2015-08-09 22:07:37.480000000 +0300
@@ -168,7 +168,7 @@
 	case $host in
 	*-*-hp*) SO=.sl;;
 	*-*-darwin*) SO=.bundle;;
-	*-*-cygwin* | *-*-mingw*) SO=.dll;;
+	*-*-cygwin* | *-*-msys* | *-*-mingw*) SO=.dll;;
 	*) SO=.so;;
 	esac
 fi
@@ -183,7 +183,7 @@
 then
 	case $host in
 	*-*-aix*) LDSHARED="\$(srcdir)/ld_so_aix \$(CC)";;
-	*-*-cygwin* | *-*-mingw*)
+	*-*-cygwin* | *-*-msys* | *-*-mingw*)
             if test "$GCC" = yes; then
                 LDSHARED="$CC -shared"
             else
@@ -249,7 +249,7 @@
              fi;;
         *-*-hp*) TRYLINKINGWITHCXX="CXXSHARED= $CXX +z ";;
         *-*-darwin*) TRYLINKINGWITHCXX="CXXSHARED= $CXX -bundle -undefined suppress -flat_namespace";;
-        *-*-cygwin* | *-*-mingw*)
+        *-*-cygwin* | *-*-msys* | *-*-mingw*)
             if test "$GCC" = yes; then
                 TRYLINKINGWITHCXX="CXXSHARED= $CXX -shared "
             else
@@ -579,7 +579,7 @@
 
 # Cygwin (Windows) needs the library for dynamic linking
 case $host in
-*-*-cygwin* | *-*-mingw*) TCLDYNAMICLINKING="$TCLLIB";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) TCLDYNAMICLINKING="$TCLLIB";;
 *)TCLDYNAMICLINKING="";;
 esac
 
@@ -729,7 +729,7 @@
 
   # Cygwin (Windows) needs the library for dynamic linking
   case $host in
-  *-*-cygwin* | *-*-mingw*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw*)
     PYTHONDYNAMICLINKING="-L$PYLIB $PYLINK"
     DEFS="-DUSE_DL_IMPORT $DEFS"
     ;;
@@ -887,7 +887,7 @@
 
   # Cygwin (Windows) needs the library for dynamic linking
   case $host in
-  *-*-cygwin* | *-*-mingw*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw*)
     PYTHON3DYNAMICLINKING="-L$PY3LIB $PY3LINK"
     DEFS="-DUSE_DL_IMPORT $DEFS"
     ;;
@@ -997,7 +997,7 @@
 
 # Cygwin (Windows) needs the library for dynamic linking
 case $host in
-*-*-cygwin* | *-*-mingw*) PERL5DYNAMICLINKING="-L$PERL5EXT -l$PERL5LIB";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) PERL5DYNAMICLINKING="-L$PERL5EXT -l$PERL5LIB";;
 *)PERL5DYNAMICLINKING="";;
 esac
 fi
@@ -1205,7 +1205,7 @@
 dnl If JAVA_HOME is defined and contains everything needed, as it's supposed to,
 dnl just use it. Otherwise try to locate the necessary files in the path.
 case $host in
-  *-*-cygwin*)
+  *-*-cygwin*|*-*-msys*)
     dnl Ensure that JAVA_HOME is in the form we can use, it's typically set to Windows path by the installer.
     if test -n "$JAVA_HOME"; then
       JAVA_HOME=`cygpath --mixed "$JAVA_HOME"`
@@ -1298,7 +1298,7 @@
 
     # Add in default installation directory on Windows for Cygwin
     case $host in
-    *-*-cygwin* | *-*-mingw*) JAVAINCDIR="c:/Program*Files*/Java/jdk*/include d:/Program*Files*/Java/jdk*/include c:/j2sdk*/include d:/j2sdk*/include c:/jdk*/include d:/jdk*/include $JAVAINCDIR";;
+    *-*-cygwin* | *-*-mingw* | *-*-msys*) JAVAINCDIR="c:/Program*Files*/Java/jdk*/include d:/Program*Files*/Java/jdk*/include c:/j2sdk*/include d:/j2sdk*/include c:/jdk*/include d:/jdk*/include $JAVAINCDIR";;
     *-*-darwin*) JAVAINCDIR="$JAVA_OSX_STD_INCDIR $JAVAINCDIR";;
     *);;
     esac
@@ -1345,7 +1345,7 @@
 
 # java.exe on Cygwin requires the Windows standard (Pascal) calling convention as it is a normal Windows executable and not a Cygwin built executable
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
     if test "$GCC" = yes; then
         JAVADYNAMICLINKING="$GCC_MNO_CYGWIN -mthreads -Wl,--add-stdcall-alias"
         JAVACFLAGS="$GCC_MNO_CYGWIN -mthreads"
@@ -1365,7 +1365,7 @@
 
 # Java on Windows platforms including Cygwin doesn't use libname.dll, rather name.dll when loading dlls
 case $host in
-*-*-cygwin* | *-*-mingw*) JAVALIBRARYPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) JAVALIBRARYPREFIX="";;
 *)JAVALIBRARYPREFIX="lib";;
 esac
 
@@ -1891,7 +1891,7 @@
 fi
 
 case $host in
-*-*-cygwin* | *-*-mingw*)	RUBYDYNAMICLINKING="-L$RUBYLIB $RUBYLINK";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*)	RUBYDYNAMICLINKING="-L$RUBYLIB $RUBYLINK";;
 *)		RUBYDYNAMICLINKING="";;
 esac
 fi
@@ -2181,7 +2181,7 @@
 
 if test -z "$CSHARPCOMPILERBIN" ; then
   case $host in
-  *-*-cygwin* | *-*-mingw*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw*)
     # prefer Mono gmcs (.NET 2.0) over mcs (.NET 1.1) - note mcs-1.2.3 has major pinvoke bug
     AC_CHECK_PROGS(CSHARPCOMPILER, csc mono-csc gmcs mcs cscc)
     if test -n "$CSHARPCOMPILER" && test "$CSHARPCOMPILER" = "csc" ; then
@@ -2238,7 +2238,7 @@
 
 # Cygwin requires the Windows standard (Pascal) calling convention as it is a Windows executable and not a Cygwin built executable
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
     if test "$GCC" = yes; then
         CSHARPDYNAMICLINKING="$GCC_MNO_CYGWIN -mthreads -Wl,--add-stdcall-alias"
         CSHARPCFLAGS="$GCC_MNO_CYGWIN -mthreads"
@@ -2254,7 +2254,7 @@
 
 # CSharp on Windows platforms including Cygwin doesn't use libname.dll, rather name.dll when loading dlls
 case $host in
-*-*-cygwin* | *-*-mingw*) CSHARPLIBRARYPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) CSHARPLIBRARYPREFIX="";;
 *)CSHARPLIBRARYPREFIX="lib";;
 esac
 
@@ -2672,7 +2672,7 @@
 
 # Do not prefix library file names with "lib" on Windows.
 case $host in
-*-*-cygwin* | *-*-mingw*) DLIBPREFIX="";;
+*-*-cygwin* | *-*-msys* | *-*-mingw*) DLIBPREFIX="";;
 *)DLIBPREFIX="lib";;
 esac
 
@@ -2883,13 +2883,13 @@
 dnl Using "mixed" path like we do here allows the path to work in both cases.
 case $host in
   *-*-mingw* ) ABS_SRCDIR=`${srcdir}/Tools/convertpath -m $ABS_SRCDIR` ;;
-  *-*-cygwin* ) ABS_SRCDIR=`cygpath --mixed $ABS_SRCDIR` ;;
+  *-*-cygwin* | *-*-msys*) ABS_SRCDIR=`cygpath --mixed $ABS_SRCDIR` ;;
 esac
 
 # Root directory
 ROOT_DIR=`pwd`
 case $host in
-*-*-cygwin*)
+*-*-cygwin* | *-*-msys*)
   # Translate path for native Windows compilers for use with 'make check'
   if (cygpath --mixed $ROOT_DIR) >/dev/null 2>/dev/null; then
     ROOT_DIR=`cygpath --mixed $ROOT_DIR`
@@ -2898,7 +2898,7 @@
 esac
 
 case $host in
-*-*-cygwin* | *-*-mingw*)
+*-*-cygwin* | *-*-msys* | *-*-mingw*)
   # Extra files generated by some Windows compilers
   EXTRA_CLEAN="*.stackdump *.exp *.lib *.pdb *.ilk"
   ;;
@@ -2918,7 +2918,7 @@
 case $build in
   # Windows does not understand unix directories. Convert into a windows directory with drive letter.
   *-*-mingw*) SWIG_LIB_WIN_UNIX=`${srcdir}/Tools/convertpath -m $SWIG_LIB`;;
-  *-*-cygwin*) SWIG_LIB_WIN_UNIX=`cygpath --mixed "$SWIG_LIB"`;;
+  *-*-cygwin* | *-*-msys*) SWIG_LIB_WIN_UNIX=`cygpath --mixed "$SWIG_LIB"`;;
   *) SWIG_LIB_WIN_UNIX="";;
 esac
 AC_DEFINE_UNQUOTED(SWIG_LIB_WIN_UNIX, ["$SWIG_LIB_WIN_UNIX"], [Directory for SWIG system-independent libraries (Unix install on native Windows)])
diff -Naur swig-3.0.7-orig/Lib/swiglabels.swg swig-3.0.7/Lib/swiglabels.swg
--- swig-3.0.7-orig/Lib/swiglabels.swg	2013-05-27 22:24:04.000000000 +0400
+++ swig-3.0.7/Lib/swiglabels.swg	2013-07-23 13:42:09.338867200 +0400
@@ -89,7 +89,7 @@
 
 /* calling conventions for Windows */
 #ifndef SWIGSTDCALL
-# if defined(_WIN32) || defined(__WIN32__) || defined(__CYGWIN__)
+# if defined(_WIN32) || defined(__WIN32__)
 #   define SWIGSTDCALL __stdcall
 # else
 #   define SWIGSTDCALL
diff -Naur swig-3.0.7-orig/Tools/config/compile swig-3.0.7/Tools/config/compile
--- swig-3.0.7-orig/Tools/config/compile	2013-05-27 22:50:41.000000000 +0400
+++ swig-3.0.7/Tools/config/compile	2013-07-23 13:53:48.043945300 +0400
@@ -54,7 +54,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -68,7 +68,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur swig-3.0.7-orig/Tools/config/config.guess swig-3.0.7/Tools/config/config.guess
--- swig-3.0.7-orig/Tools/config/config.guess	2013-05-27 22:50:41.000000000 +0400
+++ swig-3.0.7/Tools/config/config.guess	2013-07-23 13:50:30.319336000 +0400
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur swig-3.0.7-orig/Tools/mkwindows.sh swig-3.0.7/Tools/mkwindows.sh
--- swig-3.0.7-orig/Tools/mkwindows.sh	2013-05-27 22:24:04.000000000 +0400
+++ swig-3.0.7/Tools/mkwindows.sh	2013-07-23 13:42:37.279296900 +0400
@@ -32,6 +32,7 @@
 mingw=`echo "$uname" | grep -i mingw`
 linux=`echo "$uname" | grep -i linux`
 cygwin=`echo "$uname" | grep -i cygwin`
+msys=`echo "$uname" | grep -i msys`
 if test "$mingw"; then
   echo "Building native Windows executable on MinGW";
   if test x$zip = x; then
@@ -51,9 +52,17 @@
         zip=zip
       fi
       compileflags="-O2 -mno-cygwin"
-    else
-      echo "Unknown platform. Requires either Linux or MinGW."
-      exit 1;
+	else
+      if test "$msys"; then
+        echo "Building native Windows executable on MSYS"
+        if test x$zip = x; then
+          zip=zip
+        fi
+        compileflags="-O2 -mno-cygwin"
+      else
+        echo "Unknown platform. Requires either Linux or MinGW."
+        exit 1;
+	  fi
     fi
   fi
 fi
