--- origsrc/libev-4.22/Makefile.am	2014-05-22 13:53:16.000000000 -0500
+++ src/libev-4.22/Makefile.am	2016-01-28 15:47:31.578108500 -0600
@@ -14,7 +14,7 @@ include_HEADERS = ev.h ev++.h event.h
 lib_LTLIBRARIES = libev.la
 
 libev_la_SOURCES = ev.c event.c
-libev_la_LDFLAGS = -version-info $(VERSION_INFO)
+libev_la_LDFLAGS = -version-info $(VERSION_INFO) -no-undefined
 
 ev.3: ev.pod
 	pod2man -n LIBEV -r "libev-$(VERSION)" -c "libev - high performance full featured event loop" -s3 <$< >$@
