From 6d6f9ff84dd37616510629935c47645cc83b544d Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Fri, 9 Dec 2022 16:16:53 +0100
Subject: [PATCH 39/N] fixup! Rename dll from cygwin to msys

---
 winsup/cygwin/cygwin.din | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/cygwin/cygwin.din b/winsup/cygwin/cygwin.din
index d24c7d2..3167f04 100644
--- a/winsup/cygwin/cygwin.din
+++ b/winsup/cygwin/cygwin.din
@@ -1,4 +1,4 @@
-LIBRARY "cygwin1.dll" BASE=0x180040000
+LIBRARY "msys-2.0.dll" BASE=0x180040000
 
 EXPORTS
 # Exported variables
