From f9bbaaab6e48d670d2c854479f96b12d6d19d1e7 Mon Sep 17 00:00:00 2001
From: Jon Turney <jon.turney@dronecode.org.uk>
Date: Sun, 29 Dec 2019 22:26:41 +0000
Subject: [PATCH 5/5] Enable CLI styling by default on Cygwin

---
 gdb/cli/cli-style.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdb/cli/cli-style.c b/gdb/cli/cli-style.c
index 7e20c1b4826..f3d98ae4091 100644
--- a/gdb/cli/cli-style.c
+++ b/gdb/cli/cli-style.c
@@ -25,7 +25,7 @@
 
 /* True if styling is enabled.  */
 
-#if defined (__MSDOS__) || defined (__CYGWIN__)
+#if defined (__MSDOS__)
 bool cli_styling = false;
 #else
 bool cli_styling = true;
-- 
2.21.0

