--- libunistring-0.9.3/configure.ac	2010-03-29 04:35:26.000000000 +0200
+++ libunistring-0.9.3/configure.ac	2011-10-07 17:42:13.943985300 +0200
@@ -61,7 +61,7 @@ LT_LANG([Windows Resource])
 dnl On mingw and Cygwin, we can activate special Makefile rules which add
 dnl version information to the shared libraries and executables.
 case "$host_os" in
-  mingw* | cygwin*) is_woe32=yes ;;
+  mingw*) is_woe32=yes ;;
   *) is_woe32=no ;;
 esac
 AM_CONDITIONAL([WOE32], [test $is_woe32 = yes])
@@ -79,7 +79,7 @@ dnl    must arrange to define the auxili
 dnl    exported variables _also_ in the static library.
 if test "$enable_shared" = yes; then
   case "$host_os" in
-    mingw* | cygwin*) is_woe32dll=yes ;;
+    mingw*) is_woe32dll=yes ;;
     *) is_woe32dll=no ;;
   esac
 else
--- libunistring-1.0/m4/woe32-dll.m4.orig	2022-05-13 23:14:12.117697400 +0200
+++ libunistring-1.0/m4/woe32-dll.m4	2022-05-13 23:14:53.500794800 +0200
@@ -30,7 +30,7 @@
 [
   AC_REQUIRE([AC_CANONICAL_HOST])
   case "$host_os" in
-    mingw* | cygwin*)
+    mingw*)
       AC_CACHE_CHECK([for auto-import of symbols],
         [gl_cv_ld_autoimport],
         [dnl --disable-auto-import is unsupported in MSVC and in MSVC/clang.
--- libunistring-1.0/lib/Makefile.am.orig	2022-05-14 08:56:53.657968900 +0200
+++ libunistring-1.0/lib/Makefile.am	2022-05-14 08:57:05.616166100 +0200
@@ -265,8 +265,8 @@
 # Tell the mingw or Cygwin linker which symbols to export.
 if WOE32DLL
 libunistring_la_SOURCES += ../woe32dll/unistring-exports.c
-libunistring_la_LDFLAGS += -Wl,--export-all-symbols
 endif
+libunistring_la_LDFLAGS += -Wl,--export-all-symbols
 
 # Hide undesired symbols that are defined by libunistring_la_SOURCES or
 # libunistring_la_LIBADD from the global namespace, by prefixing them with
