diff -Naur a/Include/pyerrors.h b/Include/pyerrors.h
--- a/Include/pyerrors.h	2014-05-31 22:58:38.000000000 +0400
+++ b/Include/pyerrors.h	2014-06-04 10:16:43.573000000 +0400
@@ -231,7 +231,7 @@
 PyAPI_FUNC(void) PyErr_SetInterrupt(void);
 
 /* In signalmodule.c */
-int PySignal_SetWakeupFd(int fd);
+PyAPI_FUNC(int) PySignal_SetWakeupFd(int fd);
 
 /* Support for adding program text to SyntaxErrors */
 PyAPI_FUNC(void) PyErr_SyntaxLocation(const char *, int);
