diff -Naur a/setup.py b/setup.py
--- a/setup.py	2014-06-04 10:16:50.343400000 +0400
+++ b/setup.py	2014-06-04 10:16:55.039000000 +0400
@@ -551,7 +551,7 @@
 
         # Check for MacOS X, which doesn't need libm.a at all
         math_libs = ['m']
-        if host_platform in ['darwin', 'beos']:
+        if host_platform in ['darwin', 'beos', 'cygwin']:
             math_libs = []
 
         # XXX Omitted modules: gl, pure, dl, SGI-specific modules
