diff -r c087ac6fc171 Modules/_ctypes/_ctypes.c
--- a/Modules/_ctypes/_ctypes.c	Wed Jan 28 10:52:49 2015 +0200
+++ b/Modules/_ctypes/_ctypes.c	Wed Jan 28 19:53:12 2015 +0900
@@ -637,17 +637,17 @@ CDataType_in_dll(PyObject *type, PyObjec
         return NULL;
     }
 #else
     address = (void *)ctypes_dlsym(handle, name);
     if (!address) {
 #ifdef __CYGWIN__
 /* dlerror() isn't very helpful on cygwin */
         PyErr_Format(PyExc_ValueError,
-                     "symbol '%s' not found (%s) ",
+                     "symbol '%s' not found",
                      name);
 #else
         PyErr_SetString(PyExc_ValueError, ctypes_dlerror());
 #endif
         return NULL;
     }
 #endif
     return PyCData_AtAddress(type, address);
