--- a/Lib/posixpath.py	2014-02-21 10:34:28.539709900 +0000
+++ b/Lib/posixpath.py	2014-02-21 10:37:02.070491400 +0000
@@ -58,7 +58,11 @@
 
 def isabs(s):
     """Test whether a path is absolute"""
-    return s.startswith('/')
+    return s.startswith('/') or \
+	       (sys.platform == 'msys' and \
+		    len(s) > 2 and \
+			s[1] == ':' and \
+			(s[2] == os.sep or s[2] == '\\'))
 
 
 # Join pathnames.
