diff -urN a/Lib/distutils/unixccompiler.py b/Lib/distutils/unixccompiler.py
--- a/Lib/distutils/unixccompiler.py	2013-10-20 19:33:46.896828100 +0100
+++ b/Lib/distutils/unixccompiler.py	2013-10-20 19:34:22.585869400 +0100
@@ -82,6 +82,7 @@
     static_lib_format = shared_lib_format = dylib_lib_format = "lib%s%s"
     if sys.platform == "cygwin":
         exe_extension = ".exe"
+        dylib_lib_extension = ".dll.a"
 
     def preprocess(self, source,
                    output_file=None, macros=None, include_dirs=None,
