diff -urN a/Include/pyerrors.h b/Include/pyerrors.h
--- a/Include/pyerrors.h	2013-10-20 19:30:57.390132900 +0100
+++ b/Include/pyerrors.h	2013-10-20 19:31:33.580202800 +0100
@@ -231,7 +231,7 @@
 PyAPI_FUNC(void) PyErr_SetInterrupt(void);
 
 /* In signalmodule.c */
-int PySignal_SetWakeupFd(int fd);
+PyAPI_FUNC(int) PySignal_SetWakeupFd(int fd);
 
 /* Support for adding program text to SyntaxErrors */
 PyAPI_FUNC(void) PyErr_SyntaxLocation(const char *, int);
