diff -urN a/setup.py b/setup.py
--- a/setup.py	2013-10-20 19:35:36.296085400 +0100
+++ b/setup.py	2013-10-20 19:36:03.716653800 +0100
@@ -549,7 +549,7 @@
 
         # Check for MacOS X, which doesn't need libm.a at all
         math_libs = ['m']
-        if host_platform in ['darwin', 'beos']:
+        if host_platform in ['darwin', 'beos', 'cygwin']:
             math_libs = []
 
         # XXX Omitted modules: gl, pure, dl, SGI-specific modules
