diff -urN a/Modules/_sqlite/connection.c b/Modules/_sqlite/connection.c
--- a/Modules/_sqlite/connection.c	2013-10-20 19:37:19.768003700 +0100
+++ b/Modules/_sqlite/connection.c	2013-10-20 19:37:56.224088800 +0100
@@ -781,9 +781,12 @@
     }
 
 error:
+    {
 #ifdef WITH_THREAD
     PyGILState_Release(threadstate);
 #endif
+      ;
+    }
 }
 
 static void _pysqlite_drop_unused_statement_references(pysqlite_Connection* self)
