diff -urN a/Lib/uuid.py b/Lib/uuid.py
--- a/Lib/uuid.py	2013-10-20 19:40:07.732610700 +0100
+++ b/Lib/uuid.py	2013-10-20 19:40:39.586432600 +0100
@@ -408,6 +408,8 @@
             _uuid_generate_random = lib.uuid_generate_random
         if hasattr(lib, 'uuid_generate_time'):
             _uuid_generate_time = lib.uuid_generate_time
+            if _uuid_generate_random is not None:
+                break  # found everything we were looking for
 
     # The uuid_generate_* functions are broken on MacOS X 10.5, as noted
     # in issue #8621 the function generates the same sequence of values
