diff -durN yaml-0.1.7.orig/src/Makefile.am yaml-0.1.7/src/Makefile.am
--- yaml-0.1.7.orig/src/Makefile.am	2016-08-29 07:56:34.000000000 +0800
+++ yaml-0.1.7/src/Makefile.am	2017-05-12 11:19:15.890625000 +0800
@@ -1,4 +1,4 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include -Wall
 lib_LTLIBRARIES = libyaml.la
 libyaml_la_SOURCES = yaml_private.h api.c reader.c scanner.c parser.c loader.c writer.c emitter.c dumper.c
-libyaml_la_LDFLAGS = -no-undefined -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE)
+libyaml_la_LDFLAGS = -no-undefined -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE) -no-undefined
diff -durN yaml-0.1.7.orig/src/Makefile.in yaml-0.1.7/src/Makefile.in
--- yaml-0.1.7.orig/src/Makefile.in	2016-08-29 07:56:49.000000000 +0800
+++ yaml-0.1.7/src/Makefile.in	2017-05-12 11:19:57.546875000 +0800
@@ -323,7 +323,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)/include -Wall
 lib_LTLIBRARIES = libyaml.la
 libyaml_la_SOURCES = yaml_private.h api.c reader.c scanner.c parser.c loader.c writer.c emitter.c dumper.c
-libyaml_la_LDFLAGS = -no-undefined -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE)
+libyaml_la_LDFLAGS = -no-undefined -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE) -no-undefined
 all: all-am
 
 .SUFFIXES:
