diff -durN yaml-0.1.7.orig/config/config.guess yaml-0.1.7/config/config.guess
--- yaml-0.1.7.orig/config/config.guess	2017-05-12 11:17:41.015625000 +0800
+++ yaml-0.1.7/config/config.guess	2017-05-12 11:35:35.640625000 +0800
@@ -867,6 +867,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
