diff -Naur mpc-1.0.1/ar-lib mpc-1.0.1-p/ar-lib
--- mpc-1.0.1/ar-lib	2012-07-22 19:40:22.000000000 +0400
+++ mpc-1.0.1-p/ar-lib	2013-02-06 08:56:40.088867200 +0400
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
diff -Naur mpc-1.0.1/config.guess mpc-1.0.1-p/config.guess
--- mpc-1.0.1/config.guess	2012-07-22 19:40:22.000000000 +0400
+++ mpc-1.0.1-p/config.guess	2013-02-06 08:57:38.187500000 +0400
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+	amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur mpc-1.0.1/configure.ac mpc-1.0.1-p/configure.ac
--- mpc-1.0.1/configure.ac	2012-09-06 14:05:01.000000000 +0400
+++ mpc-1.0.1-p/configure.ac	2013-02-06 09:02:56.198242200 +0400
@@ -225,7 +225,7 @@
 AC_SUBST(MPC_LDFLAGS)
 AC_LIBTOOL_WIN32_DLL
 case $host in
-  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-os2*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw* | *-*-pw32* | *-*-os2*)
      MPC_WINDOWS
 esac
 
