--- origsrc/mc-4.7.5.2/lib/tty/tty-ncurses.c	2011-04-08 03:55:56.000000000 -0500
+++ src/mc-4.7.5.2/lib/tty/tty-ncurses.c	2011-07-10 21:28:07.525141000 -0500
@@ -52,7 +52,9 @@
 
 /* include at last !!! */
 #ifdef WANT_TERM_H
-#ifdef HAVE_NCURSES_TERM_H
+#if defined HAVE_NCURSESW_CURSES_H
+#include <ncursesw/term.h>
+#elif defined HAVE_NCURSES_TERM_H
 #include <ncurses/term.h>
 #else
 #include <term.h>
