diff --git mc-4.8.19.orig/configure mc-4.8.19/configure
index a89d720..e682489 100755
--- mc-4.8.19.orig/configure
+++ mc-4.8.19/configure
@@ -15789,7 +15789,7 @@ fi
 
     # on cygwin, the linker does not accept the "-z" option
     case $host_os in
-        cygwin*)
+        cygwin* | msys*)
             TESTS_LDFLAGS="-Wl,--allow-multiple-definition"
             ;;
         *)
diff --git mc-4.8.19.orig/m4/threadlib.m4 mc-4.8.19/m4/threadlib.m4
index 05cc4ff..6f85014 100644
--- mc-4.8.19.orig/m4/threadlib.m4
+++ mc-4.8.19/m4/threadlib.m4
@@ -61,7 +61,7 @@ changequote(,)dnl
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
diff --git mc-4.8.19.orig/m4.include/mc-tests.m4 mc-4.8.19/m4.include/mc-tests.m4
index 742ecb0..23207c4 100644
--- mc-4.8.19.orig/m4.include/mc-tests.m4
+++ mc-4.8.19/m4.include/mc-tests.m4
@@ -44,7 +44,7 @@ AC_DEFUN([mc_UNIT_TESTS],[
 
     # on cygwin, the linker does not accept the "-z" option
     case $host_os in
-        cygwin*)
+        cygwin* | msys*)
             TESTS_LDFLAGS="-Wl,--allow-multiple-definition"
             ;;
         *)
