diff -Naur mc-4.8.10-orig/config/compile mc-4.8.10/config/compile
--- mc-4.8.10-orig/config/compile	2012-09-26 18:03:18.000000000 +0400
+++ mc-4.8.10/config/compile	2013-08-22 19:25:17.311523500 +0400
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur mc-4.8.10-orig/config/config.guess mc-4.8.10/config/config.guess
--- mc-4.8.10-orig/config/config.guess	2012-09-26 18:03:18.000000000 +0400
+++ mc-4.8.10/config/config.guess	2013-08-22 19:25:57.655273500 +0400
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur mc-4.8.10-orig/config/config.rpath mc-4.8.10/config/config.rpath
--- mc-4.8.10-orig/config/config.rpath	2013-08-02 01:52:17.000000000 +0400
+++ mc-4.8.10/config/config.rpath	2013-08-22 19:26:49.780273500 +0400
@@ -64,7 +64,7 @@
           ;;
       esac
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -138,7 +138,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -188,7 +188,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -332,7 +332,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -523,7 +523,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur mc-4.8.10-orig/m4.include/mc-tests.m4 mc-4.8.10/m4.include/mc-tests.m4
--- mc-4.8.10-orig/m4.include/mc-tests.m4	2013-06-26 01:29:14.000000000 +0400
+++ mc-4.8.10/m4.include/mc-tests.m4	2013-08-22 19:36:58.295898500 +0400
@@ -25,7 +25,7 @@
 
     # on cygwin, the linker does not accept the "-z" option
     case $host_os in
-        cygwin*)
+        cygwin* | msys*)
             TESTS_LDFLAGS="-Wl,--allow-multiple-definition"
             ;;
         *)
