--- origsrc/mc-4.8.10/src/filemanager/filegui.c	2013-08-02 17:02:39.000000000 +0200
+++ src/mc-4.8.10/src/filemanager/filegui.c	2013-08-04 09:12:04.563871900 +0200
@@ -284,8 +284,8 @@ statvfs_works (void)
     if (statvfs_works_cache < 0)
         statvfs_works_cache = (uname (&name) == 0 && 0 <= str_verscmp (name.release, "2.6.36"));
     return statvfs_works_cache;
-}
 #endif
+}
 #endif
 
 /* --------------------------------------------------------------------------------------------- */
--- origsrc/mc-4.8.10/src/filemanager/mountlist.c	2013-08-02 17:02:39.000000000 +0200
+++ src/mc-4.8.10/src/filemanager/mountlist.c	2013-08-04 09:12:04.578872700 +0200
@@ -229,8 +229,7 @@
 #ifdef MOUNTED_GETMNTENT1
 #define ME_DUMMY(Fs_name, Fs_type, Fs_ent)      \
   (ME_DUMMY_0 (Fs_name, Fs_type)                \
-   || (strcmp (Fs_type, "none") == 0            \
-       && !hasmntopt (Fs_ent, "bind")))
+   || (strcmp (Fs_type, "none") == 0 ))           
 #else
 #define ME_DUMMY(Fs_name, Fs_type)              \
   (ME_DUMMY_0 (Fs_name, Fs_type) || strcmp (Fs_type, "none") == 0)
@@ -242,7 +241,7 @@
 /* All cygwin mount points include ':' or start with '//'; so it
    requires a native Windows call to determine remote disks.  */
 static int
-me_remote (char const *fs_name, char const *fs_type _GL_UNUSED)
+me_remote (char const *fs_name, char const *fs_type )
 {
     if (fs_name[0] && fs_name[1] == ':')
     {
