--- a/terms.c
+++ b/terms.c
@@ -224,7 +224,7 @@
 	}
 #ifdef SUPPORT_WIN9X_CONSOLE_MBCS
 	check_win9x();
-	if (isWin95 && ttyslot() != -1) {
+	if (0) {
 	    isLocalConsole = 0;
 	}
 #endif
@@ -2114,7 +2114,7 @@
  * no one complains
  */
 #ifdef __CYGWIN__
-	if (isLocalConsole && title_str == CYGWIN_TITLE) {
+	if (isLocalConsole && 0) {
 	    Str buff;
 	    buff = Sprintf(title_str, s);
 	    if (buff->length > 1024) {
