diff -Naur grep-2.23-orig/build-aux/ar-lib grep-2.23/build-aux/ar-lib
--- grep-2.23-orig/build-aux/ar-lib	2016-02-04 23:08:58.000000000 +0300
+++ grep-2.23/build-aux/ar-lib	2016-03-30 08:53:45.525087200 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -65,7 +65,7 @@
 	mingw)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin)
+	cygwin|msys)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine)
diff -Naur grep-2.23-orig/build-aux/compile grep-2.23/build-aux/compile
--- grep-2.23-orig/build-aux/compile	2016-02-04 23:08:58.000000000 +0300
+++ grep-2.23/build-aux/compile	2016-03-30 08:53:21.190520900 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur grep-2.23-orig/build-aux/config.guess grep-2.23/build-aux/config.guess
--- grep-2.23-orig/build-aux/config.guess	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/build-aux/config.guess	2016-03-30 08:53:21.207519200 +0300
@@ -903,6 +903,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
diff -Naur grep-2.23-orig/build-aux/config.rpath grep-2.23/build-aux/config.rpath
--- grep-2.23-orig/build-aux/config.rpath	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/build-aux/config.rpath	2016-03-30 08:53:21.249515000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur grep-2.23-orig/m4/btowc.m4 grep-2.23/m4/btowc.m4
--- grep-2.23-orig/m4/btowc.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/btowc.m4	2016-03-30 08:53:21.347505200 +0300
@@ -49,7 +49,7 @@
 changequote(,)dnl
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes on native Windows.
              mingw*)  gl_cv_func_btowc_nul="guessing yes" ;;
                       # Guess yes otherwise.
diff -Naur grep-2.23-orig/m4/double-slash-root.m4 grep-2.23/m4/double-slash-root.m4
--- grep-2.23-orig/m4/double-slash-root.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/double-slash-root.m4	2016-03-30 08:53:21.376502300 +0300
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur grep-2.23-orig/m4/dup2.m4 grep-2.23/m4/dup2.m4
--- grep-2.23-orig/m4/dup2.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/dup2.m4	2016-03-30 08:53:21.403499600 +0300
@@ -69,7 +69,7 @@
       [case "$host_os" in
          mingw*) # on this platform, dup2 always returns 0 for success
            gl_cv_func_dup2_works="guessing no" ;;
-         cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+         cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
            gl_cv_func_dup2_works="guessing no" ;;
          aix* | freebsd*)
                  # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
diff -Naur grep-2.23-orig/m4/fcntl.m4 grep-2.23/m4/fcntl.m4
--- grep-2.23-orig/m4/fcntl.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/fcntl.m4	2016-03-30 08:54:28.942745000 +0300
@@ -70,7 +70,7 @@
          [gl_cv_func_fcntl_f_dupfd_works=yes],
          [gl_cv_func_fcntl_f_dupfd_works=no],
          [case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac])])
diff -Naur grep-2.23-orig/m4/getcwd.m4 grep-2.23/m4/getcwd.m4
--- grep-2.23-orig/m4/getcwd.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/getcwd.m4	2016-03-30 08:53:21.438496100 +0300
@@ -54,7 +54,7 @@
                            # Guess yes on musl systems.
             *-musl*)       gl_cv_func_getcwd_null="guessing yes";;
                            # Guess yes on Cygwin.
-            cygwin*)       gl_cv_func_getcwd_null="guessing yes";;
+            cygwin* | msys*) gl_cv_func_getcwd_null="guessing yes";;
                            # If we don't know, obey --enable-cross-guesses.
             *)             gl_cv_func_getcwd_null="$gl_cross_guess_normal";;
           esac
diff -Naur grep-2.23-orig/m4/getdtablesize.m4 grep-2.23/m4/getdtablesize.m4
--- grep-2.23-orig/m4/getdtablesize.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/getdtablesize.m4	2016-03-30 08:53:21.460493900 +0300
@@ -42,7 +42,7 @@
              [gl_cv_func_getdtablesize_works=yes],
              [gl_cv_func_getdtablesize_works=no],
              [case "$host_os" in
-                cygwin*) # on cygwin 1.5.25, getdtablesize() automatically grows
+                cygwin* | msys*) # on cygwin 1.5.25, getdtablesize() automatically grows
                   gl_cv_func_getdtablesize_works="guessing no" ;;
                 *) gl_cv_func_getdtablesize_works="guessing yes" ;;
               esac
diff -Naur grep-2.23-orig/m4/malloc.m4 grep-2.23/m4/malloc.m4
--- grep-2.23-orig/m4/malloc.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/malloc.m4	2016-03-30 08:53:21.500489900 +0300
@@ -25,7 +25,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | *-musl* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull="guessing yes" ;;
           # If we don't know, obey --enable-cross-guesses.
           *) ac_cv_func_malloc_0_nonnull="$gl_cross_guess_normal" ;;
diff -Naur grep-2.23-orig/m4/printf.m4 grep-2.23/m4/printf.m4
--- grep-2.23-orig/m4/printf.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/printf.m4	2016-03-30 08:53:21.528487100 +0300
@@ -818,7 +818,7 @@
            openbsd*)        gl_cv_func_printf_directive_ls="guessing yes";;
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
-           cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           cygwin* | msys*)         gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
                             # Guess no on Android.
            linux*-android*) gl_cv_func_printf_directive_ls="guessing no";;
@@ -902,7 +902,7 @@
         [
 changequote(,)dnl
          case "$host_os" in
-           cygwin*)         gl_cv_func_printf_flag_grouping="guessing no";;
+           cygwin* | msys*)         gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)         gl_cv_func_printf_flag_grouping="guessing no";;
                             # Guess no on Android.
            linux*-android*) gl_cv_func_printf_flag_grouping="guessing no";;
@@ -1611,7 +1611,7 @@
            darwin[1-6].*)        gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
-           cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           cygwin* | msys*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Naur grep-2.23-orig/m4/realloc.m4 grep-2.23/m4/realloc.m4
--- grep-2.23-orig/m4/realloc.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/realloc.m4	2016-03-30 08:53:21.560483900 +0300
@@ -25,7 +25,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | *-musl* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull="guessing yes" ;;
           # If we don't know, obey --enable-cross-guesses.
           *) ac_cv_func_realloc_0_nonnull="$gl_cross_guess_normal" ;;
diff -Naur grep-2.23-orig/m4/setlocale.m4 grep-2.23/m4/setlocale.m4
--- grep-2.23-orig/m4/setlocale.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/setlocale.m4	2016-03-30 08:53:21.578482100 +0300
@@ -18,7 +18,7 @@
     mingw*) NEED_SETLOCALE_IMPROVED=1 ;;
     dnl On Cygwin 1.5.x, setlocale always succeeds but setlocale(LC_CTYPE,NULL)
     dnl is then still "C".
-    cygwin*)
+    cygwin* | msys*)
       case `uname -r` in
         1.5.*) NEED_SETLOCALE_IMPROVED=1 ;;
       esac
diff -Naur grep-2.23-orig/m4/threadlib.m4 grep-2.23/m4/threadlib.m4
--- grep-2.23-orig/m4/threadlib.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/threadlib.m4	2016-03-30 08:53:21.606479300 +0300
@@ -413,7 +413,7 @@
          dnl Disable multithreading by default on Cygwin 1.5.x, because it has
          dnl bugs that lead to endless loops or crashes. See
          dnl <https://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
diff -Naur grep-2.23-orig/m4/wctob.m4 grep-2.23/m4/wctob.m4
--- grep-2.23-orig/m4/wctob.m4	2016-02-03 08:46:26.000000000 +0300
+++ grep-2.23/m4/wctob.m4	2016-03-30 08:53:21.651474800 +0300
@@ -29,7 +29,7 @@
 changequote(,)dnl
         case "$host_os" in
             # Guess no on Solaris <= 9 and Cygwin.
-          solaris2.[1-9] | solaris2.[1-9].* | cygwin*)
+          solaris2.[1-9] | solaris2.[1-9].* | cygwin* | msys*)
             gl_cv_func_wctob_works="guessing no" ;;
             # Guess no on native Windows.
           mingw*)
@@ -39,7 +39,7 @@
         esac
 changequote([,])dnl
         case "$host_os" in
-          cygwin*)
+          cygwin* | msys*)
             AC_RUN_IFELSE(
               [AC_LANG_SOURCE([[
 #include <locale.h>
