--- libbobcat/bobcat/CLASSES.orig	2020-04-20 16:24:39.415646700 +0300
+++ libbobcat/bobcat/CLASSES	2020-04-20 16:25:17.450432400 +0300
@@ -104,6 +104,5 @@
 typetrait
 user
 x2a
-cryptbuf
 eoibuf
 eoi
--- libbobcat/bobcat/SSLCLASSES.orig	2020-04-20 16:24:48.569034800 +0300
+++ libbobcat/bobcat/SSLCLASSES	2020-04-20 16:25:33.020435300 +0300
@@ -3,6 +3,7 @@
 pf_iterator
 pf_iteratorstream
 bigint
+cryptbuf
 decryptbuf
 diffiehellman
 digestbuf
