--- netcat-openbsd-debian-1.219-1/netcat.c.orig	2022-12-30 13:29:36.639812800 +0100
+++ netcat-openbsd-debian-1.219-1/netcat.c	2022-12-30 13:29:47.870486300 +0100
@@ -58,6 +58,10 @@
 # define IPTOS_MINCOST IPTOS_LOWCOST
 #endif /* IPTOS_LOWDELAY */
 
+#ifndef IPTOS_LOWCOST
+# define IPTOS_LOWCOST 0x02
+#endif 
+
 # ifndef IPTOS_DSCP_AF11
 # define	IPTOS_DSCP_AF11		0x28
 # define	IPTOS_DSCP_AF12		0x30
