--- Devel-GlobalDestruction-0.14/inc/ExtUtils/HasCompiler.pm.orig	2019-09-30 15:57:14.367607800 +0300
+++ Devel-GlobalDestruction-0.14/inc/ExtUtils/HasCompiler.pm	2019-09-30 15:57:47.767266500 +0300
@@ -113,7 +113,7 @@
 			my $lib = '-l' . ($libperl =~ /lib([^.]+)\./)[0];
 			push @extra, "$abs_basename.def", $lib, $perllibs;
 		}
-		elsif ($^O eq 'cygwin') {
+		elsif ($^O eq 'cygwin' || $^O eq 'msys') {
 			push @extra, catfile($incdir, $config->get('useshrplib') ? 'libperl.dll.a' : 'libperl.a');
 		}
 		elsif ($^O eq 'aix') {
