diff -ruNa tsocks-1.8_b/validateconf.c tsocks-1.8/validateconf.c
--- tsocks-1.8_b/validateconf.c	2009-12-23 21:45:03.000000000 +0100
+++ tsocks-1.8/validateconf.c	2009-12-23 21:47:19.000000000 +0100
@@ -23,7 +23,7 @@
 */
 
 /* Global configuration variables */ 
-char *progname = "validateconf";	      /* Name for error msgs      */
+char *progname = "tsocks-validateconf";	      /* Name for error msgs      */
 
 /* Header Files */
 #include <config.h>
--- tsocks-1.8/tsocks.conf.5.orig	2017-08-29 05:58:19.311729000 -0400
+++ tsocks-1.8/tsocks.conf.5	2017-08-29 05:59:24.133886500 -0400
@@ -139,19 +139,19 @@ socks.nec.com 1080'). It then inspects t
 the version that server supports. 
 
 .TP
-validateconf
-validateconf can be used to verify the configuration file. It checks the format
+tsocks-validateconf
+tsocks-validateconf can be used to verify the configuration file. It checks the format
 of the file and also the contents for errors. Having read the file it dumps 
 the configuration to the screen in a formatted, readable manner. This can be 
 extremely useful in debugging problems.
 
-validateconf can read a configuration file from a location other than the 
+tsocks-validateconf can read a configuration file from a location other than the 
 location specified at compile time with the -f <filename> command line 
 option.
 
-Normally validateconf simply dumps the configuration read to the screen (in
+Normally tsocks-validateconf simply dumps the configuration read to the screen (in
 a nicely readable format), however it also has a useful 'test' mode. When
-passed a hostname/ip on the command line like -t <hostname/ip>, validateconf 
+passed a hostname/ip on the command line like -t <hostname/ip>, tsocks-validateconf 
 determines which of the SOCKS servers specified in the configuration file 
 would be used by tsocks to access the specified host. 
 
