--- origsrc/gpgme-1.4.2/configure.ac	2013-05-28 04:08:35.000000000 -0500
+++ src/gpgme-1.4.2/configure.ac	2013-08-05 22:46:13.901736500 -0500
@@ -984,7 +984,7 @@ if test "$have_gpg_error" = "no"; then
 *** (at least version $NEED_GPG_ERROR_VERSION is required.)
 ***]])
 fi
-if test "$have_libassuan" = "no"; then
+if test "$have_libassuan" = "no" && test "$GPGSM" != no; then
    die=yes
    AC_MSG_NOTICE([[
 ***
--- origsrc/gpgme-1.4.2/src/Makefile.am	2013-05-23 13:17:20.000000000 -0500
+++ src/gpgme-1.4.2/src/Makefile.am	2013-08-05 22:55:09.161351600 -0500
@@ -73,7 +73,7 @@ gpgsm_components =
 endif
 
 if HAVE_ASSUAN
-assuan_components = assuan-support.c engine-assuan.c
+assuan_components = assuan-support.c engine-assuan.c opassuan.c
 else
 assuan_components =
 endif
@@ -112,7 +112,7 @@ main_sources =								\
 	sign.c passphrase.c progress.c					\
 	key.c keylist.c trust-item.c trustlist.c			\
 	import.c export.c genkey.c delete.c edit.c getauditlog.c        \
-	opassuan.c passwd.c                                             \
+	passwd.c                                                        \
 	engine.h engine-backend.h engine.c engine-gpg.c status-table.c	\
 	$(gpgsm_components) $(assuan_components) $(gpgconf_components)  \
 	$(uiserver_components)						\
@@ -223,8 +223,10 @@ libgpgme_qt_la_LIBADD = $(gpgme_res) @LI
 	@GPG_ERROR_LIBS@ @QT4_CORE_LIBS@
 endif
 
+if HAVE_ASSUAN
 noinst_PROGRAMS = gpgme-tool
 gpgme_tool_LDADD = libgpgme.la @LIBASSUAN_LIBS@
+endif
 
 install-data-local: install-def-file
 
