--- gpgme-1.18.0/m4/python.m4.orig	2023-02-14 20:38:30.523605800 +0100
+++ gpgme-1.18.0/m4/python.m4	2023-02-14 20:38:34.193959700 +0100
@@ -41,7 +41,7 @@
   m4_define_default([_AM_PYTHON_INTERPRETER_LIST],
 [python2 python2.7 dnl
  python dnl
- python3 python3.10 python3.9 python3.8 python3.7 python3.6 python3.5 python3.4
+ python3 python3.11 python3.10 python3.9 python3.8 python3.7 python3.6 python3.5 python3.4
  ])
 
   AC_ARG_VAR([PYTHON], [the Python interpreter])
--- gpgme-1.18.0/configure.ac.orig	2023-02-14 20:39:33.898571900 +0100
+++ gpgme-1.18.0/configure.ac	2023-02-14 20:32:32.901864400 +0100
@@ -443,7 +443,7 @@
 	if test "$found_py" = "1" -o "$found_py3" = "1"; then
 	  # Reset everything, so that we can look for another Python.
           m4_foreach([mym4pythonver],
-                     [[2.7],[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[3.10],[all]],
+                     [[2.7],[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[3.10],[3.11],[all]],
            [unset PYTHON
 	    unset PYTHON_VERSION
 	    unset PYTHON_CPPFLAGS
