--- wcurl-2024.07.10/wcurl.orig	2024-08-03 21:22:27.193328800 +0200
+++ wcurl-2024.07.10/wcurl	2024-08-03 21:22:31.933752200 +0200
@@ -114,7 +114,7 @@
 
 # Execute curl with the list of URLs provided by the user.
 exec_curl() {
-    CMD="curl "
+    CMD="/usr/bin/curl "
 
     # Store version to check if it supports --no-clobber and --parallel.
     curl_version=$($CMD --version | cut -f2 -d' ' | head -n1)
