--- bc-1.06.orig/bc/scan.l	2015-06-11 17:02:07.886738000 +0100
+++ bc-1.06/bc/scan.l	2015-06-11 17:02:20.485238000 +0100
@@ -143,7 +143,7 @@
 
 /* Definitions for readline access. */
 extern FILE *rl_instream;
-_PROTOTYPE(char *readline, (char *));
+_PROTOTYPE(char *readline, (const char *));
 
 /* rl_input puts upto MAX characters into BUF with the number put in
    BUF placed in *RESULT.  If the yy input file is the same as
