--- src/funopen.c.bk	2015-09-23 01:56:09.000000000 +0800
+++ src/funopen.c	2017-04-13 20:34:08.080344600 +0800
@@ -98,11 +98,11 @@ funopen_close(void *cookie)
 }
 
 FILE *
-funopen(const void *cookie,
-        int (*readfn)(void *cookie, char *buf, int size),
-        int (*writefn)(void *cookie, const char *buf, int size),
-        off_t (*seekfn)(void *cookie, off_t offset, int whence),
-        int (*closefn)(void *cookie))
+funopen(const char *cookie,
+        int (*readfn)(char *cookie, char *buf, int size),
+        int (*writefn)(char *cookie, const char *buf, int size),
+        off_t (*seekfn)(char *cookie, off_t offset, int whence),
+        int (*closefn)(char *cookie))
 {
 	struct funopen_cookie *cookiewrap;
 	cookie_io_functions_t funcswrap = {
