diff -uprN libbsd-0.8.3.bk/include/bsd/nlist.h libbsd-0.8.3/include/bsd/nlist.h
--- libbsd-0.8.3.bk/include/bsd/nlist.h	2015-09-23 01:56:09.000000000 +0800
+++ libbsd-0.8.3/include/bsd/nlist.h	2017-04-13 22:19:25.487040400 +0800
@@ -27,8 +27,46 @@
 #ifndef LIBBSD_NLIST_H
 #define LIBBSD_NLIST_H
 
-#include <sys/cdefs.h>
+
+/*
+ * Symbol table entry format.  The #ifdef's are so that programs including
+ * nlist.h can initialize nlist structures statically.
+ */
+
 #include <a.out.h>
+#include <a.out.gnu.h>
+struct nlist {
+	union {
+		char *n_name;	/* symbol name (in memory) */
+		long n_strx;	/* file string table offset (on disk) */
+	} n_un;
+
+#define	N_UNDF	0x00		/* undefined */
+#define	N_ABS	0x02		/* absolute address */
+#define	N_TEXT	0x04		/* text segment */
+#define	N_DATA	0x06		/* data segment */
+#define	N_BSS	0x08		/* bss segment */
+#define	N_INDR	0x0a		/* alias definition */
+#define	N_SIZE	0x0c		/* pseudo type, defines a symbol's size */
+#define	N_COMM	0x12		/* common reference */
+#define	N_FN	0x1e		/* file name (N_EXT on) */
+#define	N_WARN	0x1e		/* warning message (N_EXT off) */
+
+#define	N_EXT	0x01		/* external (global) bit, OR'ed in */
+#define	N_TYPE	0x1e		/* mask for all the type bits */
+	unsigned char n_type;	/* type defines */
+
+	char n_other;		/* spare */
+#define	n_hash	n_desc		/* used internally by ld(1); XXX */
+	short n_desc;		/* used by stab entries */
+	unsigned long n_value;	/* address/value of the symbol */
+};
+
+#define	N_FORMAT	"%08x"	/* namelist value format; XXX */
+#define	N_STAB		0x0e0	/* mask for debugger symbols -- stab(5) */
+
+#include <sys/cdefs.h>
+
 
 __BEGIN_DECLS
 extern int nlist(const char *filename, struct nlist *list);
diff -uprN libbsd-0.8.3.bk/include/bsd/stdio.h libbsd-0.8.3/include/bsd/stdio.h
--- libbsd-0.8.3.bk/include/bsd/stdio.h	2015-09-23 13:59:34.000000000 +0800
+++ libbsd-0.8.3/include/bsd/stdio.h	2017-04-13 21:29:05.526890100 +0800
@@ -50,24 +50,6 @@ char *fgetln(FILE *fp, size_t *lenp)
 	__attribute__((deprecated("This functions cannot be safely ported, "
 	                          "use getline(3) instead, as it is supported "
 	                          "by GNU and POSIX.1-2008.")));
-
-/*
- * Note: We diverge from the FreeBSD, OpenBSD and DragonFlyBSD declarations,
- * because seekfn() there wrongly uses fpos_t, assuming it's an integral
- * type, and any code using that on a system where fpos_t is a struct
- * (such as GNU-based systems or NetBSD) will fail to build. In which case,
- * as the code has to be modified anyway, we might just as well use the
- * correct declaration here.
- */
-FILE *funopen(const void *cookie,
-              int (*readfn)(void *cookie, char *buf, int size),
-              int (*writefn)(void *cookie, const char *buf, int size),
-              off_t (*seekfn)(void *cookie, off_t offset, int whence),
-              int (*closefn)(void *cookie));
-
-#define fropen(cookie, fn) funopen(cookie, fn, NULL, NULL, NULL)
-#define fwopen(cookie, fn) funopen(cookie, NULL, fn, NULL, NULL)
-
 int fpurge(FILE *fp);
 __END_DECLS
 
diff -uprN libbsd-0.8.3.bk/src/funopen.c libbsd-0.8.3/src/funopen.c
--- libbsd-0.8.3.bk/src/funopen.c	2015-09-23 01:56:09.000000000 +0800
+++ libbsd-0.8.3/src/funopen.c	2017-04-13 21:29:20.890562400 +0800
@@ -31,112 +31,6 @@
 #include <stdio.h>
 
 #ifdef HAVE_FOPENCOOKIE
-struct funopen_cookie {
-	void *orig_cookie;
-
-	int (*readfn)(void *cookie, char *buf, int size);
-	int (*writefn)(void *cookie, const char *buf, int size);
-	off_t (*seekfn)(void *cookie, off_t offset, int whence);
-	int (*closefn)(void *cookie);
-};
-
-static ssize_t
-funopen_read(void *cookie, char *buf, size_t size)
-{
-	struct funopen_cookie *cookiewrap = cookie;
-
-	if (cookiewrap->readfn == NULL) {
-		errno = EBADF;
-		return -1;
-	}
-
-	return cookiewrap->readfn(cookiewrap->orig_cookie, buf, size);
-}
-
-static ssize_t
-funopen_write(void *cookie, const char *buf, size_t size)
-{
-	struct funopen_cookie *cookiewrap = cookie;
-
-	if (cookiewrap->writefn == NULL)
-		return EOF;
-
-	return cookiewrap->writefn(cookiewrap->orig_cookie, buf, size);
-}
-
-static int
-funopen_seek(void *cookie, off64_t *offset, int whence)
-{
-	struct funopen_cookie *cookiewrap = cookie;
-	off_t soff = *offset;
-
-	if (cookiewrap->seekfn == NULL) {
-		errno = ESPIPE;
-		return -1;
-	}
-
-	soff = cookiewrap->seekfn(cookiewrap->orig_cookie, soff, whence);
-	*offset = soff;
-
-	return *offset;
-}
-
-static int
-funopen_close(void *cookie)
-{
-	struct funopen_cookie *cookiewrap = cookie;
-	int rc;
-
-	if (cookiewrap->closefn == NULL)
-		return 0;
-
-	rc = cookiewrap->closefn(cookiewrap->orig_cookie);
-
-	free(cookiewrap);
-
-	return rc;
-}
-
-FILE *
-funopen(const void *cookie,
-        int (*readfn)(void *cookie, char *buf, int size),
-        int (*writefn)(void *cookie, const char *buf, int size),
-        off_t (*seekfn)(void *cookie, off_t offset, int whence),
-        int (*closefn)(void *cookie))
-{
-	struct funopen_cookie *cookiewrap;
-	cookie_io_functions_t funcswrap = {
-		.read = funopen_read,
-		.write = funopen_write,
-		.seek = funopen_seek,
-		.close = funopen_close,
-	};
-	const char *mode;
-
-	if (readfn) {
-		if (writefn == NULL)
-			mode = "r";
-		else
-			mode = "r+";
-	} else if (writefn) {
-		mode = "w";
-	} else {
-		errno = EINVAL;
-		return NULL;
-	}
-
-	cookiewrap = malloc(sizeof(*cookiewrap));
-	if (cookiewrap == NULL)
-		return NULL;
-
-	cookiewrap->orig_cookie = (void *)cookie;
-	cookiewrap->readfn = readfn;
-	cookiewrap->writefn = writefn;
-	cookiewrap->seekfn = seekfn;
-	cookiewrap->closefn = closefn;
-
-	return fopencookie(cookiewrap, mode, funcswrap);
-}
 #else
 #error "Function funopen() needs to be ported."
 #endif
diff -uprN libbsd-0.8.3.bk/src/getpeereid.c libbsd-0.8.3/src/getpeereid.c
--- libbsd-0.8.3.bk/src/getpeereid.c	2015-09-23 01:56:09.000000000 +0800
+++ libbsd-0.8.3/src/getpeereid.c	2017-04-13 21:35:46.559698600 +0800
@@ -40,11 +40,7 @@ int
 getpeereid(int s, uid_t *euid, gid_t *egid)
 {
 /* XXX: This should be autodetected at build time instead. */
-#if defined(__linux__)
 	struct ucred cred;
-#elif defined(__OpenBSD__)
-	struct sockpeercred cred;
-#endif
 	socklen_t credlen = sizeof(cred);
 	int ret;
 
diff -uprN libbsd-0.8.3.bk/src/nlist.c libbsd-0.8.3/src/nlist.c
--- libbsd-0.8.3.bk/src/nlist.c	2016-03-27 19:33:21.000000000 +0800
+++ libbsd-0.8.3/src/nlist.c	2017-04-13 21:54:10.025292800 +0800
@@ -44,11 +44,10 @@ static char sccsid[] = "@(#)nlist.c	8.1
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
-
+#include "nlist.h"
 #if !defined(__NO_A_OUT_SUPPORT)
 #define _NLIST_DO_AOUT
 #endif
-#define _NLIST_DO_ELF
 
 #ifdef _NLIST_DO_ELF
 #include "local-elf.h"
