--- include/bsd/stdio.h.bk	2015-09-23 13:59:34.000000000 +0800
+++ include/bsd/stdio.h	2017-04-13 20:33:19.120591600 +0800
@@ -59,11 +59,11 @@ char *fgetln(FILE *fp, size_t *lenp)
  * as the code has to be modified anyway, we might just as well use the
  * correct declaration here.
  */
-FILE *funopen(const void *cookie,
-              int (*readfn)(void *cookie, char *buf, int size),
-              int (*writefn)(void *cookie, const char *buf, int size),
-              off_t (*seekfn)(void *cookie, off_t offset, int whence),
-              int (*closefn)(void *cookie));
+FILE *funopen(const  _PTR *cookie,
+              int (*readfn)(_PTR *cookie, char *buf, int size),
+              int (*writefn)(_PTR *cookie, const char *buf, int size),
+              off_t (*seekfn)(_PTR *cookie, off_t offset, int whence),
+              int (*closefn)(_PTR *cookie));
 
 #define fropen(cookie, fn) funopen(cookie, fn, NULL, NULL, NULL)
 #define fwopen(cookie, fn) funopen(cookie, NULL, fn, NULL, NULL)
