--- WWW-RobotRules-6.03/t/rules-dbm.t.orig
+++ WWW-RobotRules-6.03/t/rules-dbm.t
@@ -12,7 +12,7 @@ my $file = "$dir/robotdb";
 my $r = WWW::RobotRules::AnyDBM_File->new("myrobot/2.0", $file);
 
 # Cache backing file(s) must have no group/world permission bits.
-if ($^O ne 'MSWin32') {
+if ($^O ne 'cygwin') {
     my @backing = glob "$file*";
     ok scalar @backing, "DBM backing file(s) exist after construction";
     for my $f (@backing) {
