--- autoconf-2.72/build-aux/config.guess.orig	2023-12-23 09:15:46.672145000 +0100
+++ autoconf-2.72/build-aux/config.guess	2023-12-23 09:16:58.657560200 +0100
@@ -954,6 +954,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	GUESS=x86_64-pc-cygwin
 	;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	GUESS=x86_64-unknown-msys
+	;;
     prep*:SunOS:5.*:*)
 	SUN_REL=`echo "$UNAME_RELEASE" | sed -e 's/[^.]*//'`
 	GUESS=powerpcle-unknown-solaris2$SUN_REL
