diff -Naur cgdb-0.6.8-orig/config/config.guess cgdb-0.6.8/config/config.guess
--- cgdb-0.6.8-orig/config/config.guess	2013-08-11 14:49:21.000000000 +0300
+++ cgdb-0.6.8/config/config.guess	2014-12-12 00:31:29.054600000 +0300
@@ -866,6 +866,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur cgdb-0.6.8-orig/configure.ac cgdb-0.6.8/configure.ac
--- cgdb-0.6.8-orig/configure.ac	2014-11-13 22:59:02.000000000 +0300
+++ cgdb-0.6.8/configure.ac	2014-12-12 00:30:40.647800000 +0300
@@ -20,7 +20,7 @@
  *-*-linux*)
    AC_DEFINE(HAVE_LINUX, 1,  Linux environment)
    ;;
- *-*-cygwin*)
+ *-*-cygwin* | *-*-msys*)
    AC_DEFINE(HAVE_CYGWIN, 1,  Cygwin environment)
    ;;
  *)
