diff -Naur cgdb-0.6.8-orig/lib/util/fs_util.c cgdb-0.6.8/lib/util/fs_util.c
--- cgdb-0.6.8-orig/lib/util/fs_util.c	2014-02-25 23:21:05.000000000 +0300
+++ cgdb-0.6.8/lib/util/fs_util.c	2014-12-12 00:49:04.299400000 +0300
@@ -56,7 +56,7 @@
 
     /* Get the directory to check */
 #ifdef HAVE_CYGWIN
-    cygwin_conv_to_full_win32_path(dir, cygwin_actual_dir);
+    cygwin_conv_path (CCP_POSIX_TO_WIN_A|CCP_ABSOLUTE, dir, cygwin_actual_dir, PATH_MAX+1);
     strncpy(actual_dir, cygwin_actual_dir, strlen(cygwin_actual_dir) + 1);
 #else
     strncpy(actual_dir, dir, strlen(dir) + 1);
@@ -94,7 +94,7 @@
 
     /* Get the directory to check */
 #ifdef HAVE_CYGWIN
-    cygwin_conv_to_full_win32_path(dir, cygwin_actual_dir);
+    cygwin_conv_path (CCP_POSIX_TO_WIN_A|CCP_ABSOLUTE, dir, cygwin_actual_dir, PATH_MAX+1);
     strncpy(actual_dir, cygwin_actual_dir, strlen(cygwin_actual_dir) + 1);
 #else
     strncpy(actual_dir, dir, strlen(dir) + 1);
@@ -143,7 +143,7 @@
     sprintf(dir, "%s/%s", base, name);
 
 #ifdef HAVE_CYGWIN
-    cygwin_conv_to_full_win32_path(dir, path);
+    cygwin_conv_path (CCP_POSIX_TO_WIN_A|CCP_ABSOLUTE, dir, path, PATH_MAX+1);
 #else
     strncpy(path, dir, strlen(dir) + 1);
 #endif
