diff --git automake-1.17-orig/t/compile5.sh automake-1.17/t/compile5.sh
index 67c46cc..ddf9683 100644
--- automake-1.17-orig/t/compile5.sh
+++ automake-1.17/t/compile5.sh
@@ -47,11 +47,11 @@ case '@host_os@' in
     ;;
 esac
 case @build_os@ in
-  mingw* | cygwin*)
+  mingw* | cygwin* | msys*)
     ;;
   *)
     winepath -w / \
-      || skip_ "not on MinGW or Cygwin, and winepath not available"
+      || skip_ "not on MinGW, Cygwin or MSYS, and winepath not available"
     ;;
 esac
 END
