--- unrar/makefile.orig	2014-05-25 19:06:51.053000000 +0400
+++ unrar/makefile	2014-05-25 19:09:28.129400000 +0400
@@ -132,8 +132,8 @@
 lib:	WHAT=RARDLL
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	clean $(OBJECTS) $(LIB_OBJ)
-	@rm -f libunrar.so
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	@rm -f libunrar.dll
+	$(LINK) -shared -Wl,--out-implib,libunrar.dll.a -o libunrar.dll $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
 			install -D unrar $(DESTDIR)/bin/unrar
@@ -142,7 +142,8 @@
 			rm -f $(DESTDIR)/bin/unrar
 
 install-lib:
-		install libunrar.so $(DESTDIR)/lib
+		install libunrar.dll.a $(DESTDIR)/lib
+		install libunrar.dll $(DESTDIR)/bin
 
 uninstall-lib:
-		rm -f $(DESTDIR)/lib/libunrar.so
+		rm -f $(DESTDIR)/lib/libunrar.dll.a
