--- unrar/makefile.orig	2020-01-29 00:00:00.000000000 +0000
+++ unrar/makefile	2020-01-29 00:00:00.000000000 +0000
@@ -154,9 +154,9 @@
 lib:	WHAT=RARDLL
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	clean $(OBJECTS) $(LIB_OBJ)
-	@rm -f libunrar.so
+	@rm -f libunrar.dll
 	@rm -f libunrar.a
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	$(LINK) -shared -Wl,--out-implib,libunrar.dll.a -o msys-unrar.dll $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 	$(AR) rcs libunrar.a $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
@@ -166,8 +166,10 @@
 			rm -f $(DESTDIR)/bin/unrar
 
 install-lib:
-		install libunrar.so $(DESTDIR)/lib
+		install libunrar.dll.a $(DESTDIR)/lib
+		install msys-unrar.dll $(DESTDIR)/bin
 		install libunrar.a $(DESTDIR)/lib
 
 uninstall-lib:
-		rm -f $(DESTDIR)/lib/libunrar.so
+		rm -f $(DESTDIR)/bin/msys-unrar.dll
+		rm -f $(DESTDIR)/lib/libunrar.dll.a
