diff -Naur automake-1.14-orig/lib/ar-lib automake-1.14/lib/ar-lib
--- automake-1.14-orig/lib/ar-lib	2013-05-09 18:13:08.000000000 +0000
+++ automake-1.14/lib/ar-lib	2013-05-16 09:13:29.442382800 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -65,7 +65,7 @@
 	mingw)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin)
+	cygwin|msys)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine)
diff -Naur automake-1.14-orig/lib/compile automake-1.14/lib/compile
--- automake-1.14-orig/lib/compile	2013-05-09 18:13:08.000000000 +0000
+++ automake-1.14/lib/compile	2013-05-16 09:14:34.254882800 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur automake-1.14-orig/lib/config.guess automake-1.14/lib/config.guess
--- automake-1.14-orig/lib/config.guess	2013-05-13 20:12:48.000000000 +0000
+++ automake-1.14/lib/config.guess	2013-05-16 09:14:15.176757800 +0000
@@ -845,6 +845,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur automake-1.14-orig/t/compile5.test automake-1.14/t/compile5.test
--- automake-1.14-orig/tests/compile5.test	2013-05-09 18:13:08.000000000 +0000
+++ automake-1.14/tests/compile5.test	2013-05-16 09:15:38.355468800 +0000
@@ -47,7 +47,7 @@
     ;;
 esac
 case @build_os@ in
-  mingw* | cygwin*)
+  mingw* | cygwin* | msys*)
     ;;
   *)
     winepath -w / \
