--- lz4-1.8.2/Makefile	2017-01-02 23:09:52.000000000 -0600
+++ lz4-1.8.2/Makefile	2017-06-19 09:25:47.933172300 -0500
@@ -89,7 +89,7 @@ clean:
 #-----------------------------------------------------------------------------
 # make install is validated only for Linux, OSX, BSD, Hurd and Solaris targets
 #-----------------------------------------------------------------------------
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS))
+ifneq (,$(filter Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS CYGWIN% MSYS%, $(shell uname)))
 HOST_OS = POSIX
 
 .PHONY: install uninstall
--- lz4-1.8.2/lib/Makefile	2017-01-02 23:09:52.000000000 -0600
+++ lz4-1.8.2/lib/Makefile	2017-06-19 09:39:30.318048400 -0500
@@ -110,7 +110,7 @@ clean:
 #-----------------------------------------------------------------------------
 # make install is validated only for Linux, OSX, BSD, Hurd and Solaris targets
 #-----------------------------------------------------------------------------
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS))
+ifneq (,$(filter Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS CYGWIN% MSYS%, $(shell uname)))
 
 ifneq (,$(filter $(shell uname),SunOS))
 INSTALL ?= ginstall
--- lz4-1.8.2/programs/Makefile	2017-01-02 23:09:52.000000000 -0600
+++ lz4-1.8.2/programs/Makefile	2017-06-19 09:27:25.812946800 -0500
@@ -111,7 +111,7 @@ preview-man: clean-man man
 #-----------------------------------------------------------------------------
 # make install is validated only for Linux, OSX, BSD, Hurd and Solaris targets
 #-----------------------------------------------------------------------------
-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS))
+ifneq (,$(filter Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS CYGWIN% MSYS%, $(shell uname)))
 
 unlz4: lz4
 	ln -s lz4 unlz4
