--- fortune-mod-9708/Makefile
+++ fortune-mod-9708/Makefile
@@ -46,13 +46,13 @@
 # NB. Under Linux, the BSD regex functions are _MUCH_ faster
 #     than the POSIX ones, but your mileage may vary.
 #
-REGEXDEFS=-DHAVE_REGEX_H -DBSD_REGEX
+REGEXDEFS=-DHAVE_REGEX_H -DPOSIX_REGEX
 
 #
 # If your system's regex functions are not in its standard C library,
 # include the appropriate link flags into REGEXLIBS
 #
-REGEXLIBS=
+REGEXLIBS=-lpcreposix
 
 DEFINES=-DFORTDIR="\"$(COOKIEDIR)\"" -DOFFDIR="\"$(OCOOKIEDIR)\""
 CFLAGS=-O2 $(DEFINES) -Wall -fomit-frame-pointer -pipe
