--- fortune-mod-9807/Makefile
+++ fortune-mod-9807/Makefile
@@ -2,25 +2,28 @@
 # Makefile for fortune-mod
 #
 
+#DESTDIR-ify
+DESTDIR:=
+
 # Where does the fortune program go?
-FORTDIR=/usr/local/games
+FORTDIR=$(DESTDIR)/usr/local/games
 # Where do the data files (fortunes, or cookies) go?
-COOKIEDIR=/usr/local/share/games/fortunes
+COOKIEDIR=$(DESTDIR)/usr/local/share/games/fortunes
 # Offensive ones?
 OCOOKIEDIR=$(COOKIEDIR)/off
 # The ones with html tags?
 WCOOKIEDIR=$(COOKIEDIR)/html
 # Where do strfile and unstr go?
-BINDIR=/usr/local/bin
+BINDIR=$(DESTDIR)/usr/local/bin
 # What is the proper mode for strfile and unstr? 755= everyone, 700= root only
 BINMODE=0755
 #BINMODE=0700
 # Where do the man pages for strfile and unstr go?
-BINMANDIR=/usr/local/man/man1
+BINMANDIR=$(DESTDIR)/usr/local/man/man1
 # What is their proper extension?
 BINMANEXT=1
 # And the same for the fortune man page
-FORTMANDIR=/usr/local/man/man6
+FORTMANDIR=$(DESTDIR)/usr/local/man/man6
 FORTMANEXT=6
 # Do we want to install the offensive files? (0 no, 1 yes)
 OFFENSIVE=1
