diff -Naur XML-Parser-2.41/inc/Devel/CheckLib.pm XML-Parser-2.41/inc/Devel/CheckLib.pm
--- XML-Parser-2.41/inc/Devel/CheckLib.pm	2011-05-25 03:39:14.000000000 +0000
+++ XML-Parser-2.41/inc/Devel/CheckLib.pm	2013-06-16 15:38:01.556218800 +0000
@@ -385,7 +385,7 @@
 
 =over
 
-=item gcc (on Linux, *BSD, Mac OS X, Solaris, Cygwin)
+=item gcc (on Linux, *BSD, Mac OS X, Solaris, Cygwin, Msys)
 
 =item Sun's compiler tools on Solaris
 
