diff -Naur XML-LibXML-2.0200-orig/t/11memory.t XML-LibXML-2.0200/t/11memory.t
--- XML-LibXML-2.0200-orig/t/11memory.t	2016-07-14 07:17:08.000000000 +0300
+++ XML-LibXML-2.0200/t/11memory.t	2019-04-27 20:54:13.399173300 +0300
@@ -10,9 +10,9 @@
 
 use constant TIMES_THROUGH => $ENV{MEMORY_TIMES} || 100_000;
 
-if (! (($^O eq 'linux') || ($^O eq 'cygwin')) )
+if (! (($^O eq 'linux') || ($^O eq 'cygwin') || ($^O eq 'msys')) )
 {
-    plan skip_all => 'Only runs on Linux and Cygwin.';
+    plan skip_all => 'Only runs on Linux and Cygwin or MSYS.';
 }
 elsif (! $ENV{MEMORY_TEST} )
 {
