diff -Naur XML-LibXML-2.0018/inc/Devel/CheckLib.pm XML-LibXML-2.0018/inc/Devel/CheckLib.pm
--- XML-LibXML-2.0018/inc/Devel/CheckLib.pm	2012-10-13 11:14:26.000000000 +0000
+++ XML-LibXML-2.0018/inc/Devel/CheckLib.pm	2013-06-16 16:03:16.958562500 +0000
@@ -417,7 +417,7 @@
 
 =over
 
-=item gcc (on Linux, *BSD, Mac OS X, Solaris, Cygwin)
+=item gcc (on Linux, *BSD, Mac OS X, Solaris, Cygwin, Msys)
 
 =item Sun's compiler tools on Solaris
 
