diff -Naur ExtUtils-Helpers-0.026-orig/lib/ExtUtils/Helpers.pm ExtUtils-Helpers-0.026/lib/ExtUtils/Helpers.pm
--- ExtUtils-Helpers-0.026-orig/lib/ExtUtils/Helpers.pm	2016-09-09 20:26:59.000000000 +0300
+++ ExtUtils-Helpers-0.026/lib/ExtUtils/Helpers.pm	2020-05-30 12:01:56.646078200 +0300
@@ -39,6 +39,7 @@
 	VMS => '__',
 	os2 => '.',
 	cygwin => '.',
+	msys => '.',
 );
 my $separator = $separator{$^O} || '::';
 
diff -Naur ExtUtils-Helpers-0.026-orig/t/man_pagename.t ExtUtils-Helpers-0.026/t/man_pagename.t
--- ExtUtils-Helpers-0.026-orig/t/man_pagename.t	2016-09-09 20:26:59.000000000 +0300
+++ ExtUtils-Helpers-0.026/t/man_pagename.t	2020-05-30 12:02:20.269088400 +0300
@@ -12,6 +12,7 @@
 	VMS => '__',
 	os2 => '.',
 	cygwin => '.',
+	msys => '.',
 );
 my $sep = $separator{$^O} || '::';
 
