From 70f8e42761bbe5f53823487f075e78481dd9c11f Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Thu, 11 Sep 2014 20:21:40 +0100
Subject: [PATCH 1/2] Modify config to support correct Mingw64 Triplet Names


diff --git a/scripts/config.guess b/scripts/config.guess
index dbfb978..555d9f0 100755
--- a/scripts/config.guess
+++ b/scripts/config.guess
@@ -822,7 +822,10 @@ EOF
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
     *:MINGW64*:*)
-	echo ${UNAME_MACHINE}-pc-mingw64
+	echo x86_64-w64-mingw32
+	exit ;;
+    *:MINGW32*:*)
+	echo i686-w64-mingw32
 	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
diff --git a/scripts/config.sub b/scripts/config.sub
index 6d2e94c..db92389 100755
--- a/scripts/config.sub
+++ b/scripts/config.sub
@@ -805,11 +805,11 @@ case $basic_machine in
 		basic_machine=microblaze-xilinx
 		;;
 	mingw64)
-		basic_machine=x86_64-pc
-		os=-mingw64
+		basic_machine=x86_64-w64
+		os=-mingw32
 		;;
 	mingw32)
-		basic_machine=i686-pc
+		basic_machine=i686-w64
 		os=-mingw32
 		;;
 	mingw32ce)
-- 
2.4.5

