--- crosstool-ng-1.27.0/scripts/config.sub.orig	2025-01-27 00:57:09.291702400 +0100
+++ crosstool-ng-1.27.0/scripts/config.sub	2025-06-07 19:31:13.928235500 +0200
@@ -389,11 +389,11 @@
 				basic_os=sysv
 				;;
 			mingw64)
-				basic_machine=x86_64-pc
-				basic_os=mingw64
+				basic_machine=x86_64-w64
+				basic_os=mingw32
 				;;
 			mingw32)
-				basic_machine=i686-pc
+				basic_machine=i686-w64
 				basic_os=mingw32
 				;;
 			mingw32ce)
--- crosstool-ng-1.27.0/scripts/config.guess.orig	2025-06-07 19:28:28.964494600 +0200
+++ crosstool-ng-1.27.0/scripts/config.guess	2025-06-07 19:30:33.048391200 +0200
@@ -924,7 +924,10 @@
 	GUESS=$UNAME_MACHINE-pc-cygwin
 	;;
     *:MINGW64*:*)
-	GUESS=$UNAME_MACHINE-pc-mingw64
+	GUESS=x86_64-w64-mingw32
+	;;
+    *:MINGW32*:*)
+	GUESS=i686-w64-mingw32
 	;;
     *:MINGW*:*)
 	GUESS=$UNAME_MACHINE-pc-mingw32
