From 70f8e42761bbe5f53823487f075e78481dd9c11f Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Thu, 11 Sep 2014 20:21:40 +0100
Subject: [PATCH 1/2] Modify config to support correct Mingw64 Triplet Names


diff -durN crosstool-ng-1.22.0.orig/scripts/config.guess crosstool-ng-1.22.0/scripts/config.guess
--- crosstool-ng-1.22.0.orig/scripts/config.guess	2015-11-18 13:29:15.000000000 +0800
+++ crosstool-ng-1.22.0/scripts/config.guess	2017-04-13 16:43:45.328125000 +0800
@@ -839,7 +839,10 @@
 	echo "$UNAME_MACHINE"-pc-cygwin
 	exit ;;
     *:MINGW64*:*)
-	echo "$UNAME_MACHINE"-pc-mingw64
+	echo x86_64-w64-mingw32
+	exit ;;
+    *:MINGW32*:*)
+	echo i686-w64-mingw32
 	exit ;;
     *:MINGW*:*)
 	echo "$UNAME_MACHINE"-pc-mingw32
diff -durN crosstool-ng-1.22.0.orig/scripts/config.sub crosstool-ng-1.22.0/scripts/config.sub
--- crosstool-ng-1.22.0.orig/scripts/config.sub	2015-11-18 13:29:15.000000000 +0800
+++ crosstool-ng-1.22.0/scripts/config.sub	2017-04-13 16:45:44.218750000 +0800
@@ -381,11 +381,11 @@
 				os=sysv
 				;;
 			mingw64)
-				basic_machine=x86_64-pc
-				os=mingw64
+				basic_machine=x86_64-w64
+				os=-mingw32
 				;;
 			mingw32)
-				basic_machine=i686-pc
+				basic_machine=i686-w64
 				os=mingw32
 				;;
 			mingw32ce)
