From d3161c922bce20a801cff07fca7850db265cec83 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Tue, 7 Jul 2015 17:41:18 +0100
Subject: [PATCH 2/2] use wide versions of libs for nconf


diff --git a/kconfig/Makefile b/kconfig/Makefile
index 3474e5c..4ed0de9 100644
--- a/kconfig/Makefile
+++ b/kconfig/Makefile
@@ -48,7 +48,7 @@ nconf_SRC = nconf.c nconf.gui.c
 nconf_OBJ = $(patsubst %.c,%.o,$(nconf_SRC))
 nconf_DEP = $(patsubst %.c,%.dep,$(nconf_SRC))
 $(nconf_OBJ) $(nconf_DEP): CFLAGS += $(INTL_CFLAGS) -I/usr/include/ncurses
-nconf: LDFLAGS += -lmenu -lpanel -lncurses
+nconf: LDFLAGS += -lmenuw -lpanelw -lncurses

 # Under Cygwin, we need to auto-import some libs (which ones, exactly?)
 # for mconf and nconf to lin properly.
--
2.4.5
