From d3161c922bce20a801cff07fca7850db265cec83 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Tue, 7 Jul 2015 17:41:18 +0100
Subject: [PATCH 2/2] use wide versions of libs for nconf


diff -durN crosstool-ng-1.22.0.orig/kconfig/Makefile crosstool-ng-1.22.0/kconfig/Makefile
--- crosstool-ng-1.22.0.orig/kconfig/Makefile	2015-11-18 13:29:15.000000000 +0800
+++ crosstool-ng-1.22.0/kconfig/Makefile	2017-04-13 16:55:14.218750000 +0800
@@ -48,7 +48,7 @@
 nconf_OBJ = $(patsubst %.c,%.o,$(nconf_SRC))
 nconf_DEP = $(patsubst %.c,%.dep,$(nconf_SRC))
 $(nconf_OBJ) $(nconf_DEP): CFLAGS += $(INTL_CFLAGS) -I/usr/include/ncurses
-nconf: LDFLAGS += -lmenu -lpanel $(LIBS)
+nconf: LDFLAGS += -lmenuw -lpanelw -lncursesw $(LIBS)
 
 # These are generated files:
 ALL_OBJS = $(sort $(COMMON_OBJ) $(LX_OBJ) $(conf_OBJ) $(mconf_OBJ) $(nconf_OBJ))
