diff -Naur crosstool-ng-1.21.0/kconfig/nconf.c crosstool-ng-1.21.0.new/kconfig/nconf.c
--- crosstool-ng-1.21.0/kconfig/nconf.c	2015-05-25 21:47:17.000000000 +0200
+++ crosstool-ng-1.21.0.new/kconfig/nconf.c	2015-09-03 11:27:08.442172100 +0200
@@ -1518,7 +1518,11 @@
 	}
 
 	notimeout(stdscr, FALSE);
-	ESCDELAY = 1;
+#if NCURSES_REENTRANT
+	set_escdelay(1);
+#else
+ 	ESCDELAY = 1;
+#endif	
 
 	/* set btns menu */
 	curses_menu = new_menu(curses_menu_items);
