From a41a68e64f6cb73b20c8b14f39d87503a412a591 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Fri, 12 Sep 2014 01:06:15 +0100
Subject: [PATCH 6/6] CT_DoForceRmdir: added support for MSYS and MINGW

---
 scripts/functions | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/scripts/functions b/scripts/functions
index 60a6793..26a637f 100644
--- a/scripts/functions
+++ b/scripts/functions
@@ -458,12 +458,15 @@ CT_DoForceRmdir() {
     for dir in "${@}"; do
         [ -d "${dir}" ] || continue
         case "$CT_SYS_OS" in
-            Linux|CYGWIN*)
+            Linux|CYGWIN*|MSYS*)
                 mode="$(stat -c '%a' "$(dirname "${dir}")")"
                 ;;
             Darwin|*BSD)
                 mode="$(stat -f '%Lp' "$(dirname "${dir}")")"
                 ;;
+            MINGW*)
+                mode=0777
+                ;;
             *)
                 CT_Abort "Unhandled host OS $CT_SYS_OS"
                 ;;
-- 
2.2.1

