--- msys2-w32api-headers/mingw-w64-headers/include/netioapi.h.orig	2015-03-03 22:12:25.029800000 +0300
+++ msys2-w32api-headers/mingw-w64-headers/include/netioapi.h	2015-03-03 22:12:45.590600000 +0300
@@ -371,8 +371,10 @@
   NETIO_STATUS NETIOAPI_API_ ConvertInterfaceAliasToLuid (CONST WCHAR *InterfaceAlias, PNET_LUID InterfaceLuid);
   NETIO_STATUS NETIOAPI_API_ ConvertInterfaceLuidToGuid (CONST NET_LUID *InterfaceLuid, GUID *InterfaceGuid);
   NETIO_STATUS NETIOAPI_API_ ConvertInterfaceGuidToLuid (CONST GUID *InterfaceGuid, PNET_LUID InterfaceLuid);
+#ifndef __CYGWIN__
   NET_IFINDEX NETIOAPI_API_ if_nametoindex (PCSTR InterfaceName);
   PCHAR NETIOAPI_API_ if_indextoname (NET_IFINDEX InterfaceIndex, PCHAR InterfaceName);
+#endif
   NET_IF_COMPARTMENT_ID NETIOAPI_API_ GetCurrentThreadCompartmentId (VOID);
   NETIO_STATUS NETIOAPI_API_ SetCurrentThreadCompartmentId (NET_IF_COMPARTMENT_ID CompartmentId);
   NET_IF_COMPARTMENT_ID NETIOAPI_API_ GetSessionCompartmentId (ULONG SessionId);
