diff -Naur texinfo-5.1-orig/build-aux/compile texinfo-5.1/build-aux/compile
--- texinfo-5.1-orig/build-aux/compile	2013-02-19 22:25:28.000000000 +0000
+++ texinfo-5.1/build-aux/compile	2013-03-25 16:51:53.019531200 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur texinfo-5.1-orig/build-aux/config.guess texinfo-5.1/build-aux/config.guess
--- texinfo-5.1-orig/build-aux/config.guess	2013-02-19 22:25:28.000000000 +0000
+++ texinfo-5.1/build-aux/config.guess	2013-03-25 16:48:21.339843700 +0000
@@ -845,6 +845,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur texinfo-5.1-orig/build-aux/config.rpath texinfo-5.1/build-aux/config.rpath
--- texinfo-5.1-orig/build-aux/config.rpath	2013-02-19 22:25:28.000000000 +0000
+++ texinfo-5.1/build-aux/config.rpath	2013-03-25 16:48:54.636718700 +0000
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur texinfo-5.1-orig/configure texinfo-5.1/configure
--- texinfo-5.1-orig/configure	2013-03-12 22:56:45.000000000 +0000
+++ texinfo-5.1/configure	2013-03-26 11:13:54.098632800 +0000
@@ -9632,7 +9632,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
@@ -12348,7 +12348,7 @@
   $as_echo_n "(cached) " >&6
 else
   case $host_os in #(
-         *cygwin*)
+         *cygwin* | *msys*)
            lt_cv_sys_argz_works=no
            if test "$cross_compiling" != no; then
              lt_cv_sys_argz_works="guessing no"
@@ -12435,7 +12435,7 @@
 
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
@@ -13811,7 +13811,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur texinfo-5.1-orig/gnulib/lib/Makefile.am texinfo-5.1/gnulib/lib/Makefile.am
--- texinfo-5.1-orig/gnulib/lib/Makefile.am	2013-03-03 00:01:13.000000000 +0000
+++ texinfo-5.1/gnulib/lib/Makefile.am	2013-03-25 16:52:51.547851600 +0000
@@ -377,7 +377,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -Naur texinfo-5.1-orig/gnulib/lib/Makefile.in texinfo-5.1/gnulib/lib/Makefile.in
--- texinfo-5.1-orig/gnulib/lib/Makefile.in	2013-03-12 22:56:43.000000000 +0000
+++ texinfo-5.1/gnulib/lib/Makefile.in	2013-03-26 11:13:49.864257800 +0000
@@ -1671,7 +1671,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -Naur texinfo-5.1-orig/gnulib/m4/argz.m4 texinfo-5.1/gnulib/m4/argz.m4
--- texinfo-5.1-orig/gnulib/m4/argz.m4	2013-02-19 22:25:29.000000000 +0000
+++ texinfo-5.1/gnulib/m4/argz.m4	2013-03-25 16:53:36.986328100 +0000
@@ -43,7 +43,7 @@
         [if argz actually works],
         [lt_cv_sys_argz_works],
         [[case $host_os in #(
-         *cygwin*)
+         *cygwin* | *msys*)
            lt_cv_sys_argz_works=no
            if test "$cross_compiling" != no; then
              lt_cv_sys_argz_works="guessing no"
diff -Naur texinfo-5.1-orig/gnulib/m4/btowc.m4 texinfo-5.1/gnulib/m4/btowc.m4
--- texinfo-5.1-orig/gnulib/m4/btowc.m4	2013-02-19 22:25:29.000000000 +0000
+++ texinfo-5.1/gnulib/m4/btowc.m4	2013-03-25 16:53:51.330078100 +0000
@@ -49,7 +49,7 @@
 changequote(,)dnl
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
diff -Naur texinfo-5.1-orig/gnulib/m4/malloc.m4 texinfo-5.1/gnulib/m4/malloc.m4
--- texinfo-5.1-orig/gnulib/m4/malloc.m4	2013-02-19 22:25:29.000000000 +0000
+++ texinfo-5.1/gnulib/m4/malloc.m4	2013-03-25 16:54:15.548828100 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur texinfo-5.1-orig/install-info/tests/defs.in texinfo-5.1/install-info/tests/defs.in
--- texinfo-5.1-orig/install-info/tests/defs.in	2013-02-23 00:11:25.000000000 +0000
+++ texinfo-5.1/install-info/tests/defs.in	2013-03-25 16:55:01.319335900 +0000
@@ -59,7 +59,7 @@
 # directories in TEXINPUTS -- except for Cygwin et al., where COMSPEC
 # might be inherited, but : is used.
 if test -n "$COMSPEC$ComSpec" \
-   && uname | $EGREP -iv 'cygwin|mingw|djgpp' >/dev/null; then
+   && uname | $EGREP -iv 'cygwin|msys|mingw|djgpp' >/dev/null; then
   path_sep=";"
 else
   path_sep=":"
diff -Naur texinfo-5.1-orig/util/texi2dvi texinfo-5.1/util/texi2dvi
--- texinfo-5.1-orig/util/texi2dvi	2013-03-12 22:56:34.000000000 +0000
+++ texinfo-5.1/util/texi2dvi	2013-03-25 16:55:49.412109400 +0000
@@ -88,7 +88,7 @@
 # directories in TEXINPUTS -- except for Cygwin et al., where COMSPEC
 # might be inherited, but : is used.
 if test -n "$COMSPEC$ComSpec" \
-   && uname | $EGREP -iv 'cygwin|mingw|djgpp' >/dev/null; then
+   && uname | $EGREP -iv 'cygwin|msys|mingw|djgpp' >/dev/null; then
   path_sep=";"
 else
   path_sep=":"
