diff -Naur texinfo-6.7-orig/tp/Texinfo/XS/configure.ac texinfo-6.7/tp/Texinfo/XS/configure.ac
--- texinfo-6.7-orig/tp/Texinfo/XS/configure.ac	2019-08-25 20:11:45.000000000 +0300
+++ texinfo-6.7/tp/Texinfo/XS/configure.ac	2020-04-20 18:01:20.794900900 +0300
@@ -107,7 +107,8 @@
   PERL_CONF_libperl=`echo $PERL_CONF_libperl \
              | sed -e 's/^lib/-l/' \
                    -e 's/\..*//' \
-                   -e 's/^cygperl.*/-lperl/' `
+                     -e 's/^cygperl.*/-lperl/' \
+                     -e 's/^msys-perl.*/-lperl/' `
   
   lookup_perl_conf_values([[archlibexp], [privlibexp]])
   
