--- /dev/null	2013-07-27 05:06:16.000000000 +0000
+++ texinfo-5.1/install-info/install-info.manifest	2013-07-27 05:05:15.146514400 +0000
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+  <assemblyIdentity version="1.0.0.0"
+     processorArchitecture="x86"
+     name="GNU.texinfo.install-info"
+     type="win32"/>
+
+  <!-- Identify the application security requirements. -->
+  <v3:trustInfo xmlns:v3="urn:schemas-microsoft-com:asm.v3">
+    <v3:security>
+      <v3:requestedPrivileges>
+	<v3:requestedExecutionLevel level="asInvoker" uiAccess="false" />
+      </v3:requestedPrivileges>
+    </v3:security>
+  </v3:trustInfo>
+</assembly>
--- texinfo-5.1/install-info/Makefile.am.orig	2013-02-23 00:11:25.000000000 +0000
+++ texinfo-5.1/install-info/Makefile.am	2013-07-27 05:03:44.534508200 +0000
@@ -28,10 +28,17 @@
 # user-specified name transformations.
 transform = s/ginstall-info/install-info/; $(program_transform_name)
 
+install_info_DEPENDENCIES = install-info.res
+
 localedir = $(datadir)/locale
 AM_CPPFLAGS = 					\
   -I$(top_srcdir)				\
   -I$(top_srcdir)/gnulib/lib			\
   -I$(top_builddir)/gnulib/lib			\
   -DLOCALEDIR=\"$(localedir)\"
-LDADD = $(top_builddir)/gnulib/lib/libgnu.a $(LIBINTL) $(LIBTHREAD)
+LDADD = $(top_builddir)/gnulib/lib/libgnu.a $(LIBINTL) $(LIBTHREAD) $(install_info_DEPENDENCIES)
+
+
+install-info.res: install-info.rc install-info.manifest
+	$(WINDRES) --input $< --output $@ --output-format=coff
+
--- texinfo-5.1/configure.ac.orig	2013-03-12 22:56:34.000000000 +0000
+++ texinfo-5.1/configure.ac	2013-07-27 05:07:52.543001200 +0000
@@ -93,6 +93,8 @@
 AC_PROG_LN_S
 AC_PROG_MKDIR_P
 
+AC_CHECK_TOOL([WINDRES],[windres],[windres])
+
 # gettext Makefiles may still need MKINSTALLDIRS.
 MKINSTALLDIRS="\$(top_srcdir)/build-aux/install-sh -d"
 AC_SUBST([MKINSTALLDIRS])
--- /dev/null	2013-07-27 05:10:46.000000000 +0000
+++ texinfo-5.1/install-info/install-info.rc	2013-07-27 05:10:35.060138300 +0000
@@ -0,0 +1 @@
+1 24 install-info.manifest
