--- texinfo-5.1/configure.ac.orig	2013-07-27 05:14:49.920501500 +0000
+++ texinfo-5.1/configure.ac	2013-07-27 05:17:02.699362200 +0000
@@ -331,7 +331,7 @@
 # i18n support.  To update to a new version of gettext, run:
 #   gettextize -f -c
 # See http://gnu.org/s/gnulib/manual/html_node/gettextize-and-autopoint.html.
-AM_GNU_GETTEXT_VERSION([0.18.2])
+AM_GNU_GETTEXT_VERSION([0.18.1])
 AM_GNU_GETTEXT([external])
 
 USE_EXTERNAL_LIBINTL=no
