--- texinfo-5.1/util/Makefile.am.orig	2013-02-23 00:11:25.000000000 +0000
+++ texinfo-5.1/util/Makefile.am	2013-07-27 05:31:52.695877400 +0000
@@ -18,9 +18,6 @@
 nodist_noinst_SCRIPTS = txixml2texi
 
 w32_bat = makeinfo.bat pod2texi.bat texi2any.bat
-if HOST_IS_WINDOWS
- bin_SCRIPTS += $(w32_bat)
-endif
 
 # for auctex.
 pdftexi2dvi: texi2pdf
--- texinfo-5.1/install-info/Makefile.am.orig	2013-07-27 05:17:59.390561100 +0000
+++ texinfo-5.1/install-info/Makefile.am	2013-07-27 05:31:42.722611000 +0000
@@ -15,9 +15,6 @@
 bin_PROGRAMS = ginstall-info
 
 w32_script = ginstall-info.exe.manifest
-if HOST_IS_WINDOWS
- dist_bin_SCRIPTS = $(w32_script)
-endif
 
 # Use `ginstall-info' in the definition of PROGRAMS and in dependencies
 # because automake generates an install-info target in every Makefile
