--- texinfo-6.7/configure.ac.orig	2020-04-20 18:30:32.028533900 +0300
+++ texinfo-6.7/configure.ac	2020-04-20 18:30:58.115891400 +0300
@@ -314,7 +314,7 @@
 # i18n support.  To update to a new version of gettext, run:
 #   gettextize -f --po-dir=po --po-dir=po_document 
 # See http://gnu.org/s/gnulib/manual/html_node/gettextize-and-autopoint.html.
-AM_GNU_GETTEXT_VERSION([0.20])
+AM_GNU_GETTEXT_VERSION([0.19.8])
 AM_GNU_GETTEXT([external])
 
 USE_EXTERNAL_LIBINTL=no
