--- origsrc/texinfo-6.1/info/dir.c	2016-02-06 07:20:23.000000000 -0500
+++ src/texinfo-6.1/info/dir.c	2016-03-04 16:27:54.478422100 -0500
@@ -74,7 +74,7 @@ build_dir_node (void)
 "A few useful Info commands:\n"
 "\n"
 "  'q' quits;\n"
-"  '?' lists all Info commands;\n"
+"  'H' lists all Info commands;\n"
 "  'h' starts the Info tutorial;\n"
 "  'mTexinfo RET' visits the Texinfo manual, etc.\n"
 
--- origsrc/texinfo-6.1/install-info/install-info.c	2016-02-06 07:20:24.000000000 -0500
+++ src/texinfo-6.1/install-info/install-info.c	2016-03-04 16:27:54.478422100 -0500
@@ -2,7 +2,7 @@
    $Id: install-info.c 6906 2016-01-01 18:33:45Z karl $
 
    Copyright 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004,
-   2005, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015
+   2005, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016
    Free Software Foundation, Inc.
 
    This program is free software: you can redistribute it and/or modify
@@ -646,7 +646,7 @@ The first time you invoke Info you start
 %s\tThis is the top of the INFO tree\n\
 \n\
   This (the Directory node) gives a menu of major topics.\n\
-  Typing \"q\" exits, \"?\" lists all Info commands, \"d\" returns here,\n\
+  Typing \"q\" exits, \"H\" lists all Info commands, \"d\" returns here,\n\
   \"h\" gives a primer for first-timers,\n\
   \"mEmacs<Return>\" visits the Emacs manual, etc.\n\
 \n\
--- origsrc/texinfo-6.1/install-info/tests/ii-0018-expected-dir-file	2015-12-16 05:45:49.000000000 -0500
+++ src/texinfo-6.1/install-info/tests/ii-0018-expected-dir-file	2016-03-04 16:30:18.949743200 -0500
@@ -5,7 +5,7 @@ The first time you invoke Info you start
 File: dir,	Node: Top	This is the top of the INFO tree
 
   This (the Directory node) gives a menu of major topics.
-  Typing "q" exits, "?" lists all Info commands, "d" returns here,
+  Typing "q" exits, "H" lists all Info commands, "d" returns here,
   "h" gives a primer for first-timers,
   "mEmacs<Return>" visits the Emacs manual, etc.
 
--- origsrc/texinfo-6.1/install-info/tests/ii-0019-expected-dir-file	2015-12-16 06:01:06.000000000 -0500
+++ src/texinfo-6.1/install-info/tests/ii-0019-expected-dir-file	2016-03-04 16:30:59.488994700 -0500
@@ -5,7 +5,7 @@ The first time you invoke Info you start
 File: dir,	Node: Top	This is the top of the INFO tree
 
   This (the Directory node) gives a menu of major topics.
-  Typing "q" exits, "?" lists all Info commands, "d" returns here,
+  Typing "q" exits, "H" lists all Info commands, "d" returns here,
   "h" gives a primer for first-timers,
   "mEmacs<Return>" visits the Emacs manual, etc.
 
--- origsrc/texinfo-6.1/util/dir-example	2016-02-06 07:20:25.000000000 -0500
+++ src/texinfo-6.1/util/dir-example	2016-03-04 16:27:54.478422100 -0500
@@ -17,7 +17,7 @@ This is the Info main menu (aka director
 A few useful Info commands:
 
   `q' quits;
-  `?' lists all Info commands;
+  `H' lists all Info commands;
   `h' starts the Info tutorial;
   `mTexinfo RET' visits the Texinfo manual, etc.
 
