From d2d18d5ea279c9b5e00e4388ec6bb0982771ec98 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 5 Sep 2014 21:06:56 +0100
Subject: [PATCH 2/3] path: Fix cygwin path_conv of root

---
 winsup/cygwin/path.cc | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index be08707..2157358 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -709,6 +709,12 @@ path_conv::check (const char *src, unsigned opt,
 	      need_directory = 1;
 	      *--tail = '\0';
 	    }
+	    /* Special case for "/" must set need_directory, without removing
+	    trailing slash */
+	  else if (tail == path_copy+1 && isslash (tail[-1]))
+	    {
+	      need_directory = 1;
+	    }
 	  path_end = tail;
 
 	  /* Scan path_copy from right to left looking either for a symlink
-- 
2.1.0

