--- msys2-runtime/winsup/cygwin/path.cc.orig	2014-03-20 20:46:27.520673000 +0000
+++ msys2-runtime/winsup/cygwin/path.cc	2014-03-20 21:29:26.113160000 +0000
@@ -692,6 +692,12 @@
 	      need_directory = 1;
 	      *--tail = '\0';
 	    }
+	    /* Special case for "/" must set need_directory, without removing
+	    trailing slash */
+	  else if (tail == path_copy+1 && isslash (tail[-1]))
+	    {
+	      need_directory = 1;
+	    }
 	  path_end = tail;
 
 	  /* Scan path_copy from right to left looking either for a symlink
