From 90d13a556d561c4153f5c8db48b8df531a9fa4b5 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 1 Oct 2014 19:55:18 +0100
Subject: [PATCH] In create_root_entry, ensure native_root ends with a '/'

---
 winsup/cygwin/mount.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/winsup/cygwin/mount.cc b/winsup/cygwin/mount.cc
index 80e39b6..14a8f93 100644
--- a/winsup/cygwin/mount.cc
+++ b/winsup/cygwin/mount.cc
@@ -450,6 +450,11 @@ mount_info::create_root_entry (const PWCHAR root)
     The entry is immutable, unless the "override" option is given in /etc/fstab. */
   char native_root[PATH_MAX];
   sys_wcstombs (native_root, PATH_MAX, root);
+  /* Ensure that native_root ends with a / since src and dst entries in the
+     mount table must end consistently - otherwise a path conversion of '/'
+     will not end with '/' and things will break. */
+  if (strlen(native_root) < PATH_MAX - 1)
+      strcat(native_root, "/");
   assert (*native_root != '\0');
   if (add_item (native_root, "/",
 		MOUNT_SYSTEM | MOUNT_BINARY | MOUNT_IMMUTABLE | MOUNT_AUTOMATIC | MOUNT_NOACL)
-- 
2.1.0

