From 38579dc718d7d20fcd7ab7a84ff12684acf18ef3 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Mon, 2 Mar 2015 14:09:29 +0300
Subject: [PATCH 10/23] convert(): check safely whether we could allocate the
 buffer

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/path.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index b14a49ec4..c5b0547a9 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -3536,12 +3536,12 @@ arg_heuristic_with_exclusions (char const * const arg, char const * exclusions,
 
   size_t stack_len = arglen + MAX_PATH;
   char * stack_path = (char *)malloc (stack_len);
-  memset (stack_path, 0, MAX_PATH);
-  if (!stack_len)
+  if (!stack_path)
     {
       debug_printf ("out of stack space?");
       return (char *)arg;
     }
+  memset (stack_path, 0, MAX_PATH);
   convert (stack_path, stack_len - 1, arg);
   debug_printf ("convert()'ed: %s (length %d)\n.....->: %s", arg, arglen, stack_path);
   // Don't allocate memory if no conversion happened.
-- 
2.18.0

