From 488e69e40890cdbf162411e9560b38f01873adb3 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 21 Aug 2015 12:18:52 +0100
Subject: [PATCH 13/15] Add debugging for build_argv

---
 winsup/cygwin/dcrt0.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/winsup/cygwin/dcrt0.cc b/winsup/cygwin/dcrt0.cc
index 46200f17c..2dc4aae25 100644
--- a/winsup/cygwin/dcrt0.cc
+++ b/winsup/cygwin/dcrt0.cc
@@ -309,6 +309,8 @@ build_argv (char *cmd, char **&argv, int &argc, int winshell, int glob)
   argvlen = 0;
   argv = NULL;
 
+  debug_printf ("cmd = '%s', winshell = %d, glob = %d", cmd, winshell, glob);
+
   /* Scan command line until there is nothing left. */
   while (*cmd)
     {
-- 
2.21.0

