From ca6ce6de6beba21688ae06cc7c68847c5d82954d Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 21 Aug 2015 09:52:47 +0100
Subject: [PATCH 14/15] Add debugging for strace make_command_line

---
 winsup/utils/strace.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/winsup/utils/strace.cc b/winsup/utils/strace.cc
index 35953b85a..6c94158f3 100644
--- a/winsup/utils/strace.cc
+++ b/winsup/utils/strace.cc
@@ -351,6 +351,7 @@ create_child (char **argv)
     flags |= CREATE_NEW_CONSOLE | CREATE_NEW_PROCESS_GROUP;
 
   make_command_line (one_line, argv);
+  printf ("create_child: %s\n", one_line.buf);
 
   SetConsoleCtrlHandler (NULL, 0);
 /* Commit message for this code was:
-- 
2.21.0

