From 0c08a7f8af0e4a700caf43251c9e76ff9d818486 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 21 Aug 2015 12:18:52 +0100
Subject: [PATCH 20/23] Add debugging for build_argv

---
 winsup/cygwin/dcrt0.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/winsup/cygwin/dcrt0.cc b/winsup/cygwin/dcrt0.cc
index 7f69b33d4..705817679 100644
--- a/winsup/cygwin/dcrt0.cc
+++ b/winsup/cygwin/dcrt0.cc
@@ -309,6 +309,8 @@ build_argv (char *cmd, char **&argv, int &argc, int winshell, int glob)
   argvlen = 0;
   argv = NULL;
 
+  debug_printf ("cmd = '%s', winshell = %d, glob = %d", cmd, winshell, glob);
+
   /* Scan command line until there is nothing left. */
   while (*cmd)
     {
-- 
2.18.0

