From fb738ad5e0e3a95f14e5a3d47242ba13fe0dbdaf Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 21 Aug 2015 09:52:47 +0100
Subject: [PATCH 21/23] Add debugging for strace make_command_line

---
 winsup/utils/strace.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/winsup/utils/strace.cc b/winsup/utils/strace.cc
index 638b66293..ca148c372 100644
--- a/winsup/utils/strace.cc
+++ b/winsup/utils/strace.cc
@@ -351,6 +351,7 @@ create_child (char **argv)
     flags |= CREATE_NEW_CONSOLE | CREATE_NEW_PROCESS_GROUP;
 
   make_command_line (one_line, argv);
+  printf ("create_child: %s\n", one_line.buf);
 
   SetConsoleCtrlHandler (NULL, 0);
 /* Commit message for this code was:
-- 
2.18.0

