From 08904f622bae5918d52a024a774039a124b9a50d Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 15 May 2016 09:31:51 +0100
Subject: [PATCH 23/23] path.cc: Ignore zero-length exclusions

---
 winsup/cygwin/path.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index 3850a75..e266a94 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -3454,7 +3454,7 @@ arg_heuristic_with_exclusions (char const * const arg, char const * exclusions,
     {
       /* Since we've got regex linked we should maybe switch to that, but
          running regexes for every argument could be too slow. */
-      if ( strcmp (exclusions, "*") == 0 || strstr (arg, exclusions) == arg )
+      if ( strcmp (exclusions, "*") == 0 || (strlen (exclusions) && strstr (arg, exclusions) == arg) )
         return (char*)arg;
       exclusions += strlen (exclusions) + 1;
     }
-- 
2.9.0

