From 07dc2f408e5da9615c2eb6db07979a1a7592ef78 Mon Sep 17 00:00:00 2001
From: Einbert_Xeride <einbert-xeride@outlook.com>
Date: Tue, 25 Apr 2017 02:39:07 +0800
Subject: [PATCH 24/24] path.cc: Allow native unprivileged symlink

---
 winsup/cygwin/path.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index 13f4e39f5..e3aaae3f6 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -1813,8 +1813,9 @@ symlink_native (const char *oldpath, path_conv &win32_newpath)
     }
   /* Try to create native symlink. */
   if (!CreateSymbolicLinkW (final_newpath->Buffer, final_oldpath->Buffer,
-			    win32_oldpath.isdir ()
-			    ? SYMBOLIC_LINK_FLAG_DIRECTORY : 0))
+				(win32_oldpath.isdir ()
+				? SYMBOLIC_LINK_FLAG_DIRECTORY : 0)
+				| 0x2 /* SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE */ ))
     {
       /* Repair native newpath, we still need it. */
       final_newpath->Buffer[1] = L'?';
-- 
2.12.1

