From 8696868b113e1c2325b8ba5363e90f3634da1106 Mon Sep 17 00:00:00 2001
From: Einbert-Xeride <einbert-xeride@outlook.com>
Date: Tue, 25 Apr 2017 00:27:14 +0800
Subject: [PATCH] Allow unprivileged symlink

---
 winsup/cygwin/path.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index ee7636dbf..2304db8f5 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -1725,8 +1725,9 @@ symlink_native (const char *oldpath, path_conv &win32_newpath)
	 }
   /* Try to create native symlink. */
   if (!CreateSymbolicLinkW (final_newpath->Buffer, final_oldpath->Buffer,
-				win32_oldpath.isdir ()
-				? SYMBOLIC_LINK_FLAG_DIRECTORY : 0))
+				(win32_oldpath.isdir ()
+				? SYMBOLIC_LINK_FLAG_DIRECTORY : 0)
+				| 0x2 /* SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE */ ))
	 {
	   /* Repair native newpath, we still need it. */
	   final_newpath->Buffer[1] = L'?';
-- 
2.12.2

