From e70eb5fdbf98f6cbf92efef4c533c373df77ceaf Mon Sep 13 00:00:00 2017
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Sun, 13 Sep 2017 09:31:51 +0100
Subject: [PATCH] newlib: Fix possible segmentation fault in strnstr() on 64-bit systems

---
 newlib/libc/string/strnstr.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/newlib/libc/string/strnstr.c b/newlib/libc/string/strnstr.c
index ce32d0f73e..f6b007813f 100644
--- a/newlib/libc/string/strnstr.c
+++ b/newlib/libc/string/strnstr.c
@@ -31,6 +31,7 @@ QUICKREF
 */
 
 #undef __STRICT_ANSI__
+#define _GNU_SOURCE
 #include <_ansi.h>
 #include <string.h>
 
-- 
2.14.1
