From 8894ce93fad39ba0522245799bba8f8d714f3ea6 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Fri, 24 May 2019 13:41:47 +0300
Subject: [PATCH] QueryUnbiasedInterruptTime must be load from kernel32.dll

---
 winsup/cygwin/autoload.cc | 2 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/winsup/cygwin/autoload.cc b/winsup/cygwin/autoload.cc
index c4d91611e..86dd47547 100644
--- a/winsup/cygwin/autoload.cc
+++ b/winsup/cygwin/autoload.cc
@@ -593,7 +593,7 @@ LoadDLLfunc (SetThreadGroupAffinity, 12, kernel32)
    available via KernelBase.dll. */
 LoadDLLfunc (QueryInterruptTime, 4, KernelBase)
 LoadDLLfunc (QueryInterruptTimePrecise, 4, KernelBase)
-LoadDLLfunc (QueryUnbiasedInterruptTime, 4, KernelBase)
+LoadDLLfunc (QueryUnbiasedInterruptTime, 4, kernel32)
 LoadDLLfunc (QueryUnbiasedInterruptTimePrecise, 4, KernelBase)
 
 /* ldap functions are cdecl! */
-- 
2.21.0

