From 12ee69ae288878ed4dfec45f3d5abf69954cc435 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Wed, 24 Mar 2021 21:14:54 +0100
Subject: [PATCH 39/N] fixup! uname: limit sysname to "MSYS" or "MINGW"

---
 winsup/cygwin/uname.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/winsup/cygwin/uname.cc b/winsup/cygwin/uname.cc
index ad554ec..daed73f 100644
--- a/winsup/cygwin/uname.cc
+++ b/winsup/cygwin/uname.cc
@@ -39,7 +39,7 @@ uname_x (struct utsname *name)
       char* msystem = getenv("MSYSTEM");
       const char* msystem_sysname = "MSYS";
       if (msystem != NULL && *msystem && strcmp(msystem, "MSYS") != 0)
-        msystem_sysname = "MINGW";
+        msystem_sysname = (strstr(msystem, "32") != NULL) ? "MINGW32" : "MINGW64";;
       __small_sprintf (name->sysname, "%s_%s-%u%s",
 		       msystem_sysname,
 		       wincap.osname (), wincap.build_number (),
@@ -107,7 +107,7 @@ uname (struct utsname *in_name)
       char* msystem = getenv("MSYSTEM");
       const char* msystem_sysname = "MSYS";
       if (msystem != NULL && *msystem && strcmp(msystem, "MSYS") != 0)
-        msystem_sysname = "MINGW";
+        msystem_sysname = (strstr(msystem, "32") != NULL) ? "MINGW32" : "MINGW64";
       __small_sprintf (name->sysname, "%s_%s", msystem_sysname, wincap.osname ());
 #else
       __small_sprintf (name->sysname, "CYGWIN_%s", wincap.osname ());
-- 
2.31.0

