From e054c53f7de5eaa1dfca742a7d345e7b1e017927 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Mon, 23 Apr 2018 11:58:47 +0200
Subject: [PATCH 52/N] srandomdev(): avoid warning about "uninitialized use"

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/random.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/cygwin/random.cc b/winsup/cygwin/random.cc
index 163fc04..1df6b60 100644
--- a/winsup/cygwin/random.cc
+++ b/winsup/cygwin/random.cc
@@ -302,7 +302,7 @@ srandomdev()
 
 	if (getentropy ((void *) state, len)) {
 		struct timeval tv;
-		unsigned long junk;
+		unsigned long junk = 0;
 
 		gettimeofday(&tv, NULL);
 		/* Avoid a compiler warning when we really want to get at the
-- 
2.9.0

