From c87a1769878c7f4d5f7348126a5db7d72e184dd3 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Sun, 22 Apr 2018 23:40:33 +0200
Subject: [PATCH 54/N] fixup! kill: kill Win32 processes more gently

---
 winsup/utils/kill.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/winsup/utils/kill.cc b/winsup/utils/kill.cc
index 911eedb..955d4eb 100644
--- a/winsup/utils/kill.cc
+++ b/winsup/utils/kill.cc
@@ -174,7 +174,7 @@ forcekill (int pid, int sig, int wait)
   if (!wait || WaitForSingleObject (h, 200) != WAIT_OBJECT_0)
     {
       HANDLE cur = GetCurrentProcess (), h2;
-      /* duplicate handle with access rights required for exit_process() */
+      /* duplicate handle with access rights required for exit_process_tree() */
       if (DuplicateHandle (cur, h, cur, &h2, PROCESS_CREATE_THREAD |
 			      PROCESS_QUERY_INFORMATION |
 			      PROCESS_VM_OPERATION |
@@ -184,7 +184,7 @@ forcekill (int pid, int sig, int wait)
         CloseHandle(h);
         h = h2;
       }
-      exit_process (h2, 128 + sig, 1);
+      exit_process_tree (h2, 128 + sig);
     }
   CloseHandle (h);
 }
-- 
2.9.0

