From fa84a1f06b83b54847e6505df2bdb6db1860f47f Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Mon, 23 Apr 2018 11:58:47 +0200
Subject: [PATCH 55/N] srandomdev(): avoid warning about "uninitialized use"

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/random.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/cygwin/random.cc b/winsup/cygwin/random.cc
index 802c33b..ff2c5cc 100644
--- a/winsup/cygwin/random.cc
+++ b/winsup/cygwin/random.cc
@@ -310,7 +310,7 @@ srandomdev()
 
 	if (getentropy ((void *) state, len)) {
 		struct timeval tv;
-		unsigned long junk;
+		unsigned long junk = 0;
 
 		gettimeofday(&tv, NULL);
 		srandom((getpid() << 16) ^ tv.tv_sec ^ tv.tv_usec ^ dummy(&junk));
-- 
2.9.0

