From f9ea2574b90e18d1213ba28858aa2ce31570c451 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 10 Jan 2018 23:54:52 +0100
Subject: [PATCH 59/N] fixup! exit_process.h: fix handling of SIGINT and
 SIGTERM

---
 winsup/utils/kill.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/utils/kill.cc b/winsup/utils/kill.cc
index 734e656..05a282a 100644
--- a/winsup/utils/kill.cc
+++ b/winsup/utils/kill.cc
@@ -187,7 +187,7 @@ forcekill (int pid, int sig, int wait)
 	      CloseHandle (h2);
 	    }
 	  else
-	    terminate_process_tree(h, 128 + sig);
+	    exit_process (h, 128 + sig);
 	}
       else if (sig && !TerminateProcess (h, sig << 8)
           && WaitForSingleObject (h, 200) != WAIT_OBJECT_0)
-- 
2.9.0

