--- origsrc/src/winsup/cygwin/Makefile.in	2013-02-13 10:34:03.000000000 -0600
+++ src/src/winsup/cygwin/Makefile.in	2013-02-13 15:16:23.767777400 -0600
@@ -36,7 +36,6 @@ build_alias:=@build_alias@
 host_alias:=@host_alias@
 prefix:=@prefix@
 
-program_transform_name:=@program_transform_name@
 exec_prefix:=@exec_prefix@
 bindir:=@bindir@
 libdir:=@libdir@
--- origsrc/src/winsup/utils/Makefile.in	2013-01-29 15:54:42.000000000 -0600
+++ src/src/winsup/utils/Makefile.in	2013-02-13 15:17:36.678947700 -0600
@@ -37,7 +37,6 @@ prefix:=@prefix@
 exec_prefix:=@exec_prefix@
 
 bindir:=@bindir@
-program_transform_name:=@program_transform_name@
 
 override INSTALL:=@INSTALL@
 override INSTALL_PROGRAM:=@INSTALL_PROGRAM@
