--- msys2-runtime/winsup/cygwin/nlsfuncs.cc.orig	2023-09-14 22:15:33.262109300 +0200
+++ msys2-runtime/winsup/cygwin/nlsfuncs.cc	2023-09-14 22:07:27.497844600 +0200
@@ -19,6 +19,8 @@
 #include "lc_msg.h"
 #include "lc_era.h"
 
+#pragma GCC diagnostic ignored "-Wuse-after-free"
+
 #define _LC(x)	&lc_##x##_ptr,lc_##x##_end-lc_##x##_ptr
 
 #define getlocaleinfo(category,type) \
@@ -1613,3 +1615,5 @@
   if (ret)
     internal_setlocale ();
 }
+
+#pragma GCC diagnostic pop
\ No newline at end of file
