--- origsrc/util-linux-2.24.2/libblkid/src/Makemodule.am	2014-04-24 02:37:04.354704100 -0500
+++ src/util-linux-2.24.2/libblkid/src/Makemodule.am	2014-07-08 21:17:41.022588300 -0500
@@ -94,6 +94,7 @@ libblkid_la_SOURCES = \
 	libblkid/src/superblocks/xfs.c \
 	libblkid/src/superblocks/zfs.c \
 	\
+	libblkid/src/topology/ioctl.c \
 	libblkid/src/topology/topology.c \
 	libblkid/src/topology/topology.h
 
@@ -101,7 +102,6 @@ if LINUX
 libblkid_la_SOURCES += \
 	libblkid/src/topology/dm.c \
 	libblkid/src/topology/evms.c \
-	libblkid/src/topology/ioctl.c \
 	libblkid/src/topology/lvm.c \
 	libblkid/src/topology/md.c \
 	libblkid/src/topology/sysfs.c
--- origsrc/util-linux-2.24.2/libblkid/src/topology/topology.c	2014-04-23 08:28:29.146326300 -0500
+++ src/util-linux-2.24.2/libblkid/src/topology/topology.c	2014-07-08 21:17:17.354582900 -0500
@@ -74,7 +74,9 @@ struct blkid_struct_topology {
  */
 static const struct blkid_idinfo *idinfos[] =
 {
-#ifdef __linux__
+#if defined(__CYGWIN__)
+	&ioctl_tp_idinfo
+#elif defined(__linux__)
 	&ioctl_tp_idinfo,
 	&sysfs_tp_idinfo,
 	&md_tp_idinfo,
--- origsrc/util-linux-2.24.2/libblkid/src/topology/topology.h	2013-02-27 10:46:29.873020600 -0600
+++ src/util-linux-2.24.2/libblkid/src/topology/topology.h	2014-07-08 21:17:17.357583300 -0500
@@ -11,8 +11,10 @@ extern int blkid_topology_set_physical_s
 /*
  * topology probers
  */
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
 extern const struct blkid_idinfo ioctl_tp_idinfo;
+#endif
+#ifdef __linux__
 extern const struct blkid_idinfo md_tp_idinfo;
 extern const struct blkid_idinfo evms_tp_idinfo;
 extern const struct blkid_idinfo sysfs_tp_idinfo;
