ipcrm and ipcs are provided by ipc-utils

--- origsrc/util-linux-2.24.2/sys-utils/Makemodule.am	2014-04-24 02:37:04.454705800 -0500
+++ src/util-linux-2.24.2/sys-utils/Makemodule.am	2014-07-08 21:15:11.943157700 -0500
@@ -9,6 +9,7 @@ dist_man_MANS += sys-utils/ipcmk.1
 ipcmk_SOURCES = sys-utils/ipcmk.c
 ipcmk_LDADD = $(LDADD) libcommon.la
 
+if LINUX
 usrbin_exec_PROGRAMS += ipcrm
 dist_man_MANS += sys-utils/ipcrm.1
 ipcrm_SOURCES = sys-utils/ipcrm.c
@@ -20,6 +21,7 @@ ipcs_SOURCES =	sys-utils/ipcs.c \
 		sys-utils/ipcutils.c \
 		sys-utils/ipcutils.h
 ipcs_LDADD = $(LDADD) libcommon.la
+endif
 
 
 usrbin_exec_PROGRAMS += renice
