--- origsrc/util-linux-2.25.1/configure.ac	2014-10-20 12:35:57.424262800 -0500
+++ src/util-linux-2.25.1/configure.ac	2014-10-22 19:25:17.942302200 -0500
@@ -1101,8 +1101,11 @@ UL_REQUIRES_HAVE([lscpu], [cpu_set_t], [
 AM_CONDITIONAL([BUILD_LSCPU], [test "x$build_lscpu" = xyes])
 
 
-UL_BUILD_INIT([lslogins], [check])
-UL_REQUIRES_BUILD([lslogins], [libsmartcols])
+AC_ARG_ENABLE([lslogins],
+  AS_HELP_STRING([--disable-lslogins], [do not build lslogins]),
+  [], [UL_DEFAULT_ENABLE([lslogins], [check])]
+)
+UL_BUILD_INIT([lslogins])
 AM_CONDITIONAL([BUILD_LSLOGINS], [test "x$build_lslogins" = xyes])
 
 
