diff -Naur origsrc/util-linux-2.25.1/lib/ttyutils.c src/util-linux-2.25.1/lib/ttyutils.c
--- origsrc/util-linux-2.25.1/lib/ttyutils.c	2013-09-18 08:50:12.671263500 -0500
+++ src/util-linux-2.25.1/lib/ttyutils.c	2014-11-16 17:25:14.277411500 -0600
@@ -22,10 +22,14 @@
 #ifdef TIOCGSIZE
 	if (ioctl (STDIN_FILENO, TIOCGSIZE, &t_win) == 0)
 		return t_win.ts_cols;
+	if (ioctl (STDOUT_FILENO, TIOCGSIZE, &t_win) == 0)
+		return t_win.ts_cols;
 #endif
 #ifdef TIOCGWINSZ
 	if (ioctl (STDIN_FILENO, TIOCGWINSZ, &w_win) == 0)
 		return w_win.ws_col;
+	if (ioctl (STDOUT_FILENO, TIOCGWINSZ, &w_win) == 0)
+		return w_win.ws_col;
 #endif
         cp = getenv("COLUMNS");
 	if (cp) {
