--- origsrc/util-linux-2.25.1/include/blkdev.h	2014-08-27 06:06:31.182865100 -0500
+++ src/util-linux-2.25.1/include/blkdev.h	2014-10-20 12:23:01.868981200 -0500
@@ -21,6 +21,10 @@
 
 #define DEFAULT_SECTOR_SIZE       512
 
+#ifdef __CYGWIN__
+#include <cygwin/fs.h>
+#endif
+
 #ifdef __linux__
 /* very basic ioctls, should be available everywhere */
 # ifndef BLKROSET
--- origsrc/util-linux-2.25.1/include/ttyutils.h	2014-04-24 02:37:04.352704100 -0500
+++ src/util-linux-2.25.1/include/ttyutils.h	2014-10-20 12:23:01.878981200 -0500
@@ -80,7 +80,10 @@ static inline void reset_virtual_console
 	tp->c_oflag &= ~(OLCUC | OCRNL | ONOCR | ONLRET | OFILL | \
 			    NLDLY|CRDLY|TABDLY|BSDLY|VTDLY|FFDLY);
 	tp->c_lflag |=  (ISIG | ICANON | IEXTEN | ECHO|ECHOE|ECHOK|ECHOKE|ECHOCTL);
-	tp->c_lflag &= ~(ECHONL|ECHOPRT | NOFLSH | TOSTOP);
+	tp->c_lflag &= ~(ECHONL | NOFLSH | TOSTOP);
+#ifdef ECHOPRT
+	tp->c_lflag &= ~ECHOPRT;
+#endif
 
 	if ((flags & UL_TTY_KEEPCFLAGS) == 0) {
 		tp->c_cflag |=  (CREAD | CS8 | HUPCL);
--- origsrc/util-linux-2.25.1/lib/randutils.c	2014-08-26 09:48:45.079654000 -0500
+++ src/util-linux-2.25.1/lib/randutils.c	2014-10-20 12:23:01.878981200 -0500
@@ -13,8 +13,6 @@
 #include <string.h>
 #include <sys/time.h>
 
-#include <sys/syscall.h>
-
 #include "c.h"
 #include "randutils.h"
 #include "nls.h"
@@ -26,6 +24,7 @@
 #endif
 
 #if defined(__linux__) && defined(__NR_gettid) && defined(HAVE_JRAND48)
+#include <sys/syscall.h>
 #define DO_JRAND_MIX
 THREAD_LOCAL unsigned short ul_jrand_seed[3];
 #endif
