newlib scanf(3) does not support 'm'

--- origsrc/util-linux-2.25.1/lib/colors.c	2014-09-03 02:50:42.187957900 -0500
+++ src/util-linux-2.25.1/lib/colors.c	2014-10-22 17:25:41.257979100 -0500
@@ -561,7 +561,8 @@ static int colors_read_schemes(struct ul
 	}
 
 	while (fgets(buf, sizeof(buf), f)) {
-		char *cn = NULL, *seq = NULL;
+		char *cn = malloc (sizeof(buf) + 1);
+		char *seq = malloc (sizeof(buf) + 1);
 		char *p = strchr(buf, '\n');
 
 		if (!p) {
@@ -577,9 +578,9 @@ static int colors_read_schemes(struct ul
 		if (*p == '\0' || *p == '#')
 			continue;
 
-		rc = sscanf(p,  UL_SCNsA" "	/* name */
-				UL_SCNsA,	/* color */
-				&cn, &seq);
+		rc = sscanf(p,  "%s "	/* name */
+				"%s",	/* color */
+				cn, seq);
 		if (rc == 2 && cn && seq)
 			rc = colors_add_scheme(cc, cn, seq);	/* set rc=0 on success */
 		if (rc) {
