--- origsrc/util-linux-2.25.1/disk-utils/cfdisk.c	2014-09-03 02:50:42.185957900 -0500
+++ src/util-linux-2.25.1/disk-utils/cfdisk.c	2014-10-22 19:09:20.709249100 -0500
@@ -199,10 +199,11 @@ static void resize(void)
 
 	if (ioctl(fileno(stdout), TIOCGWINSZ, &ws) != -1
 	    && ws.ws_row && ws.ws_col) {
-		LINES = ws.ws_row;
-		COLS = ws.ws_col;
 #if HAVE_RESIZETERM
 		resizeterm(ws.ws_row, ws.ws_col);
+#else
+		LINES = ws.ws_row;
+		COLS = ws.ws_col;
 #endif
 		clearok(stdscr, TRUE);
 	}
