--- a/tests/helpers/Makemodule.am	2018-06-04 09:57:02.000000000 +0200
+++ b/tests/helpers//Makemodule.am	2018-09-23 09:52:49.157048900 +0200
@@ -21,9 +21,6 @@
 test_sigreceive_SOURCES = tests/helpers/test_sigreceive.c
 test_sigreceive_LDADD = $(LDADD) libcommon.la
 
-check_PROGRAMS += test_tiocsti
-test_tiocsti_SOURCES = tests/helpers/test_tiocsti.c
-
 check_PROGRAMS += test_uuid_namespace
 test_uuid_namespace_SOURCES = tests/helpers/test_uuid_namespace.c \
 	libuuid/src/predefined.c libuuid/src/unpack.c libuuid/src/unparse.c
