--- origsrc/util-linux-2.39.3/include/blkdev.h	2023-11-21 00:46:50.040993300 -0800
+++ src/util-linux-2.39.3/include/blkdev.h	2024-03-06 21:49:24.330239000 -0800
@@ -23,6 +23,10 @@
 
 #define DEFAULT_SECTOR_SIZE       512
 
+#ifdef __CYGWIN__
+#include <cygwin/fs.h>
+#endif
+
 #ifdef __linux__
 /* very basic ioctls, should be available everywhere */
 # ifndef BLKROSET
--- origsrc/util-linux-2.39.3/include/ttyutils.h	2023-12-01 03:25:15.854048300 -0800
+++ src/util-linux-2.39.3/include/ttyutils.h	2024-03-07 02:05:32.567596700 -0800
@@ -164,7 +164,11 @@ static inline void reset_virtual_console
 	tp->c_oflag &= ~(OLCUC | OCRNL | ONOCR | ONLRET | OFILL | \
 			    NLDLY|CRDLY|TABDLY|BSDLY|VTDLY|FFDLY);
 	tp->c_lflag |=  (ISIG | ICANON | IEXTEN | ECHO|ECHOE|ECHOK|ECHOKE|ECHOCTL);
+#ifdef ECHOPRT
 	tp->c_lflag &= ~(ECHONL|ECHOPRT | NOFLSH | TOSTOP);
+#else
+	tp->c_lflag &= ~(ECHONL| NOFLSH | TOSTOP);
+#endif
 
 	if ((flags & UL_TTY_KEEPCFLAGS) == 0) {
 		tp->c_cflag |=  (CREAD | CS8 | HUPCL);
