--- origsrc/util-linux-2.39.3/tests/helpers/test_tiocsti.c	2023-11-21 00:46:50.245992100 -0800
+++ src/util-linux-2.39.3/tests/helpers/test_tiocsti.c	2024-03-06 22:26:34.506716700 -0800
@@ -11,6 +11,7 @@
 
 int main(void)
 {
+#ifdef TIOCSTI
 	int rc = 0;
 	char *cmd = "id -u -n\n";
 
@@ -18,4 +19,7 @@ int main(void)
 		rc += ioctl(0, TIOCSTI, cmd++);
 
 	exit(rc ? EXIT_FAILURE : EXIT_SUCCESS);
+#else
+	return 77;
+#endif
 }
