--- origsrc/util-linux-2.39.3/configure.ac	2024-03-06 22:38:09.882861600 -0800
+++ src/util-linux-2.39.3/configure.ac	2024-03-07 01:27:42.682134800 -0800
@@ -2367,7 +2367,7 @@ AC_ARG_ENABLE([more],
 )
 UL_BUILD_INIT([more])
 UL_REQUIRES_HAVE([more], [ncursesw, ncurses], [ncursesw or ncurses libraries])
-UL_REQUIRES_LINUX([more])
+# Cygwin OK here too! UL_REQUIRES_LINUX([more])
 AM_CONDITIONAL([BUILD_MORE], [test "x$build_more" = xyes])
 
 
--- origsrc/util-linux-2.39.3/text-utils/more.c	2023-12-01 03:25:16.037047200 -0800
+++ src/util-linux-2.39.3/text-utils/more.c	2024-03-07 02:29:48.127056400 -0800
@@ -57,7 +57,9 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/file.h>
-#include <sys/ttydefaults.h>
+#ifdef HAVE_SYS_TTYDEFAULTS_H
+# include <sys/ttydefaults.h>
+#endif
 #include <sys/wait.h>
 #include <regex.h>
 #include <assert.h>
