--- origsrc/util-linux-2.40.2/libblkid/src/probe.c	2024-07-04 00:54:41.225242100 -0700
+++ src/util-linux-2.40.2/libblkid/src/probe.c	2024-12-21 14:48:51.789312800 -0800
@@ -970,7 +970,15 @@ failed:
 #ifdef BLKIOOPT
 static uint64_t blkid_get_io_size(int fd)
 {
+#ifdef BLKBSZGET
+#ifdef __CYGWIN__
+#undef BLKBSZGET
+#define BLKBSZGET  _IOR(0x12,112,size_t)
+#endif
 	static const int ioctls[] = { BLKIOOPT, BLKIOMIN, BLKBSZGET };
+#else
+	static const int ioctls[] = { BLKIOOPT, BLKIOMIN };
+#endif
 	unsigned int s;
 	size_t i;
 	int r;
