--- origsrc/util-linux-2.40.2/configure.ac	2025-02-02 18:00:40.679521900 -0800
+++ src/util-linux-2.40.2/configure.ac	2025-02-02 18:08:21.224583300 -0800
@@ -2225,8 +2225,9 @@ UL_REQUIRES_HAVE([scriptlive], [pty], [o
 AM_CONDITIONAL([BUILD_SCRIPTLIVE], [test "x$build_scriptlive" = xyes])
 
 
-UL_BUILD_INIT([col], [check])
-UL_REQUIRES_COMPILE([col], [#include <limits.h>], [__GLIBC__], [building for glibc])
+UL_BUILD_INIT([col], [yes])
+#Cygwin OK here too! UL_BUILD_INIT([col], [check])
+#Cygwin OK here too! UL_REQUIRES_COMPILE([col], [#include <limits.h>], [__GLIBC__], [building for glibc])
 AM_CONDITIONAL([BUILD_COL], [test "x$build_col" = xyes])
 
 UL_BUILD_INIT([colcrt], [yes])
