--- origsrc/util-linux-2.40.2/libuuid/src/uuid_time.c	2024-07-04 00:54:41.230242000 -0700
+++ src/util-linux-2.40.2/libuuid/src/uuid_time.c	2024-10-25 01:04:10.470713100 -0700
@@ -85,6 +85,8 @@ time_t __uuid_time(const uuid_t uu, stru
 }
 #if defined(__USE_TIME_BITS64) && defined(__GLIBC__)
 extern time_t uuid_time64(const uuid_t uu, struct timeval *ret_tv) __attribute__((weak, alias("__uuid_time")));
+#elif defined(__CYGWIN__)
+extern time_t uuid_time(const uuid_t uu, struct timeval *ret_tv) __attribute__((alias("__uuid_time")));
 #else
 extern time_t uuid_time(const uuid_t uu, struct timeval *ret_tv) __attribute__((weak, alias("__uuid_time")));
 #endif
