--- util-linux-2.40.2/configure.ac.orig      2025-02-06 11:29:31.296507100 -0800
+++ util-linux-2.40.2/configure.ac   2025-02-06 11:33:08.141674200 -0800
@@ -2535,9 +2535,8 @@
 UL_ENABLE_ALIAS([taskset], [schedutils])
 UL_BUILD_INIT([taskset])
 UL_REQUIRES_HAVE([taskset], [cpu_set_t], [cpu_set_t type])
-UL_REQUIRES_SYSCALL_CHECK([taskset],
-	[UL_CHECK_SYSCALL([sched_getaffinity])],
-	[sched_getaffinity])
+AC_CHECK_FUNCS([sched_getaffinity sched_setaffinity], [have_schedaffinity=yes],[have_schedaffinity=no])
+UL_REQUIRES_HAVE([taskset], [schedaffinity], [sched affinity functions])
 AM_CONDITIONAL([BUILD_TASKSET], [test "x$build_taskset" = xyes])
