--- rtorrent-0.9.6-orig/src/utils/directory.cc  2015-09-03 19:03:45.000000000 +0000
+++ rtorrent-0.9.6/src/utils/directory.cc       2016-11-07 16:38:18.298862400 +0000
@@ -71,7 +71,7 @@
     return false;

   struct dirent* entry;
-#ifdef __sun__
+#if defined(__sun__) || defined(__MSYS__)
   struct stat s;
 #endif

@@ -81,7 +81,7 @@

     iterator itr = base_type::insert(end(), value_type());

-#ifdef __sun__
+#if defined(__sun__) || defined(__MSYS__)
     stat(entry->d_name, &s);
     itr->d_fileno = entry->d_ino;
     itr->d_reclen = 0;

