diff -Naur rsync-3.1.0-orig/config.guess rsync-3.1.0/config.guess
--- rsync-3.1.0-orig/config.guess	2006-11-02 05:01:25.000000000 +0300
+++ rsync-3.1.0/config.guess	2013-07-21 07:35:52.411359400 +0400
@@ -811,6 +814,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur rsync-3.0.9-orig/configure.ac rsync-3.0.9/configure.ac
--- rsync-3.0.9-orig/configure.ac	2011-09-23 20:13:53.000000000 +0400
+++ rsync-3.0.9/configure.ac	2013-07-21 07:37:32.989484400 +0400
@@ -299,7 +299,7 @@
 				ipv6libdir=/usr/local/v6/lib;
 				AC_DEFINE(INET6, 1, [true if you have IPv6])])
 			;;
-		cygwin)
+		cygwin|msys)
 			AC_EGREP_CPP(yes, [
 #include <netinet/in.h>
 #ifdef _CYGWIN_IN6_H
@@ -330,7 +330,7 @@
 
 AC_MSG_CHECKING([whether to call shutdown on all sockets])
 case $host_os in
-	*cygwin* ) AC_MSG_RESULT(yes)
+	*cygwin*|*msys* ) AC_MSG_RESULT(yes)
                    AC_DEFINE(SHUTDOWN_ALL_SOCKETS, 1,
 			    [Define to 1 if sockets need to be shutdown])
 		   ;;
@@ -632,7 +632,7 @@
 if test x"$ac_cv_func_posix_fallocate" = x"yes"; then
     AC_MSG_CHECKING([whether posix_fallocate is efficient])
     case $host_os in
-    *cygwin*)
+    *cygwin*|*msys*)
 	AC_MSG_RESULT(yes)
 	AC_DEFINE(HAVE_EFFICIENT_POSIX_FALLOCATE, 1,
 		  [Define if posix_fallocate is efficient (Cygwin)])
@@ -976,7 +976,7 @@
 	AC_DEFINE(HAVE_UNIXWARE_ACLS, 1, [true if you have UnixWare ACLs])
 	AC_DEFINE(SUPPORT_ACLS, 1, [Define to 1 to add support for ACLs])
 	;;
-    solaris*|*cygwin*)
+    solaris*|*cygwin*|*msys*)
 	AC_MSG_RESULT(Using solaris ACLs)
 	AC_DEFINE(HAVE_SOLARIS_ACLS, 1, [true if you have solaris ACLs])
 	AC_DEFINE(SUPPORT_ACLS, 1)
diff -Naur rsync-3.1.0-orig/configure.sh rsync-3.1.0/configure.sh
--- rsync-3.1.0-orig/configure.sh	2011-09-23 20:41:30.000000000 +0400
+++ rsync-3.1.0/configure.sh	2013-07-21 07:38:41.786359400 +0400
@@ -4819,7 +4819,7 @@
 rm -f conftest*
 
 			;;
-		cygwin)
+		cygwin|msys)
 			cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -4918,7 +4918,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to call shutdown on all sockets" >&5
 $as_echo_n "checking whether to call shutdown on all sockets... " >&6; }
 case $host_os in
-	*cygwin* ) { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+	*cygwin*|*msys* ) { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
 $as_echo "#define SHUTDOWN_ALL_SOCKETS 1" >>confdefs.h
@@ -7751,7 +7751,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether posix_fallocate is efficient" >&5
 $as_echo_n "checking whether posix_fallocate is efficient... " >&6; }
     case $host_os in
-    *cygwin*)
+    *cygwin*|*msys*)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
@@ -8661,7 +8661,7 @@
 $as_echo "#define SUPPORT_ACLS 1" >>confdefs.h
 
 	;;
-    solaris*|*cygwin*)
+    solaris*|*cygwin*|*msys*)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: Using solaris ACLs" >&5
 $as_echo "Using solaris ACLs" >&6; }
 
