diff -Naur libpipeline-1.5.1-orig/build-aux/ar-lib libpipeline-1.5.1/build-aux/ar-lib
--- libpipeline-1.5.1-orig/build-aux/ar-lib	2019-01-27 17:02:33.000000000 +0300
+++ libpipeline-1.5.1/build-aux/ar-lib	2019-02-20 09:02:03.388256200 +0300
@@ -56,6 +56,9 @@
 	  CYGWIN*)
 	    file_conv=cygwin
 	    ;;
+	  MSYS*)
+	    file_conv=cygwin
+	    ;;
 	  *)
 	    file_conv=wine
 	    ;;
@@ -68,6 +71,9 @@
 	cygwin)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
+	msys)
+	  file=`cygpath -m "$file" || echo "$file"`
+	  ;;
 	wine)
 	  file=`winepath -w "$file" || echo "$file"`
 	  ;;
diff -Naur libpipeline-1.5.1-orig/build-aux/compile libpipeline-1.5.1/build-aux/compile
--- libpipeline-1.5.1-orig/build-aux/compile	2019-01-27 17:02:33.000000000 +0300
+++ libpipeline-1.5.1/build-aux/compile	2019-02-20 09:02:55.572548100 +0300
@@ -56,6 +56,9 @@
 	  CYGWIN*)
 	    file_conv=cygwin
 	    ;;
+	  MSYS*)
+	    file_conv=cygwin
+	    ;;
 	  *)
 	    file_conv=wine
 	    ;;
@@ -70,6 +73,9 @@
 	cygwin/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
+	msys/*)
+	  file=`cygpath -m "$file" || echo "$file"`
+	  ;;
 	wine/*)
 	  file=`winepath -w "$file" || echo "$file"`
 	  ;;
diff -Naur libpipeline-1.5.1-orig/build-aux/config.guess libpipeline-1.5.1/build-aux/config.guess
--- libpipeline-1.5.1-orig/build-aux/config.guess	2019-01-27 16:48:01.000000000 +0300
+++ libpipeline-1.5.1/build-aux/config.guess	2019-02-20 09:03:27.399604100 +0300
@@ -896,6 +896,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-pc-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
diff -Naur libpipeline-1.5.1-orig/build-aux/config.rpath libpipeline-1.5.1/build-aux/config.rpath
--- libpipeline-1.5.1-orig/build-aux/config.rpath	2019-01-05 14:20:49.000000000 +0300
+++ libpipeline-1.5.1/build-aux/config.rpath	2019-02-20 08:59:56.838833800 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur libpipeline-1.5.1-orig/gl/m4/double-slash-root.m4 libpipeline-1.5.1/gl/m4/double-slash-root.m4
--- libpipeline-1.5.1-orig/gl/m4/double-slash-root.m4	2019-01-05 14:20:51.000000000 +0300
+++ libpipeline-1.5.1/gl/m4/double-slash-root.m4	2019-02-20 09:04:33.340328000 +0300
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur libpipeline-1.5.1-orig/gl/m4/malloc.m4 libpipeline-1.5.1/gl/m4/malloc.m4
--- libpipeline-1.5.1-orig/gl/m4/malloc.m4	2019-01-05 14:20:51.000000000 +0300
+++ libpipeline-1.5.1/gl/m4/malloc.m4	2019-02-20 09:23:03.458478800 +0300
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur libpipeline-1.5.1-orig/gl/m4/printf.m4 libpipeline-1.5.1/gl/m4/printf.m4
--- libpipeline-1.5.1-orig/gl/m4/printf.m4	2019-01-27 16:48:24.000000000 +0300
+++ libpipeline-1.5.1/gl/m4/printf.m4	2019-02-20 09:23:33.878532200 +0300
@@ -785,6 +785,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
                             # Guess no on Android.
            linux*-android*) gl_cv_func_printf_directive_ls="guessing no";;
