diff -durN libpipeline-1.4.1.orig/build-aux/config.guess libpipeline-1.4.1/build-aux/config.guess
--- libpipeline-1.4.1.orig/build-aux/config.guess	2017-05-17 10:35:06.968750000 +0800
+++ libpipeline-1.4.1/build-aux/config.guess	2017-05-17 10:58:47.937500000 +0800
@@ -891,6 +891,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -durN libpipeline-1.4.1.orig/build-aux/config.rpath libpipeline-1.4.1/build-aux/config.rpath
--- libpipeline-1.4.1.orig/build-aux/config.rpath	2017-05-17 10:35:06.968750000 +0800
+++ libpipeline-1.4.1/build-aux/config.rpath	2017-05-17 10:58:47.937500000 +0800
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -durN libpipeline-1.4.1.orig/gnulib/m4/double-slash-root.m4 libpipeline-1.4.1/gnulib/m4/double-slash-root.m4
--- libpipeline-1.4.1.orig/gnulib/m4/double-slash-root.m4	2017-05-17 10:35:41.640625000 +0800
+++ libpipeline-1.4.1/gnulib/m4/double-slash-root.m4	2017-05-17 11:04:32.203125000 +0800
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -durN libpipeline-1.4.1.orig/gnulib/m4/malloc.m4 libpipeline-1.4.1/gnulib/m4/malloc.m4
--- libpipeline-1.4.1.orig/gnulib/m4/malloc.m4	2017-05-17 10:35:49.015625000 +0800
+++ libpipeline-1.4.1/gnulib/m4/malloc.m4	2017-05-17 11:06:30.671875000 +0800
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -durN libpipeline-1.4.1.orig/gnulib/m4/printf.m4 libpipeline-1.4.1/gnulib/m4/printf.m4
--- libpipeline-1.4.1.orig/gnulib/m4/printf.m4	2017-05-17 10:35:52.421875000 +0800
+++ libpipeline-1.4.1/gnulib/m4/printf.m4	2017-05-17 11:07:41.343750000 +0800
@@ -730,6 +730,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+		   msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -807,6 +808,7 @@
 changequote(,)dnl
          case "$host_os" in
            cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+		   msys*)        gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -1459,6 +1461,8 @@
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
            cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+                                 # Guess yes on msys.
+           msys*)                gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -durN libpipeline-1.4.1.orig/gnulib/m4/threadlib.m4 libpipeline-1.4.1/gnulib/m4/threadlib.m4
--- libpipeline-1.4.1.orig/gnulib/m4/threadlib.m4	2017-05-17 10:36:01.656250000 +0800
+++ libpipeline-1.4.1/gnulib/m4/threadlib.m4	2017-05-17 11:08:22.250000000 +0800
@@ -68,7 +68,7 @@
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
