diff -Naur xorriso-1.4.8-orig/config.guess xorriso-1.4.8/config.guess
--- xorriso-1.4.8-orig/config.guess	2017-09-12 18:39:26.000000000 +0300
+++ xorriso-1.4.8/config.guess	2017-11-17 09:00:38.173592100 +0300
@@ -851,6 +851,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur xorriso-1.4.8-orig/configure.ac xorriso-1.4.8/configure.ac
--- xorriso-1.4.8-orig/configure.ac	2017-09-12 18:39:26.000000000 +0300
+++ xorriso-1.4.8/configure.ac	2017-11-17 08:59:18.170844200 +0300
@@ -379,7 +379,7 @@
 
 
 case $host_os in
-    cygwin*|mingw*)
+    cygwin*|msys*|mingw*)
         default_libcdio=yes
         ;;
     *)
