--- automake-1.15.1/bin/automake.in.orig	2017-10-01 16:51:34.179938300 -0400
+++ automake-1.15.1/bin/automake.in	2017-10-01 18:13:47.444365300 -0400
@@ -3879,7 +3879,7 @@ sub substitute_ac_subst_variables_worker
 sub substitute_ac_subst_variables
 {
   my ($text) = @_;
-  $text =~ s/\$[{]([^ \t=:+{}]+)}/substitute_ac_subst_variables_worker ($1)/ge;
+  $text =~ s/\$[\{]([^ \t=:+{}]+)}/substitute_ac_subst_variables_worker ($1)/ge;
   return $text;
 }
 
