diff -ur irssi-orig/irssi.conf irssi/irssi.conf
--- irssi-orig/irssi.conf	2015-03-14 12:09:29.769857100 -0500
+++ irssi/irssi.conf	2015-03-14 12:10:59.681229900 -0500
@@ -97,6 +97,7 @@
   { name = "#lobby";     chatnet = "EsperNet";  autojoin = "No"; },
   { name = "#freenode";  chatnet = "Freenode";  autojoin = "No"; },
   { name = "#irssi";     chatnet = "Freenode";  autojoin = "No"; },
+  { name = "#msys2";     chatnet = "Freenode";  autojoin = "No"; },
   { name = "#gamesurge"; chatnet = "GameSurge"; autojoin = "No"; },
   { name = "#irssi";     chatnet = "IRCNet";    autojoin = "No"; },
   { name = "#ircsource"; chatnet = "IRCSource"; autojoin = "No"; },
