diff -Naur a/src/perl/common/Makefile.PL.in b/src/perl/common/Makefile.PL.in
--- a/src/perl/common/Makefile.PL.in    2014-10-11 11:42:36.000000000 +0300
+++ b/src/perl/common/Makefile.PL.in    2015-02-26 04:00:00.000000000 +0200
@@ -3,5 +3,6 @@
 WriteMakefile('NAME' => 'Irssi',
               'LIBS' => '',
              'OBJECT' => '$(O_FILES)',
+             'dynamic_lib' => { 'OTHERLDFLAGS' => '--no-undefined' },
               'INC' => '-I../../.. -I@top_srcdir@ -I@top_srcdir@/src -I@top_srcdir@/src/core @GLIB_CFLAGS@',
              'VERSION_FROM' => '@srcdir@/Irssi.pm');
