--- origsrc/autoconf-2.13/CYGWIN-PATCHES/autoconf2.1.README	1969-12-31 19:00:00.000000000 -0500
+++ src/autoconf-2.13/CYGWIN-PATCHES/autoconf2.1.README	2013-07-08 21:15:03.011317700 -0400
@@ -0,0 +1,78 @@
+autoconf2.1
+========================================================================
+Stable version of the autoconf automatic configure script builder.
+It is packaged to work alongside autoconf2.5 and autoconf (wrappers).
+
+Build requirements
+  cygport-0.12.2
+  cygwin-1.7.20-1
+  perl
+  coreutils
+  m4
+  make
+  texinfo
+  gawk
+  sed
+  dejagnu (make check only)
+  expect  (make check only)
+
+Canonical homepage:
+  http://www.gnu.org/software/autoconf/
+
+Canonical download:
+  ftp://ftp.gnu.org/pub/gnu/autoconf/
+
+License:
+  GPL
+
+Language:
+  perl, m4
+
+========================================================================
+Test results:
+
+32 bit - 
+  # of expected passes            258
+  # of unexpected failures        1
+  FAIL: AC_F77_LIBRARY_LDFLAGS, problem with executing
+  (because modern gcc provides only gfortran, not f77)
+
+64 bit - 
+  # of expected passes            258
+  # of unexpected failures        1
+  FAIL: AC_F77_LIBRARY_LDFLAGS, problem with executing
+  (because modern gcc provides only gfortran, not f77)
+ 
+========================================================================
+
+Port Notes:
+----- autoconf2.1-2.13-12 -- 2013 Jul 08 -----
+* Rebuild using latest cygport
+* Rely on cygport auto-generated setup.hint
+* First cygwin64 build
+
+----- autoconf2.1-2.13-11 --  unreleased -----
+* 64bit only release; release-number skipped
+
+----- autoconf2.1-2.13-10 -- 18 Oct 2008 -----
+* routine rebuild; use stock cygport
+* This version marks the beginning of cygwin-1.7 specific
+  ports. Releases -4 thru -9 are reserved for cygwin-1.5.
+
+----- autoconf2.1-2.13-3 -- 15 Nov 2006 -----
+* harmonize .info files
+
+----- autoconf2.1-2.13-2 -- 14 Oct 2006 -----
+* switch to cygport build
+
+
+2.13-1
+---------
+first release in unified distribution -- older releases on cygwin
+installed autoconf-2.13 into /usr/autotool/stable and relied on 
+kludgy wraper scripts.  This release is the first to install both
+versions of autoconf into /usr, and use Red Hat-derived non-kludgy
+wrapper scripts.
+
+----------------------------------------
+Cygwin port maintained by: Chuck Wilson <cygwin@cygwin.com>
