From fd9c1a9435f8c869b61b62abc28d9be641b54e25 Mon Sep 17 00:00:00 2001
From: David Rothenberger <d.rothenberger@entomo.com>
Date: Thu, 7 Nov 2013 23:02:43 +0400
Subject: [PATCH 01/N] Create directories in build tree.

---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 31f5368..48697bb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1759,6 +1759,11 @@ AC_DEFINE_UNQUOTED([SVN_BUILD_TARGET], "${target}",
                    [Defined to the config.guess name of the build target])
 
 AC_OUTPUT
+if test "$abs_srcdir" != "$abs_builddir"
+then
+  AC_MSG_NOTICE([Creating build directories])
+  (cd "$abs_srcdir"; find subversion tools contrib -type d) | xargs $MKDIR
+fi
 
 # ==== Print final messages to user ==========================================
 
