From cf658430a6cd0afe489bbf53b004232b721958eb Mon Sep 17 00:00:00 2001
From: David Rothenberger <d.rothenberger@entomo.com>
Date: Thu, 7 Nov 2013 23:02:43 +0400
Subject: [PATCH 06/N] Use binary mode for svnadmin dump/load.

---
 subversion/svnadmin/svnadmin.c | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/subversion/svnadmin/svnadmin.c b/subversion/svnadmin/svnadmin.c
index cc14ee9..7ba128b 100644
--- a/subversion/svnadmin/svnadmin.c
+++ b/subversion/svnadmin/svnadmin.c
@@ -54,6 +54,12 @@
 
 #include "svn_private_config.h"
 
+#include <unistd.h>
+#ifdef __WIN32__
+#include <io.h>
+#endif
+#include <fcntl.h>
+
 
 /*** Code. ***/
 
@@ -1454,7 +1460,12 @@ subcommand_dump(apr_getopt_t *os, void *baton, apr_pool_t *pool)
       out_stream = svn_stream_from_aprfile2(file, FALSE, pool);
     }
   else
-    SVN_ERR(svn_stream_for_stdout(&out_stream, pool));
+    {
+#ifdef __WIN32__
+      setmode(STDOUT_FILENO, O_BINARY);
+#endif
+      SVN_ERR(svn_stream_for_stdout(&out_stream, pool));
+    }
 
   /* Progress feedback goes to STDERR, unless they asked to suppress it. */
   if (! opt_state->quiet)
@@ -1730,7 +1741,12 @@ subcommand_load(apr_getopt_t *os, void *baton, apr_pool_t *pool)
     SVN_ERR(svn_stream_open_readonly(&in_stream, opt_state->file,
                                      pool, pool));
   else
-    SVN_ERR(svn_stream_for_stdin2(&in_stream, TRUE, pool));
+    {
+#ifdef __WIN32__
+      setmode(STDIN_FILENO, O_BINARY);
+#endif
+      SVN_ERR(svn_stream_for_stdin2(&in_stream, TRUE, pool));
+    }
 
   /* Progress feedback goes to STDOUT, unless they asked to suppress it. */
   if (! opt_state->quiet)
