From bd976d3a45b2e066c88911f9014955e0417e62bd Mon Sep 17 00:00:00 2001
From: David Rothenberger <daveroth@acm.org>
Date: Mon, 14 Sep 2015 14:39:28 +0300
Subject: [PATCH 16/N] Fix serf config

---
 build/ac-macros/serf.m4 | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/build/ac-macros/serf.m4 b/build/ac-macros/serf.m4
index 0a549b3..61d9990 100644
--- a/build/ac-macros/serf.m4
+++ b/build/ac-macros/serf.m4
@@ -170,8 +170,7 @@ AC_DEFUN(SVN_SERF_PKG_CONFIG,
           serf_found=yes
           SVN_SERF_INCLUDES=[`$PKG_CONFIG $serf_pc_arg --cflags-only-I`]
           SVN_SERF_LIBS=[`$PKG_CONFIG $serf_pc_arg --libs-only-l`]
-          dnl don't use --libs-only-L because then we might miss some options
-          LDFLAGS=["$LDFLAGS `$PKG_CONFIG $serf_pc_arg --libs | $SED -e 's/-l[^ ]*//g'`"]
+          LDFLAGS=["$LDFLAGS `$PKG_CONFIG $serf_pc_arg --libs-only-L`"]
           break
         else
           AC_MSG_RESULT([no])
