From c7d158f7c6ce46d12bf66a5db452c7570a5fba70 Mon Sep 17 00:00:00 2001
From: David Rothenberger <daveroth@acm.org>
Date: Mon, 14 Sep 2015 14:39:28 +0300
Subject: [PATCH 17/N] Remove contrib

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 48697bb..d40a583 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1762,7 +1762,7 @@ AC_OUTPUT
 if test "$abs_srcdir" != "$abs_builddir"
 then
   AC_MSG_NOTICE([Creating build directories])
-  (cd "$abs_srcdir"; find subversion tools contrib -type d) | xargs $MKDIR
+  (cd "$abs_srcdir"; find subversion tools -type d) | xargs $MKDIR
 fi
 
 # ==== Print final messages to user ==========================================
