From 54acd3abbc7ea4cc53fbead63aa584baf9beaff1 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Mon, 14 Sep 2015 14:39:28 +0300
Subject: [PATCH 18/N] subversion-1.9.1-msys2

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 7369075..a6dfd09 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -941,7 +941,7 @@ copy-swig-py: autogen-swig-py swig-py $(SWIG_PY_DIR)/libsvn
 	@for f in $(SWIG_PY_SRC_DIR)/*.py $(SWIG_PY_DIR)/*.py; do \
 	  ! [ -f "$$f" ] || cp -pf $$f $(SWIG_PY_DIR)/libsvn; \
 	done
-	@cd $(SWIG_PY_DIR)/libsvn;ln -sf ../.libs/*.so .
+	@cd $(SWIG_PY_DIR)/libsvn;ln -sf ../.libs/*.dll .
 	@touch $(SWIG_PY_DIR)/libsvn/__init__.py
 
 check-swig-py: swig-py copy-swig-py
