From b862ae27a6d4a186085bac29b15728e40f14dbac Mon Sep 17 00:00:00 2001
From: Alexpux <alexey.pawlow@gmail.com>
Date: Tue, 17 Dec 2013 14:05:47 +0400
Subject: [PATCH 19/N] remove-checking-symlink

---
 configure.ac | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index d40a583..bd3b0f5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -925,9 +925,6 @@ AC_FUNC_MEMCMP
 dnl svn_error's default warning handler uses vfprintf()
 AC_FUNC_VPRINTF
 
-dnl check for functions needed in special file handling
-AC_CHECK_FUNCS(symlink readlink)
-
 dnl check for uname and ELF headers
 AC_CHECK_HEADERS(sys/utsname.h, [AC_CHECK_FUNCS(uname)], [])
 AC_CHECK_HEADERS(elf.h)
