Remove contrib

From: David Rothenberger <daveroth@acm.org>


---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index bc91a22..44c8a26 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1610,7 +1610,7 @@ AC_OUTPUT
 if test "$abs_srcdir" != "$abs_builddir"
 then
   AC_MSG_NOTICE([Creating build directories])
-  (cd "$abs_srcdir"; find subversion tools contrib -type d) | xargs $MKDIR
+  (cd "$abs_srcdir"; find subversion tools -type d) | xargs $MKDIR
 fi
 
 # ==== Print final messages to user ==========================================
