diff -Naur subversion-1.12.0-orig/build.conf subversion-1.12.0/build.conf
--- subversion-1.12.0-orig/build.conf	2019-06-11 10:48:44.113467100 +0300
+++ subversion-1.12.0/build.conf	2019-06-11 11:01:24.586603000 +0300
@@ -101,7 +101,7 @@
 
 bdb-test-scripts =
 
-swig-python-opts = $(SWIG_FEATURES) -python $(SWIG_PY_FEATURES) -classic
+swig-python-opts = $(SWIG_FEATURES) -python $(SWIG_PY_FEATURES) -nofastunpack
 swig-perl-opts = $(SWIG_FEATURES) -perl $(SWIG_PL_FEATURES) -nopm -noproxy
 swig-ruby-opts = $(SWIG_FEATURES) -ruby $(SWIG_RB_FEATURES)
 swig-languages = python perl ruby
