--- subversion-1.8.10/autogen.sh.orig	2014-10-30 06:41:03.288200000 +0300
+++ subversion-1.8.10/autogen.sh	2014-10-30 06:41:16.751000000 +0300
@@ -105,7 +105,7 @@
 
 if [ $lt_major_version -ge 2 ]; then
     for file in config.guess config.sub; do
-        configfile=${LIBTOOL_CONFIG-`cd $ltpath/../share/libtool/config ; pwd`/$file}
+        configfile=${LIBTOOL_CONFIG-`cd $ltpath/../share/libtool/build-aux ; pwd`/$file}
 
         if [ ! -f $configfile ]; then
             echo "$configfile not found (try setting the LIBTOOL_CONFIG environment variable)"
