--- subversion-1.8.5/configure.ac.orig	2013-10-24 08:02:16.000000000 +0400
+++ subversion-1.8.5/configure.ac	2013-12-17 13:27:49.179600000 +0400
@@ -872,9 +872,6 @@
 dnl svn_error's default warning handler uses vfprintf()
 AC_FUNC_VPRINTF
 
-dnl check for functions needed in special file handling
-AC_CHECK_FUNCS(symlink readlink)
-
 dnl check for uname and ELF headers
 AC_CHECK_HEADERS(sys/utsname.h, [AC_CHECK_FUNCS(uname)], [])
 AC_CHECK_HEADERS(elf.h)
