diff -Naur subversion-1.8.0-orig/Makefile.in subversion-1.8.0/Makefile.in
--- subversion-1.8.0-orig/Makefile.in	2009-11-19 22:25:52.000000000 +0300
+++ subversion-1.8.0/Makefile.in	2013-06-24 09:18:54.944336000 +0400
@@ -936,7 +936,7 @@
 	@for f in $(SWIG_PY_SRC_DIR)/*.py $(SWIG_PY_DIR)/*.py; do \
 	  ! [ -f "$$f" ] || cp -pf $$f $(SWIG_PY_DIR)/libsvn; \
 	done
-	@cd $(SWIG_PY_DIR)/libsvn;ln -sf ../.libs/*.so .
+	@cd $(SWIG_PY_DIR)/libsvn;ln -sf ../.libs/*.dll .
 	@touch $(SWIG_PY_DIR)/libsvn/__init__.py
 
 swig-py: autogen-swig-py copy-swig-py
