diff -rupN protobuf-3.6.1-orig/configure.ac protobuf-3.6.1/configure.ac
--- protobuf-3.6.1-orig/configure.ac	2018-08-02 21:26:04.124338200 +1000
+++ protobuf-3.6.1/configure.ac	2018-08-02 21:26:12.444697000 +1000
@@ -190,7 +190,7 @@ ACX_PTHREAD
 AC_CXX_STL_HASH
 
 case "$target_os" in
-  mingw* | cygwin* | win* | aix*)
+  mingw* | cygwin* | win* | msys* | aix*)
     ;;
   *)
     # Need to link against rt on Solaris
