diff -rupN protobuf-2.6.1-orig/configure.ac protobuf-2.6.1/configure.ac
--- protobuf-2.6.1-orig/configure.ac	2015-07-09 21:26:04.124338200 +1000
+++ protobuf-2.6.1/configure.ac	2015-07-09 21:26:12.444697000 +1000
@@ -140,7 +140,7 @@ ACX_PTHREAD
 AC_CXX_STL_HASH
 
 case "$target_os" in
-  mingw* | cygwin* | win*)
+  mingw* | cygwin* | win* | msys*)
     ;;
   *)
     # Need to link against rt on Solaris
