--- autotools-wrappers/ac-wrapper.sh.orig	2023-12-23 09:24:41.791445600 +0100
+++ autotools-wrappers/ac-wrapper.sh	2023-12-23 09:25:17.970262800 +0100
@@ -79,7 +79,7 @@
 # Set up bindings between actual version and WANT_AUTOCONF;
 # Start with last known versions to speed up lookup process.
 #
-LAST_KNOWN_VER="71"
+LAST_KNOWN_VER="72"
 vers="9999"
 vers="${vers} $(printf '2.%s ' `seq ${LAST_KNOWN_VER} -1 50`)"
 
