diff -Naur automake-1.16-orig/t/compile5.sh automake-1.16/t/compile5.sh
--- automake-1.16-orig/t/compile5.sh	2014-12-30 16:49:41.000000000 +0300
+++ automake-1.16/t/compile5.sh	2015-01-07 10:48:12.160000000 +0300
@@ -47,11 +47,11 @@
     ;;
 esac
 case @build_os@ in
-  mingw* | cygwin*)
+  mingw* | cygwin* | msys*)
     ;;
   *)
     winepath -w / \
-      || skip_ "not on MinGW or Cygwin, and winepath not available"
+      || skip_ "not on MinGW or Cygwin or MSYS, and winepath not available"
     ;;
 esac
 END
