diff --git a/tests/scripts/options/symlinks b/tests/scripts/options/symlinks
index bf8a892..7532f7f 100644
--- a/tests/scripts/options/symlinks
+++ b/tests/scripts/options/symlinks
@@ -10,7 +10,7 @@ $details = "Verify that symlink handling with and without -L works properly.";
 # (in that the symlink() function doesn't fail) but it really doesn't, so
 # check for it explicitly.
 
-if ($port_type eq 'W32' || !( eval { symlink("",""); 1 })) {
+if ($port_type eq 'W32' || $osname =~ /msys/i || !( eval { symlink("",""); 1 })) {
   # This test is N/A
   return -1;
 }
