diff -durN make-git.orig/maintMakefile make-git/maintMakefile
--- make-git.orig/maintMakefile	2017-05-17 08:51:04.859375000 +0800
+++ make-git/maintMakefile	2017-05-17 09:06:44.453125000 +0800
@@ -20,7 +20,7 @@
 MAKEWEBDIR ?= $(SRCROOTDIR)/make/make-web
 
 # We like mondo-warnings!
-AM_CFLAGS += -Wall -Wextra -Werror -Wwrite-strings -Wshadow -Wpointer-arith \
+AM_CFLAGS += -Wall -Wextra -Wwrite-strings -Wshadow -Wpointer-arith \
 	-Wdeclaration-after-statement -Wbad-function-cast -Wformat-security \
 	-Wtype-limits -Wunused-but-set-parameter -Wlogical-op \
 	-Wignored-qualifiers -Wformat-signedness -Wduplicated-cond
