From 0dde44f5f1a398bab333f66d000f5a67e20c6bb5 Mon Sep 17 00:00:00 2001
From: pedro martelletto <pedro@yubico.com>
Date: Tue, 15 Feb 2022 10:25:37 +0100
Subject: [PATCH 2/2] winhello: sort according to CTAP2 canonical CBOR encoding
 form

---
 src/winhello.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/winhello.c b/src/winhello.c
index 2fc6a2a..63dd82c 100644
--- a/src/winhello.c
+++ b/src/winhello.c
@@ -883,7 +883,7 @@ fido_winhello_get_cbor_info(fido_dev_t *dev, fido_cbor_info_t *ci)
 	const char *v[3] = { "U2F_V2", "FIDO_2_0", "FIDO_2_1_PRE" };
 	const char *e[2] = { "credProtect", "hmac-secret" };
 	const char *t[2] = { "nfc", "usb" };
-	const char *o[4] = { "rk", "up", "plat", "uv" };
+	const char *o[4] = { "rk", "up", "uv", "plat" };
 
 	(void)dev;
 
-- 
2.35.1.windows.2

