--- libxml2-2.11.4/configure.ac.orig	2023-06-21 11:18:44.777229500 +0200
+++ libxml2-2.11.4/configure.ac	2023-06-21 11:18:54.986273900 +0200
@@ -767,9 +767,12 @@
 PYTHON_LDFLAGS=
 if test "${PYTHON}" != ""; then
     case "$host" in
-        *-*-cygwin* | *-*-mingw* | *-*-msys* )
+        *-*-mingw* )
             PYTHON_LDFLAGS="-no-undefined -shrext .pyd"
             ;;
+        *-*-cygwin* |*-*-msys* )
+            PYTHON_LDFLAGS="-no-undefined -shrext .dll"
+            ;;
     esac
 fi
 AC_SUBST(PYTHON_LDFLAGS)
