--- libxml2-2.7.3/python/Makefile.am	2008-02-21 15:53:06.000000000 -0600
+++ libxml2-2.7.3/python/Makefile.am	2009-03-16 21:08:12.859375000 -0500
@@ -9,1 +9,1 @@ INCLUDES = \
-docsdir = $(datadir)/doc/libxml2-python-$(LIBXML_VERSION)
+docsdir = $(datadir)/doc/python-libxml2
--- libxml2-2.7.3/python/tests/Makefile.am	2008-09-25 11:21:03.000000000 -0500
+++ libxml2-2.7.3/python/tests/Makefile.am	2009-03-16 21:07:47.468750000 -0500
@@ -1,4 +1,4 @@
-exampledir = $(datadir)/doc/libxml2-python-$(LIBXML_VERSION)/examples
+exampledir = $(datadir)/doc/python-libxml2/examples
 dist_example_DATA = $(PYTESTS) $(XMLS)
 
 PYTESTS=	\
