From 10ec19ab0a49936ca8ca0d6962d24d3f889960a5 Mon Sep 17 00:00:00 2001
From: dcb <dcb314@hotmail.com>
Date: Thu, 2 May 2013 16:11:46 +0800
Subject: [PATCH] Fix missing break on last() function for attributes

pointed out by cppcheck
---
 python/libxml.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/python/libxml.c b/python/libxml.c
index 03cfb9f..3338b83 100644
--- a/python/libxml.c
+++ b/python/libxml.c
@@ -2683,6 +2683,7 @@ libxml_last(ATTRIBUTE_UNUSED PyObject * self, PyObject * args)
                 xmlAttrPtr attr = (xmlAttrPtr) cur;
 
                 res = attr->last;
+		break;
             }
         default:
             res = NULL;
-- 
1.8.3.2

