diff -Naur libtasn1-3.3-orig/build-aux/ar-lib libtasn1-3.3/build-aux/ar-lib
--- libtasn1-3.3-orig/build-aux/ar-lib	2013-03-24 10:26:53.000000000 +0000
+++ libtasn1-3.3/build-aux/ar-lib	2013-04-25 09:50:06.293945300 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -65,7 +65,7 @@
 	mingw)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin)
+	cygwin | msys)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine)
diff -Naur libtasn1-3.3-orig/build-aux/config.guess libtasn1-3.3/build-aux/config.guess
--- libtasn1-3.3-orig/build-aux/config.guess	2013-03-24 10:26:53.000000000 +0000
+++ libtasn1-3.3/build-aux/config.guess	2013-04-25 09:50:38.147460900 +0000
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur libtasn1-3.3-orig/gl/m4/malloc.m4 libtasn1-3.3/gl/m4/malloc.m4
--- libtasn1-3.3-orig/gl/m4/malloc.m4	2013-03-24 09:16:44.000000000 +0000
+++ libtasn1-3.3/gl/m4/malloc.m4	2013-04-25 09:54:50.157226600 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur libtasn1-3.3-orig/gl/m4/realloc.m4 libtasn1-3.3/gl/m4/realloc.m4
--- libtasn1-3.3-orig/gl/m4/realloc.m4	2013-03-24 09:16:44.000000000 +0000
+++ libtasn1-3.3/gl/m4/realloc.m4	2013-04-25 09:55:08.762695300 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
