diff -Naur libtasn1-3.3-orig/build-aux/config.guess libtasn1-3.3/build-aux/config.guess
--- libtasn1-3.3-orig/build-aux/config.guess	2013-03-24 10:26:53.000000000 +0000
+++ libtasn1-3.3/build-aux/config.guess	2013-04-25 09:50:38.147460900 +0000
@@ -947,6 +947,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	GUESS=x86_64-pc-cygwin
 	;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	GUESS=x86_64-unknown-msys
+	;;
     prep*:SunOS:5.*:*)
 	SUN_REL=`echo "$UNAME_RELEASE" | sed -e 's/[^.]*//'`
 	GUESS=powerpcle-unknown-solaris2$SUN_REL
