diff -Naur libtasn1-3.3-orig/build-aux/ar-lib libtasn1-3.3/build-aux/ar-lib
--- libtasn1-3.3-orig/build-aux/ar-lib	2013-03-24 10:26:53.000000000 +0000
+++ libtasn1-3.3/build-aux/ar-lib	2013-04-25 09:50:06.293945300 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -65,7 +65,7 @@
 	mingw)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin)
+	cygwin | msys)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine)
diff -Naur libtasn1-3.3-orig/build-aux/config.guess libtasn1-3.3/build-aux/config.guess
--- libtasn1-3.3-orig/build-aux/config.guess	2013-03-24 10:26:53.000000000 +0000
+++ libtasn1-3.3/build-aux/config.guess	2013-04-25 09:50:38.147460900 +0000
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
