diff --git aspell-0.60.6.1.orig/prog/compress.c aspell-0.60.6.1/prog/compress.c
index c47ddf2..f4271b7 100644
--- aspell-0.60.6.1.orig/prog/compress.c
+++ aspell-0.60.6.1/prog/compress.c
@@ -25,7 +25,7 @@
 #  include <io.h>
 #  include <fcntl.h>
 
-#  define SETBIN(fno)  _setmode( _fileno( fno ), _O_BINARY )
+#  define SETBIN(fno)  _setmode( _fileno( fno ), O_BINARY )
 
 #else
 
diff --git aspell-0.60.6.1.orig/prog/prezip.c aspell-0.60.6.1/prog/prezip.c
index 3835114..4d2677c 100644
--- aspell-0.60.6.1.orig/prog/prezip.c
+++ aspell-0.60.6.1/prog/prezip.c
@@ -39,7 +39,7 @@
 #  include <io.h>
 #  include <fcntl.h>
 
-#  define SETBIN(fno)  _setmode( _fileno( fno ), _O_BINARY )
+#  define SETBIN(fno)  _setmode( _fileno( fno ), O_BINARY )
 
 #else
 
