--- aspell-0.60.6.1/prog/compress.c	2004-11-07 13:29:48.000000000 +0100
+++ aspell-0.60.6.1/prog/compress.c	2013-04-30 09:03:17.471947700 +0200
@@ -25,7 +25,11 @@
 #  include <io.h>
 #  include <fcntl.h>
 
+#if defined(__CYGWIN__)
+#  define SETBIN(fno)  _setmode( fileno( fno ), O_BINARY )
+#else
 #  define SETBIN(fno)  _setmode( _fileno( fno ), _O_BINARY )
+#endif
 
 #else
 
--- aspell-0.60.6.1/prog/prezip.c	2011-07-02 23:09:09.000000000 +0200
+++ aspell-0.60.6.1/prog/prezip.c	2013-04-30 09:02:30.307958600 +0200
@@ -39,7 +39,11 @@
 #  include <io.h>
 #  include <fcntl.h>
 
+#if defined(__CYGWIN__)
+#  define SETBIN(fno)  _setmode( fileno( fno ), O_BINARY )
+#else
 #  define SETBIN(fno)  _setmode( _fileno( fno ), _O_BINARY )
+#endif
 
 #else
 
