--- aspell-0.60.6.1/lib/new_filter.cpp	2011-07-02 23:09:09.000000000 +0200
+++ aspell-0.60.6.1/lib/new_filter.cpp	2011-11-11 01:35:59.046875000 +0100
@@ -458,11 +458,19 @@ namespace acommon
       module->file.assign(option_file.str(), slash + 1 - option_file.str());
       //module->file += "lib";
       module->file += filter_name;
+#ifdef __CYGWIN__
+      module->file += "-filter.dll";
+#else
       module->file += "-filter.so";
+#endif
     } else {
       if (module->file[0] != '/')
         module->file.insert(0, option_file.str(), slash + 1 - option_file.str());
+#ifdef __CYGWIN__
+      module->file += ".dll";
+#else
       module->file += ".so";
+#endif
     }
 
     return module.release();
