--- aspell-0.60.6.1/Makefile.am	2011-07-03 01:42:34.000000000 +0200
+++ aspell-0.60.6.1/Makefile.am	2011-11-11 01:35:59.046875000 +0100
@@ -194,7 +194,7 @@ libaspell_la_SOURCES +=\
 else # not COMPILE_IN_FILTERS
 
 dynamic_optfiles += ${optfiles}
-filter_ldflags = -module -avoid-version
+filter_ldflags = -module -avoid-version -no-undefined
 
 ### Add name of filter library containing your filter. Name always
 ### must look like lib<filtername>-filter.la see development manual
@@ -203,31 +203,31 @@ filter_LTLIBRARIES = email-filter.la tex
                      nroff-filter.la texinfo-filter.la
 
 email_filter_la_SOURCES = modules/filter/email.cpp
-email_filter_la_LIBADD = libaspell.la 
+email_filter_la_LIBADD = libaspell.la $(LTLIBINTL)
 email_filter_la_LDFLAGS  = ${filter_ldflags}
 
 tex_filter_la_SOURCES = modules/filter/tex.cpp
-tex_filter_la_LIBADD = libaspell.la
+tex_filter_la_LIBADD = libaspell.la $(LTLIBINTL)
 tex_filter_la_LDFLAGS  = ${filter_ldflags}
 
 sgml_filter_la_SOURCES  = modules/filter/sgml.cpp
-sgml_filter_la_LIBADD = libaspell.la
+sgml_filter_la_LIBADD = libaspell.la $(LTLIBINTL)
 sgml_filter_la_LDFLAGS  = ${filter_ldflags}
 
 markdown_filter_la_SOURCES  = modules/filter/markdown.cpp
-markdown_filter_la_LIBADD = libaspell.la
+markdown_filter_la_LIBADD = libaspell.la $(LTLIBINTL)
 markdown_filter_la_LDFLAGS  = ${filter_ldflags}
 
 context_filter_la_SOURCES  = modules/filter/context.cpp
-context_filter_la_LIBADD = libaspell.la
+context_filter_la_LIBADD = libaspell.la $(LTLIBINTL)
 context_filter_la_LDFLAGS  = ${filter_ldflags}
 
 nroff_filter_la_SOURCES  = modules/filter/nroff.cpp
-nroff_filter_la_LIBADD = libaspell.la
+nroff_filter_la_LIBADD = libaspell.la $(LTLIBINTL)
 nroff_filter_la_LDFLAGS  = ${filter_ldflags}
 
 texinfo_filter_la_SOURCES  = modules/filter/texinfo.cpp
-texinfo_filter_la_LIBADD = libaspell.la
+texinfo_filter_la_LIBADD = libaspell.la $(LTLIBINTL)
 texinfo_filter_la_LDFLAGS  = ${filter_ldflags}
 
 ### Before this line add the corresponding <yourfilterlibrary>_SOURCES and
