diff --git a/configure.ac b/configure.ac
index f7a09a9..b20bf4e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -71,10 +71,6 @@ dnl Support BSD4.4 "sa_len" extension when calculating sockaddrs arrays
 AC_CHECK_MEMBERS(struct sockaddr.sa_len, , , [#include <sys/types.h>
 #include <sys/socket.h>])
 
-dnl Check if it is available the RFC2292 IPv4 extension and in_port_t.
-AC_CHECK_TYPES([struct in_pktinfo, in_port_t], , , [#include <sys/types.h>
-#include <netinet/in.h>])
-
 # check if this compiler supports the common parameter `-Wall'
 if test -n "$GCC"; then
   CFLAGS="$CFLAGS -Wall"
