--- origsrc/pinentry-1.0.0/w32/Makefile.am	2016-04-14 09:39:51.000000000 -0500
+++ src/pinentry-1.0.0/w32/Makefile.am	2017-06-29 02:06:33.906651400 -0500
@@ -35,7 +35,7 @@ pinentry_w32_SOURCES = main.c pinentry-w
 pinentry_w32_LDFLAGS = -mwindows
 pinentry_w32_LDADD = pinentry-w32.o \
 	../pinentry/libpinentry.a ../secmem/libsecmem.a \
-        $(COMMON_LIBS)
+        $(COMMON_LIBS) $(LIBICONV)
 
 pinentry-w32.o: pinentry-w32.rc resource.h $(logos)
 	$(WINDRES) -I.. -v -o $@ $<
