diff -ru pinentry-1.1.0-orig/configure.ac pinentry-1.1.0/configure.ac
--- pinentry-1.1.0-orig/configure.ac	2017-12-03 17:49:54.000000000 +0100
+++ pinentry-1.1.0/configure.ac	2018-06-21 22:44:58.560006300 +0200
@@ -576,7 +576,7 @@
 # Check whether we should build the W32 pinentry.  This is actually
 # the simplest check as we do this only for that platform.
 #
-pinentry_w32=no
+pinentry_w32=yes
 test $have_w32_system = yes && pinentry_w32=yes
 AM_CONDITIONAL(BUILD_PINENTRY_W32, test "$pinentry_w32" = "yes")
 
diff -ru pinentry-1.1.0-orig/w32/main.c pinentry-1.1.0/w32/main.c
--- pinentry-1.1.0-orig/w32/main.c	2017-12-03 17:13:05.000000000 +0100
+++ pinentry-1.1.0/w32/main.c	2018-06-21 22:44:34.529631800 +0200
@@ -19,6 +19,7 @@
 #include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
 #if WINVER < 0x0403
 # define WINVER 0x0403  /* Required for SendInput.  */
 #endif
diff -ru pinentry-1.1.0-orig/w32/pinentry-w32.rc pinentry-1.1.0/w32/pinentry-w32.rc
--- pinentry-1.1.0-orig/w32/pinentry-w32.rc	2017-12-03 17:13:05.000000000 +0100
+++ pinentry-1.1.0/w32/pinentry-w32.rc	2018-06-21 22:46:08.010978600 +0200
@@ -63,8 +63,8 @@
     RTEXT           "", IDC_PINENT_PROMPT, 6,  60,  60, 12
     EDITTEXT        IDC_PINENT_TEXT,      70,  59, 110, 12, ES_PASSWORD | ES_AUTOHSCROLL
     CTEXT           "", IDC_PINENT_ERR,    6,  76, 174, 12
-    DEFPUSHBUTTON   "O&K", IDOK,          74,  96,  50, 14
-    PUSHBUTTON      "&Cancel", IDCANCEL, 130,  96,  50, 14
+    DEFPUSHBUTTON   "O&K", IDOK,          10,  96,  80, 14
+    PUSHBUTTON      "&Cancel", IDCANCEL,  96,  96,  80, 14
 END
 
 
