--- pinentry-1.3.0/configure.ac.orig	2024-03-18 12:13:19.000000000 +0100
+++ pinentry-1.3.0/configure.ac	2024-06-09 17:50:57.791957400 +0200
@@ -762,7 +762,7 @@
 # Check whether we should build the W32 pinentry.  This is actually
 # the simplest check as we do this only for that platform.
 #
-pinentry_w32=no
+pinentry_w32=yes
 test $have_w32_system = yes && pinentry_w32=yes
 AM_CONDITIONAL(BUILD_PINENTRY_W32, test "$pinentry_w32" = "yes")
 
