diff -Naur pinentry-0.9.1-orig/build-aux/config.guess pinentry-0.9.1/build-aux/config.guess
--- pinentry-0.9.1-orig/build-aux/config.guess	2015-03-18 13:38:54.000000000 +0300
+++ pinentry-0.9.1/build-aux/config.guess	2015-05-02 10:51:40.813200000 +0300
@@ -867,6 +867,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur pinentry-0.9.1-orig/build-aux/config.rpath pinentry-0.9.1/build-aux/config.rpath
--- pinentry-0.9.1-orig/build-aux/config.rpath	2015-03-18 13:39:29.000000000 +0300
+++ pinentry-0.9.1/build-aux/config.rpath	2015-05-02 10:51:14.106000000 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur pinentry-0.9.1-orig/pinentry/pinentry-curses.c pinentry-0.9.1/pinentry/pinentry-curses.c
--- pinentry-0.9.1-orig/pinentry/pinentry-curses.c	2015-03-18 14:02:08.000000000 +0300
+++ pinentry-0.9.1/pinentry/pinentry-curses.c	2015-05-02 10:52:54.760400000 +0300
@@ -22,7 +22,7 @@
 #include <config.h>
 #endif
 #include <assert.h>
-#include <curses.h>
+#include <ncurses/curses.h>
 #include <signal.h>
 #include <fcntl.h>
 #include <unistd.h>
