--- openssl-3.0.7/Configurations/platform/Cygwin.pm.orig	2022-12-12 21:04:51.674110200 +0100
+++ openssl-3.0.7/Configurations/platform/Cygwin.pm	2022-11-01 15:14:36.000000000 +0100
@@ -15,7 +15,7 @@
 sub sharedname {
     my $class = shift;
     my $lib = platform::mingw->sharedname(@_);
-    $lib =~ s|^lib|cyg| if defined $lib;
+    $lib =~ s|^lib|msys-| if defined $lib;
     return $lib;
 }
 
diff --git a/tools/c_rehash.in b/tools/c_rehash.in
index 421fd89..640db8c 100644
--- a/tools/c_rehash.in
+++ b/tools/c_rehash.in
@@ -21,6 +21,7 @@ my $x509hash = "-subject_hash";
 my $crlhash = "-hash";
 my $verbose = 0;
 my $symlink_exists=eval {symlink("",""); 1};
+$symlink_exists=0;
 my $removelinks = 1;
 
 ##  Parse flags.
