--- openssl-1.1.1s/build.info.orig	2022-11-01 13:36:10.000000000 +0100
+++ openssl-1.1.1s/build.info	2022-12-11 19:26:26.709402900 +0100
@@ -80,8 +80,8 @@
 ENDIF
 
 IF[{- $config{target} =~ /^Cygwin/ -}]
- SHARED_NAME[libcrypto]=cygcrypto-{- $sover_filename -}
- SHARED_NAME[libssl]=cygssl-{- $sover_filename -}
+ SHARED_NAME[libcrypto]=msys-crypto-{- $sover_filename -}
+ SHARED_NAME[libssl]=msys-ssl-{- $sover_filename -}
 ELSIF[{- $config{target} =~ /^mingw/ -}]
  SHARED_NAME[libcrypto]=libcrypto-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
  SHARED_NAME[libssl]=libssl-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
diff --git a/tools/c_rehash.in b/tools/c_rehash.in
index 421fd89..640db8c 100644
--- a/tools/c_rehash.in
+++ b/tools/c_rehash.in
@@ -21,6 +21,7 @@ my $x509hash = "-subject_hash";
 my $crlhash = "-hash";
 my $verbose = 0;
 my $symlink_exists=eval {symlink("",""); 1};
+$symlink_exists=0;
 my $removelinks = 1;
 
 ##  Parse flags.
diff --git a/util/process_docs.pl b/util/process_docs.pl
index 30b149e..0f97ce1 100755
--- a/util/process_docs.pl
+++ b/util/process_docs.pl
@@ -79,6 +79,7 @@ if ($options{debug}) {
 }
 
 my $symlink_exists = eval { symlink("",""); 1 };
+$symlink_exists=0;
 
 foreach my $section (sort @{$options{section}}) {
     my $subdir = "man$section";
