--- openssl-3.0.7/Configurations/unix-Makefile.tmpl.orig	2022-12-12 20:42:42.640227800 +0100
+++ openssl-3.0.7/Configurations/unix-Makefile.tmpl	2022-12-12 20:43:33.810678900 +0100
@@ -304,7 +304,7 @@
 # $(libdir) is chosen to be compatible with the GNU coding standards
 libdir={- file_name_is_absolute($libdir)
           ? $libdir : '$(INSTALLTOP)/$(LIBDIR)' -}
-ENGINESDIR=$(libdir)/engines-{- $sover_dirname -}
+ENGINESDIR=$(libdir)/openssl/engines-{- $sover_dirname -}
 MODULESDIR=$(libdir)/ossl-modules
 
 # Convenience variable for those who want to set the rpath in shared
