--- po4a-0.48/po/pod.cfg.orig	2017-02-16 12:55:57.972637800 +0300
+++ po4a-0.48/po/pod.cfg	2017-02-16 12:57:48.428543800 +0300
@@ -32,58 +32,58 @@
 [type: pod] doc/po4a-runtime.7.pod          $lang:blib/man/$lang/man7/po4a-runtime.7.pod \
             add_$lang:?doc/addendum.$lang
 
-[type: pod] lib/Locale/Po4a/AsciiDoc.pm     $lang:blib/man/$lang/man3/Locale::Po4a::AsciiDoc.3pm.pod \
+[type: pod] lib/Locale/Po4a/AsciiDoc.pm     $lang:blib/man/$lang/man3/Locale.Po4a.AsciiDoc.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/BibTeX.pm       $lang:blib/man/$lang/man3/Locale::Po4a::BibTeX.3pm.pod \
+[type: pod] lib/Locale/Po4a/BibTeX.pm       $lang:blib/man/$lang/man3/Locale.Po4a.BibTeX.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Chooser.pm      $lang:blib/man/$lang/man3/Locale::Po4a::Chooser.3pm.pod \
+[type: pod] lib/Locale/Po4a/Chooser.pm      $lang:blib/man/$lang/man3/Locale.Po4a.Chooser.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Common.pm       $lang:blib/man/$lang/man3/Locale::Po4a::Common.3pm.pod \
+[type: pod] lib/Locale/Po4a/Common.pm       $lang:blib/man/$lang/man3/Locale.Po4a.Common.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Dia.pm          $lang:blib/man/$lang/man3/Locale::Po4a::Dia.3pm.pod \
+[type: pod] lib/Locale/Po4a/Dia.pm          $lang:blib/man/$lang/man3/Locale.Po4a.Dia.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Docbook.pm      $lang:blib/man/$lang/man3/Locale::Po4a::Docbook.3pm.pod \
+[type: pod] lib/Locale/Po4a/Docbook.pm      $lang:blib/man/$lang/man3/Locale.Po4a.Docbook.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Guide.pm        $lang:blib/man/$lang/man3/Locale::Po4a::Guide.3pm.pod \
+[type: pod] lib/Locale/Po4a/Guide.pm        $lang:blib/man/$lang/man3/Locale.Po4a.Guide.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Halibut.pm      $lang:blib/man/$lang/man3/Locale::Po4a::Halibut.3pm.pod \
+[type: pod] lib/Locale/Po4a/Halibut.pm      $lang:blib/man/$lang/man3/Locale.Po4a.Halibut.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Ini.pm          $lang:blib/man/$lang/man3/Locale::Po4a::Ini.3pm.pod \
+[type: pod] lib/Locale/Po4a/Ini.pm          $lang:blib/man/$lang/man3/Locale.Po4a.Ini.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/KernelHelp.pm   $lang:blib/man/$lang/man3/Locale::Po4a::KernelHelp.3pm.pod \
+[type: pod] lib/Locale/Po4a/KernelHelp.pm   $lang:blib/man/$lang/man3/Locale.Po4a.KernelHelp.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/LaTeX.pm        $lang:blib/man/$lang/man3/Locale::Po4a::LaTeX.3pm.pod \
+[type: pod] lib/Locale/Po4a/LaTeX.pm        $lang:blib/man/$lang/man3/Locale.Po4a.LaTeX.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Man.pm          $lang:blib/man/$lang/man3/Locale::Po4a::Man.3pm.pod \
+[type: pod] lib/Locale/Po4a/Man.pm          $lang:blib/man/$lang/man3/Locale.Po4a.Man.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Po.pm           $lang:blib/man/$lang/man3/Locale::Po4a::Po.3pm.pod \
+[type: pod] lib/Locale/Po4a/Po.pm           $lang:blib/man/$lang/man3/Locale.Po4a.Po.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Pod.pm          $lang:blib/man/$lang/man3/Locale::Po4a::Pod.3pm.pod \
+[type: pod] lib/Locale/Po4a/Pod.pm          $lang:blib/man/$lang/man3/Locale.Po4a.Pod.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Sgml.pm         $lang:blib/man/$lang/man3/Locale::Po4a::Sgml.3pm.pod \
+[type: pod] lib/Locale/Po4a/Sgml.pm         $lang:blib/man/$lang/man3/Locale.Po4a.Sgml.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/TeX.pm          $lang:blib/man/$lang/man3/Locale::Po4a::TeX.3pm.pod \
+[type: pod] lib/Locale/Po4a/TeX.pm          $lang:blib/man/$lang/man3/Locale.Po4a.TeX.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Texinfo.pm      $lang:blib/man/$lang/man3/Locale::Po4a::Texinfo.3pm.pod \
+[type: pod] lib/Locale/Po4a/Texinfo.pm      $lang:blib/man/$lang/man3/Locale.Po4a.Texinfo.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Text.pm         $lang:blib/man/$lang/man3/Locale::Po4a::Text.3pm.pod \
+[type: pod] lib/Locale/Po4a/Text.pm         $lang:blib/man/$lang/man3/Locale.Po4a.Text.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/TransTractor.pm $lang:blib/man/$lang/man3/Locale::Po4a::TransTractor.3pm.pod \
+[type: pod] lib/Locale/Po4a/TransTractor.pm $lang:blib/man/$lang/man3/Locale.Po4a.TransTractor.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Wml.pm          $lang:blib/man/$lang/man3/Locale::Po4a::Wml.3pm.pod \
+[type: pod] lib/Locale/Po4a/Wml.pm          $lang:blib/man/$lang/man3/Locale.Po4a.Wml.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Xhtml.pm        $lang:blib/man/$lang/man3/Locale::Po4a::Xhtml.3pm.pod \
+[type: pod] lib/Locale/Po4a/Xhtml.pm        $lang:blib/man/$lang/man3/Locale.Po4a.Xhtml.3pm.pod \
             add_$lang:?doc/addendum.$lang
-[type: pod] lib/Locale/Po4a/Xml.pm          $lang:blib/man/$lang/man3/Locale::Po4a::Xml.3pm.pod \
+[type: pod] lib/Locale/Po4a/Xml.pm          $lang:blib/man/$lang/man3/Locale.Po4a.Xml.3pm.pod \
             add_$lang:?doc/addendum.$lang
 
 # Deactivated modules
 #  not tested
-#[type: pod] lib/Locale/Po4a/Debconf.pm      $lang:blib/man/$lang/man3/Locale::Po4a::Debconf.3pm.pod \
+#[type: pod] lib/Locale/Po4a/Debconf.pm      $lang:blib/man/$lang/man3/Locale.Po4a.Debconf.3pm.pod \
 #            add_$lang:?doc/addendum.$lang
 #  deprecated, use Xhtml instead
-#[type: pod] lib/Locale/Po4a/Html.pm         $lang:blib/man/$lang/man3/Locale::Po4a::Html.3pm.pod \
+#[type: pod] lib/Locale/Po4a/Html.pm         $lang:blib/man/$lang/man3/Locale.Po4a.Html.3pm.pod \
 #            add_$lang:?doc/addendum.$lang
 #  deprecated, use Text instead
-#[type: pod] lib/Locale/Po4a/NewsDebian.pm   $lang:blib/man/$lang/man3/Locale::Po4a::NewsDebian.3pm.pod \
+#[type: pod] lib/Locale/Po4a/NewsDebian.pm   $lang:blib/man/$lang/man3/Locale.Po4a.NewsDebian.3pm.pod \
 #            add_$lang:?doc/addendum.$lang
--- po4a-0.48/Po4aBuilder.pm.orig	2017-02-16 13:06:40.166758600 +0300
+++ po4a-0.48/Po4aBuilder.pm	2017-02-16 13:08:11.218209000 +0300
@@ -218,7 +218,7 @@
     }
     foreach $file (@{$self->rscan_dir('lib',qr{\.pm$})}) {
         $file =~ m,([^/]*).pm$,;
-        copy($file, File::Spec->catdir($man3path, "Locale::Po4a::$1.3pm.pod")) or die;
+        copy($file, File::Spec->catdir($man3path, "Locale.Po4a.$1.3pm.pod")) or die;
     }
     $self->delete_filetree( File::Spec->catdir("blib", "bindoc") );
     $self->delete_filetree( File::Spec->catdir("blib", "libdoc") );
