diff -u a/bootstrap b/bootstrap
--- a/bootstrap	2022-07-24 00:19:04.972571900 -0700
+++ b/bootstrap	2022-07-24 00:20:09.875444300 -0700
@@ -369,8 +369,8 @@
           _G_path_prog=$_G_dir/$_G_prog_name$_exeext
           func_executable_p "$_G_path_prog" || continue
           case `"$_G_path_prog" --version 2>&1` in
-            *GNU*) func_path_progs_result=$_G_path_prog _G_path_prog_found=: ;;
-            *)     $_G_check_func $_G_path_prog
+            *GNU*) func_path_progs_result="$_G_path_prog" _G_path_prog_found=: ;;
+            *)     $_G_check_func "$_G_path_prog"
 		   func_path_progs_result=$func_check_prog_result
 		   ;;
           esac
@@ -418,7 +418,7 @@
       _G_count=`expr $_G_count + 1`
       if test "$_G_count" -gt "$_G_path_prog_max"; then
         # Best one so far, save it but keep looking for a better one
-        func_check_prog_result=$_G_path_prog
+        func_check_prog_result="$_G_path_prog"
         _G_path_prog_max=$_G_count
       fi
       # 10*(2^10) chars as input seems more than enough
@@ -454,7 +454,7 @@
       _G_count=`expr $_G_count + 1`
       if test "$_G_count" -gt "$_G_path_prog_max"; then
         # Best one so far, save it but keep looking for a better one
-        func_check_prog_result=$_G_path_prog
+        func_check_prog_result="$_G_path_prog"
         _G_path_prog_max=$_G_count
       fi
       # 10*(2^10) chars as input seems more than enough
@@ -2503,15 +2503,15 @@
 	  IFS=$_G_find_tool_save_IFS
 	  _G_progpath=$_G_dir/$_G_prog
           test -r "$_G_progpath" && {
-            _G_curver=`func_tool_version_number $_G_progpath`
+            _G_curver=`func_tool_version_number "$_G_progpath"`
 	    case $_G_bestver,$_G_curver in
 	    ,)
 	      # first non--version responsive prog sticks!
-              test -n "$_G_progpath" || _G_find_tool_res=$_G_progpath
+              test -n "$_G_progpath" || _G_find_tool_res="$_G_progpath"
               ;;
             ,*)
 	      # first --version responsive prog beats non--version responsive!
-	      _G_find_tool_res=$_G_progpath
+	      _G_find_tool_res="$_G_progpath"
 	      _G_bestver=$_G_curver
 	      ;;
 	    *,*)
@@ -2519,7 +2519,7 @@
 	      test "x$_G_curver" = "x$_G_bestver" \
 		|| func_lt_ver "$_G_curver" "$_G_bestver" \
 		|| {
-		     _G_find_tool_res=$_G_progpath
+		     _G_find_tool_res="$_G_progpath"
 		     _G_bestver=$_G_curver
 	           }
 	      ;;
