diff --git a/libltdl/config/ltmain.m4sh b/libltdl/config/ltmain.m4sh
index 0418007..91276c2 100644
--- a/libltdl/config/ltmain.m4sh
+++ b/libltdl/config/ltmain.m4sh
@@ -4163,6 +4163,7 @@
 # include <unistd.h>
 # include <stdint.h>
 # ifdef __CYGWIN__
+#  include <process.h>
 #  include <io.h>
 # endif
 #endif
diff --git a/libltdl/config/ltmain.sh b/libltdl/config/ltmain.sh
index 0418007..91276c2 100644
--- a/libltdl/config/ltmain.sh
+++ b/libltdl/config/ltmain.sh
@@ -4163,6 +4163,7 @@
 # include <unistd.h>
 # include <stdint.h>
 # ifdef __CYGWIN__
+#  include <process.h>
 #  include <io.h>
 # endif
 #endif
