From a8c42da283ff288b5a425183bd26af5df665f90e Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Mon, 20 Apr 2026 07:02:29 +0800
Subject: [PATCH] Support building under cygwin

---
 Make.defaults       | 2 +-
 lib/x86_64/setjmp.S | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Make.defaults b/Make.defaults
index cf7a88e..6daaa85 100755
--- a/Make.defaults
+++ b/Make.defaults
@@ -265,7 +265,7 @@ INCDIR  += -I$(SRCDIR) -I$(TOPDIR)/inc -I$(TOPDIR)/inc/$(ARCH) \
 
 # Only enable -fPIC for non MinGW compilers (unneeded on MinGW)
 GCCMACHINE := $(shell $(CC) -dumpmachine)
-IS_MINGW32 := $(findstring mingw32, $(GCCMACHINE))
+IS_MINGW32 := 1
 ifeq ($(IS_MINGW32),)
   EFI_CFLAGS += -fPIC
 endif
diff --git a/lib/x86_64/setjmp.S b/lib/x86_64/setjmp.S
index 719b021..1c381cf 100644
--- a/lib/x86_64/setjmp.S
+++ b/lib/x86_64/setjmp.S
@@ -1,6 +1,6 @@
 	.text
 	.globl	base_setjmp
-#ifndef __MINGW32__
+#if !defined(__MINGW32__) && !defined(__CYGWIN__)
 	.type base_setjmp, @function
 #else
 	.def base_setjmp; .scl 2; .type 32; .endef
@@ -20,7 +20,7 @@ base_setjmp:
 	ret
 
 	.globl	base_longjmp
-#ifndef __MINGW32__
+#if !defined(__MINGW32__) && !defined(__CYGWIN__)
 	.type	base_longjmp, @function
 #else
 	.def base_longjmp; .scl 2; .type 32; .endef
-- 
2.52.0.windows.1

