--- squashfs-tools-4.7/squashfs-tools/nprocessors_compat.c.orig	2025-06-03 21:09:33.000000000 +0200
+++ squashfs-tools-4.7/squashfs-tools/nprocessors_compat.c	2025-07-10 19:30:14.754787900 +0200
@@ -23,7 +23,7 @@
 
 static int processors = -1;
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
 #include <sched.h>
 #include <unistd.h>
 
--- squashfs-tools-4.7/squashfs-tools/print_pager.h.orig	2025-07-10 19:33:03.683114600 +0200
+++ squashfs-tools-4.7/squashfs-tools/print_pager.h	2025-07-10 19:32:33.860852300 +0200
@@ -23,6 +23,8 @@
  * print_pager.h
  */
 
+#include <sys/types.h>
+
 #define TRUE 1
 #define FALSE 0
 
--- squashfs-tools-4.7/squashfs-tools/endian_compat.h.orig	2025-07-10 21:04:58.301215400 +0200
+++ squashfs-tools-4.7/squashfs-tools/endian_compat.h	2025-07-10 21:05:21.558736300 +0200
@@ -23,7 +23,7 @@
  *
  * endian_compat.h
  */
-#ifndef linux
+#if !defined(linux) && !defined(__CYGWIN__)
 #include <sys/types.h>
 #define __BYTE_ORDER BYTE_ORDER
 #define __BIG_ENDIAN BIG_ENDIAN
