diff --git a/jhbuild/defaults.jhbuildrc b/jhbuild/defaults.jhbuildrc
index e467a49..de715c4 100644
--- a/jhbuild/defaults.jhbuildrc
+++ b/jhbuild/defaults.jhbuildrc
@@ -78,7 +78,15 @@ repos = {}
 cvsroots = {}
 svnroots = {}
 branches = {}
-module_autogenargs = {}
+# Arch-specific setting: we need to pass PYTHON=/usr/bin/python2 when building some modules.
+module_autogenargs = {
+    'cairo': autogenargs + ' ac_cv_prog_RANLIB=gcc-ranlib RANLIB=gcc-ranlib AR=gcc-ar',
+    'evolution-data-server': autogenargs + ' PYTHON=/usr/bin/python2',
+    'gobject-introspection': autogenargs + ' PYTHON=/usr/bin/python2',
+    'itstool': autogenargs + ' PYTHON=/usr/bin/python2',
+    'telepathy-mission-control': autogenargs + ' PYTHON=/usr/bin/python2',
+    'WebKit': autogenargs + ' PYTHON=/usr/bin/python2',
+}
 module_cmakeargs = {}
 module_makeargs = {}
 module_extra_env = {}
