--- sqlite-src-3490100/main.mk.orig	2025-02-18 15:09:33.000000000 +0100
+++ sqlite-src-3490100/main.mk	2025-05-02 19:24:26.778929900 +0200
@@ -1929,7 +1929,7 @@
 # specifically does not have an --enable-rbu flag in the configure
 # script.
 rbu$(T.exe): $(TOP)/ext/rbu/rbu.c $(TOP)/ext/rbu/sqlite3rbu.c sqlite3.o
-	$(T.link) -I. -o $@ $(TOP)/ext/rbu/rbu.c sqlite3.o $(LDFLAGS.libsqlite3)
+	$(T.link) -I. -o $@ $(TOP)/ext/rbu/rbu.c $(TOP)/ext/rbu/sqlite3rbu.c sqlite3.o $(LDFLAGS.libsqlite3)
 
 loadfts$(T.exe): $(TOP)/tool/loadfts.c $(libsqlite3.LIB)
 	$(T.link) $(TOP)/tool/loadfts.c $(libsqlite3.LIB) \
