--- lemon.c.orig        2023-12-04 17:56:38.260892600 +0100
+++ lemon.c     2023-12-04 19:20:21.727095900 +0100
@@ -3688,11 +3688,19 @@
     tpltname = templatename;
   }else{
     toFree = tpltname = pathsearch(lemp->argv[0],templatename,0);
+    if ( tpltname==0 || access(tpltname,004)!=0 ) {
+      tpltname = 0;
+      sprintf(buf,"/usr/share/lemon/%s",templatename);
+      if ( access(buf,004)==0 ) {
+        tpltname = buf;
+      }
+    }
   }
   if( tpltname==0 ){
     fprintf(stderr,"Can't find the parser driver template file \"%s\".\n",
     templatename);
     lemp->errorcnt++;
+    free(toFree);
     return 0;
   }
   in = fopen(tpltname,"rb");
