--- lemon.c.orig	2020-12-01 21:30:32.470000000 +0100
+++ lemon.c	2020-12-01 21:28:20.268940000 +0100
@@ -3672,6 +3672,10 @@
     toFree = tpltname = pathsearch(lemp->argv0,templatename,0);
   }
   if( tpltname==0 ){
+    sprintf(buf,"/usr/share/lemon/%s",templatename);
+    tpltname = buf;
+  }
+  if( tpltname==0 ){
     fprintf(stderr,"Can't find the parser driver template file \"%s\".\n",
     templatename);
     lemp->errorcnt++;
@@ -3679,6 +3683,11 @@
   }
   in = fopen(tpltname,"rb");
   if( in==0 ){
+    sprintf(buf,"/usr/share/lemon/%s",tpltname);
+    tpltname = buf;
+    in = fopen(tpltname,"rb");
+  }
+  if( in==0 ){
     fprintf(stderr,"Can't open the template file \"%s\".\n",tpltname);
     lemp->errorcnt++;
   }
