diff -Naur sqlite-autoconf-3100000-orig/sqlite3.c sqlite-autoconf-3100000/sqlite3.c
--- sqlite-autoconf-3100000-orig/sqlite3.c	2014-11-08 20:34:01.593800000 +0300
+++ sqlite-autoconf-3100000/sqlite3.c	2014-11-08 20:37:42.146600000 +0300
@@ -100964,7 +100964,9 @@
     sqlite3_free(zAltFile);
 #if defined(_WIN32) || defined(__CYGWIN__)
     if( handle==0 ){
-#  if defined(__CYGWIN__)
+#  if defined(__MSYS__)
+      zAltFile = sqlite3_mprintf("msys-sqlite3%s-0.%s", zFile, azEnding);
+#  elif defined(__CYGWIN__)
       zAltFile = sqlite3_mprintf("cygsqlite3%s-0.%s", zFile, azEnding);
 #  else /* how about Msys/Msys-2 ??? */
       zAltFile = sqlite3_mprintf("sqlite3%s.%s", zFile, azEnding);
