--- origsrc/sqlite-autoconf-3210000/CYGWIN-PATCHES/README	1970-01-01 01:00:00.000000000 +0100
+++ src/sqlite-autoconf-3210000/CYGWIN-PATCHES/README	2017-11-06 12:39:14.477098700 +0100
@@ -0,0 +1,349 @@
+sqlite3
+-------------------------------------------
+SQLite is a C library that implements an embeddable SQL database engine.
+Programs that link with the SQLite library can have SQL database access
+without running a separate RDBMS process.  The distribution comes with a
+standalone command-line access program (sqlite) that can be used to
+administer an SQLite database and which serves as an example of how to
+use the SQLite library.
+
+Runtime requirements:
+  cygwin-2.9
+  libreadline7
+
+Build requirements:
+(besides corresponding devel packages)
+  binutils-2.28
+  cygport-0.24
+  gawk-4.1
+  gcc-core-6.4.0
+  libtool-2.4.6
+  make-4.2
+  sed-4.4
+  icu-59.1
+
+Canonical homepage:
+  http://www.sqlite.org/
+
+-------------------------------------------
+
+Build instructions:
+  unpack sqlite3-3.21.0-1-src.tar.bz2
+    if you use setup to install this src package, it will be
+	 unpacked under /usr/src automatically
+  cd /usr/src
+  cygport ./sqlite3-3.21.0-1.cygport all
+
+This will create:
+  /usr/src/libsqlite3-devel-3.21.0-1.tar.bz2
+  /usr/src/libsqlite3_0-3.21.0-1.tar.bz2
+  /usr/src/sqlite3-3.21.0-1-src.tar.bz2
+  /usr/src/sqlite3-3.21.0-1.tar.bz2
+
+-------------------------------------------
+
+Files included in the binary packages:
+
+(sqlite3)
+  /usr/bin/sqlite3.exe
+  /usr/bin/sqldiff.exe
+  /usr/share/doc/Cygwin/sqlite3.README
+  /usr/share/doc/sqlite3/README
+  /usr/share/man/man1/sqlite3.1.gz
+
+(tcl-sqlite3)
+  /usr/lib/sqlite3/pkgIndex.tcl
+  /usr/lib/sqlite3/tclsqlite3.dll
+  /usr/share/man/mann/sqlite3.n.gz
+
+(sqlite3-extensions)
+  /usr/bin/cygsqlite3amatch-0.dll
+  /usr/bin/cygsqlite3closure-0.dll
+  /usr/bin/cygsqlite3completion-0.dll
+  /usr/bin/cygsqlite3eval-0.dll
+  /usr/bin/cygsqlite3fileio-0.dll
+  /usr/bin/cygsqlite3fuzzer-0.dll
+  /usr/bin/cygsqlite3ieee754-0.dll
+  /usr/bin/cygsqlite3nextchar-0.dll
+  /usr/bin/cygsqlite3percentile-0.dll
+  /usr/bin/cygsqlite3regexp-0.dll
+  /usr/bin/cygsqlite3remember-0.dll
+  /usr/bin/cygsqlite3rot13-0.dll
+  /usr/bin/cygsqlite3series-0.dll
+  /usr/lib/libsqlite3sha1.dll.a
+  /usr/lib/libsqlite3shathree.dll.a
+  /usr/bin/cygsqlite3showauth-0.dll
+  /usr/bin/cygsqlite3spellfix-0.dll
+  /usr/bin/cygsqlite3stmt-0.dll
+  /usr/bin/cygsqlite3totype-0.dll
+  /usr/bin/cygsqlite3unionvtab-0.dll
+  /usr/bin/cygsqlite3vtshim-0.dll
+  /usr/bin/cygsqlite3wholenumber-0.dll
+
+(sqlite3-compress)
+  /usr/bin/cygsqlite3compress-0.dll
+
+(sqlite3-icu)
+  /usr/bin/cygsqlite3icu-0.dll
+
+(sqlite3-vfslog)
+  /usr/bin/cygsqlite3vfslog-0.dll
+
+(sqlite3-rbu)
+  /usr/bin/rbu.exe
+  /usr/bin/cygsqlite3rbu-0.dll
+
+(libsqlite3_0)
+  /usr/bin/cygsqlite3-0.dll
+
+(libsqlite3-devel)
+  /usr/include/sqlite3.h
+  /usr/include/sqlite3ext.h
+  /usr/lib/libsqlite3.dll.a
+  /usr/lib/libsqlite3amatch.dll.a
+  /usr/lib/libsqlite3carray.dll.a
+  /usr/lib/libsqlite3closure.dll.a
+  /usr/lib/libsqlite3compress.dll.a
+  /usr/lib/libsqlite3csv.dll.a
+  /usr/lib/libsqlite3eval.dll.a
+  /usr/lib/libsqlite3fileio.dll.a
+  /usr/lib/libsqlite3fuzzer.dll.a
+  /usr/lib/libsqlite3icu.dll.a
+  /usr/lib/libsqlite3ieee754.dll.a
+  /usr/lib/libsqlite3memvfs.dll.a
+  /usr/lib/libsqlite3nextchar.dll.a
+  /usr/lib/libsqlite3percentile.dll.a
+  /usr/lib/libsqlite3rbu.dll.a
+  /usr/lib/libsqlite3regexp.dll.a
+  /usr/lib/libsqlite3remember.dll.a
+  /usr/lib/libsqlite3rot13.dll.a
+  /usr/lib/libsqlite3series.dll.a
+  /usr/lib/libsqlite3sha1.dll.a
+  /usr/lib/libsqlite3shathree.dll.a
+  /usr/lib/libsqlite3showauth.dll.a
+  /usr/lib/libsqlite3spellfix.dll.a
+  /usr/lib/libsqlite3totype.dll.a
+  /usr/lib/libsqlite3vfslog.dll.a
+  /usr/lib/libsqlite3vtshim.dll.a
+  /usr/lib/libsqlite3wholenumber.dll.a
+  /usr/lib/pkgconfig/sqlite3.pc
+
+------------------
+
+Port Notes:
+
+The Cygwin build contains VFS'es which are either win32 or unix compatible:
+
+   "win32": (default)     Uses Win32 API, has 4096 byte path limit.
+   "win32-longpath":      Uses Win32 API, has 65535 byte and 32768 chars path
+                          limit.
+   "win32-none":          Same as "win32", but without locking.
+   "none":                Same as "win32-none".
+   "win32-longpath-none": Same as "win32-longpath", but without locking.
+   "unix"/"unix-excl":    Uses Win32 file locking as supplied by cygwin1.dll.
+                          Works almost the same as "win32", but does not have
+                          a retry mechanism. This locking is always exclusive.
+
+   "unix-none":           Same as corresponding UNIX locking methods.
+   "unix-posix":
+   "unix-dotfile":
+   "unix-flock":
+
+----- version 3.21.0-1 -----
+Corresponds to upstream SQLite 3.21.0
+
+----- version 3.20.1-1 -----
+Corresponds to upstream SQLite 3.20.1
+
+----- version 3.19.3-1 -----
+Corresponds to upstream SQLite 3.19.3
+
+----- version 3.18.0-1 -----
+Corresponds to upstream SQLite 3.18.0
+
+----- version 3.16.2-1 -----
+Corresponds to upstream SQLite 3.16.2
+
+----- version 3.15.2-1 -----
+Corresponds to upstream SQLite 3.15.2
+
+----- version 3.14.1-1 -----
+Corresponds to upstream SQLite 3.14.1
+
+----- version 3.13.0-1 -----
+Corresponds to upstream SQLite 3.13.0
+
+----- version 3.12.2-1 -----
+Corresponds to upstream SQLite 3.12.2
+
+Additional changes:
+- Upgraded dependency to ICU 5.7
+
+----- version 3.11.1-1 -----
+Corresponds to upstream SQLite 3.11.1
+
+----- version 3.10.1-1 -----
+Corresponds to upstream SQLite 3.10.1
+
+----- version 3.10.0-1 -----
+Corresponds to upstream SQLite 3.10.0
+
+----- version 3.9.2-1 -----
+Corresponds to upstream SQLite 3.9.2
+
+----- version 3.9.0-1 (should have been numbered 3.9.1-1) -----
+Corresponds to upstream SQLite 3.9.1
+
+The FTS5 extension is no longer available as a separate dll, it is
+now part of cygsqlite3-0.dll, just as the JSON1 extension.
+
+----- version 3.8.11.1-1 -----
+Corresponds to upstream SQLite 3.8.11.1
+
+Additional changes:
+- Compiled with the following additional compilation flag:
+	-DSQLITE_ENABLE_DBSTAT_VTAB=1
+
+----- version 3.8.9-1 -----
+Corresponds to upstream SQLite 3.8.9
+
+Additional changes:
+- Compiled with the following additional compilation flags:
+	-DSQLITE_ENABLE_UPDATE_DELETE_LIMIT=1
+	-DSQLITE_OMIT_LOOKASIDE=1
+	-DSQLITE_SECURE_DELETE=1
+	-DSQLITE_SOUNDEX=1
+  Those are the same set of flags enabled in the corresponding
+  Ubuntu SQLite package, for improved linux compatibility.
+- New extension "showauth".
+- "compress" extension modified such that it can decompress
+  blobs which are compressed with the "zlib" extension. "zlib"
+  extension is now deprecated, and will be removed in the SQLite
+  3.8.10 Cygwin package. Please switch to use "compress" in stead.
+
+----- version 3.8.8.3-1 -----
+Corresponds to upstream SQLite 3.8.8.3
+
+Additional changes:
+- Upgraded dependency to ICU 5.4
+- Compiled with SQLITE_ENABLE_FTS3_PARENTHESIS enabled. This option
+  modifies the query pattern parser in FTS3 such that it supports
+  operators AND and NOT (in addition to the usual OR and NEAR) and
+  also allows query expressions to contain nested parenthesis.
+
+Use of CYGWIN_SQLITE_LOCKING environment variable now fully
+removed from the SQLite code.
+
+If you want to restore the old behavior, add the following
+lines of code somewhere at the start of your program:
+
+    const char *zVfs = getenv("CYGWIN_SQLITE_LOCKING");
+    if (zVfs) {
+        sqlite3_vfs *pVfs = sqlite3_vfs_find(zVfs);
+        if (pVfs) sqlite3_vfs_register(pVfs, 1);
+    }
+
+----- version 3.8.7.4-1 -----
+Corresponds to upstream SQLite 3.8.7.4
+
+Deprecate use of CYGWIN_SQLITE_LOCKING
+environment variable, it's a non-standard
+way of switching VFS and it appears that
+no-one should actually use it.
+
+----- version 3.8.6-1 -----
+Corresponds to upstream SQLite 3.8.6
+
+Does not use win32 mutexes internally any more, but
+cygwin (pthread) mutexes. This should improve the
+signal-handling of applications using SQLite.
+
+----- version 3.8.5-1 -----
+Corresponds to upstream SQLite 3.8.5
+
+----- version 3.8.4.3-1 -----
+Corresponds to upstream SQLite 3.8.4.3
+
+Default VFS is now "win32", which is extended being
+able to handle up to 4096 byte path lenghts.
+You can switch to the old behavior by selecting the
+"unix" VFS, the environment variable CYGWIN_SQLITE_LOCKING
+will then determine the real VFS in use.
+
+----- version 3.8.3.1-1 -----
+Corresponds to upstream SQLite 3.8.3.1
+
+----- version 3.8.2-3 -----
+"winnt" VFS renamed to "unix".
+
+Corresponds to upstream SQLite 3.8.2
+
+Better UNC/Extended windows path handling in "win32", "winnt"
+and "win32-longpath" VFS's
+
+----- version 3.8.1-1 -----
+Version bump.
+New "win32", "win32-longpath", and "winnt" VFS. The default is
+"unix", which behaves identical to earlier Cygwin versions.
+
+Versions below starting with 3.5.8-1 were done by Warren Young
+
+----- version 3.7.x-y -----
+Version bump, not further specified here.
+
+----- version 3.6.2-1 -----
+Restored DLL version of library
+Distributing *.a in devel package instead of lib package.
+
+----- version 3.5.8-1 -----
+Fork of Cygwin Ports version, removing Tcl support since Cygwin doesn't
+distribute Tcl anymore.
+
+Maintainer of versions below unknown.
+
+----- version 3.5.6-1 -----
+Version bump.
+Enabled the load_extension API for the PHP bindings.
+
+----- version 3.5.4-1 -----
+Version bump.
+
+----- version 3.5.3-1 -----
+Version bump.
+
+----- version 3.5.2-1 -----
+Version bump.
+
+----- version 3.3.17-1 -----
+Version bump.
+
+----- version 3.3.12-1 -----
+Version bump.
+
+----- version 3.3.10-1 -----
+Version bump.
+
+----- version 3.3.7-1 -----
+Version bump.  Broke out separate binary packages.
+
+----- version 3.3.5-1 -----
+Version bump.
+
+----- version 3.2.8-1 -----
+Version bump.
+
+----- version 3.2.7-1 -----
+Version bump.
+
+----- version 3.2.6-1 -----
+Version bump.
+
+----- version 3.2.1-1 -----
+Version bump for 3.0 branch.
+
+----- version 3.0.8-1 -----
+Initial release of the 3.0 branch for Cygwin.
+This package can be installed in parallel with sqlite-2.8.x.
+
+
+Cygwin port maintained by Jan Nijtmans
