diff -Naur sqlite-autoconf-3080701-orig/sqlite3.c sqlite-autoconf-3080701/sqlite3.c
--- sqlite-autoconf-3080701-orig/sqlite3.c	2014-11-08 20:34:01.593800000 +0300
+++ sqlite-autoconf-3080701/sqlite3.c	2014-11-08 20:37:42.146600000 +0300
@@ -33758,14 +33758,14 @@
   { "GetModuleHandleW",         (SYSCALL)0,                      0 },
 #endif
 
-#define osGetModuleHandleW ((HMODULE(*)(LPCWSTR))aSyscall[76].pCurrent)
+#define osGetModuleHandleW ((HMODULE(WINAPI*)(LPCWSTR))aSyscall[76].pCurrent)
 
 #if defined(SQLITE_WIN32_HAS_WIDE) && !defined(SQLITE_OMIT_LOAD_EXTENSION)
   { "SetDllDirectoryW",         (SYSCALL)SetDllDirectoryW,       0 },
 #else
   { "SetDllDirectoryW",         (SYSCALL)0,                      0 },
 #endif
 
-#define osSetDllDirectoryW ((BOOL(*)(LPCWSTR))aSyscall[77].pCurrent)
+#define osSetDllDirectoryW ((BOOL(WINAPI*)(LPCWSTR))aSyscall[77].pCurrent)
 
 #if defined(__CYGWIN__)
   { "getenv",                   (SYSCALL)getenv,                 0 },
@@ -38527,7 +38527,7 @@
 #ifdef _WIN32
   module = osGetModuleHandleW(L"CYGWIN1.DLL");
   if( !module){
-    module = osGetModuleHandleW(L"MSYS-1.0.DLL");
+    module = osGetModuleHandleW(L"MSYS-2.0.DLL");
   }
   if( module ){
     for( i=78; i<ArraySize(aSyscall); ++i ){
@@ -100964,6 +100964,12 @@
       handle = sqlite3OsDlOpen(pVfs, zAltFile);
       sqlite3_free(zAltFile);
     }
+    if( handle==0 ){
+      zAltFile = sqlite3_mprintf("msys-sqlite3%s-0.%s", zFile, azEnding);
+      if( zAltFile==0 ) return SQLITE_NOMEM;
+      handle = sqlite3OsDlOpen(pVfs, zAltFile);
+      sqlite3_free(zAltFile);
+    }
 #endif
   }
   if( handle==0 ){
