diff -Naur sqlite-autoconf-3080001-orig/config.guess sqlite-autoconf-3080001/config.guess
--- sqlite-autoconf-3080001-orig/config.guess	2013-08-29 21:47:30.000000000 +0400
+++ sqlite-autoconf-3080001/config.guess	2013-08-30 13:09:08.752929700 +0400
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur sqlite-autoconf-3080001-orig/tea/tclconfig/tcl.m4 sqlite-autoconf-3080001/tea/tclconfig/tcl.m4
--- sqlite-autoconf-3080001-orig/tea/tclconfig/tcl.m4	2013-08-29 21:47:30.000000000 +0400
+++ sqlite-autoconf-3080001/tea/tclconfig/tcl.m4	2013-08-30 13:04:21.379882800 +0400
@@ -1423,7 +1423,7 @@
 	    CC_SEARCH_FLAGS=""
 	    LD_SEARCH_FLAGS=""
 	    ;;
-	CYGWIN_*)
+	CYGWIN_* | *MSYS_*)
 	    SHLIB_CFLAGS=""
 	    SHLIB_LD='${CC} -shared'
 	    SHLIB_SUFFIX=".dll"
@@ -1997,7 +1997,7 @@
 	case $system in
 	    AIX-*) ;;
 	    BSD/OS*) ;;
-	    CYGWIN_*|MINGW32_*) ;;
+	    CYGWIN_*|MINGW32_*|MSYS_*) ;;
 	    IRIX*) ;;
 	    NetBSD-*|FreeBSD-*|OpenBSD-*) ;;
 	    Darwin-*) ;;
@@ -2870,7 +2870,7 @@
 	    EXEEXT=".exe"
 	    TEA_PLATFORM="windows"
 	    ;;
-	*CYGWIN_*)
+	*CYGWIN_* | *MSYS_*)
 	    CYGPATH=echo
 	    EXEEXT=".exe"
 	    # TEA_PLATFORM is determined later in LOAD_TCLCONFIG
