diff -Naur sqlite-autoconf-3080701-orig/sqlite3.c sqlite-autoconf-3080701/sqlite3.c
--- sqlite-autoconf-3080701-orig/sqlite3.c	2014-11-08 20:34:01.593800000 +0300
+++ sqlite-autoconf-3080701/sqlite3.c	2014-11-08 20:37:42.146600000 +0300
@@ -38527,7 +38527,7 @@
 #ifdef _WIN32
   module = osGetModuleHandleW(L"CYGWIN1.DLL");
   if( !module){
-    module = osGetModuleHandleW(L"MSYS-1.0.DLL");
+    module = osGetModuleHandleW(L"MSYS-2.0.DLL");
   }
   if( module ){
     for( i=78; i<ArraySize(aSyscall); ++i ){
@@ -100964,6 +100964,12 @@
       handle = sqlite3OsDlOpen(pVfs, zAltFile);
       sqlite3_free(zAltFile);
     }
+    if( handle==0 ){
+      zAltFile = sqlite3_mprintf("msys-sqlite3%s-0.%s", zFile, azEnding);
+      if( zAltFile==0 ) return SQLITE_NOMEM;
+      handle = sqlite3OsDlOpen(pVfs, zAltFile);
+      sqlite3_free(zAltFile);
+    }
 #endif
   }
   if( handle==0 ){
