From 277489f19bd0e58396606973bb79b6faa8e1e0a5 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Thu, 30 Oct 2014 20:45:19 +0000
Subject: [PATCH] fix printk to use FMT-cell instead of %x where it should

---
 arch/unix/unix.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/unix/unix.c b/arch/unix/unix.c
index 1f628eb..db70e0a 100644
--- a/arch/unix/unix.c
+++ b/arch/unix/unix.c
@@ -324,10 +324,10 @@ segv_handler(int signo __attribute__ ((unused)),
 	if (PC >= (ucell) dict && PC <= (ucell) dict + dicthead)
 		addr = *(ucell *) PC;
 
-	printk("panic: segmentation violation at %x\n", (ucell)si->si_addr);
-	printk("dict=0x%x here=0x%x(dict+0x%x) pc=0x%x(dict+0x%x)\n",
+	printk("panic: segmentation violation at " FMT_cell "\n", (ucell)si->si_addr);
+	printk("dict=0x" FMT_cell " here=0x" FMT_cell " (dict+0x" FMT_cell ") pc=0x" FMT_cell "(dict+0x" FMT_cell ")\n",
 	       (ucell)dict, (ucell)dict + dicthead, dicthead, PC, PC - (ucell) dict);
-	printk("dstackcnt=%d rstackcnt=%d instruction=%x\n",
+	printk("dstackcnt=%d rstackcnt=%d instruction=" FMT_cell "\n",
 	       dstackcnt, rstackcnt, addr);
 
 #ifdef CONFIG_DEBUG_DSTACK
@@ -566,7 +566,7 @@ int main(int argc, char *argv[])
 			if (optind + 1 != argc)
 				printk("Warning: only first dictionary used.\n");
 
-			printk("dictionary loaded (%d bytes).\n", dicthead);
+			printk("dictionary loaded (" FMT_cell "bytes).\n", dicthead);
 			printk("Initializing memory...");
 		}
 		init_memory();
-- 
2.1.2

