diff -Naur pcre2-10.31-orig/ar-lib pcre2-10.31/ar-lib
--- pcre2-10.31-orig/ar-lib	2018-02-12 14:15:50.000000000 +0300
+++ pcre2-10.31/ar-lib	2018-06-26 10:00:07.074240500 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -65,7 +65,7 @@
 	mingw)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin)
+	cygwin|msys)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine)
diff -Naur pcre2-10.31-orig/compile pcre2-10.31/compile
--- pcre2-10.31-orig/compile	2018-02-12 14:15:50.000000000 +0300
+++ pcre2-10.31/compile	2018-06-26 10:00:07.084240500 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur pcre2-10.31-orig/config.guess pcre2-10.31/config.guess
--- pcre2-10.31-orig/config.guess	2018-02-12 14:15:50.000000000 +0300
+++ pcre2-10.31/config.guess	2018-06-26 10:00:07.094240500 +0300
@@ -892,6 +892,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur pcre2-10.31-orig/configure.ac pcre2-10.31/configure.ac
--- pcre2-10.31-orig/configure.ac	2018-02-12 14:15:35.000000000 +0300
+++ pcre2-10.31/configure.ac	2018-06-26 10:00:07.094240500 +0300
@@ -787,7 +787,7 @@
 NO_UNDEFINED=
 EXPORT_ALL_SYMBOLS=
 case $host_os in
-  cygwin* | mingw* )
+  cygwin* | msys* | mingw* )
     if test X"$enable_shared" = Xyes; then
       NO_UNDEFINED="-no-undefined"
       EXPORT_ALL_SYMBOLS="-Wl,--export-all-symbols"
