--- remake/configure.ac.orig	2017-12-04 11:47:01.148856700 +0300
+++ remake/configure.ac	2017-12-04 11:47:33.690518200 +0300
@@ -181,9 +181,10 @@
 # comes with it's own PC file so we have to specify them as individual
 # packages.  Ugh.
 AS_IF([test "x$with_guile" != xno],
-[ PKG_CHECK_MODULES([GUILE], [guile-2.0], [have_guile=yes],
-  [PKG_CHECK_MODULES([GUILE], [guile-1.8], [have_guile=yes],
-    [have_guile=no])])
+[ PKG_CHECK_MODULES([GUILE], [guile-2.2], [have_guile=yes],
+  [ PKG_CHECK_MODULES([GUILE], [guile-2.0], [have_guile=yes],
+    [PKG_CHECK_MODULES([GUILE], [guile-1.8], [have_guile=yes],
+      [have_guile=no])])])
 ])
 
 AS_IF([test "$have_guile" = yes],
