--- origsrc/gettext-1.05/Makefile.PL	2005-01-30 17:30:22.000000000 -0600
+++ src/gettext-1.05/Makefile.PL	2008-03-10 23:41:52.343750000 -0500
@@ -50,7 +50,7 @@
 	close TEST;
 	open(SAVE, ">&STDERR");
 	open(STDERR, ">/dev/null");
-	system($cc . " -o conftest " . $libs . " conftest.c");
+	system($cc . " -o conftest " . " conftest.c " . $libs);
 	my $exitstatus = $?;
 	open(STDERR, ">&SAVE");
 	if ($exitstatus != 0) {
