From 2e911f22e68e05fe728dedefe9a54447834811d3 Mon Sep 17 00:00:00 2001
From: Jonathan Yong <10walls@gmail.com>
Date: Tue, 8 Sep 2020 15:21:56 +0200
Subject: [PATCH 10/12] Cygwin: g++ time

---
 libstdc++-v3/acinclude.m4 | 3 +++
 libstdc++-v3/configure    | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/libstdc++-v3/acinclude.m4 b/libstdc++-v3/acinclude.m4
index e668d2dba27..ee054ad5727 100644
--- a/libstdc++-v3/acinclude.m4
+++ b/libstdc++-v3/acinclude.m4
@@ -1764,7 +1764,10 @@ AC_DEFUN([GLIBCXX_ENABLE_LIBSTDCXX_TIME], [
 
     case "${target_os}" in
       cygwin*)
+        ac_has_clock_monotonic=yes
+        ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
+        ac_has_sched_yield=yes
         ;;
       mingw*)
         ac_has_win32_sleep=yes
diff --git a/libstdc++-v3/configure b/libstdc++-v3/configure
index 67d2b8c7b72..00f2cec330c 100755
--- a/libstdc++-v3/configure
+++ b/libstdc++-v3/configure
@@ -21347,7 +21347,10 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
     case "${target_os}" in
       cygwin*)
+        ac_has_clock_monotonic=yes
+        ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
+        ac_has_sched_yield=yes
         ;;
       mingw*)
         ac_has_win32_sleep=yes
-- 
2.45.1

