From 109c129db070da8a1935ca6d325a8f5eb5ed0930 Mon Sep 17 00:00:00 2001
From: Jonathan Yong <10walls@gmail.com>
Date: Tue, 8 Sep 2020 15:21:56 +0200
Subject: [PATCH 12/18] Cygwin: g++ time

---
 libstdc++-v3/acinclude.m4 | 3 +++
 libstdc++-v3/configure    | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/libstdc++-v3/acinclude.m4 b/libstdc++-v3/acinclude.m4
index ee5e0336f2c..bc7d137dc74 100644
--- a/libstdc++-v3/acinclude.m4
+++ b/libstdc++-v3/acinclude.m4
@@ -1410,7 +1410,10 @@ AC_DEFUN([GLIBCXX_ENABLE_LIBSTDCXX_TIME], [
 
     case "${target_os}" in
       cygwin*)
+        ac_has_clock_monotonic=yes
+        ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
+        ac_has_sched_yield=yes
         ;;
       darwin*)
         ac_has_nanosleep=yes
diff --git a/libstdc++-v3/configure b/libstdc++-v3/configure
index 9f9c5a2419a..cc6e45592e1 100755
--- a/libstdc++-v3/configure
+++ b/libstdc++-v3/configure
@@ -21060,7 +21060,10 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
     case "${target_os}" in
       cygwin*)
+        ac_has_clock_monotonic=yes
+        ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
+        ac_has_sched_yield=yes
         ;;
       darwin*)
         ac_has_nanosleep=yes
-- 
2.28.0

