--- a/libgomp/config/cygwin/plugin-suffix.h	1969-12-31 18:00:00.000000000 -0600
+++ b/libgomp/config/cygwin/plugin-suffix.h	2016-01-31 17:06:42.166913600 -0600
@@ -0,0 +1,2 @@
+#define SONAME_PREFIX "cyg"
+#define SONAME_SUFFIX(n) ("-" #n ".dll")
--- a/libgomp/config/darwin/plugin-suffix.h	2015-01-28 15:23:14.000000000 -0600
+++ b/libgomp/config/darwin/plugin-suffix.h	2016-01-31 17:06:46.445956900 -0600
@@ -23,4 +23,5 @@
    see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    <http://www.gnu.org/licenses/>.  */
 
+#define SONAME_PREFIX "lib"
 #define SONAME_SUFFIX(n) ("." #n ".dylib")
--- a/libgomp/config/hpux/plugin-suffix.h	2015-01-28 15:23:14.000000000 -0600
+++ b/libgomp/config/hpux/plugin-suffix.h	2016-01-31 17:06:52.059169700 -0600
@@ -23,4 +23,5 @@
    see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    <http://www.gnu.org/licenses/>.  */
 
+#define SONAME_PREFIX "lib"
 #define SONAME_SUFFIX(n) (".sl." #n)
--- a/libgomp/config/posix/plugin-suffix.h	2015-01-28 15:23:14.000000000 -0600
+++ b/libgomp/config/posix/plugin-suffix.h	2016-01-31 17:06:58.481985300 -0600
@@ -23,4 +23,5 @@
    see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    <http://www.gnu.org/licenses/>.  */
 
+#define SONAME_PREFIX "lib"
 #define SONAME_SUFFIX(n) (".so." #n)
--- a/libgomp/configure.tgt	2015-03-13 05:57:07.000000000 -0500
+++ b/libgomp/configure.tgt	2016-01-31 16:25:19.321132200 -0600
@@ -132,6 +132,10 @@ case "${target}" in
 	esac
 	;;
 
+  *-*-cygwin*)
+	config_path="cygwin posix"
+	;;
+
   *-*-mingw32*)
 	config_path="mingw32 posix"
 	;;
--- a/libgomp/target.c	2015-04-08 10:58:33.000000000 -0500
+++ b/libgomp/target.c	2016-01-31 17:07:08.238724300 -0600
@@ -1215,7 +1215,7 @@ gomp_load_plugin_for_device (struct gomp
 static void
 gomp_target_init (void)
 {
-  const char *prefix ="libgomp-plugin-";
+  const char *prefix = SONAME_PREFIX "gomp-plugin-";
   const char *suffix = SONAME_SUFFIX (1);
   const char *cur, *next;
   char *plugin_name;
