https://cygwin.com/ml/cygwin/2016-02/msg00393.html

--- a/gcc/config/i386/cygwin.h	2016-02-11 17:21:55.058829500 -0600
+++ b/gcc/config/i386/cygwin.h	2016-02-25 11:48:51.241207300 -0600
@@ -39,7 +39,6 @@ along with GCC; see the file COPYING3.
 
 #undef STARTFILE_SPEC
 #define STARTFILE_SPEC "\
-  -L%R/usr/lib/w32api \
   %{!shared: %{!mdll: crt0%O%s \
   %{pg:gcrt0%O%s}}}\
   %{shared:crtbeginS.o%s;:crtbegin.o%s} \
@@ -56,6 +55,13 @@
     fvtable-verify=std:vtv_end.o%s} \
    crtend.o%s"
 
+#undef STANDARD_STARTFILE_PREFIX
+#define STANDARD_STARTFILE_PREFIX "/usr/lib/"
+#undef STANDARD_STARTFILE_PREFIX_1
+#define STANDARD_STARTFILE_PREFIX_1 "/lib/"
+#undef STANDARD_STARTFILE_PREFIX_2
+#define STANDARD_STARTFILE_PREFIX_2 "/usr/lib/w32api/"
+
 /* Normally, -lgcc is not needed since everything in it is in the DLL, but we
    want to allow things to be added to it when installing new versions of
    GCC without making a new CYGWIN.DLL, so we leave it.  Profiling is handled
