--- origsrc/gcc-6.4.0/gcc/glimits.h	2016-01-04 08:30:50.652828000 -0600
+++ src/gcc-6.4.0/gcc/glimits.h	2017-11-13 18:46:28.887539100 -0600
@@ -111,7 +111,7 @@
 # define ULLONG_MAX (LLONG_MAX * 2ULL + 1ULL)
 #endif
 
-#if defined (__GNU_LIBRARY__) ? defined (__USE_GNU) : !defined (__STRICT_ANSI__)
+#if defined (__GNU_LIBRARY__) ? defined (__USE_GNU) : defined (__NEWLIB__) ? __GNU_VISIBLE : !defined (__STRICT_ANSI__)
 /* Minimum and maximum values a `signed long long int' can hold.  */
 # undef LONG_LONG_MIN
 # define LONG_LONG_MIN (-LONG_LONG_MAX - 1LL)
