diff -Naur gcc-4.9.2-orig/libjava/classpath/examples/gnu/classpath/examples/awt/HintingDemo.java gcc-4.9.2/libjava/classpath/examples/gnu/classpath/examples/awt/HintingDemo.java
--- gcc-4.9.2-orig/libjava/classpath/examples/gnu/classpath/examples/awt/HintingDemo.java	2011-01-11 22:46:05.000000000 +0300
+++ gcc-4.9.2/libjava/classpath/examples/gnu/classpath/examples/awt/HintingDemo.java	2014-11-03 23:44:02.743000000 +0300
@@ -211,7 +211,7 @@
 
   HintingDemo()
   {
-    File file = new File("/usr/share/fonts/truetype/freefont/FreeSans.ttf");
+    File file = new File("/usr/share/fonts/TTF/Vera.ttf");
     loadFont(file);
     setLayout(new BorderLayout());
     chooser = new Chooser();
diff -Naur gcc-4.9.2-orig/libjava/classpath/gnu/java/awt/peer/ClasspathDesktopPeer.java gcc-4.9.2/libjava/classpath/gnu/java/awt/peer/ClasspathDesktopPeer.java
--- gcc-4.9.2-orig/libjava/classpath/gnu/java/awt/peer/ClasspathDesktopPeer.java	2011-01-11 22:46:05.000000000 +0300
+++ gcc-4.9.2/libjava/classpath/gnu/java/awt/peer/ClasspathDesktopPeer.java	2014-11-03 23:44:02.743000000 +0300
@@ -58,7 +58,7 @@
     implements DesktopPeer
 {
   /** This is the fallback browser, if no desktop was detected. */
-  protected static final String _DEFAULT_BROWSER = "firefox";
+  protected static final String _DEFAULT_BROWSER = "xdg-open";
 
   /** gnu.java.awt.peer.Desktop.html.command */
   protected static final String _BROWSE = "html";
diff -Naur gcc-4.9.2-orig/libjava/classpath/gnu/java/awt/peer/ClasspathFontPeer.java gcc-4.9.2/libjava/classpath/gnu/java/awt/peer/ClasspathFontPeer.java
--- gcc-4.9.2-orig/libjava/classpath/gnu/java/awt/peer/ClasspathFontPeer.java	2011-01-11 22:46:05.000000000 +0300
+++ gcc-4.9.2/libjava/classpath/gnu/java/awt/peer/ClasspathFontPeer.java	2014-11-03 23:44:02.743000000 +0300
@@ -158,6 +158,8 @@
 
   protected static boolean isLogicalFontName(String name)
   {
+    return false;
+/*
     String uname = name.toUpperCase ();
     return (uname.equals ("SANSSERIF") ||
             uname.equals ("SERIF") ||
@@ -165,6 +167,7 @@
             uname.equals ("DIALOG") ||
             uname.equals ("DIALOGINPUT") ||
             uname.equals ("DEFAULT"));
+*/
   }
 
   protected static String logicalFontNameToFaceName (String name)
diff -Naur gcc-4.9.2-orig/libjava/classpath/resource/gnu/java/awt/font/fonts.properties gcc-4.9.2/libjava/classpath/resource/gnu/java/awt/font/fonts.properties
--- gcc-4.9.2-orig/libjava/classpath/resource/gnu/java/awt/font/fonts.properties	2007-10-21 15:06:09.000000000 +0300
+++ gcc-4.9.2/libjava/classpath/resource/gnu/java/awt/font/fonts.properties	2014-11-03 23:44:02.743000000 +0300
@@ -36,20 +36,20 @@
 # exception statement from your version.
 
 # List of available fonts.
-FreeMono/p=/usr/share/fonts/truetype/freefont/FreeMono.ttf
-FreeMono/b=/usr/share/fonts/truetype/freefont/FreeMonoBold.ttf
-FreeMono/bi=/usr/share/fonts/truetype/freefont/FreeMonoBoldOblique.ttf
-FreeMono/i=/usr/share/fonts/truetype/freefont/FreeMonoOblique.ttf
+FreeMono/p=/usr/share/fonts/TTF/VeraMono.ttf
+FreeMono/b=/usr/share/fonts/TTF/VeraMoBd.ttf
+FreeMono/bi=/usr/share/fonts/TTF/VeraMoBI.ttf
+FreeMono/i=/usr/share/fonts/TTF/VeraMoIt.ttf
 
-FreeSans/p=/usr/share/fonts/truetype/freefont/FreeSans.ttf
-FreeSans/b=/usr/share/fonts/truetype/freefont/FreeSansBold.ttf
-FreeSans/bi=/usr/share/fonts/truetype/freefont/FreeSansBoldOblique.ttf
-FreeSans/i=/usr/share/fonts/truetype/freefont/FreeSansOblique.ttf
+FreeSans/p=/usr/share/fonts/TTF/Vera.ttf
+FreeSans/b=/usr/share/fonts/TTF/VeraBd.ttf
+FreeSans/bi=/usr/share/fonts/TTF/VeraBI.ttf
+FreeSans/i=/usr/share/fonts/TTF/VeraIt.ttf
 
-FreeSerif/p=/usr/share/fonts/truetype/freefont/FreeSerif.ttf
-FreeSerif/b=/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf
-FreeSerif/bi=/usr/share/fonts/truetype/freefont/FreeSerifBoldOblique.ttf
-FreeSerif/i=/usr/share/fonts/truetype/freefont/FreeSerifOblique.ttf
+FreeSerif/p=/usr/share/fonts/TTF/VeraSe.ttf
+FreeSerif/b=/usr/share/fonts/TTF/VeraSeBd.ttf
+FreeSerif/bi=/usr/share/fonts/TTF/VeraSeBd.ttf
+FreeSerif/i=/usr/share/fonts/TTF/VeraSe.ttf
 
 # List of aliases.
 Dialog/a=FreeSans
