diff -Naur gcc-4.9.2-orig/libjava/gnu/java/nio/channels/natFileChannelPosix.cc gcc-4.9.2/libjava/gnu/java/nio/channels/natFileChannelPosix.cc
--- gcc-4.9.2-orig/libjava/gnu/java/nio/channels/natFileChannelPosix.cc	2007-04-03 04:05:50.000000000 +0300
+++ gcc-4.9.2/libjava/gnu/java/nio/channels/natFileChannelPosix.cc	2014-11-03 23:51:11.446600000 +0300
@@ -386,7 +386,7 @@
 
 #if defined (FIONREAD)
   r = ::ioctl (fd, FIONREAD, &num);
-  if (r == -1 && errno == ENOTTY)
+  if (r == -1 && (errno == ENOTTY || errno == EINVAL))
     {
       // If the ioctl doesn't work, we don't care.
       r = 0;
