diff -Naur gcc-4.9.2-orig/gcc/config/i386/cygwin.h gcc-4.9.2/gcc/config/i386/cygwin.h
--- gcc-4.9.2-orig/gcc/config/i386/cygwin.h	2014-11-03 23:44:00.902200000 +0300
+++ gcc-4.9.2/gcc/config/i386/cygwin.h	2014-11-04 00:03:50.792200000 +0300
@@ -143,7 +143,7 @@
 #define LIBGCC_SONAME "cyggcc_s" LIBGCC_EH_EXTN "-1.dll"
 
 /* We should find a way to not have to update this manually.  */
-#define LIBGCJ_SONAME "cyggcj" /*LIBGCC_EH_EXTN*/ "-15.dll"
+#define LIBGCJ_SONAME "cyggcj" /*LIBGCC_EH_EXTN*/ "-16.dll"
 
 /* Make stack executable to avoid DEP problems with trampolines.  */
 #define HAVE_ENABLE_EXECUTE_STACK
diff -Naur gcc-4.9.2-orig/gcc/config/i386/mingw32.h gcc-4.9.2/gcc/config/i386/mingw32.h
--- gcc-4.9.2-orig/gcc/config/i386/mingw32.h	2014-04-23 12:46:13.000000000 +0300
+++ gcc-4.9.2/gcc/config/i386/mingw32.h	2014-11-04 00:03:50.792200000 +0300
@@ -246,4 +246,4 @@
 #define LIBGCC_SONAME "libgcc_s" LIBGCC_EH_EXTN "-1.dll"
 
 /* We should find a way to not have to update this manually.  */
-#define LIBGCJ_SONAME "libgcj" /*LIBGCC_EH_EXTN*/ "-13.dll"
+#define LIBGCJ_SONAME "libgcj" /*LIBGCC_EH_EXTN*/ "-14.dll"
