diff -Naur gcc-4.9.2-orig/gcc/config.host gcc-4.9.2/gcc/config.host
--- gcc-4.9.2-orig/gcc/config.host	2014-01-03 01:23:26.000000000 +0300
+++ gcc-4.9.2/gcc/config.host	2014-11-04 00:33:32.998600000 +0300
@@ -219,14 +219,14 @@
     out_host_hook_obj=host-cygwin.o
     host_xmake_file="${host_xmake_file} i386/x-cygwin"
     host_exeext=.exe
-    host_lto_plugin_soname=cyglto_plugin-0.dll
+    host_lto_plugin_soname=cyglto_plugin.dll
     ;;
   i[34567]86-*-mingw32*)
     host_xm_file=i386/xm-mingw32.h
     host_xmake_file="${host_xmake_file} i386/x-mingw32"
     host_exeext=.exe
     out_host_hook_obj=host-mingw32.o
-    host_lto_plugin_soname=liblto_plugin-0.dll
+    host_lto_plugin_soname=liblto_plugin.dll
     ;;
   x86_64-*-mingw*)
     use_long_long_for_widest_fast_int=yes
@@ -234,7 +234,7 @@
     host_xmake_file="${host_xmake_file} i386/x-mingw32"
     host_exeext=.exe
     out_host_hook_obj=host-mingw32.o
-    host_lto_plugin_soname=liblto_plugin-0.dll
+    host_lto_plugin_soname=liblto_plugin.dll
     ;;
   i[34567]86-*-uwin*)
     echo "*** UWIN may not be used as a host platform because"
diff -Naur gcc-4.9.2-orig/lto-plugin/Makefile.am gcc-4.9.2/lto-plugin/Makefile.am
--- gcc-4.9.2-orig/lto-plugin/Makefile.am	2014-03-17 16:04:12.000000000 +0300
+++ gcc-4.9.2/lto-plugin/Makefile.am	2014-11-04 00:33:32.998600000 +0300
@@ -24,7 +24,7 @@
 	$(if $(wildcard ../libiberty/pic/libiberty.a),$(Wc)../libiberty/pic/libiberty.a,)
 # Note that we intentionally override the bindir supplied by ACX_LT_HOST_FLAGS
 liblto_plugin_la_LDFLAGS = $(AM_LDFLAGS) \
-	$(lt_host_flags) -module -bindir $(libexecsubdir) \
+	$(lt_host_flags) -module -avoid-version -bindir $(libexecsubdir) \
 	$(if $(wildcard ../libiberty/pic/libiberty.a),,-Wc,../libiberty/libiberty.a)
 liblto_plugin_la_DEPENDENCIES = $(if $(wildcard \
 	../libiberty/pic/libiberty.a),../libiberty/pic/libiberty.a,)
diff -Naur gcc-4.9.2-orig/lto-plugin/Makefile.in gcc-4.9.2/lto-plugin/Makefile.in
--- gcc-4.9.2-orig/lto-plugin/Makefile.in	2014-03-17 16:04:12.000000000 +0300
+++ gcc-4.9.2/lto-plugin/Makefile.in	2014-11-04 00:33:33.014200000 +0300
@@ -245,7 +245,7 @@
 
 # Note that we intentionally override the bindir supplied by ACX_LT_HOST_FLAGS
 liblto_plugin_la_LDFLAGS = $(AM_LDFLAGS) \
-	$(lt_host_flags) -module -bindir $(libexecsubdir) \
+	$(lt_host_flags) -module -avoid-version -bindir $(libexecsubdir) \
 	$(if $(wildcard ../libiberty/pic/libiberty.a),,-Wc,../libiberty/libiberty.a)
 
 liblto_plugin_la_DEPENDENCIES = $(if $(wildcard \
