From 748d4e91cce2f58ca76a223ab5108395fcdf784e Mon Sep 17 00:00:00 2001
From: Achim Gratz <Achim.Gratz@Stromeko.DE>
Date: Sun, 20 Oct 2024 16:52:06 +0200
Subject: [PATCH 401/404] fix build: stage1 now depends on libintl / libiconv

---
 gcc/Makefile.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gcc/Makefile.in b/gcc/Makefile.in
index 55b4cd7dbed..4f71b7f6d93 100644
--- a/gcc/Makefile.in
+++ b/gcc/Makefile.in
@@ -3348,7 +3348,8 @@ gengtype$(exeext) : gengtype.o gengtype-lex.o gengtype-parse.o \
 # Rule for the generator programs:
 $(genprog:%=build/gen%$(build_exeext)): build/gen%$(build_exeext): build/gen%.o $(BUILD_LIBDEPS)
 	+$(LINKER_FOR_BUILD) $(BUILD_LINKERFLAGS) $(BUILD_LDFLAGS) -o $@ \
-	    $(filter-out $(BUILD_LIBDEPS), $^) $(BUILD_LIBS)
+	    $(filter-out $(BUILD_LIBDEPS), $^) $(BUILD_LIBS) \
+	    $(LIBINTL) $(LIBICONV)
 
 omp-general.o: omp-device-properties.h
 
-- 
2.45.1

