From f26079a3ea4b4ddf4e1c500e82cd5511ca770984 Mon Sep 17 00:00:00 2001
From: Achim Gratz <Achim.Gratz@Stromeko.DE>
Date: Mon, 10 Feb 2025 18:45:54 +0100
Subject: [PATCH 404/404] fix build: opts

---
 gcc/opts.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gcc/opts.cc b/gcc/opts.cc
index ffcbdfef0bd..705899b0434 100644
--- a/gcc/opts.cc
+++ b/gcc/opts.cc
@@ -1751,8 +1751,11 @@ print_filtered_help (unsigned int include_flags,
 		      sizeof new_help - strlen (new_help),
 		      "  %s", _(use_diagnosed_msg));
 	  else
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
 	    snprintf (new_help, sizeof new_help,
 		      "%s  %s", help, _(use_diagnosed_msg));
+#pragma GCC diagnostic pop
 
 	  help = new_help;
 	}
-- 
2.45.1

