diff -Naur gcc-4.9.2-orig/gcc/testsuite/gcc.target/i386/pr25993.c gcc-4.9.2/gcc/testsuite/gcc.target/i386/pr25993.c
--- gcc-4.9.2-orig/gcc/testsuite/gcc.target/i386/pr25993.c	2010-05-21 18:03:26.000000000 +0300
+++ gcc-4.9.2/gcc/testsuite/gcc.target/i386/pr25993.c	2014-11-04 00:36:02.945800000 +0300
@@ -1,5 +1,5 @@
 /* { dg-do assemble } */
-/* { dg-skip-if "" { "*-*-darwin*" "*-*-mingw*" } { "*" } { "" } } */
+/* { dg-skip-if "" { "*-*-darwin*" "*-*-mingw*" "*-*-cygwin*" } { "*" } { "" } } */
 /* { dg-options "-std=c99 -x assembler-with-cpp" } */
 
 #ifndef __ASSEMBLER__
