diff -Naur gcc-4.9.1-orig/gcc/gimple-fold.c gcc-4.9.1/gcc/gimple-fold.c
--- gcc-4.9.1-orig/gcc/gimple-fold.c	2014-03-26 12:17:10.000000000 +0300
+++ gcc-4.9.1/gcc/gimple-fold.c	2014-10-29 16:01:44.235000000 +0300
@@ -146,7 +146,8 @@
          The second is important when devirtualization happens during final
          compilation stage when making a new reference no longer makes callee
          to be compiled.  */
-      if (!node || !node->definition || node->global.inlined_to)
+      if (!node || !node->definition
+	  || DECL_EXTERNAL (decl) || node->global.inlined_to)
 	{
 	  gcc_checking_assert (!TREE_ASM_WRITTEN (decl));
 	  return false;
