diff -Naur gcc-6.3.0-orig/gcc/config/i386/cygwin.h gcc-6.3.0/gcc/config/i386/cygwin.h
--- gcc-6.3.0-orig/gcc/config/i386/cygwin.h	2017-01-24 11:59:41.138236300 +0300
+++ gcc-6.3.0/gcc/config/i386/cygwin.h	2017-01-24 12:05:25.522747400 +0300
@@ -56,6 +56,10 @@
     fvtable-verify=std:vtv_end.o%s} \
    crtend.o%s"
 
+/* There is a bug when building i686 dw-2 exceptions
+   where gcc_s gets stripped which this works around */
+#define PREVENT_STRIP_REG_FRAME_INFO "--undefined=___deregister_frame_info --undefined=___register_frame_info"
+
 /* Normally, -lgcc is not needed since everything in it is in the DLL, but we
    want to allow things to be added to it when installing new versions of
    GCC without making a new CYGWIN.DLL, so we leave it.  Profiling is handled
@@ -66,7 +70,7 @@
  %{static|static-libgcc:-lgcc -lgcc_eh} \
  %{!static: \
    %{!static-libgcc: \
-     -lgcc_s -lgcc \
+     -lgcc_s " PREVENT_STRIP_REG_FRAME_INFO " -lgcc \
     } \
   } "
 #else
diff -Naur gcc-6.3.0-orig/gcc/config/i386/mingw32.h gcc-6.3.0/gcc/config/i386/mingw32.h
--- gcc-6.3.0-orig/gcc/config/i386/mingw32.h	2016-04-08 16:47:37.000000000 +0300
+++ gcc-6.3.0/gcc/config/i386/mingw32.h	2017-01-24 12:05:25.538347400 +0300
@@ -120,6 +120,10 @@
   %{shared|mdll: " SUB_LINK_ENTRY " --enable-auto-image-base} \
   %(shared_libgcc_undefs)"
 
+/* There is a bug when building i686 dw-2 exceptions
+   where gcc_s gets stripped which this works around */
+#define PREVENT_STRIP_REG_FRAME_INFO "--undefined=___deregister_frame_info --undefined=___register_frame_info"
+
 /* Include in the mingw32 libraries with libgcc */
 #ifdef ENABLE_SHARED_LIBGCC
 #define SHARED_LIBGCC_SPEC " \
@@ -128,9 +132,9 @@
    %{!static-libgcc: \
      %{!shared: \
        %{!shared-libgcc:-lgcc -lgcc_eh} \
-       %{shared-libgcc:-lgcc_s -lgcc} \
+       %{shared-libgcc:-lgcc_s " PREVENT_STRIP_REG_FRAME_INFO " -lgcc} \
       } \
-     %{shared:-lgcc_s -lgcc} \
+     %{shared:-lgcc_s " PREVENT_STRIP_REG_FRAME_INFO " -lgcc} \
     } \
   } "
 #else
diff -Naur gcc-6.3.0-orig/gcc/config/i386/msys.h gcc-6.3.0/gcc/config/i386/msys.h
--- gcc-6.3.0-orig/gcc/config/i386/msys.h	2017-01-24 12:04:53.694488600 +0300
+++ gcc-6.3.0/gcc/config/i386/msys.h	2017-01-24 12:05:25.538347400 +0300
@@ -63,6 +63,10 @@
 #undef STANDARD_STARTFILE_PREFIX_2
 #define STANDARD_STARTFILE_PREFIX_2 "/usr/lib/w32api/"
 
+/* There is a bug when building i686 dw-2 exceptions
+   where gcc_s gets stripped which this works around */
+#define PREVENT_STRIP_REG_FRAME_INFO "--undefined=___deregister_frame_info --undefined=___register_frame_info"
+
 /* Normally, -lgcc is not needed since everything in it is in the DLL, but we
    want to allow things to be added to it when installing new versions of
    GCC without making a new msys-2.0.dll, so we leave it.  Profiling is handled
@@ -73,7 +77,7 @@
  %{static|static-libgcc:-lgcc -lgcc_eh} \
  %{!static: \
    %{!static-libgcc: \
-     -lgcc_s -lgcc \
+     -lgcc_s " PREVENT_STRIP_REG_FRAME_INFO " -lgcc \
     } \
   } "
 #else
