Index: gcc/ada/s-oscons-tmplt.c
===================================================================
--- gcc/ada/s-oscons-tmplt.c    (revision 202887)
+++ gcc/ada/s-oscons-tmplt.c    (working copy)
@@ -973,7 +973,12 @@

 #endif /* HAVE_TERMIOS */

-#ifdef __MINGW32__
+#ifdef __CYGWIN__
+#define RTS_CONTROL_ENABLE 0x1
+#define DTR_CONTROL_ENABLE 0x1
+#endif
+
+#if defined (__CYGWIN__) || defined (__MINGW32__)
 CNU(DTR_CONTROL_ENABLE, "Enable DTR flow ctrl")
 CNU(RTS_CONTROL_ENABLE, "Enable RTS flow ctrl")
 #endif

