--- origsrc/gcc-4.8.0/gcc/config/i386/winnt.c	2013-03-23 23:59:41.134655300 -0500
+++ src/gcc-4.8.0/gcc/config/i386/winnt.c	2013-03-30 22:43:58.408210500 -0500
@@ -696,6 +696,15 @@ i386_pe_record_stub (const char *name)
   if (!name || *name == 0)
     return;
 
+  p = stub_head;
+  while (p != NULL)
+    {
+      if (p->name[0] == *name
+          && !strcmp (p->name, name))
+	return;
+      p = p->next;
+    }
+
   p = ggc_alloc_stub_list ();
   p->next = stub_head;
   p->name = name;
