--- gcc-4.9.2/libgcc/config/i386/cygming-crtbegin.c.orig	2014-11-02 18:57:00.662422200 +0000
+++ gcc-4.9.2/libgcc/config/i386/cygming-crtbegin.c	2014-11-02 18:58:25.330673700 +0000
@@ -58,7 +58,9 @@
 /* Work around for current cygwin32 build problems (Bug gas/16858).
    Compile weak default functions only for 64-bit systems,
    when absolutely necessary.  */
-#ifdef __x86_64__
+/* On MSYS2, it seems we may need this fix for all arches - or else changing the specs to
+    -lgcc_s --undefined=___deregister_frame_info --undefined=___register_frame_info -lgcc
+#ifdef __x86_64__ */
 TARGET_ATTRIBUTE_WEAK void
 __register_frame_info (__attribute__((unused)) const void *p,
 		       __attribute__((unused)) struct object *o)
@@ -70,7 +72,7 @@
 {
   return (void*) 0;
 }
-#endif
+/*#endif */
 #endif /* DWARF2_UNWIND_INFO */
 
 #if TARGET_USE_JCR_SECTION
