--- origsrc/gcc-4.8.2/CYGWIN-PATCHES/fix-libtool-scripts-for-latest-gcc-runtimes.sh	1970-01-01 07:30:00.000000000 +0730
+++ src/gcc-4.8.2/CYGWIN-PATCHES/fix-libtool-scripts-for-latest-gcc-runtimes.sh	2013-11-08 22:19:30.903592600 +0800
@@ -0,0 +1,16 @@
+libdir=${@:-/usr/lib}
+
+[ ! -d ${libdir} ] && exit 1
+
+extrafiles=$(ls -1 2>/dev/null /usr/bin/libtool \
+      /usr/share/apr/build-1/libtool \
+      /usr/lib/R/bin/libtool /usr/lib/rep/i686-pc-cygwin/libtool)
+
+echo "${extrafiles}" \
+  | xargs -r grep -l '^sys_lib_search_path_spec=.*/usr/lib/gcc/i686-pc-cygwin/[34].[3-9].[0-9]' \
+  | xargs -r sed -i -e "/^sys_lib_search_path_spec=/ s|\(pc-cygwin\)/[34].[3-9].[0-9]|\1/$(/usr/bin/gcc-4 -dumpversion)|g"
+
+find ${libdir} -type f -name '*.la' \
+  | xargs -r grep -l '^dependency_libs=.*/usr/lib/gcc/i686-pc-cygwin/[34].[3-9].[0-9]/lib[a-z_+]*.la' \
+  | xargs -r sed -i -e "/^dependency_libs=/ s|[^ ]*pc-cygwin/[34].[3-9].[0-9]/lib\([a-z_+]*\)\.la|-l\1|g"
+
