--- origsrc/gcc-4.8.2/gcc/ada/s-oscons-tmplt.c	2013-11-08 06:57:30.076795400 +0800
+++ src/gcc-4.8.2/gcc/ada/s-oscons-tmplt.c	2013-11-08 17:45:25.509372700 +0800
@@ -974,8 +974,8 @@ CND(VEOL2, "Alternative EOL")
 #endif /* HAVE_TERMIOS */
 
 #ifdef __CYGWIN__
-#define DTR_CONTROL_DISABLE 0x0
 #define DTR_CONTROL_ENABLE 0x1
+#define RTS_CONTROL_ENABLE 0x1
 #endif
 
 #if defined (__CYGWIN__) || defined (__MINGW32__)
