diff -Naur file-5.38/config.guess file-5.38-msys2/config.guess
--- file-5.38/config.guess	2014-02-13 03:29:38.000000000 +0400
+++ file-5.38-msys2/config.guess	2014-02-16 13:26:11.040000000 +0400
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	 echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	 exit ;;
