From 70c65d2e1841491f59168db1f905e8b14083fb1c Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Tue, 4 Mar 2014 17:42:19 +0000
Subject: [PATCH] off by one in out of bounds calculations (Jan Kaluza)

---
 src/softmagic.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/softmagic.c b/src/softmagic.c
index 170de95..5ed3998 100644
--- a/src/softmagic.c
+++ b/src/softmagic.c
@@ -72,7 +72,7 @@ private int mcopy(struct magic_set *, union VALUETYPE *, int, int,
 private void cvt_32(union VALUETYPE *, const struct magic *);
 private void cvt_64(union VALUETYPE *, const struct magic *);
 
-#define OFFSET_OOB(n, o, i)	((n) < (o) || (i) >= ((n) - (o)))
+#define OFFSET_OOB(n, o, i)	((n) < (o) || (i) > ((n) - (o)))
 /*
  * softmagic - lookup one file in parsed, in-memory copy of database
  * Passed the name and FILE * of one file to be typed.
-- 
1.8.5.5

