diff -Naur file-5.38-orig/compile file-5.38/compile
--- file-5.38-orig/compile	2019-12-17 05:13:46.000000000 +0300
+++ file-5.38/compile	2020-04-18 10:58:46.889418800 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur file-5.38-orig/config.guess file-5.38/config.guess
--- file-5.38-orig/config.guess	2019-12-17 05:13:46.000000000 +0300
+++ file-5.38/config.guess	2020-04-18 10:55:41.924167300 +0300
@@ -911,6 +911,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
