--- file-5.15/magic/Magdir/windows.orig	2013-11-29 09:41:14.357200000 +0400
+++ file-5.15/magic/Magdir/windows	2013-11-29 09:41:21.658000000 +0400
@@ -160,7 +160,7 @@
 
 # Windows *.INF *.INI files updated by Joerg Jenderek at Apr 2013
 # emtpy ,comment , section , unicode line
-0	regex/s		\\`(\r\n|;|[[]|\xFF\xFE)			
+0	regex/s		\\`(\r\n|;|[[]|\\xFF\\xFE)			
 # left bracket in section line
 >&0	search/8192	[						
 # http://en.wikipedia.org/wiki/Autorun.inf
