diff -Naur File-Which-1.09/t/03_simple.t File-Which-1.09/t/03_simple.t
--- File-Which-1.09/t/03_simple.t	2009-09-26 17:29:48.000000000 +0000
+++ File-Which-1.09/t/03_simple.t	2013-06-16 15:26:59.334539100 +0000
@@ -13,7 +13,7 @@
 use constant IS_VMS    => ($^O eq 'VMS');
 use constant IS_MAC    => ($^O eq 'MacOS');
 use constant IS_DOS    => ($^O eq 'MSWin32' or $^O eq 'dos' or $^O eq 'os2');
-use constant IS_CYGWIN => ($^O eq 'cygwin');
+use constant IS_CYGWIN => ($^O eq 'cygwin' || $^O eq 'msys');
 
 # Check that it returns undef if no file is passed
 is(
