diff --unified --recursive --text googletest-release-1.11.0.orig/googlemock/CMakeLists.txt googletest-release-1.11.0/googlemock/CMakeLists.txt
--- googletest-release-1.11.0.orig/googlemock/CMakeLists.txt	2020-11-13 14:33:36.022632459 -0500
+++ googletest-release-1.11.0/googlemock/CMakeLists.txt	2020-11-13 14:35:01.806861990 -0500
@@ -40,6 +40,7 @@
 # Language "C" is required for find_package(Threads).
 cmake_minimum_required(VERSION 3.5)
 cmake_policy(SET CMP0048 NEW)
+  set(GOOGLETEST_VERSION __GOOGLETEST_VERSION__)
 project(gmock VERSION ${GOOGLETEST_VERSION} LANGUAGES CXX C)
 
 if (COMMAND set_up_hermetic_build)
diff --unified --recursive --text googletest-release-1.11.0.orig/googletest/CMakeLists.txt googletest-release-1.11.0/googletest/CMakeLists.txt
--- googletest-release-1.11.0.orig/googletest/CMakeLists.txt	2020-11-13 14:33:36.039299268 -0500
+++ googletest-release-1.11.0/googletest/CMakeLists.txt	2020-11-13 14:37:10.274915300 -0500
@@ -46,6 +46,7 @@
 
 cmake_minimum_required(VERSION 3.5)
 cmake_policy(SET CMP0048 NEW)
+set(GOOGLETEST_VERSION __GOOGLETEST_VERSION__)
 project(gtest VERSION ${GOOGLETEST_VERSION} LANGUAGES CXX C)
 
 if (POLICY CMP0063) # Visibility
