diff -Naur rcs-5.8.1-orig/build-aux/compile rcs-5.8.1/build-aux/compile
--- rcs-5.8.1-orig/build-aux/compile	2012-04-26 19:29:42.000000000 +0000
+++ rcs-5.8.1/build-aux/compile	2013-04-26 19:03:45.291015600 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur rcs-5.8.1-orig/build-aux/config.guess rcs-5.8.1/build-aux/config.guess
--- rcs-5.8.1-orig/build-aux/config.guess	2012-04-26 19:29:42.000000000 +0000
+++ rcs-5.8.1/build-aux/config.guess	2013-04-26 19:03:23.525390600 +0000
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur rcs-5.8.1-orig/build-aux/config.rpath rcs-5.8.1/build-aux/config.rpath
--- rcs-5.8.1-orig/build-aux/config.rpath	2012-05-20 18:07:23.000000000 +0000
+++ rcs-5.8.1/build-aux/config.rpath	2013-04-26 19:02:41.197265600 +0000
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur rcs-5.8.1-orig/configure rcs-5.8.1/configure
--- rcs-5.8.1-orig/configure	2012-06-05 09:46:17.000000000 +0000
+++ rcs-5.8.1/configure	2013-04-26 19:01:56.106445300 +0000
@@ -7549,7 +7549,7 @@
      else
        case "$host_os" in
                                                                osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
@@ -9345,6 +9345,8 @@
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
             cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+                     # Guess yes on MSYS.
+            msys*) gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
@@ -12345,6 +12347,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -12429,6 +12432,7 @@
 
          case "$host_os" in
            cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           msys*)        gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -16959,7 +16963,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -22670,6 +22674,8 @@
              gl_cv_func_dup2_works="guessing no" ;;
            cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
+           msys*) # on MSYS, dup2(1,1) returns 0
+             gl_cv_func_dup2_works="guessing no" ;;
            linux*) # On linux between 2008-07-27 and 2009-05-11, dup2 of a
                    # closed fd may yield -EBADF instead of -1 / errno=EBADF.
              gl_cv_func_dup2_works="guessing no" ;;
diff -Naur rcs-5.8.1-orig/m4/double-slash-root.m4 rcs-5.8.1/m4/double-slash-root.m4
--- rcs-5.8.1-orig/m4/double-slash-root.m4	2012-05-20 18:07:26.000000000 +0000
+++ rcs-5.8.1/m4/double-slash-root.m4	2013-04-26 19:04:37.125976500 +0000
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur rcs-5.8.1-orig/m4/dup2.m4 rcs-5.8.1/m4/dup2.m4
--- rcs-5.8.1-orig/m4/dup2.m4	2012-05-26 09:32:33.000000000 +0000
+++ rcs-5.8.1/m4/dup2.m4	2013-04-26 19:05:09.375976500 +0000
@@ -48,6 +48,8 @@
              gl_cv_func_dup2_works="guessing no" ;;
            cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
+           msys*) # on MSYS, dup2(1,1) returns 0
+             gl_cv_func_dup2_works="guessing no" ;;
            linux*) # On linux between 2008-07-27 and 2009-05-11, dup2 of a
                    # closed fd may yield -EBADF instead of -1 / errno=EBADF.
              gl_cv_func_dup2_works="guessing no" ;;
diff -Naur rcs-5.8.1-orig/m4/getcwd.m4 rcs-5.8.1/m4/getcwd.m4
--- rcs-5.8.1-orig/m4/getcwd.m4	2012-05-26 09:32:34.000000000 +0000
+++ rcs-5.8.1/m4/getcwd.m4	2013-04-26 19:05:43.219726500 +0000
@@ -50,6 +50,8 @@
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
             cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+                     # Guess yes on MSYS.
+            msys*)   gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
diff -Naur rcs-5.8.1-orig/m4/malloc.m4 rcs-5.8.1/m4/malloc.m4
--- rcs-5.8.1-orig/m4/malloc.m4	2012-05-26 09:32:34.000000000 +0000
+++ rcs-5.8.1/m4/malloc.m4	2013-04-26 19:06:04.235351500 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur rcs-5.8.1-orig/m4/printf.m4 rcs-5.8.1/m4/printf.m4
--- rcs-5.8.1-orig/m4/printf.m4	2012-05-20 18:07:27.000000000 +0000
+++ rcs-5.8.1/m4/printf.m4	2013-04-26 19:07:08.157226500 +0000
@@ -736,6 +736,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -813,6 +814,7 @@
 changequote(,)dnl
          case "$host_os" in
            cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           msys*)        gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -1464,6 +1466,8 @@
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
            cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+                                 # Guess yes on MSYS.
+           msys*)                gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Naur rcs-5.8.1-orig/m4/realloc.m4 rcs-5.8.1/m4/realloc.m4
--- rcs-5.8.1-orig/m4/realloc.m4	2012-05-26 09:32:34.000000000 +0000
+++ rcs-5.8.1/m4/realloc.m4	2013-04-26 19:07:34.079101500 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
diff -Naur rcs-5.8.1-orig/m4/threadlib.m4 rcs-5.8.1/m4/threadlib.m4
--- rcs-5.8.1-orig/m4/threadlib.m4	2012-05-20 18:07:27.000000000 +0000
+++ rcs-5.8.1/m4/threadlib.m4	2013-04-26 19:07:51.719726500 +0000
@@ -68,7 +68,7 @@
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
