diff -durN rcs-5.9.4.orig/build-aux/config.guess rcs-5.9.4/build-aux/config.guess
--- rcs-5.9.4.orig/build-aux/config.guess	2017-05-18 14:45:41.762941200 +0800
+++ rcs-5.9.4/build-aux/config.guess	2017-05-18 15:05:51.747178800 +0800
@@ -892,6 +892,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -durN rcs-5.9.4.orig/build-aux/config.rpath rcs-5.9.4/build-aux/config.rpath
--- rcs-5.9.4.orig/build-aux/config.rpath	2017-05-18 14:40:21.477591600 +0800
+++ rcs-5.9.4/build-aux/config.rpath	2017-05-18 15:06:52.482331200 +0800
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -durN rcs-5.9.4.orig/configure rcs-5.9.4/configure
--- rcs-5.9.4.orig/configure	2017-05-18 14:45:24.278342400 +0800
+++ rcs-5.9.4/configure	2017-05-18 15:10:28.391276300 +0800
@@ -7721,7 +7721,7 @@
      else
        case "$host_os" in
                                                                osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
@@ -8558,7 +8558,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -9613,6 +9613,8 @@
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
             cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+                     # Guess yes on msys.
+            msys*)   gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
@@ -12577,6 +12579,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+		   msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -12661,6 +12664,7 @@
 
          case "$host_os" in
            cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+		   msys*)        gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -17508,7 +17512,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
@@ -18131,7 +18135,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -18244,7 +18248,7 @@
 else
   if test "$cross_compiling" = yes; then :
   case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac
@@ -24194,6 +24198,8 @@
              gl_cv_func_dup2_works="guessing no" ;;
            cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
+           msys*) # on MSYS, dup2(1,1) returns 0
+             gl_cv_func_dup2_works="guessing no" ;;
            aix* | freebsd*)
                    # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
                    # not EBADF.
@@ -25253,7 +25259,7 @@
 else
   if test "$cross_compiling" = yes; then :
   case "$host_os" in
-          cygwin*) # on cygwin 1.5.25, getdtablesize() automatically grows
+          cygwin* | msys*) # on cygwin 1.5.25, getdtablesize() automatically grows
             gl_cv_func_getdtablesize_works="guessing no" ;;
           *) gl_cv_func_getdtablesize_works="guessing yes" ;;
          esac
diff -durN rcs-5.9.4.orig/m4/double-slash-root.m4 rcs-5.9.4/m4/double-slash-root.m4
--- rcs-5.9.4.orig/m4/double-slash-root.m4	2017-05-18 14:41:20.868976800 +0800
+++ rcs-5.9.4/m4/double-slash-root.m4	2017-05-18 15:11:51.626184000 +0800
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -durN rcs-5.9.4.orig/m4/dup2.m4 rcs-5.9.4/m4/dup2.m4
--- rcs-5.9.4.orig/m4/dup2.m4	2017-05-18 14:41:21.447109200 +0800
+++ rcs-5.9.4/m4/dup2.m4	2017-05-18 15:12:29.860803700 +0800
@@ -80,6 +80,8 @@
              gl_cv_func_dup2_works="guessing no" ;;
            cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
+           msys*) # on MSYS, dup2(1,1) returns 0
+             gl_cv_func_dup2_works="guessing no" ;;
            aix* | freebsd*)
                    # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
                    # not EBADF.
diff -durN rcs-5.9.4.orig/m4/fcntl.m4 rcs-5.9.4/m4/fcntl.m4
--- rcs-5.9.4.orig/m4/fcntl.m4	2017-05-18 14:41:24.603399600 +0800
+++ rcs-5.9.4/m4/fcntl.m4	2017-05-18 15:13:03.486018900 +0800
@@ -69,7 +69,7 @@
          [gl_cv_func_fcntl_f_dupfd_works=yes],
          [gl_cv_func_fcntl_f_dupfd_works=no],
          [case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac])])
diff -durN rcs-5.9.4.orig/m4/getcwd.m4 rcs-5.9.4/m4/getcwd.m4
--- rcs-5.9.4.orig/m4/getcwd.m4	2017-05-18 14:41:28.759702800 +0800
+++ rcs-5.9.4/m4/getcwd.m4	2017-05-18 15:13:58.517621100 +0800
@@ -52,6 +52,8 @@
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
             cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+                     # Guess yes on msys.
+            msys*)   gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
diff -durN rcs-5.9.4.orig/m4/getdtablesize.m4 rcs-5.9.4/m4/getdtablesize.m4
--- rcs-5.9.4.orig/m4/getdtablesize.m4	2017-05-18 14:41:28.994080800 +0800
+++ rcs-5.9.4/m4/getdtablesize.m4	2017-05-18 15:14:30.267824300 +0800
@@ -28,7 +28,7 @@
         [gl_cv_func_getdtablesize_works=yes],
         [gl_cv_func_getdtablesize_works=no],
         [case "$host_os" in
-          cygwin*) # on cygwin 1.5.25, getdtablesize() automatically grows
+          cygwin* | msys*) # on cygwin 1.5.25, getdtablesize() automatically grows
             gl_cv_func_getdtablesize_works="guessing no" ;;
           *) gl_cv_func_getdtablesize_works="guessing yes" ;;
          esac])
diff -durN rcs-5.9.4.orig/m4/malloc.m4 rcs-5.9.4/m4/malloc.m4
--- rcs-5.9.4.orig/m4/malloc.m4	2017-05-18 14:41:34.447275600 +0800
+++ rcs-5.9.4/m4/malloc.m4	2017-05-18 15:14:50.189826800 +0800
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -durN rcs-5.9.4.orig/m4/printf.m4 rcs-5.9.4/m4/printf.m4
--- rcs-5.9.4.orig/m4/printf.m4	2017-05-18 14:41:41.275488000 +0800
+++ rcs-5.9.4/m4/printf.m4	2017-05-18 15:15:23.361914100 +0800
@@ -730,6 +730,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+		   msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
diff -durN rcs-5.9.4.orig/m4/realloc.m4 rcs-5.9.4/m4/realloc.m4
--- rcs-5.9.4.orig/m4/realloc.m4	2017-05-18 14:41:42.447378000 +0800
+++ rcs-5.9.4/m4/realloc.m4	2017-05-18 15:15:44.658925400 +0800
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
diff -durN rcs-5.9.4.orig/m4/threadlib.m4 rcs-5.9.4/m4/threadlib.m4
--- rcs-5.9.4.orig/m4/threadlib.m4	2017-05-18 14:41:51.259990800 +0800
+++ rcs-5.9.4/m4/threadlib.m4	2017-05-18 15:16:35.190498800 +0800
@@ -68,7 +68,7 @@
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
diff -durN rcs-5.9.4.orig/tests/known-failures.in rcs-5.9.4/tests/known-failures.in
--- rcs-5.9.4.orig/tests/known-failures.in	2017-05-18 14:33:03.206356800 +0800
+++ rcs-5.9.4/tests/known-failures.in	2017-05-18 15:17:51.925364900 +0800
@@ -62,7 +62,7 @@
 # preconditions or filters.  Ugly and possibly buggy in itself!
 more=
 case '@host_os@' in
-    cygwin | darwin)
+    cygwin | msys | darwin)
         case $1 in
             5.8 | 5.8.* | 5.9.[012])
                 more=t805
