diff -Naur readline-6.1/configure readline-6.1-p/configure
--- readline-6.1/configure	2009-12-30 01:33:49.000000000 +0300
+++ readline-6.1-p/configure	2013-02-13 18:33:36.087890600 +0400
@@ -2028,7 +2028,7 @@
 CROSS_COMPILE=
 if test "x$cross_compiling" = "xyes"; then
     case "${host}" in
-    *-cygwin*)
+    *-cygwin* | *-msys*)
         cross_cache=${srcdir}/cross-build/cygwin.cache
         ;;
     *-mingw*)
diff -Naur readline-6.1/configure.in readline-6.1-p/configure.in
--- readline-6.1/configure.in	2009-10-12 18:12:15.000000000 +0400
+++ readline-6.1-p/configure.in	2013-02-13 18:34:11.056640600 +0400
@@ -76,7 +76,7 @@
 CROSS_COMPILE=
 if test "x$cross_compiling" = "xyes"; then
     case "${host}" in
-    *-cygwin*)
+    *-cygwin* | *-msys*)
         cross_cache=${srcdir}/cross-build/cygwin.cache
         ;;
     *-mingw*)
diff -Naur readline-6.1/support/config.guess readline-6.1-p/support/config.guess
--- readline-6.1/support/config.guess	2009-01-04 22:41:33.000000000 +0300
+++ readline-6.1-p/support/config.guess	2013-02-13 18:36:03.711914100 +0400
@@ -781,6 +781,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
@@ -818,6 +821,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur readline-6.1/support/config.rpath readline-6.1-p/support/config.rpath
--- readline-6.1/support/config.rpath	2008-08-20 17:21:09.000000000 +0400
+++ readline-6.1-p/support/config.rpath	2013-02-13 18:37:28.040039100 +0400
@@ -108,7 +108,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -148,7 +148,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -267,7 +267,7 @@
       ;;
     bsdi4*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -439,7 +439,7 @@
     ;;
   bsdi4*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
diff -Naur readline-6.1/support/config.sub readline-6.1-p/support/config.sub
--- readline-6.1/support/config.sub	2009-11-19 21:43:37.000000000 +0300
+++ readline-6.1-p/support/config.sub	2013-02-13 18:37:55.071289100 +0400
@@ -1256,7 +1256,7 @@
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* \
-	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
+	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
diff -Naur readline-6.1/support/shlib-install readline-6.1-p/support/shlib-install
--- readline-6.1/support/shlib-install	2009-10-28 16:30:18.000000000 +0300
+++ readline-6.1-p/support/shlib-install	2013-02-13 18:40:57.883789100 +0400
@@ -74,9 +74,9 @@
 	if [ -z "$uninstall" ]; then
 		chmod 555 ${INSTALLDIR}/${LIBNAME}
 	fi ;;
-cygwin*|mingw*)
+cygwin*|msys*|mingw*)
 	IMPLIBNAME=`echo ${LIBNAME} \
-		| sed -e 's,^cyg,lib,' -e 's,[0-9]*.dll$,.dll.a,'`
+		| sed -e 's,^msys-,lib,' -e 's,[0-9]*.dll$,.dll.a,'`
 	if [ -z "$uninstall" ]; then
 		${echo} $RM ${BINDIR}/${LIBNAME}.${OLDSUFF}
 		if [ -f "$BINDIR/$LIBNAME" ]; then
@@ -193,7 +193,7 @@
 	fi
 	;;
 
-cygwin*|mingw*)
+cygwin*|msys*|mingw*)
 	# Links to .dlls don't work.  Hence shobj-conf used DLLVERSION.dll
 	# instead of so.SHLIB_MAJOR.SHLIB_MINOR.  The postinstall above
 	# took care of everything else.
diff -Naur readline-6.1/support/shobj-conf readline-6.1-p/support/shobj-conf
--- readline-6.1/support/shobj-conf	2009-10-28 16:20:21.000000000 +0300
+++ readline-6.1-p/support/shobj-conf	2013-02-13 18:39:31.899414100 +0400
@@ -520,6 +520,24 @@
 	fi
 	;;
 
+msys*)
+	SHOBJ_LD='$(CC)'
+	SHOBJ_LDFLAGS='-shared -Wl,--enable-auto-import -Wl,--enable-auto-image-base -Wl,--export-all -Wl,--out-implib=$(@).a'
+	SHLIB_LIBPREF='msys-'
+	SHLIB_LIBSUFF='dll'
+	SHLIB_LIBVERSION='$(SHLIB_DLLVERSION).$(SHLIB_LIBSUFF)'
+	SHLIB_LIBS='$(TERMCAP_LIB)'
+
+	SHLIB_DOT=
+	# For official cygwin releases, DLLVERSION will be defined in the
+	# environment of configure, and will be incremented any time the API
+	# changes in a non-backwards compatible manner.  Otherwise, it is just
+	# SHLIB_MAJOR.
+	if [ -n "$DLLVERSION" ] ; then
+		SHLIB_DLLVERSION="$DLLVERSION"
+	fi
+	;;
+
 mingw*)
 	SHOBJ_LD='$(CC)'
 	SHOBJ_LDFLAGS='-shared -Wl,--enable-auto-import -Wl,--enable-auto-image-base -Wl,--export-all -Wl,--out-implib=$(@).a'
