--- origsrc/readline-7.0/Makefile.in	2015-08-26 09:15:59.000000000 -0500
+++ src/readline-7.0/Makefile.in	2017-02-11 16:38:31.337560100 -0600
@@ -313,7 +313,8 @@ install-dvi:
 install-html:
 install-pdf:
 install-ps:
-check:
+check: force
+	-( cd examples && $(MAKE) $(MFLAGS) check )
 installcheck:
 
 dist:   force
--- origsrc/readline-7.0/examples/Makefile.in	2016-04-26 08:50:18.000000000 -0500
+++ src/readline-7.0/examples/Makefile.in	2017-02-11 16:38:31.337560100 -0600
@@ -54,7 +54,7 @@ CPPFLAGS = @CPPFLAGS@
 
 CCFLAGS  = $(ASAN_CFLAGS) $(DEFS) $(LOCAL_CFLAGS) $(INCLUDES) $(CPPFLAGS) \
 	$(CFLAGS)
-LDFLAGS = -g -L.. @LDFLAGS@ $(ASAN_LDFLAGS)
+LDFLAGS = -g -L../shlib -L.. @LDFLAGS@ $(ASAN_LDFLAGS)
 
 ASAN_XCFLAGS = -fsanitize=address -fno-omit-frame-pointer
 ASAN_XLDFLAGS = -fsanitize=address
@@ -84,7 +84,7 @@ all: $(EXECUTABLES)
 everything: all
 
 check:	rlversion$(EXEEXT)
-	@echo Readline version: `rlversion$(EXEEXT)`
+	@echo Readline version: `PATH=../shlib:$$PATH ./rlversion$(EXEEXT)`
 
 installdirs:
 	-$(SHELL) $(top_srcdir)/support/mkdirs $(DESTDIR)$(installdir)
