diff -rwu orig/funmap.c readline-7.0/funmap.c
--- orig/funmap.c	2018-03-01 18:12:03.726007400 +0900
+++ readline-7.0/funmap.c	2018-03-01 18:12:09.403227100 +0900
@@ -116,7 +116,7 @@
   { "non-incremental-reverse-search-history-again", rl_noninc_reverse_search_again },
   { "old-menu-complete", rl_old_menu_complete },
   { "overwrite-mode", rl_overwrite_mode },
-#if defined (_WIN32)
+#if defined (_WIN32) || defined (__CYGWIN__)
   { "paste-from-clipboard", rl_paste_from_clipboard },
 #endif
   { "possible-completions", rl_possible_completions },
diff -rwu orig/kill.c readline-7.0/kill.c
--- orig/kill.c	2018-03-01 18:12:03.813767700 +0900
+++ readline-7.0/kill.c	2018-03-01 18:12:12.417899000 +0900
@@ -741,7 +741,7 @@
 }
 
 /* A special paste command for Windows users. */
-#if defined (_WIN32)
+#if defined (_WIN32) || defined (__CYGWIN__)
 #include <windows.h>
 
 static char*
diff -rwu orig/readline.h readline-7.0/readline.h
--- orig/readline.h	2018-03-01 18:12:03.932404000 +0900
+++ readline-7.0/readline.h	2018-03-01 18:12:08.364068200 +0900
@@ -174,7 +174,7 @@
 extern int rl_yank_last_arg PARAMS((int, int));
 extern int rl_bracketed_paste_begin PARAMS((int, int));
 /* Not available unless _WIN32 is defined. */
-#if defined (_WIN32)
+#if defined (_WIN32) || defined (__CYGWIN__)
 extern int rl_paste_from_clipboard PARAMS((int, int));
 #endif
 
