diff --git a/support/shobj-conf b/support/shobj-conf
index c885950..443034e 100644
--- a/support/shobj-conf
+++ b/support/shobj-conf
@@ -477,6 +477,7 @@ msdos*)
 	;;
 
 cygwin*)
+	SHOBJ_CFLAGS=-DSHARED_READLINE
 	SHOBJ_LD='$(CC)'
 	SHOBJ_LDFLAGS='-shared -Wl,--enable-auto-import -Wl,--enable-auto-image-base -Wl,--export-all -Wl,--out-implib=$(@).a'
 	SHLIB_LIBPREF='msys-'
diff --git a/tcap.h b/tcap.h
index 859e6ee..f0c79fc 100644
--- a/tcap.h
+++ b/tcap.h
@@ -26,6 +26,10 @@
 #  include "config.h"
 #endif
 
+#if !defined (SHARED_READLINE)
+#  define NCURSES_STATIC
+#endif
+
 #if defined (HAVE_TERMCAP_H)
 #  if defined (__linux__) && !defined (SPEED_T_IN_SYS_TYPES)
 #    include "rltty.h"
