--- binutils-2.36.1/ld/emultempl/pep.em.orig	2021-05-10 08:58:40.005554200 +0200
+++ binutils-2.36.1/ld/emultempl/pep.em	2021-05-10 09:01:58.002803000 +0200
@@ -117,7 +117,7 @@
 					: 0x1C0000000LL))
 #undef NT_DLL_AUTO_IMAGE_MASK
 #define NT_DLL_AUTO_IMAGE_MASK \
-  ((bfd_vma) (${move_default_addr_high} ? 0x1ffff0000LL \
+  ((bfd_vma) (${move_default_addr_high} ? 0x7fffff0000LL \
 					: 0x1ffff0000LL))
 #else
 #undef  NT_EXE_IMAGE_BASE
