--- configure.ac.orig	2017-01-28 19:26:38.253796500 +0100
+++ configure.ac	2017-01-28 19:53:12.562170100 +0100
@@ -8,7 +8,7 @@
 AC_PROG_INSTALL
 
 case "${target}" in
-  *-cygwin)
+  *-msys)
   	;;
   *)
   	echo "The only supported target is cygwin."
@@ -16,7 +16,7 @@
 	;;
 esac
 
-AC_CHECK_TOOL(CC, ${target_cpu}-pc-cygwin-gcc, ${target_cpu}-pc-cygwin-gcc)
+AC_CHECK_TOOL(CC, ${target_cpu}-pc-msys-gcc, ${target_cpu}-pc-msys-gcc)
 if test -z "$CC"; then
   AC_CHECK_TOOL(CC, gcc, gcc)
   if test -z "$CC"; then
