diff -Naur libxslt-1.1.28-orig/config.guess libxslt-1.1.28/config.guess
--- libxslt-1.1.28-orig/config.guess	2008-01-30 20:27:02.000000000 +0000
+++ libxslt-1.1.28/config.guess	2013-04-29 06:32:13.080710900 +0000
@@ -781,6 +781,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
@@ -815,6 +818,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur libxslt-1.1.28-orig/config.sub libxslt-1.1.28/config.sub
--- libxslt-1.1.28-orig/config.sub	2008-01-30 20:27:02.000000000 +0000
+++ libxslt-1.1.28/config.sub	2013-04-29 06:31:39.627585900 +0000
@@ -1217,7 +1217,7 @@
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* \
-	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
+	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
diff -Naur libxslt-1.1.28-orig/configure.in libxslt-1.1.28/configure.in
--- libxslt-1.1.28-orig/configure.in	2012-11-21 07:10:58.000000000 +0000
+++ libxslt-1.1.28/configure.in	2013-04-29 06:36:40.971335900 +0000
@@ -622,7 +622,7 @@
 WIN32_EXTRA_LIBADD=
 WIN32_EXTRA_LDFLAGS=
 case "$host" in
- *-*-cygwin*)
+ *-*-cygwin* | *-*-msys*)
  WIN32_EXTRA_LDFLAGS="-no-undefined"
  ;;
  *-*-mingw*)
