diff -durN libxslt-1.1.32.orig/config.guess libxslt-1.1.32/config.guess
--- libxslt-1.1.32.orig/config.guess	2017-05-11 20:29:02.515625000 +0800
+++ libxslt-1.1.32/config.guess	2017-05-11 20:48:19.640625000 +0800
@@ -821,6 +821,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW64*:*)
 	echo ${UNAME_MACHINE}-pc-mingw64
 	exit ;;
@@ -867,6 +870,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
diff -durN libxslt-1.1.29.orig/configure.ac libxslt-1.1.29/configure.ac
--- libxslt-1.1.32.orig/configure.ac	2017-05-11 20:29:03.953125000 +0800
+++ libxslt-1.1.32/configure.ac	2017-05-11 20:50:04.625000000 +0800
@@ -623,7 +623,7 @@
 AC_SUBST(LIBXSLT_DEFAULT_PLUGINS_PATH)
 
 case "$host" in
- *-*-cygwin*|*-*-mingw*)
+ *-*-cygwin*|*-*-mingw*|*-*-msys*)
  LDFLAGS="$LDFLAGS -no-undefined"
  ;;
 esac
