--- libxslt-1.1.33/libexslt/exsltexports.h.orig	2019-02-07 11:57:33.833426800 +0300
+++ libxslt-1.1.33/libexslt/exsltexports.h	2019-02-07 11:57:41.851840900 +0300
@@ -128,6 +128,9 @@
     #endif
   #endif
   #define EXSLTCALL __cdecl
+  #if !defined _REENTRANT
+    #define _REENTRANT
+  #endif
 #endif
 
 /* Compatibility */
--- libxslt-1.1.33/libxslt/xsltexports.h.orig	2019-02-07 11:57:55.080664100 +0300
+++ libxslt-1.1.33/libxslt/xsltexports.h	2019-02-07 11:58:02.833877800 +0300
@@ -130,6 +130,9 @@
     #endif
   #endif
   #define XSLTCALL __cdecl
+  #if !defined _REENTRANT
+    #define _REENTRANT
+  #endif
 #endif
 
 /* Compatibility */
