diff -Naur mercurial-5.2-orig/mercurial/posix.py mercurial-5.2/mercurial/posix.py
--- mercurial-5.2-orig/mercurial/posix.py	2013-05-14 23:43:40.000000000 +0000
+++ mercurial-5.2/mercurial/posix.py	2013-05-16 09:11:10.796875000 +0000
@@ -468,7 +468,7 @@
         return encoding.hfsignoreclean(enc)
 
 
-if pycompat.sysplatform == b'cygwin':
+if pycompat.sysplatform in [b'cygwin', b'msys']:
     # workaround for cygwin, in which mount point part of path is
     # treated as case sensitive, even though underlying NTFS is case
     # insensitive.
