From 6599ac30d78850ebe092592885dcfb75e112d53b Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Tue, 4 Aug 2015 18:27:23 +0100
Subject: [PATCH] Driver: add support to mingw64 for msys2 host


diff --git a/lib/Driver/MinGWToolChain.cpp b/lib/Driver/MinGWToolChain.cpp
index 6de2c50..6c711de 100644
--- a/lib/Driver/MinGWToolChain.cpp
+++ b/lib/Driver/MinGWToolChain.cpp
@@ -68,7 +68,10 @@ MinGW::MinGW(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
 
 // In Windows there aren't any standard install locations, we search
 // for gcc on the PATH. In Linux the base is always /usr.
-#ifdef LLVM_ON_WIN32
+// In Msys2 the base is always /opt
+#ifdef __MSYS__
+    Base = "/opt";
+#elif defined(LLVM_ON_WIN32)
   if (getDriver().SysRoot.size())
     Base = getDriver().SysRoot;
   else if (llvm::ErrorOr<std::string> GPPName =
-- 
2.4.5

