From a1d23fe12949bd41e664b61d1bd34fb9dc630a69 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Mon, 30 Nov 2015 12:30:43 -0800
Subject: [PATCH] set up args for lld and work around library with path bug


diff --git a/lib/Driver/ToolChains.h b/lib/Driver/ToolChains.h
index c836ac3..db390c1 100644
--- a/lib/Driver/ToolChains.h
+++ b/lib/Driver/ToolChains.h
@@ -649,6 +649,10 @@ public:
       const llvm::opt::ArgList &DriverArgs,
       llvm::opt::ArgStringList &CC1Args) const override;
 
+  RuntimeLibType GetDefaultRuntimeLibType() const override {
+    return ToolChain::RLT_CompilerRT;
+  }
+
 protected:
   Tool *getTool(Action::ActionClass AC) const override;
   Tool *buildLinker() const override;
diff --git a/lib/Driver/Tools.cpp b/lib/Driver/Tools.cpp
index 9ae9348..87eb676 100644
--- a/lib/Driver/Tools.cpp
+++ b/lib/Driver/Tools.cpp
@@ -9456,7 +9456,11 @@ void MinGW::Linker::AddLibGCC(const ArgList &Args,
       CmdArgs.push_back("-lgcc");
     }
   } else {
-    AddRunTimeLibs(getToolChain(), getToolChain().getDriver(), CmdArgs, Args);
+    CmdArgs.push_back(
+      Args.MakeArgString("-L" +
+        Twine(getToolChain().getDriver().ResourceDir.c_str()) + "/lib/windows"));
+    CmdArgs.push_back(
+      Args.MakeArgString("-lclang_rt.builtins-" + getToolChain().getArchName()));
   }
 
   CmdArgs.push_back("-lmoldname");
@@ -9483,7 +9487,7 @@ void MinGW::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   // handled somewhere else.
   Args.ClaimAllArgs(options::OPT_w);
 
-  StringRef LinkerName = Args.getLastArgValue(options::OPT_fuse_ld_EQ, "ld");
+  StringRef LinkerName = Args.getLastArgValue(options::OPT_fuse_ld_EQ, "lld");
   if (LinkerName.equals_lower("lld")) {
     CmdArgs.push_back("-flavor");
     CmdArgs.push_back("gnu");
-- 
2.6.3

