From 33b0c456ae8fe009ae1989d2055d6f0c9212a244 Mon Sep 17 00:00:00 2001
From: Martell Malone <martellmalone@gmail.com>
Date: Wed, 28 Oct 2015 19:32:16 +0000
Subject: [PATCH] add support for gnu in coff


diff --git a/COFF/Options.td b/COFF/Options.td
index 2402b80..02e5d94 100644
--- a/COFF/Options.td
+++ b/COFF/Options.td
@@ -89,6 +89,21 @@ def help_q : Flag<["/?", "-?"], "">, Alias<help>;
 // Flags for debugging
 def lldmap : Joined<["/", "-"], "lldmap:">;
 
+
+def output : Separate<["-"], "o">, Alias<out>;
+def L : Joined<["-"], "L">, Alias<libpath>;
+def e : Separate<["-"], "e">, Alias<entry>;
+def subs : Separate<["--"], "subsystem">, Alias<subsystem>;
+
+
+// Used to override gnu pe targets
+def m : Separate<["-"], "m">, MetaVarName<"<emulation>">,
+     HelpText<"Select target emulation">;
+
+// Select an optional flavor
+def flavor: Separate<["-"], "flavor">,
+     HelpText<"Flavor for linking, options are gnu/darwin/link">;
+
 //==============================================================================
 // The flags below do nothing. They are defined only for link.exe compatibility.
 //==============================================================================
diff --git a/lib/Driver/UniversalDriver.cpp b/lib/Driver/UniversalDriver.cpp
index 45bcbc8..0adc9fd 100644
--- a/lib/Driver/UniversalDriver.cpp
+++ b/lib/Driver/UniversalDriver.cpp
@@ -94,6 +94,21 @@ static Flavor strToFlavor(StringRef str) {
       .Default(Flavor::invalid);
 }
 
+//std::pair<llvm::Triple::ObjectFileFormat, IMAGE_FILE_MACHINE_TYPE> getTargetFromEmulation
+
+static bool isPETarget(const llvm::opt::InputArgList &parsedArgs) {
+  llvm::opt::Arg *argMachine = parsedArgs.getLastArg(OPT_m);
+  if (!argMachine)
+    return false;
+  if (argMachine->containsValue("i386pe"))
+    return true;
+  if (argMachine->containsValue("i386pep"))
+    return true;
+  if (argMachine->containsValue("thumb2pe"))
+    return true;
+  return false;
+}
+
 static ProgramNameParts parseProgramName(StringRef programName) {
   SmallVector<StringRef, 3> components;
   llvm::SplitString(programName, components, "-");
@@ -195,6 +210,12 @@ bool UniversalDriver::link(llvm::MutableArrayRef<const char *> args,
     return true;
   }
 
+  //Override gnu for pe targets to use coff
+  if(isPETarget(parsedArgs)) {
+    coff::link(args);
+    return true;
+  }
+
   Flavor flavor = getFlavor(args, parsedArgs);
 
   // Switch to appropriate driver.
diff --git a/lib/Driver/UniversalDriverOptions.td b/lib/Driver/UniversalDriverOptions.td
index 14abc9c..38e620b 100644
--- a/lib/Driver/UniversalDriverOptions.td
+++ b/lib/Driver/UniversalDriverOptions.td
@@ -11,6 +11,10 @@ def core : Flag<["-"], "core">,
 def target: Separate<["-"], "target">,
      HelpText<"Select the target">;
 
+// Used to override gnu pe targets
+def m : Separate<["-"], "m">, MetaVarName<"<emulation>">,
+     HelpText<"Select target emulation">;
+
 def version: Flag<["-"], "version">,
      HelpText<"Display the version">;
 
-- 
2.6.2

