diff -durN gnutls-3.5.11.orig/gl/m4/malloc.m4 gnutls-3.5.11/gl/m4/malloc.m4
--- gnutls-3.5.11.orig/gl/m4/malloc.m4	2016-05-28 15:57:01.000000000 +0800
+++ gnutls-3.5.11/gl/m4/malloc.m4	2017-05-10 15:44:51.140625000 +0800
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -durN gnutls-3.5.11.orig/gl/m4/printf.m4 gnutls-3.5.11/gl/m4/printf.m4
--- gnutls-3.5.11.orig/gl/m4/printf.m4	2016-12-19 22:10:59.000000000 +0800
+++ gnutls-3.5.11/gl/m4/printf.m4	2017-05-10 15:46:08.312500000 +0800
@@ -722,7 +722,7 @@
            openbsd*)        gl_cv_func_printf_directive_ls="guessing no";;
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
-           cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           cygwin* | msys*)         gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -799,7 +799,7 @@
         [
 changequote(,)dnl
          case "$host_os" in
-           cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           cygwin* | msys*)      gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -1451,7 +1451,7 @@
            darwin[1-6].*)        gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
-           cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           cygwin* | msys*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -durN gnutls-3.5.11.orig/gl/m4/realloc.m4 gnutls-3.5.11/gl/m4/realloc.m4
--- gnutls-3.5.11.orig/gl/m4/realloc.m4	2016-05-28 15:57:01.000000000 +0800
+++ gnutls-3.5.11/gl/m4/realloc.m4	2017-05-10 15:47:17.500000000 +0800
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
