diff -Naur gnutls-3.4.0-orig/gl/m4/malloc.m4 gnutls-3.4.0/gl/m4/malloc.m4
--- gnutls-3.4.0-orig/gl/m4/malloc.m4	2015-01-19 20:38:09.000000000 +0300
+++ gnutls-3.4.0/gl/m4/malloc.m4	2015-05-01 22:03:57.665000000 +0300
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur gnutls-3.4.0-orig/gl/m4/printf.m4 gnutls-3.4.0/gl/m4/printf.m4
--- gnutls-3.4.0-orig/gl/m4/printf.m4	2015-03-23 21:21:01.000000000 +0300
+++ gnutls-3.4.0/gl/m4/printf.m4	2015-05-01 22:04:32.999000000 +0300
@@ -720,7 +720,7 @@
            openbsd*)        gl_cv_func_printf_directive_ls="guessing no";;
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
-           cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           cygwin* | msys*)         gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -797,7 +797,7 @@
         [
 changequote(,)dnl
          case "$host_os" in
-           cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           cygwin* | msys*)      gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -1449,7 +1449,7 @@
            darwin[1-6].*)        gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
-           cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           cygwin* | msys*)      gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Naur gnutls-3.4.0-orig/gl/m4/realloc.m4 gnutls-3.4.0/gl/m4/realloc.m4
--- gnutls-3.4.0-orig/gl/m4/realloc.m4	2015-01-19 20:38:09.000000000 +0300
+++ gnutls-3.4.0/gl/m4/realloc.m4	2015-05-01 22:11:16.368200000 +0300
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
