diff -Naur pkg-config-0.28/config.guess pkg-config-0.28-msys/config.guess
--- pkg-config-0.28/config.guess	2012-10-01 13:15:51.000000000 +0000
+++ pkg-config-0.28-msys/config.guess	2013-04-15 18:56:46.103515600 +0000
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
