--- a/absl/base/internal/sysinfo.cc	2026-08-18 14:53:30.000000000 +0200
+++ b/absl/base/internal/sysinfo.cc	2026-09-29 01:06:39.472086400 +0200
@@ -478,7 +478,7 @@
 pid_t GetTID() {
   // `pthread_t` need not be arithmetic per POSIX; platforms where it isn't
   // should be handled above.
-  return static_cast<pid_t>(pthread_self());
+  return static_cast<pid_t>(reinterpret_cast<uintptr_t>(pthread_self()));
 }
 
 #endif
