diff -Naur crosstool-ng.orig/configure.ac crosstool-ng/configure.ac
--- crosstool-ng.orig/configure.ac
+++ crosstool-ng/configure.ac
@@ -240,7 +240,7 @@ AC_CACHE_CHECK([whether stat takes GNU or BSD format],
      rm -f conftest
      AS_IF([test "$attr_bsd" = "642"],
        [acx_cv_stat_flavor=BSD],
-       [test "$attr_gnu" = "642"],
+       [test "$attr_gnu" = "644"],
        [acx_cv_stat_flavor=GNU],
        [AC_MSG_ERROR([cannot determine stat(1) format option])])])
