diff -aurN 0001/breezy/transport/http/_urllib2_wrappers.py 0002/breezy/transport/http/_urllib2_wrappers.py
--- 0001/breezy/transport/http/_urllib2_wrappers.py	2016-04-15 17:08:16.998146000 -0300
+++ 0002/breezy/transport/http/_urllib2_wrappers.py	2016-04-15 17:12:58.067000000 -0300
@@ -88,6 +88,7 @@
 # Note for packagers: if there is no package providing certs for your platform,
 # the curl project produces http://curl.haxx.se/ca/cacert.pem weekly.
 _ssl_ca_certs_known_locations = [
+    u'/usr/ssl/certs/ca-bundle.crt', # MSYS2
     u'/etc/ssl/certs/ca-certificates.crt',  # Ubuntu/debian/gentoo
     u'/etc/pki/tls/certs/ca-bundle.crt',  # Fedora/CentOS/RH
     u'/etc/ssl/ca-bundle.pem',  # OpenSuse
