From 6cf7f5b440e7f1196b4a89e5de8ee1ca1eb083ee Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Thu, 22 Jan 2026 05:22:54 +0800
Subject: [PATCH] add msys2 certs location

---
 breezy/transport/http/__init__.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/breezy/transport/http/__init__.py b/breezy/transport/http/__init__.py
index ba5301e..88a8cf7 100644
--- a/breezy/transport/http/__init__.py
+++ b/breezy/transport/http/__init__.py
@@ -36,6 +36,7 @@ def default_user_agent():
 # Note for packagers: if there is no package providing certs for your platform,
 # the curl project produces http://curl.haxx.se/ca/cacert.pem weekly.
 _ssl_ca_certs_known_locations = [
+    "/usr/ssl/certs/ca-bundle.crt", # MSYS2
     "/etc/ssl/certs/ca-certificates.crt",  # Ubuntu/debian/gentoo
     "/etc/pki/tls/certs/ca-bundle.crt",  # Fedora/CentOS/RH
     "/etc/ssl/ca-bundle.pem",  # OpenSuse
-- 
2.52.0.windows.1

