From d6819221d58742bfc1c57397db716820b4d8f46a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 30 Jun 2015 18:48:00 +0100
Subject: [PATCH] msys2ize

---
 configure.host | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.host b/configure.host
index 4062a48..9985b3e 100644
--- a/configure.host
+++ b/configure.host
@@ -64,7 +64,7 @@ case "${host}" in
 	TARGET=X86_FREEBSD; TARGETDIR=x86
 	;;
 
-  i?86-*-cygwin* | i?86-*-mingw* | i?86-*-os2* | i?86-*-interix* \
+  i?86-*-cygwin* | i?86-*-msys* | i?86-*-mingw* | i?86-*-os2* | i?86-*-interix* \
   | x86_64-*-cygwin* | x86_64-*-mingw* | x86_64-*-win* )
 	TARGETDIR=x86
 	if test $ac_cv_sizeof_size_t = 4; then
@@ -81,7 +81,7 @@ case "${host}" in
 	    UNSUPPORTED=1
           fi
 	fi
-	# All mingw/cygwin/win32 builds require -no-undefined for sharedlib.
+	# All mingw/cygwin/msys/win32 builds require -no-undefined for sharedlib.
 	# We must also check with_cross_host to decide if this is a native
 	# or cross-build and select where to install dlls appropriately.
 	if test -n "$with_cross_host" &&
