diff -Naur mpfr-4.1.0-orig/acinclude.m4 mpfr-4.1.0/acinclude.m4
--- mpfr-4.1.0-orig/acinclude.m4	2020-06-09 13:53:21.000000000 +0300
+++ mpfr-4.1.0/acinclude.m4	2020-07-17 09:37:56.366441200 +0300
@@ -1473,7 +1473,7 @@
 [AC_REQUIRE([AC_CANONICAL_HOST])
 AC_SUBST(MPFR_LIBM,'')
 case $host in
-  *-*-beos* | *-*-cygwin* | *-*-pw32*)
+  *-*-beos* | *-*-cygwin* | *-*-msys* | *-*-pw32*)
     # According to libtool AC CHECK LIBM, these systems don't have libm
     ;;
   *-*-solaris*)
diff -Naur mpfr-4.1.0-orig/config.guess mpfr-4.1.0/config.guess
--- mpfr-4.1.0-orig/config.guess	2020-07-10 14:59:02.000000000 +0300
+++ mpfr-4.1.0/config.guess	2020-07-17 09:42:00.167676100 +0300
@@ -883,6 +883,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
diff -Naur mpfr-4.1.0-orig/configure.ac mpfr-4.1.0/configure.ac
--- mpfr-4.1.0-orig/configure.ac	2020-07-10 14:52:33.000000000 +0300
+++ mpfr-4.1.0/configure.ac	2020-07-17 09:37:56.388596200 +0300
@@ -589,7 +589,7 @@
 AC_SUBST(MPFR_LDFLAGS)
 AC_SUBST(LIBMPFR_LDFLAGS)
 case $host in
-  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-os2*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw* | *-*-pw32* | *-*-os2*)
    AC_MSG_CHECKING(for DLL/static GMP)
    if test "$enable_shared" = yes; then
      MPFR_LDFLAGS="$MPFR_LDFLAGS -no-undefined"
