--- a/syscall.c	2017-06-29 13:25:21.707103000 -0700
+++ b/syscall.c	2017-06-29 13:25:47.096037500 -0700
@@ -485,13 +485,6 @@
 			(ffd.dwReserved0 == IO_REPARSE_TAG_SYMLINK)) {
 			// is symlink!
 			isSymlink = 1;
-		} else {
-			if (DEBUG_GTE(TIME, 3)) {
-				rprintf(FINFO, "do_lstat %S dwFileAttributes of 0x%x res0 0x%x\n",
-					szFname,
-					ffd.dwFileAttributes,
-					ffd.dwReserved0);
-			}
 		}
 		CloseHandle(hFind);
 	}
