--- a/testsuite/rsync.fns	2017-04-19 12:14:33.343959000 -0700
+++ b/testsuite/rsync.fns	2017-06-29 14:08:54.075718600 -0700
@@ -357,7 +357,8 @@
 }
 
 # Determine what shell command will appropriately test for links.
-ln -s foo "$scratchdir/testlink"
+touch "$scratchdir/foo"
+ln -s "$scratchdir/foo" "$scratchdir/testlink"
 for cmd in test /bin/test /usr/bin/test /usr/ucb/bin/test /usr/ucb/test
 do
     for switch in -h -L
@@ -373,7 +374,7 @@
 done
 # ok, now get rid of it
 rm "$scratchdir/testlink"
-
+rm "$scratchdir/foo"
 
 if [ "x$TEST_SYMLINK_CMD" = 'x' ]
 then
--- a/testsuite/daemon.test.orig	2017-06-29 14:39:22.348185100 -0700
+++ b/testsuite/daemon.test	2017-06-29 14:39:40.582379800 -0700
@@ -21,6 +21,10 @@
 
 . "$suitedir/rsync.fns"
 
+case "`uname -s`" in
+	MSYS_NT*) test_skipped "Can't test on MSYS2";;
+esac
+
 chkfile="$scratchdir/rsync.chk"
 outfile="$scratchdir/rsync.out"
 
--- a/testsuite/exclude.test.orig	2017-06-29 14:32:41.906903100 -0700
+++ b/testsuite/exclude.test	2017-06-29 14:35:59.810040100 -0700
@@ -12,6 +12,10 @@
 
 . "$suitedir/rsync.fns"
 
+case "`uname -s`" in
+	MSYS_NT*) test_skipped "Can't test on MSYS2";;
+esac
+
 CVSIGNORE='*.junk'
 export CVSIGNORE
 
--- a/testsuite/executability.test.orig	2017-06-29 14:36:31.680000700 -0700
+++ b/testsuite/executability.test	2017-06-29 14:36:54.257392900 -0700
@@ -7,6 +7,10 @@
 
 . $suitedir/rsync.fns
 
+case "`uname -s`" in
+	MSYS_NT*) test_skipped "Can't test on MSYS2";;
+esac
+
 # Put some files in the From directory
 mkdir "$fromdir"
 cat <<EOF >"$fromdir/1"
--- a/testsuite/itemize.test.orig	2017-06-29 14:37:25.666094600 -0700
+++ b/testsuite/itemize.test	2017-06-29 14:37:43.983984200 -0700
@@ -10,6 +10,10 @@
 
 . "$suitedir/rsync.fns"
 
+case "`uname -s`" in
+	MSYS_NT*) test_skipped "Can't test on MSYS2";;
+esac
+
 to2dir="$tmpdir/to2"
 
 chkfile="$scratchdir/rsync.chk"
