diff -Naur autoconf-2.69/build-aux/config.guess autoconf-2.69m/build-aux/config.guess
--- autoconf-2.69/build-aux/config.guess	2012-04-25 02:15:10.000000000 +0000
+++ autoconf-2.69m/build-aux/config.guess	2013-02-23 20:44:30.000000000 +0000
@@ -844,6 +844,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur autoconf-2.69/lib/autoconf/fortran.m4 autoconf-2.69m/lib/autoconf/fortran.m4
--- autoconf-2.69/lib/autoconf/fortran.m4	2012-03-07 17:35:25.000000000 +0000
+++ autoconf-2.69m/lib/autoconf/fortran.m4	2013-02-23 20:46:35.000000000 +0000
@@ -666,7 +666,7 @@
 	  ;;
 	-lkernel32)
 	  case $host_os in
-	  *cygwin*) ;;
+	  *cygwin* | *msys*) ;;
 	  *) ac_cv_[]_AC_LANG_ABBREV[]_libs="$ac_cv_[]_AC_LANG_ABBREV[]_libs $ac_arg"
 	    ;;
 	  esac
diff -Naur autoconf-2.69/lib/autoconf/specific.m4 autoconf-2.69m/lib/autoconf/specific.m4
--- autoconf-2.69/lib/autoconf/specific.m4	2012-01-21 13:46:39.000000000 +0000
+++ autoconf-2.69m/lib/autoconf/specific.m4	2013-02-23 20:47:16.000000000 +0000
@@ -321,7 +321,7 @@
 AU_DEFUN([AC_CYGWIN],
 [AC_CANONICAL_HOST
 case $host_os in
-  *cygwin* ) CYGWIN=yes;;
+  *cygwin* | *msys*) CYGWIN=yes;;
 	 * ) CYGWIN=no;;
 esac
 ], [$0 is obsolete: use AC_CANONICAL_HOST and check if $host_os
diff -Naur autoconf-2.69/tests/local.at autoconf-2.69m/tests/local.at
--- autoconf-2.69/tests/local.at	2012-04-24 20:44:15.000000000 +0000
+++ autoconf-2.69m/tests/local.at	2013-02-23 20:48:14.000000000 +0000
@@ -313,7 +313,7 @@
       [OPENMP_CFLAGS],
       [LIBS|LIB@&t@OBJS|LTLIBOBJS|LDFLAGS],
       [INSTALL(_(DATA|PROGRAM|SCRIPT))?],
-      [CYGWIN|ISC|MINGW32|MINIX|EMXOS2|XENIX|EXEEXT|OBJEXT],
+      [CYGWIN|ISC|MSYS|MINGW32|MINIX|EMXOS2|XENIX|EXEEXT|OBJEXT],
       [X_(CFLAGS|(EXTRA_|PRE_)?LIBS)|x_(includes|libraries)|(have|no)_x],
       [(host|build|target)(_(alias|cpu|vendor|os))?],
       [cross_compiling|U],
