--- wget-1.18-orig/lib/sys_select.in.h	2015-12-12 04:22:18.000000000 +0800
+++ wget-1.18/lib/sys_select.in.h	2016-10-24 17:25:15.902538500 +0800
@@ -82,7 +82,7 @@
    Also, Mac OS X, AIX, HP-UX, IRIX, Solaris, Interix declare select()
    in <sys/time.h>.
    But avoid namespace pollution on glibc systems.  */
-# ifndef __GLIBC__
+# if !(defined __GLIBC__ || defined __NEWLIB__)
 #  include <sys/time.h>
 # endif

@@ -103,7 +103,7 @@
    But avoid namespace pollution on glibc systems.
    Do this after the include_next (for the sake of OpenBSD 5.0) but before
    the split double-inclusion guard (for the sake of Solaris).  */
-#if !(defined __GLIBC__ && !defined __UCLIBC__)
+#if !((defined __GLIBC__ || defined __NEWLIB__) && !defined __UCLIBC__)
 # include <signal.h>
 #endif

