--- wget-1.16.2/configure.ac.orig	2015-03-04 09:15:13.615800000 +0300
+++ wget-1.16.2/configure.ac	2015-03-04 09:15:33.817800000 +0300
@@ -48,7 +48,7 @@
 dnl
 dnl Automake setup
 dnl
-AM_INIT_AUTOMAKE([1.9])
+AM_INIT_AUTOMAKE([1.14])
 
 dnl
 dnl Get cannonical host
@@ -200,7 +200,7 @@
 dnl Gettext
 dnl
 AM_GNU_GETTEXT([external],[need-ngettext])
-AM_GNU_GETTEXT_VERSION([0.17])
+AM_GNU_GETTEXT_VERSION([0.19])
 
 AC_PROG_RANLIB
 
