From 55d8b959cee7b461fb8ee6d08c29fb06417b9f96 Mon Sep 17 00:00:00 2001
From: "j0t" <j0t@inbox.lv>
Date: Sun, 7 Apr 2019 11:31:32 +0300
Subject: [PATCH] fix pcsc_io_request_s layout for SCardTransmit in 64-bit

---
 scd/apdu.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scd/apdu.c b/scd/apdu.c
index 816938ac5..05f380dd9 100644
--- a/scd/apdu.c
+++ b/scd/apdu.c
@@ -266,8 +266,8 @@ static npth_mutex_t reader_table_lock;
 
 struct pcsc_io_request_s
 {
-  unsigned long protocol;
-  unsigned long pci_len;
+  pcsc_dword_t protocol;
+  pcsc_dword_t pci_len;
 };
 
 typedef struct pcsc_io_request_s *pcsc_io_request_t;
-- 
2.17.0

