From 39fb2c89cf77b55ed72f5501c0503ad65c20d85e Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Mon, 21 Aug 2023 14:22:50 +0900
Subject: avoid-systemd-deprecation-warning

---
 dirmngr/dirmngr.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/dirmngr/dirmngr.c b/dirmngr/dirmngr.c
index 5a2a45bc1..2e077356b 100644
--- a/dirmngr/dirmngr.c
+++ b/dirmngr/dirmngr.c
@@ -1331,9 +1331,6 @@ main (int argc, char **argv)
     {
       struct stat statbuf;
 
-      if (!opt.quiet)
-        log_info(_("WARNING: \"%s\" is a deprecated option\n"), "--supervised");
-
       is_supervised = 1;
 
       /* In supervised mode, we expect file descriptor 3 to be an
-- 
2.43.0

