From 8ee9269a8e616485e7261d6cc6357bbdecad111e Mon Sep 17 00:00:00 2001
From: Andrew Gallagher <andrewg@andrewg.com>
Date: Tue, 6 Jan 2026 17:54:26 +0000
Subject: skip trust packets during import-restore

---
 g10/import.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/g10/import.c b/g10/import.c
index 882b3c453..29bcacbf5 100644
--- a/g10/import.c
+++ b/g10/import.c
@@ -1084,9 +1084,7 @@ read_block( IOBUF a, unsigned int options,
           break;
 
         case PKT_RING_TRUST:
-          /* Skip those packets unless we are in restore mode.  */
-          if ((opt.import_options & IMPORT_RESTORE))
-            goto x_default;
+          /* Skip trust packets.  */
           free_packet (pkt, &parsectx);
           init_packet(pkt);
           break;
-- 
2.39.3 (Apple Git-145)

