diff -Naur gnupg-1.4.20-orig/configure.ac gnupg-1.4.20/configure.ac
--- gnupg-1.4.20-orig/configure.ac	2013-10-04 22:34:26.000000000 +0400
+++ gnupg-1.4.20/configure.ac	2013-11-06 07:06:03.372000000 +0400
@@ -820,6 +820,9 @@
     *-*-cygwin*)
         PRINTABLE_OS_NAME="Cygwin"
         ;;
+    *-*-msys*)
+        PRINTABLE_OS_NAME="MSYS"
+        ;;
     i?86-emx-os2 | i?86-*-os2*emx )
         PRINTABLE_OS_NAME="OS/2"
         ;;
@@ -1156,7 +1159,7 @@
         random_modules="rndlinux"
      else
          case "${host}" in
-            *-mingw32*|*-*-cygwin*)
+            *-mingw32*|*-*-cygwin*|*-*-msys*)
                 random_modules="rndw32"
                 ;;
             i?86-emx-os2|i?86-*-os2*emx)
diff -Naur gnupg-1.4.20-orig/intl/Makefile.in gnupg-1.4.20/intl/Makefile.in
--- gnupg-1.4.20-orig/intl/Makefile.in	2013-10-04 22:34:26.000000000 +0400
+++ gnupg-1.4.20/intl/Makefile.in	2013-11-06 07:06:40.297200000 +0400
@@ -414,7 +414,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
@@ -500,7 +500,7 @@
 	    case '@host_os@' in \
 	      darwin[56]*) \
 	        need_charset_alias=true ;; \
-	      darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	      darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	        need_charset_alias=false ;; \
 	      *) \
 	        need_charset_alias=true ;; \
diff -Naur gnupg-1.4.20-orig/m4/intl.m4 gnupg-1.4.20/m4/intl.m4
--- gnupg-1.4.20-orig/m4/intl.m4	2013-10-04 22:34:26.000000000 +0400
+++ gnupg-1.4.20/m4/intl.m4	2013-11-06 07:06:59.235600000 +0400
@@ -125,7 +125,7 @@
   dnl On mingw and Cygwin, we can activate special Makefile rules which add
   dnl version information to the shared libraries and executables.
   case "$host_os" in
-    mingw* | cygwin*) is_woe32=yes ;;
+    mingw* | cygwin* | msys*) is_woe32=yes ;;
     *) is_woe32=no ;;
   esac
   WOE32=$is_woe32
diff -Naur gnupg-1.4.20-orig/scripts/compile gnupg-1.4.20/scripts/compile
--- gnupg-1.4.20-orig/scripts/compile	2013-10-04 22:34:26.000000000 +0400
+++ gnupg-1.4.20/scripts/compile	2013-11-06 07:07:30.591600000 +0400
@@ -54,7 +54,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -68,7 +68,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur gnupg-1.4.20-orig/scripts/config.guess gnupg-1.4.20/scripts/config.guess
--- gnupg-1.4.20-orig/scripts/config.guess	2013-10-04 22:34:26.000000000 +0400
+++ gnupg-1.4.20/scripts/config.guess	2013-11-06 07:08:38.997600000 +0400
@@ -851,6 +851,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur gnupg-1.4.20-orig/scripts/config.rpath gnupg-1.4.20/scripts/config.rpath
--- gnupg-1.4.20-orig/scripts/config.rpath	2013-10-04 22:34:26.000000000 +0400
+++ gnupg-1.4.20/scripts/config.rpath	2013-11-06 07:08:09.544800000 +0400
@@ -64,7 +64,7 @@
           ;;
       esac
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -138,7 +138,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -188,7 +188,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -332,7 +332,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -523,7 +523,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
