diff -Naur windows-default-manifest-orig/config.guess windows-default-manifest/config.guess
--- windows-default-manifest-orig/config.guess	2014-04-07 19:55:55.000000000 +0300
+++ windows-default-manifest/config.guess	2014-10-26 20:18:53.028800000 +0300
@@ -866,6 +866,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur windows-default-manifest-orig/configure.ac windows-default-manifest/configure.ac
--- windows-default-manifest-orig/configure.ac	2014-04-07 21:59:29.000000000 +0300
+++ windows-default-manifest/configure.ac	2014-10-26 20:19:22.372400000 +0300
@@ -10,10 +10,12 @@
 case "${host}" in
   *-cygwin)
   	;;
+  *-msys)
+  	;;
   *-mingw*)
   	;;
   *)
-  	echo "The only supported hosts are Cygwin and MingW."
+  	echo "The only supported hosts are Cygwin/MSYS2 and MingW."
 	exit 1
 	;;
 esac
