--- YAML-Syck-1.31/inc/ExtUtils/HasCompiler.pm.orig	2013-11-10 13:42:56.977200000 +0400
+++ YAML-Syck-1.31/inc/ExtUtils/HasCompiler.pm	2013-11-10 13:48:13.062400000 +0400
@@ -104,7 +104,7 @@
 			my $lib = '-l' . ($libperl =~ /lib([^.]+)\./)[0];
 			push @extra, "$abs_basename.def", $lib, $perllibs;
 		}
-		elsif ($^O eq 'cygwin') {
+		elsif ($^O eq 'cygwin' or $^O eq 'msys') {
 			push @extra, catfile($incdir, $config->get('useshrplib') ? 'libperl.dll.a' : 'libperl.a');
 		}
 		elsif ($^O eq 'aix') {
