--- YAML-Syck-1.27/inc/Module/Install/Can.pm.orig	2013-11-10 13:42:56.977200000 +0400
+++ YAML-Syck-1.27/inc/Module/Install/Can.pm	2013-11-10 13:48:13.062400000 +0400
@@ -132,7 +132,7 @@
 }
 
 # Fix Cygwin bug on maybe_command();
-if ( $^O eq 'cygwin' ) {
+if ( $^O eq 'cygwin' or $^O eq 'msys' ) {
 	require ExtUtils::MM_Cygwin;
 	require ExtUtils::MM_Win32;
 	if ( ! defined(&ExtUtils::MM_Cygwin::maybe_command) ) {
