diff --git a/configure.ac b/configure.ac
index e7f79b0..c3d586e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -174,7 +174,7 @@ case "${host_os}" in
 		appendLib -lsocket -lnsl -lresolv
 		ISSUN=1
 	;;
-	cygwin)
+	cygwin|msys*)
 		# We don't want to use -std=gnu++11 instead of -std=c++11, but among other things, -std=c++11 defines __STRICT_ANSI__ which makes cygwin not to compile: undefined references to strerror_r, to fdopen, to strcasecmp, etc (their declarations in system headers are between ifdef)
 		appendCXX -U__STRICT_ANSI__
 		ISCYGWIN=1
@@ -485,7 +485,14 @@
 else
 	# But cygwin does want most of ZNC in a shared lib
 	# See https://cygwin.com/ml/cygwin-apps/2015-07/msg00108.html for the reasoning behind the name.
-	LIBZNC="cygznc-${LIBZNC_VERSION}.dll"
+	case "${host_os}" in
+	cygwin*)
+		LIBZNC="cygznc-${LIBZNC_VERSION}.dll"
+	;;
+	msys*)
+		LIBZNC="msys-znc-${LIBZNC_VERSION}.dll"
+	;;
+	esac
 	LIBZNCDIR="$bindir"
 	# See above about __STRICT_ANSI__
 	qt_CFLAGS="$qt_CFLAGS -U__STRICT_ANSI__"
