--- origsrc/db-5.3.21/dist/aclocal/sql.m4	2012-05-11 12:57:48.000000000 -0500
+++ src/db-5.3.21/dist/aclocal/sql.m4	2013-03-08 01:47:48.453498100 -0600
@@ -91,7 +91,7 @@ esac
 # !!! END COPIED from autoconf distribution
 
 sqlite_dir=$srcdir/../lang/sql/sqlite
-(cd sql && eval "\$SHELL ../$sqlite_dir/configure --disable-option-checking $ac_sub_configure_args CPPFLAGS=\"-I.. $CPPFLAGS\" --enable-amalgamation=$db_cv_sql_amalgamation --enable-readline=$with_readline" && cat build_config.h >> config.h) || exit 1
+(cd sql && eval "\$SHELL $sqlite_dir/configure --disable-option-checking $ac_sub_configure_args CPPFLAGS=\"-I.. $CPPFLAGS\" --enable-amalgamation=$db_cv_sql_amalgamation --enable-readline=$with_readline" && cat build_config.h >> config.h) || exit 1
 
 # Configure JDBC if --enable-jdbc
 if test "$db_cv_jdbc" != "no"; then
