--- origsrc/db-5.3.21/dist/aclocal_java/ac_jni_include_dirs.m4	2012-05-11 12:57:48.000000000 -0500
+++ src/db-5.3.21/dist/aclocal_java/ac_jni_include_dirs.m4	2013-03-08 00:21:17.541594900 -0600
@@ -75,7 +75,6 @@ if test "$_JTOPDIR" != "/usr"; then
 	case "$host_os" in
 	aix*)		_JNI_INC_SUBDIRS="aix";;
 	bsdi*)		_JNI_INC_SUBDIRS="bsdos";;
-	cygwin*)	_JNI_INC_SUBDIRS="win32";;
 	freebsd*)	_JNI_INC_SUBDIRS="freebsd";;
 	hp*)		_JNI_INC_SUBDIRS="hp-ux";;
 	linux*)		_JNI_INC_SUBDIRS="linux genunix";;
--- origsrc/db-5.3.21/dist/configure.ac	2012-05-11 12:57:48.000000000 -0500
+++ src/db-5.3.21/dist/configure.ac	2013-03-08 00:20:48.257920000 -0600
@@ -435,7 +435,7 @@ if test "$db_cv_java" = "yes" -o "$db_cv
         # A classpath that includes . is needed to check for Java
 	# Since Cygwin uses Windows' javac, we need Windows path separators
 	case "$host_os" in 
-	cygwin*)	CLASSPATH=".;$CLASSPATH";;
+	mingw*)		CLASSPATH=".;$CLASSPATH";;
 	*)		CLASSPATH=".:$CLASSPATH";;
 	esac
 	export CLASSPATH
