diff -Naur libedit-20120311-3.0-orig/config.guess libedit-20120311-3.0/config.guess
--- libedit-20120311-3.0-orig/config.guess	2012-03-11 22:01:33.000000000 +0000
+++ libedit-20120311-3.0/config.guess	2013-04-26 16:07:04.552734300 +0000
@@ -846,6 +846,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
