diff --git a/src/el.h b/src/el.h
index b0d174e..3940b07 100644
--- a/src/el.h
+++ b/src/el.h
@@ -117,8 +117,8 @@ struct editline {
 	int		  el_errfd;	/* Error file descriptor	*/
 	int		  el_flags;	/* Various flags.		*/
 	coord_t		  el_cursor;	/* Cursor location		*/
-	wint_t		**el_display;	/* Real screen image = what is there */
-	wint_t		**el_vdisplay;	/* Virtual screen image = what we see */
+	wchar_t		**el_display;	/* Real screen image = what is there */
+	wchar_t		**el_vdisplay;	/* Virtual screen image = what we see */
 	void		 *el_data;	/* Client data			*/
 	el_line_t	  el_line;	/* The current line information	*/
 	el_state_t	  el_state;	/* Current editor state		*/
