diff -Naur libedit-20120311-3.0-orig/config.guess libedit-20120311-3.0/config.guess
--- libedit-20120311-3.0-orig/config.guess	2012-03-11 22:01:33.000000000 +0000
+++ libedit-20120311-3.0/config.guess	2013-04-26 16:07:04.552734300 +0000
@@ -947,6 +947,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	GUESS=x86_64-pc-cygwin
 	;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	GUESS=x86_64-unknown-msys
+	;;
     prep*:SunOS:5.*:*)
 	SUN_REL=`echo "$UNAME_RELEASE" | sed -e 's/[^.]*//'`
 	GUESS=powerpcle-unknown-solaris2$SUN_REL
