diff -Naur pth-2.0.7-orig/aclocal.m4 pth-2.0.7/aclocal.m4
--- pth-2.0.7-orig/aclocal.m4	2006-06-08 21:54:01.000000000 +0400
+++ pth-2.0.7/aclocal.m4	2014-08-01 11:23:55.456400000 +0400
@@ -1101,7 +1101,7 @@
         #   can use the Interactive UNIX dispatching code.
         ac_cv_check_sjlj=sjljisc
         ;;
-    *-*-cygwin* )
+    *-*-cygwin* | *-*-44msys* )
         ac_cv_check_sjlj=sjljw32
         ;;
 esac
diff -Naur pth-2.0.7-orig/config.guess pth-2.0.7/config.guess
--- pth-2.0.7-orig/config.guess	2006-03-07 14:10:49.000000000 +0300
+++ pth-2.0.7/config.guess	2014-08-01 10:58:24.523400000 +0400
@@ -769,6 +769,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     i*:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
@@ -797,6 +800,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur pth-2.0.7-orig/config.sub pth-2.0.7/config.sub
--- pth-2.0.7-orig/config.sub	2006-03-07 14:10:49.000000000 +0300
+++ pth-2.0.7/config.sub	2014-08-01 10:57:51.451400000 +0400
@@ -1199,7 +1199,7 @@
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* \
-	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
+	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
diff -Naur pth-2.0.7-orig/shtool pth-2.0.7/shtool
--- pth-2.0.7-orig/shtool	2006-06-08 22:14:44.000000000 +0400
+++ pth-2.0.7/shtool	2014-08-01 11:02:00.447400000 +0400
@@ -528,7 +528,7 @@
                 term_bold=`awk 'BEGIN { printf("%c%c%c%c", 27, 91, 49, 109); }' </dev/null 2>/dev/null`
                 term_norm=`awk 'BEGIN { printf("%c%c%c", 27, 91, 109); }' </dev/null 2>/dev/null`
                 ;;
-            vt100|vt100*|cygwin)
+            vt100|vt100*|cygwin|msys)
                 term_bold=`awk 'BEGIN { printf("%c%c%c%c%c%c", 27, 91, 49, 109, 0, 0); }' </dev/null 2>/dev/null`
                 term_norm=`awk 'BEGIN { printf("%c%c%c%c%c", 27, 91, 109, 0, 0); }' </dev/null 2>/dev/null`
                 ;;
