--- libgpg-error-1.19/src/Makefile.am.orig	2015-04-10 17:51:48.000000000 +0200
+++ libgpg-error-1.19/src/Makefile.am	2015-04-30 15:06:04.203067300 +0200
@@ -282,8 +284,8 @@
 pre_mkheader_cmds = :
 parts_of_gpg_error_h += ./lock-obj-pub.native.h
 
-./lock-obj-pub.native.h: Makefile gen-posix-lock-obj posix-lock-obj.h
-	./gen-posix-lock-obj >$@
+./lock-obj-pub.native.h: Makefile gen-posix-lock-obj$(EXEEXT) posix-lock-obj.h
+	./gen-posix-lock-obj$(EXEEXT) >$@
 endif
 
 # We also depend on versioninfo.rc because that is build by
