--- guile-2.0.14/test-suite/standalone/test-ffi	2013-03-01 17:50:05.000000000 +0100
+++ b/test-suite/standalone/test-ffi	2017-03-25 12:49:48.062647800 +0100
@@ -266,11 +266,11 @@
 (define global (dynamic-link))
 
 (define strerror
-  (pointer->procedure '* (dynamic-func "strerror" global)
+  (pointer->procedure '* (dynamic-func "strerror" (dynamic-link "msys-2.0")) 
                       (list int)))
 
 (define strlen
-  (pointer->procedure size_t (dynamic-func "strlen" global)
+  (pointer->procedure size_t (dynamic-func "strlen" (dynamic-link "msys-2.0"))
                       (list '*)))
 
 (let* ((ptr (strerror ENOENT))
