From 95a61f2a2f6fb42f235c658b8f8374dc50d3c0d5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hannes=20M=C3=BCller?= <>
Date: Thu, 8 Sep 2022 11:06:21 +0200
Subject: [PATCH] Include ITIMER_REAL test in signals.test for Cygwin

---
 test-suite/tests/signals.test | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/test-suite/tests/signals.test b/test-suite/tests/signals.test
index 0b5570ae9..2c4fc931d 100644
--- a/test-suite/tests/signals.test
+++ b/test-suite/tests/signals.test
@@ -39,14 +39,8 @@
     (with-test-prefix "current itimers are 0"
 
       (pass-if "ITIMER_REAL"
-        ;; setitimer may have already been called in other tests.  For
-        ;; some versions of Cygwin, the return value of setitimer is
-        ;; invalid after an alarm has occurred.  See
-        ;; https://www.cygwin.com/ml/cygwin/2019-02/msg00395.html
-        (if (string-contains %host-type "cygwin")
-            (throw 'unresolved)
-            (equal? (setitimer ITIMER_REAL 0 0 0 0)
-                    '((0 . 0) (0 . 0)))))
+        (equal? (setitimer ITIMER_REAL 0 0 0 0)
+                '((0 . 0) (0 . 0))))
 
       (pass-if "ITIMER_VIRTUAL"
         (if (not (provided? 'ITIMER_VIRTUAL))
-- 
2.30.2

