From 6ed20304c4d6e68fa74b01228700e6aebffc129a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hannes=20M=C3=BCller?= <>
Date: Wed, 7 Sep 2022 15:35:45 +0200
Subject: [PATCH] In foreign.test add linked dll for host-type msys explicitly

---
 test-suite/tests/foreign.test | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test-suite/tests/foreign.test b/test-suite/tests/foreign.test
index 28d7b5df8..2e9d6312f 100644
--- a/test-suite/tests/foreign.test
+++ b/test-suite/tests/foreign.test
@@ -241,6 +241,9 @@
         ;; into linked DLLs. Thus, one needs to link to the core C
         ;; library DLL explicitly.
         "cygwin1")
+       ((string-contains %host-type "msys")
+        ;; MSYS2 behaves like Cygwin
+        "msys-2.0")
        (else #f))
       "qsort" #:arg-types (list '* size_t size_t '*))))
 
-- 
2.30.2

