From 7888fca3944bf131253074261dde0965f0500c21 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hannes=20M=C3=BCller?= <>
Date: Wed, 7 Sep 2022 16:12:02 +0200
Subject: [PATCH 5/5] In i18n.test add unresolved for host type msys

---
 test-suite/tests/i18n.test | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test-suite/tests/i18n.test b/test-suite/tests/i18n.test
index 83b53d017..01c71d4a5 100644
--- a/test-suite/tests/i18n.test
+++ b/test-suite/tests/i18n.test
@@ -177,6 +177,7 @@
           (string-contains %host-type "solaris2.10")
           (string-contains %host-type "darwin8")
           (string-contains %host-type "mingw32")
+          (string-contains %host-type "msys")
           (string-contains %host-type "cygwin"))
       (throw 'unresolved)
       (under-locale-or-unresolved %turkish-utf8-locale thunk)))
-- 
2.30.2

