--- origsrc/guile-2.0.9/guile-readline/Makefile.am	2012-07-02 04:28:13.000000000 -0500
+++ src/guile-2.0.9/guile-readline/Makefile.am	2013-04-21 21:48:37.315207200 -0500
@@ -51,7 +51,7 @@ libguilereadline_v_@LIBGUILEREADLINE_MAJ
 
 libguilereadline_v_@LIBGUILEREADLINE_MAJOR@_la_LDFLAGS =	\
   -version-info @LIBGUILEREADLINE_INTERFACE@ -export-dynamic	\
-  -no-undefined
+  -module -no-undefined
 
 
 BUILT_SOURCES = readline.x
