diff -Naur guile-2.0.9-orig/build-aux/ar-lib guile-2.0.9/build-aux/ar-lib
--- guile-2.0.9-orig/build-aux/ar-lib	2013-01-28 20:35:24.000000000 +0000
+++ guile-2.0.9/build-aux/ar-lib	2013-04-25 07:34:38.530273400 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -65,7 +65,7 @@
 	mingw)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin)
+	cygwin | msys)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine)
diff -Naur guile-2.0.9-orig/build-aux/compile guile-2.0.9/build-aux/compile
--- guile-2.0.9-orig/build-aux/compile	2013-01-28 20:35:24.000000000 +0000
+++ guile-2.0.9/build-aux/compile	2013-04-25 07:35:03.280273400 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur guile-2.0.9-orig/build-aux/config.guess guile-2.0.9/build-aux/config.guess
--- guile-2.0.9-orig/build-aux/config.guess	2013-01-28 20:35:24.000000000 +0000
+++ guile-2.0.9/build-aux/config.guess	2013-04-25 07:35:30.301757800 +0000
@@ -849,6 +849,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur guile-2.0.9-orig/build-aux/config.rpath guile-2.0.9/build-aux/config.rpath
--- guile-2.0.9-orig/build-aux/config.rpath	2013-02-18 20:37:42.000000000 +0000
+++ guile-2.0.9/build-aux/config.rpath	2013-04-25 07:36:09.202148400 +0000
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur guile-2.0.9-orig/config.rpath guile-2.0.9/config.rpath
--- guile-2.0.9-orig/config.rpath	2010-12-08 09:06:14.000000000 +0000
+++ guile-2.0.9/config.rpath	2013-04-25 07:30:05.668945300 +0000
@@ -64,7 +64,7 @@
           ;;
       esac
       ;;
-    mingw* | cygwin* | pw32* | os2*)
+    mingw* | cygwin* | msys* | pw32* | os2*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -132,7 +132,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -182,7 +182,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -326,7 +326,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -517,7 +517,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur guile-2.0.9-orig/guile-readline/config.guess guile-2.0.9/guile-readline/config.guess
--- guile-2.0.9-orig/guile-readline/config.guess	2010-12-13 19:29:24.000000000 +0000
+++ guile-2.0.9/guile-readline/config.guess	2013-04-25 07:42:24.603515600 +0000
@@ -800,6 +800,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
@@ -840,6 +843,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur guile-2.0.9-orig/guile-readline/config.sub guile-2.0.9/guile-readline/config.sub
--- guile-2.0.9-orig/guile-readline/config.sub	2010-12-13 19:29:24.000000000 +0000
+++ guile-2.0.9/guile-readline/config.sub	2013-04-25 07:42:48.841796900 +0000
@@ -1293,7 +1293,7 @@
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
-	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
+	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
diff -Naur guile-2.0.9-orig/lib/Makefile.am guile-2.0.9/lib/Makefile.am
--- guile-2.0.9-orig/lib/Makefile.am	2013-04-05 20:54:14.000000000 +0000
+++ guile-2.0.9/lib/Makefile.am	2013-04-25 07:50:28.660156300 +0000
@@ -878,7 +878,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -Naur guile-2.0.9-orig/lib/Makefile.in guile-2.0.9/lib/Makefile.in
--- guile-2.0.9-orig/lib/Makefile.in	2013-04-09 23:31:58.000000000 +0000
+++ guile-2.0.9/lib/Makefile.in	2013-04-25 07:50:45.152343800 +0000
@@ -2566,7 +2566,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -Naur guile-2.0.9-orig/m4/btowc.m4 guile-2.0.9/m4/btowc.m4
--- guile-2.0.9-orig/m4/btowc.m4	2013-02-18 20:37:43.000000000 +0000
+++ guile-2.0.9/m4/btowc.m4	2013-04-25 07:51:52.476562500 +0000
@@ -50,6 +50,8 @@
            case "$host_os" in
                       # Guess no on Cygwin.
              cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+                      # Guess no on MSYS.
+             msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
diff -Naur guile-2.0.9-orig/m4/double-slash-root.m4 guile-2.0.9/m4/double-slash-root.m4
--- guile-2.0.9-orig/m4/double-slash-root.m4	2013-02-18 20:37:43.000000000 +0000
+++ guile-2.0.9/m4/double-slash-root.m4	2013-04-25 07:52:12.550781300 +0000
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur guile-2.0.9-orig/m4/dup2.m4 guile-2.0.9/m4/dup2.m4
--- guile-2.0.9-orig/m4/dup2.m4	2013-02-18 20:37:43.000000000 +0000
+++ guile-2.0.9/m4/dup2.m4	2013-04-25 07:53:07.474609400 +0000
@@ -51,6 +51,8 @@
              gl_cv_func_dup2_works="guessing no" ;;
            cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
+           msys*) # on MSYS, dup2(1,1) returns 0
+             gl_cv_func_dup2_works="guessing no" ;;
            linux*) # On linux between 2008-07-27 and 2009-05-11, dup2 of a
                    # closed fd may yield -EBADF instead of -1 / errno=EBADF.
              gl_cv_func_dup2_works="guessing no" ;;
diff -Naur guile-2.0.9-orig/m4/malloc.m4 guile-2.0.9/m4/malloc.m4
--- guile-2.0.9-orig/m4/malloc.m4	2013-02-18 20:37:43.000000000 +0000
+++ guile-2.0.9/m4/malloc.m4	2013-04-25 07:56:52.291015600 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur guile-2.0.9-orig/m4/printf.m4 guile-2.0.9/m4/printf.m4
--- guile-2.0.9-orig/m4/printf.m4	2013-02-18 20:37:43.000000000 +0000
+++ guile-2.0.9/m4/printf.m4	2013-04-25 07:57:37.988281300 +0000
@@ -736,6 +736,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -1465,6 +1466,8 @@
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
            cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+                                 # Guess yes on MSYS.
+           msys*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
