diff -urN ccache-3.2.4.orig/ccache.h ccache-3.2.4/ccache.h
--- ccache-3.2.4.orig/ccache.h	2015-10-08 21:14:23.000000000 +0200
+++ ccache-3.2.4/ccache.h	2016-04-17 19:42:45.063373100 +0200
@@ -248,8 +248,9 @@
 #endif
 
 /* mkstemp() on some versions of cygwin don't handle binary files, so
-   override */
-#ifdef __CYGWIN__
+   override,
+   however MSYS2 is fine */
+#if defined(__CYGWIN__) && !defined(__MSYS__)
 #undef HAVE_MKSTEMP
 #endif
 
diff -urN ccache-3.2.4.orig/lockfile.c ccache-3.2.4/lockfile.c
--- ccache-3.2.4.orig/lockfile.c	2015-10-08 21:14:23.000000000 +0200
+++ ccache-3.2.4/lockfile.c	2016-04-17 19:38:29.151513300 +0200
@@ -36,7 +36,7 @@
 	char *my_content = NULL, *content = NULL, *initial_content = NULL;
 	const char *hostname = get_hostname();
 	bool acquired = false;
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__MSYS__)
 	const size_t bufsize = 1024;
 	int fd, len;
 #else
@@ -48,7 +48,7 @@
 		free(my_content);
 		my_content = format("%s:%d:%d", hostname, (int)getpid(), (int)time(NULL));
 
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__MSYS__)
 		fd = open(lockfile, O_WRONLY|O_CREAT|O_EXCL|O_BINARY, 0666);
 		if (fd == -1) {
 			saved_errno = errno;
