--- psmisc-v23.6/src/killall.c.orig	2024-02-01 20:04:58.803238600 +0100
+++ psmisc-v23.6/src/killall.c	2024-02-01 20:04:42.372098800 +0100
@@ -38,7 +38,9 @@
 #include <errno.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#ifdef HAVE_SYS_SYSCALL_H
 #include <sys/syscall.h>
+#endif
 #include <fcntl.h>
 #include <getopt.h>
 #include <pwd.h>
--- psmisc-v23.7/src/statx.c.orig	2024-06-09 20:06:35.559989500 +0200
+++ psmisc-v23.7/src/statx.c	2024-06-09 20:07:07.457617800 +0200
@@ -26,6 +26,8 @@
 #include <config.h>
 #endif
 
+#ifdef WITH_STATX
+
 #include <sys/sysmacros.h>
 #include <sys/syscall.h>
 #include <sys/types.h>
@@ -33,7 +35,6 @@
 #include <fcntl.h>		/* Definition of AT_* constants */
 
 int stat_flags = AT_NO_AUTOMOUNT|AT_STATX_DONT_SYNC;
-#ifdef WITH_STATX
 
 #include <errno.h>
 #ifndef HAVE_STATX

