diff -r 39b9b05c3085 Lib/distutils/sysconfig.py
--- a/Lib/distutils/sysconfig.py	Wed Apr 10 00:27:23 2013 +0200
+++ b/Lib/distutils/sysconfig.py	Wed Apr 10 10:14:18 2013 +0200
@@ -370,7 +370,10 @@
                     done[n] = item = ""
                 if found:
                     after = value[m.end():]
-                    value = value[:m.start()] + item + after
+                    value = value[:m.start()]
+                    if item.strip() not in value:
+                        value += item
+                    value += after
                     if "$" in after:
                         notdone[name] = value
                     else:
diff -r 39b9b05c3085 Lib/sysconfig.py
--- a/Lib/sysconfig.py	Wed Apr 10 00:27:23 2013 +0200
+++ b/Lib/sysconfig.py	Wed Apr 10 10:14:18 2013 +0200
@@ -294,7 +294,10 @@
 
                 if found:
                     after = value[m.end():]
-                    value = value[:m.start()] + item + after
+                    value = value[:m.start()]
+                    if item.strip() not in value:
+                        value += item
+                    value += after
                     if "$" in after:
                         notdone[name] = value
                     else:
