diff -r 28c04e954bb6 Lib/lib2to3/main.py
--- a/Lib/lib2to3/main.py	Tue Oct 29 22:25:55 2013 -0400
+++ b/Lib/lib2to3/main.py	Wed Nov 06 14:33:07 2013 +0100
@@ -213,6 +213,7 @@
 
     # Set up logging handler
     level = logging.DEBUG if options.verbose else logging.INFO
+    logging.root.handlers = []
     logging.basicConfig(format='%(name)s: %(message)s', level=level)
     logger = logging.getLogger('lib2to3.main')
 
