diff -up Python-3.5.0/configure.ac Python-3.5.0/configure.ac
--- Python-3.5.0/configure.ac	2015-09-23 13:52:20.756909744 +0200
+++ Python-3.5.0/configure.ac	2015-09-23 13:52:46.859163629 +0200
@@ -799,9 +799,9 @@ cat >> conftest.c <<EOF
         alpha-linux-gnu
 # elif defined(__ARM_EABI__) && defined(__ARM_PCS_VFP)
 #  if defined(__ARMEL__)
-        arm-linux-gnueabihf
+        arm-linux-gnueabi
 #  else
-        armeb-linux-gnueabihf
+        armeb-linux-gnueabi
 #  endif
 # elif defined(__ARM_EABI__) && !defined(__ARM_PCS_VFP)
 #  if defined(__ARMEL__)
