diff -urp Python-3.5.0/configure p/configure
--- Python-3.5.0/configure	2016-02-25 16:12:12.615184011 +0000
+++ p/configure	2016-02-25 16:13:01.293412517 +0000
@@ -5267,7 +5267,7 @@ cat >> conftest.c <<EOF
 #  elif _MIPS_SIM == _ABIN32
         mips64el-linux-gnuabin32
 #  elif _MIPS_SIM == _ABI64
-        mips64el-linux-gnuabi64
+        mips64el-linux-gnu
 #  else
 #   error unknown platform triplet
 #  endif
@@ -5277,7 +5277,7 @@ cat >> conftest.c <<EOF
 #  elif _MIPS_SIM == _ABIN32
         mips64-linux-gnuabin32
 #  elif _MIPS_SIM == _ABI64
-        mips64-linux-gnuabi64
+        mips64-linux-gnu
 #  else
 #   error unknown platform triplet
 #  endif
diff -urp Python-3.5.0/configure.ac p/configure.ac
--- Python-3.5.0/configure.ac	2016-02-25 16:12:11.663159985 +0000
+++ p/configure.ac	2016-02-25 16:13:18.814854710 +0000
@@ -821,7 +821,7 @@ cat >> conftest.c <<EOF
 #  elif _MIPS_SIM == _ABIN32
         mips64el-linux-gnuabin32
 #  elif _MIPS_SIM == _ABI64
-        mips64el-linux-gnuabi64
+        mips64el-linux-gnu
 #  else
 #   error unknown platform triplet
 #  endif
@@ -831,7 +831,7 @@ cat >> conftest.c <<EOF
 #  elif _MIPS_SIM == _ABIN32
         mips64-linux-gnuabin32
 #  elif _MIPS_SIM == _ABI64
-        mips64-linux-gnuabi64
+        mips64-linux-gnu
 #  else
 #   error unknown platform triplet
 #  endif
