diff --git a/Makefile.pre.in b/Makefile.pre.in
index 28df0e1..42f811c 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -879,6 +879,7 @@ Python/frozen.o: Python/importlib.h Python/importlib_external.h
 # follow our naming conventions. dtrace(1) uses the output filename to generate
 # an include guard, so we can't use a pipeline to transform its output.
 Include/pydtrace_probes.h: $(srcdir)/Include/pydtrace.d
+	@$(MKDIR_P) Include
 	$(DTRACE) $(DFLAGS) -o $@ -h -s $<
 	: sed in-place edit with POSIX-only tools
 	sed 's/PYTHON_/PyDTrace_/' $@ > $@.tmp
