--- Python-3.4.3/Lib/distutils/command/build_ext.py.orig	2015-02-25 05:27:44.000000000 -0600
+++ Python-3.4.3/Lib/distutils/command/build_ext.py	2015-05-05 11:15:40.666175000 -0500
@@ -696,9 +696,9 @@ class build_ext(Command):
             else:
                 return ext.libraries
         elif sys.platform[:6] == "cygwin":
-            template = "python%d.%d"
+            template = "python%d.%d%s"
             pythonlib = (template %
-                   (sys.hexversion >> 24, (sys.hexversion >> 16) & 0xff))
+                   (sys.hexversion >> 24, (sys.hexversion >> 16) & 0xff, sys.abiflags))
             # don't extend ext.libraries, it may be shared with other
             # extensions, it is a reference to the original list
             return ext.libraries + [pythonlib]
--- Python-3.4.3/Makefile.pre.in.orig	2015-02-25 05:27:45.000000000 -0600
+++ Python-3.4.3/Makefile.pre.in	2015-05-05 11:15:40.672175700 -0500
@@ -638,7 +638,7 @@ $(PYTHONFRAMEWORKDIR)/Versions/$(VERSION
 
 # This rule builds the Cygwin Python DLL and import library if configured
 # for a shared core library; otherwise, this rule is a noop.
-$(DLLLIBRARY) libpython$(VERSION).dll.a: $(LIBRARY_OBJS)
+$(DLLLIBRARY) libpython$(LDVERSION).dll.a: $(LIBRARY_OBJS)
 	if test -n "$(DLLLIBRARY)"; then \
 		$(LDSHARED) -Wl,--out-implib=$@ -o $(DLLLIBRARY) $^ \
 			$(LIBS) $(MODLIBS) $(SYSLIBS) $(LDLAST); \
--- Python-3.4.3/Modules/makesetup.orig	2015-02-25 05:27:46.000000000 -0600
+++ Python-3.4.3/Modules/makesetup	2015-05-05 11:15:40.676676300 -0500
@@ -91,7 +91,7 @@ CYGWIN*) if test $libdir = .
 	 else
 	 	ExtraLibDir='$(LIBPL)'
 	 fi
-	 ExtraLibs="-L$ExtraLibDir -lpython\$(VERSION)";;
+	 ExtraLibs="-L$ExtraLibDir -lpython\$(LDVERSION)";;
 esac
 
 # Main loop
--- Python-3.4.3/configure.ac.orig	2015-02-25 05:27:46.000000000 -0600
+++ Python-3.4.3/configure.ac	2015-05-05 11:24:25.915873200 -0500
@@ -944,6 +944,7 @@ if test $enable_shared = "yes"; then
   case $ac_sys_system in
     CYGWIN*)
           LDLIBRARY='libpython$(LDVERSION).dll.a'
+          BLDLIBRARY='-L. -lpython$(LDVERSION)'
           DLLLIBRARY='libpython$(LDVERSION).dll'
           ;;
     SunOS*)
