--- Python-3.4.3/Python/dynload_shlib.c.orig	2015-02-25 05:27:46.000000000 -0600
+++ Python-3.4.3/Python/dynload_shlib.c	2015-05-05 11:25:41.882019700 -0500
@@ -34,6 +34,8 @@
 
 const char *_PyImport_DynLoadFiletab[] = {
 #ifdef __CYGWIN__
+    "." SOABI ".dll",
+    ".abi" PYTHON_ABI_STRING ".dll",
     ".dll",
 #else  /* !__CYGWIN__ */
     "." SOABI ".so",
--- Python-3.4.3/configure.ac.orig	2015-02-25 05:27:46.000000000 -0600
+++ Python-3.4.3/configure.ac	2015-05-05 11:24:25.915873200 -0500
@@ -4096,7 +4097,7 @@ AC_MSG_RESULT($SOABI)
 
 AC_SUBST(EXT_SUFFIX)
 case $ac_sys_system in
-    Linux*|GNU*)
+    Linux*|GNU*|CYGWIN*)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
