--- Python-3.4.3/Include/pythread.h.orig	2015-02-25 05:27:44.000000000 -0600
+++ Python-3.4.3/Include/pythread.h	2015-05-05 11:27:20.117994100 -0500
@@ -77,11 +77,11 @@ PyAPI_FUNC(int) PyThread_set_stacksize(s
 PyAPI_FUNC(PyObject*) PyThread_GetInfo(void);
 
 /* Thread Local Storage (TLS) API */
-PyAPI_FUNC(int) PyThread_create_key(void);
-PyAPI_FUNC(void) PyThread_delete_key(int);
-PyAPI_FUNC(int) PyThread_set_key_value(int, void *);
-PyAPI_FUNC(void *) PyThread_get_key_value(int);
-PyAPI_FUNC(void) PyThread_delete_key_value(int key);
+PyAPI_FUNC(long) PyThread_create_key(void);
+PyAPI_FUNC(void) PyThread_delete_key(long);
+PyAPI_FUNC(int) PyThread_set_key_value(long, void *);
+PyAPI_FUNC(void *) PyThread_get_key_value(long);
+PyAPI_FUNC(void) PyThread_delete_key_value(long key);
 
 /* Cleanup after a fork */
 PyAPI_FUNC(void) PyThread_ReInitTLS(void);
--- Python-3.6.0/Python/pystate.c.orig	2017-02-08 04:54:10.265051500 -0500
+++ Python-3.6.0/Python/pystate.c	2017-02-08 05:00:43.539089400 -0500
@@ -47,7 +47,7 @@
    GILState implementation
 */
 static PyInterpreterState *autoInterpreterState = NULL;
-static int autoTLSkey = -1;
+static long autoTLSkey = -1L;
 #else
 #define HEAD_INIT() /* Nothing */
 #define HEAD_LOCK() /* Nothing */
--- Python-3.6.0/Python/thread_pthread.h.orig	2016-12-22 21:21:22.000000000 -0500
+++ Python-3.6.0/Python/thread_pthread.h	2017-02-08 05:18:45.791168100 -0500
@@ -603,44 +603,44 @@
 
 #define Py_HAVE_NATIVE_TLS
 
-int
+long
 PyThread_create_key(void)
 {
     pthread_key_t key;
     int fail = pthread_key_create(&key, NULL);
     if (fail)
-        return -1;
+        return -1L;
     if (key > INT_MAX) {
         /* Issue #22206: handle integer overflow */
         pthread_key_delete(key);
         errno = ENOMEM;
-        return -1;
+        return -1L;
     }
-    return (int)key;
+    return (long)key;
 }
 
 void
-PyThread_delete_key(int key)
+PyThread_delete_key(long key)
 {
     pthread_key_delete(key);
 }
 
 void
-PyThread_delete_key_value(int key)
+PyThread_delete_key_value(long key)
 {
     pthread_setspecific(key, NULL);
 }
 
 int
-PyThread_set_key_value(int key, void *value)
+PyThread_set_key_value(long key, void *value)
 {
     int fail;
     fail = pthread_setspecific(key, value);
-    return fail ? -1 : 0;
+    return fail ? -1L : 0L;
 }
 
 void *
-PyThread_get_key_value(int key)
+PyThread_get_key_value(long key)
 {
     return pthread_getspecific(key);
 }
