--- Python-3.4.3/Modules/signalmodule.c.orig	2015-02-25 05:27:46.000000000 -0600
+++ Python-3.4.3/Modules/signalmodule.c	2015-05-05 16:10:10.357940900 -0500
@@ -741,7 +741,11 @@ fill_siginfo(siginfo_t *si)
     PyStructSequence_SET_ITEM(result, 4, _PyLong_FromUid(si->si_uid));
     PyStructSequence_SET_ITEM(result, 5,
                                 PyLong_FromLong((long)(si->si_status)));
+#ifdef __CYGWIN__
+    PyStructSequence_SET_ITEM(result, 6, PyLong_FromLong(0L));
+#else
     PyStructSequence_SET_ITEM(result, 6, PyLong_FromLong(si->si_band));
+#endif
     if (PyErr_Occurred()) {
         Py_DECREF(result);
         return NULL;
