--- Python-3.4.3/Modules/_struct.c.orig	2015-02-25 05:27:45.000000000 -0600
+++ Python-3.4.3/Modules/_struct.c	2015-05-05 16:17:01.509150400 -0500
@@ -1627,7 +1627,7 @@ unpackiter_iternext(unpackiterobject *se
 }
 
 static PyTypeObject unpackiter_type = {
-    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    PyVarObject_HEAD_INIT(NULL, 0)
     "unpack_iterator",                          /* tp_name */
     sizeof(unpackiterobject),                   /* tp_basicsize */
     0,                                          /* tp_itemsize */
