--- Python-3.4.4/Lib/cgi.py	2015-12-21 00:00:58.000000000 -0600
+++ Python-3.4.4/Lib/cgi.py	2016-05-24 00:33:59.673777200 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 # NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
 # intentionally NOT "/usr/bin/env python".  On many systems
--- Python-3.4.4/Lib/encodings/rot_13.py	2015-12-21 00:00:58.000000000 -0600
+++ Python-3.4.4/Lib/encodings/rot_13.py	2016-05-24 00:34:13.983796900 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """ Python Character Mapping Codec for ROT13.
 
 This codec de/encodes from str to str.
