--- Python-3.3.2-orig/Lib/distutils/unixccompiler.py	2013-05-15 20:32:54.000000000 +0400
+++ Python-3.3.2/Lib/distutils/unixccompiler.py	2013-08-01 23:49:16.568359300 +0400
@@ -222,7 +222,7 @@
         # the configuration data stored in the Python installation, so
         # we use this hack.
         compiler = os.path.basename(sysconfig.get_config_var("CC"))
-        if sys.platform[:6] == "darwin":
+        if sys.platform[:6] == "darwin" or sys.platform[:6] == "cygwin":
             # MacOSX's linker doesn't understand the -R flag at all
             return "-L" + dir
         elif sys.platform[:5] == "hp-ux":
