--- Python-3.3.2-orig/Lib/ctypes/__init__.py	2013-08-02 00:29:33.877929600 +0400
+++ Python-3.3.2/Lib/ctypes/__init__.py	2013-05-15 20:32:54.000000000 +0400
@@ -436,7 +436,8 @@
 if _os.name in ("nt", "ce"):
     pythonapi = PyDLL("python dll", None, _sys.dllhandle)
 elif _sys.platform == "cygwin":
-    pythonapi = PyDLL("libpython%d.%d.dll" % _sys.version_info[:2])
+    pythonapi = PyDLL("libpython%d.%d%s.dll" % \
+                      (_sys.version_info[:2] + tuple(_sys.abiflags)))
 else:
     pythonapi = PyDLL(None)
 
