--- Python-3.3.2-orig/Lib/distutils/cygwinccompiler.py	2013-05-15 20:32:54.000000000 +0400
+++ Python-3.3.2/Lib/distutils/cygwinccompiler.py	2013-08-02 00:24:11.022460900 +0400
@@ -91,9 +91,7 @@
     compiler_type = 'cygwin'
     obj_extension = ".o"
     static_lib_extension = ".a"
-    shared_lib_extension = ".dll"
-    static_lib_format = "lib%s%s"
-    shared_lib_format = "%s%s"
+    shared_lib_extension = ".dll.a"
     exe_extension = ".exe"
 
     def __init__(self, verbose=0, dry_run=0, force=0):
--- Python-3.3.2-orig/Lib/distutils/unixccompiler.py	2013-08-01 23:49:16.568359300 +0400
+++ Python-3.3.2/Lib/distutils/unixccompiler.py	2013-08-02 00:25:54.459960900 +0400
@@ -79,6 +79,7 @@
     static_lib_format = shared_lib_format = dylib_lib_format = "lib%s%s"
     if sys.platform == "cygwin":
         exe_extension = ".exe"
+        dylib_lib_extension = ".dll.a"
 
     def preprocess(self, source, output_file=None, macros=None,
                    include_dirs=None, extra_preargs=None, extra_postargs=None):
