--- Python-3.3.2-orig/Include/pyport.h	2013-05-15 20:32:54.000000000 +0400
+++ Python-3.3.2/Include/pyport.h	2013-08-02 11:50:31.637695300 +0400
@@ -760,7 +760,15 @@
         /* http://docs.python.org/extending/windows.html#a-cookbook-approach */
 #                       if !defined(__CYGWIN__)
 #                               define PyAPI_FUNC(RTYPE) __declspec(dllimport) RTYPE
+#                       else
+#                               define PyAPI_FUNC(RTYPE) RTYPE
 #                       endif /* !__CYGWIN__ */
+            /* NOTE: The issue3945 "compile error in _fileio.c (cygwin)"
+             * was resolved with modification of code.
+             * This issue was resolved for gcc(mingw) with enabling auto
+             * import feature. Since _fileio.c problem now disappear there
+             * is no more reasons to avoid dllimport for gcc(mingw).
+             */
 #                       define PyAPI_DATA(RTYPE) extern __declspec(dllimport) RTYPE
         /* module init functions outside the core must be exported */
 #                       if defined(__cplusplus)
