--- Python-3.3.2-orig/configure.ac	2013-08-02 00:18:22.859375000 +0400
+++ Python-3.3.2/configure.ac	2013-08-02 00:35:10.071289000 +0400
@@ -3945,7 +3945,7 @@
 
 AC_SUBST(EXT_SUFFIX)
 case $ac_sys_system in
-    Linux*|GNU*)
+    Linux*|GNU*|CYGWIN*)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
--- Python-3.3.2-orig/Python/dynload_shlib.c	2013-05-15 20:33:00.000000000 +0400
+++ Python-3.3.2/Python/dynload_shlib.c	2013-08-02 00:38:38.291015600 +0400
@@ -38,6 +38,7 @@
 
 const char *_PyImport_DynLoadFiletab[] = {
 #ifdef __CYGWIN__
+    "." SOABI ".dll",
     ".dll",
 #else  /* !__CYGWIN__ */
 #if defined(PYOS_OS2) && defined(PYCC_GCC)
