--- Python-3.3.2-orig/Lib/distutils/command/build_ext.py	2013-05-15 20:32:54.000000000 +0400
+++ Python-3.3.2/Lib/distutils/command/build_ext.py	2013-08-02 00:15:05.345703100 +0400
@@ -730,9 +730,9 @@
             # extensions, it is a reference to the original list
             return ext.libraries + [pythonlib]
         elif sys.platform[:6] == "cygwin":
-            template = "python%d.%d"
+            template = "python%d.%d%s"
             pythonlib = (template %
-                   (sys.hexversion >> 24, (sys.hexversion >> 16) & 0xff))
+                   (sys.hexversion >> 24, (sys.hexversion >> 16) & 0xff, sys.abiflags))
             # don't extend ext.libraries, it may be shared with other
             # extensions, it is a reference to the original list
             return ext.libraries + [pythonlib]
--- Python-3.3.2-orig/Modules/makesetup	2013-05-15 20:32:59.000000000 +0400
+++ Python-3.3.2/Modules/makesetup	2013-08-02 00:17:46.703125000 +0400
@@ -91,7 +91,7 @@
 	 else
 	 	ExtraLibDir='$(LIBPL)'
 	 fi
-	 ExtraLibs="-L$ExtraLibDir -lpython\$(VERSION)";;
+	 ExtraLibs="-L$ExtraLibDir -lpython\$(LDVERSION)";;
 esac
 
 # Main loop
--- Python-3.3.2-orig/configure.ac	2013-05-15 20:33:00.000000000 +0400
+++ Python-3.3.2/configure.ac	2013-08-02 00:18:22.859375000 +0400
@@ -921,6 +921,7 @@
   case $ac_sys_system in
     CYGWIN*)
           LDLIBRARY='libpython$(LDVERSION).dll.a'
+          BLDLIBRARY='-L. -lpython$(LDVERSION)'
           DLLLIBRARY='libpython$(LDVERSION).dll'
           ;;
     SunOS*)
--- Python-3.3.2-orig/Makefile.pre.in	2013-05-15 20:32:57.000000000 +0400
+++ Python-3.3.2/Makefile.pre.in	2013-08-02 00:16:23.352539000 +0400
@@ -560,7 +560,7 @@
 
 # This rule builds the Cygwin Python DLL and import library if configured
 # for a shared core library; otherwise, this rule is a noop.
-$(DLLLIBRARY) libpython$(VERSION).dll.a: $(LIBRARY_OBJS)
+$(DLLLIBRARY) libpython$(LDVERSION).dll.a: $(LIBRARY_OBJS)
 	if test -n "$(DLLLIBRARY)"; then \
 		$(LDSHARED) -Wl,--out-implib=$@ -o $(DLLLIBRARY) $^ \
 			$(LIBS) $(MODLIBS) $(SYSLIBS) $(LDLAST); \
