--- Python-3.3.2-orig/Modules/signalmodule.c	2013-05-15 20:32:59.000000000 +0400
+++ python-3.3.2/Modules/signalmodule.c	2013-10-21 13:17:31.434200000 +0400
@@ -8,8 +8,18 @@
 #include "posixmodule.h"
 #endif
 
+#if defined(__CYGWIN__)
+/* Type siginfo_t is different on cygwin. For instance si_band is not
+ * a member of structure, so until fill_siginfo(siginfo_t *si) is
+ * ported properly lets "undef" sigwaitinfo and sigtimedwait to
+ * avoid build failure in fill_siginfo(..).
+ */
+# undef HAVE_SIGWAITINFO
+# undef HAVE_SIGTIMEDWAIT
+#endif
+
 #ifdef MS_WINDOWS
-#include <Windows.h>
+#include <windows.h>
 #ifdef HAVE_PROCESS_H
 #include <process.h>
 #endif
