--- dtc/akefile.orig	2018-01-03 07:04:32.000000000 +0100
+++ dtc/Makefile	2018-07-26 17:52:18.156225100 +0200
@@ -18,7 +18,7 @@
 CPPFLAGS = -I libfdt -I .
 WARNINGS = -Wall -Wpointer-arith -Wcast-qual -Wnested-externs \
 	-Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Wshadow
-CFLAGS = -g -Os $(SHAREDLIB_CFLAGS) -Werror $(WARNINGS)
+CFLAGS = -g -Os $(SHAREDLIB_CFLAGS)  $(WARNINGS)
 
 BISON = bison
 LEX = flex
@@ -40,9 +40,9 @@
 SHAREDLIB_CFLAGS  = -fPIC
 SHAREDLIB_LDFLAGS = -fPIC -dynamiclib -Wl,-install_name -Wl,
 else ifeq ($(HOSTOS),$(filter $(HOSTOS),msys cygwin))
-SHAREDLIB_EXT     = so
+SHAREDLIB_EXT     = dll
 SHAREDLIB_CFLAGS  =
 SHAREDLIB_LDFLAGS = -shared -Wl,--version-script=$(LIBFDT_version) -Wl,-soname,
 else
 SHAREDLIB_EXT     = so
 SHAREDLIB_CFLAGS  = -fPIC
@@ -177,7 +179,7 @@
 libfdt_clean:
 	@$(VECHO) CLEAN "(libfdt)"
 	rm -f $(addprefix $(LIBFDT_objdir)/,$(STD_CLEANFILES))
-	rm -f $(LIBFDT_objdir)/*.so
+	rm -f $(LIBFDT_objdir)/*.$(SHAREDLIB_EXT)
 
 ifneq ($(DEPTARGETS),)
 -include $(LIBFDT_OBJS:%.o=$(LIBFDT_objdir)/%.d)
@@ -246,7 +248,7 @@
 include $(PYLIBFDT_srcdir)/Makefile.pylibfdt
 
 .PHONY: pylibfdt
-pylibfdt: $(PYLIBFDT_objdir)/_libfdt.so
+pylibfdt: $(PYLIBFDT_objdir)/_libfdt.$(SHAREDLIB_EXT)
 
 pylibfdt_clean:
 	@$(VECHO) CLEAN "(pylibfdt)"
--- dtc/pylibfdt/Makefile.pylibfdt.orig	2018-01-03 07:04:32.000000000 +0100
+++ dtc/pylibfdt/Makefile.pylibfdt	2018-07-26 17:52:36.463209700 +0200
@@ -3,7 +3,7 @@
 
 PYLIBFDT_srcs = $(addprefix $(LIBFDT_srcdir)/,$(LIBFDT_SRCS)) \
 		$(PYLIBFDT_srcdir)/libfdt.i
-PYMODULE = $(PYLIBFDT_objdir)/_libfdt.so
+PYMODULE = $(PYLIBFDT_objdir)/_libfdt.$(SHAREDLIB_EXT)
 
 define run_setup
 	SOURCES="$(1)" CPPFLAGS="$(CPPFLAGS)" OBJDIR="$(PYLIBFDT_objdir)"
@@ -14,11 +14,11 @@
 $(PYMODULE): $(PYLIBFDT_srcs)
 	@$(VECHO) PYMOD $@
 	$(call run_setup, $^, build_ext --inplace)
-	mv _libfdt.so $@
+	mv _libfdt.$(SHAREDLIB_EXT) $@
 
 install_pylibfdt: $(PYMODULE)
 	$(VECHO) INSTALL-PYLIB; \
 	$(call run_setup, $(PYLIBFDT_srcs), \
 		install $(if $(SETUP_PREFIX),--prefix=$(SETUP_PREFIX)))
 
-PYLIBFDT_cleanfiles = libfdt_wrap.c libfdt.py libfdt.pyc _libfdt.so
+PYLIBFDT_cleanfiles = libfdt_wrap.c libfdt.py libfdt.pyc _libfdt.$(SHAREDLIB_EXT)
