--- dtc-1.5.0/pylibfdt/Makefile.pylibfdt.orig	2019-07-17 10:50:34.316040400 +0300
+++ dtc-1.5.0/pylibfdt/Makefile.pylibfdt	2019-07-17 10:50:43.566856700 +0300
@@ -21,7 +21,7 @@
 
 install_pylibfdt: $(PYMODULE)
 	@$(VECHO) INSTALL-PYLIB
-	$(PYTHON) $(SETUP) $(SETUPFLAGS) install --prefix=$(PREFIX)
+	$(PYTHON) $(SETUP) $(SETUPFLAGS) install --prefix=$(DESTDIR)$(PREFIX)
 
 pylibfdt_clean:
 	@$(VECHO) CLEAN "(pylibfdt)"
