diff -durN serf-1.3.9.orig/SConstruct serf-1.3.9/SConstruct
--- serf-1.3.9.orig/SConstruct	2015-09-17 20:46:24.000000000 +0800
+++ serf-1.3.9/SConstruct	2017-05-20 12:08:24.781250000 +0800
@@ -117,6 +117,9 @@
                   None),
   RawListVariable('CPPFLAGS', "Extra flags for the C preprocessor "
                   "(space separated)", None), 
+  PathVariable('DESTDIR', 'Directory to install under',
+               '',
+               PathVariable.PathAccept),
   )
 
 if sys.platform == 'win32':
@@ -189,6 +192,7 @@
 apu = str(env['APU'])
 zlib = str(env['ZLIB'])
 gssapi = env.get('GSSAPI', None)
+destdir = str(env.get('DESTDIR', ''))
 
 if gssapi and os.path.isdir(gssapi):
   krb5_config = os.path.join(gssapi, 'bin', 'krb5-config')
@@ -416,8 +420,8 @@
 
 # INSTALLATION STUFF
 
-install_static = env.Install(libdir, lib_static)
-install_shared = env.InstallVersionedLib(libdir, lib_shared)
+install_static = env.Install(destdir + libdir, lib_static)
+install_shared = env.InstallVersionedLib(destdir + libdir, lib_shared)
 
 if sys.platform == 'darwin':
   # Change the shared library install name (id) to its final name and location.
@@ -436,8 +440,8 @@
 
 env.Alias('install-lib', [install_static, install_shared,
                           ])
-env.Alias('install-inc', env.Install(incdir, HEADER_FILES))
-env.Alias('install-pc', env.Install(os.path.join(libdir, 'pkgconfig'),
+env.Alias('install-inc', env.Install(destdir + incdir, HEADER_FILES))
+env.Alias('install-pc', env.Install(os.path.join(destdir + libdir, 'pkgconfig'),
                                     pkgconfig))
 env.Alias('install', ['install-lib', 'install-inc', 'install-pc', ])
 
