--- origsrc/serf-1.3.2/auth/auth.c	2013-09-28 23:37:46.000000000 -0700
+++ src/serf-1.3.2/auth/auth.c	2013-11-01 23:30:11.314705300 -0700
@@ -13,6 +13,7 @@
  * limitations under the License.
  */
 
+#include <strings.h>
 #include "serf.h"
 #include "serf_private.h"
 #include "auth.h"
--- origsrc/serf-1.3.2/auth/auth_basic.c	2013-08-17 14:58:47.000000000 -0700
+++ src/serf-1.3.2/auth/auth_basic.c	2013-11-01 23:30:26.198556600 -0700
@@ -15,6 +15,7 @@
 
 /*** Basic authentication ***/
 
+#include <strings.h>
 #include <serf.h>
 #include <serf_private.h>
 #include <auth/auth.h>
--- origsrc/serf-1.3.2/auth/auth_spnego.c	2013-08-15 02:00:57.000000000 -0700
+++ src/serf-1.3.2/auth/auth_spnego.c	2013-11-01 23:30:42.282476600 -0700
@@ -22,6 +22,7 @@
  *  The HTTP message exchange is documented in RFC 4559.
  **/
 
+#include <strings.h>
 #include <serf.h>
 #include <serf_private.h>
 #include <auth/auth.h>
--- origsrc/serf-1.3.2/buckets/headers_buckets.c	2013-07-21 07:27:03.000000000 -0700
+++ src/serf-1.3.2/buckets/headers_buckets.c	2013-11-01 23:29:05.527942500 -0700
@@ -14,6 +14,7 @@
  */
 
 #include <stdlib.h>
+#include <strings.h>
 
 #include <apr_general.h>  /* for strcasecmp() */
 
--- origsrc/serf-1.3.2/buckets/response_buckets.c	2013-09-28 23:37:46.000000000 -0700
+++ src/serf-1.3.2/buckets/response_buckets.c	2013-11-01 23:29:49.220441600 -0700
@@ -13,6 +13,7 @@
  * limitations under the License.
  */
 
+#include <strings.h>
 #include <apr_lib.h>
 #include <apr_strings.h>
 #include <apr_date.h>
--- origsrc/serf-1.3.2/outgoing.c	2013-09-28 23:37:46.000000000 -0700
+++ src/serf-1.3.2/outgoing.c	2013-11-01 23:29:11.353275700 -0700
@@ -13,6 +13,7 @@
  * limitations under the License.
  */
 
+#include <strings.h>
 #include <apr_pools.h>
 #include <apr_poll.h>
 #include <apr_version.h>
