diff -durN serf-1.3.9.orig/auth/auth.c serf-1.3.9/auth/auth.c
--- serf-1.3.9.orig/auth/auth.c	2017-05-20 12:10:45.093750000 +0800
+++ serf-1.3.9/auth/auth.c	2017-05-20 12:14:05.500000000 +0800
@@ -18,6 +18,7 @@
  * ====================================================================
  */
 
+#include <strings.h>
 #include "serf.h"
 #include "serf_private.h"
 #include "auth.h"
diff -durN serf-1.3.9.orig/auth/auth_basic.c serf-1.3.9/auth/auth_basic.c
--- serf-1.3.9.orig/auth/auth_basic.c	2017-05-20 12:10:45.093750000 +0800
+++ serf-1.3.9/auth/auth_basic.c	2017-05-20 12:14:35.671875000 +0800
@@ -20,6 +20,7 @@
 
 /*** Basic authentication ***/
 
+#include <strings.h>
 #include <serf.h>
 #include <serf_private.h>
 #include <auth/auth.h>
diff -durN serf-1.3.9.orig/auth/auth_spnego.c serf-1.3.9/auth/auth_spnego.c
--- serf-1.3.9.orig/auth/auth_spnego.c	2017-05-20 12:10:45.093750000 +0800
+++ serf-1.3.9/auth/auth_spnego.c	2017-05-20 12:15:05.406250000 +0800
@@ -28,6 +28,7 @@
  *  The HTTP message exchange is documented in RFC 4559.
  **/
 
+#include <strings.h>
 #include <serf.h>
 #include <serf_private.h>
 #include <auth/auth.h>
diff -durN serf-1.3.9.orig/buckets/headers_buckets.c serf-1.3.9/buckets/headers_buckets.c
--- serf-1.3.9.orig/buckets/headers_buckets.c	2017-05-20 12:10:44.937500000 +0800
+++ serf-1.3.9/buckets/headers_buckets.c	2017-05-20 12:15:56.562500000 +0800
@@ -21,6 +21,7 @@
 #include <stdlib.h>
 
 #include <apr_general.h>  /* for strcasecmp() */
+#include <strings.h>
 
 #include "serf.h"
 #include "serf_bucket_util.h"
diff -durN serf-1.3.9.orig/buckets/response_buckets.c serf-1.3.9/buckets/response_buckets.c
--- serf-1.3.9.orig/buckets/response_buckets.c	2017-05-20 12:10:44.968750000 +0800
+++ serf-1.3.9/buckets/response_buckets.c	2017-05-20 12:16:25.265625000 +0800
@@ -18,6 +18,7 @@
  * ====================================================================
  */
 
+#include <strings.h>
 #include <apr_lib.h>
 #include <apr_strings.h>
 #include <apr_date.h>
diff -durN serf-1.3.9.orig/outgoing.c serf-1.3.9/outgoing.c
--- serf-1.3.9.orig/outgoing.c	2017-05-20 12:10:45.000000000 +0800
+++ serf-1.3.9/outgoing.c	2017-05-20 12:16:52.593750000 +0800
@@ -18,6 +18,7 @@
  * ====================================================================
  */
 
+#include <strings.h>
 #include <apr_pools.h>
 #include <apr_poll.h>
 #include <apr_version.h>
