--- src/Makefile.orig	2022-01-19 23:03:40.000000000 +0100
+++ src/Makefile	2022-05-09 07:29:28.401456100 +0200
@@ -33,7 +33,7 @@
 #	make zonenames
 # to get a list of the values you can use for LOCALTIME.
 
-LOCALTIME=	Factory
+LOCALTIME=	Europe/Berlin
 
 # The POSIXRULES macro controls interpretation of nonstandard and obsolete
 # POSIX-like TZ settings like TZ='EET-2EEST' that lack DST transition rules.
@@ -111,7 +111,7 @@
 ZDUMPDIR = $(BINDIR)
 
 # The "zic" command goes in:
-ZICDIR = $(TOPDIR)/$(USRDIR)/sbin
+ZICDIR = $(TOPDIR)/$(USRDIR)/bin
 
 # Manual pages go in subdirectories of. . .
 MANDIR = $(TOPDIR)/$(USRSHAREDIR)/man
@@ -403,7 +403,7 @@
 # Also, it is better (though not essential) if 'awk' supports UTF-8,
 # and unfortunately mawk and busybox awk do not support UTF-8.
 # Try AWK=gawk or AWK=nawk if your awk has the abovementioned problems.
-AWK=		awk
+AWK=		gawk
 
 # The full path name of a Posix-compliant shell, preferably one that supports
 # the Korn shell's 'select' statement as an extension.
