--- src/Makefile.orig	2013-11-05 11:36:15.455800000 +0400
+++ src/Makefile	2013-11-05 11:40:42.433600000 +0400
@@ -350,9 +350,9 @@
 		-rm -f $(DESTDIR)$(TZDIR)/iso3166.tab \
 			$(DESTDIR)$(TZDIR)/zone.tab
 		cp iso3166.tab zone.tab $(DESTDIR)$(TZDIR)/.
-		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(ETCDIR)
-		cp tzselect zic zdump $(DESTDIR)$(ETCDIR)/.
-		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(MANDIR) \
+		-mkdir -p $(DESTDIR)$(TOPDIR) $(DESTDIR)$(ETCDIR)
+		cp tzselect zic.exe zdump.exe $(DESTDIR)$(ETCDIR)/.
+		-mkdir -p $(DESTDIR)$(TOPDIR) $(DESTDIR)$(MANDIR) \
 			$(DESTDIR)$(MANDIR)/man3 $(DESTDIR)$(MANDIR)/man5 \
 			$(DESTDIR)$(MANDIR)/man8
 		-rm -f $(DESTDIR)$(MANDIR)/man3/newctime.3 \
@@ -366,9 +366,9 @@
 		cp tzselect.8 zdump.8 zic.8 $(DESTDIR)$(MANDIR)/man8/.
 
 INSTALL:	ALL install date.1
-		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(BINDIR)
-		cp date $(DESTDIR)$(BINDIR)/.
-		-mkdir $(DESTDIR)$(TOPDIR) $(DESTDIR)$(MANDIR) \
+		-mkdir -p $(DESTDIR)$(TOPDIR) $(DESTDIR)$(BINDIR)
+		cp date.exe $(DESTDIR)$(BINDIR)/tz-date.exe
+		-mkdir -p $(DESTDIR)$(TOPDIR) $(DESTDIR)$(MANDIR) \
 			$(DESTDIR)$(MANDIR)/man1
 		-rm -f $(DESTDIR)$(MANDIR)/man1/date.1
 		cp date.1 $(DESTDIR)$(MANDIR)/man1/.
@@ -519,7 +519,7 @@
 check_public:	$(ENCHILADA)
 		make maintainer-clean
 		make "CFLAGS=$(GCC_DEBUG_FLAGS)" $(ENCHILADA) all
-		mkdir tzpublic
+		mkdir -p tzpublic
 		for i in $(TDATA) ; do \
 		  $(zic) -v -d tzpublic $$i 2>&1 || exit; \
 		done
@@ -529,10 +529,10 @@
 # Check that the code works under various alternative
 # implementations of time_t.
 check_time_t_alternatives:
-		mkdir tzpublic
+		mkdir -p tzpublic
 		zones=`$(AWK) '/^[^#]/ { print $$3 }' <zone.tab` && \
 		for type in $(TIME_T_ALTERNATIVES); do \
-		  mkdir tzpublic/$$type && \
+		  mkdir -p tzpublic/$$type && \
 		  make clean_misc && \
 		  make TOPDIR=`pwd`/tzpublic/$$type \
 		    CFLAGS='$(CFLAGS) -Dtime_tz='"'$$type'" \
