--- src/Makefile.orig	2018-01-17 18:05:36.000000000 +0900
+++ src/Makefile	2018-01-19 17:58:12.857248200 +0900
@@ -19,7 +19,7 @@
 #	make zonenames
 # to get a list of the values you can use for LOCALTIME.
 
-LOCALTIME=	GMT
+LOCALTIME=	Europe/Berlin
 
 # If you want something other than Eastern United States time as a template
 # for handling POSIX-style time zone environment variables,
@@ -88,7 +88,7 @@
 ZDUMPDIR = $(BINDIR)
 
 # The "zic" command goes in:
-ZICDIR = $(TOPDIR)/$(USRDIR)/sbin
+ZICDIR = $(TOPDIR)/$(USRDIR)/bin
 
 # Manual pages go in subdirectories of. . .
 MANDIR = $(TOPDIR)/$(USRSHAREDIR)/man
@@ -322,7 +322,7 @@
 # January's first Monday when a "%V" format is used and January 1
 # falls on a Friday, Saturday, or Sunday.
 
-CFLAGS=
+CFLAGS+= -DHAVE_ADJTIME=0
 
 # Linker flags.  Default to $(LFLAGS) for backwards compatibility
 # to release 2012h and earlier.
@@ -346,7 +346,7 @@
 ZIC_INSTALL=	$(ZIC) -d '$(DESTDIR)$(TZDIR)' $(LEAPSECONDS)
 
 # The name of a Posix-compliant 'awk' on your system.
-AWK=		awk
+AWK=		gawk
 
 # The full path name of a Posix-compliant shell, preferably one that supports
 # the Korn shell's 'select' statement as an extension.
@@ -425,7 +425,7 @@
 
 #MAKE=		make
 
-cc=		cc
+cc=		gcc
 CC=		$(cc) -DTZDIR='"$(TZDIR)"'
 
 AR=		ar
