--- src/Makefile.orig	2018-01-17 18:05:36.000000000 +0900
+++ src/Makefile	2018-01-19 17:58:12.857248200 +0900
@@ -33,7 +33,7 @@
 #	make zonenames
 # to get a list of the values you can use for LOCALTIME.
 
-LOCALTIME=	GMT
+LOCALTIME=	Europe/Berlin
 
 # If you want something other than Eastern United States time as a template
 # for handling POSIX-style time zone environment variables,
@@ -101,7 +101,7 @@
 ZDUMPDIR = $(BINDIR)
 
 # The "zic" command goes in:
-ZICDIR = $(TOPDIR)/$(USRDIR)/sbin
+ZICDIR = $(TOPDIR)/$(USRDIR)/bin
 
 # Manual pages go in subdirectories of. . .
 MANDIR = $(TOPDIR)/$(USRSHAREDIR)/man
@@ -345,7 +345,7 @@
 # January's first Monday when a "%V" format is used and January 1
 # falls on a Friday, Saturday, or Sunday.
 
-CFLAGS=
+CFLAGS+= -DHAVE_ADJTIME=0
 
 # Linker flags.  Default to $(LFLAGS) for backwards compatibility
 # to release 2012h and earlier.
@@ -375,7 +375,7 @@
 # Older 'mawk' versions, such as the 'mawk' in Ubuntu 16.04, might dump core;
 # on Ubuntu you can work around this with
 #	AWK=		gawk
-AWK=		awk
+AWK=		gawk
 
 # The full path name of a Posix-compliant shell, preferably one that supports
 # the Korn shell's 'select' statement as an extension.
@@ -469,7 +469,7 @@
 
 #MAKE=		make
 
-cc=		cc
+cc=		gcc
 CC=		$(cc) -DTZDIR='"$(TZDIR)"'
 
 AR=		ar
