--- src/Makefile.orig	2013-11-05 11:36:15.455800000 +0400
+++ src/Makefile	2013-11-05 11:40:42.433600000 +0400
@@ -350,7 +350,7 @@
 		$(ZIC) -y $(YEARISTYPE) \
 			-d $(DESTDIR)$(TZDIR) -l $(LOCALTIME) -p $(POSIXRULES)
 		cp -f iso3166.tab zone.tab $(DESTDIR)$(TZDIR)/.
-		cp tzselect zic zdump $(DESTDIR)$(ETCDIR)/.
+		cp tzselect zic.exe zdump.exe $(DESTDIR)$(ETCDIR)/.
 		cp libtz.a $(DESTDIR)$(LIBDIR)/.
 		$(RANLIB) $(DESTDIR)$(LIBDIR)/libtz.a
 		cp -f newctime.3 newtzset.3 $(DESTDIR)$(MANDIR)/man3/.
@@ -366,7 +366,7 @@
 
 INSTALL:	ALL install date.1
 		mkdir -p $(DESTDIR)$(BINDIR) $(DESTDIR)$(MANDIR)/man1
-		cp date $(DESTDIR)$(BINDIR)/.
+		cp date.exe $(DESTDIR)$(BINDIR)/.
 		cp -f date.1 $(DESTDIR)$(MANDIR)/man1/.
 
 version.h:
