--- src/Makefile.orig	2026-06-17 22:54:04.000000000 +0200
+++ src/Makefile	2026-07-15 07:12:08.927406100 +0200
@@ -64,7 +64,7 @@
 #	make zonenames
 # to get a list of the values you can use for LOCALTIME.
 
-LOCALTIME=	Factory
+LOCALTIME=	Europe/Berlin
 
 
 # Installation locations.
@@ -112,7 +112,7 @@
 ZDUMPDIR = $(BINDIR)
 
 # The "zic" command goes in:
-ZICDIR = $(TOPDIR)/$(USRDIR)/sbin
+ZICDIR = $(TOPDIR)/$(USRDIR)/bin
 
 # Manual pages go in subdirectories of:
 MANDIR = $(TOPDIR)/$(USRSHAREDIR)/man
@@ -535,7 +535,7 @@
 # Also, it is better (though not essential) if 'awk' supports UTF-8,
 # and unfortunately mawk and busybox awk do not support UTF-8.
 # Try AWK=gawk or AWK=nawk if your awk has the problems mentioned above.
-AWK=		awk
+AWK=		gawk
 
 # The full path name of a POSIX-compliant shell, preferably one that supports
 # the Korn shell's 'select' statement as an extension.
