From 97eaec05a18ff77e97706d2611c11dfd5e5928fc Mon Sep 17 00:00:00 2001
From: Jonathan Yong <10walls@gmail.com>
Date: Tue, 8 Sep 2020 15:24:34 +0200
Subject: [PATCH 13/18] Cygwin: newlib ftm

---
 gcc/glimits.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/glimits.h b/gcc/glimits.h
index a37f496ef1b..37ec2d039aa 100644
--- a/gcc/glimits.h
+++ b/gcc/glimits.h
@@ -111,7 +111,7 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 # define ULLONG_MAX (LLONG_MAX * 2ULL + 1ULL)
 #endif
 
-#if defined (__GNU_LIBRARY__) ? defined (__USE_GNU) : !defined (__STRICT_ANSI__)
+#if defined (__GNU_LIBRARY__) ? defined (__USE_GNU) : defined (__NEWLIB__) ? __GNU_VISIBLE : !defined (__STRICT_ANSI__)
 /* Minimum and maximum values a `signed long long int' can hold.  */
 # undef LONG_LONG_MIN
 # define LONG_LONG_MIN (-LONG_LONG_MAX - 1LL)
-- 
2.28.0

