From 189efb9574652865e808b98b77b618712a60539e Mon Sep 17 00:00:00 2001
From: Jonathan Yong <10walls@gmail.com>
Date: Tue, 8 Sep 2020 15:26:02 +0200
Subject: [PATCH 14/18] Cygwin: define STD_UNIX

---
 gcc/config/i386/cygwin.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/gcc/config/i386/cygwin.h b/gcc/config/i386/cygwin.h
index 13a7d57eb65..573df1c3cf0 100644
--- a/gcc/config/i386/cygwin.h
+++ b/gcc/config/i386/cygwin.h
@@ -24,14 +24,12 @@ along with GCC; see the file COPYING3.  If not see
       builtin_define ("__CYGWIN__");				\
       if (!TARGET_64BIT)					\
 	builtin_define ("__CYGWIN32__");			\
-      builtin_define ("__unix__");				\
-      builtin_define ("__unix");				\
+      builtin_define_std ("unix");				\
     }								\
   while (0)
 
 #undef CPP_SPEC
 #define CPP_SPEC "%(cpp_cpu) %{posix:-D_POSIX_SOURCE} \
-  %{!ansi:-Dunix} \
   %{pthread:-D_REENTRANT} \
   %{mwin32:-DWIN32 -D_WIN32 -D__WIN32 -D__WIN32__ %{!ansi:-DWINNT}} \
   %{!nostdinc:%{!mno-win32:-idirafter ../include/w32api%s -idirafter ../../include/w32api%s}}\
-- 
2.28.0

