From eac663bd31d1d34cea6a344d5a26fdf4ef12791b Mon Sep 17 00:00:00 2001
From: Jonathan Yong <10walls@gmail.com>
Date: Sun, 22 Mar 2020 09:57:42 +0800
Subject: [PATCH 16/18] libstdc++: regenerate src/Makefile.in for lt_host_flags

Signed-off-by: Jonathan Yong <10walls@gmail.com>
---
 libstdc++-v3/src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libstdc++-v3/src/Makefile.in b/libstdc++-v3/src/Makefile.in
index ea37a962a9a..00d863331e8 100644
--- a/libstdc++-v3/src/Makefile.in
+++ b/libstdc++-v3/src/Makefile.in
@@ -516,7 +516,7 @@ libstdc___la_DEPENDENCIES = \
 libstdc___la_LDFLAGS = \
 	-version-info $(libtool_VERSION) ${version_arg} -lm
 
-libstdc___la_LINK = $(CXXLINK) $(libstdc___la_LDFLAGS)
+libstdc___la_LINK = $(CXXLINK) $(libstdc___la_LDFLAGS) $(lt_host_flags)
 
 # A note on compatibility and static libraries.
 #
-- 
2.28.0

