From cc937b29a9b772240a9cfbb41428a08db66a91af Mon Sep 17 00:00:00 2001
From: Jonathan Yong <10walls@gmail.com>
Date: Sat, 19 Sep 2015 15:41:44 +0800
Subject: [PATCH 21/21] search /usr/lib/w32api explicitly

---
 gcc/config/i386/cygwin.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/config/i386/cygwin.h b/gcc/config/i386/cygwin.h
index 2a2a0bf..fd3bc0a 100644
--- a/gcc/config/i386/cygwin.h
+++ b/gcc/config/i386/cygwin.h
@@ -39,6 +39,7 @@ along with GCC; see the file COPYING3.  If not see
 
 #undef STARTFILE_SPEC
 #define STARTFILE_SPEC "\
+  -L%R/usr/lib/w32api \
   %{!shared: %{!mdll: crt0%O%s \
   %{pg:gcrt0%O%s}}}\
   %{shared:crtbeginS.o%s;:crtbegin.o%s} \
-- 
2.4.5

