--- gcc-5.3.0/config/mt-cygwin	1969-12-31 18:00:00.000000000 -0600
+++ gcc-5.3.0/config/mt-cygwin	2016-02-04 01:48:00.058340400 -0600
@@ -0,0 +1 @@
+CXXFLAGS_FOR_TARGET += -D_GNU_SOURCE
--- gcc-5.3.0/configure.ac	2015-05-03 12:29:57.000000000 -0500
+++ gcc-5.3.0/configure.ac	2016-02-03 17:34:52.238164400 -0600
@@ -2519,6 +2519,9 @@ case "${target}" in
   nios2-*-elf*)
     target_makefile_frag="config/mt-nios2-elf"
     ;;
+  *-*-cygwin*)
+    target_makefile_frag="config/mt-cygwin"
+    ;;
   *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu | *-*-kopensolaris*-gnu)
     target_makefile_frag="config/mt-gnu"
     ;;
--- gcc-5.3.0/configure	2015-05-03 12:29:57.000000000 -0500
+++ gcc-5.3.0/configure	2016-02-04 02:06:24.917639700 -0600
@@ -6873,6 +6873,9 @@ case "${target}" in
   nios2-*-elf*)
     target_makefile_frag="config/mt-nios2-elf"
     ;;
+  *-*-cygwin*)
+    target_makefile_frag="config/mt-cygwin"
+    ;;
   *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu | *-*-kopensolaris*-gnu)
     target_makefile_frag="config/mt-gnu"
     ;;
--- gcc-5.3.0/gcc/config/i386/cygwin.h	2016-02-03 16:49:24.261755400 -0600
+++ gcc-5.3.0/gcc/config/i386/cygwin.h	2016-02-04 02:04:29.163940800 -0600
@@ -105,6 +105,9 @@ along with GCC; see the file COPYING3.
   --wrap _ZdlPvRKSt9nothrow_t \
   --wrap _ZdaPvRKSt9nothrow_t \
 "
+/* The GNU C++ standard library requires that these macros be defined.  */
+#undef CPLUSPLUS_CPP_SPEC
+#define CPLUSPLUS_CPP_SPEC "-D_GNU_SOURCE %(cpp)"
 
 #if defined (USE_CYGWIN_LIBSTDCXX_WRAPPERS)
 
