This was just a bit outdated...

--- origsrc/gcc-5.3.0/libstdc++-v3/acinclude.m4	2015-11-24 07:25:07.000000000 -0600
+++ src/gcc-5.3.0/libstdc++-v3/acinclude.m4	2016-04-01 00:21:40.077426000 -0500
@@ -1196,7 +1196,10 @@ AC_DEFUN([GLIBCXX_ENABLE_LIBSTDCXX_TIME]
 
     case "${target_os}" in
       cygwin*)
+        ac_has_clock_monotonic=yes
+        ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
+        ac_has_sched_yield=yes
         ;;
       darwin*)
         ac_has_nanosleep=yes
--- origsrc/gcc-5.3.0/libstdc++-v3/configure	2016-03-31 21:24:38.558164300 -0500
+++ src/gcc-5.3.0/libstdc++-v3/configure	2016-04-01 00:26:14.110723800 -0500
@@ -19632,7 +19632,10 @@ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
 
     case "${target_os}" in
       cygwin*)
+        ac_has_clock_monotonic=yes
+        ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
+        ac_has_sched_yield=yes
         ;;
       darwin*)
         ac_has_nanosleep=yes
