diff -Naur gcc-4.9.2-orig/libffi/src/x86/ffi.c gcc-4.9.2/libffi/src/x86/ffi.c
--- gcc-4.9.2-orig/libffi/src/x86/ffi.c	2012-12-26 19:28:02.000000000 +0300
+++ gcc-4.9.2/libffi/src/x86/ffi.c	2014-11-04 00:25:28.993000000 +0300
@@ -28,7 +28,7 @@
    DEALINGS IN THE SOFTWARE.
    ----------------------------------------------------------------------- */
 
-#if !defined(__x86_64__) || defined(_WIN64)
+#if !defined(__x86_64__) || defined(_WIN64) || defined(__CYGWIN__)
 
 #ifdef _WIN64
 #include <windows.h>
