--- gcc/graphite-clast-to-gimple.c.orig	2014-03-03 11:39:22.000000000 +0000
+++ gcc/graphite-clast-to-gimple.c
@@ -28,6 +28,7 @@ along with GCC; see the file COPYING3.
 #include <isl/constraint.h>
 #include <isl/ilp.h>
 #include <isl/aff.h>
+#include <isl/deprecated/ilp_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
--- gcc/graphite-interchange.c.orig	2014-01-02 22:23:26.000000000 +0000
+++ gcc/graphite-interchange.c
@@ -28,7 +28,8 @@ along with GCC; see the file COPYING3.
 #include <isl/set.h>
 #include <isl/map.h>
 #include <isl/union_map.h>
-#include <isl/ilp.h>
+#include <isl/deprecated/ilp_int.h>
+#include <isl/deprecated/constraint_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
--- gcc/graphite-optimize-isl.c.orig	2014-01-02 22:23:26.000000000 +0000
+++ gcc/graphite-optimize-isl.c
@@ -28,6 +28,7 @@ along with GCC; see the file COPYING3.
 #include <isl/band.h>
 #include <isl/aff.h>
 #include <isl/options.h>
+#include <isl/deprecated/aff_int.h>
 #endif
 
 #include "system.h"
@@ -373,7 +374,7 @@ getScheduleForBandList (isl_band_list *B
 	{
 	  for (i = ScheduleDimensions - 1 ;  i >= 0 ; i--)
 	    {
-	      if (isl_band_member_is_zero_distance (Band, i))
+	      if (isl_band_member_is_coincident (Band, i))
 		{
 		  isl_map *TileMap;
 		  isl_union_map *TileUMap;
@@ -448,6 +449,7 @@ optimize_isl (scop_p scop)
   isl_union_set *domain;
   isl_union_map *validity, *proximity, *dependences;
   isl_union_map *schedule_map;
+  isl_schedule_constraints *ScheduleConstraints;
 
   domain = scop_get_domains (scop);
   dependences = scop_get_dependences (scop);
@@ -463,7 +465,15 @@ optimize_isl (scop_p scop)
   isl_options_set_schedule_maximize_band_depth (scop->ctx, 1);
   isl_options_set_schedule_fuse (scop->ctx, ISL_SCHEDULE_FUSE_MIN);
   isl_options_set_on_error (scop->ctx, ISL_ON_ERROR_CONTINUE);
-  schedule = isl_union_set_compute_schedule (domain, validity, proximity);
+
+  ScheduleConstraints = isl_schedule_constraints_on_domain(domain);
+  ScheduleConstraints =
+      isl_schedule_constraints_set_proximity(ScheduleConstraints, proximity);
+  ScheduleConstraints = isl_schedule_constraints_set_validity(
+      ScheduleConstraints, isl_union_map_copy(validity));
+  ScheduleConstraints =
+      isl_schedule_constraints_set_coincidence(ScheduleConstraints, validity);
+  schedule = isl_schedule_constraints_compute_schedule(ScheduleConstraints);
   isl_options_set_on_error (scop->ctx, ISL_ON_ERROR_ABORT);
 
   if (!schedule)
--- gcc/graphite-poly.c.orig	2014-01-02 22:23:26.000000000 +0000
+++ gcc/graphite-poly.c
@@ -28,6 +28,7 @@ along with GCC; see the file COPYING3.
 #include <isl/constraint.h>
 #include <isl/ilp.h>
 #include <isl/aff.h>
+#include <isl/deprecated/ilp_int.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
 #endif
--- gcc/graphite-sese-to-poly.c.orig	2014-04-08 10:59:40.000000000 +0000
+++ gcc/graphite-sese-to-poly.c
@@ -26,6 +26,8 @@ along with GCC; see the file COPYING3.
 #include <isl/union_map.h>
 #include <isl/constraint.h>
 #include <isl/aff.h>
+#include <isl/deprecated/aff_int.h>
+#include <isl/deprecated/constraint_int.h>
 #include <cloog/cloog.h>
 #include <cloog/cloog.h>
 #include <cloog/isl/domain.h>
