--- origsrc/gcc-4.5.0/gcc/testsuite/gcc.target/i386/pr25993.c	2008-02-26 01:38:34.000000000 +0000
+++ src/gcc-4.5.0/gcc/testsuite/gcc.target/i386/pr25993.c	2010-04-25 16:17:21.703125000 +0100
@@ -1,5 +1,5 @@
 /* { dg-do assemble } */
-/* { dg-skip-if "" { "*-*-darwin*" "*-*-mingw*" } { "*" } { "" } } */
+/* { dg-skip-if "" { "*-*-darwin*" "*-*-mingw*" "*-*-cygwin*" } { "*" } { "" } } */
 /* { dg-options "-std=c99 -x assembler-with-cpp" } */
 
 #ifndef __ASSEMBLER__
