--- origsrc/gcc-4.5.0/libjava/classpath/configure.ac	2009-02-05 17:26:08.000000000 -0600
+++ src/gcc-4.5.0/libjava/classpath/configure.ac	2009-12-06 01:58:25.091013300 -0600
@@ -26,7 +26,7 @@ case "$host_os" in
 	cp_module=""
 	;;
 	*)
-	cp_module="-module"
+	cp_module="-module -avoid-version"
 	;;
 esac
 
--- origsrc/gcc-4.7.2/libjava/classpath/configure	2011-11-29 14:12:00.000000000 -0600
+++ src/gcc-4.7.2/libjava/classpath/configure	2013-03-01 01:48:05.585762700 -0600
@@ -3172,7 +3172,7 @@ case "$host_os" in
 	cp_module=""
 	;;
 	*)
-	cp_module="-module"
+	cp_module="-module -avoid-version"
 	;;
 esac
 
--- origsrc/gcc-4.7.2/libjava/include/posix.h	2011-01-03 14:52:22.000000000 -0600
+++ src/gcc-4.7.2/libjava/include/posix.h	2013-03-06 20:26:16.157459700 -0600
@@ -47,11 +47,17 @@ details.  */
 #include <java/util/Properties.h>
 
 // Prefix and suffix for shared libraries.
+#ifdef __CYGWIN__
+#define _Jv_platform_solib_prefix "cyg"
+#else
 #define _Jv_platform_solib_prefix "lib"
+#endif
 #if defined(__APPLE__) && defined(__MACH__)
 #define _Jv_platform_solib_suffix ".dylib"
 #elif defined(HPUX) && defined(HP_PA)
 #define _Jv_platform_solib_suffix ".sl"
+#elif defined(__CYGWIN__)
+#define _Jv_platform_solib_suffix ".dll"
 #else
 #define _Jv_platform_solib_suffix ".so"
 #endif
--- origsrc/gcc-4.7.2/libjava/java/net/natVMURLConnection.cc	2009-12-01 16:05:25.000000000 -0600
+++ src/gcc-4.7.2/libjava/java/net/natVMURLConnection.cc	2013-03-06 20:20:42.202970700 -0600
@@ -32,7 +32,11 @@ void
 java::net::VMURLConnection::init ()
 {
 #if defined (HAVE_MAGIC_T) && defined (HAVE_MAGIC_H) && defined (USE_LTDL)
+#ifdef __CYGWIN__
+  lt_dlhandle handle = lt_dlopenext ("cygmagic-1.dll");
+#else
   lt_dlhandle handle = lt_dlopenext ("libmagic.so");
+#endif
   if (!handle)
     return;
 
