--- origsrc/gcc-4.8-20130310/config/picflag.m4	2012-01-23 00:25:28.000000000 -0600
+++ src/gcc-4.8-20130310/config/picflag.m4	2013-03-17 03:30:20.891423200 -0500
@@ -17,7 +17,9 @@ case "${$2}" in
     hppa*64*-*-hpux*)
 	# PIC is the default for 64-bit PA HP-UX.
 	;;
-    i[[34567]]86-*-cygwin* | i[[34567]]86-*-mingw* | x86_64-*-mingw*)
+    i[[34567]]86-*-cygwin* | x86_64-*-cygwin*)
+	;;
+    i[[34567]]86-*-mingw* | x86_64-*-mingw*)
 	;;
     i[[34567]]86-*-interix[[3-9]]*)
 	# Interix 3.x gcc -fpic/-fPIC options generate broken code.
--- origsrc/gcc-4.8-20130310/gcc/configure	2013-03-17 03:47:51.178944300 -0500
+++ src/gcc-4.8-20130310/gcc/configure	2013-03-17 11:48:48.692743100 -0500
@@ -4893,7 +4893,9 @@ case "${target}" in
     hppa*64*-*-hpux*)
 	# PIC is the default for 64-bit PA HP-UX.
 	;;
-    i[34567]86-*-cygwin* | i[34567]86-*-mingw* | x86_64-*-mingw*)
+    i[34567]86-*-cygwin* | x86_64-*-cygwin*)
+	;;
+    i[34567]86-*-mingw* | x86_64-*-mingw*)
 	;;
     i[34567]86-*-interix[3-9]*)
 	# Interix 3.x gcc -fpic/-fPIC options generate broken code.
--- origsrc/gcc-4.8-20130310/libada/configure	2013-03-10 17:37:35.000000000 -0500
+++ src/gcc-4.8-20130310/libada/configure	2013-03-17 03:35:42.763922700 -0500
@@ -2834,7 +2834,9 @@ case "${host}" in
     hppa*64*-*-hpux*)
 	# PIC is the default for 64-bit PA HP-UX.
 	;;
-    i[34567]86-*-cygwin* | i[34567]86-*-mingw* | x86_64-*-mingw*)
+    i[34567]86-*-cygwin* | x86_64-*-cygwin*)
+	;;
+    i[34567]86-*-mingw* | x86_64-*-mingw*)
 	;;
     i[34567]86-*-interix[3-9]*)
 	# Interix 3.x gcc -fpic/-fPIC options generate broken code.
--- origsrc/gcc-4.8-20130310/libgcc/configure	2013-03-17 03:33:36.033747800 -0500
+++ src/gcc-4.8-20130310/libgcc/configure	2013-03-17 03:35:57.173942600 -0500
@@ -2228,7 +2228,9 @@ case "${host}" in
     hppa*64*-*-hpux*)
 	# PIC is the default for 64-bit PA HP-UX.
 	;;
-    i[34567]86-*-cygwin* | i[34567]86-*-mingw* | x86_64-*-mingw*)
+    i[34567]86-*-cygwin* | x86_64-*-cygwin*)
+	;;
+    i[34567]86-*-mingw* | x86_64-*-mingw*)
 	;;
     i[34567]86-*-interix[3-9]*)
 	# Interix 3.x gcc -fpic/-fPIC options generate broken code.
--- origsrc/gcc-4.8-20130310/libiberty/configure	2012-09-18 11:03:01.000000000 -0500
+++ src/gcc-4.8-20130310/libiberty/configure	2013-03-17 03:36:10.983961700 -0500
@@ -4865,7 +4865,9 @@ case "${host}" in
     hppa*64*-*-hpux*)
 	# PIC is the default for 64-bit PA HP-UX.
 	;;
-    i[34567]86-*-cygwin* | i[34567]86-*-mingw* | x86_64-*-mingw*)
+    i[34567]86-*-cygwin* | x86_64-*-cygwin*)
+	;;
+    i[34567]86-*-mingw* | x86_64-*-mingw*)
 	;;
     i[34567]86-*-interix[3-9]*)
 	# Interix 3.x gcc -fpic/-fPIC options generate broken code.
