diff -Naur libidn-1.26-orig/build-aux/compile libidn-1.26/build-aux/compile
--- libidn-1.26-orig/build-aux/compile	2012-12-11 15:12:08.000000000 +0000
+++ libidn-1.26/build-aux/compile	2013-04-25 11:01:39.880859400 +0000
@@ -54,7 +54,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -68,7 +68,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur libidn-1.35-orig/build-aux/config.guess libidn-1.35/build-aux/config.guess
--- libidn-1.35-orig/build-aux/config.guess	2012-12-11 15:12:08.000000000 +0000
+++ libidn-1.35/build-aux/config.guess	2013-04-25 11:02:03.271484400 +0000
@@ -870,6 +870,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2`echo ${UNAME_RELEASE}|sed -e 's/[^.]*//'`
 	exit ;;
diff -Naur libidn-1.26-orig/build-aux/config.rpath libidn-1.26/build-aux/config.rpath
--- libidn-1.26-orig/build-aux/config.rpath	2012-11-22 09:40:21.000000000 +0000
+++ libidn-1.26/build-aux/config.rpath	2013-04-25 11:02:37.177734400 +0000
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur libidn-1.26-orig/gl/m4/csharpexec.m4 libidn-1.26/gl/m4/csharpexec.m4
--- libidn-1.26-orig/gl/m4/csharpexec.m4	2012-04-27 08:18:42.000000000 +0000
+++ libidn-1.26/gl/m4/csharpexec.m4	2013-04-25 11:08:35.791992200 +0000
@@ -53,7 +53,7 @@
            ifelse([$1], , , [&& clix $2/$1 >/dev/null 2>/dev/null]); then
           HAVE_CLIX=1
           case $host_os in
-            cygwin* | mingw* | pw32*)
+            cygwin* | msys* |mingw* | pw32*)
               CLIX_PATH_VAR=PATH
               ;;
             darwin* | rhapsody*)
diff -Naur libidn-1.26-orig/gl/m4/dup2.m4 libidn-1.26/gl/m4/dup2.m4
--- libidn-1.26-orig/gl/m4/dup2.m4	2012-05-22 17:55:20.000000000 +0000
+++ libidn-1.26/gl/m4/dup2.m4	2013-04-25 11:09:03.147460900 +0000
@@ -68,7 +68,7 @@
         [case "$host_os" in
            mingw*) # on this platform, dup2 always returns 0 for success
              gl_cv_func_dup2_works="guessing no" ;;
-           cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+           cygwin*|msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
            aix* | freebsd*)
                    # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
diff -Naur libidn-1.26-orig/gl/m4/getcwd.m4 libidn-1.26/gl/m4/getcwd.m4
--- libidn-1.26-orig/gl/m4/getcwd.m4	2012-05-22 17:55:20.000000000 +0000
+++ libidn-1.26/gl/m4/getcwd.m4	2013-04-25 11:09:32.528320300 +0000
@@ -50,6 +50,8 @@
             *-gnu* | gnu*) gl_cv_func_getcwd_null="guessing yes";;
                            # Guess yes on Cygwin.
             cygwin*)       gl_cv_func_getcwd_null="guessing yes";;
+                           # Guess yes on MSYS.
+            msys*)         gl_cv_func_getcwd_null="guessing yes";;
                            # If we don't know, assume the worst.
             *)             gl_cv_func_getcwd_null="guessing no";;
           esac
diff -Naur libidn-1.26-orig/gl/m4/malloc.m4 libidn-1.26/gl/m4/malloc.m4
--- libidn-1.26-orig/gl/m4/malloc.m4	2012-05-22 17:55:20.000000000 +0000
+++ libidn-1.26/gl/m4/malloc.m4	2013-04-25 11:10:01.426757800 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur libidn-1.26-orig/lib/gl/m4/setlocale.m4 libidn-1.26/lib/gl/m4/setlocale.m4
--- libidn-1.26-orig/lib/gl/m4/setlocale.m4	2012-04-27 08:18:42.000000000 +0000
+++ libidn-1.26/lib/gl/m4/setlocale.m4	2013-04-25 11:12:28.392578100 +0000
@@ -14,7 +14,7 @@
     mingw*) REPLACE_SETLOCALE=1 ;;
     dnl On Cygwin 1.5.x, setlocale always succeeds but setlocale(LC_CTYPE,NULL)
     dnl is then still "C".
-    cygwin*)
+    cygwin* | msys*)
       case `uname -r` in
         1.5.*) REPLACE_SETLOCALE=1 ;;
       esac
diff -Naur libidn-1.26-orig/m4/threadlib.m4 libidn-1.26/m4/threadlib.m4
--- libidn-1.26-orig/m4/threadlib.m4	2012-12-11 15:11:52.000000000 +0000
+++ libidn-1.26/m4/threadlib.m4	2013-04-25 11:17:48.372070300 +0000
@@ -61,7 +61,7 @@
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
