diff -Naur YAML-0.84/inc/Module/Install/Can.pm YAML-0.84/inc/Module/Install/Can.pm
--- YAML-0.84/inc/Module/Install/Can.pm	2012-07-13 18:18:19.000000000 +0000
+++ YAML-0.84/inc/Module/Install/Can.pm	2013-06-16 16:04:01.880437500 +0000
@@ -132,7 +132,7 @@
 }
 
 # Fix Cygwin bug on maybe_command();
-if ( $^O eq 'cygwin' ) {
+if ( $^O eq 'cygwin' or $^O eq 'msys' ) {
 	require ExtUtils::MM_Cygwin;
 	require ExtUtils::MM_Win32;
 	if ( ! defined(&ExtUtils::MM_Cygwin::maybe_command) ) {
