diff --git a/makefile b/makefile
index 2d1fd8c..cf2e915 100644
--- a/makefile
+++ b/makefile
@@ -56,7 +56,6 @@ depend:
 	$(MAKE) -C CPP/7zip/UI/Client7z           depend
 	$(MAKE) -C CPP/7zip/UI/Console            depend
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree  depend
-	$(MAKE) -C CPP/7zip/Compress/Rar          depend
 	$(MAKE) -C check/my_86_filter             depend
 	$(MAKE) -C CPP/7zip/UI/GUI                depend
 	$(MAKE) -C CPP/7zip/UI/FileManager        depend
@@ -68,7 +67,6 @@ sfx: common
 common7z:common
 	$(MKDIR) bin/Codecs
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree all
-	$(MAKE) -C CPP/7zip/Compress/Rar         all
 
 7z: common7z
 	$(MAKE) -C CPP/7zip/UI/Console           all
@@ -93,7 +91,6 @@ clean:
 	$(MAKE) -C CPP/7zip/UI/FileManager       clean
 	$(MAKE) -C CPP/7zip/UI/GUI               clean
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree clean
-	$(MAKE) -C CPP/7zip/Compress/Rar         clean
 	$(MAKE) -C CPP/7zip/Bundles/LzmaCon      clean2
 	$(MAKE) -C CPP/7zip/Bundles/AloneGCOV    clean
 	$(MAKE) -C CPP/7zip/TEST/TestUI          clean
diff --git a/makefile.machine b/makefile.machine
index 6326a51..e9d330a 100644
--- a/makefile.machine
+++ b/makefile.machine
@@ -4,13 +4,13 @@
 
 OPTFLAGS=-O
 
-ALLFLAGS=${OPTFLAGS} -pipe -s \
+ALLFLAGS=${OPTFLAGS} -pipe \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	-D_7ZIP_LARGE_PAGES \
 	$(LOCAL_FLAGS)
 
-CXX=g++
+CXX=g++ -DHAVE_GCCVISIBILITYPATCH -fvisibility=hidden -fvisibility-inlines-hidden
 CC=gcc
 CC_SHARED=-fPIC
 LINK_SHARED=-fPIC -shared
