diff -urN p7zip_9.20.1/makefile.orig p7zip_9.20.1/makefile
--- p7zip_9.20.1/makefile.orig	2011-03-13 07:52:45.000000000 -0400
+++ p7zip_9.20.1/makefile	2011-06-26 16:25:33.265000000 -0400
@@ -56,7 +56,6 @@
 	$(MAKE) -C CPP/7zip/UI/Client7z           depend
 	$(MAKE) -C CPP/7zip/UI/Console            depend
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree  depend
-	$(MAKE) -C CPP/7zip/Compress/Rar          depend
 	$(MAKE) -C CPP/7zip/UI/GUI                depend
 	$(MAKE) -C CPP/7zip/UI/FileManager        depend
 	$(MAKE) -C check/my_86_filter             depend
@@ -68,7 +67,6 @@
 common7z:common
 	$(MKDIR) bin/Codecs
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree all
-	$(MAKE) -C CPP/7zip/Compress/Rar         all
 
 7z: common7z
 	$(MAKE) -C CPP/7zip/UI/Console           all
@@ -93,7 +91,6 @@
 	$(MAKE) -C CPP/7zip/UI/FileManager       clean
 	$(MAKE) -C CPP/7zip/UI/GUI               clean
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree clean
-	$(MAKE) -C CPP/7zip/Compress/Rar         clean
 	$(MAKE) -C CPP/7zip/Compress/LZMA_Alone  clean
 	$(MAKE) -C CPP/7zip/Bundles/AloneGCOV    clean
 	$(MAKE) -C CPP/7zip/TEST/TestUI          clean
diff -urN p7zip_9.20.1/makefile.machine.orig p7zip_9.20.1/makefile.machine
--- p7zip_9.20.1/makefile.machine.orig	2011-03-13 07:54:57.000000000 -0400
+++ p7zip_9.20.1/makefile.machine	2011-06-26 16:26:33.595000000 -0400
@@ -4,13 +4,13 @@
 
 OPTFLAGS=-O
 
-ALLFLAGS=${OPTFLAGS} -pipe -s \
+ALLFLAGS=${OPTFLAGS} -pipe \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	-D_7ZIP_LARGE_PAGES \
 	$(LOCAL_FLAGS)
 
-CXX=g++ $(ALLFLAGS)
+CXX=g++ $(ALLFLAGS) -DHAVE_GCCVISIBILITYPATCH -fvisibility=hidden -fvisibility-inlines-hidden
 CC=gcc $(ALLFLAGS)
 CC_SHARED=-fPIC
 LINK_SHARED=-fPIC -shared
