diff -urN p7zip_16.02/makefile.orig p7zip_16.02/makefile
--- p7zip_16.02/makefile.orig	2017-04-20 16:55:35.041001600 +0900
+++ p7zip_16.02/makefile	2017-04-20 16:59:15.262613900 +0900
@@ -31,7 +31,6 @@
 	$(MAKE) -C CPP/7zip/UI/Client7z           depend
 	$(MAKE) -C CPP/7zip/UI/Console            depend
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree  depend
-	$(MAKE) -C CPP/7zip/Compress/Rar          depend
 	$(MAKE) -C CPP/7zip/UI/GUI                depend
 	$(MAKE) -C CPP/7zip/UI/FileManager        depend
 
@@ -42,7 +41,6 @@
 common7z:common
 	$(MKDIR) bin/Codecs
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree all
-	$(MAKE) -C CPP/7zip/Compress/Rar         all
 
 lzham:common
 	$(MKDIR) bin/Codecs
@@ -67,7 +65,6 @@
 	$(MAKE) -C CPP/7zip/UI/FileManager       clean
 	$(MAKE) -C CPP/7zip/UI/GUI               clean
 	$(MAKE) -C CPP/7zip/Bundles/Format7zFree clean
-	$(MAKE) -C CPP/7zip/Compress/Rar         clean
 	$(MAKE) -C CPP/7zip/Compress/Lzham       clean
 	$(MAKE) -C CPP/7zip/Bundles/LzmaCon      clean2
 	$(MAKE) -C CPP/7zip/Bundles/AloneGCOV    clean
diff -urN p7zip_16.02/makefile.machine.orig p7zip_16.02/makefile.machine
--- p7zip_16.02/makefile.machine.orig   2017-04-20 16:34:26.212102400 +0900
+++ p7zip_16.02/makefile.machine        2017-04-20 16:46:23.502622500 +0900
@@ -10,7 +10,7 @@
        -D_7ZIP_LARGE_PAGES \
        $(LOCAL_FLAGS)

-CXX=g++
+CXX=g++ -DHAVE_GCCVISIBILITYPATCH -fvisibility=hidden -fvisibility-inlines-hidden
 CC=gcc
 CC_SHARED=-fPIC
 LINK_SHARED=-fPIC -shared
