diff --git a/CPP/myWindows/myAddExeFlag.cpp b/CPP/myWindows/myAddExeFlag.cpp
index 6bdc34a..4e49ff3 100644
--- a/CPP/myWindows/myAddExeFlag.cpp
+++ b/CPP/myWindows/myAddExeFlag.cpp
@@ -12,9 +12,14 @@
 
 void myAddExeFlag(const UString &u_name)
 {
+	struct stat fileinfo;
 	AString filename = UnicodeStringToMultiByte(u_name, CP_ACP); // FIXME
 	const char * name = nameWindowToUnix(filename);
 	// printf("myAddExeFlag(%s)\n",name);
-	chmod(name,0700);
+
+	// should detect error, but this interface gives no capability to do so
+	stat(name, &fileinfo);
+	mode_t filemode = fileinfo.st_mode;
+	chmod(name, filemode | S_IXUSR | S_IXGRP | S_IXOTH);
 }
 
