diff --git a/CPP/Common/StringConvert.cpp b/CPP/Common/StringConvert.cpp
index e87da8d..a4a3036 100644
--- a/CPP/Common/StringConvert.cpp
+++ b/CPP/Common/StringConvert.cpp
@@ -179,7 +179,7 @@ AString UnicodeStringToMultiByte(const UString &src, UINT /* codePage */ )
     }
   }
 #else
-  const UString &srcString = src;
+  UString srcString(src);
 #endif
 
   if ((global_use_utf16_conversion) && (!srcString.IsEmpty()))
