diff --git a/CPP/Common/StringConvert.cpp b/CPP/Common/StringConvert.cpp
index fe3ff42..038120c 100644
--- a/CPP/Common/StringConvert.cpp
+++ b/CPP/Common/StringConvert.cpp
@@ -178,7 +178,7 @@ AString UnicodeStringToMultiByte(const UString &src, UINT /* codePage */ )
     }
   }
 #else
-  UString &srcString = src;
+  UString srcString(src);
 #endif
 
   if ((global_use_utf16_conversion) && (!srcString.IsEmpty()))
