diff --git a/contrib/gzip-like_CLI_wrapper_for_7z/p7zip b/contrib/gzip-like_CLI_wrapper_for_7z/p7zip
index 0b9f19b..e290faf 100755
--- a/contrib/gzip-like_CLI_wrapper_for_7z/p7zip
+++ b/contrib/gzip-like_CLI_wrapper_for_7z/p7zip
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # gzip-like CLI wrapper for p7zip
 # version 2.0
 #
@@ -65,7 +65,7 @@ doit()
   trap "rm -f -- ${tmp}" 0
 
   if ${compress} ; then
-    if tty > /dev/null ; then
+    if tty -s <&1 > /dev/null ; then
       echo "$0: compressed data not written to a terminal."
       echo "For help, type: $0 -h"
       exit 1
@@ -120,6 +120,9 @@ while [ "$#" != "0" ] ; do
       shift
       break
       ;;
+    -*)
+      usage
+      ;;
     *)
 	doit ${compress} "$1"
         file="$1"
diff --git a/install.sh b/install.sh
index b3b39d7..81edb40 100755
--- a/install.sh
+++ b/install.sh
@@ -3,28 +3,29 @@
 installShared()
 {
   prg=$1
-  if [ -x bin/${prg} ]
+  if [ -x bin/${prg}${EXEEXT} ]
   then
     echo "- installing ${DEST_DIR}${DEST_BIN}/${prg}"
-    cp bin/${prg} "${DEST_DIR}${DEST_SHARE}/${prg}"
-    chmod 777 "${DEST_DIR}${DEST_SHARE}/${prg}"
-    strip     "${DEST_DIR}${DEST_SHARE}/${prg}"
-    chmod 555 "${DEST_DIR}${DEST_SHARE}/${prg}"
+    cp bin/${prg} "${DEST_DIR}${DEST_SHARE}/${prg}${EXEEXT}"
+    chmod 777 "${DEST_DIR}${DEST_SHARE}/${prg}${EXEEXT}"
+    strip     "${DEST_DIR}${DEST_SHARE}/${prg}${EXEEXT}"
+    chmod 555 "${DEST_DIR}${DEST_SHARE}/${prg}${EXEEXT}"
     echo "#! /bin/sh" > "${DEST_DIR}${DEST_BIN}/${prg}"
-    echo "\"${DEST_SHARE}/${prg}\" \"\$@\"" >> "${DEST_DIR}${DEST_BIN}/${prg}"
+    echo "\"${DEST_SHARE}/${prg}${EXEEXT}\" \"\$@\"" >> "${DEST_DIR}${DEST_BIN}/${prg}"
     chmod 555 "${DEST_DIR}${DEST_BIN}/${prg}"
   fi
 }
 
 # global install
-DEST_HOME=/usr/local
+DEST_HOME=/usr
 # for testing 
 # DEST_HOME=${HOME}/INSTALL/usr/local
 DEST_BIN=${DEST_HOME}/bin
 DEST_SHARE=${DEST_HOME}/lib/p7zip
-DEST_MAN=${DEST_HOME}/man
+DEST_MAN=${DEST_HOME}/share/man
 DEST_SHARE_DOC=${DEST_HOME}/share/doc/p7zip
 DEST_DIR=
+EXEEXT=
 [ "$1" ] && DEST_BIN="$1"
 [ "$2" ] && DEST_SHARE="$2"
 [ "$3" ] && DEST_MAN="$3"
@@ -38,34 +39,34 @@ then
   use_share="o"
 fi
 
-if [ -x bin/7z ]
+if [ -x bin/7z${EXEEXT} ]
 then
   use_share="o"
 fi
 
-if [ -x bin/7zG ]
+if [ -x bin/7zG${EXEEXT} ]
 then
   use_share="o"
 fi
 
-if [ -x bin/7zFM ]
+if [ -x bin/7zFM${EXEEXT} ]
 then
   use_share="o"
 fi
 
 # cleaning previous install
-rm -f  "${DEST_DIR}${DEST_BIN}/7z"
-rm -f  "${DEST_DIR}${DEST_BIN}/7za"
-rm -f  "${DEST_DIR}${DEST_BIN}/7zr"
-rm -f  "${DEST_DIR}${DEST_BIN}/7zG"
-rm -f  "${DEST_DIR}${DEST_BIN}/p7zipForFilemanager"
-rm -f  "${DEST_DIR}${DEST_BIN}/7zFM"
-rm -f  "${DEST_DIR}${DEST_SHARE}/7z"
-rm -f  "${DEST_DIR}${DEST_SHARE}/7za"
+rm -f  "${DEST_DIR}${DEST_BIN}/7z${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_BIN}/7za${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_BIN}/7zr${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_BIN}/7zG${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_BIN}/p7zipForFilemanager${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_BIN}/7zFM${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_SHARE}/7z${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_SHARE}/7za${EXEEXT}"
 rm -f  "${DEST_DIR}${DEST_SHARE}/7zCon.sfx"
-rm -f  "${DEST_DIR}${DEST_SHARE}/7zr"
-rm -f  "${DEST_DIR}${DEST_SHARE}/7zG"
-rm -f  "${DEST_DIR}${DEST_SHARE}/7zFM"
+rm -f  "${DEST_DIR}${DEST_SHARE}/7zr${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_SHARE}/7zG${EXEEXT}"
+rm -f  "${DEST_DIR}${DEST_SHARE}/7zFM${EXEEXT}"
 rm -f  "${DEST_DIR}${DEST_SHARE}/7z.so"
 chmod -R 777 "${DEST_DIR}${DEST_SHARE}/Codecs" 2> /dev/null
 chmod -R 777 "${DEST_DIR}${DEST_SHARE}/Formats" 2> /dev/null
@@ -112,20 +113,20 @@ then
   fi
 
 else
-  if [ -x bin/7za ]
+  if [ -x bin/7za${EXEEXT} ]
   then
     echo "- installing ${DEST_DIR}${DEST_BIN}/7za"
     mkdir -p "${DEST_DIR}${DEST_BIN}"
-    cp bin/7za "${DEST_DIR}${DEST_BIN}/7za"
-    chmod 555  "${DEST_DIR}${DEST_BIN}/7za"
+    cp bin/7za${EXEEXT} "${DEST_DIR}${DEST_BIN}/7za${EXEEXT}"
+    chmod 555  "${DEST_DIR}${DEST_BIN}/7za${EXEEXT}"
   fi
 
-  if [ -x bin/7zr ]
+  if [ -x bin/7zr${EXEEXT} ]
   then
     echo "- installing ${DEST_DIR}${DEST_BIN}/7zr"
     mkdir -p "${DEST_DIR}${DEST_BIN}"
-    cp bin/7zr "${DEST_DIR}${DEST_BIN}/7zr"
-    chmod 555  "${DEST_DIR}${DEST_BIN}/7zr"
+    cp bin/7zr${EXEEXT} "${DEST_DIR}${DEST_BIN}/7zr${EXEEXT}"
+    chmod 555  "${DEST_DIR}${DEST_BIN}/7zr${EXEEXT}"
   fi
 fi
 
@@ -134,27 +135,27 @@ if [ -d DOC ]
 then
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7z.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOC"?g man1/7z.1 > "${DEST_DIR}${DEST_MAN}/man1/7z.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7za.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOC"?g man1/7za.1 > "${DEST_DIR}${DEST_MAN}/man1/7za.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7zr.1"
    sed -e s?"{DEST_SHARE_DOC}"?"${DEST_SHARE_DOC}/DOC"?g man1/7zr.1 > "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
 else
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7z.1"
    grep -v "{DEST_SHARE_DOC}" man1/7z.1 > "${DEST_DIR}${DEST_MAN}/man1/7z.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7z.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7za.1"
    grep -v "{DEST_SHARE_DOC}" man1/7za.1 > "${DEST_DIR}${DEST_MAN}/man1/7za.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7za.1"
 
    echo "- installing ${DEST_DIR}${DEST_MAN}/man1/7zr.1"
    grep -v "{DEST_SHARE_DOC}" man1/7zr.1 > "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
-   chmod 444 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
+   chmod 644 "${DEST_DIR}${DEST_MAN}/man1/7zr.1"
 fi
 
 if [ -f README ]
@@ -162,7 +163,7 @@ then
   echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/README"
   mkdir -p  "${DEST_DIR}${DEST_SHARE_DOC}"
   cp README "${DEST_DIR}${DEST_SHARE_DOC}/README"
-  chmod 444 "${DEST_DIR}${DEST_SHARE_DOC}/README"
+  chmod 644 "${DEST_DIR}${DEST_SHARE_DOC}/README"
 fi
 
 if [ -f ChangeLog ]
@@ -170,28 +171,28 @@ then
   echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
   mkdir -p     "${DEST_DIR}${DEST_SHARE_DOC}"
   cp ChangeLog "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
-  chmod 444    "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
+  chmod 644    "${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
 fi
 
 if [ -d DOC ]
 then
   echo "- installing HTML help in ${DEST_DIR}${DEST_SHARE_DOC}/DOC"
   mkdir -p "${DEST_DIR}${DEST_SHARE_DOC}"
-  cp -r DOC "${DEST_DIR}${DEST_SHARE_DOC}/DOC"
-  find "${DEST_DIR}${DEST_SHARE_DOC}/DOC" -type d -exec chmod 555 {} \;
-  find "${DEST_DIR}${DEST_SHARE_DOC}/DOC" -type f -exec chmod 444 {} \;
+  cp -Lr DOC "${DEST_DIR}${DEST_SHARE_DOC}/DOC"
+  find "${DEST_DIR}${DEST_SHARE_DOC}/DOC" -type d -exec chmod 755 {} \;
+  find "${DEST_DIR}${DEST_SHARE_DOC}/DOC" -type f -exec chmod 644 {} \;
 fi
 
 use_lang="n"
 
-if [ -x bin/7zG ]
+if [ -x bin/7zG${EXEEXT} ]
 then
   use_lang="o"
-  cp GUI/p7zipForFilemanager "${DEST_DIR}${DEST_BIN}/"
-  chmod 555 "${DEST_DIR}${DEST_BIN}/"
+  cp GUI/p7zipForFilemanager${EXEEXT} "${DEST_DIR}${DEST_BIN}/"
+  chmod 755 "${DEST_DIR}${DEST_BIN}/"
 fi
 
-if [ -x bin/7zFM ]
+if [ -x bin/7zFM${EXEEXT} ]
 then
   use_lang="o"
 fi
diff --git a/makefile b/makefile
index cf2e915..dd149bb 100644
--- a/makefile
+++ b/makefile
@@ -1,10 +1,10 @@
 MKDIR = mkdir -p
 
-DEST_HOME=/usr/local
+DEST_HOME=/usr
 DEST_BIN=$(DEST_HOME)/bin
 DEST_SHARE=$(DEST_HOME)/lib/p7zip
 DEST_SHARE_DOC=$(DEST_HOME)/share/doc/p7zip
-DEST_MAN=$(DEST_HOME)/man
+DEST_MAN=$(DEST_HOME)/share/man
 
 .PHONY: default all all2 7za 7zG 7zFM sfx 7z 7zr Client7z common common7z clean tar_bin depend test test_7z test_7zr test_7zG test_Client7z all_test app cppcheck
 
diff --git a/makefile.cygwin_asm b/makefile.cygwin_asm
index f65ab4a..9adb1d1 100644
--- a/makefile.cygwin_asm
+++ b/makefile.cygwin_asm
@@ -11,7 +11,7 @@ ALLFLAGS=${OPTFLAGS} -s \
 CXX=g++
 CC=gcc
 LINK_SHARED=-shared
-ASM=nasm -f gnuwin32
+ASM=nasm -f win32
 
 # Cygwin does not need -ldl
 LOCAL_LIBS=-lpthread
