From 81a57fbd4c10db5f35a1ad233abd27a71f4009cf Mon Sep 17 00:00:00 2001
From: Jannick <thirdedition@gmx.net>
Date: Wed, 9 Dec 2020 00:36:38 +0100
Subject: [PATCH 1/3] autoconf-2.70-msys-specifics-AC_CYGWIN

---
 lib/autoconf/specific.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/autoconf/specific.m4 b/lib/autoconf/specific.m4
index 275938c..3ffb417 100644
--- a/lib/autoconf/specific.m4
+++ b/lib/autoconf/specific.m4
@@ -319,11 +319,11 @@ AU_DEFUN([AC_GNU_SOURCE], [AC_USE_SYSTEM_EXTENSIONS])
 AU_DEFUN([AC_CYGWIN],
 [AC_CANONICAL_HOST
 case $host_os in
-  *cygwin* ) CYGWIN=yes;;
+  *cygwin* | *msys* ) CYGWIN=yes;;
 	 * ) CYGWIN=no;;
 esac
 ], [$0 is obsolete: use AC_CANONICAL_HOST and check if $host_os
-matches *cygwin*])# AC_CYGWIN
+matches *cygwin* or *msys*])# AC_CYGWIN
 
 
 # AC_EMXOS2
-- 
2.29.2.windows.2

