--- procps-ng-4.0.5/library/signames.c.orig	2024-12-26 16:11:12.020934500 +0100
+++ procps-ng-4.0.5/library/signames.c	2024-12-26 16:12:31.274809200 +0100
@@ -100,9 +100,6 @@
         SIGABBREV(PWR);
 #endif
         SIGABBREV(SYS);
-#ifdef SIGLOST              // Hurd (gitlab#93)
-        SIGABBREV(LOST);
-#endif
 #if defined __sun || defined __SUN || defined __solaris__ || defined __SOLARIS__
         case 0: return "EXIT";
 #endif
