--- procps-ng-4.0.5/testsuite/ps.test/test-hugetlb.c.orig	2024-12-26 16:12:35.138200500 +0100
+++ procps-ng-4.0.5/testsuite/ps.test/test-hugetlb.c	2024-12-26 16:12:43.726064700 +0100
@@ -59,7 +59,7 @@
         usage(argv[0], "Invalid size");
 
     if (MAP_FAILED == (addr = mmap(NULL, (kbytes*1024), (PROT_READ | PROT_WRITE),
-                    (MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB), 0, 0))) {
+                    (MAP_PRIVATE | MAP_ANONYMOUS), 0, 0))) {
         perror("mmap");
         exit(EXIT_FAILURE);
     }
