--- origsrc/procps-v4.0.4/Makefile.am
+++ src/procps-v4.0.4/Makefile.am
@@ -136,7 +136,7 @@
 endif
 dist_man_MANS += man/kill.1
 src_kill_SOURCES = src/kill.c local/strutils.c local/fileutils.c local/signals.c
-src_kill_LDADD = $(LTLIBINTL)
+src_kill_LDADD = $(LDADD) $(LTLIBINTL)
 else
   EXTRA_DIST += man/kill.1
 endif
@@ -179,7 +179,7 @@
 endif
 src_watch_SOURCES = src/watch.c local/strutils.c local/fileutils.c
 src_watch_CFLAGS = @NCURSES_CFLAGS@
-src_watch_LDADD = @NCURSES_LIBS@ $(CYGWINFLAGS)
+src_watch_LDADD = @NCURSES_LIBS@ $(LDADD) $(CYGWINFLAGS)
 src_top_top_SOURCES = \
 	src/top/top.h \
 	src/top/top.c \
@@ -225,7 +225,7 @@
 endif
 if !CYGWIN
 src_pwdx_SOURCES = src/pwdx.c local/fileutils.c
-src_pwdx_LDADD= $(CYGWINFLAGS)
+src_pwdx_LDADD= $(LDADD) $(CYGWINFLAGS)
 src_sysctl_SOURCES = \
 	src/sysctl.c \
 	local/fileutils.c \

