--- origsrc/procps-v4.0.4/Makefile.am
+++ src/procps-v4.0.4/Makefile.am
@@ -38,6 +38,7 @@ sbin_PROGRAMS = \
 	src/sysctl
 else
 transform += s/pscommand/procps/; $(program_transform_name)
+transform += ;s/^ps/procps/; $(program_transform_name)
 endif
 
 if !CYGWIN
@@ -77,15 +78,15 @@ dist_man_MANS = \
 	man/vmstat.8 \
 	man/procps.3 \
 	man/procps_pids.3 \
-	man/procps_misc.3
+	man/procps_misc.3 \
+	man/ps.1
 
 if !CYGWIN
 dist_man_MANS += \
 	man/pwdx.1 \
 	man/tload.1 \
 	man/sysctl.8 \
-	man/sysctl.conf.5 \
-	man/ps.1
+	man/sysctl.conf.5
 
 if BUILD_PIDWAIT
 dist_man_MANS += man/pidwait.1
