diff -Naur procps-ng-3.3.12-orig/configure.ac procps-ng-3.3.12/configure.ac
--- procps-ng-3.3.12-orig/configure.ac	2016-07-10 05:49:25.000000000 +0800
+++ procps-ng-3.3.12/configure.ac	2016-10-26 16:29:11.941814400 +0800
@@ -169,6 +169,7 @@
 AM_CONDITIONAL(BUILD_KILL, test "x$enable_kill" = xyes)
 AM_CONDITIONAL(LINUX, test "x$host_os" = xlinux-gnu)
 AM_CONDITIONAL(CYGWIN, test "x$host_os" = xcygwin)
+AM_CONDITIONAL(CYGWIN, test "x$host_os" = xmsys)
 
 AC_ARG_ENABLE([skill],
   AS_HELP_STRING([--enable-skill], [build skill and snice]),
