--- cargo-c-0.10.18/src/build_targets.rs.orig	2025-11-02 20:12:15.955580000 +0100
+++ cargo-c-0.10.18/src/build_targets.rs	2025-11-02 20:15:46.016610700 +0100
@@ -132,7 +132,7 @@
         match self.lib_type() {
             LibType::Windows => {
                 if self.target.os == "cygwin" {
-                    Some(format!("cyg{}.dll", self.name).into())
+                    Some(format!("msys-{}.dll", self.name).into())
                 } else if self.shared_lib.is_some()
                     && self.use_meson_naming_convention
                     && self.target.env == "gnu"
