diff -Naur bison-3.0.5-orig/configure.ac bison-3.0.5/configure.ac
--- bison-3.0.5-orig/configure.ac	2018-05-27 12:34:40.000000000 +0300
+++ bison-3.0.5/configure.ac	2018-05-31 08:49:39.118626800 +0300
@@ -209,7 +209,7 @@
 # We use gnulib, which is only guaranteed to work properly with the
 # latest Gettext.
 AM_GNU_GETTEXT([external], [need-ngettext])
-AM_GNU_GETTEXT_VERSION([0.18])
+AM_GNU_GETTEXT_VERSION([0.19])
 BISON_I18N
 
 # Internationalized parsers.
