diff -Naur time-1.7-orig/Makefile.in time-1.7-3/Makefile.in
--- time-1.7-orig/Makefile.in	1996-07-12 00:07:03.000000000 +0300
+++ time-1.7/Makefile.in	2013-10-16 21:03:31.000000000 +0300
@@ -7,22 +7,24 @@
 
 SHELL = /bin/sh
 
+DESTDIR=
 srcdir = @srcdir@
 top_srcdir = @top_srcdir@
 VPATH = @srcdir@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 
-bindir = @bindir@
-sbindir = @sbindir@
-libexecdir = @libexecdir@
-datadir = @datadir@
-sysconfdir = @sysconfdir@
-sharedstatedir = @sharedstatedir@
-localstatedir = @localstatedir@
-libdir = @libdir@
-infodir = @infodir@
-mandir = @mandir@
+EXEEXT = @EXEEXT@
+bindir = ${DESTDIR}@bindir@
+sbindir = ${DESTDIR}@sbindir@
+libexecdir = ${DESTDIR}@libexecdir@
+datadir = ${DESTDIR}@datadir@
+sysconfdir = ${DESTDIR}@sysconfdir@
+sharedstatedir = ${DESTDIR}@sharedstatedir@
+localstatedir = ${DESTDIR}@localstatedir@
+libdir = ${DESTDIR}@libdir@
+infodir = ${DESTDIR}@infodir@
+mandir = ${DESTDIR}@mandir@
 includedir = @includedir@
 oldincludedir = /usr/include
 
@@ -38,7 +40,7 @@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 transform = @program_transform_name@
 
-bin_PROGRAMS = time
+bin_PROGRAMS = time${EXEEXT}
 info_TEXINFOS = time.texi
 time_SOURCES = time.c getopt.c getopt1.c error.c resuse.c version.c \
 	getopt.h getpagesize.h resuse.h port.h wait.h
@@ -139,7 +141,7 @@
 
 maintainer-clean-compile:
 
-time: $(time_OBJECTS) $(time_DEPENDENCIES)
+time${EXEEXT}: $(time_OBJECTS) $(time_DEPENDENCIES)
 	$(LINK) $(time_OBJECTS) $(time_LDADD) $(LIBS)
 
 version.texi: stamp-vti
diff -Naur time-1.7-orig/time.mknetrel time-1.7-3/time.mknetrel
--- time-1.7-orig/time.mknetrel	1970-01-01 03:00:00.000000000 +0300
+++ time-1.7-3/time.mknetrel	2013-10-22 00:37:36.000000000 +0300
@@ -0,0 +1,4 @@
+preconfig() {
+    export ac_cv_func_wait3_rusage=yes
+    export useDESTDIR=1
+}
