diff -urN lftp-4.7.1.orig/m4/readline.m4 lftp-4.7.1/m4/readline.m4
--- lftp-4.7.1.orig/m4/readline.m4	2016-02-20 08:57:52.000000000 -0500
+++ lftp-4.7.1/m4/readline.m4	2016-04-05 05:09:43.308102900 -0400
@@ -12,9 +12,14 @@
 [
    old_LIBS="$LIBS"
    LIBS="-lreadline $READLINE_SUPPLIB $LIBS"
-   AC_LINK_IFELSE([AC_LANG_PROGRAM([[extern int (*rl_getc_function)();]], [[rl_getc_function=0;
-       rl_completion_matches(0,0);]])],[AC_PREPROC_IFELSE([AC_LANG_SOURCE([[#include <readline/readline.h>]])],[lftp_cv_precompiled_readline=yes],[lftp_cv_precompiled_readline=no])
-      ],[lftp_cv_precompiled_readline=no])
+   AC_LINK_IFELSE(
+     [AC_LANG_PROGRAM(
+       [[#include <stdio.h>
+	     #include <readline/readline.h>]],
+       [[rl_getc_function=0;
+         rl_completion_matches(0,0);]])],
+     [lftp_cv_precompiled_readline=yes],
+     [lftp_cv_precompiled_readline=no])
    LIBS="$old_LIBS"
 ])
    ;;
