diff -Naur lftp-4.4.15-orig/src/Makefile.am lftp-4.4.15/src/Makefile.am
--- lftp-4.4.15-orig/src/Makefile.am	2013-12-11 13:51:19.000000000 +0400
+++ lftp-4.4.15/src/Makefile.am	2014-04-14 10:03:31.824400000 +0400
@@ -47,16 +47,16 @@
 endif
 GNULIB = $(top_builddir)/lib/libgnu.la
 
-proto_ftp_la_LDFLAGS  = -module -avoid-version -rpath $(pkgverlibdir)
-proto_http_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
-proto_file_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
-proto_fish_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
-proto_sftp_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
-cmd_mirror_la_LDFLAGS = -module -avoid-version -rpath $(pkgverlibdir)
-cmd_sleep_la_LDFLAGS  = -module -avoid-version -rpath $(pkgverlibdir)
-cmd_torrent_la_LDFLAGS= -module -avoid-version -rpath $(pkgverlibdir)
-liblftp_pty_la_LDFLAGS     = -avoid-version -rpath $(pkgverlibdir)
-liblftp_network_la_LDFLAGS = -avoid-version -rpath $(pkgverlibdir) $(OPENSSL_LDFLAGS)
+proto_ftp_la_LDFLAGS  = -module -avoid-version -no-undefined -rpath $(pkgverlibdir)
+proto_http_la_LDFLAGS = -module -avoid-version -no-undefined -rpath $(pkgverlibdir)
+proto_file_la_LDFLAGS = -module -avoid-version -no-undefined -no-undefined -rpath $(pkgverlibdir)
+proto_fish_la_LDFLAGS = -module -avoid-version -no-undefined -rpath $(pkgverlibdir)
+proto_sftp_la_LDFLAGS = -module -avoid-version -no-undefined -rpath $(pkgverlibdir)
+cmd_mirror_la_LDFLAGS = -module -avoid-version -no-undefined -rpath $(pkgverlibdir)
+cmd_sleep_la_LDFLAGS  = -module -avoid-version -no-undefined -rpath $(pkgverlibdir)
+cmd_torrent_la_LDFLAGS= -module -avoid-version -no-undefined -rpath $(pkgverlibdir)
+liblftp_pty_la_LDFLAGS     = -avoid-version -no-undefined -rpath $(pkgverlibdir)
+liblftp_network_la_LDFLAGS = -avoid-version -no-undefined -rpath $(pkgverlibdir) $(OPENSSL_LDFLAGS)
 liblftp_network_la_LIBADD  = $(SOCKSLIBS) $(OPENSSL_LIBS) $(LIBGNUTLS_LIBS) $(GNULIB)
 
 proto_ftp_la_LIBADD  = -L$(DESTDIR)$(pkgverlibdir) liblftp-network.la
