diff -Naur npth-1.2-orig/build-aux/compile npth-1.2/build-aux/compile
--- npth-1.2-orig/build-aux/compile	2015-01-28 13:26:31.000000000 +0300
+++ npth-1.2/build-aux/compile	2015-04-30 09:57:10.015800000 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur npth-1.2-orig/build-aux/config.guess npth-1.2/build-aux/config.guess
--- npth-1.2-orig/build-aux/config.guess	2015-01-28 13:26:06.000000000 +0300
+++ npth-1.2/build-aux/config.guess	2015-04-30 09:56:37.191400000 +0300
@@ -867,6 +867,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
