diff -Naur diffutils-3.3-orig/build-aux/compile diffutils-3.3/build-aux/compile
--- diffutils-3.3-orig/build-aux/compile	2013-03-24 17:55:40.000000000 +0000
+++ diffutils-3.3/build-aux/compile	2013-05-08 10:37:14.156250000 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur diffutils-3.3-orig/build-aux/config.guess diffutils-3.3/build-aux/config.guess
--- diffutils-3.3-orig/build-aux/config.guess	2013-03-22 04:20:48.000000000 +0000
+++ diffutils-3.3/build-aux/config.guess	2013-05-08 10:37:14.171875000 +0000
@@ -802,6 +802,9 @@
     *:MINGW64*:*)
 	echo ${UNAME_MACHINE}-pc-mingw64
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
@@ -845,6 +848,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur diffutils-3.3-orig/build-aux/config.rpath diffutils-3.3/build-aux/config.rpath
--- diffutils-3.3-orig/build-aux/config.rpath	2013-03-22 04:20:48.000000000 +0000
+++ diffutils-3.3/build-aux/config.rpath	2013-05-08 10:37:14.171875000 +0000
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur diffutils-3.3-orig/configure diffutils-3.3/configure
--- diffutils-3.3-orig/configure	2013-03-24 18:03:03.000000000 +0000
+++ diffutils-3.3/configure	2013-05-08 10:40:07.202148400 +0000
@@ -8937,7 +8937,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -13862,7 +13862,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
@@ -16357,7 +16357,7 @@
                      # Guess yes on glibc systems.
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
-            cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+            cygwin* | msys*) gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
@@ -16629,7 +16629,7 @@
 
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
@@ -17024,7 +17024,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -17066,7 +17066,7 @@
   case "$host_os" in
            mingw*) # on this platform, dup2 always returns 0 for success
              gl_cv_func_dup2_works="guessing no" ;;
-           cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+           cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
            linux*) # On linux between 2008-07-27 and 2009-05-11, dup2 of a
                    # closed fd may yield -EBADF instead of -1 / errno=EBADF.
@@ -20122,7 +20122,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
@@ -29433,7 +29433,7 @@
 
   case "$host_os" in
             mingw*) REPLACE_SETLOCALE=1 ;;
-            cygwin*)
+            cygwin* | msys*)
       case `uname -r` in
         1.5.*) REPLACE_SETLOCALE=1 ;;
       esac
@@ -31292,13 +31292,13 @@
 
                         case "$host_os" in
             # Guess no on Solaris <= 9 and Cygwin.
-          solaris2.[1-9] | solaris2.[1-9].* | cygwin*)
+          solaris2.[1-9] | solaris2.[1-9].* | cygwin* | msys*)
             gl_cv_func_wctob_works="guessing no" ;;
             # Guess yes otherwise.
           *) gl_cv_func_wctob_works="guessing yes" ;;
         esac
         case "$host_os" in
-          cygwin*)
+          cygwin* | msys*)
             if test "$cross_compiling" = yes; then :
   :
 else
diff -Naur diffutils-3.3-orig/lib/gnulib.mk diffutils-3.3/lib/gnulib.mk
--- diffutils-3.3-orig/lib/gnulib.mk	2013-03-23 14:52:02.000000000 +0000
+++ diffutils-3.3/lib/gnulib.mk	2013-05-08 10:37:14.250000000 +0000
@@ -804,7 +804,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -Naur diffutils-3.3-orig/lib/Makefile.in diffutils-3.3/lib/Makefile.in
--- diffutils-3.3-orig/lib/Makefile.in	2013-03-24 17:55:42.000000000 +0000
+++ diffutils-3.3/lib/Makefile.in	2013-05-08 10:37:14.234375000 +0000
@@ -2065,7 +2065,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -Naur diffutils-3.3-orig/m4/btowc.m4 diffutils-3.3/m4/btowc.m4
--- diffutils-3.3-orig/m4/btowc.m4	2013-03-22 04:20:49.000000000 +0000
+++ diffutils-3.3/m4/btowc.m4	2013-05-08 10:37:14.250000000 +0000
@@ -49,7 +49,7 @@
 changequote(,)dnl
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
diff -Naur diffutils-3.3-orig/m4/double-slash-root.m4 diffutils-3.3/m4/double-slash-root.m4
--- diffutils-3.3-orig/m4/double-slash-root.m4	2013-03-22 04:20:49.000000000 +0000
+++ diffutils-3.3/m4/double-slash-root.m4	2013-05-08 10:37:14.250000000 +0000
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur diffutils-3.3-orig/m4/dup2.m4 diffutils-3.3/m4/dup2.m4
--- diffutils-3.3-orig/m4/dup2.m4	2013-03-22 04:20:49.000000000 +0000
+++ diffutils-3.3/m4/dup2.m4	2013-05-08 10:41:12.903320300 +0000
@@ -49,7 +49,7 @@
         [case "$host_os" in
            mingw*) # on this platform, dup2 always returns 0 for success
              gl_cv_func_dup2_works="guessing no" ;;
-           cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+           cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
            linux*) # On linux between 2008-07-27 and 2009-05-11, dup2 of a
                    # closed fd may yield -EBADF instead of -1 / errno=EBADF.
diff -Naur diffutils-3.3-orig/m4/getcwd.m4 diffutils-3.3/m4/getcwd.m4
--- diffutils-3.3-orig/m4/getcwd.m4	2013-03-22 04:20:49.000000000 +0000
+++ diffutils-3.3/m4/getcwd.m4	2013-05-08 10:41:31.262695300 +0000
@@ -49,7 +49,7 @@
                      # Guess yes on glibc systems.
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
-            cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+            cygwin* | msys*) gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
diff -Naur diffutils-3.3-orig/m4/malloc.m4 diffutils-3.3/m4/malloc.m4
--- diffutils-3.3-orig/m4/malloc.m4	2013-03-22 04:20:49.000000000 +0000
+++ diffutils-3.3/m4/malloc.m4	2013-05-08 10:44:46.582031200 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur diffutils-3.3-orig/m4/printf.m4 diffutils-3.3/m4/printf.m4
--- diffutils-3.3-orig/m4/printf.m4	2013-03-22 04:20:49.000000000 +0000
+++ diffutils-3.3/m4/printf.m4	2013-05-08 10:37:14.281250000 +0000
@@ -735,7 +735,7 @@
            openbsd*)        gl_cv_func_printf_directive_ls="guessing no";;
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
-           cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           cygwin* | msys*)         gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -812,7 +812,7 @@
         [
 changequote(,)dnl
          case "$host_os" in
-           cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           cygwin* | msys*)      gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -1464,7 +1464,7 @@
            darwin[1-6].*)        gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
-           cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           cygwin* | msys*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Naur diffutils-3.3-orig/m4/setlocale.m4 diffutils-3.3/m4/setlocale.m4
--- diffutils-3.3-orig/m4/setlocale.m4	2013-03-22 04:20:49.000000000 +0000
+++ diffutils-3.3/m4/setlocale.m4	2013-05-08 10:37:14.281250000 +0000
@@ -14,7 +14,7 @@
     mingw*) REPLACE_SETLOCALE=1 ;;
     dnl On Cygwin 1.5.x, setlocale always succeeds but setlocale(LC_CTYPE,NULL)
     dnl is then still "C".
-    cygwin*)
+    cygwin* | msys*)
       case `uname -r` in
         1.5.*) REPLACE_SETLOCALE=1 ;;
       esac
diff -Naur diffutils-3.3-orig/m4/wctob.m4 diffutils-3.3/m4/wctob.m4
--- diffutils-3.3-orig/m4/wctob.m4	2013-03-22 04:20:49.000000000 +0000
+++ diffutils-3.3/m4/wctob.m4	2013-05-08 10:37:14.281250000 +0000
@@ -29,14 +29,14 @@
 changequote(,)dnl
         case "$host_os" in
             # Guess no on Solaris <= 9 and Cygwin.
-          solaris2.[1-9] | solaris2.[1-9].* | cygwin*)
+          solaris2.[1-9] | solaris2.[1-9].* | cygwin* | msys*)
             gl_cv_func_wctob_works="guessing no" ;;
             # Guess yes otherwise.
           *) gl_cv_func_wctob_works="guessing yes" ;;
         esac
 changequote([,])dnl
         case "$host_os" in
-          cygwin*)
+          cygwin* | msys*)
             AC_RUN_IFELSE(
               [AC_LANG_SOURCE([[
 #include <locale.h>
