--- zstd-1.4.4/programs/util.c.orig	2020-01-06 06:41:47.793486500 +0100
+++ zstd-1.4.4/programs/util.c	2020-01-06 06:41:57.234481400 +0100
@@ -621,7 +621,7 @@
     return numPhysicalCores;
 }
 
-#elif defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
+#elif defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__) || defined(__CYGWIN__)
 
 /* Use POSIX sysconf
  * see: man 3 sysconf */
