diff -Naur DBI-1.627_94/t/09trace.t DBI-1.627_94/t/09trace.t
--- DBI-1.627_94/t/09trace.t	2013-04-05 02:17:19.000000000 +0400
+++ DBI-1.627_94/t/09trace.t	2013-07-17 08:42:48.167968700 +0400
@@ -37,7 +37,7 @@
 DBI->trace(0, undef);   # turn off and restore to STDERR
 
 SKIP: {
-        skip "cygwin has buffer flushing bug", 1 if ($^O =~ /cygwin/i);
+        skip "cygwin has buffer flushing bug", 1 if ($^O =~ /cygwin|msys/i);
         ok( -s $trace_file, "trace file size = " . -s $trace_file);
 }
 
