diff -Naur pcre-8.33-orig/config.guess pcre-8.33/config.guess
--- pcre-8.33-orig/config.guess	2013-05-27 15:56:00.000000000 +0000
+++ pcre-8.33/config.guess	2013-06-02 08:33:13.797851600 +0000
@@ -903,6 +903,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"$(echo "$UNAME_RELEASE"|sed -e 's/[^.]*//')"
 	exit ;;
diff -Naur pcre-8.33-orig/configure.ac pcre-8.33/configure.ac
--- pcre-8.33-orig/configure.ac	2013-05-27 15:54:34.000000000 +0000
+++ pcre-8.33/configure.ac	2013-06-02 08:34:06.329101600 +0000
@@ -892,7 +892,7 @@
 NO_UNDEFINED=
 EXPORT_ALL_SYMBOLS=
 case $host_os in
-  cygwin* | mingw* )
+  cygwin* | msys* | mingw* )
     if test X"$enable_shared" = Xyes; then
       NO_UNDEFINED="-no-undefined"
       EXPORT_ALL_SYMBOLS="-Wl,--export-all-symbols"
