diff -Naur pcre-8.33-orig/ar-lib pcre-8.33/ar-lib
--- pcre-8.33-orig/ar-lib	2013-05-27 15:56:00.000000000 +0000
+++ pcre-8.33/ar-lib	2013-06-02 08:32:18.782226600 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -65,7 +65,7 @@
 	mingw)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin)
+	cygwin|msys)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine)
diff -Naur pcre-8.33-orig/compile pcre-8.33/compile
--- pcre-8.33-orig/compile	2013-05-27 15:56:00.000000000 +0000
+++ pcre-8.33/compile	2013-06-02 08:32:45.157226600 +0000
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur pcre-8.33-orig/config.guess pcre-8.33/config.guess
--- pcre-8.33-orig/config.guess	2013-05-27 15:56:00.000000000 +0000
+++ pcre-8.33/config.guess	2013-06-02 08:33:13.797851600 +0000
@@ -845,6 +845,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
diff -Naur pcre-8.33-orig/configure.ac pcre-8.33/configure.ac
--- pcre-8.33-orig/configure.ac	2013-05-27 15:54:34.000000000 +0000
+++ pcre-8.33/configure.ac	2013-06-02 08:34:06.329101600 +0000
@@ -856,7 +856,7 @@
 NO_UNDEFINED=
 EXPORT_ALL_SYMBOLS=
 case $host_os in
-  cygwin* | mingw* )
+  cygwin* | msys* | mingw* )
     if test X"$enable_shared" = Xyes; then
       NO_UNDEFINED="-no-undefined"
       EXPORT_ALL_SYMBOLS="-Wl,--export-all-symbols"
