diff -urN shellspec-0.28.1.org/Makefile shellspec-0.28.1/Makefile
--- shellspec-0.28.1.org/Makefile	2021-01-10 21:25:23.000000000 -0800
+++ shellspec-0.28.1/Makefile	2022-06-09 04:03:17.406666800 -0700
@@ -13,11 +13,11 @@
 	tar -czf shellspec-dist.tar.gz $^ --transform 's,^,shellspec/,'
 
 install:
-	install -d "$(BINDIR)" "$(LIBDIR)"
-	install stub/shellspec "$(BINDIR)/shellspec"
-	find lib libexec -type d -exec install -d "$(LIBDIR)/shellspec/{}" \;
-	find LICENSE lib -type f -exec install -m 644 {} "$(LIBDIR)/shellspec/{}" \;
-	find shellspec libexec -type f -exec install {} "$(LIBDIR)/shellspec/{}" \;
+	install -d "$(DESTDIR)$(BINDIR)" "$(DESTDIR)$(LIBDIR)"
+	install stub/shellspec "$(DESTDIR)/$(BINDIR)/shellspec"
+	find lib libexec -type d -exec install -d "$(DESTDIR)$(LIBDIR)/shellspec/{}" \;
+	find LICENSE lib -type f -exec install -m 644 {} "$(DESTDIR)/$(LIBDIR)/shellspec/{}" \;
+	find shellspec libexec -type f -exec install {} "$(DESTDIR)/$(LIBDIR)/shellspec/{}" \;
 
 uninstall:
 	rm -rf "$(BINDIR)/shellspec" "$(LIBDIR)/shellspec"
