--- which-2.20/Makefile.am.orig	2007-10-10 00:11:42.000000000 +0000
+++ which-2.20/Makefile.am	2013-07-25 11:44:32.103848400 +0000
@@ -1,5 +1,5 @@
 AUTOMAKE_OPTIONS = no-texinfo.tex
-ACLOCAL_AMFLAGS=@ACLOCAL_CWFLAGS@
+ACLOCAL_AMFLAGS = @ACLOCAL_CWFLAGS@
 
 SUBDIRS = tilde .
 
--- which-2.20/tilde/Makefile.am.orig	2008-01-16 17:51:57.000000000 +0000
+++ which-2.20/tilde/Makefile.am	2013-07-25 11:45:06.456210600 +0000
@@ -1,4 +1,4 @@
-ACLOCAL_AMFLAGS=@ACLOCAL_CWFLAGS@
+ACLOCAL_AMFLAGS = @ACLOCAL_CWFLAGS@
 AUTOMAKE_OPTIONS = foreign
 noinst_LIBRARIES = libtilde.a
 libtilde_a_SOURCES = tilde.c shell.c tilde.h ansi_stdlib.h bashansi.h xmalloc.h
