diff -Nadur findutils-4.7.0/build-aux/config.rpath findutils-4.8.0/build-aux/config.rpath
--- findutils-4.7.0/build-aux/config.rpath
+++ findutils-4.8.0/build-aux/config.rpath
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Nadur findutils-4.7.0/gl/m4/btowc.m4 findutils-4.8.0/gl/m4/btowc.m4
--- findutils-4.7.0/gl/m4/btowc.m4
+++ findutils-4.8.0/gl/m4/btowc.m4
@@ -42,6 +42,7 @@
            case "$host_os" in
                       # Guess no on Cygwin.
              cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             msys*)   gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes on native Windows.
              mingw*)  gl_cv_func_btowc_nul="guessing yes" ;;
                       # Guess yes otherwise.
diff -Nadur findutils-4.7.0/gl/m4/double-slash-root.m4 findutils-4.8.0/gl/m4/double-slash-root.m4
--- findutils-4.7.0/gl/m4/double-slash-root.m4
+++ findutils-4.8.0/gl/m4/double-slash-root.m4
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Nadur findutils-4.7.0/gl/m4/dup2.m4 findutils-4.8.0/gl/m4/dup2.m4
--- findutils-4.7.0/gl/m4/dup2.m4
+++ findutils-4.8.0/gl/m4/dup2.m4
@@ -69,7 +69,7 @@
       [case "$host_os" in
          mingw*) # on this platform, dup2 always returns 0 for success
            gl_cv_func_dup2_works="guessing no" ;;
-         cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+         cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
            gl_cv_func_dup2_works="guessing no" ;;
          aix* | freebsd*)
                  # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
diff -Nadur findutils-4.7.0/gl/m4/fcntl.m4 findutils-4.8.0/gl/m4/fcntl.m4
--- findutils-4.7.0/gl/m4/fcntl.m4
+++ findutils-4.8.0/gl/m4/fcntl.m4
@@ -70,7 +70,7 @@
          [gl_cv_func_fcntl_f_dupfd_works=yes],
          [gl_cv_func_fcntl_f_dupfd_works=no],
          [case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac])])
diff -Nadur findutils-4.7.0/gl/m4/getcwd.m4 findutils-4.8.0/gl/m4/getcwd.m4
--- findutils-4.7.0/gl/m4/getcwd.m4
+++ findutils-4.8.0/gl/m4/getcwd.m4
@@ -55,6 +55,7 @@
             *-musl*)       gl_cv_func_getcwd_null="guessing yes";;
                            # Guess yes on Cygwin.
             cygwin*)       gl_cv_func_getcwd_null="guessing yes";;
+            msys*)         gl_cv_func_getcwd_null="guessing yes";;
                            # If we don't know, obey --enable-cross-guesses.
             *)             gl_cv_func_getcwd_null="$gl_cross_guess_normal";;
           esac
diff -Nadur findutils-4.7.0/gl/m4/malloc.m4 findutils-4.8.0/gl/m4/malloc.m4
--- findutils-4.7.0/gl/m4/malloc.m4
+++ findutils-4.8.0/gl/m4/malloc.m4
@@ -25,7 +25,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | *-musl* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull="guessing yes" ;;
           # If we don't know, obey --enable-cross-guesses.
           *) ac_cv_func_malloc_0_nonnull="$gl_cross_guess_normal" ;;
diff -Nadur findutils-4.7.0/gl/m4/printf.m4 findutils-4.8.0/gl/m4/printf.m4
--- findutils-4.7.0/gl/m4/printf.m4
+++ findutils-4.8.0/gl/m4/printf.m4
@@ -813,6 +813,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
                             # Guess no on Android.
            linux*-android*) gl_cv_func_printf_directive_ls="guessing no";;
@@ -897,6 +898,7 @@
 changequote(,)dnl
          case "$host_os" in
            cygwin*)         gl_cv_func_printf_flag_grouping="guessing no";;
+           msys*)           gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)         gl_cv_func_printf_flag_grouping="guessing no";;
                             # Guess no on Android.
            linux*-android*) gl_cv_func_printf_flag_grouping="guessing no";;
@@ -1606,6 +1608,7 @@
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
            cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           msys*)                gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Nadur findutils-4.7.0/gl/m4/realloc.m4 findutils-4.8.0/gl/m4/realloc.m4
--- findutils-4.7.0/gl/m4/realloc.m4
+++ findutils-4.8.0/gl/m4/realloc.m4
@@ -25,7 +25,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | *-musl* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull="guessing yes" ;;
           # If we don't know, obey --enable-cross-guesses.
           *) ac_cv_func_realloc_0_nonnull="$gl_cross_guess_normal" ;;
diff -Nadur findutils-4.7.0/gl/m4/setlocale_null.m4 findutils-4.8.0/gl/m4/setlocale_null.m4
--- findutils-4.7.0/gl/m4/setlocale_null.m4
+++ findutils-4.8.0/gl/m4/setlocale_null.m4
@@ -14,7 +14,7 @@
     [gl_cv_func_setlocale_null_all_mtsafe],
     [case "$host_os" in
        # Guess no on musl libc, macOS, FreeBSD, NetBSD, OpenBSD, AIX, Haiku, Cygwin.
-       *-musl* | darwin* | freebsd* | netbsd* | openbsd* | aix* | haiku* | cygwin*)
+       *-musl* | darwin* | freebsd* | netbsd* | openbsd* | aix* | haiku* | cygwin* | msys*)
          gl_cv_func_setlocale_null_all_mtsafe=no ;;
        # Guess yes on glibc, HP-UX, IRIX, Solaris, native Windows.
        *-gnu* | gnu* | hpux* | irix* | solaris* | mingw*)
@@ -48,7 +48,7 @@
        openbsd* | aix*)
          gl_cv_func_setlocale_null_one_mtsafe=no ;;
        # Guess yes on glibc, musl libc, macOS, FreeBSD, NetBSD, HP-UX, IRIX, Solaris, Haiku, Cygwin, native Windows.
-       *-gnu* | gnu* | *-musl* | darwin* | freebsd* | netbsd* | hpux* | irix* | solaris* | haiku* | cygwin* | mingw*)
+       *-gnu* | gnu* | *-musl* | darwin* | freebsd* | netbsd* | hpux* | irix* | solaris* | haiku* | cygwin* | msys* | mingw*)
          gl_cv_func_setlocale_null_one_mtsafe=yes ;;
        # If we don't know, obey --enable-cross-guesses.
        *)
diff -Nadur findutils-4.7.0/gl/m4/unlinkdir.m4 findutils-4.8.0/gl/m4/unlinkdir.m4
--- findutils-4.7.0/gl/m4/unlinkdir.m4
+++ findutils-4.8.0/gl/m4/unlinkdir.m4
@@ -25,6 +25,7 @@
   linux-* | linux | \
   freebsd2.2* | freebsd[[3-9]]* | freebsd[[1-9]][[0-9]]* | \
   cygwin | \
+  msys*  | \
   mingw*)
     AC_DEFINE([UNLINK_CANNOT_UNLINK_DIR], [1],
       [Define to 1 if unlink (dir) cannot possibly succeed.]);;
diff -Nadur findutils-4.7.0/gl/m4/wctob.m4 findutils-4.8.0/gl/m4/wctob.m4
--- findutils-4.7.0/gl/m4/wctob.m4
+++ findutils-4.8.0/gl/m4/wctob.m4
@@ -29,7 +29,7 @@
 changequote(,)dnl
         case "$host_os" in
             # Guess no on Solaris <= 9 and Cygwin.
-          solaris2.[1-9] | solaris2.[1-9].* | cygwin*)
+          solaris2.[1-9] | solaris2.[1-9].* | cygwin* | msys*)
             gl_cv_func_wctob_works="guessing no" ;;
             # Guess no on native Windows.
           mingw*)
@@ -39,7 +39,7 @@
         esac
 changequote([,])dnl
         case "$host_os" in
-          cygwin*)
+          cygwin* | msys*)
             AC_RUN_IFELSE(
               [AC_LANG_SOURCE([[
 #include <locale.h>
