diff -Naur findutils-4.7.0-orig/build-aux/compile findutils-4.7.0/build-aux/compile
--- findutils-4.7.0-orig/build-aux/compile	2019-08-29 22:15:43.000000000 +0300
+++ findutils-4.7.0/build-aux/compile	2019-10-01 08:35:50.674508600 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur findutils-4.7.0-orig/build-aux/config.guess findutils-4.7.0/build-aux/config.guess
--- findutils-4.7.0-orig/build-aux/config.guess	2019-06-23 12:00:48.000000000 +0300
+++ findutils-4.7.0/build-aux/config.guess	2019-10-01 08:36:15.587752400 +0300
@@ -896,6 +896,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-pc-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
diff -Naur findutils-4.7.0-orig/build-aux/config.rpath findutils-4.7.0/build-aux/config.rpath
--- findutils-4.7.0-orig/build-aux/config.rpath	2019-08-29 22:14:52.000000000 +0300
+++ findutils-4.7.0/build-aux/config.rpath	2019-10-01 08:36:50.017012900 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur findutils-4.7.0-orig/gl/m4/btowc.m4 findutils-4.7.0/gl/m4/btowc.m4
--- findutils-4.7.0-orig/gl/m4/btowc.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/btowc.m4	2019-10-01 08:37:52.198722100 +0300
@@ -51,6 +51,8 @@
                       # Guess no on Cygwin.
              cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes on native Windows.
+             msys*)   gl_cv_func_btowc_nul="guessing no" ;;
+                      # Guess yes on native Windows.
              mingw*)  gl_cv_func_btowc_nul="guessing yes" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
diff -Naur findutils-4.7.0-orig/gl/m4/double-slash-root.m4 findutils-4.7.0/gl/m4/double-slash-root.m4
--- findutils-4.7.0-orig/gl/m4/double-slash-root.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/double-slash-root.m4	2019-10-01 08:38:12.853158400 +0300
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur findutils-4.7.0-orig/gl/m4/dup2.m4 findutils-4.7.0/gl/m4/dup2.m4
--- findutils-4.7.0-orig/gl/m4/dup2.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/dup2.m4	2019-10-01 08:38:58.686038900 +0300
@@ -78,7 +78,7 @@
         [case "$host_os" in
            mingw*) # on this platform, dup2 always returns 0 for success
              gl_cv_func_dup2_works="guessing no" ;;
-           cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+           cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
            aix* | freebsd*)
                    # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
diff -Naur findutils-4.7.0-orig/gl/m4/fcntl.m4 findutils-4.7.0/gl/m4/fcntl.m4
--- findutils-4.7.0-orig/gl/m4/fcntl.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/fcntl.m4	2019-10-01 08:39:18.747674100 +0300
@@ -69,7 +69,7 @@
          [gl_cv_func_fcntl_f_dupfd_works=yes],
          [gl_cv_func_fcntl_f_dupfd_works=no],
          [case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac])])
diff -Naur findutils-4.7.0-orig/gl/m4/getcwd.m4 findutils-4.7.0/gl/m4/getcwd.m4
--- findutils-4.7.0-orig/gl/m4/getcwd.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/getcwd.m4	2019-10-01 08:40:55.155843400 +0300
@@ -55,6 +55,8 @@
                            # Guess yes on Cygwin.
             cygwin*)       gl_cv_func_getcwd_null="guessing yes";;
                            # If we don't know, assume the worst.
+            msys*)         gl_cv_func_getcwd_null="guessing yes";;
+                           # If we don't know, assume the worst.
             *)             gl_cv_func_getcwd_null="guessing no";;
           esac
         ]])])
diff -Naur findutils-4.7.0-orig/gl/m4/getdtablesize.m4 findutils-4.7.0/gl/m4/getdtablesize.m4
--- findutils-4.7.0-orig/gl/m4/getdtablesize.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/getdtablesize.m4	2019-10-01 08:41:15.296478900 +0300
@@ -39,7 +39,7 @@
              [gl_cv_func_getdtablesize_works=yes],
              [gl_cv_func_getdtablesize_works=no],
              [case "$host_os" in
-                cygwin*) # on cygwin 1.5.25, getdtablesize() automatically grows
+                cygwin* | msys*) # on cygwin 1.5.25, getdtablesize() automatically grows
                   gl_cv_func_getdtablesize_works="guessing no" ;;
                 *) gl_cv_func_getdtablesize_works="guessing yes" ;;
               esac
diff -Naur findutils-4.7.0-orig/gl/m4/malloc.m4 findutils-4.7.0/gl/m4/malloc.m4
--- findutils-4.7.0-orig/gl/m4/malloc.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/malloc.m4	2019-10-01 08:41:47.502664700 +0300
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | *-musl* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull="guessing yes" ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull="guessing no" ;;
diff -Naur findutils-4.7.0-orig/gl/m4/printf.m4 findutils-4.7.0/gl/m4/printf.m4
--- findutils-4.7.0-orig/gl/m4/printf.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/printf.m4	2019-10-01 08:42:39.310355700 +0300
@@ -795,6 +795,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
                             # Guess no on Android.
            linux*-android*) gl_cv_func_printf_directive_ls="guessing no";;
@@ -1582,6 +1583,8 @@
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
            cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+                                 # Guess yes on MSYS.
+           msys*)                gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Naur findutils-4.7.0-orig/gl/m4/realloc.m4 findutils-4.7.0/gl/m4/realloc.m4
--- findutils-4.7.0-orig/gl/m4/realloc.m4	2019-08-29 22:14:55.000000000 +0300
+++ findutils-4.7.0/gl/m4/realloc.m4	2019-10-01 08:43:25.923237600 +0300
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | *-musl* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull="guessing yes" ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull="guessing no" ;;
diff -Naur findutils-4.7.0-orig/gl/m4/wctob.m4 findutils-4.7.0/gl/m4/wctob.m4
--- findutils-4.7.0-orig/gl/m4/wctob.m4	2019-08-29 22:14:56.000000000 +0300
+++ findutils-4.7.0/gl/m4/wctob.m4	2019-10-01 08:44:46.902979800 +0300
@@ -39,7 +39,7 @@
         esac
 changequote([,])dnl
         case "$host_os" in
-          cygwin*)
+          cygwin* | msys*)
             AC_RUN_IFELSE(
               [AC_LANG_SOURCE([[
 #include <locale.h>
