--- rhash-1.4.1/configure.orig	2021-05-10 23:43:36.396000000 +0200
+++ rhash-1.4.1/configure	2021-05-10 23:44:36.134593900 +0200
@@ -665,7 +665,7 @@
 fi
 
 WIN_LDFLAGS=
-if win32; then
+if win32 && ! cygwin && ! msys; then
   start_check "linker support for --nxcompat --no-seh --dynamicbase"
   cc_check_cflag "-Wl,--nxcompat,--no-seh,--dynamicbase" &&
     WIN_LDFLAGS="-Wl,--nxcompat,--no-seh,--dynamicbase"
