--- rhash-1.3.5/Makefile.orig	2017-11-17 07:54:50.450792800 +0300
+++ rhash-1.3.5/Makefile	2017-11-17 07:56:16.689351900 +0300
@@ -130,8 +130,8 @@
 
 # dependencies should be properly set, otherwise 'make -j<n>' can fail
 install-symlinks: mkdir-bin install-man
-	for f in $(SYMLINKS); do ln -fs rhash $(DESTDIR)$(BINDIR)/$$f; done
-	cd $(DESTDIR)$(MANDIR)/man1 && for f in $(SYMLINKS); do ln -fs rhash.1* $$f.1; done
+	for f in $(SYMLINKS); do cp -f rhash.exe $(DESTDIR)$(BINDIR)/$$f.exe; done
+	cd $(DESTDIR)$(MANDIR)/man1 && for f in $(SYMLINKS); do cp -f rhash.1* $$f.1; done
 
 uninstall-binary:
 	rm -f $(DESTDIR)$(BINDIR)/$(PROGNAME)
