--- RHash-1.3.6/configure.orig	2018-03-27 18:22:44.445286700 -0400
+++ RHash-1.3.6/configure	2018-03-27 18:22:53.683713800 -0400
@@ -649,12 +649,12 @@ if test "$OPT_GETTEXT" != "no"; then
   if cc_check_headers "libintl.h"; then
     if cc_check_statement "libintl.h" "gettext(\"\");"; then
       GETTEXT_FOUND=found
-    elif cc_check_statement "libintl.h" "gettext(\"\");" "-lintl"; then
-      GETTEXT_LDFLAGS="-lintl"
-      GETTEXT_FOUND=found
     elif cc_check_statement "libintl.h" "gettext(\"\");" "-lintl -liconv"; then
       GETTEXT_LDFLAGS="-lintl -liconv"
       GETTEXT_FOUND=found
+    elif cc_check_statement "libintl.h" "gettext(\"\");" "-lintl"; then
+      GETTEXT_LDFLAGS="-lintl"
+      GETTEXT_FOUND=found
     fi
   fi
   test "$GETTEXT_FOUND" = "found" && RHASH_DEFINES=$(str_concat $RHASH_DEFINES -DUSE_GETTEXT)
@@ -897,4 +897,4 @@ Libs: -L\${libdir} -lrhash
 Libs.private: ${OPENSSL_LDFLAGS}
 
 EOF
-fi
\ No newline at end of file
+fi
