diff --git a/Makefile b/Makefile
index 1cfb459..db8c05a 100644
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ SHAR = shar
 PERL = perl
 RM = rm -rf
 VERSION = 2.1.5
-MKDIR = mkdir
+MKDIR = mkdir -p
 prefix = /usr/local
 PYTHON2 = python
 PYTHON3 = python
@@ -46,18 +46,18 @@ python3:
 install: install-main install-man install-man-ja
 
 install-main:
-	-$(MKDIR) $(prefix)/bin
-	cp -f nkf $(prefix)/bin/
+	-$(MKDIR) $(DESTDIR)$(prefix)/bin
+	cp -f nkf $(DESTDIR)$(prefix)/bin/
 
 install-man:
-	-$(MKDIR) $(prefix)/man
-	-$(MKDIR) $(prefix)/man/man1
-	cp -f nkf.1 $(prefix)/man/man1/
+	-$(MKDIR) $(DESTDIR)$(prefix)/share/man
+	-$(MKDIR) $(DESTDIR)$(prefix)/share/man/man1
+	cp -f nkf.1 $(DESTDIR)$(prefix)/share/man/man1/
 
 install-man-ja:
-	-$(MKDIR) $(prefix)/man/ja
-	-$(MKDIR) $(prefix)/man/ja/man1
-	cp -f nkf.1j $(prefix)/man/ja/man1/nkf.1
+	-$(MKDIR) $(DESTDIR)$(prefix)/share/man/ja
+	-$(MKDIR) $(DESTDIR)$(prefix)/share/man/ja/man1
+	iconv -f ISO-2022-JP -t UTF-8  nkf.1j > $(DESTDIR)$(prefix)/share/man/ja/man1/nkf.1
 
 shar:
 	-mkdir nkf-$(VERSION)
