diff -ruN libtorrent-0.13.6-orig/libtorrent.pc.in libtorrent-0.13.6/libtorrent.pc.in
--- libtorrent-0.13.6-orig/libtorrent.pc.in	2015-09-03 18:30:55.000000000 +0000
+++ libtorrent-0.13.6/libtorrent.pc.in	2016-11-08 09:42:49.681229600 +0000
@@ -6,5 +6,5 @@
 Name: libtorrent
 Description: A BitTorrent library
 Version: @VERSION@
-Libs: -L${libdir} -ltorrent
+Libs: -L${libdir} -ltorrent -lcrypto -lz
 Cflags: -I${includedir}
diff -ruN libtorrent-0.13.6-orig/src/Makefile.am libtorrent-0.13.6/src/Makefile.am
--- libtorrent-0.13.6-orig/src/Makefile.am	2015-09-03 18:30:55.000000000 +0000
+++ libtorrent-0.13.6/src/Makefile.am	2016-11-08 09:35:05.411954700 +0000
@@ -10,7 +10,7 @@
 
 lib_LTLIBRARIES = libtorrent.la
 
-libtorrent_la_LDFLAGS = -version-info $(LIBTORRENT_INTERFACE_VERSION_INFO)
+libtorrent_la_LDFLAGS = -version-info $(LIBTORRENT_INTERFACE_VERSION_INFO) -no-undefined
 libtorrent_la_LIBADD = \
 	torrent/libsub_torrent.la \
 	torrent/data/libsub_torrentdata.la \
