--- Python-3.1.2/Include/py_curses.h.orig	2009-09-06 16:26:46.000000000 -0500
+++ Python-3.1.2/Include/py_curses.h	2010-05-03 01:25:11.226933700 -0500
@@ -17,6 +17,13 @@
 #endif
 #endif /* __APPLE__ */
 
+#ifdef __CYGWIN__
+/* the following define is necessary for Cygwin; without it, the
+   Cygwin-supplied ncurses.h sets NCURSES_OPAQUE to 1, and then Python
+   can't get at the WINDOW flags field. */
+#define NCURSES_INTERNALS
+#endif /* __CYGWIN__ */
+
 /* On FreeBSD, [n]curses.h and stdlib.h/wchar.h use different guards
    against multiple definition of wchar_t and wint_t. */
 #if defined(__FreeBSD__) && defined(_XOPEN_SOURCE_EXTENDED)
