--- Python-3.8.0a3-orig/setup.py	2019-03-29 13:42:31.711914000 -0400
+++ Python-3.8.0a3/setup.py	2019-03-29 13:40:09.958984300 -0400
@@ -1208,7 +1208,7 @@ class PyBuildExt(build_ext):
         dbm_setup_debug = False   # verbose debug prints from this script?
         dbm_order = ['gdbm']
         # The standard Unix dbm module:
-        if not CYGWIN:
+        if not MS_WINDOWS:
             config_args = [arg.strip("'")
                            for arg in sysconfig.get_config_var("CONFIG_ARGS").split()]
             dbm_args = [arg for arg in config_args
