--- Python-3.8.0a4-orig/Lib/distutils/cygwinccompiler.py	2019-05-06 14:30:25.000000000 -0400
+++ Python-3.8.0a4/Lib/distutils/cygwinccompiler.py	2019-05-19 14:35:55.872267300 -0400
@@ -92,9 +92,7 @@ class CygwinCCompiler(UnixCCompiler):
     compiler_type = 'cygwin'
     obj_extension = ".o"
     static_lib_extension = ".a"
-    shared_lib_extension = ".dll"
-    static_lib_format = "lib%s%s"
-    shared_lib_format = "%s%s"
+    shared_lib_extension = ".dll.a"
     exe_extension = ".exe"
 
     def __init__(self, verbose=0, dry_run=0, force=0):
--- Python-3.8.0a4-orig/Lib/distutils/unixccompiler.py	2019-05-19 14:24:27.779442200 -0400
+++ Python-3.8.0a4/Lib/distutils/unixccompiler.py	2019-05-19 14:35:55.872267300 -0400
@@ -81,6 +81,7 @@ class UnixCCompiler(CCompiler):
     xcode_stub_lib_format = dylib_lib_format
     if sys.platform == "cygwin":
         exe_extension = ".exe"
+        dylib_lib_extension = ".dll.a"
 
     def preprocess(self, source, output_file=None, macros=None,
                    include_dirs=None, extra_preargs=None, extra_postargs=None):
--- Python-3.8.0b2-orig/Misc/python.pc.in	2019-07-04 06:50:19.000000000 -0400
+++ Python-3.8.0b2/Misc/python.pc.in	2019-07-21 15:33:35.374399700 -0400
@@ -9,5 +9,5 @@ Description: Build a C extension for Pyt
 Requires:
 Version: @VERSION@
 Libs.private: @LIBS@
-Libs:
+Libs: @LIBPYTHON@
 Cflags: -I${includedir}/python@VERSION@@ABIFLAGS@
