--- Python-3.8.0a2-orig/Include/cpython/pyerrors.h	2019-02-25 07:08:32.000000000 -0500
+++ Python-3.8.0a2/Include/cpython/pyerrors.h	2019-03-11 11:35:44.868187500 -0400
@@ -132,7 +132,7 @@ PyAPI_FUNC(PyObject *) _PyErr_TrySetFrom
 
 /* In signalmodule.c */
 
-int PySignal_SetWakeupFd(int fd);
+PyAPI_FUNC(int) PySignal_SetWakeupFd(int fd);
 PyAPI_FUNC(int) _PyErr_CheckSignals(void);
 
 /* Support for adding program text to SyntaxErrors */
