--- Python-3.4.3/Python/dynload_shlib.c.orig	2015-02-25 05:27:46.000000000 -0600
+++ Python-3.4.3/Python/dynload_shlib.c	2015-05-05 11:25:41.882019700 -0500
@@ -34,6 +34,8 @@
 
 const char *_PyImport_DynLoadFiletab[] = {
 #ifdef __CYGWIN__
+    "." SOABI ".dll",
+    ".abi" PYTHON_ABI_STRING ".dll",
     ".dll",
 #else  /* !__CYGWIN__ */
     "." SOABI ".so",
--- Python-3.6.0/configure.ac.orig	2017-02-08 04:39:39.793333600 -0500
+++ Python-3.6.0/configure.ac	2017-02-08 04:45:44.649752400 -0500
@@ -4614,7 +4614,7 @@
 
 AC_SUBST(EXT_SUFFIX)
 case $ac_sys_system in
-    Linux*|GNU*|Darwin)
+    Linux*|GNU*|Darwin|CYGWIN*)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
