--- Python-3.8.0a4-orig/setup.py	2019-05-19 12:20:42.157823300 -0400
+++ Python-3.8.0a4/setup.py	2019-05-19 14:13:32.603602100 -0400
@@ -2216,7 +2216,7 @@ class PyBuildExt(build_ext):
                            depends=sha3_deps))
 
     def detect_nis(self):
-        if MS_WINDOWS or CYGWIN or HOST_PLATFORM == 'qnx6':
+        if MS_WINDOWS or HOST_PLATFORM == 'qnx6':
             self.missing.append('nis')
             return
 
