Modules/parsermodule.o:parsermodule.c: undefined reference to `_PyParser_Grammar'

--- Python-3.8.0a4-orig/Modules/Setup	2019-05-06 14:30:25.000000000 -0400
+++ Python-3.8.0a4/Modules/Setup	2019-05-19 14:14:54.663297000 -0400
@@ -327,7 +327,7 @@ _symtable symtablemodule.c
 #binascii binascii.c
 
 # Fred Drake's interface to the Python parser
-#parser parsermodule.c
+parser parsermodule.c
 
 
 # Andrew Kuchling's zlib module.
