--- Python-3.8.0a3-orig/Python/dynload_shlib.c	2019-03-25 15:36:40.000000000 -0400
+++ Python-3.8.0a3/Python/dynload_shlib.c	2019-03-29 13:18:23.024414000 -0400
@@ -35,6 +35,8 @@
 
 const char *_PyImport_DynLoadFiletab[] = {
 #ifdef __CYGWIN__
+    "." SOABI ".dll",
+    ".abi" PYTHON_ABI_STRING ".dll",
     ".dll",
 #else  /* !__CYGWIN__ */
     "." SOABI ".so",
--- Python-3.13.7/Misc/platform_triplet.c.orig	2025-08-21 08:17:40.030191600 +0200
+++ Python-3.13.7/Misc/platform_triplet.c	2025-08-21 08:18:24.329873900 +0200
@@ -265,6 +265,14 @@
 #  endif
 #elif defined(__VXWORKS__)
 PLATFORM_TRIPLET=vxworks
+#elif defined(__CYGWIN__)
+#  if defined(__x86_64__)
+PLATFORM_TRIPLET=x86_64-cygwin
+#  elif defined(__i386__)
+PLATFORM_TRIPLET=i386-cygwin
+#  else
+#   error unknown platform triplet
+#  endif
 #elif defined(__wasm32__)
 #  if defined(__EMSCRIPTEN__)
 PLATFORM_TRIPLET=wasm32-emscripten
