--- Python-3.8.0a3-orig/Python/dynload_shlib.c	2019-03-25 15:36:40.000000000 -0400
+++ Python-3.8.0a3/Python/dynload_shlib.c	2019-03-29 13:18:23.024414000 -0400
@@ -35,6 +35,8 @@
 
 const char *_PyImport_DynLoadFiletab[] = {
 #ifdef __CYGWIN__
+    "." SOABI ".dll",
+    ".abi" PYTHON_ABI_STRING ".dll",
     ".dll",
 #else  /* !__CYGWIN__ */
     "." SOABI ".so",
--- Python-3.8.0a3-orig/configure.ac	2019-03-29 13:16:29.207031200 -0400
+++ Python-3.8.0a3/configure.ac	2019-03-29 13:20:15.713867100 -0400
@@ -837,6 +837,14 @@ cat >> conftest.c <<EOF
         darwin
 #elif defined(__VXWORKS__)
         vxworks
+#elif defined(__CYGWIN__)
+# if defined(__x86_64__)
+        x86_64-cygwin
+# elif defined(__i386__)
+        i386-cygwin
+# else
+#   error unknown platform triplet
+# endif
 #else
 # error unknown platform triplet
 #endif
@@ -4590,7 +4598,7 @@ AC_MSG_RESULT($SOABI)
 
 AC_SUBST(EXT_SUFFIX)
 case $ac_sys_system in
-    Linux*|GNU*|Darwin|VxWorks)
+    Linux*|GNU*|Darwin|VxWorks|CYGWIN*)
 	EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX};;
     *)
 	EXT_SUFFIX=${SHLIB_SUFFIX};;
