--- Python-3.8.0a3-orig/setup.py	2019-03-29 13:13:05.105468700 -0400
+++ Python-3.8.0a3/setup.py	2019-03-29 13:21:39.245117100 -0400
@@ -1871,12 +1871,6 @@ class PyBuildExt(build_ext):
             include_dirs.append('/usr/X11/include')
             added_lib_dirs.append('/usr/X11/lib')
 
-        # If Cygwin, then verify that X is installed before proceeding
-        if CYGWIN:
-            x11_inc = find_file('X11/Xlib.h', [], include_dirs)
-            if x11_inc is None:
-                return False
-
         # Check for BLT extension
         if self.compiler.find_library_file(self.lib_dirs + added_lib_dirs,
                                                'BLT8.0'):
@@ -1894,9 +1888,8 @@ class PyBuildExt(build_ext):
         if HOST_PLATFORM in ['aix3', 'aix4']:
             libs.append('ld')
 
-        # Finally, link with the X11 libraries (not appropriate on cygwin)
-        if not CYGWIN:
-            libs.append('X11')
+        # Finally, link with the X11 libraries
+        libs.append('X11')
 
         # XXX handle these, but how to detect?
         # *** Uncomment and edit for PIL (TkImaging) extension only:
