--- Python-3.8.0a3-orig/Modules/_xxsubinterpretersmodule.c	2019-03-25 15:36:40.000000000 -0400
+++ Python-3.8.0a3/Modules/_xxsubinterpretersmodule.c	2019-03-29 13:33:54.880859300 -0400
@@ -1735,7 +1735,7 @@ PyDoc_STRVAR(channelid_doc,
 "A channel ID identifies a channel and may be used as an int.");
 
 static PyTypeObject ChannelIDtype = {
-    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    PyVarObject_HEAD_INIT(NULL, 0)
     "_xxsubinterpreters.ChannelID", /* tp_name */
     sizeof(channelid),              /* tp_basicsize */
     0,                              /* tp_itemsize */
