--- Python-3.10.5/Makefile.pre.in.orig	2022-07-02 15:11:36.746831700 +0200
+++ Python-3.10.5/Makefile.pre.in	2022-07-02 15:12:27.208270400 +0200
@@ -1336,6 +1336,7 @@
 		if test -f $(LDLIBRARY) && test "$(PYTHONFRAMEWORKDIR)" = "no-framework" ; then \
 			if test -n "$(DLLLIBRARY)" ; then \
 				$(INSTALL_SHARED) $(DLLLIBRARY) $(DESTDIR)$(BINDIR); \
+				$(INSTALL_DATA) $(LDLIBRARY) $(DESTDIR)$(LIBDIR); \
 			else \
 				$(INSTALL_SHARED) $(LDLIBRARY) $(DESTDIR)$(LIBDIR)/$(INSTSONAME); \
 				if test $(LDLIBRARY) != $(INSTSONAME); then \
