--- Python-3.8.2/Makefile.pre.in.orig	2020-04-16 14:57:03.823588900 +0300
+++ Python-3.8.2/Makefile.pre.in	2020-04-16 15:06:58.593432800 +0300
@@ -1235,6 +1235,7 @@
 	if test -f $(LDLIBRARY) && test "$(PYTHONFRAMEWORKDIR)" = "no-framework" ; then \
 		if test -n "$(DLLLIBRARY)" ; then \
 			$(INSTALL_SHARED) $(DLLLIBRARY) $(DESTDIR)$(BINDIR); \
+			$(INSTALL_DATA) $(LDLIBRARY) $(DESTDIR)$(LIBDIR); \
 		else \
 			$(INSTALL_SHARED) $(LDLIBRARY) $(DESTDIR)$(LIBDIR)/$(INSTSONAME); \
 			if test $(LDLIBRARY) != $(INSTSONAME); then \
