--- Python-3.8.7.cygwin/configure.ac	2021-05-11 21:07:52.839122100 +0200
+++ Python-3.8.7/configure.ac	2021-05-11 21:10:48.366063600 +0200
@@ -1137,7 +1148,7 @@
   case $ac_sys_system in
     CYGWIN*)
           LDLIBRARY='libpython$(LDVERSION).dll.a'
-          DLLLIBRARY='libpython$(LDVERSION).dll'
+          DLLLIBRARY='msys-python$(LDVERSION).dll'
           ;;
     SunOS*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
--- Python-3.8.7.orig/Lib/ctypes/__init__.py	2020-12-21 17:25:24.000000000 +0100
+++ Python-3.8.7/Lib/ctypes/__init__.py	2021-05-11 22:26:25.866873900 +0200
@@ -456,7 +456,7 @@
 if _os.name == "nt":
     pythonapi = PyDLL("python dll", None, _sys.dllhandle)
 elif _sys.platform == "cygwin":
-    pythonapi = PyDLL("libpython%d.%d.dll" % _sys.version_info[:2])
+    pythonapi = PyDLL("msys-python%d.%d.dll" % _sys.version_info[:2])
 else:
     pythonapi = PyDLL(None)
 
--- Python-3.12.4/Lib/test/test_ctypes/test_loading.py.orig	2024-06-06 20:26:44.000000000 +0200
+++ Python-3.12.4/Lib/test/test_ctypes/test_loading.py	2024-07-04 07:35:25.827763700 +0200
@@ -16,7 +16,7 @@
     if os.name == "nt":
         libc_name = find_library("c")
     elif sys.platform == "cygwin":
-        libc_name = "cygwin1.dll"
+        libc_name = "msys-2.0.dll"
     else:
         libc_name = find_library("c")
 
--- Python-3.8.7/Lib/ctypes/util.py.orig	2021-05-11 22:31:37.409285900 +0200
+++ Python-3.8.7/Lib/ctypes/util.py	2021-05-11 22:33:22.371550100 +0200
@@ -387,8 +387,8 @@
             print(f"crypto\t:: {find_library('crypto')}")
             print(f"crypto\t:: {cdll.LoadLibrary(find_library('crypto'))}")
         elif sys.platform == "cygwin":
-            print(cdll.LoadLibrary("cygbz2-1.dll"))
-            print(cdll.LoadLibrary("cygcrypt-0.dll"))
+            print(cdll.LoadLibrary("msys-bz2-1.dll"))
+            print(cdll.LoadLibrary("msys-crypt-0.dll"))
             print(find_library("crypt"))
         else:
             print(cdll.LoadLibrary("libm.so"))
diff -Naur Python-3.8.2-orig/Lib/sysconfig.py Python-3.8.2/Lib/sysconfig.py
--- Python-3.8.2-orig/Lib/sysconfig.py	2020-02-25 00:36:25.000000000 +0300
+++ Python-3.8.2/Lib/sysconfig.py	2020-04-16 11:24:20.643039800 +0300
@@ -673,6 +673,13 @@
         m = rel_re.match(release)
         if m:
             release = m.group()
+    elif osname[:4] == "msys":
+        osname = "msys"
+        import re
+        rel_re = re.compile (r'[\d.]+')
+        m = rel_re.match(release)
+        if m:
+            release = m.group()
     elif osname[:6] == "darwin":
         import _osx_support
         osname, release, machine = _osx_support.get_platform_osx(
