--- Python-3.8.5/Makefile.pre.in.orig	2020-08-10 16:49:23.697127200 -0700
+++ Python-3.8.5/Makefile.pre.in	2020-08-10 16:51:30.546101400 -0700
@@ -667,6 +667,10 @@
 	if test -n "$(DLLLIBRARY)"; then \
 		$(LDSHARED) -Wl,--out-implib=$@ -o $(DLLLIBRARY) $^ \
 			$(LIBS) $(MODLIBS) $(SYSLIBS); \
+		if [[ `uname -m` == i?86 ]]; then \
+			rebase -sO $(DLLLIBRARY); \
+		else true; \
+		fi; \
 	else true; \
 	fi
 
