--- Python-3.8.5/Makefile.pre.in.orig	2020-08-13 17:35:32.636969100 -0700
+++ Python-3.8.5/Makefile.pre.in	2020-08-13 23:24:10.054938400 -0700
@@ -609,6 +609,7 @@
 	$(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' \
 		_TCLTK_INCLUDES='$(TCLTK_INCLUDES)' _TCLTK_LIBS='$(TCLTK_LIBS)' \
 		$(PYTHON_FOR_BUILD) $(srcdir)/setup.py $$quiet build
+	[[ `uname -m` == i?86 ]] && find . -name \*.dll | rebase -sOT - || true
 
 libpython$(LDVERSION).so: $(LIBRARY_OBJS) $(DTRACE_OBJS)
 	if test $(INSTSONAME) != $(LDLIBRARY); then \
