--- Python-3.12.4/Makefile.pre.in.orig	2024-07-04 07:37:58.547224000 +0200
+++ Python-3.12.4/Makefile.pre.in	2024-07-04 07:41:07.755948400 +0200
@@ -995,6 +995,7 @@
 			$(LN) -sf ../../$$mod $$target/`basename $$mod`; \
 		fi; \
 	done
+	[[ `uname -m` == i?86 ]] && find . -name \*.dll | rebase -sOT - || true
 
 # dependency on BUILDPYTHON ensures that the target is run last
 .PHONY: checksharedmods
