--- Python-3.8.5/Makefile.pre.in.orig	2020-08-14 12:32:57.031236300 -0700
+++ Python-3.8.5/Makefile.pre.in	2020-08-14 12:33:19.884645300 -0700
@@ -1256,7 +1256,7 @@
 			$(DESTDIR)$(BINDIR)/python$(VERSION)$(EXE); \
 	fi
 
-bininstall: altbininstall
+bininstall: altbininstall commoninstall
 	if test ! -d $(DESTDIR)$(LIBPC); then \
 		echo "Creating directory $(LIBPC)"; \
 		$(INSTALL) -d -m $(DIRMODE) $(DESTDIR)$(LIBPC); \
