--- Python-3.11.2/Modules/ossaudiodev.c.orig	2023-02-07 14:37:51.000000000 +0100
+++ Python-3.11.2/Modules/ossaudiodev.c	2023-02-13 20:51:13.398293500 +0100
@@ -964,7 +964,7 @@
 };
 
 static PyTypeObject OSSAudioType = {
-    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    PyVarObject_HEAD_INIT(NULL, 0)
     "ossaudiodev.oss_audio_device", /*tp_name*/
     sizeof(oss_audio_t),        /*tp_basicsize*/
     0,                          /*tp_itemsize*/
@@ -998,7 +998,7 @@
 };
 
 static PyTypeObject OSSMixerType = {
-    PyVarObject_HEAD_INIT(&PyType_Type, 0)
+    PyVarObject_HEAD_INIT(NULL, 0)
     "ossaudiodev.oss_mixer_device", /*tp_name*/
     sizeof(oss_mixer_t),            /*tp_basicsize*/
     0,                              /*tp_itemsize*/
