--- Python-3.13.7/configure.ac.orig	2025-08-14 13:12:11.000000000 +0200
+++ Python-3.13.7/configure.ac	2025-09-12 13:23:33.609233300 +0200
@@ -6320,6 +6320,10 @@
 if test "$PY_ENABLE_SHARED" = "1" && ( test -n "$ANDROID_API_LEVEL" || test "$MACHDEP" = "cygwin"); then
   MODULE_DEPS_SHARED="$MODULE_DEPS_SHARED \$(LDLIBRARY)"
   LIBPYTHON="\$(BLDLIBRARY)"
+  LIBPYTHON="$BLDLIBRARY"
+  LIBPYTHON=$(echo "$LIBPYTHON" | sed "s/\$(LDVERSION)/$LDVERSION/g")
+  LIBPYTHON=$(echo "$LIBPYTHON" | sed "s/\$(VERSION)/$VERSION/g")
+  LIBPYTHON=$(echo "$LIBPYTHON" | sed "s/\$(ABIFLAGS)/$ABIFLAGS/g")
 fi
 
 # On iOS the shared libraries must be linked with the Python framework
