--- pkgfile-26/meson.build	2026-09-03 21:44:36.300937000 +0000
+++ pkgfile-26/meson.build	2026-09-03 21:48:16.000935100 +0000
@@ -27,7 +27,7 @@
 libpcre = dependency('libpcre', version: '>= 8.30')
 libarchive = dependency('libarchive', version: '>= 3.2.0')
 libcurl = dependency('libcurl')
-libsystemd = dependency('libsystemd')
+libsystemd = dependency('', required: false)
 pthreads = dependency('threads')
 stdcppfs = cpp.find_library('stdc++fs')
 gtest = dependency('gtest', required: false)
@@ -77,14 +77,6 @@
     install: true,
 )
 
-executable(
-    'pkgfiled',
-    'src/pkgfiled.cc',
-    link_with: [libcommon],
-    dependencies: [libsystemd, stdcppfs],
-    install: true,
-)
-
 # Debugging aid for inspecting the PFDB repo database format; not part of
 # the supported user-facing tooling, so it's never installed.
 executable(
@@ -112,23 +104,6 @@
     install_dir: join_paths(get_option('mandir'), 'man1'),
 )
 
-custom_target(
-    'pkgfiled.1',
-    output: 'pkgfiled.1',
-    input: 'man/pkgfiled.pod',
-    command: [
-        pod2man,
-        '--section=1',
-        '--center=pkgfiled Manual',
-        '--name=PKGFILED',
-        '--release=pkgfiled @0@'.format(meson.project_version()),
-        '@INPUT@',
-        '@OUTPUT@',
-    ],
-    install: true,
-    install_dir: join_paths(get_option('mandir'), 'man1'),
-)
-
 if get_option('systemd_units')
     systemd = dependency(
         'systemd',
@@ -144,13 +119,6 @@
         configuration: conf,
         install_dir: systemunitdir,
     )
-
-    configure_file(
-        input: 'systemd/pkgfiled.service.in',
-        output: 'pkgfiled.service',
-        configuration: conf,
-        install_dir: systemunitdir,
-    )
 endif
 
 command_not_found_hooks = files(
@@ -218,7 +186,6 @@
     integration_tests = [
       'tests/database.py',
       'tests/list.py',
-      'tests/pkgfiled.py',
       'tests/search.py',
       'tests/update.py'
     ]
