diff -urN a/argp/gllib/Makefile.am b/argp/gllib/Makefile.am
--- a/argp/gllib/Makefile.am	2023-07-11 06:59:35.845600000 -0400
+++ b/argp/gllib/Makefile.am	2023-07-10 15:16:10.822421600 -0400
@@ -41,7 +41,8 @@
 AM_CPPFLAGS = -DGNULIB_STRICT_CHECKING=1
 AM_CFLAGS =
 
-noinst_LTLIBRARIES += libargp.la
+lib_LTLIBRARIES = libargp.la
+include_HEADERS = argp.h
 
 libargp_la_SOURCES =
 libargp_la_LIBADD = $(gl_LTLIBOBJS)
