--- crypt/Makefile.orig	2014-10-31 00:14:16.525400000 +0300
+++ crypt/Makefile	2014-10-31 00:14:39.395000000 +0300
@@ -7,12 +7,12 @@
 bindir=$(prefix)/bin
 libdir=$(prefix)/lib
 incdir=$(prefix)/include
-docdir=$(prefix)/share/doc/Cygwin
+docdir=$(prefix)/share/doc/MSYS
 DLLVER=0
 EXE=.exe
 
 STATICLIB=libcrypt.a
-SHAREDLIB=cygcrypt-$(DLLVER).dll
+SHAREDLIB=msys-crypt-$(DLLVER).dll
 IMPORTLIB=libcrypt.dll.a
 
 APPS=crypt$(EXE)
