diff -urN Sys-CPU-0.61.orig/CPU.xs Sys-CPU-0.61/CPU.xs
--- Sys-CPU-0.61.orig/CPU.xs	2015-01-19 21:41:05.230451600 +0000
+++ Sys-CPU-0.61/CPU.xs	2015-01-19 21:41:08.702650200 +0000
@@ -11,7 +11,7 @@
 **************************************************************************************/
 
 #define MAX_IDENT_SIZE 256
-#if defined(_WIN32) || defined(WIN32)
+#if defined(_WIN32) || defined(WIN32) || defined(__CYGWIN__)
   #define _have_cpu_type
   #define _have_cpu_clock
   #define WINDOWS
