--- origsrc/gdbm-1.10/compat/Makefile.am	2011-08-14 15:17:54.000000000 -0500
+++ src/gdbm-1.10/compat/Makefile.am	2013-02-19 17:24:54.209585900 -0600
@@ -50,5 +50,5 @@ NDBM_CF=\
 
 libgdbm_compat_la_SOURCES = $(DBM_CF) $(NDBM_CF)
 
-libgdbm_compat_la_LDFLAGS = -version-info $(VI_CURRENT):$(VI_REVISION):$(VI_AGE)
+libgdbm_compat_la_LDFLAGS = -version-info $(VI_CURRENT):$(VI_REVISION):$(VI_AGE) -no-undefined
 
--- origsrc/gdbm-1.10/src/Makefile.am	2011-11-12 13:13:01.000000000 -0600
+++ src/gdbm-1.10/src/Makefile.am	2013-02-19 17:25:10.264504100 -0600
@@ -60,7 +60,7 @@ libgdbm_la_SOURCES = \
  update.c\
  version.c
 
-libgdbm_la_LDFLAGS = -version-info $(VI_CURRENT):$(VI_REVISION):$(VI_AGE)
+libgdbm_la_LDFLAGS = -version-info $(VI_CURRENT):$(VI_REVISION):$(VI_AGE) -no-undefined
 
 # Programs
 bin_PROGRAMS = testgdbm 
