--- origsrc/gdbm-1.11/src/Makefile.am	2014-05-26 21:27:28.220164600 -0500
+++ src/gdbm-1.11/src/Makefile.am	2014-05-26 21:27:43.507105800 -0500
@@ -89,8 +89,8 @@ libgdbmapp_a_SOURCES =\
 bin_PROGRAMS = gdbmtool gdbm_load gdbm_dump
 
 gdbmtool_LDADD  = \
- ./libgdbmapp.a\
- ./libgdbm.la\
+ libgdbmapp.a \
+ libgdbm.la \
  @READLINE_LIBS@
 
 gdbmtool_SOURCES = \
@@ -92,5 +92,5 @@ gdbmtool_SOURCES = \
 AM_YFLAGS = -dtv
 #AM_LFLAGS = -d
 
-gdbm_load_LDADD  = ./libgdbmapp.a ./libgdbm.la 
-gdbm_dump_LDADD  = ./libgdbmapp.a ./libgdbm.la 
+gdbm_load_LDADD  = libgdbmapp.a libgdbm.la 
+gdbm_dump_LDADD  = libgdbmapp.a libgdbm.la 
