diff -Naur gdbm-1.15.orig/src/gdbmopen.c gdbm-1.15/src/gdbmopen.c
--- gdbm-1.15.orig/src/gdbmopen.c	2018-05-06 21:50:23.243439700 -0400
+++ gdbm-1.15/src/gdbmopen.c	2018-05-07 14:15:12.620590400 -0400
@@ -656,7 +656,7 @@
       for (index = 0; index < size; index++)
         {
 	  (dbf->bucket_cache[index]).ca_bucket =
-	    malloc (dbf->header->bucket_size);
+	    calloc (1, dbf->header->bucket_size);
           if ((dbf->bucket_cache[index]).ca_bucket == NULL)
 	    {
               GDBM_SET_ERRNO (dbf, GDBM_MALLOC_ERROR, TRUE);