--- crypt/Makefile.orig	2014-10-31 00:14:16.525400000 +0300
+++ crypt/Makefile	2014-10-31 00:14:39.395000000 +0300
@@ -8,11 +8,11 @@
 bindir=$(prefix)/bin
 libdir=$(prefix)/lib
 incdir=$(prefix)/include
-docdir=$(prefix)/share/doc/Cygwin
+docdir=$(prefix)/share/doc/Msys
 DLLVER=0
 
 STATICLIB=libcrypt.a
-SHAREDLIB=cygcrypt-$(DLLVER).dll
+SHAREDLIB=msys-crypt-$(DLLVER).dll
 IMPORTLIB=libcrypt.dll.a
 
 LIBS=$(STATICLIB) $(SHAREDLIB)
--- crypt/crypt.def.orig	2014-11-01 00:41:22.483000000 +0300
+++ crypt/crypt.def	2014-11-01 00:41:27.865000000 +0300
@@ -1,4 +1,4 @@
-LIBRARY "cygcrypt-0.dll"
+LIBRARY "msys-crypt-0.dll"
 
 EXPORTS
 encrypt
