--- gamin-0.1.10/libgamin/Makefile.am.orig	2007-07-04 17:36:48.000000000 +0400
+++ gamin-0.1.10/libgamin/Makefile.am	2014-09-26 10:31:01.008000000 +0400
@@ -40,12 +40,12 @@
 libgamin_1_la_LIBADD =
 
 libgamin_1_la_LDFLAGS = -Wl,--version-script=$(srcdir)/gamin_sym.version \
-                        -version-info @GAMIN_VERSION_INFO@ @THREAD_LIBS@
+                        -version-info @GAMIN_VERSION_INFO@ @THREAD_LIBS@ -no-undefined
 
 libfam_la_SOURCES = $(libgamin_1_la_SOURCES)
 libfam_la_LIBADD = $(libgamin_1_la_LIBADD)
 libfam_la_LDFLAGS = -Wl,--version-script=$(srcdir)/gamin_sym.version	\
-                    -version-info @FAM_VERSION_INFO@ @THREAD_LIBS@
+                    -version-info @FAM_VERSION_INFO@ @THREAD_LIBS@ -no-undefined
 
 #
 # Compile a program locally to check
--- gamin-0.1.10/python/Makefile.am.orig	2007-07-04 17:36:48.000000000 +0400
+++ gamin-0.1.10/python/Makefile.am	2014-09-26 10:31:01.008000000 +0400
@@ -11,7 +11,7 @@
 
 _gamin_la_SOURCES = gamin.c
 _gamin_la_LIBADD = $(top_builddir)/libgamin/libgamin-1.la
-_gamin_la_LDFLAGS = -module -avoid-version
+_gamin_la_LDFLAGS = -module -avoid-version -no-undefined
 
 python_PYTHON = gamin.py
 
