diff -urN ack-2.12.orig/Ack.pm ack-2.12/Ack.pm
--- ack-2.12.orig/Ack.pm	2013-12-03 13:05:35.000000000 +0000
+++ ack-2.12/Ack.pm	2014-04-20 12:24:27.100739600 +0100
@@ -46,7 +46,7 @@
     $output_to_pipe  = not -t *STDOUT;
     $is_filter_mode = -p STDIN;
 
-    $is_cygwin       = ($^O eq 'cygwin');
+    $is_cygwin       = ($^O eq 'cygwin' || $^O eq 'msys');
     $is_windows      = ($^O eq 'MSWin32');
     $dir_sep_chars   = $is_windows ? quotemeta( '\\/' ) : quotemeta( File::Spec->catfile( '', '' ) );
 }
diff -urN ack-2.12.orig/t/Util.pm ack-2.12/t/Util.pm
--- ack-2.12.orig/t/Util.pm	2013-11-30 18:34:00.000000000 +0000
+++ ack-2.12/t/Util.pm	2014-04-20 12:24:27.104739800 +0100
@@ -32,7 +32,7 @@
 }
 
 sub is_cygwin {
-    return $^O eq 'cygwin';
+    return $^O eq 'cygwin' or $^O eq 'msys';
 }
 
 sub is_empty_array {
