diff -urN ack-v3.0.0.orig/lib/App/Ack/ConfigDefault.pm ack-v.3.0.0/lib/App/Ack/ConfigDefault.pm
--- ack-v3.0.0.orig/lib/App/Ack/ConfigDefault.pm	2013-11-30 18:34:00.000000000 +0000
+++ ack-v.3.0.0/lib/App/Ack/ConfigDefault.pm	2014-04-20 12:19:57.886341400 +0100
@@ -176,6 +176,7 @@
 --type-add=make:is:Makefile.Debug
 --type-add=make:is:Makefile.Release
 --type-add=make:is:GNUmakefile
+--type-add=make:match:.*(M|m)akefile.*
 
 # Rakefiles
 # http://rake.rubyforge.org/
