diff -urN ack-2.12.orig/ConfigDefault.pm ack-2.12/ConfigDefault.pm
--- ack-2.12.orig/ConfigDefault.pm	2013-11-30 18:34:00.000000000 +0000
+++ ack-2.12/ConfigDefault.pm	2014-04-20 12:19:57.886341400 +0100
@@ -132,9 +132,7 @@
 # Makefiles http://www.gnu.org/s/make/
 --type-add=make:ext:mk
 --type-add=make:ext:mak
---type-add=make:is:makefile
---type-add=make:is:Makefile
---type-add=make:is:GNUmakefile
+--type-add=make:match:.*(M|m)akefile.*
 
 # Rakefiles http://rake.rubyforge.org/
 --type-add=rake:is:Rakefile
