diff -Naur ack-2.10-orig/Ack.pm ack-2.10/Ack.pm
--- ack-2.10-orig/Ack.pm	2013-09-25 01:35:57.000000000 +0400
+++ ack-2.10/Ack.pm	2013-11-17 19:55:43.793800000 +0400
@@ -48,7 +48,7 @@
     $output_to_pipe  = not -t *STDOUT;
     $is_filter_mode = -p STDIN;
 
-    $is_cygwin       = ($^O eq 'cygwin');
+    $is_cygwin       = ($^O eq 'cygwin' || $^O eq 'msys');
     $is_windows      = ($^O eq 'MSWin32');
     $dir_sep_chars   = $is_windows ? quotemeta( '\\/' ) : quotemeta( File::Spec->catfile( '', '' ) );
 }
diff -Naur ack-2.10-orig/t/Util.pm ack-2.10/t/Util.pm
--- ack-2.10-orig/t/Util.pm	2013-09-25 01:35:57.000000000 +0400
+++ ack-2.10/t/Util.pm	2013-11-17 19:56:15.040600000 +0400
@@ -31,7 +31,7 @@
 }
 
 sub is_cygwin {
-    return $^O eq 'cygwin';
+    return $^O eq 'cygwin' or $^O eq 'msys';
 }
 
 sub build_ack_invocation {
