--- sed-4.4/sed/sed.c.orig	2017-02-23 22:01:05.503921600 +0300
+++ sed-4.4/sed/sed.c	2017-02-23 22:01:46.289254400 +0300
@@ -61,7 +61,11 @@
 char *in_place_extension = NULL;
 
 /* The mode to use to read/write files, either "r"/"w" or "rb"/"wb".  */
+#ifdef __MSYS__
+char const *read_mode = "rt";
+#else
 char const *read_mode = "r";
+#endif
 char const *write_mode = "w";
 
 /* Do we need to be pedantically POSIX compliant? */
