diff -Naur ExtUtils-InstallPaths-0.012-orig/t/destinations.t ExtUtils-InstallPaths-0.012/t/destinations.t
--- ExtUtils-InstallPaths-0.012-orig/t/destinations.t	2018-05-10 22:03:35.000000000 +0300
+++ ExtUtils-InstallPaths-0.012/t/destinations.t	2020-05-30 12:11:33.129752200 +0300
@@ -272,11 +272,11 @@
 sub have_same_ending {
 	my ($dir1, $dir2, $message) = @_;
 
-	$dir1 =~ s{/$}{} if $^O eq 'cygwin'; # remove any trailing slash
+	$dir1 =~ s{/$}{} if $^O eq 'cygwin' || $^O eq 'msys'; # remove any trailing slash
 	my (undef, $dirs1, undef) = splitpath $dir1;
 	my @dir1 = splitdir $dirs1;
 
-	$dir2 =~ s{/$}{} if $^O eq 'cygwin'; # remove any trailing slash
+	$dir2 =~ s{/$}{} if $^O eq 'cygwin' || $^O eq 'msys'; # remove any trailing slash
 	my (undef, $dirs2, undef) = splitpath $dir2;
 	my @dir2 = splitdir $dirs2;
 
