From e578267041203e225c8da57064dac125edb24cc8 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Thu, 18 Dec 2025 08:24:47 +0800
Subject: [PATCH] fixes building with gcc 15

---
 srclib/stddef.in.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/srclib/stddef.in.h b/srclib/stddef.in.h
index e39268c..2bedc05 100644
--- a/srclib/stddef.in.h
+++ b/srclib/stddef.in.h
@@ -22,6 +22,7 @@
  * <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/stddef.h.html>
  */
 
+
 #if __GNUC__ >= 3
 @PRAGMA_SYSTEM_HEADER@
 #endif
@@ -97,6 +98,8 @@ typedef long max_align_t;
 
 /* The include_next requires a split double-inclusion guard.  */
 
+#define __need_wint_t
+#define _STDDEF_H
 #  @INCLUDE_NEXT@ @NEXT_STDDEF_H@
 
 /* On NetBSD 5.0, the definition of NULL lacks proper parentheses.  */
-- 
2.51.0.windows.1

