commit 4309f6f45a3b735649f4baa7dad53bd69e9fbefe
Author: Lemures Lemniscati <lemures.lemniscati@gmail.com>
Date:   Sat Jul 11 10:10:24 2020 +0900

    1.16-aliases.patch

diff --git a/lib/encodings.def b/lib/encodings.def
index 402b5e3..3d925e3 100644
--- a/lib/encodings.def
+++ b/lib/encodings.def
@@ -46,7 +46,7 @@ DEFENCODING(( "US-ASCII",               /* IANA */
               "IBM367",                 /* IANA */
               "US",                     /* IANA */
               "csASCII",                /* IANA */
-            /*"ISO646.1991-IRV",           X11R6.4 */
+              "ISO646.1991-IRV",        /* X11R6.4 */
             ),
             ascii,
             { ascii_mbtowc, NULL },       { ascii_wctomb, NULL })
@@ -58,17 +58,14 @@ DEFALIAS(     "646",                    /* Solaris */
 /* General multi-byte encodings */
 
 DEFENCODING(( "UTF-8",                  /* IANA, RFC 2279 */
-            /*"UTF8",                      JDK 1.1 */
-            /*"CP65001",                   Windows */
+              "UTF8",                   /* JDK 1.1 */
+              "CP65001",                /* Windows */
             ),
             utf8,
             { utf8_mbtowc, NULL },        { utf8_wctomb, NULL })
-#ifdef USE_HPUX_ALIASES
-DEFALIAS(     "UTF8",                   /* HP-UX */
-            utf8)
-#endif
 
 DEFENCODING(( "UCS-2",                  /* glibc */
+              "UCS2",                   /* glibc */
               "ISO-10646-UCS-2",        /* IANA */
               "csUnicode",              /* IANA */
             ),
@@ -76,22 +73,25 @@ DEFENCODING(( "UCS-2",                  /* glibc */
             { ucs2_mbtowc, NULL },        { ucs2_wctomb, NULL })
 
 DEFENCODING(( "UCS-2BE",                /* glibc */
+              "UCS2BE",                 /* glibc */
               "UNICODEBIG",             /* glibc */
               "UNICODE-1-1",            /* IANA */
               "csUnicode11",            /* IANA */
-            /*"CP1201",                    Windows */
+              "CP1201",                 /* Windows */
             ),
             ucs2be,
             { ucs2be_mbtowc, NULL },      { ucs2be_wctomb, NULL })
 
 DEFENCODING(( "UCS-2LE",                /* glibc */
+              "UCS2LE",                 /* glibc */
               "UNICODELITTLE",          /* glibc */
-            /*"CP1200",                    Windows */
+              "CP1200",                 /* Windows */
             ),
             ucs2le,
             { ucs2le_mbtowc, NULL },      { ucs2le_wctomb, NULL })
 
 DEFENCODING(( "UCS-4",                  /* glibc */
+              "UCS4",                   /* glibc */
               "ISO-10646-UCS-4",        /* IANA */
               "csUCS4",                 /* IANA */
             ),
@@ -99,43 +99,52 @@ DEFENCODING(( "UCS-4",                  /* glibc */
             { ucs4_mbtowc, NULL },        { ucs4_wctomb, NULL })
 
 DEFENCODING(( "UCS-4BE",                /* glibc */
-            /*"CP12001",                   Windows */
+              "UCS4BE",                 /* glibc */
+              "CP12001",                /* Windows */
             ),
             ucs4be,
             { ucs4be_mbtowc, NULL },      { ucs4be_wctomb, NULL })
 
 DEFENCODING(( "UCS-4LE",                /* glibc */
-            /*"CP12000",                   Windows */
+              "UCS4LE",                 /* glibc */
+              "CP12000",                /* Windows */
             ),
             ucs4le,
             { ucs4le_mbtowc, NULL },      { ucs4le_wctomb, NULL })
 
 DEFENCODING(( "UTF-16",                 /* IANA, RFC 2781 */
+              "UTF16",                  /* glibc */
             ),
             utf16,
             { utf16_mbtowc, NULL },       { utf16_wctomb, NULL })
 
 DEFENCODING(( "UTF-16BE",               /* IANA, RFC 2781 */
+              "UTF16BE",                /* glibc */
             ),
             utf16be,
             { utf16be_mbtowc, NULL },     { utf16be_wctomb, NULL })
 
 DEFENCODING(( "UTF-16LE",               /* IANA, RFC 2781 */
+              "UTF16LE",                /* glibc */
+              "UNICODEFFFE",            /* glibc */
             ),
             utf16le,
             { utf16le_mbtowc, NULL },     { utf16le_wctomb, NULL })
 
 DEFENCODING(( "UTF-32",                 /* IANA, Unicode 3.1 */
+              "UTF32",                  /* glibc */
             ),
             utf32,
             { utf32_mbtowc, NULL },       { utf32_wctomb, NULL })
 
 DEFENCODING(( "UTF-32BE",               /* IANA, Unicode 3.1 */
+              "UTF32BE",                /* glibc */
             ),
             utf32be,
             { utf32be_mbtowc, NULL },     { utf32be_wctomb, NULL })
 
 DEFENCODING(( "UTF-32LE",               /* IANA, Unicode 3.1 */
+              "UTF32LE",                /* glibc */
             ),
             utf32le,
             { utf32le_mbtowc, NULL },     { utf32le_wctomb, NULL })
@@ -143,7 +152,7 @@ DEFENCODING(( "UTF-32LE",               /* IANA, Unicode 3.1 */
 DEFENCODING(( "UTF-7",                  /* IANA, RFC 2152 */
               "UNICODE-1-1-UTF-7",      /* IANA, RFC 1642 */
               "csUnicode11UTF7",        /* IANA */
-            /*"CP65000",                   Windows */
+              "CP65000",                /* Windows */
             ),
             utf7,
             { utf7_mbtowc, NULL },        { utf7_wctomb, utf7_reset })
@@ -190,8 +199,8 @@ DEFENCODING(( "ISO-8859-1",             /* IANA */
               "L1",                     /* IANA */
               "csISOLatin1",            /* IANA */
               "ISO8859-1",              /* X11R6.4, glibc, FreeBSD, AIX, IRIX, OSF/1, Solaris */
-            /*"ISO8859_1",                 JDK 1.1 */
-            /*"CP28591",                   Windows */
+              "ISO8859_1",              /* JDK 1.1 */
+              "CP28591",                /* Windows */
             ),
             iso8859_1,
             { iso8859_1_mbtowc, NULL },   { iso8859_1_wctomb, NULL })
@@ -208,8 +217,8 @@ DEFENCODING(( "ISO-8859-2",             /* IANA */
               "L2",                     /* IANA */
               "csISOLatin2",            /* IANA */
               "ISO8859-2",              /* X11R6.4, glibc, FreeBSD, AIX, IRIX, OSF/1, Solaris */
-            /*"ISO8859_2",                 JDK 1.1 */
-            /*"CP28592",                   Windows */
+              "ISO8859_2",              /* JDK 1.1 */
+              "CP28592",                /* Windows */
             ),
             iso8859_2,
             { iso8859_2_mbtowc, NULL },   { iso8859_2_wctomb, NULL })
@@ -226,8 +235,8 @@ DEFENCODING(( "ISO-8859-3",             /* IANA */
               "L3",                     /* IANA */
               "csISOLatin3",            /* IANA */
               "ISO8859-3",              /* X11R6.4, glibc, FreeBSD, Solaris */
-            /*"ISO8859_3",                 JDK 1.1 */
-            /*"CP28593",                   Windows */
+              "ISO8859_3",              /* JDK 1.1 */
+              "CP28593",                /* Windows */
             ),
             iso8859_3,
             { iso8859_3_mbtowc, NULL },   { iso8859_3_wctomb, NULL })
@@ -240,8 +249,8 @@ DEFENCODING(( "ISO-8859-4",             /* IANA */
               "L4",                     /* IANA */
               "csISOLatin4",            /* IANA */
               "ISO8859-4",              /* X11R6.4, glibc, FreeBSD, OSF/1, Solaris */
-            /*"ISO8859_4",                 JDK 1.1 */
-            /*"CP28594",                   Windows */
+              "ISO8859_4",              /* JDK 1.1 */
+              "CP28594",                /* Windows */
             ),
             iso8859_4,
             { iso8859_4_mbtowc, NULL },   { iso8859_4_wctomb, NULL })
@@ -253,8 +262,8 @@ DEFENCODING(( "ISO-8859-5",             /* IANA */
               "CYRILLIC",               /* IANA */
               "csISOLatinCyrillic",     /* IANA */
               "ISO8859-5",              /* X11R6.4, glibc, FreeBSD, AIX, IRIX, OSF/1, Solaris */
-            /*"ISO8859_5",                 JDK 1.1 */
-            /*"CP28595",                   Windows */
+              "ISO8859_5",              /* JDK 1.1 */
+              "CP28595",                /* Windows */
             ),
             iso8859_5,
             { iso8859_5_mbtowc, NULL },   { iso8859_5_wctomb, NULL })
@@ -272,8 +281,8 @@ DEFENCODING(( "ISO-8859-6",             /* IANA */
               "ARABIC",                 /* IANA */
               "csISOLatinArabic",       /* IANA */
               "ISO8859-6",              /* X11R6.4, glibc, FreeBSD, AIX, Solaris */
-            /*"ISO8859_6",                 JDK 1.1 */
-            /*"CP28596",                   Windows */
+              "ISO8859_6",              /* JDK 1.1 */
+              "CP28596",                /* Windows */
             ),
             iso8859_6,
             { iso8859_6_mbtowc, NULL },   { iso8859_6_wctomb, NULL })
@@ -293,8 +302,8 @@ DEFENCODING(( "ISO-8859-7",             /* IANA, RFC 1947 */
               "GREEK",                  /* IANA */
               "csISOLatinGreek",        /* IANA */
               "ISO8859-7",              /* X11R6.4, glibc, FreeBSD, AIX, IRIX, OSF/1, Solaris */
-            /*"ISO8859_7",                 JDK 1.1 */
-            /*"CP28597",                   Windows */
+              "ISO8859_7",              /* JDK 1.1 */
+              "CP28597",                /* Windows */
             ),
             iso8859_7,
             { iso8859_7_mbtowc, NULL },   { iso8859_7_wctomb, NULL })
@@ -310,9 +319,9 @@ DEFENCODING(( "ISO-8859-8",             /* IANA */
               "HEBREW",                 /* IANA */
               "csISOLatinHebrew",       /* IANA */
               "ISO8859-8",              /* X11R6.4, glibc, FreeBSD, AIX, OSF/1, Solaris */
-            /*"ISO8859_8",                 JDK 1.1 */
-            /*"CP28598",                   Windows */
-            /*"CP38598",                   Windows */
+              "ISO8859_8",              /* JDK 1.1 */
+              "CP28598",                /* Windows */
+              "CP38598",                /* Windows */
             ),
             iso8859_8,
             { iso8859_8_mbtowc, NULL },   { iso8859_8_wctomb, NULL })
@@ -329,8 +338,8 @@ DEFENCODING(( "ISO-8859-9",             /* IANA */
               "L5",                     /* IANA */
               "csISOLatin5",            /* IANA */
               "ISO8859-9",              /* X11R6.4, glibc, FreeBSD, AIX, IRIX, OSF/1, Solaris */
-            /*"ISO8859_9",                 JDK 1.1 */
-            /*"CP28599",                   Windows */
+              "ISO8859_9",              /* JDK 1.1 */
+              "CP28599",                /* Windows */
             ),
             iso8859_9,
             { iso8859_9_mbtowc, NULL },   { iso8859_9_wctomb, NULL })
@@ -364,7 +373,7 @@ DEFENCODING(( "ISO-8859-13",            /* IANA, glibc */
               "LATIN7",                 /* glibc */
               "L7",                     /* glibc */
               "ISO8859-13",             /* glibc, FreeBSD */
-            /*"CP28603",                   Windows */
+              "CP28603",                /* Windows */
             ),
             iso8859_13,
             { iso8859_13_mbtowc, NULL },  { iso8859_13_wctomb, NULL })
@@ -391,7 +400,7 @@ DEFENCODING(( "ISO-8859-15",            /* IANA, glibc */
               "ISO-IR-203",
               "LATIN-9",                /* IANA */
               "ISO8859-15",             /* glibc, FreeBSD, AIX, OSF/1, Solaris */
-            /*"CP28605",                   Windows */
+              "CP28605",                /* Windows */
             ),
             iso8859_15,
             { iso8859_15_mbtowc, NULL },  { iso8859_15_wctomb, NULL })
@@ -413,7 +422,7 @@ DEFENCODING(( "ISO-8859-16",            /* IANA */
 
 DEFENCODING(( "KOI8-R",                 /* IANA, RFC 1489, X11R6.4, JDK 1.1 */
               "csKOI8R",                /* IANA */
-            /*"CP20866",                   Windows */
+              "CP20866",                /* Windows */
             ),
             koi8_r,
             { koi8_r_mbtowc, NULL },      { koi8_r_wctomb, NULL })
@@ -424,7 +433,7 @@ DEFENCODING(( "KOI8-U",                 /* IANA, RFC 2319 */
             { koi8_u_mbtowc, NULL },      { koi8_u_wctomb, NULL })
 
 DEFENCODING(( "KOI8-RU",
-            /*"CP21866",                   Windows */
+              "CP21866",                /* Windows */
             ),
             koi8_ru,
             { koi8_ru_mbtowc, NULL },     { koi8_ru_wctomb, NULL })
@@ -548,73 +557,73 @@ DEFENCODING(( "MacRoman",               /* JDK 1.1 */
               "MACINTOSH",              /* IANA */
               "MAC",                    /* IANA */
               "csMacintosh",            /* IANA */
-            /*"CP10000",                   Windows */
+              "CP10000",                /* Windows */
             ),
             mac_roman,
             { mac_roman_mbtowc, NULL },   { mac_roman_wctomb, NULL })
 
 DEFENCODING(( "MacCentralEurope",       /* JDK 1.1 */
-            /*"CP10029",                   Windows */
+              "CP10029",                /* Windows */
             ),
             mac_centraleurope,
             { mac_centraleurope_mbtowc, NULL }, { mac_centraleurope_wctomb, NULL })
 
 DEFENCODING(( "MacIceland",             /* JDK 1.1 */
-            /*"CP10079",                   Windows */
+              "CP10079",                /* Windows */
             ),
             mac_iceland,
             { mac_iceland_mbtowc, NULL }, { mac_iceland_wctomb, NULL })
 
 DEFENCODING(( "MacCroatian",            /* JDK 1.1 */
-            /*"CP10082",                   Windows */
+              "CP10082",                /* Windows */
             ),
             mac_croatian,
             { mac_croatian_mbtowc, NULL }, { mac_croatian_wctomb, NULL })
 
 DEFENCODING(( "MacRomania",             /* JDK 1.1 */
-            /*"CP10010",                   Windows */
+              "CP10010",                /* Windows */
             ),
             mac_romania,
             { mac_romania_mbtowc, NULL }, { mac_romania_wctomb, NULL })
 
 DEFENCODING(( "MacCyrillic",            /* JDK 1.1 */
-            /*"CP10007",                   Windows */
+              "CP10007",                /* Windows */
             ),
             mac_cyrillic,
             { mac_cyrillic_mbtowc, NULL }, { mac_cyrillic_wctomb, NULL })
 
 DEFENCODING(( "MacUkraine",             /* JDK 1.1 */
-            /*"CP10017",                   Windows */
+              "CP10017",                /* Windows */
             ),
             mac_ukraine,
             { mac_ukraine_mbtowc, NULL }, { mac_ukraine_wctomb, NULL })
 
 DEFENCODING(( "MacGreek",               /* JDK 1.1 */
-            /*"CP10006",                   Windows */
+              "CP10006",                /* Windows */
             ),
             mac_greek,
             { mac_greek_mbtowc, NULL },   { mac_greek_wctomb, NULL })
 
 DEFENCODING(( "MacTurkish",             /* JDK 1.1 */
-            /*"CP10081",                   Windows */
+              "CP10081",                /* Windows */
             ),
             mac_turkish,
             { mac_turkish_mbtowc, NULL }, { mac_turkish_wctomb, NULL })
 
 DEFENCODING(( "MacHebrew",              /* JDK 1.1 */
-            /*"CP10005",                   Windows */
+              "CP10005",                /* Windows */
             ),
             mac_hebrew,
             { mac_hebrew_mbtowc, NULL },  { mac_hebrew_wctomb, NULL })
 
 DEFENCODING(( "MacArabic",              /* JDK 1.1 */
-            /*"CP10004",                   Windows */
+              "CP10004",                /* Windows */
             ),
             mac_arabic,
             { mac_arabic_mbtowc, NULL },  { mac_arabic_wctomb, NULL })
 
 DEFENCODING(( "MacThai",                /* JDK 1.1 */
-            /*"CP10021",                   Windows */
+              "CP10021",                /* Windows */
             ),
             mac_thai,
             { mac_thai_mbtowc, NULL },    { mac_thai_wctomb, NULL })
@@ -739,8 +748,8 @@ DEFENCODING(( "JIS_X0201",              /* IANA */
               "JISX0201-1976",
               "X0201",                  /* IANA */
               "csHalfWidthKatakana",    /* IANA */
-            /*"JISX0201.1976-0",           X11R6.4 */
-            /*"JIS0201",                   JDK 1.1 */
+              "JISX0201.1976-0",        /* X11R6.4 */
+              "JIS0201",                /* JDK 1.1 */
             ),
             jisx0201,
             { jisx0201_mbtowc, NULL },    { jisx0201_wctomb, NULL })
@@ -748,14 +757,13 @@ DEFENCODING(( "JIS_X0201",              /* IANA */
 DEFENCODING(( "JIS_X0208",
               "JIS_X0208-1983",         /* IANA */
               "JIS_X0208-1990",
-              "JIS0208",
               "X0208",                  /* IANA */
               "ISO-IR-87",              /* IANA */
               "JIS_C6226-1983",         /* IANA */
               "csISO87JISX0208",        /* IANA */
-            /*"JISX0208.1983-0",           X11R6.4 */
-            /*"JISX0208.1990-0",           X11R6.4 */
-            /*"JIS0208",                   JDK 1.1 */
+              "JISX0208.1983-0",        /* X11R6.4 */
+              "JISX0208.1990-0",        /* X11R6.4 */
+              "JIS0208",                /* JDK 1.1 */
             ),
             jisx0208,
             { jisx0208_mbtowc, NULL },    { jisx0208_wctomb, NULL })
@@ -766,8 +774,8 @@ DEFENCODING(( "JIS_X0212",
               "X0212",                  /* IANA */
               "ISO-IR-159",             /* IANA */
               "csISO159JISX02121990",   /* IANA */
-            /*"JISX0212.1990-0",           X11R6.4 */
-            /*"JIS0212",                   JDK 1.1 */
+              "JISX0212.1990-0",        /* X11R6.4 */
+              "JIS0212",                /* JDK 1.1 */
             ),
             jisx0212,
             { jisx0212_mbtowc, NULL },    { jisx0212_wctomb, NULL })
@@ -785,7 +793,7 @@ DEFENCODING(( "GB_2312-80",             /* IANA */
               "ISO-IR-58",              /* IANA */
               "csISO58GB231280",        /* IANA */
               "CHINESE",                /* IANA */
-            /*"GB2312.1980-0",             X11R6.4 */
+              "GB2312.1980-0",          /* X11R6.4 */
             ),
             gb2312,
             { gb2312_mbtowc, NULL },      { gb2312_wctomb, NULL })
@@ -802,8 +810,8 @@ DEFENCODING(( "KSC_5601",               /* IANA */
               "ISO-IR-149",             /* IANA */
               "csKSC56011987",          /* IANA */
               "KOREAN",                 /* IANA */
-            /*"KSC5601.1987-0",            X11R6.4 */
-            /*"KSX1001:1992",              Ken Lunde */
+              "KSC5601.1987-0",         /* X11R6.4 */
+              "KSX1001:1992",           /* Ken Lunde */
             ),
             ksc5601,
             { ksc5601_mbtowc, NULL },     { ksc5601_wctomb, NULL })
@@ -814,8 +822,8 @@ DEFENCODING(( "EUC-JP",                 /* IANA */
               "EUCJP",                  /* glibc, HP-UX, IRIX, OSF/1, Solaris */
               "Extended_UNIX_Code_Packed_Format_for_Japanese", /* IANA */
               "csEUCPkdFmtJapanese",    /* IANA */
-            /*"EUC_JP",                    JDK 1.1 */
-            /*"CP51932",                   Windows */
+              "EUC_JP",                 /* JDK 1.1 */
+              "CP51932",                /* Windows */
             ),
             euc_jp,
             { euc_jp_mbtowc, NULL },      { euc_jp_wctomb, NULL })
@@ -842,6 +850,7 @@ DEFALIAS(     "PCK",                    /* Solaris */
 #endif
 
 DEFENCODING(( "CP932",                  /* glibc */
+              "WINDOWS-31J",            /* IANA */
             ),
             cp932,
             { cp932_mbtowc, NULL },       { cp932_wctomb, NULL })
@@ -852,7 +861,7 @@ DEFALIAS(     "IBM-932",                /* AIX */
 
 DEFENCODING(( "ISO-2022-JP",            /* IANA, RFC 1468 */
               "csISO2022JP",            /* IANA */
-            /*"ISO2022JP",                 JDK 1.1 */
+              "ISO2022JP",              /* JDK 1.1 */
             ),
             iso2022_jp,
             { iso2022_jp_mbtowc, NULL },  { iso2022_jp_wctomb, iso2022_jp_reset })
@@ -870,7 +879,7 @@ DEFENCODING(( "ISO-2022-JP-2",          /* IANA, RFC 1554 */
 
 DEFENCODING(( "ISO-2022-JP-MS",
               "CP50221",
-            /*"ISO-2022-JP-ESC",           Windows */
+              "ISO-2022-JP-ESC",        /* Windows */
             ),
             iso2022_jpms,
             { iso2022_jpms_mbtowc, NULL }, { iso2022_jpms_wctomb, iso2022_jpms_reset })
@@ -880,8 +889,8 @@ DEFENCODING(( "EUC-CN",                 /* glibc */
               "GB2312",                 /* IANA */
               "CN-GB",                  /* RFC 1922 */
               "csGB2312",               /* IANA */
-            /*"EUC_CN",                    JDK 1.1 */
-            /*"CP51936",                   Windows */
+              "EUC_CN",                 /* JDK 1.1 */
+              "CP51936",                /* Windows */
             ),
             euc_cn,
             { euc_cn_mbtowc, NULL },      { euc_cn_wctomb, NULL })
@@ -911,14 +920,14 @@ DEFENCODING(( "CP936",                  /* IANA */
             { cp936_mbtowc, NULL },       { cp936_wctomb, NULL })
 
 DEFENCODING(( "GB18030",                /* IANA, glibc */
-            /*"CP54936",                   Windows */
+              "CP54936",                /* Windows */
             ),
             gb18030,
             { gb18030_mbtowc, NULL },     { gb18030_wctomb, NULL })
 
 DEFENCODING(( "ISO-2022-CN",            /* IANA, RFC 1922 */
               "csISO2022CN",
-            /*"ISO2022CN",                 JDK 1.1 */
+              "ISO2022CN",              /* JDK 1.1 */
             ),
             iso2022_cn,
             { iso2022_cn_mbtowc, NULL },  { iso2022_cn_wctomb, iso2022_cn_reset })
@@ -930,7 +939,7 @@ DEFENCODING(( "ISO-2022-CN-EXT",        /* IANA, RFC 1922 */
 
 DEFENCODING(( "HZ",                     /* RFC 1843 */
               "HZ-GB-2312",             /* IANA, RFC 1842 */
-            /*"CP52936",                   Windows */
+              "CP52936",                /* Windows */
             ),
             hz,
             { hz_mbtowc, NULL },          { hz_wctomb, hz_reset })
@@ -938,8 +947,8 @@ DEFENCODING(( "HZ",                     /* RFC 1843 */
 DEFENCODING(( "EUC-TW",                 /* glibc */
               "EUCTW",                  /* glibc, HP-UX, IRIX, OSF/1 */
               "csEUCTW",
-            /*"EUC_TW",                    JDK 1.1 */
-            /*"CP51950",                   Windows */
+              "EUC_TW",                 /* JDK 1.1 */
+              "CP51950",                /* Windows */
             ),
             euc_tw,
             { euc_tw_mbtowc, NULL },      { euc_tw_wctomb, NULL })
@@ -992,8 +1001,8 @@ DEFENCODING(( "BIG5-HKSCS",             /* IANA */
 DEFENCODING(( "EUC-KR",                 /* IANA, RFC 1557 */
               "EUCKR",                  /* glibc, HP-UX, IRIX, OSF/1 */
               "csEUCKR",                /* IANA */
-            /*"EUC_KR",                    JDK 1.1 */
-            /*"CP51949",                   Windows */
+              "EUC_KR",                 /* JDK 1.1 */
+              "CP51949",                /* Windows */
             ),
             euc_kr,
             { euc_kr_mbtowc, NULL },      { euc_kr_wctomb, NULL })
@@ -1032,8 +1041,8 @@ DEFALIAS(     "KO_KR.JOHAP92",          /* Solaris */
 
 DEFENCODING(( "ISO-2022-KR",            /* IANA, RFC 1557 */
               "csISO2022KR",            /* IANA */
-            /*"ISO2022KR",                 JDK 1.1 */
-            /*"CP50225",                   Windows */
+              "ISO2022KR",              /* JDK 1.1 */
+              "CP50225",                /* Windows */
             ),
             iso2022_kr,
             { iso2022_kr_mbtowc, NULL },  { iso2022_kr_wctomb, iso2022_kr_reset })
