diff --git a/mesonbuild/build.py b/mesonbuild/build.py
index 071dbdfd4..85249ebbf 100644
--- a/mesonbuild/build.py
+++ b/mesonbuild/build.py
@@ -2289,7 +2289,7 @@ class SharedLibrary(BuildTarget):
             suffix = 'dll'
             # Shared library is of the form cygfoo.dll
             # (ld --dll-search-prefix=cyg is the default)
-            prefix = 'cyg'
+            prefix = 'msys-'
             # Import library is called libfoo.dll.a
             import_prefix = self.prefix if self.prefix is not None else 'lib'
             import_filename_tpl = import_prefix + '{0.name}.dll.a'
diff --git a/mesonbuild/compilers/mixins/clike.py b/mesonbuild/compilers/mixins/clike.py
index 70e81a4b8..ecabdf8f4 100644
--- a/mesonbuild/compilers/mixins/clike.py
+++ b/mesonbuild/compilers/mixins/clike.py
@@ -1032,7 +1032,7 @@ class CLikeCompiler(Compiler):
             stlibext += ['lib']
         elif env.machines[self.for_machine].is_cygwin():
             shlibext = ['dll', 'dll.a']
-            prefixes = ['cyg'] + prefixes
+            prefixes = ['msys-', 'cyg'] + prefixes
         elif self.id.lower() == 'c6000' or self.id.lower() == 'ti':
             # TI C6000 compiler can use both extensions for static or dynamic libs.
             stlibext = ['a', 'lib']
