diff --git src/util.c src/util.c
index 476b236..94b3c70 100644
--- src/util.c
+++ src/util.c
@@ -14,6 +14,19 @@
 #define getc_unlocked(x) getc(x)
 #endif
 
+#ifdef __MSYS__
+#include <windows.h>
+#include <psapi.h>
+#include <shellapi.h>
+#include <gdiplus.h>
+#include <process.h>
+#include <winuser.h>
+//#include <conio.h>
+#include <stdint.h>
+#include "w32api/minwindef.h"
+#define _MAX_PATH MAX_PATH
+#endif /* __MSYS__ */
+
 #define CHECK_AND_RETURN(ptr)             \
     if (ptr == NULL) {                    \
         die("Memory allocation failed."); \
