--- elfutils-0.173/libelf/Makefile.am.orig	2018-07-30 10:25:51.581470000 +0300
+++ elfutils-0.173/libelf/Makefile.am	2018-07-30 10:26:08.741500100 +0300
@@ -96,7 +96,7 @@
 am_libelf_pic_a_OBJECTS = $(libelf_a_SOURCES:.c=.os)
 
 libelf_so_DEPS = ../lib/libeu.a
-libelf_so_LDLIBS = $(libelf_so_DEPS) -lz
+libelf_so_LDLIBS = $(libelf_so_DEPS) -lz -lintl
 if USE_LOCKS
 libelf_so_LDLIBS += -lpthread
 endif
--- elfutils-0.173/libdw/Makefile.am.orig	2018-07-30 10:53:44.228407500 +0300
+++ elfutils-0.173/libdw/Makefile.am	2018-07-30 10:53:49.173616200 +0300
@@ -108,7 +108,7 @@
 libdw_so_LIBS = libdw_pic.a ../libdwelf/libdwelf_pic.a \
 	  ../libdwfl/libdwfl_pic.a ../libebl/libebl.a
 libdw_so_DEPS = ../lib/libeu.a ../libelf/libelf.so
-libdw_so_LDLIBS = $(libdw_so_DEPS) -ldl -lz $(argp_LDADD) $(zip_LIBS)
+libdw_so_LDLIBS = $(libdw_so_DEPS) -ldl -lz -lintl $(argp_LDADD) $(zip_LIBS)
 libdw_so_SOURCES =
 libdw.so$(EXEEXT): $(srcdir)/libdw.map $(libdw_so_LIBS) $(libdw_so_DEPS)
 # The rpath is necessary for libebl because its $ORIGIN use will
--- elfutils-0.173/libasm/Makefile.am.orig	2018-07-30 11:02:16.191106800 +0300
+++ elfutils-0.173/libasm/Makefile.am	2018-07-30 11:02:37.048343400 +0300
@@ -56,7 +56,7 @@
 am_libasm_pic_a_OBJECTS = $(libasm_a_SOURCES:.c=.os)
 
 libasm_so_DEPS = ../lib/libeu.a ../libebl/libebl.a ../libelf/libelf.so ../libdw/libdw.so
-libasm_so_LDLIBS = $(libasm_so_DEPS)
+libasm_so_LDLIBS = $(libasm_so_DEPS) -lintl
 if USE_LOCKS
 libasm_so_LDLIBS += -lpthread
 endif
