--- elfutils-0.173/libdw/Makefile.am.orig	2018-07-30 10:45:42.116360700 +0300
+++ elfutils-0.173/libdw/Makefile.am	2018-07-30 10:46:09.978009700 +0300
@@ -115,7 +115,6 @@
 # not fly in a setuid executable that links in libdw.
 	$(AM_V_CCLD)$(LINK) $(dso_LDFLAGS) -o $@ \
 		-Wl,--soname,$@.$(VERSION) \
-		-Wl,--enable-new-dtags,-rpath,$(pkglibdir) \
 		-Wl,--version-script,$<,--no-undefined \
 		-Wl,--whole-archive $(libdw_so_LIBS) -Wl,--no-whole-archive \
 		$(libdw_so_LDLIBS)
