--- origsrc/automake-1.7.9/lib/am/libtool.am	2003-06-04 15:23:26.000000000 -0400
+++ src/automake-1.7.9/lib/am/libtool.am	2008-10-25 23:02:40.663657600 -0400
@@ -28,4 +28,4 @@
 
 ?TOPDIR?distclean-am: distclean-libtool
 ?TOPDIR?distclean-libtool:
-?TOPDIR?	-rm -f libtool
+?TOPDIR?	-rm -f libtool config.lt
--- origsrc/automake-1.7.9/tests/libtool5.test	2003-09-07 15:58:39.000000000 -0400
+++ src/automake-1.7.9/tests/libtool5.test	2008-10-25 18:39:40.252540800 -0400
@@ -46,4 +46,4 @@
 
 libtoolize
 $ACLOCAL
-$AUTOMAKE
+$AUTOMAKE --add-missing
--- origsrc/automake-1.7.9/tests/ltcond.test	2003-08-06 19:34:41.000000000 -0400
+++ src/automake-1.7.9/tests/ltcond.test	2008-10-25 18:39:40.262555200 -0400
@@ -64,7 +64,7 @@
 libtoolize
 $ACLOCAL
 $AUTOCONF
-$AUTOMAKE
+$AUTOMAKE --add-missing
 
 # Install libraries in lib/, and the rest in empty/.
 # (in fact there is no "rest", so as the name imply empty/ is
--- origsrc/automake-1.7.9/tests/ltcond2.test	2003-07-30 17:38:38.000000000 -0400
+++ src/automake-1.7.9/tests/ltcond2.test	2008-10-25 18:39:40.302612800 -0400
@@ -75,7 +75,7 @@
 libtoolize
 $ACLOCAL
 $AUTOCONF
-$AUTOMAKE
+$AUTOMAKE --add-missing
 ./configure
 $MAKE check
 test -f check-ok
--- origsrc/automake-1.7.9/tests/ltconv.test	2003-08-05 15:42:40.000000000 -0400
+++ src/automake-1.7.9/tests/ltconv.test	2008-10-25 18:39:40.302612800 -0400
@@ -113,7 +113,7 @@
 libtoolize
 $ACLOCAL
 $AUTOCONF
-$AUTOMAKE
+$AUTOMAKE --add-missing
 
 # Install libraries in lib/, programs in bin/, and the rest in empty/.
 # (in fact there is no "rest", so as the name imply empty/ is
--- origsrc/automake-1.7.9/tests/subst.test	2003-06-04 15:23:28.000000000 -0400
+++ src/automake-1.7.9/tests/subst.test	2008-10-25 22:53:11.645449600 -0400
@@ -24,8 +24,14 @@
 . ./defs || exit 1
 
 cat >> configure.in << 'END'
-AC_SUBST($1)
-AC_SUBST([$]$1)  dnl this is the actual invocation that was used
+dnl This test used to have the following lines, which cannot have
+dnl worked sensibly with Autoconf for years, however:
+dnl AC_SUBST($1)
+dnl AC_SUBST([$]$1)  dnl this is the actual invocation that was used
+dnl
+AC_DEFUN([FOO],
+[AC_SUBST([$1])])
+FOO([BAR])
 END
 
 : > Makefile.am
