--- p11-kit-0.25.1/p11-kit/meson.build.orig	2023-10-27 18:02:38.287295300 +0200
+++ p11-kit-0.25.1/p11-kit/meson.build	2023-10-27 18:02:48.975648600 +0200
@@ -461,7 +461,7 @@
                   name_suffix: module_suffix,
                   link_args: p11_module_ldflags,
                   link_depends: [p11_module_symbol_map],
-                  dependencies: [libp11_test_dep] + libp11_asn1_deps)
+                  dependencies: [libp11_test_dep] + libp11_asn1_deps + libintl_deps)
   endforeach
 endif
 
