Workaround the exception c00000005 in verifyable_object_isvalid
on Win8 x64.

--- p11-kit-0.18.7-orig/trust/module.c	2013-10-14 07:15:57.000000000 -0500
+++ p11-kit-0.18.7/trust/module.c	2014-01-23 21:48:07.621401600 -0600
@@ -1798,7 +1798,7 @@ void p11_trust_module_init (void);
 void p11_trust_module_fini (void);
 
 #ifdef __GNUC__
-__attribute__((constructor))
+__attribute__((constructor, optimize(1)))
 #endif
 void
 p11_trust_module_init (void)
