--- p11-kit-0.23.5-orig/common/compat.c	2017-02-16 10:01:07.000000000 -0600
+++ p11-kit-0.23.5/common/compat.c	2017-04-27 22:33:18.917949600 -0500
@@ -40,8 +40,12 @@
  */
 #define _XOPEN_SOURCE 700
 
-#if defined(HAVE_ISSETUGID) && defined(__FreeBSD__)
+#if defined(HAVE_ISSETUGID)
+#if defined(__FreeBSD__)
 #define __BSD_VISIBLE 1
+#elif defined(__CYGWIN__)
+#define _BSD_SOURCE
+#endif
 #endif
 
 #include "compat.h"
