--- p11-kit-0.18.7-orig/p11-kit/Makefile.am	2013-10-14 16:15:57.000000000 +0400
+++ p11-kit-0.18.7/p11-kit/Makefile.am	2013-10-21 15:14:43.475600000 +0400
@@ -103,5 +103,4 @@
 
 # Proxy module is actually same as library, so install a link
 install-exec-hook:
-	$(LN_S) -f `readlink $(DESTDIR)$(libdir)/libp11-kit.so` $(DESTDIR)$(libdir)/p11-kit-proxy.so
 	$(MKDIR_P) $(DESTDIR)$(p11_package_config_modules)
