--- coreutils-8.32/configure.ac
+++ coreutils-8.32/configure.ac
@@ -509,7 +509,7 @@
   ],
   [stdbuf_supported=yes])
 AC_MSG_RESULT([$stdbuf_supported])
-if test "$stdbuf_supported" = "yes" && test -z "$EXEEXT"; then
+if test "$stdbuf_supported" = "yes"; then
   gl_ADD_PROG([optional_bin_progs], [stdbuf])
 fi
 CFLAGS=$ac_save_CFLAGS
@@ -530,7 +530,7 @@
 # Note it adding to pkglibexec_PROGRAMS, $(transform) in src/local.mk
 # may need to be updated accordingly.
 case " $optional_bin_progs " in
-  *' stdbuf '*) pkglibexec_PROGRAMS='src/libstdbuf.so';;
+  *' stdbuf '*) pkglibexec_PROGRAMS='src/libstdbuf.so$(EXEEXT)';;
   *) pkglibexec_PROGRAMS='';;
 esac
 
--- coreutils-8.32/src/stdbuf.c
+++ coreutils-8.32/src/stdbuf.c
@@ -33,7 +33,7 @@
 
 /* The official name of this program (e.g., no 'g' prefix).  */
 #define PROGRAM_NAME "stdbuf"
-#define LIB_NAME "libstdbuf.so" /* FIXME: don't hardcode  */
+#define LIB_NAME "libstdbuf.dll" /* FIXME: don't hardcode  */
 
 #define AUTHORS proper_name ("Padraig Brady")
 
