--- coreutils-9.0/configure.ac.orig
+++ coreutils-9.0/configure.ac
@@ -644,7 +644,7 @@
 # Note it adding to pkglibexec_PROGRAMS, $(transform) in src/local.mk
 # may need to be updated accordingly.
 case " $optional_bin_progs " in
-  *' stdbuf '*) pkglibexec_PROGRAMS='src/libstdbuf.so';;
+  *' stdbuf '*) pkglibexec_PROGRAMS='src/libstdbuf.so$(EXEEXT)';;
   *) pkglibexec_PROGRAMS='';;
 esac
 
--- coreutils-9.0/src/stdbuf.c.orig
+++ coreutils-9.0/src/stdbuf.c
@@ -33,7 +33,7 @@
 
 /* The official name of this program (e.g., no 'g' prefix).  */
 #define PROGRAM_NAME "stdbuf"
-#define LIB_NAME "libstdbuf.so" /* FIXME: don't hardcode  */
+#define LIB_NAME "libstdbuf.dll" /* FIXME: don't hardcode  */
 
 #define AUTHORS proper_name ("Padraig Brady")
 
