diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2014-07-19 02:22:24.000000000 +0400
+++ b/Makefile.in	2014-07-20 20:19:43.363400000 +0400
@@ -14076,8 +14076,8 @@
 	  && $(run_help2man)						\
 		     --source='$(PACKAGE_STRING)'			\
 		     --include=$(srcdir)/man/$$name.x			\
-		     --output=$$t/$$name.1 $$t/$$argv			\
 		     --info-page='coreutils \(aq'$$name' invocation\(aq' \
+		     --output=$$t/$$name.1 $$t/$$argv			\
 	  && sed \
 	       -e 's|$*\.td/||g' \
 	       -e '/For complete documentation/d' \
