diff -Naur a/configure b/configure
--- a/configure	2014-07-19 02:07:39.000000000 +0400
+++ b/configure	2014-07-20 20:19:45.422600000 +0400
@@ -63373,7 +63373,7 @@
 # Now that we know which programs will actually be built, determine
 # which optional helper progs should be compiled.
 case " $optional_bin_progs " in
-  *' stdbuf '*) pkglibexec_PROGRAMS='src/libstdbuf.so';;
+  *' stdbuf '*) pkglibexec_PROGRAMS='src/libstdbuf.so$(EXEEXT)';;
   *) pkglibexec_PROGRAMS='';;
 esac
 
diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2014-07-20 20:19:44.237000000 +0400
+++ b/Makefile.in	2014-07-20 20:19:45.438200000 +0400
@@ -4595,7 +4595,7 @@
 
 # Ensure we don't link against libcoreutils.a as that lib is
 # not compiled with -fPIC which causes issues on 64 bit at least
-src_libstdbuf_so_LDADD = $(LIBINTL)
+src_libstdbuf_so_LDADD = $(LIBINTL) $(LDADD)
 
 # Note libstdbuf is only compiled if GCC is available
 # (as per the check in configure.ac), so these flags should be available.
