diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2014-07-20 20:19:46.499000000 +0400
+++ b/Makefile.in	2014-07-20 20:19:47.591000000 +0400
@@ -3767,6 +3767,7 @@
 WERROR_CFLAGS = @WERROR_CFLAGS@
 WINDOWS_64_BIT_OFF_T = @WINDOWS_64_BIT_OFF_T@
 WINDOWS_64_BIT_ST_SIZE = @WINDOWS_64_BIT_ST_SIZE@
+WINDRES = windres
 WINT_T_SUFFIX = @WINT_T_SUFFIX@
 XGETTEXT = @XGETTEXT@
 XGETTEXT_015 = @XGETTEXT_015@
@@ -7666,9 +7667,9 @@
 src/src_ginstall-selinux.$(OBJEXT): src/$(am__dirstamp) \
 	src/$(DEPDIR)/$(am__dirstamp)
 
-src/ginstall$(EXEEXT): $(src_ginstall_OBJECTS) $(src_ginstall_DEPENDENCIES) $(EXTRA_src_ginstall_DEPENDENCIES) src/$(am__dirstamp)
+src/ginstall$(EXEEXT): $(src_ginstall_OBJECTS) src/src_ginstall-manifest.res $(src_ginstall_DEPENDENCIES) $(EXTRA_src_ginstall_DEPENDENCIES) src/$(am__dirstamp)
 	@rm -f src/ginstall$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(src_ginstall_OBJECTS) $(src_ginstall_LDADD) $(LIBS)
+	$(AM_V_CCLD)$(LINK) $(src_ginstall_OBJECTS) src/src_ginstall-manifest.res $(src_ginstall_LDADD) $(LIBS)
 src/groups.$(OBJEXT): src/$(am__dirstamp) \
 	src/$(DEPDIR)/$(am__dirstamp)
 src/group-list.$(OBJEXT): src/$(am__dirstamp) \
@@ -10699,6 +10700,9 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(AM_V_CC@am__nodep@)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(src_coreutils_CFLAGS) $(CFLAGS) -c -o src/src_coreutils-coreutils.obj `if test -f 'src/coreutils.c'; then $(CYGPATH_W) 'src/coreutils.c'; else $(CYGPATH_W) '$(srcdir)/src/coreutils.c'; fi`
 
+src/src_ginstall-manifest.res: src/install.rc src/install.manifest
+	$(WINDRES) --input $< --output $@ --output-format=coff
+
 src/src_ginstall-install.o: src/install.c
 @am__fastdepCC_TRUE@	$(AM_V_CC)$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(src_ginstall_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT src/src_ginstall-install.o -MD -MP -MF src/$(DEPDIR)/src_ginstall-install.Tpo -c -o src/src_ginstall-install.o `test -f 'src/install.c' || echo '$(srcdir)/'`src/install.c
 @am__fastdepCC_TRUE@	$(AM_V_at)$(am__mv) src/$(DEPDIR)/src_ginstall-install.Tpo src/$(DEPDIR)/src_ginstall-install.Po
diff -Naur a/src/install.manifest b/src/install.manifest
--- a/src/install.manifest	1970-01-01 03:00:00.000000000 +0300
+++ b/src/install.manifest	2014-07-20 20:19:47.591000000 +0400
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+  <assemblyIdentity version="1.0.0.0"
+     processorArchitecture="x86"
+     name="GNU.coreutils.install"
+     type="win32"/>
+
+  <!-- Identify the application security requirements. -->
+  <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
+    <security>
+      <requestedPrivileges>
+        <requestedExecutionLevel level="asInvoker" uiAccess="false"/>
+      </requestedPrivileges>
+    </security>
+  </trustInfo>
+</assembly>
diff -Naur a/src/install.rc b/src/install.rc
--- a/src/install.rc	1970-01-01 03:00:00.000000000 +0300
+++ b/src/install.rc	2014-07-20 20:19:47.606600000 +0400
@@ -0,0 +1 @@
+1 24 install.manifest
