diff -Naur a/lib/gnulib.mk b/lib/gnulib.mk
--- a/lib/gnulib.mk	2014-07-19 02:01:51.000000000 +0400
+++ b/lib/gnulib.mk	2014-07-20 20:19:49.743800000 +0400
@@ -483,6 +483,15 @@
 
 ## end   gnulib module cycle-check
 
+## begin gnulib module cygwin
+
+
+EXTRA_DIST += lib/cygwin.c
+
+EXTRA_lib_libcoreutils_a_SOURCES += lib/cygwin.c
+
+## end   gnulib module cygwin
+
 ## begin gnulib module dev-ino
 
 
