diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2014-07-20 20:19:48.636200000 +0400
+++ b/Makefile.in	2014-07-20 20:19:51.771800000 +0400
@@ -627,7 +627,7 @@
 	lib/argv-iter.c lib/argv-iter.h lib/backupfile.c lib/base32.h \
 	lib/base32.c lib/base64.h lib/base64.c lib/bitrotate.h \
 	lib/bitrotate.c lib/buffer-lcm.c lib/buffer-lcm.h \
-	lib/c-ctype.h lib/c-ctype.c lib/c-strcase.h lib/c-strcasecmp.c \
+	lib/cygwin.c lib/cygwin.h lib/c-ctype.h lib/c-ctype.c lib/c-strcase.h lib/c-strcasecmp.c \
 	lib/c-strncasecmp.c lib/c-strtod.c lib/c-strtold.c \
 	lib/canon-host.c lib/canonicalize.c lib/careadlinkat.c \
 	lib/cloexec.c lib/close-stream.c lib/closein.c lib/closeout.c \
@@ -709,7 +709,7 @@
 	lib/argv-iter.$(OBJEXT) lib/backupfile.$(OBJEXT) \
 	lib/base32.$(OBJEXT) lib/base64.$(OBJEXT) \
 	lib/bitrotate.$(OBJEXT) lib/buffer-lcm.$(OBJEXT) \
-	lib/c-ctype.$(OBJEXT) lib/c-strcasecmp.$(OBJEXT) \
+	lib/cygwin.$(OBJEXT) lib/c-ctype.$(OBJEXT) lib/c-strcasecmp.$(OBJEXT) \
 	lib/c-strncasecmp.$(OBJEXT) lib/c-strtod.$(OBJEXT) \
 	lib/c-strtold.$(OBJEXT) lib/canon-host.$(OBJEXT) \
 	lib/canonicalize.$(OBJEXT) lib/careadlinkat.$(OBJEXT) \
@@ -4223,7 +4223,7 @@
 	lib/argv-iter.c lib/argv-iter.h lib/backupfile.c lib/base32.h \
 	lib/base32.c lib/base64.h lib/base64.c lib/bitrotate.h \
 	lib/bitrotate.c lib/buffer-lcm.c lib/buffer-lcm.h \
-	lib/c-ctype.h lib/c-ctype.c lib/c-strcase.h lib/c-strcasecmp.c \
+	lib/cygwin.c lib/cygwin.h lib/c-ctype.h lib/c-ctype.c lib/c-strcase.h lib/c-strcasecmp.c \
 	lib/c-strncasecmp.c lib/c-strtod.c lib/c-strtold.c \
 	lib/canon-host.c lib/canonicalize.c lib/careadlinkat.c \
 	lib/cloexec.c lib/close-stream.c lib/closein.c lib/closeout.c \
@@ -5849,6 +5849,8 @@
 	lib/$(DEPDIR)/$(am__dirstamp)
 lib/buffer-lcm.$(OBJEXT): lib/$(am__dirstamp) \
 	lib/$(DEPDIR)/$(am__dirstamp)
+lib/cygwin.$(OBJEXT): lib/$(am__dirstamp) \
+	lib/$(DEPDIR)/$(am__dirstamp)
 lib/c-ctype.$(OBJEXT): lib/$(am__dirstamp) \
 	lib/$(DEPDIR)/$(am__dirstamp)
 lib/c-strcasecmp.$(OBJEXT): lib/$(am__dirstamp) \
