diff -Naur coreutils-8.21-cygwin/build-aux/compile coreutils-8.21-msys/build-aux/compile
--- coreutils-8.21-cygwin/build-aux/compile	2013-02-14 14:58:02.000000000 +0000
+++ coreutils-8.21-msys/build-aux/compile	2013-05-17 10:27:58.446289100 +0000
@@ -54,7 +54,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -68,7 +68,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur coreutils-8.21-cygwin/build-aux/config.guess coreutils-8.21-msys/build-aux/config.guess
--- coreutils-8.21-cygwin/build-aux/config.guess	2013-02-07 16:44:47.000000000 +0000
+++ coreutils-8.21-msys/build-aux/config.guess	2013-05-17 10:31:55.508789100 +0000
@@ -845,6 +845,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur coreutils-8.21-cygwin/build-aux/config.rpath coreutils-8.21-msys/build-aux/config.rpath
--- coreutils-8.21-cygwin/build-aux/config.rpath	2013-01-02 12:34:45.000000000 +0000
+++ coreutils-8.21-msys/build-aux/config.rpath	2013-05-17 10:27:58.446289100 +0000
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -537,7 +537,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur coreutils-8.21-cygwin/configure coreutils-8.21-msys/configure
--- coreutils-8.21-cygwin/configure	2013-02-14 15:54:18.000000000 +0000
+++ coreutils-8.21-msys/configure	2013-05-17 10:33:57.739257800 +0000
@@ -7171,7 +7171,7 @@
      else
        case "$host_os" in
                                                                osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
@@ -9051,7 +9051,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -13066,7 +13066,7 @@
                      # Guess yes on glibc systems.
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
-            cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+            cygwin* | msys*) gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
@@ -23531,7 +23531,7 @@
            openbsd*)        gl_cv_func_printf_directive_ls="guessing no";;
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
-           cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           cygwin* | msys*)         gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -23615,7 +23615,7 @@
       if test "$cross_compiling" = yes; then :
 
          case "$host_os" in
-           cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           cygwin* | msys*)      gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -25145,7 +25145,7 @@
 
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
@@ -26321,7 +26321,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
@@ -26411,7 +26411,7 @@
   case "$host_os" in
            mingw*) # on this platform, dup2 always returns 0 for success
              gl_cv_func_dup2_works="guessing no" ;;
-           cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+           cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
            linux*) # On linux between 2008-07-27 and 2009-05-11, dup2 of a
                    # closed fd may yield -EBADF instead of -1 / errno=EBADF.
@@ -36608,7 +36608,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
@@ -43008,7 +43008,7 @@
   case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
@@ -57682,7 +57682,7 @@
 
   case "$host_os" in
             mingw*) REPLACE_SETLOCALE=1 ;;
-            cygwin*)
+            cygwin* | msys*)
       case `uname -r` in
         1.5.*) REPLACE_SETLOCALE=1 ;;
       esac
@@ -58733,7 +58733,7 @@
   gnu[0-9]* | \
   linux-* | linux | \
   freebsd2.2* | freebsd[3-9]* | freebsd[1-9][0-9]* | \
-  cygwin | \
+  cygwin | msys | \
   mingw*)
 
 $as_echo "#define UNLINK_CANNOT_UNLINK_DIR 1" >>confdefs.h
@@ -59396,13 +59396,13 @@
 
                         case "$host_os" in
             # Guess no on Solaris <= 9 and Cygwin.
-          solaris2.[1-9] | solaris2.[1-9].* | cygwin*)
+          solaris2.[1-9] | solaris2.[1-9].* | cygwin* | msys*)
             gl_cv_func_wctob_works="guessing no" ;;
             # Guess yes otherwise.
           *) gl_cv_func_wctob_works="guessing yes" ;;
         esac
         case "$host_os" in
-          cygwin*)
+          cygwin* | msys*)
             if test "$cross_compiling" = yes; then :
   :
 else
diff -Naur coreutils-8.21-cygwin/gnulib-tests/test-copy-acl.sh coreutils-8.21-msys/gnulib-tests/test-copy-acl.sh
--- coreutils-8.21-cygwin/gnulib-tests/test-copy-acl.sh	2012-07-12 13:49:26.000000000 +0000
+++ coreutils-8.21-msys/gnulib-tests/test-copy-acl.sh	2013-05-17 10:27:58.524414100 +0000
@@ -128,7 +128,7 @@
   # the programs.
   # func_test_same_acls file1 file2
   case $acl_flavor in
-    linux | cygwin | freebsd | solaris)
+    linux | cygwin | msys | freebsd | solaris)
       func_test_same_acls ()
       {
         getfacl "$1" | sed -e "s/$1/FILENAME/g" > tmpaclout1
@@ -302,7 +302,7 @@
 
         ;;
 
-      cygwin)
+      cygwin | msys)
 
         # Set an ACL for a group.
         setfacl -m group:0:1 tmpfile0
diff -Naur coreutils-8.21-cygwin/gnulib-tests/test-file-has-acl.sh coreutils-8.21-msys/gnulib-tests/test-file-has-acl.sh
--- coreutils-8.21-cygwin/gnulib-tests/test-file-has-acl.sh	2012-07-12 13:49:26.000000000 +0000
+++ coreutils-8.21-msys/gnulib-tests/test-file-has-acl.sh	2013-05-17 10:27:58.540039100 +0000
@@ -247,7 +247,7 @@
         fi
         ;;
 
-      cygwin)
+      cygwin | msys)
 
         # Set an ACL for a group.
         if setfacl -m group:0:1 tmpfile0; then
diff -Naur coreutils-8.21-cygwin/gnulib-tests/test-set-mode-acl.sh coreutils-8.21-msys/gnulib-tests/test-set-mode-acl.sh
--- coreutils-8.21-cygwin/gnulib-tests/test-set-mode-acl.sh	2012-07-12 13:49:26.000000000 +0000
+++ coreutils-8.21-msys/gnulib-tests/test-set-mode-acl.sh	2013-05-17 10:27:58.540039100 +0000
@@ -177,7 +177,7 @@
           linux | freebsd | solaris)
             setfacl -m user:$auid:1 tmpfile0
             ;;
-          cygwin)
+          cygwin | msys)
             setfacl -m group:0:1 tmpfile0
             ;;
           hpux)
diff -Naur coreutils-8.21-cygwin/install.exe.manifest coreutils-8.21-msys/install.exe.manifest
--- coreutils-8.21-cygwin/install.exe.manifest	1970-01-01 00:00:00.000000000 +0000
+++ coreutils-8.21-msys/install.exe.manifest	2013-05-17 10:27:58.540039100 +0000
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+  <v3:trustInfo xmlns:v3="urn:schemas-microsoft-com:asm.v3">
+    <v3:security>
+      <v3:requestedPrivileges>
+        <v3:requestedExecutionLevel level="asInvoker" />
+      </v3:requestedPrivileges>
+    </v3:security>
+  </v3:trustInfo>
+</assembly>
+
diff -Naur coreutils-8.21-cygwin/lib/gnulib.mk coreutils-8.21-msys/lib/gnulib.mk
--- coreutils-8.21-cygwin/lib/gnulib.mk	2013-02-14 14:57:23.000000000 +0000
+++ coreutils-8.21-msys/lib/gnulib.mk	2013-05-17 10:27:58.540039100 +0000
@@ -1887,7 +1887,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
diff -Naur coreutils-8.21-cygwin/m4/btowc.m4 coreutils-8.21-msys/m4/btowc.m4
--- coreutils-8.21-cygwin/m4/btowc.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/btowc.m4	2013-05-17 10:28:02.352539100 +0000
@@ -49,7 +49,7 @@
 changequote(,)dnl
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes otherwise.
              *)       gl_cv_func_btowc_nul="guessing yes" ;;
            esac
diff -Naur coreutils-8.21-cygwin/m4/double-slash-root.m4 coreutils-8.21-msys/m4/double-slash-root.m4
--- coreutils-8.21-cygwin/m4/double-slash-root.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/double-slash-root.m4	2013-05-17 10:28:02.368164100 +0000
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur coreutils-8.21-cygwin/m4/dup2.m4 coreutils-8.21-msys/m4/dup2.m4
--- coreutils-8.21-cygwin/m4/dup2.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/dup2.m4	2013-05-17 10:38:24.763671900 +0000
@@ -49,7 +49,7 @@
         [case "$host_os" in
            mingw*) # on this platform, dup2 always returns 0 for success
              gl_cv_func_dup2_works="guessing no" ;;
-           cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+           cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
            linux*) # On linux between 2008-07-27 and 2009-05-11, dup2 of a
                    # closed fd may yield -EBADF instead of -1 / errno=EBADF.
diff -Naur coreutils-8.21-cygwin/m4/getcwd.m4 coreutils-8.21-msys/m4/getcwd.m4
--- coreutils-8.21-cygwin/m4/getcwd.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/getcwd.m4	2013-05-17 10:38:59.138671900 +0000
@@ -49,7 +49,7 @@
                      # Guess yes on glibc systems.
             *-gnu*)  gl_cv_func_getcwd_null="guessing yes";;
                      # Guess yes on Cygwin.
-            cygwin*) gl_cv_func_getcwd_null="guessing yes";;
+            cygwin* | msys*) gl_cv_func_getcwd_null="guessing yes";;
                      # If we don't know, assume the worst.
             *)       gl_cv_func_getcwd_null="guessing no";;
           esac
diff -Naur coreutils-8.21-cygwin/m4/malloc.m4 coreutils-8.21-msys/m4/malloc.m4
--- coreutils-8.21-cygwin/m4/malloc.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/malloc.m4	2013-05-17 10:39:37.419921900 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur coreutils-8.21-cygwin/m4/printf.m4 coreutils-8.21-msys/m4/printf.m4
--- coreutils-8.21-cygwin/m4/printf.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/printf.m4	2013-05-17 10:28:02.383789100 +0000
@@ -735,7 +735,7 @@
            openbsd*)        gl_cv_func_printf_directive_ls="guessing no";;
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
-           cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           cygwin* | msys*)         gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
            *)               gl_cv_func_printf_directive_ls="guessing yes";;
          esac
@@ -812,7 +812,7 @@
         [
 changequote(,)dnl
          case "$host_os" in
-           cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           cygwin* | msys*)      gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
            *)            gl_cv_func_printf_flag_grouping="guessing yes";;
@@ -1464,7 +1464,7 @@
            darwin[1-6].*)        gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
-           cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           cygwin* | msys*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Naur coreutils-8.21-cygwin/m4/realloc.m4 coreutils-8.21-msys/m4/realloc.m4
--- coreutils-8.21-cygwin/m4/realloc.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/realloc.m4	2013-05-17 10:40:41.154296900 +0000
@@ -30,7 +30,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
diff -Naur coreutils-8.21-cygwin/m4/setlocale.m4 coreutils-8.21-msys/m4/setlocale.m4
--- coreutils-8.21-cygwin/m4/setlocale.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/setlocale.m4	2013-05-17 10:28:02.383789100 +0000
@@ -14,7 +14,7 @@
     mingw*) REPLACE_SETLOCALE=1 ;;
     dnl On Cygwin 1.5.x, setlocale always succeeds but setlocale(LC_CTYPE,NULL)
     dnl is then still "C".
-    cygwin*)
+    cygwin* | msys*)
       case `uname -r` in
         1.5.*) REPLACE_SETLOCALE=1 ;;
       esac
diff -Naur coreutils-8.21-cygwin/m4/threadlib.m4 coreutils-8.21-msys/m4/threadlib.m4
--- coreutils-8.21-cygwin/m4/threadlib.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/threadlib.m4	2013-05-17 10:28:02.383789100 +0000
@@ -68,7 +68,7 @@
          dnl bugs that lead to endless loops or crashes. See
          dnl <http://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
          osf*) gl_use_threads=no ;;
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
diff -Naur coreutils-8.21-cygwin/m4/unlinkdir.m4 coreutils-8.21-msys/m4/unlinkdir.m4
--- coreutils-8.21-cygwin/m4/unlinkdir.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/unlinkdir.m4	2013-05-17 10:28:02.383789100 +0000
@@ -24,7 +24,7 @@
   gnu[[0-9]]* | \
   linux-* | linux | \
   freebsd2.2* | freebsd[[3-9]]* | freebsd[[1-9]][[0-9]]* | \
-  cygwin | \
+  cygwin | msys | \
   mingw*)
     AC_DEFINE([UNLINK_CANNOT_UNLINK_DIR], [1],
       [Define to 1 if unlink (dir) cannot possibly succeed.]);;
diff -Naur coreutils-8.21-cygwin/m4/wctob.m4 coreutils-8.21-msys/m4/wctob.m4
--- coreutils-8.21-cygwin/m4/wctob.m4	2013-01-02 12:34:46.000000000 +0000
+++ coreutils-8.21-msys/m4/wctob.m4	2013-05-17 10:41:41.685546900 +0000
@@ -29,14 +29,14 @@
 changequote(,)dnl
         case "$host_os" in
             # Guess no on Solaris <= 9 and Cygwin.
-          solaris2.[1-9] | solaris2.[1-9].* | cygwin*)
+          solaris2.[1-9] | solaris2.[1-9].* | cygwin* | msys*)
             gl_cv_func_wctob_works="guessing no" ;;
             # Guess yes otherwise.
           *) gl_cv_func_wctob_works="guessing yes" ;;
         esac
 changequote([,])dnl
         case "$host_os" in
-          cygwin*)
+          cygwin* | msys*)
             AC_RUN_IFELSE(
               [AC_LANG_SOURCE([[
 #include <locale.h>
diff -Naur coreutils-8.21-cygwin/Makefile.in coreutils-8.21-msys/Makefile.in
--- coreutils-8.21-cygwin/Makefile.in	2013-05-17 10:18:44.589843800 +0000
+++ coreutils-8.21-msys/Makefile.in	2013-05-17 10:36:31.981445300 +0000
@@ -8230,7 +8230,7 @@
 	  case '$(host_os)' in \
 	    darwin[56]*) \
 	      need_charset_alias=true ;; \
-	    darwin* | cygwin* | mingw* | pw32* | cegcc*) \
+	    darwin* | cygwin* | msys* | mingw* | pw32* | cegcc*) \
 	      need_charset_alias=false ;; \
 	    *) \
 	      need_charset_alias=true ;; \
