diff -Naur coreutils-8.21-fpending/lib/config.hin coreutils-8.21/lib/config.hin
--- coreutils-8.21-fpending/lib/config.hin	2013-10-20 10:11:10.369000000 +0400
+++ coreutils-8.21/lib/config.hin	2013-10-20 10:24:12.124800000 +0400
@@ -1298,10 +1298,6 @@
    don't. */
 #undef HAVE_DECL__SYS_SIGLIST
 
-/* Define to 1 if you have the declaration of `__fpending', and to 0 if you
-   don't. */
-#undef HAVE_DECL___FPENDING
-
 /* Define to 1 if you have the declaration of `__fsetlocking', and to 0 if you
    don't. */
 #undef HAVE_DECL___FSETLOCKING
@@ -2519,9 +2515,6 @@
    member named physmem. */
 #undef HAVE__SYSTEM_CONFIGURATION
 
-/* Define to 1 if you have the `__fpending' function. */
-#undef HAVE___FPENDING
-
 /* Define to 1 if you have the `__fpurge' function. */
 #undef HAVE___FPURGE
 
diff -Naur coreutils-8.21-fpending/lib/fpending.h coreutils-8.21/lib/fpending.h
--- coreutils-8.21-fpending/lib/fpending.h	2013-09-22 19:25:57.000000000 +0400
+++ coreutils-8.21/lib/fpending.h	2013-10-20 10:26:22.710600000 +0400
@@ -20,11 +20,10 @@
 
 #include <stddef.h>
 #include <stdio.h>
+#if HAVE_STDIO_EXT_H
+# include <stdio_ext.h>
+#endif
 
-#if HAVE_DECL___FPENDING
-# if HAVE_STDIO_EXT_H
-#  include <stdio_ext.h>
-# endif
-#else
+#ifndef __fpending
 size_t __fpending (FILE *);
 #endif
diff -Naur coreutils-8.21-fpending/m4/fpending.m4 coreutils-8.21/m4/fpending.m4
--- coreutils-8.21-fpending/m4/fpending.m4	2013-09-22 19:25:57.000000000 +0400
+++ coreutils-8.21/m4/fpending.m4	2013-10-20 10:27:26.603600000 +0400
@@ -16,14 +16,19 @@
 AC_DEFUN([gl_FUNC_FPENDING],
 [
   AC_CHECK_HEADERS_ONCE([stdio_ext.h])
-  AC_CHECK_FUNCS_ONCE([__fpending])
-  fp_headers='
-#     include <stdio.h>
-#     if HAVE_STDIO_EXT_H
-#      include <stdio_ext.h>
-#     endif
-'
-  AC_CHECK_DECLS([__fpending], , , $fp_headers)
+  AC_CACHE_CHECK([for __fpending], [gl_cv_func___fpending],
+    [
+      AC_LINK_IFELSE(
+	[AC_LANG_PROGRAM(
+	   [[#include <stdio.h>
+	     #if HAVE_STDIO_EXT_H
+	     # include <stdio_ext.h>
+	     #endif
+	   ]],
+	   [[return ! __fpending (stdin);]])],
+	[gl_cv_func___fpending=yes],
+	[gl_cv_func___fpending=no])
+    ])
 ])
 
 AC_DEFUN([gl_PREREQ_FPENDING],
diff -Naur coreutils-8.21-fpending/m4/gnulib-comp.m4 coreutils-8.21/m4/gnulib-comp.m4
--- coreutils-8.21-fpending/m4/gnulib-comp.m4	2013-09-22 19:29:16.000000000 +0400
+++ coreutils-8.21/m4/gnulib-comp.m4	2013-10-20 10:28:16.554800000 +0400
@@ -679,7 +679,7 @@
   gl_STDIO_MODULE_INDICATOR([fopen])
   gl_MODULE_INDICATOR([fopen-safer])
   gl_FUNC_FPENDING
-  if test $ac_cv_func___fpending = no; then
+  if test $gl_cv_func___fpending = no; then
     AC_LIBOBJ([fpending])
     gl_PREREQ_FPENDING
   fi
