--- src/coreutils-8.22/Makefile.in.orig	2014-01-30 11:49:05.090504600 +0400
+++ src/coreutils-8.22/Makefile.in	2014-01-30 12:04:56.602188428 +0400
@@ -9980,8 +9980,8 @@
 	  && $(run_help2man)						\
 		     --source='$(PACKAGE_STRING)'			\
 		     --include=$(srcdir)/man/$$name.x			\
-		     --output=$$t/$$name.1 $$t/$$name			\
 		     --info-page='coreutils \(aq'$$name' invocation\(aq' \
+		     --output=$$t/$$name.1 $$t/$$name			\
 	  && sed \
 	       -e 's|$*\.td/||g' \
 	       -e '/For complete documentation/d' \
