--- origsrc/src/chown-core.h	2023-07-05 05:09:55.000000000 -0600
+++ src/src/chown-core.h	2023-09-09 20:22:55.569928600 -0600
@@ -51,7 +51,7 @@ struct Chown_option
 
   /* Pointer to the device and inode numbers of '/', when --recursive.
      Need not be freed.  Otherwise nullptr.  */
-  struct dev_ino *root_dev_ino;
+  struct root_dev_ino *root_dev_ino;
 
   /* This corresponds to the --dereference (opposite of -h) option.  */
   bool affect_symlink_referent;
