--- origsrc/src/chcon.c	2023-07-05 05:09:55.000000000 -0600
+++ src/src/chcon.c	2023-09-09 20:07:55.163173700 -0600
@@ -47,7 +47,7 @@ static bool verbose;
 
 /* Pointer to the device and inode numbers of '/', when --recursive.
    Otherwise nullptr.  */
-static struct dev_ino *root_dev_ino;
+static struct root_dev_ino *root_dev_ino;
 
 /* The name of the context file is being given. */
 static char const *specified_context;
@@ -571,7 +571,7 @@ main (int argc, char **argv)
 
   if (recurse && preserve_root)
     {
-      static struct dev_ino dev_ino_buf;
+      static struct root_dev_ino dev_ino_buf;
       root_dev_ino = get_root_dev_ino (&dev_ino_buf);
       if (root_dev_ino == nullptr)
         die (EXIT_FAILURE, errno, _("failed to get attributes of %s"),
