--- origsrc/src/install.c	2023-07-05 05:09:55.000000000 -0600
+++ src/src/install.c	2023-09-09 22:14:29.695309700 -0600
@@ -44,6 +44,10 @@
 #include "utimens.h"
 #include "xstrtol.h"
 
+#ifdef __CYGWIN__
+# include "cygwin.h"
+#endif
+
 /* The official name of this program (e.g., no 'g' prefix).  */
 #define PROGRAM_NAME "install"
 
@@ -504,6 +508,16 @@ strip (char const *name)
         char const *safe_name = name;
         if (name && *name == '-')
           safe_name = file_name_concat (".", name, nullptr);
+#if __CYGWIN__
+        {
+	 /* Check for .exe here, since strip doesn't.  */
+	   char *p;
+	   if (((p = strchr (safe_name, '\0') - 4) <= safe_name
+	         || strcasecmp (p, ".exe") != 0)
+	         && 0 < cygwin_spelling (safe_name))
+            CYGWIN_APPEND_EXE (p, safe_name);
+        }
+#endif /* __CYGWIN__ */
         execlp (strip_program, strip_program, safe_name, nullptr);
         error (EXIT_FAILURE, errno, _("cannot run %s"),
                quoteaf (strip_program));
