--- origsrc/lib/hash-pjw.h	2016-07-15 14:47:39.000000000 -0500
+++ src/lib/hash-pjw.h	2017-02-03 13:32:48.725613200 -0600
@@ -21,3 +21,4 @@
    The result is platform dependent: it depends on the size of the 'size_t'
    type and on the signedness of the 'char' type.  */
 extern size_t hash_pjw (void const *x, size_t tablesize) _GL_ATTRIBUTE_PURE;
+extern size_t hash_pjw_case (void const *x, size_t tablesize) _GL_ATTRIBUTE_PURE;
