--- origsrc/src/chroot.c	2016-11-06 16:15:29.000000000 -0600
+++ src/src/chroot.c	2017-02-03 13:32:48.741217700 -0600
@@ -171,7 +171,8 @@ static bool
 is_root (const char* dir)
 {
   char *resolved = canonicalize_file_name (dir);
-  bool is_res_root = resolved && STREQ ("/", resolved);
+  bool is_res_root = resolved && (STREQ ("/", resolved)
+				  || STREQ ( "//", resolved));
   free (resolved);
   return is_res_root;
 }
