--- origsrc/src/rm.c	2016-11-06 16:15:30.000000000 -0600
+++ src/src/rm.c	2017-02-03 13:32:48.756816900 -0600
@@ -327,7 +327,7 @@ main (int argc, char **argv)
 
   if (x.recursive && preserve_root)
     {
-      static struct dev_ino dev_ino_buf;
+      static struct root_dev_ino dev_ino_buf;
       x.root_dev_ino = get_root_dev_ino (&dev_ino_buf);
       if (x.root_dev_ino == NULL)
         die (EXIT_FAILURE, errno, _("failed to get attributes of %s"),
