--- origsrc/configure.ac	2016-11-29 12:03:45.000000000 -0600
+++ src/configure.ac	2017-02-03 13:32:48.725613200 -0600
@@ -470,7 +470,8 @@ gl_WARN_ADD([-Werror], [CFLAGS])
 gl_WARN_ADD([-errwarn], [CFLAGS])
 # Put this message here, after gl_WARN_ADD's chatter.
 AC_MSG_CHECKING([whether this system supports stdbuf])
-CFLAGS="-fPIC $CFLAGS"
+# -fPIC on cygwin gives lots of warnings
+#CFLAGS="-fPIC $CFLAGS"
 LDFLAGS="-shared $LDFLAGS"
 stdbuf_supported=no
 # Note we only LINK here rather than RUN to support cross compilation
@@ -488,7 +489,7 @@ AC_LINK_IFELSE(
   ],
   [stdbuf_supported=yes])
 AC_MSG_RESULT([$stdbuf_supported])
-if test "$stdbuf_supported" = "yes" && test -z "$EXEEXT"; then
+if test "$stdbuf_supported" = "yes"; then
   gl_ADD_PROG([optional_bin_progs], [stdbuf])
 fi
 CFLAGS=$ac_save_CFLAGS
