--- origsrc/src/remove.h	2023-09-10 13:37:14.816502400 -0600
+++ src/src/remove.h	2023-09-10 13:36:42.028328800 -0600
@@ -54,7 +54,7 @@ struct rm_options
 
   /* Pointer to the device and inode numbers of '/', when --recursive
      and preserving '/'.  Otherwise null.  */
-  struct dev_ino *root_dev_ino;
+  struct root_dev_ino *root_dev_ino;
 
   /* If true, do not traverse into (or remove) any directory that is
      the root of a file system.  I.e., a separate device.  */
