--- src/coreutils-8.21/Makefile.in.orig	2013-06-29 22:56:26.000000000 +0400
+++ src/coreutils-8.21/Makefile.in	2013-06-29 23:02:02.000000000 +0400
@@ -3294,7 +3294,7 @@
 
 # Ensure we don't link against libcoreutils.a as that lib is
 # not compiled with -fPIC which causes issues on 64 bit at least
-src_libstdbuf_so_LDADD = 
+src_libstdbuf_so_LDADD = $(LDADD)
 
 # Note libstdbuf is only compiled if GCC is available
 # (as per the check in configure.ac), so these flags should be available.
