--- origsrc/src/remove.h	2016-03-09 04:15:39.000000000 -0600
+++ src/src/remove.h	2017-02-03 13:32:48.756816900 -0600
@@ -54,7 +54,7 @@ struct rm_options
 
   /* Pointer to the device and inode numbers of '/', when --recursive
      and preserving '/'.  Otherwise NULL.  */
-  struct dev_ino *root_dev_ino;
+  struct root_dev_ino *root_dev_ino;
 
   /* If nonzero, stdin is a tty.  */
   bool stdin_tty;
