--- origsrc/src/dircolors.c	2016-11-06 16:15:30.000000000 -0600
+++ src/src/dircolors.c	2017-02-03 13:32:48.741217700 -0600
@@ -496,8 +496,12 @@ main (int argc, char **argv)
             }
           else
             {
+              /* tcsh treats LS_COLORS as a magic shell variable for its
+                 builtin ls-F, but does not recognize all the categories
+                 that coreutils ls does.  Therefore, silence stderr to
+                 avoid messages like "Unknown colorls variable `su'.".  */
               prefix = "setenv LS_COLORS '";
-              suffix = "'\n";
+              suffix = "' >&/dev/null\n";
             }
           fputs (prefix, stdout);
           fwrite (s, 1, len, stdout);
