--- a/src/dircolors.hin
+++ b/src/dircolors.hin
@@ -33,6 +33,7 @@ TERM konsole
 TERM kterm
 TERM linux
 TERM linux-c
+TERM mintty
 TERM mlterm
 TERM putty
 TERM rxvt*
@@ -41,6 +42,9 @@ TERM st
 TERM terminator
 TERM tmux*
 TERM vt100
+TERM vt241
+TERM vt34*
+TERM vt525
 TERM xterm*
 
 # ===================================================================
@@ -62,6 +66,8 @@ RESET 0		# reset to "normal" color
 DIR 01;34	# directory
 LINK 01;36	# symbolic link.  (If you set this to 'target' instead of a
                 # numerical value, the color is as for the file pointed to.)
+.lnk 01;36	# Windows shortcut
+.url 01;36	# Windows Internet shortcut
 MULTIHARDLINK 00	# regular file with more than one link
 FIFO 40;33	# pipe
 SOCK 01;35	# socket
@@ -96,10 +102,15 @@ EXEC 01;32
 #.com 01;32
 #.btm 01;32
 #.bat 01;32
+#.dll 01;32 # Windows Dynamic Link Library
+#.xll 01;32 # Excel extension DLLs
 # Or if you want to color scripts even if they do not have the
 # executable bit actually set.
 #.sh  01;32
 #.csh 01;32
+#.dash 01;32
+#.ksh 01;32
+#.zsh 01;32
 
 # archives or compressed (bright red)
 .tar 01;31
@@ -148,6 +159,15 @@ EXEC 01;32
 .swm 01;31
 .dwm 01;31
 .esd 01;31
+.epub 01;31
+.ods 01;31
+.odt 01;31
+.docx 01;31
+.pptx 01;31
+.xlsx 01;31
+.xps 01;31
+.accumap 01;31
+.lwell 01;31
 
 # image formats
 .avif 01;35
@@ -199,10 +219,14 @@ EXEC 01;32
 .yuv 01;35
 .cgm 01;35
 .emf 01;35
+.ico 01;35
+.icl 01;35
 
 # https://wiki.xiph.org/MIME_Types_and_File_Extensions
 .ogv 01;35
 .ogx 01;35
+.anx 01;35
+.axv 01;35
 
 # audio formats
 .aac 00;36
@@ -223,6 +247,7 @@ EXEC 01;32
 .opus 00;36
 .spx 00;36
 .xspf 00;36
+.axa 00;36
 
 # backup files
 *~ 00;90
