--- origsrc/src/chown.c	2016-11-06 16:15:29.000000000 -0600
+++ src/src/chown.c	2017-02-03 13:32:48.741217700 -0600
@@ -313,7 +313,7 @@ main (int argc, char **argv)
 
   if (chopt.recurse && preserve_root)
     {
-      static struct dev_ino dev_ino_buf;
+      static struct root_dev_ino dev_ino_buf;
       chopt.root_dev_ino = get_root_dev_ino (&dev_ino_buf);
       if (chopt.root_dev_ino == NULL)
         die (EXIT_FAILURE, errno, _("failed to get attributes of %s"),
