--- origsrc/src/chown-core.h	2016-03-09 04:15:39.000000000 -0600
+++ src/src/chown-core.h	2017-02-03 13:32:48.741217700 -0600
@@ -50,7 +50,7 @@ struct Chown_option
 
   /* Pointer to the device and inode numbers of '/', when --recursive.
      Need not be freed.  Otherwise NULL.  */
-  struct dev_ino *root_dev_ino;
+  struct root_dev_ino *root_dev_ino;
 
   /* This corresponds to the --dereference (opposite of -h) option.  */
   bool affect_symlink_referent;
