--- origsrc/src/chmod.c	2023-08-29 05:39:27.000000000 -0600
+++ src/src/chmod.c	2023-09-09 20:16:41.171411900 -0600
@@ -87,7 +87,7 @@ static enum Verbosity verbosity = V_off;
 
 /* Pointer to the device and inode numbers of '/', when --recursive.
    Otherwise nullptr.  */
-static struct dev_ino *root_dev_ino;
+static struct root_dev_ino *root_dev_ino;
 
 /* For long options that have no equivalent short option, use a
    non-character as a pseudo short option, starting with CHAR_MAX + 1.  */
@@ -553,7 +553,7 @@ main (int argc, char **argv)
 
   if (recurse && preserve_root)
     {
-      static struct dev_ino dev_ino_buf;
+      static struct root_dev_ino dev_ino_buf;
       root_dev_ino = get_root_dev_ino (&dev_ino_buf);
       if (root_dev_ino == nullptr)
         die (EXIT_FAILURE, errno, _("failed to get attributes of %s"),
