diff -Naur IPC-Run3-0.046/t/fork.t IPC-Run3-0.046/t/fork.t
--- IPC-Run3-0.046/t/fork.t	2009-01-15 15:28:27.000000000 +0000
+++ IPC-Run3-0.046/t/fork.t	2013-06-16 15:27:23.818914100 +0000
@@ -2,7 +2,7 @@
 
 use Test::More;
 plan skip_all => "Test::More 0.31 required for no_ending()" if $Test::More::VERSION <= 0.31;
-plan skip_all => "tests fail on Win32 and Cygwin" if $^O =~ /^(MSWin32|cygwin)$/;
+plan skip_all => "tests fail on Win32 and Cygwin" if $^O =~ /^(MSWin32|cygwin|msys)$/;
 plan tests => 5;
 
 use IPC::Run3;
