--- origsrc/gmp-5.1.1/configure.ac	2013-02-11 09:29:14.000000000 -0600
+++ src/gmp-5.1.1/configure.ac	2013-03-05 03:19:44.015673500 -0600
@@ -2427,16 +2427,6 @@ if test "${with_tags+set}" != set; then
   fi
 fi
 
-# The dead hand of AC_REQUIRE makes AC_PROG_LIBTOOL expand and execute
-# AC_PROG_F77, even when F77 is not in the selected with_tags.  This is
-# probably harmless, but it's unsightly and bloats our configure, so pretend
-# AC_PROG_F77 has been expanded already.
-#
-# FIXME: Rumour has it libtool will one day provide a way for a configure.in
-# to say what it wants from among supported languages etc.
-#
-AC_PROVIDE([AC_PROG_F77])
-
 AC_PROG_LIBTOOL
 
 # Generate an error here if attempting to build both shared and static when
