--- gmp-5.1.2-orig/configfsf.guess	2013-05-20 14:59:56.000000000 +0000
+++ gmp-5.1.2/configfsf.guess	2013-06-02 05:43:41.018554700 +0000
@@ -851,6 +851,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur gmp-5.1.2-orig/configure.ac gmp-5.1.2/configure.ac
--- gmp-5.1.2-orig/configure.ac	2013-05-20 14:59:56.000000000 +0000
+++ gmp-5.1.2/configure.ac	2013-06-02 05:42:33.099609400 +0000
@@ -1703,7 +1703,7 @@
 	    cclist_64="$cclist_64 cc"
 	    cc_64_cflags="-xO3 -m64"
 	    ;;
-	  *-*-mingw* | *-*-cygwin)
+	  *-*-mingw* | *-*-cygwin | *-*-msys)
 	    limb_64=longlong
 	    CALLING_CONVENTIONS_OBJS_64=""
 	    AC_DEFINE(HOST_DOS64,1,[Define to 1 for Windos/64])
@@ -2363,7 +2363,7 @@
 
 AC_SUBST(LIBGMP_DLL,0)
 case $host in
-  *-*-cygwin* | *-*-mingw* | *-*-pw32* | *-*-os2*)
+  *-*-cygwin* | *-*-msys* | *-*-mingw* | *-*-pw32* | *-*-os2*)
     # By default, build only static.
     if test -z "$enable_shared"; then
       enable_shared=no
@@ -3457,7 +3457,7 @@
 	  case $host in
 	    *-*-darwin*)
 	      GMP_INCLUDE_MPN(x86_64/darwin.m4) ;;
-	    *-*-mingw* | *-*-cygwin)
+	    *-*-mingw* | *-*-cygwin | *-*-msys)
 	      GMP_INCLUDE_MPN(x86_64/dos64.m4) ;;
 	  esac
           ;;
@@ -3686,7 +3686,7 @@
 
 if test x$cross_compiling = xyes ; then
    case "$host" in
-     *-*-mingw* | *-*-cygwin)
+     *-*-mingw* | *-*-cygwin | *-*-msys)
      if test x$ABI = x64 ; then
 	AC_MSG_NOTICE([If wine64 is installed, use make check TESTS_ENVIRONMENT=wine64.])
      else
