diff --git a/glib/glib-private.h b/glib/glib-private.h
index 7ab2579b0..3318e4c91 100644
--- a/glib/glib-private.h
+++ b/glib/glib-private.h
@@ -64,7 +64,7 @@
  * as we'd like: https://stackoverflow.com/a/11529277/210151 and
  * https://devblogs.microsoft.com/oldnewthing/20200731-00/?p=104024
  */
-#elif defined (G_OS_UNIX) && !defined (__APPLE__) && g_macro__has_attribute (weak)
+#elif defined (G_OS_UNIX) && !defined (__APPLE__) && g_macro__has_attribute (weak) && !defined(__CYGWIN__)
 
 #define HAS_DYNAMIC_ASAN_LOADING
 
