diff --git a/gio/glib-compile-schemas.c b/gio/glib-compile-schemas.c
index 83184e1e8..32b02fa39 100644
--- a/gio/glib-compile-schemas.c
+++ b/gio/glib-compile-schemas.c
@@ -2166,6 +2166,7 @@ main (int argc, char **argv)
   gboolean strict = FALSE;
   gchar **schema_files = NULL;
   gchar **override_files = NULL;
+  gchar **command_line = NULL;
   GOptionContext *context = NULL;
   gint retval;
   GOptionEntry entries[] = {
@@ -2207,12 +2208,22 @@ main (int argc, char **argv)
        "and the cache file is called gschemas.compiled."));
   g_option_context_add_main_entries (context, entries, GETTEXT_PACKAGE);
 
-  if (!g_option_context_parse (context, &argc, &argv, &error))
+#ifdef G_OS_WIN32
+  command_line = g_win32_get_command_line ();
+  if (!g_option_context_parse_strv(context, command_line, &error))
+    {
+      fprintf (stderr, "%s\n", error->message);
+      retval = 1;
+      goto done;
+    }
+#else
+  if (!g_option_context_parse(context, &argc, &argv, &error))
     {
       fprintf (stderr, "%s\n", error->message);
       retval = 1;
       goto done;
     }
+#endif
 
   if (show_version_and_exit)
     {
@@ -2317,6 +2328,7 @@ done:
   g_free (target);
   g_strfreev (schema_files);
   g_strfreev (override_files);
+  g_strfreev (command_line);
   g_option_context_free (context);
 
 #ifdef G_OS_WIN32
