--- origsrc/glib-2.36.3/configure.ac	2013-08-04 18:11:14.140206800 -0500
+++ src/glib-2.36.3/configure.ac	2013-08-04 18:30:21.852852200 -0500
@@ -3584,7 +3584,7 @@ dnl visibility attributes for symbol exp
 dnl
 GLIB_HIDDEN_VISIBILITY_CFLAGS=""
 case "$host" in
-  *-*-mingw*)
+  *-*-cygwin*|*-*-mingw*)
     dnl on mingw32 we do -fvisibility=hidden and __declspec(dllexport)
     AC_DEFINE([_GLIB_EXTERN], [__attribute__((visibility("default"))) __declspec(dllexport) extern],
               [defines how to decorate public symbols while building])
