--- origsrc/glib-2.36.4/glib/goption.c	2013-08-07 09:36:32.000000000 -0500
+++ src/glib-2.36.4/glib/goption.c	2013-10-21 13:25:01.242539100 -0500
@@ -1671,7 +1671,7 @@ free_pending_nulls (GOptionContext *cont
 static char *
 platform_get_argv0 (void)
 {
-#if defined __linux
+#if defined __linux || defined(__CYGWIN__)
   char *cmdline;
   char *base_arg0;
   gsize len;
--- origsrc/glib-2.36.4/glib/tests/option-argv0.c	2013-05-07 13:26:07.000000000 -0500
+++ src/glib-2.36.4/glib/tests/option-argv0.c	2013-10-21 13:26:28.800547100 -0500
@@ -27,7 +27,7 @@
 #include <stdio.h>
 #include <string.h>
 
-#if defined __linux || defined __OpenBSD__
+#if defined __linux || defined __OpenBSD__ || defined __CYGWIN__
 static void
 test_platform_argv0 (void)
 {
@@ -57,7 +57,7 @@ main (int   argc,
   /* Note - we can't actually use g_test_* because g_test_init mutates
    * g_get_prgname() which is exactly what we wanted to test =/
    */
-#if defined __linux || defined __OpenBSD__
+#if defined __linux || defined __OpenBSD__ || defined __CYGWIN__
   g_print ("/option/argv0: ");
   test_platform_argv0 ();
   g_print ("OK\n");
