--- origsrc/glib-2.36.4/docs/reference/gio/gdbus-object-manager-example/Makefile.am	2013-08-07 09:34:32.000000000 -0500
+++ src/glib-2.36.4/docs/reference/gio/gdbus-object-manager-example/Makefile.am	2013-10-22 00:54:44.900548900 -0500
@@ -11,10 +11,10 @@ DOC_MAIN_SGML_FILE=gdbus-object-manager-
 SCAN_OPTIONS=--deprecated-guards="G_DISABLE_DEPRECATED"
 
 # The directory containing the source code. Relative to $(srcdir)
-DOC_SOURCE_DIR=$(top_builddir)/gio/tests/gdbus-object-manager-example
+DOC_SOURCE_DIR=$(top_srcdir)/gio/tests/gdbus-object-manager-example
 
-HFILE_GLOB=$(top_builddir)/gio/tests/gdbus-object-manager-example/*.h
-CFILE_GLOB=$(top_builddir)/gio/tests/gdbus-object-manager-example/*.c
+HFILE_GLOB=$(top_srcdir)/gio/tests/gdbus-object-manager-example/*.h
+CFILE_GLOB=$(top_srcdir)/gio/tests/gdbus-object-manager-example/*.c
 
 # Headers to ignore
 IGNORE_HFILES =				\
