diff -Naur glib-2.38.1-orig/compile glib-2.38.1/compile
--- glib-2.38.1-orig/compile	2013-08-07 18:43:05.000000000 +0400
+++ glib-2.38.1/compile	2013-08-23 10:47:48.875000000 +0400
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur glib-2.38.1-orig/config.guess glib-2.38.1/config.guess
--- glib-2.38.1-orig/config.guess	2013-08-07 18:43:05.000000000 +0400
+++ glib-2.38.1/config.guess	2013-08-23 10:48:19.218750000 +0400
@@ -866,6 +866,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
--- glib-2.42.0/configure.ac.orig	2014-09-26 10:16:30.460314900 +0900
+++ glib-2.42.0/configure.ac	2014-09-26 10:18:01.902282100 +0900
@@ -1459,7 +1459,7 @@
 dnl *** force native WIN32 shared lib loader 
 if test -z "$G_MODULE_IMPL"; then
   case "$host" in
-  *-*-mingw*|*-*-cygwin*) G_MODULE_IMPL=G_MODULE_IMPL_WIN32 ;;
+  *-*-mingw*|*-*-cygwin*|*-*-msys*) G_MODULE_IMPL=G_MODULE_IMPL_WIN32 ;;
   esac
 fi
 dnl *** force native AIX library loader
@@ -1834,7 +1834,7 @@
 
 AC_MSG_CHECKING([whether to compile timeloop])
 case "$host" in
-  *-*-cygwin*|*-*-mingw*|*-*-minix)
+  *-*-cygwin*|*-*-msys*|*-*-mingw*|*-*-minix)
     enable_timeloop=no
     ;;
   *)
@@ -1846,7 +1846,7 @@
 
 AC_MSG_CHECKING([if building for some Win32 platform])
 case "$host" in
-  *-*-mingw*|*-*-cygwin*)
+  *-*-mingw*|*-*-cygwin*|*-*-msys*)
     platform_win32=yes
     ;;
   *)
@@ -1962,7 +1962,7 @@
   # One of them does for most gcc versions and some other platforms/compilers
   # too and could be considered as the canonical way to go. 
   case $host in
-    *-*-cygwin*|*-*-darwin*)
+    *-*-cygwin*|*-*-msys*|*-*-darwin*)
        # skip cygwin and darwin -pthread or -pthreads test
        ;;
     *-solaris*)
@@ -3360,7 +3360,7 @@
 
 g_pid_type="$glib_pid_type"
 case $host in
-  *-*-cygwin*)
+  *-*-cygwin* | *-*-msys*)
     glib_os="#define G_OS_UNIX
 #define G_PLATFORM_WIN32
 #define G_WITH_CYGWIN"
@@ -3496,7 +3496,7 @@
 dnl
 GLIB_HIDDEN_VISIBILITY_CFLAGS=""
 case "$host" in
-  *-*-cygwin*|*-*-mingw*)
+  *-*-cygwin*|*-*-msys*|*-*-mingw*)
     dnl on mingw32 we do -fvisibility=hidden and __declspec(dllexport)
     AC_DEFINE([_GLIB_EXTERN], [__attribute__((visibility("default"))) __declspec(dllexport) extern],
               [defines how to decorate public symbols while building])
