From d8a64ed1d63710ad764fbf3b1fb28d08204f9e72 Mon Sep 17 00:00:00 2001
From: Kamil Dudka <kdudka@redhat.com>
Date: Wed, 13 Feb 2013 15:28:36 +0100
Subject: [PATCH] avoid build failure caused by mistakenly excluded <locale.h>

---
 src/LYMain.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/LYMain.c b/src/LYMain.c
index 9528be2..aaddb7b 100644
--- a/src/LYMain.c
+++ b/src/LYMain.c
@@ -58,7 +58,7 @@
 #include <io.h>
 #endif
 
-#if defined(LOCALE) && (!defined(HAVE_LIBINTL_H) || !defined(LC_ALL))
+#if defined(LOCALE)
 #undef gettext			/* Solaris locale.h prototypes gettext() */
 #include <locale.h>
 #ifndef HAVE_GETTEXT
-- 
1.7.1

