--- lynx2.8.9rel.1/configure.orig	2021-11-23 02:27:37.585543700 -0500
+++ lynx2.8.9rel.1/configure	2021-11-23 02:32:35.001980900 -0500
@@ -4457,6 +4457,10 @@ fi
 	TRY_CFLAGS="$TRY_CFLAGS -DWIN_EX -D_WINDOWS_NSL"
 	LIBS="$LIBS -luser32"
 	;;
+msys*)
+	TRY_CFLAGS="$TRY_CFLAGS -DWIN_EX -D_WINDOWS_NSL"
+	LIBS="$LIBS -luser32"
+	;;
 (hpux*)
 	TRY_CFLAGS="$TRY_CFLAGS -D_XOPEN_SOURCE_EXTENDED"
 	test -z "$GCC" && TRY_CFLAGS="$TRY_CFLAGS -Wp,-H600000 -DSYS_HPUX"
@@ -9265,7 +9269,7 @@ case $host_os in
 	cf_xopen_source="-D_SGI_SOURCE"
 	cf_XOPEN_SOURCE=
 	;;
-(linux*|uclinux*|gnu*|mint*|k*bsd*-gnu|cygwin)
+(linux*|uclinux*|gnu*|mint*|k*bsd*-gnu|cygwin|mys*)
 
 echo "$as_me:9270: checking if we must define _GNU_SOURCE" >&5
 echo $ECHO_N "checking if we must define _GNU_SOURCE... $ECHO_C" >&6
