diff --git a/options-table.c b/options-table.c
index f5e973e..29f16f4 100644
--- a/options-table.c
+++ b/options-table.c
@@ -508,7 +508,7 @@ const struct options_table_entry options_table[] = {
 	  .type = OPTIONS_TABLE_ARRAY,
 	  .scope = OPTIONS_TABLE_SESSION,
 	  .default_str = "DISPLAY KRB5CCNAME SSH_ASKPASS SSH_AUTH_SOCK "
-	  		 "SSH_AGENT_PID SSH_CONNECTION WINDOWID XAUTHORITY"
+			 "SSH_AGENT_PID SSH_CONNECTION WINDOWID XAUTHORITY MSYSTEM"
 	},
 
 	{ .name = "visual-activity",
