--- options-table.c.orig	2021-04-15 05:28:08.620465500 +0900
+++ options-table.c	2021-04-15 05:32:12.676737900 +0900
@@ -706,7 +706,7 @@
 	  .scope = OPTIONS_TABLE_SESSION,
 	  .flags = OPTIONS_TABLE_IS_ARRAY,
 	  .default_str = "DISPLAY KRB5CCNAME SSH_ASKPASS SSH_AUTH_SOCK "
-			 "SSH_AGENT_PID SSH_CONNECTION WINDOWID XAUTHORITY",
+			 "SSH_AGENT_PID SSH_CONNECTION WINDOWID XAUTHORITY MSYSTEM",
 	  .text = "List of environment variables to update in the session "
 		  "environment when a client is attached."
 	},
