--- tmux.c.orig	2022-06-01 17:37:46.135702500 +0900
+++ tmux.c	2022-06-01 17:43:45.993181000 +0900
@@ -222,7 +222,11 @@
 		xasprintf(cause, "%s is not a directory", base);
 		goto fail;
 	}
+#ifdef __MSYS__
+	if (sb.st_uid != uid) {
+#else /* __MSYS__ */
 	if (sb.st_uid != uid || (sb.st_mode & S_IRWXO) != 0) {
+#endif /* __MSYS__ */
 		xasprintf(cause, "directory %s has unsafe permissions", base);
 		goto fail;
 	}
