--- tar-1.28/lib/rmt.h.orig	2018-04-04 14:13:27.434833600 +0200
+++ tar-1.28/lib/rmt.h	2018-04-04 14:19:17.243518800 +0200
@@ -35,10 +35,14 @@
    Distributed File System (DFS).  However, when --force-local, a
    filename is never remote.  */
 
+/*
 #define _remdev(dev_name) \
   (!force_local_option && (rmt_dev_name__ = strchr (dev_name, ':')) \
    && rmt_dev_name__ > (dev_name) \
    && ! memchr (dev_name, '/', rmt_dev_name__ - (dev_name)))
+*/
+
+#define _remdev(dev_name) 0
 
 #define _isrmt(fd) \
   ((fd) >= __REM_BIAS)
