diff -Naur tar-1.33-orig/build-aux/compile tar-1.33/build-aux/compile
--- tar-1.33-orig/build-aux/compile	2021-01-07 17:25:19.000000000 +0300
+++ tar-1.33/build-aux/compile	2021-01-12 09:07:31.903196000 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur tar-1.33-orig/build-aux/config.guess tar-1.33/build-aux/config.guess
--- tar-1.33-orig/build-aux/config.guess	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/build-aux/config.guess	2021-01-12 09:07:31.906821100 +0300
@@ -915,6 +915,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"$(echo "$UNAME_RELEASE"|sed -e 's/[^.]*//')"
 	exit ;;
diff -Naur tar-1.33-orig/build-aux/config.rpath tar-1.33/build-aux/config.rpath
--- tar-1.33-orig/build-aux/config.rpath	2021-01-07 17:29:12.000000000 +0300
+++ tar-1.33/build-aux/config.rpath	2021-01-12 09:07:31.910547300 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur tar-1.33-orig/m4/btowc.m4 tar-1.33/m4/btowc.m4
--- tar-1.33-orig/m4/btowc.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/btowc.m4	2021-01-12 09:07:31.912322700 +0300
@@ -41,7 +41,7 @@
 changequote(,)dnl
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes on native Windows.
              mingw*)  gl_cv_func_btowc_nul="guessing yes" ;;
                       # Guess yes otherwise.
diff -Naur tar-1.33-orig/m4/double-slash-root.m4 tar-1.33/m4/double-slash-root.m4
--- tar-1.33-orig/m4/double-slash-root.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/double-slash-root.m4	2021-01-12 09:07:31.916347700 +0300
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur tar-1.33-orig/m4/dup2.m4 tar-1.33/m4/dup2.m4
--- tar-1.33-orig/m4/dup2.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/dup2.m4	2021-01-12 09:20:50.807069000 +0300
@@ -71,6 +71,8 @@
            gl_cv_func_dup2_works="guessing no" ;;
          cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
            gl_cv_func_dup2_works="guessing no" ;;
+         msys*) # on msys 1.5.x, dup2(1,1) returns 0
+           gl_cv_func_dup2_works="guessing no" ;;
          aix* | freebsd*)
                  # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
                  # not EBADF.
diff -Naur tar-1.33-orig/m4/fchmodat.m4 tar-1.33/m4/fchmodat.m4
--- tar-1.33-orig/m4/fchmodat.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/fchmodat.m4	2021-01-12 09:21:31.409253900 +0300
@@ -58,7 +58,7 @@
          [gl_cv_func_fchmodat_works=no],
          [case "$host_os" in
             dnl Guess no on Linux with glibc and Cygwin, yes otherwise.
-            linux-gnu* | cygwin*) gl_cv_func_fchmodat_works="guessing no" ;;
+            linux-gnu* | cygwin* | msys*) gl_cv_func_fchmodat_works="guessing no" ;;
             *)                    gl_cv_func_fchmodat_works="$gl_cross_guess_normal" ;;
           esac
          ])
diff -Naur tar-1.33-orig/m4/fcntl.m4 tar-1.33/m4/fcntl.m4
--- tar-1.33-orig/m4/fcntl.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/fcntl.m4	2021-01-12 09:07:31.923235200 +0300
@@ -70,7 +70,7 @@
          [gl_cv_func_fcntl_f_dupfd_works=yes],
          [gl_cv_func_fcntl_f_dupfd_works=no],
          [case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac])])
diff -Naur tar-1.33-orig/m4/getcwd.m4 tar-1.33/m4/getcwd.m4
--- tar-1.33-orig/m4/getcwd.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/getcwd.m4	2021-01-12 09:22:34.149099000 +0300
@@ -55,6 +55,8 @@
             *-musl*)       gl_cv_func_getcwd_null="guessing yes";;
                            # Guess yes on Cygwin.
             cygwin*)       gl_cv_func_getcwd_null="guessing yes";;
+                           # Guess yes on MSYS.
+            msys*)         gl_cv_func_getcwd_null="guessing yes";;
                            # If we don't know, obey --enable-cross-guesses.
             *)             gl_cv_func_getcwd_null="$gl_cross_guess_normal";;
           esac
diff -Naur tar-1.33-orig/m4/getdtablesize.m4 tar-1.33/m4/getdtablesize.m4
--- tar-1.33-orig/m4/getdtablesize.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/getdtablesize.m4	2021-01-12 09:07:31.931994800 +0300
@@ -42,7 +42,7 @@
              [gl_cv_func_getdtablesize_works=yes],
              [gl_cv_func_getdtablesize_works=no],
              [case "$host_os" in
-                cygwin*) # on cygwin 1.5.25, getdtablesize() automatically grows
+                cygwin* | msys* ) # on cygwin 1.5.25, getdtablesize() automatically grows
                   gl_cv_func_getdtablesize_works="guessing no" ;;
                 *) gl_cv_func_getdtablesize_works="guessing yes" ;;
               esac
diff -Naur tar-1.33-orig/m4/malloc.m4 tar-1.33/m4/malloc.m4
--- tar-1.33-orig/m4/malloc.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/malloc.m4	2021-01-12 09:23:28.730950400 +0300
@@ -25,7 +25,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | *-musl* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull="guessing yes" ;;
           # If we don't know, obey --enable-cross-guesses.
           *) ac_cv_func_malloc_0_nonnull="$gl_cross_guess_normal" ;;
diff -Naur tar-1.33-orig/m4/printf.m4 tar-1.33/m4/printf.m4
--- tar-1.33-orig/m4/printf.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/printf.m4	2021-01-12 09:24:25.722450100 +0300
@@ -813,6 +813,7 @@
            irix*)           gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)        gl_cv_func_printf_directive_ls="guessing no";;
            cygwin*)         gl_cv_func_printf_directive_ls="guessing no";;
+           msys*)           gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*)  gl_cv_func_printf_directive_ls="guessing no";;
                             # Guess no on Android.
            linux*-android*) gl_cv_func_printf_directive_ls="guessing no";;
@@ -897,6 +898,7 @@
 changequote(,)dnl
          case "$host_os" in
            cygwin*)         gl_cv_func_printf_flag_grouping="guessing no";;
+           msys*)           gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)         gl_cv_func_printf_flag_grouping="guessing no";;
                             # Guess no on Android.
            linux*-android*) gl_cv_func_printf_flag_grouping="guessing no";;
@@ -1605,7 +1607,7 @@
            darwin[1-6].*)        gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
-           cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           cygwin* | msys*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Naur tar-1.33-orig/m4/realloc.m4 tar-1.33/m4/realloc.m4
--- tar-1.33-orig/m4/realloc.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/realloc.m4	2021-01-12 09:24:50.342352500 +0300
@@ -25,7 +25,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | *-musl* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull="guessing yes" ;;
           # If we don't know, obey --enable-cross-guesses.
           *) ac_cv_func_realloc_0_nonnull="$gl_cross_guess_normal" ;;
diff -Naur tar-1.33-orig/m4/setlocale_null.m4 tar-1.33/m4/setlocale_null.m4
--- tar-1.33-orig/m4/setlocale_null.m4	2021-01-07 17:29:12.000000000 +0300
+++ tar-1.33/m4/setlocale_null.m4	2021-01-12 09:25:23.879954100 +0300
@@ -14,7 +14,7 @@
     [gl_cv_func_setlocale_null_all_mtsafe],
     [case "$host_os" in
        # Guess no on musl libc, macOS, FreeBSD, NetBSD, OpenBSD, AIX, Haiku, Cygwin.
-       *-musl* | darwin* | freebsd* | netbsd* | openbsd* | aix* | haiku* | cygwin*)
+       *-musl* | darwin* | freebsd* | netbsd* | openbsd* | aix* | haiku* | cygwin* | msys*)
          gl_cv_func_setlocale_null_all_mtsafe=no ;;
        # Guess yes on glibc, HP-UX, IRIX, Solaris, native Windows.
        *-gnu* | gnu* | hpux* | irix* | solaris* | mingw*)
@@ -48,7 +48,7 @@
        openbsd* | aix*)
          gl_cv_func_setlocale_null_one_mtsafe=no ;;
        # Guess yes on glibc, musl libc, macOS, FreeBSD, NetBSD, HP-UX, IRIX, Solaris, Haiku, Cygwin, native Windows.
-       *-gnu* | gnu* | *-musl* | darwin* | freebsd* | netbsd* | hpux* | irix* | solaris* | haiku* | cygwin* | mingw*)
+       *-gnu* | gnu* | *-musl* | darwin* | freebsd* | netbsd* | hpux* | irix* | solaris* | haiku* | cygwin* | msys* | mingw*)
          gl_cv_func_setlocale_null_one_mtsafe=yes ;;
        # If we don't know, obey --enable-cross-guesses.
        *)
diff -Naur tar-1.33-orig/m4/threadlib.m4 tar-1.33/m4/threadlib.m4
--- tar-1.33-orig/m4/threadlib.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/threadlib.m4	2021-01-12 09:25:47.309442100 +0300
@@ -413,7 +413,7 @@
          dnl Disable multithreading by default on Cygwin 1.5.x, because it has
          dnl bugs that lead to endless loops or crashes. See
          dnl <https://cygwin.com/ml/cygwin/2009-08/msg00283.html>.
-         cygwin*)
+         cygwin* | msys*)
                case `uname -r` in
                  1.[0-5].*) gl_use_threads=no ;;
                  *)         gl_use_threads=yes ;;
diff -Naur tar-1.33-orig/m4/unlinkdir.m4 tar-1.33/m4/unlinkdir.m4
--- tar-1.33-orig/m4/unlinkdir.m4	2021-01-07 17:27:55.000000000 +0300
+++ tar-1.33/m4/unlinkdir.m4	2021-01-12 09:07:31.950057500 +0300
@@ -25,6 +25,7 @@
   linux-* | linux | \
   freebsd2.2* | freebsd[[3-9]]* | freebsd[[1-9]][[0-9]]* | \
   cygwin | \
+  msys | \
   mingw*)
     AC_DEFINE([UNLINK_CANNOT_UNLINK_DIR], [1],
       [Define to 1 if unlink (dir) cannot possibly succeed.]);;
