diff -Naur tar-1.31-orig/build-aux/compile tar-1.31/build-aux/compile
--- tar-1.31-orig/build-aux/compile	2019-01-02 21:16:45.000000000 +0300
+++ tar-1.31/build-aux/compile	2019-01-15 11:43:57.260624400 +0300
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur tar-1.31-orig/build-aux/config.guess tar-1.31/build-aux/config.guess
--- tar-1.31-orig/build-aux/config.guess	2019-01-02 21:15:07.000000000 +0300
+++ tar-1.31/build-aux/config.guess	2019-01-15 11:43:57.276224400 +0300
@@ -896,6 +896,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-pc-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
diff -Naur tar-1.31-orig/build-aux/config.rpath tar-1.31/build-aux/config.rpath
--- tar-1.31-orig/build-aux/config.rpath	2019-01-02 21:15:56.000000000 +0300
+++ tar-1.31/build-aux/config.rpath	2019-01-15 11:43:57.291824500 +0300
@@ -57,7 +57,7 @@
     aix*)
       wl='-Wl,'
       ;;
-    mingw* | cygwin* | pw32* | os2* | cegcc*)
+    mingw* | cygwin* | msys* | pw32* | os2* | cegcc*)
       ;;
     hpux9* | hpux10* | hpux11*)
       wl='-Wl,'
@@ -149,7 +149,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -198,7 +198,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -348,7 +348,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32* | cegcc*)
+    cygwin* | msys* | mingw* | pw32* | cegcc*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -533,7 +533,7 @@
   bsdi[45]*)
     library_names_spec='$libname$shrext'
     ;;
-  cygwin* | mingw* | pw32* | cegcc*)
+  cygwin* | msys* | mingw* | pw32* | cegcc*)
     shrext=.dll
     library_names_spec='$libname.dll.a $libname.lib'
     ;;
diff -Naur tar-1.31-orig/m4/btowc.m4 tar-1.31/m4/btowc.m4
--- tar-1.31-orig/m4/btowc.m4	2019-01-02 21:16:02.000000000 +0300
+++ tar-1.31/m4/btowc.m4	2019-01-15 11:43:57.307424500 +0300
@@ -49,7 +49,7 @@
 changequote(,)dnl
            case "$host_os" in
                       # Guess no on Cygwin.
-             cygwin*) gl_cv_func_btowc_nul="guessing no" ;;
+             cygwin* | msys*) gl_cv_func_btowc_nul="guessing no" ;;
                       # Guess yes on native Windows.
              mingw*)  gl_cv_func_btowc_nul="guessing yes" ;;
                       # Guess yes otherwise.
diff -Naur tar-1.31-orig/m4/double-slash-root.m4 tar-1.31/m4/double-slash-root.m4
--- tar-1.31-orig/m4/double-slash-root.m4	2019-01-02 21:16:02.000000000 +0300
+++ tar-1.31/m4/double-slash-root.m4	2019-01-15 11:43:57.323024500 +0300
@@ -16,7 +16,7 @@
         # special semantics and is distinct from /, please report it to
         # <bug-gnulib@gnu.org>.
         case $host in
-          *-cygwin | i370-ibm-openedition)
+          *-cygwin | *-msys | i370-ibm-openedition)
             gl_cv_double_slash_root=yes ;;
           *)
             # Be optimistic and assume that / and // are the same when we
diff -Naur tar-1.31-orig/m4/dup2.m4 tar-1.31/m4/dup2.m4
--- tar-1.31-orig/m4/dup2.m4	2019-01-02 21:16:02.000000000 +0300
+++ tar-1.31/m4/dup2.m4	2019-01-15 11:43:57.323024500 +0300
@@ -78,7 +78,7 @@
         [case "$host_os" in
            mingw*) # on this platform, dup2 always returns 0 for success
              gl_cv_func_dup2_works="guessing no" ;;
-           cygwin*) # on cygwin 1.5.x, dup2(1,1) returns 0
+           cygwin* | msys*) # on cygwin 1.5.x, dup2(1,1) returns 0
              gl_cv_func_dup2_works="guessing no" ;;
            aix* | freebsd*)
                    # on AIX 7.1 and FreeBSD 6.1, dup2 (1,toobig) gives EMFILE,
diff -Naur tar-1.31-orig/m4/fcntl.m4 tar-1.31/m4/fcntl.m4
--- tar-1.31-orig/m4/fcntl.m4	2019-01-02 21:16:03.000000000 +0300
+++ tar-1.31/m4/fcntl.m4	2019-01-15 11:43:57.354224600 +0300
@@ -69,7 +69,7 @@
          [gl_cv_func_fcntl_f_dupfd_works=yes],
          [gl_cv_func_fcntl_f_dupfd_works=no],
          [case $host_os in
-            aix* | cygwin* | haiku*)
+            aix* | cygwin* | msys* | haiku*)
                gl_cv_func_fcntl_f_dupfd_works="guessing no" ;;
             *) gl_cv_func_fcntl_f_dupfd_works="guessing yes" ;;
           esac])])
diff -Naur tar-1.31-orig/m4/getcwd.m4 tar-1.31/m4/getcwd.m4
--- tar-1.31-orig/m4/getcwd.m4	2019-01-02 21:16:03.000000000 +0300
+++ tar-1.31/m4/getcwd.m4	2019-01-16 07:49:19.431055700 +0300
@@ -51,7 +51,7 @@
                            # Guess yes on glibc systems.
             *-gnu* | gnu*) gl_cv_func_getcwd_null="guessing yes";;
                            # Guess yes on Cygwin.
-            cygwin*)       gl_cv_func_getcwd_null="guessing yes";;
+            cygwin* | msys*) gl_cv_func_getcwd_null="guessing yes";;
                            # If we don't know, assume the worst.
             *)             gl_cv_func_getcwd_null="guessing no";;
           esac
diff -Naur tar-1.31-orig/m4/getdtablesize.m4 tar-1.31/m4/getdtablesize.m4
--- tar-1.31-orig/m4/getdtablesize.m4	2019-01-02 21:16:03.000000000 +0300
+++ tar-1.31/m4/getdtablesize.m4	2019-01-15 11:43:57.385424600 +0300
@@ -39,7 +39,7 @@
              [gl_cv_func_getdtablesize_works=yes],
              [gl_cv_func_getdtablesize_works=no],
              [case "$host_os" in
-                cygwin*) # on cygwin 1.5.25, getdtablesize() automatically grows
+                cygwin* | msys* ) # on cygwin 1.5.25, getdtablesize() automatically grows
                   gl_cv_func_getdtablesize_works="guessing no" ;;
                 *) gl_cv_func_getdtablesize_works="guessing yes" ;;
               esac
diff -Naur tar-1.31-orig/m4/malloc.m4 tar-1.31/m4/malloc.m4
--- tar-1.31-orig/m4/malloc.m4	2019-01-02 21:16:03.000000000 +0300
+++ tar-1.31/m4/malloc.m4	2019-01-16 07:49:52.916322700 +0300
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_malloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_malloc_0_nonnull=no ;;
diff -Naur tar-1.31-orig/m4/printf.m4 tar-1.31/m4/printf.m4
--- tar-1.31-orig/m4/printf.m4	2019-01-02 21:16:04.000000000 +0300
+++ tar-1.31/m4/printf.m4	2019-01-15 11:43:57.401024700 +0300
@@ -769,7 +769,7 @@
            openbsd*)       gl_cv_func_printf_directive_ls="guessing no";;
            irix*)          gl_cv_func_printf_directive_ls="guessing no";;
            solaris*)       gl_cv_func_printf_directive_ls="guessing no";;
-           cygwin*)        gl_cv_func_printf_directive_ls="guessing no";;
+           cygwin* | msys*) gl_cv_func_printf_directive_ls="guessing no";;
            beos* | haiku*) gl_cv_func_printf_directive_ls="guessing no";;
                            # Guess yes on native Windows.
            mingw*)         gl_cv_func_printf_directive_ls="guessing yes";;
@@ -849,7 +849,7 @@
         [
 changequote(,)dnl
          case "$host_os" in
-           cygwin*)      gl_cv_func_printf_flag_grouping="guessing no";;
+           cygwin* | msys*)      gl_cv_func_printf_flag_grouping="guessing no";;
            netbsd*)      gl_cv_func_printf_flag_grouping="guessing no";;
                          # Guess no on native Windows.
            mingw* | pw*) gl_cv_func_printf_flag_grouping="guessing no";;
@@ -1524,7 +1524,7 @@
            darwin[1-6].*)        gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
            darwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Cygwin.
-           cygwin*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
+           cygwin* | msys*)              gl_cv_func_vsnprintf_zerosize_c99="guessing yes";;
                                  # Guess yes on Solaris >= 2.6.
            solaris2.[0-5] | solaris2.[0-5].*)
                                  gl_cv_func_vsnprintf_zerosize_c99="guessing no";;
diff -Naur tar-1.31-orig/m4/realloc.m4 tar-1.31/m4/realloc.m4
--- tar-1.31-orig/m4/realloc.m4	2019-01-02 21:16:04.000000000 +0300
+++ tar-1.31/m4/realloc.m4	2019-01-16 07:50:20.583779600 +0300
@@ -33,7 +33,7 @@
        [case "$host_os" in
           # Guess yes on platforms where we know the result.
           *-gnu* | gnu* | freebsd* | netbsd* | openbsd* \
-          | hpux* | solaris* | cygwin* | mingw*)
+          | hpux* | solaris* | cygwin* | msys* | mingw*)
             ac_cv_func_realloc_0_nonnull=yes ;;
           # If we don't know, assume the worst.
           *) ac_cv_func_realloc_0_nonnull=no ;;
diff -Naur tar-1.31-orig/m4/unlinkdir.m4 tar-1.31/m4/unlinkdir.m4
--- tar-1.31-orig/m4/unlinkdir.m4	2019-01-02 21:16:05.000000000 +0300
+++ tar-1.31/m4/unlinkdir.m4	2019-01-15 11:43:57.416624700 +0300
@@ -25,6 +25,7 @@
   linux-* | linux | \
   freebsd2.2* | freebsd[[3-9]]* | freebsd[[1-9]][[0-9]]* | \
   cygwin | \
+  msys | \
   mingw*)
     AC_DEFINE([UNLINK_CANNOT_UNLINK_DIR], [1],
       [Define to 1 if unlink (dir) cannot possibly succeed.]);;
