diff -Naur tar-1.27-fpending/gnu/fpending.h tar-1.27/gnu/fpending.h
--- tar-1.27-fpending/gnu/fpending.h	2013-06-29 09:54:24.000000000 +0400
+++ tar-1.27/gnu/fpending.h	2013-10-20 10:37:23.409600000 +0400
@@ -22,11 +22,10 @@
 
 #include <stddef.h>
 #include <stdio.h>
+#if HAVE_STDIO_EXT_H
+# include <stdio_ext.h>
+#endif
 
-#if HAVE_DECL___FPENDING
-# if HAVE_STDIO_EXT_H
-#  include <stdio_ext.h>
-# endif
-#else
+#ifndef __fpending
 size_t __fpending (FILE *);
 #endif
diff -Naur tar-1.27-fpending/m4/fpending.m4 tar-1.27/m4/fpending.m4
--- tar-1.27-fpending/m4/fpending.m4	2013-06-29 09:53:43.000000000 +0400
+++ tar-1.27/m4/fpending.m4	2013-10-20 10:37:59.518000000 +0400
@@ -16,14 +16,19 @@
 AC_DEFUN([gl_FUNC_FPENDING],
 [
   AC_CHECK_HEADERS_ONCE([stdio_ext.h])
-  AC_CHECK_FUNCS_ONCE([__fpending])
-  fp_headers='
-#     include <stdio.h>
-#     if HAVE_STDIO_EXT_H
-#      include <stdio_ext.h>
-#     endif
-'
-  AC_CHECK_DECLS([__fpending], , , $fp_headers)
+  AC_CACHE_CHECK([for __fpending], [gl_cv_func___fpending],
+    [
+      AC_LINK_IFELSE(
+	[AC_LANG_PROGRAM(
+	   [[#include <stdio.h>
+	     #if HAVE_STDIO_EXT_H
+	     # include <stdio_ext.h>
+	     #endif
+	   ]],
+	   [[return ! __fpending (stdin);]])],
+	[gl_cv_func___fpending=yes],
+	[gl_cv_func___fpending=no])
+    ])
 ])
 
 AC_DEFUN([gl_PREREQ_FPENDING],
diff -Naur tar-1.27-fpending/m4/gnulib-comp.m4 tar-1.27/m4/gnulib-comp.m4
--- tar-1.27-fpending/m4/gnulib-comp.m4	2013-06-29 09:54:27.000000000 +0400
+++ tar-1.27/m4/gnulib-comp.m4	2013-10-20 10:38:42.168400000 +0400
@@ -451,7 +451,7 @@
     gl_PREREQ_FNMATCH
   fi
   gl_FUNC_FPENDING
-  if test $ac_cv_func___fpending = no; then
+  if test $gl_cv_func___fpending = no; then
     AC_LIBOBJ([fpending])
     gl_PREREQ_FPENDING
   fi
