diff -Naur Net-SSLeay-1.55/t/local/61_threads-cb-crash.t Net-SSLeay-1.55-patched/t/local/61_threads-cb-crash.t
--- Net-SSLeay-1.55/t/local/61_threads-cb-crash.t	2012-04-06 01:29:23.000000000 +0400
+++ Net-SSLeay-1.55-patched/t/local/61_threads-cb-crash.t	2013-11-14 13:22:49.414800000 +0400
@@ -9,7 +9,7 @@
 };
 
 #XXX-TODO perhaps perl+ithreads related issue (needs more investigation)
-plan skip_all => "this test sometimes crashes on cygwin" if $^O eq 'cygwin';
+plan skip_all => "this test sometimes crashes on cygwin" if $^O eq 'cygwin' or $^O eq 'msys';
 
 # NOTE: expect warnings about threads still running under perl 5.8 and threads 1.71
 plan tests => 1;
diff -Naur Net-SSLeay-1.55/t/local/62_threads-ctx_new-deadlock.t Net-SSLeay-1.55-patched/t/local/62_threads-ctx_new-deadlock.t
--- Net-SSLeay-1.55/t/local/62_threads-ctx_new-deadlock.t	2012-04-06 01:29:23.000000000 +0400
+++ Net-SSLeay-1.55-patched/t/local/62_threads-ctx_new-deadlock.t	2013-11-14 13:23:01.177200000 +0400
@@ -9,7 +9,7 @@
 };
 
 #XXX-TODO perhaps perl+ithreads related issue (needs more investigation)
-plan skip_all => "this test sometimes crashes on cygwin" if $^O eq 'cygwin';
+plan skip_all => "this test sometimes crashes on cygwin" if $^O eq 'cygwin' or $^O eq 'msys';
 
 plan tests => 1;
 
