--- a/rebaseall.in	2013-02-18 14:04:43.000000000 +0400
+++ b/rebaseall.in	2014-02-28 18:44:17.784400000 +0400
@@ -218,6 +218,7 @@
     # some interpreters include a method for installing addons outside of the
     # package manager, such as CPAN and RubyGems.
     for d in /usr/lib/perl5/*_perl /usr/lib/py*/site-packages \
+             /usr/lib/py*/lib-dynload \
              /usr/lib/php /usr/lib/R/site-library /usr/lib/rub*/gems \
              /usr/lib/octave/site
     do
