--- rebase-4.4.1-orig/rebaseall.in	2013-02-18 14:04:43.000000000 +0400
+++ rebase-4.4.1/rebaseall.in	2014-06-24 14:32:45.015200000 +0400
@@ -1,4 +1,4 @@
-#!/bin/@ASH@
+#!/usr/bin/@ASH@
 
 #
 # Copyright (c) 2003, 2005, 2006, 2008, 2011, 2012 Jason Tishler
@@ -134,12 +134,12 @@
   ProcessResult=0
   case $Platform in
     mingw|msys )
-      /bin/ps -s | /bin/gawk '\
+      /usr/bin/ps -s | /usr/bin/gawk '\
 	# Count number of running ash or dash. \
-	/\/bin\/(d)?ash(\.exe)?$/{ ash_cnt++; } \
+	/\/usr\/bin\/(d)?ash(\.exe)?$/{ ash_cnt++; } \
 	# Count number of ps and gawk. \
-	/\/bin\/ps(\.exe)?$/{ cnt++; } \
-	/\/bin\/gawk(\.exe)?$/{ cnt++; } \
+	/\/usr\/bin\/ps(\.exe)?$/{ cnt++; } \
+	/\/usr\/bin\/gawk(\.exe)?$/{ cnt++; } \
 	END{ \
 	  # Uncomment for testing: \
 	  # printf "TOTAL: %d CNT: %d ASH_CNT: %d\n", NR, cnt, ash_cnt; \
@@ -165,7 +165,7 @@
       echo "${ProgramName}: only ash or dash processes are allowed during rebasing"
       echo "    Exit all Cygwin processes and stop all Cygwin services."
       echo "    Execute ash (or dash) from Start/Run... or a cmd or command window."
-      echo "    Execute '/bin/rebaseall' from ash (or dash)."
+      echo "    Execute '/usr/bin/rebaseall' from ash (or dash)."
       exit 2
   fi
 fi
@@ -231,7 +231,7 @@
     NoDyn='-n'
     ;;
   mingw|msys)
-    for f in /bin /lib
+    for f in /usr/bin /usr/lib
     do
       find $f -type f |
         grep -E "\.($Suffixes)\$" |
--- rebase-4.4.1-orig/peflagsall.in	2012-04-30 17:37:15.000000000 +0400
+++ rebase-4.4.1/peflagsall.in	2014-06-24 14:32:39.617600000 +0400
@@ -1,4 +1,4 @@
-#!/bin/@ASH@
+#!/usr/bin/@ASH@
 
 #
 # Copyright (c) 2009,2011 Charles Wilson
@@ -153,12 +153,12 @@
 ProcessResult=0
 case $Platform in
   mingw|msys )
-    /bin/ps -s | /bin/gawk '\
+    /usr/bin/ps -s | /usr/bin/gawk '\
       # Count number of running ash or dash. \
-      /\/bin\/(d)?ash(\.exe)?$/{ ash_cnt++; } \
+      /\/usr\/bin\/(d)?ash(\.exe)?$/{ ash_cnt++; } \
       # Count number of ps and gawk. \
-      /\/bin\/ps(\.exe)?$/{ cnt++; } \
-      /\/bin\/gawk(\.exe)?$/{ cnt++; } \
+      /\/usr\/bin\/ps(\.exe)?$/{ cnt++; } \
+      /\/usr\/bin\/gawk(\.exe)?$/{ cnt++; } \
       END{ \
         # Uncomment for testing: \
         # printf "TOTAL: %d CNT: %d ASH_CNT: %d\n", NR, cnt, ash_cnt; \
@@ -184,7 +184,7 @@
     echo "$ProgramName: only ash or dash processes are allowed during this process."
     echo "    Exit all Cygwin processes and stop all Cygwin services."
     echo "    Execute ash (or dash) from Start/Run... or a cmd or command window."
-    echo "    Execute '/bin/peflagsall' from ash (or dash)."
+    echo "    Execute '/usr/bin/peflagsall' from ash (or dash)."
     exit 2
 fi
 
@@ -264,7 +264,7 @@
           -e '/d?ash\.exe$/d' -e '/peflags\.exe$/d' >"${TmpFile}"
     ;;
   msys)
-    for f in /bin /lib
+    for f in /usr/bin /usr/lib
     do
       find $f -type f |
         grep -E "\.($Suffixes)\$" |
