From d5098f3924649936b561b5a2a4bdb7fd34c9736a Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 14 Feb 2017 14:45:21 +0300
Subject: [PATCH 1/7] perl.cygwin-auto-image-base

---
 Makefile.SH | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.SH b/Makefile.SH
index efeb8d6..e2490da 100755
--- a/Makefile.SH
+++ b/Makefile.SH
@@ -73,7 +73,7 @@ true)
 		esac
 		;;
 	cygwin*)
-		shrpldflags="$shrpldflags -Wl,--out-implib=libperl.dll.a -Wl,--image-base,0x52000000"
+		shrpldflags="$shrpldflags -Wl,--out-implib=libperl.dll.a"
 		linklibperl="-L. -lperl"
 		;;
 	sunos*)
-- 
2.37.1

