From 7b14270505bd2c7c73d64cfaef9314a900b1e910 Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 14 Feb 2017 14:45:21 +0300
Subject: [PATCH 01/10] perl.cygwin-hints

---
 hints/cygwin.sh | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/hints/cygwin.sh b/hints/cygwin.sh
index a4d390e..158e89d 100644
--- a/hints/cygwin.sh
+++ b/hints/cygwin.sh
@@ -25,13 +25,13 @@ libswanted=`echo " $libswanted " | sed -e 's/ c / /g'`
 libswanted=`echo " $libswanted " | sed -e 's/ m / /g'`
 # - eliminate -lutil, symbols are all in libcygwin.a
 libswanted=`echo " $libswanted " | sed -e 's/ util / /g'`
-test -z "$ignore_versioned_solibs" && ignore_versioned_solibs='y'
-test -z "$usenm" && usenm='no'
-test -z "$libc" && libc='/usr/lib/libcygwin.a'
-test -z "$loclibpth" && loclibpth=' '
-test -z "$glibpth" && glibpth=' '
-test -z "$plibpth" && plibpth=' '
-test -z "$libpth" && libpth=' '
+ignore_versioned_solibs='y'
+usenm='no'
+libc='/usr/lib/libcygwin.a'
+loclibpth=' '
+glibpth=' '
+plibpth=' '
+libpth=' '
 PATH='.:/usr/bin/'
 # - add libgdbm_compat $libswanted
 libswanted="$libswanted gdbm_compat"
