From 53179b6ddb0be481062410ef7e80721dffcf11fc Mon Sep 17 00:00:00 2001
From: Alexey Pavlov <alexpux@gmail.com>
Date: Tue, 14 Feb 2017 14:45:21 +0300
Subject: [PATCH 2/7] perl.cygwin-hints

---
 hints/cygwin.sh | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/hints/cygwin.sh b/hints/cygwin.sh
index 40edf5f..bda3da8 100644
--- a/hints/cygwin.sh
+++ b/hints/cygwin.sh
@@ -25,13 +25,13 @@ libswanted=`echo " $libswanted " | sed -e 's/ c / /g'`
 libswanted=`echo " $libswanted " | sed -e 's/ m / /g'`
 # - eliminate -lutil, symbols are all in libcygwin.a
 libswanted=`echo " $libswanted " | sed -e 's/ util / /g'`
-test -z "$ignore_versioned_solibs" && ignore_versioned_solibs='y'
-test -z "$usenm" && usenm='no'
-test -z "$libc" && libc='/usr/lib/libcygwin.a'
-test -z "$loclibpth" && loclibpth=' '
-test -z "$glibpth" && glibpth=' '
-test -z "$plibpth" && plibpth=' '
-test -z "$libpth" && libpth=' '
+ignore_versioned_solibs='y'
+usenm='no'
+libc='/usr/lib/libcygwin.a'
+loclibpth=' '
+glibpth=' '
+plibpth=' '
+libpth=' '
 PATH='.:/usr/bin/'
 # - add libgdbm_compat $libswanted
 libswanted="$libswanted gdbm_compat"
@@ -62,6 +71,13 @@ case "$osvers" in
         d_inetpton='undef'
 esac
 
+case "$osvers" in
+    2.[0-4].*|1.*)
+        # newlib finitel is buggy before cygwin-2.5.0
+        d_finitel='undef'
+        ;;
+esac
+
 # compile Win32CORE "module" as static. try to avoid the space.
 if test -z "$static_ext"; then
   static_ext="Win32CORE"
-- 
2.37.1

