diff --git a/perly.act b/perly.act
index fe8a5e9..c6df745 100644
--- a/perly.act
+++ b/perly.act
@@ -5,7 +5,7 @@
  */
 
 case 2:
-#line 119 "perly.y" /* yacc.c:1646  */
+#line 119 "perly.y" /* yacc.c:1648  */
     {
 			  parser->expect = XSTATE;
                           (yyval.ival) = 0;
@@ -14,7 +14,7 @@ case 2:
     break;
 
   case 3:
-#line 124 "perly.y" /* yacc.c:1646  */
+#line 124 "perly.y" /* yacc.c:1648  */
     {
 			  newPROG(block_end((ps[-1].val.ival),(ps[0].val.opval)));
 			  PL_compiling.cop_seq = 0;
@@ -24,7 +24,7 @@ case 2:
     break;
 
   case 4:
-#line 130 "perly.y" /* yacc.c:1646  */
+#line 130 "perly.y" /* yacc.c:1648  */
     {
 			  parser->expect = XTERM;
                           (yyval.ival) = 0;
@@ -33,7 +33,7 @@ case 2:
     break;
 
   case 5:
-#line 135 "perly.y" /* yacc.c:1646  */
+#line 135 "perly.y" /* yacc.c:1648  */
     {
 			  PL_eval_root = (ps[0].val.opval);
 			  (yyval.ival) = 0;
@@ -42,7 +42,7 @@ case 2:
     break;
 
   case 6:
-#line 140 "perly.y" /* yacc.c:1646  */
+#line 140 "perly.y" /* yacc.c:1648  */
     {
 			  parser->expect = XBLOCK;
                           (yyval.ival) = 0;
@@ -51,7 +51,7 @@ case 2:
     break;
 
   case 7:
-#line 145 "perly.y" /* yacc.c:1646  */
+#line 145 "perly.y" /* yacc.c:1648  */
     {
 			  PL_pad_reset_pending = TRUE;
 			  PL_eval_root = (ps[0].val.opval);
@@ -63,7 +63,7 @@ case 2:
     break;
 
   case 8:
-#line 153 "perly.y" /* yacc.c:1646  */
+#line 153 "perly.y" /* yacc.c:1648  */
     {
 			  parser->expect = XSTATE;
                           (yyval.ival) = 0;
@@ -72,7 +72,7 @@ case 2:
     break;
 
   case 9:
-#line 158 "perly.y" /* yacc.c:1646  */
+#line 158 "perly.y" /* yacc.c:1648  */
     {
 			  PL_pad_reset_pending = TRUE;
 			  PL_eval_root = (ps[0].val.opval);
@@ -84,7 +84,7 @@ case 2:
     break;
 
   case 10:
-#line 166 "perly.y" /* yacc.c:1646  */
+#line 166 "perly.y" /* yacc.c:1648  */
     {
 			  parser->expect = XSTATE;
                           (yyval.ival) = 0;
@@ -93,7 +93,7 @@ case 2:
     break;
 
   case 11:
-#line 171 "perly.y" /* yacc.c:1646  */
+#line 171 "perly.y" /* yacc.c:1648  */
     {
 			  PL_pad_reset_pending = TRUE;
 			  PL_eval_root = (ps[0].val.opval);
@@ -105,7 +105,7 @@ case 2:
     break;
 
   case 12:
-#line 179 "perly.y" /* yacc.c:1646  */
+#line 179 "perly.y" /* yacc.c:1648  */
     {
 			  parser->expect = XSTATE;
                           (yyval.ival) = 0;
@@ -114,7 +114,7 @@ case 2:
     break;
 
   case 13:
-#line 184 "perly.y" /* yacc.c:1646  */
+#line 184 "perly.y" /* yacc.c:1648  */
     {
 			  PL_eval_root = (ps[0].val.opval);
 			  (yyval.ival) = 0;
@@ -123,7 +123,7 @@ case 2:
     break;
 
   case 14:
-#line 192 "perly.y" /* yacc.c:1646  */
+#line 192 "perly.y" /* yacc.c:1648  */
     { if (parser->copline > (line_t)(ps[-3].val.ival))
 			      parser->copline = (line_t)(ps[-3].val.ival);
 			  (yyval.opval) = block_end((ps[-2].val.ival), (ps[-1].val.opval));
@@ -132,7 +132,7 @@ case 2:
     break;
 
   case 15:
-#line 200 "perly.y" /* yacc.c:1646  */
+#line 200 "perly.y" /* yacc.c:1648  */
     { if (parser->copline > (line_t)(ps[-6].val.ival))
 			      parser->copline = (line_t)(ps[-6].val.ival);
 			  (yyval.opval) = block_end((ps[-5].val.ival), (ps[-2].val.opval));
@@ -141,14 +141,14 @@ case 2:
     break;
 
   case 16:
-#line 207 "perly.y" /* yacc.c:1646  */
+#line 207 "perly.y" /* yacc.c:1648  */
     { (yyval.ival) = block_start(TRUE);
 			  parser->parsed_sub = 0; }
 
     break;
 
   case 17:
-#line 212 "perly.y" /* yacc.c:1646  */
+#line 212 "perly.y" /* yacc.c:1648  */
     { if (parser->copline > (line_t)(ps[-3].val.ival))
 			      parser->copline = (line_t)(ps[-3].val.ival);
 			  (yyval.opval) = block_end((ps[-2].val.ival), (ps[-1].val.opval));
@@ -157,20 +157,20 @@ case 2:
     break;
 
   case 18:
-#line 219 "perly.y" /* yacc.c:1646  */
+#line 219 "perly.y" /* yacc.c:1648  */
     { (yyval.ival) = block_start(FALSE);
 			  parser->parsed_sub = 0; }
 
     break;
 
   case 19:
-#line 225 "perly.y" /* yacc.c:1646  */
+#line 225 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 20:
-#line 227 "perly.y" /* yacc.c:1646  */
+#line 227 "perly.y" /* yacc.c:1648  */
     {   (yyval.opval) = op_append_list(OP_LINESEQ, (ps[-1].val.opval), (ps[0].val.opval));
 			    PL_pad_reset_pending = TRUE;
 			    if ((ps[-1].val.opval) && (ps[0].val.opval))
@@ -180,13 +180,13 @@ case 2:
     break;
 
   case 21:
-#line 236 "perly.y" /* yacc.c:1646  */
+#line 236 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 22:
-#line 238 "perly.y" /* yacc.c:1646  */
+#line 238 "perly.y" /* yacc.c:1648  */
     {   (yyval.opval) = op_append_list(OP_LINESEQ, (ps[-1].val.opval), (ps[0].val.opval));
 			    PL_pad_reset_pending = TRUE;
 			    if ((ps[-1].val.opval) && (ps[0].val.opval))
@@ -196,7 +196,7 @@ case 2:
     break;
 
   case 23:
-#line 247 "perly.y" /* yacc.c:1646  */
+#line 247 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = (ps[0].val.opval) ? newSTATEOP(0, NULL, (ps[0].val.opval)) : NULL;
 			}
@@ -204,13 +204,13 @@ case 2:
     break;
 
   case 24:
-#line 251 "perly.y" /* yacc.c:1646  */
+#line 251 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 25:
-#line 255 "perly.y" /* yacc.c:1646  */
+#line 255 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = newSTATEOP(SVf_UTF8 * (ps[-1].val.pval)[strlen((ps[-1].val.pval))+1], (ps[-1].val.pval), (ps[0].val.opval));
 			}
@@ -218,7 +218,7 @@ case 2:
     break;
 
   case 26:
-#line 259 "perly.y" /* yacc.c:1646  */
+#line 259 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = newSTATEOP(SVf_UTF8 * (ps[-1].val.pval)[strlen((ps[-1].val.pval))+1], (ps[-1].val.pval), (ps[0].val.opval));
 			}
@@ -226,13 +226,13 @@ case 2:
     break;
 
   case 27:
-#line 266 "perly.y" /* yacc.c:1646  */
+#line 266 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 28:
-#line 268 "perly.y" /* yacc.c:1646  */
+#line 268 "perly.y" /* yacc.c:1648  */
     {
 			  CV *fmtcv = PL_compcv;
 			  newFORM((ps[-2].val.ival), (ps[-1].val.opval), (ps[0].val.opval));
@@ -246,7 +246,7 @@ case 2:
     break;
 
   case 29:
-#line 280 "perly.y" /* yacc.c:1646  */
+#line 280 "perly.y" /* yacc.c:1648  */
     {
                           init_named_cv(PL_compcv, (ps[-1].val.opval));
 			  parser->in_my = 0;
@@ -256,7 +256,7 @@ case 2:
     break;
 
   case 30:
-#line 286 "perly.y" /* yacc.c:1646  */
+#line 286 "perly.y" /* yacc.c:1648  */
     {
 			  SvREFCNT_inc_simple_void(PL_compcv);
 			  (ps[-5].val.opval)->op_type == OP_CONST
@@ -271,7 +271,7 @@ case 2:
     break;
 
   case 31:
-#line 301 "perly.y" /* yacc.c:1646  */
+#line 301 "perly.y" /* yacc.c:1648  */
     {
                           init_named_cv(PL_compcv, (ps[-1].val.opval));
 			  parser->in_my = 0;
@@ -281,7 +281,7 @@ case 2:
     break;
 
   case 32:
-#line 307 "perly.y" /* yacc.c:1646  */
+#line 307 "perly.y" /* yacc.c:1648  */
     {
 			  SvREFCNT_inc_simple_void(PL_compcv);
 			  (ps[-4].val.opval)->op_type == OP_CONST
@@ -296,7 +296,7 @@ case 2:
     break;
 
   case 33:
-#line 318 "perly.y" /* yacc.c:1646  */
+#line 318 "perly.y" /* yacc.c:1648  */
     {
 			  package((ps[-1].val.opval));
 			  if ((ps[-2].val.opval))
@@ -307,13 +307,13 @@ case 2:
     break;
 
   case 34:
-#line 325 "perly.y" /* yacc.c:1646  */
+#line 325 "perly.y" /* yacc.c:1648  */
     { CvSPECIAL_on(PL_compcv); /* It's a BEGIN {} */ }
 
     break;
 
   case 35:
-#line 327 "perly.y" /* yacc.c:1646  */
+#line 327 "perly.y" /* yacc.c:1648  */
     {
 			  SvREFCNT_inc_simple_void(PL_compcv);
 			  utilize((ps[-6].val.ival), (ps[-5].val.ival), (ps[-3].val.opval), (ps[-2].val.opval), (ps[-1].val.opval));
@@ -324,7 +324,7 @@ case 2:
     break;
 
   case 36:
-#line 334 "perly.y" /* yacc.c:1646  */
+#line 334 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-4].val.ival),
 			      newCONDOP(0, (ps[-3].val.opval), op_scope((ps[-1].val.opval)), (ps[0].val.opval)));
@@ -334,7 +334,7 @@ case 2:
     break;
 
   case 37:
-#line 340 "perly.y" /* yacc.c:1646  */
+#line 340 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-4].val.ival),
                               newCONDOP(0, (ps[-3].val.opval), (ps[0].val.opval), op_scope((ps[-1].val.opval))));
@@ -344,7 +344,7 @@ case 2:
     break;
 
   case 38:
-#line 346 "perly.y" /* yacc.c:1646  */
+#line 346 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-3].val.ival), newGIVENOP((ps[-2].val.opval), op_scope((ps[0].val.opval)), 0));
 			  parser->copline = (line_t)(ps[-5].val.ival);
@@ -353,19 +353,19 @@ case 2:
     break;
 
   case 39:
-#line 351 "perly.y" /* yacc.c:1646  */
+#line 351 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = block_end((ps[-3].val.ival), newWHENOP((ps[-2].val.opval), op_scope((ps[0].val.opval)))); }
 
     break;
 
   case 40:
-#line 353 "perly.y" /* yacc.c:1646  */
+#line 353 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newWHENOP(0, op_scope((ps[0].val.opval))); }
 
     break;
 
   case 41:
-#line 355 "perly.y" /* yacc.c:1646  */
+#line 355 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-5].val.ival),
 				  newWHILEOP(0, 1, NULL,
@@ -376,7 +376,7 @@ case 2:
     break;
 
   case 42:
-#line 362 "perly.y" /* yacc.c:1646  */
+#line 362 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-5].val.ival),
 				  newWHILEOP(0, 1, NULL,
@@ -387,19 +387,19 @@ case 2:
     break;
 
   case 43:
-#line 369 "perly.y" /* yacc.c:1646  */
+#line 369 "perly.y" /* yacc.c:1648  */
     { parser->expect = XTERM; }
 
     break;
 
   case 44:
-#line 371 "perly.y" /* yacc.c:1646  */
+#line 371 "perly.y" /* yacc.c:1648  */
     { parser->expect = XTERM; }
 
     break;
 
   case 45:
-#line 374 "perly.y" /* yacc.c:1646  */
+#line 374 "perly.y" /* yacc.c:1648  */
     {
 			  OP *initop = (ps[-9].val.opval);
 			  OP *forop = newWHILEOP(0, 1, NULL,
@@ -418,7 +418,7 @@ case 2:
     break;
 
   case 46:
-#line 389 "perly.y" /* yacc.c:1646  */
+#line 389 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-6].val.ival), newFOROP(0, (ps[-5].val.opval), (ps[-3].val.opval), (ps[-1].val.opval), (ps[0].val.opval)));
 			  parser->copline = (line_t)(ps[-8].val.ival);
@@ -427,7 +427,7 @@ case 2:
     break;
 
   case 47:
-#line 394 "perly.y" /* yacc.c:1646  */
+#line 394 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-4].val.ival), newFOROP(0,
 				      op_lvalue((ps[-6].val.opval), OP_ENTERLOOP), (ps[-3].val.opval), (ps[-1].val.opval), (ps[0].val.opval)));
@@ -437,13 +437,13 @@ case 2:
     break;
 
   case 48:
-#line 400 "perly.y" /* yacc.c:1646  */
+#line 400 "perly.y" /* yacc.c:1648  */
     { parser->in_my = 0; (yyval.opval) = my((ps[0].val.opval)); }
 
     break;
 
   case 49:
-#line 402 "perly.y" /* yacc.c:1646  */
+#line 402 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end(
 				(ps[-7].val.ival),
@@ -460,7 +460,7 @@ case 2:
     break;
 
   case 50:
-#line 415 "perly.y" /* yacc.c:1646  */
+#line 415 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-4].val.ival), newFOROP(
 				0, op_lvalue(newUNOP(OP_REFGEN, 0,
@@ -472,7 +472,7 @@ case 2:
     break;
 
   case 51:
-#line 423 "perly.y" /* yacc.c:1646  */
+#line 423 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = block_end((ps[-4].val.ival),
 				  newFOROP(0, NULL, (ps[-3].val.opval), (ps[-1].val.opval), (ps[0].val.opval)));
@@ -482,7 +482,7 @@ case 2:
     break;
 
   case 52:
-#line 429 "perly.y" /* yacc.c:1646  */
+#line 429 "perly.y" /* yacc.c:1648  */
     {
 			  /* a block is a loop that happens once */
 			  (yyval.opval) = newWHILEOP(0, 1, NULL,
@@ -492,7 +492,7 @@ case 2:
     break;
 
   case 53:
-#line 435 "perly.y" /* yacc.c:1646  */
+#line 435 "perly.y" /* yacc.c:1648  */
     {
 			  package((ps[-2].val.opval));
 			  if ((ps[-3].val.opval)) {
@@ -503,7 +503,7 @@ case 2:
     break;
 
   case 54:
-#line 442 "perly.y" /* yacc.c:1646  */
+#line 442 "perly.y" /* yacc.c:1648  */
     {
 			  /* a block is a loop that happens once */
 			  (yyval.opval) = newWHILEOP(0, 1, NULL,
@@ -515,7 +515,7 @@ case 2:
     break;
 
   case 55:
-#line 450 "perly.y" /* yacc.c:1646  */
+#line 450 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = (ps[-1].val.opval);
 			}
@@ -523,7 +523,7 @@ case 2:
     break;
 
   case 56:
-#line 454 "perly.y" /* yacc.c:1646  */
+#line 454 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = newLISTOP(OP_DIE, 0, newOP(OP_PUSHMARK, 0),
 				newSVOP(OP_CONST, 0, newSVpvs("Unimplemented")));
@@ -532,7 +532,7 @@ case 2:
     break;
 
   case 57:
-#line 459 "perly.y" /* yacc.c:1646  */
+#line 459 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = NULL;
 			  parser->copline = NOLINE;
@@ -541,7 +541,7 @@ case 2:
     break;
 
   case 58:
-#line 467 "perly.y" /* yacc.c:1646  */
+#line 467 "perly.y" /* yacc.c:1648  */
     { OP *list;
 			  if ((ps[0].val.opval)) {
 			      OP *term = (ps[0].val.opval);
@@ -560,74 +560,74 @@ case 2:
     break;
 
   case 59:
-#line 484 "perly.y" /* yacc.c:1646  */
+#line 484 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 60:
-#line 486 "perly.y" /* yacc.c:1646  */
+#line 486 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_unscope((ps[-1].val.opval)); }
 
     break;
 
   case 61:
-#line 491 "perly.y" /* yacc.c:1646  */
+#line 491 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 62:
-#line 493 "perly.y" /* yacc.c:1646  */
+#line 493 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 63:
-#line 495 "perly.y" /* yacc.c:1646  */
+#line 495 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOGOP(OP_AND, 0, (ps[0].val.opval), (ps[-2].val.opval)); }
 
     break;
 
   case 64:
-#line 497 "perly.y" /* yacc.c:1646  */
+#line 497 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOGOP(OP_OR, 0, (ps[0].val.opval), (ps[-2].val.opval)); }
 
     break;
 
   case 65:
-#line 499 "perly.y" /* yacc.c:1646  */
+#line 499 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOOPOP(OPf_PARENS, 1, scalar((ps[0].val.opval)), (ps[-2].val.opval)); }
 
     break;
 
   case 66:
-#line 501 "perly.y" /* yacc.c:1646  */
+#line 501 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOOPOP(OPf_PARENS, 1, (ps[0].val.opval), (ps[-2].val.opval)); }
 
     break;
 
   case 67:
-#line 503 "perly.y" /* yacc.c:1646  */
+#line 503 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newFOROP(0, NULL, (ps[0].val.opval), (ps[-2].val.opval), NULL);
 			  parser->copline = (line_t)(ps[-1].val.ival); }
 
     break;
 
   case 68:
-#line 506 "perly.y" /* yacc.c:1646  */
+#line 506 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newWHENOP((ps[0].val.opval), op_scope((ps[-2].val.opval))); }
 
     break;
 
   case 69:
-#line 511 "perly.y" /* yacc.c:1646  */
+#line 511 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 70:
-#line 513 "perly.y" /* yacc.c:1646  */
+#line 513 "perly.y" /* yacc.c:1648  */
     {
 			  ((ps[0].val.opval))->op_flags |= OPf_PARENS;
 			  (yyval.opval) = op_scope((ps[0].val.opval));
@@ -636,7 +636,7 @@ case 2:
     break;
 
   case 71:
-#line 518 "perly.y" /* yacc.c:1646  */
+#line 518 "perly.y" /* yacc.c:1648  */
     { parser->copline = (line_t)(ps[-5].val.ival);
 			    (yyval.opval) = newCONDOP(0,
 				newSTATEOP(OPf_SPECIAL,NULL,(ps[-3].val.opval)),
@@ -647,19 +647,19 @@ case 2:
     break;
 
   case 72:
-#line 528 "perly.y" /* yacc.c:1646  */
+#line 528 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 73:
-#line 530 "perly.y" /* yacc.c:1646  */
+#line 530 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_scope((ps[0].val.opval)); }
 
     break;
 
   case 74:
-#line 535 "perly.y" /* yacc.c:1646  */
+#line 535 "perly.y" /* yacc.c:1648  */
     { (yyval.ival) = (PL_min_intro_pending &&
 			    PL_max_intro_pending >=  PL_min_intro_pending);
 			  intro_my(); }
@@ -667,13 +667,13 @@ case 2:
     break;
 
   case 75:
-#line 541 "perly.y" /* yacc.c:1646  */
+#line 541 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 77:
-#line 547 "perly.y" /* yacc.c:1646  */
+#line 547 "perly.y" /* yacc.c:1648  */
     { YYSTYPE tmplval;
 			  (void)scan_num("1", &tmplval);
 			  (yyval.opval) = tmplval.opval; }
@@ -681,118 +681,118 @@ case 2:
     break;
 
   case 79:
-#line 555 "perly.y" /* yacc.c:1646  */
+#line 555 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = invert(scalar((ps[0].val.opval))); }
 
     break;
 
   case 80:
-#line 560 "perly.y" /* yacc.c:1646  */
+#line 560 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); intro_my(); }
 
     break;
 
   case 81:
-#line 564 "perly.y" /* yacc.c:1646  */
+#line 564 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); intro_my(); }
 
     break;
 
   case 82:
-#line 567 "perly.y" /* yacc.c:1646  */
+#line 567 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 83:
-#line 568 "perly.y" /* yacc.c:1646  */
+#line 568 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 84:
-#line 572 "perly.y" /* yacc.c:1646  */
+#line 572 "perly.y" /* yacc.c:1648  */
     { (yyval.ival) = start_subparse(FALSE, 0);
 			    SAVEFREESV(PL_compcv); }
 
     break;
 
   case 85:
-#line 578 "perly.y" /* yacc.c:1646  */
+#line 578 "perly.y" /* yacc.c:1648  */
     { (yyval.ival) = start_subparse(FALSE, CVf_ANON);
 			    SAVEFREESV(PL_compcv); }
 
     break;
 
   case 86:
-#line 583 "perly.y" /* yacc.c:1646  */
+#line 583 "perly.y" /* yacc.c:1648  */
     { (yyval.ival) = start_subparse(TRUE, 0);
 			    SAVEFREESV(PL_compcv); }
 
     break;
 
   case 89:
-#line 594 "perly.y" /* yacc.c:1646  */
+#line 594 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 91:
-#line 600 "perly.y" /* yacc.c:1646  */
+#line 600 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 92:
-#line 602 "perly.y" /* yacc.c:1646  */
+#line 602 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 93:
-#line 604 "perly.y" /* yacc.c:1646  */
+#line 604 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 94:
-#line 609 "perly.y" /* yacc.c:1646  */
+#line 609 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 95:
-#line 611 "perly.y" /* yacc.c:1646  */
+#line 611 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 96:
-#line 622 "perly.y" /* yacc.c:1646  */
+#line 622 "perly.y" /* yacc.c:1648  */
     { parser->in_my = 0; (yyval.opval) = NULL; }
 
     break;
 
   case 97:
-#line 624 "perly.y" /* yacc.c:1646  */
+#line 624 "perly.y" /* yacc.c:1648  */
     { parser->in_my = 0; (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 98:
-#line 629 "perly.y" /* yacc.c:1646  */
+#line 629 "perly.y" /* yacc.c:1648  */
     { (yyval.ival) = '@'; }
 
     break;
 
   case 99:
-#line 631 "perly.y" /* yacc.c:1646  */
+#line 631 "perly.y" /* yacc.c:1648  */
     { (yyval.ival) = '%'; }
 
     break;
 
   case 100:
-#line 635 "perly.y" /* yacc.c:1646  */
+#line 635 "perly.y" /* yacc.c:1648  */
     {
                             I32 sigil   = (ps[-2].val.ival);
                             OP *var     = (ps[-1].val.opval);
@@ -812,25 +812,25 @@ case 2:
     break;
 
   case 101:
-#line 654 "perly.y" /* yacc.c:1646  */
+#line 654 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 102:
-#line 656 "perly.y" /* yacc.c:1646  */
+#line 656 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newOP(OP_NULL, 0); }
 
     break;
 
   case 103:
-#line 658 "perly.y" /* yacc.c:1646  */
+#line 658 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 104:
-#line 664 "perly.y" /* yacc.c:1646  */
+#line 664 "perly.y" /* yacc.c:1648  */
     {
                             OP *var     = (ps[-1].val.opval);
                             OP *defexpr = (ps[0].val.opval);
@@ -852,8 +852,8 @@ case 2:
                                                     "lacks default expression");
                                     op_free(defexpr);
                                 }
-                                else { 
-                                    /* a normal '=default' expression */ 
+                                else {
+                                    /* a normal '=default' expression */
                                     OP *defop = (OP*)alloc_LOGOP(OP_ARGDEFELEM,
                                                         defexpr,
                                                         LINKLIST(defexpr));
@@ -895,25 +895,25 @@ case 2:
     break;
 
   case 105:
-#line 729 "perly.y" /* yacc.c:1646  */
+#line 729 "perly.y" /* yacc.c:1648  */
     { parser->in_my = KEY_sigvar; (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 106:
-#line 731 "perly.y" /* yacc.c:1646  */
+#line 731 "perly.y" /* yacc.c:1648  */
     { parser->in_my = KEY_sigvar; (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 107:
-#line 737 "perly.y" /* yacc.c:1646  */
+#line 737 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[-1].val.opval); }
 
     break;
 
   case 108:
-#line 739 "perly.y" /* yacc.c:1646  */
+#line 739 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = op_append_list(OP_LINESEQ, (ps[-2].val.opval), (ps[0].val.opval));
 			}
@@ -921,37 +921,37 @@ case 2:
     break;
 
   case 109:
-#line 743 "perly.y" /* yacc.c:1646  */
+#line 743 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 110:
-#line 748 "perly.y" /* yacc.c:1646  */
+#line 748 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 111:
-#line 750 "perly.y" /* yacc.c:1646  */
+#line 750 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 112:
-#line 754 "perly.y" /* yacc.c:1646  */
+#line 754 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 113:
-#line 756 "perly.y" /* yacc.c:1646  */
+#line 756 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 114:
-#line 760 "perly.y" /* yacc.c:1646  */
+#line 760 "perly.y" /* yacc.c:1648  */
     {
                             ENTER;
                             SAVEIV(parser->sig_elems);
@@ -966,7 +966,7 @@ case 2:
     break;
 
   case 115:
-#line 772 "perly.y" /* yacc.c:1646  */
+#line 772 "perly.y" /* yacc.c:1648  */
     {
                             OP            *sigops = (ps[-1].val.opval);
                             UNOP_AUX_item *aux;
@@ -1014,19 +1014,19 @@ case 2:
     break;
 
   case 116:
-#line 818 "perly.y" /* yacc.c:1646  */
+#line 818 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 117:
-#line 819 "perly.y" /* yacc.c:1646  */
+#line 819 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 118:
-#line 825 "perly.y" /* yacc.c:1646  */
+#line 825 "perly.y" /* yacc.c:1648  */
     {
 			  if (parser->copline > (line_t)(ps[-2].val.ival))
 			      parser->copline = (line_t)(ps[-2].val.ival);
@@ -1036,19 +1036,19 @@ case 2:
     break;
 
   case 119:
-#line 835 "perly.y" /* yacc.c:1646  */
+#line 835 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 120:
-#line 836 "perly.y" /* yacc.c:1646  */
+#line 836 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 121:
-#line 840 "perly.y" /* yacc.c:1646  */
+#line 840 "perly.y" /* yacc.c:1648  */
     {
 			  if (parser->copline > (line_t)(ps[-2].val.ival))
 			      parser->copline = (line_t)(ps[-2].val.ival);
@@ -1059,31 +1059,31 @@ case 2:
     break;
 
   case 122:
-#line 851 "perly.y" /* yacc.c:1646  */
+#line 851 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOGOP(OP_AND, 0, (ps[-2].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 123:
-#line 853 "perly.y" /* yacc.c:1646  */
+#line 853 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOGOP((ps[-1].val.ival), 0, (ps[-2].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 124:
-#line 855 "perly.y" /* yacc.c:1646  */
+#line 855 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOGOP(OP_DOR, 0, (ps[-2].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 126:
-#line 861 "perly.y" /* yacc.c:1646  */
+#line 861 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[-1].val.opval); }
 
     break;
 
   case 127:
-#line 863 "perly.y" /* yacc.c:1646  */
+#line 863 "perly.y" /* yacc.c:1648  */
     {
 			  OP* term = (ps[0].val.opval);
 			  (yyval.opval) = op_append_elem(OP_LIST, (ps[-2].val.opval), term);
@@ -1092,7 +1092,7 @@ case 2:
     break;
 
   case 129:
-#line 872 "perly.y" /* yacc.c:1646  */
+#line 872 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list((ps[-2].val.ival), OPf_STACKED,
 				op_prepend_elem(OP_LIST, newGVREF((ps[-2].val.ival),(ps[-1].val.opval)), (ps[0].val.opval)) );
 			}
@@ -1100,7 +1100,7 @@ case 2:
     break;
 
   case 130:
-#line 876 "perly.y" /* yacc.c:1646  */
+#line 876 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list((ps[-4].val.ival), OPf_STACKED,
 				op_prepend_elem(OP_LIST, newGVREF((ps[-4].val.ival),(ps[-2].val.opval)), (ps[-1].val.opval)) );
 			}
@@ -1108,7 +1108,7 @@ case 2:
     break;
 
   case 131:
-#line 880 "perly.y" /* yacc.c:1646  */
+#line 880 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list(OP_ENTERSUB, OPf_STACKED,
 				op_append_elem(OP_LIST,
 				    op_prepend_elem(OP_LIST, scalar((ps[-5].val.opval)), (ps[-1].val.opval)),
@@ -1118,7 +1118,7 @@ case 2:
     break;
 
   case 132:
-#line 886 "perly.y" /* yacc.c:1646  */
+#line 886 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list(OP_ENTERSUB, OPf_STACKED,
 				op_append_elem(OP_LIST, scalar((ps[-2].val.opval)),
 				    newMETHOP(OP_METHOD, 0, (ps[0].val.opval))));
@@ -1127,7 +1127,7 @@ case 2:
     break;
 
   case 133:
-#line 891 "perly.y" /* yacc.c:1646  */
+#line 891 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list(OP_ENTERSUB, OPf_STACKED,
 				op_append_elem(OP_LIST,
 				    op_prepend_elem(OP_LIST, (ps[-1].val.opval), (ps[0].val.opval)),
@@ -1137,7 +1137,7 @@ case 2:
     break;
 
   case 134:
-#line 897 "perly.y" /* yacc.c:1646  */
+#line 897 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list(OP_ENTERSUB, OPf_STACKED,
 				op_append_elem(OP_LIST,
 				    op_prepend_elem(OP_LIST, (ps[-3].val.opval), (ps[-1].val.opval)),
@@ -1147,26 +1147,26 @@ case 2:
     break;
 
   case 135:
-#line 903 "perly.y" /* yacc.c:1646  */
+#line 903 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list((ps[-1].val.ival), 0, (ps[0].val.opval)); }
 
     break;
 
   case 136:
-#line 905 "perly.y" /* yacc.c:1646  */
+#line 905 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list((ps[-3].val.ival), 0, (ps[-1].val.opval)); }
 
     break;
 
   case 137:
-#line 907 "perly.y" /* yacc.c:1646  */
+#line 907 "perly.y" /* yacc.c:1648  */
     { SvREFCNT_inc_simple_void(PL_compcv);
 			  (yyval.opval) = newANONATTRSUB((ps[-1].val.ival), 0, NULL, (ps[0].val.opval)); }
 
     break;
 
   case 138:
-#line 910 "perly.y" /* yacc.c:1646  */
+#line 910 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED,
 				 op_append_elem(OP_LIST,
 				   op_prepend_elem(OP_LIST, (ps[-1].val.opval), (ps[0].val.opval)), (ps[-4].val.opval)));
@@ -1175,20 +1175,20 @@ case 2:
     break;
 
   case 141:
-#line 925 "perly.y" /* yacc.c:1646  */
+#line 925 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP(OP_GELEM, 0, (ps[-4].val.opval), scalar((ps[-2].val.opval))); }
 
     break;
 
   case 142:
-#line 927 "perly.y" /* yacc.c:1646  */
+#line 927 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP(OP_AELEM, 0, oopsAV((ps[-3].val.opval)), scalar((ps[-1].val.opval)));
 			}
 
     break;
 
   case 143:
-#line 930 "perly.y" /* yacc.c:1646  */
+#line 930 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP(OP_AELEM, 0,
 					ref(newAVREF((ps[-4].val.opval)),OP_RV2AV),
 					scalar((ps[-1].val.opval)));
@@ -1197,7 +1197,7 @@ case 2:
     break;
 
   case 144:
-#line 935 "perly.y" /* yacc.c:1646  */
+#line 935 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP(OP_AELEM, 0,
 					ref(newAVREF((ps[-3].val.opval)),OP_RV2AV),
 					scalar((ps[-1].val.opval)));
@@ -1206,14 +1206,14 @@ case 2:
     break;
 
   case 145:
-#line 940 "perly.y" /* yacc.c:1646  */
+#line 940 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP(OP_HELEM, 0, oopsHV((ps[-4].val.opval)), jmaybe((ps[-2].val.opval)));
 			}
 
     break;
 
   case 146:
-#line 943 "perly.y" /* yacc.c:1646  */
+#line 943 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP(OP_HELEM, 0,
 					ref(newHVREF((ps[-5].val.opval)),OP_RV2HV),
 					jmaybe((ps[-2].val.opval))); }
@@ -1221,7 +1221,7 @@ case 2:
     break;
 
   case 147:
-#line 947 "perly.y" /* yacc.c:1646  */
+#line 947 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP(OP_HELEM, 0,
 					ref(newHVREF((ps[-4].val.opval)),OP_RV2HV),
 					jmaybe((ps[-2].val.opval))); }
@@ -1229,7 +1229,7 @@ case 2:
     break;
 
   case 148:
-#line 951 "perly.y" /* yacc.c:1646  */
+#line 951 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED,
 				   newCVREF(0, scalar((ps[-3].val.opval))));
 			  if (parser->expect == XBLOCK)
@@ -1239,7 +1239,7 @@ case 2:
     break;
 
   case 149:
-#line 957 "perly.y" /* yacc.c:1646  */
+#line 957 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED,
 				   op_append_elem(OP_LIST, (ps[-1].val.opval),
 				       newCVREF(0, scalar((ps[-4].val.opval)))));
@@ -1250,7 +1250,7 @@ case 2:
     break;
 
   case 150:
-#line 965 "perly.y" /* yacc.c:1646  */
+#line 965 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED,
 				   op_append_elem(OP_LIST, (ps[-1].val.opval),
 					       newCVREF(0, scalar((ps[-3].val.opval)))));
@@ -1261,7 +1261,7 @@ case 2:
     break;
 
   case 151:
-#line 972 "perly.y" /* yacc.c:1646  */
+#line 972 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED,
 				   newCVREF(0, scalar((ps[-2].val.opval))));
 			  if (parser->expect == XBLOCK)
@@ -1271,37 +1271,37 @@ case 2:
     break;
 
   case 152:
-#line 978 "perly.y" /* yacc.c:1646  */
+#line 978 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newSLICEOP(0, (ps[-1].val.opval), (ps[-4].val.opval)); }
 
     break;
 
   case 153:
-#line 980 "perly.y" /* yacc.c:1646  */
+#line 980 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newSLICEOP(0, (ps[-1].val.opval), (ps[-3].val.opval)); }
 
     break;
 
   case 154:
-#line 982 "perly.y" /* yacc.c:1646  */
+#line 982 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newSLICEOP(0, (ps[-1].val.opval), NULL); }
 
     break;
 
   case 155:
-#line 987 "perly.y" /* yacc.c:1646  */
+#line 987 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newASSIGNOP(OPf_STACKED, (ps[-2].val.opval), (ps[-1].val.ival), (ps[0].val.opval)); }
 
     break;
 
   case 156:
-#line 989 "perly.y" /* yacc.c:1646  */
+#line 989 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP((ps[-1].val.ival), 0, scalar((ps[-2].val.opval)), scalar((ps[0].val.opval))); }
 
     break;
 
   case 157:
-#line 991 "perly.y" /* yacc.c:1646  */
+#line 991 "perly.y" /* yacc.c:1648  */
     {   if ((ps[-1].val.ival) != OP_REPEAT)
 				scalar((ps[-2].val.opval));
 			    (yyval.opval) = newBINOP((ps[-1].val.ival), 0, (ps[-2].val.opval), scalar((ps[0].val.opval)));
@@ -1310,111 +1310,111 @@ case 2:
     break;
 
   case 158:
-#line 996 "perly.y" /* yacc.c:1646  */
+#line 996 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP((ps[-1].val.ival), 0, scalar((ps[-2].val.opval)), scalar((ps[0].val.opval))); }
 
     break;
 
   case 159:
-#line 998 "perly.y" /* yacc.c:1646  */
+#line 998 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP((ps[-1].val.ival), 0, scalar((ps[-2].val.opval)), scalar((ps[0].val.opval))); }
 
     break;
 
   case 160:
-#line 1000 "perly.y" /* yacc.c:1646  */
+#line 1000 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP((ps[-1].val.ival), 0, scalar((ps[-2].val.opval)), scalar((ps[0].val.opval))); }
 
     break;
 
   case 161:
-#line 1002 "perly.y" /* yacc.c:1646  */
+#line 1002 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP((ps[-1].val.ival), 0, scalar((ps[-2].val.opval)), scalar((ps[0].val.opval))); }
 
     break;
 
   case 162:
-#line 1004 "perly.y" /* yacc.c:1646  */
+#line 1004 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP((ps[-1].val.ival), 0, scalar((ps[-2].val.opval)), scalar((ps[0].val.opval))); }
 
     break;
 
   case 163:
-#line 1006 "perly.y" /* yacc.c:1646  */
+#line 1006 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newBINOP((ps[-1].val.ival), 0, scalar((ps[-2].val.opval)), scalar((ps[0].val.opval))); }
 
     break;
 
   case 164:
-#line 1008 "perly.y" /* yacc.c:1646  */
+#line 1008 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newRANGE((ps[-1].val.ival), scalar((ps[-2].val.opval)), scalar((ps[0].val.opval))); }
 
     break;
 
   case 165:
-#line 1010 "perly.y" /* yacc.c:1646  */
+#line 1010 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOGOP(OP_AND, 0, (ps[-2].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 166:
-#line 1012 "perly.y" /* yacc.c:1646  */
+#line 1012 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOGOP(OP_OR, 0, (ps[-2].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 167:
-#line 1014 "perly.y" /* yacc.c:1646  */
+#line 1014 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOGOP(OP_DOR, 0, (ps[-2].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 168:
-#line 1016 "perly.y" /* yacc.c:1646  */
+#line 1016 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = bind_match((ps[-1].val.ival), (ps[-2].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 169:
-#line 1021 "perly.y" /* yacc.c:1646  */
+#line 1021 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_NEGATE, 0, scalar((ps[0].val.opval))); }
 
     break;
 
   case 170:
-#line 1023 "perly.y" /* yacc.c:1646  */
+#line 1023 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 171:
-#line 1026 "perly.y" /* yacc.c:1646  */
+#line 1026 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_NOT, 0, scalar((ps[0].val.opval))); }
 
     break;
 
   case 172:
-#line 1028 "perly.y" /* yacc.c:1646  */
+#line 1028 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP((ps[-1].val.ival), 0, scalar((ps[0].val.opval))); }
 
     break;
 
   case 173:
-#line 1030 "perly.y" /* yacc.c:1646  */
+#line 1030 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_POSTINC, 0,
 					op_lvalue(scalar((ps[-1].val.opval)), OP_POSTINC)); }
 
     break;
 
   case 174:
-#line 1033 "perly.y" /* yacc.c:1646  */
+#line 1033 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_POSTDEC, 0,
 					op_lvalue(scalar((ps[-1].val.opval)), OP_POSTDEC));}
 
     break;
 
   case 175:
-#line 1036 "perly.y" /* yacc.c:1646  */
+#line 1036 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_convert_list(OP_JOIN, 0,
 				       op_append_elem(
 					OP_LIST,
@@ -1429,155 +1429,155 @@ case 2:
     break;
 
   case 176:
-#line 1047 "perly.y" /* yacc.c:1646  */
+#line 1047 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_PREINC, 0,
 					op_lvalue(scalar((ps[0].val.opval)), OP_PREINC)); }
 
     break;
 
   case 177:
-#line 1050 "perly.y" /* yacc.c:1646  */
+#line 1050 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_PREDEC, 0,
 					op_lvalue(scalar((ps[0].val.opval)), OP_PREDEC)); }
 
     break;
 
   case 178:
-#line 1057 "perly.y" /* yacc.c:1646  */
+#line 1057 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newANONLIST((ps[-1].val.opval)); }
 
     break;
 
   case 179:
-#line 1059 "perly.y" /* yacc.c:1646  */
+#line 1059 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newANONLIST(NULL);}
 
     break;
 
   case 180:
-#line 1061 "perly.y" /* yacc.c:1646  */
+#line 1061 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newANONHASH((ps[-2].val.opval)); }
 
     break;
 
   case 181:
-#line 1063 "perly.y" /* yacc.c:1646  */
+#line 1063 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newANONHASH(NULL); }
 
     break;
 
   case 182:
-#line 1065 "perly.y" /* yacc.c:1646  */
+#line 1065 "perly.y" /* yacc.c:1648  */
     { SvREFCNT_inc_simple_void(PL_compcv);
 			  (yyval.opval) = newANONATTRSUB((ps[-3].val.ival), (ps[-2].val.opval), (ps[-1].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 183:
-#line 1068 "perly.y" /* yacc.c:1646  */
+#line 1068 "perly.y" /* yacc.c:1648  */
     { SvREFCNT_inc_simple_void(PL_compcv);
 			  (yyval.opval) = newANONATTRSUB((ps[-2].val.ival), NULL, (ps[-1].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 184:
-#line 1074 "perly.y" /* yacc.c:1646  */
+#line 1074 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = dofile((ps[0].val.opval), (ps[-1].val.ival));}
 
     break;
 
   case 185:
-#line 1076 "perly.y" /* yacc.c:1646  */
+#line 1076 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_NULL, OPf_SPECIAL, op_scope((ps[0].val.opval)));}
 
     break;
 
   case 190:
-#line 1084 "perly.y" /* yacc.c:1646  */
+#line 1084 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newCONDOP(0, (ps[-4].val.opval), (ps[-2].val.opval), (ps[0].val.opval)); }
 
     break;
 
   case 191:
-#line 1086 "perly.y" /* yacc.c:1646  */
+#line 1086 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_REFGEN, 0, (ps[0].val.opval)); }
 
     break;
 
   case 192:
-#line 1088 "perly.y" /* yacc.c:1646  */
+#line 1088 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_REFGEN, 0, localize((ps[0].val.opval),1)); }
 
     break;
 
   case 193:
-#line 1090 "perly.y" /* yacc.c:1646  */
+#line 1090 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 194:
-#line 1092 "perly.y" /* yacc.c:1646  */
+#line 1092 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = localize((ps[0].val.opval),0); }
 
     break;
 
   case 195:
-#line 1094 "perly.y" /* yacc.c:1646  */
+#line 1094 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = sawparens((ps[-1].val.opval)); }
 
     break;
 
   case 196:
-#line 1096 "perly.y" /* yacc.c:1646  */
+#line 1096 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 197:
-#line 1098 "perly.y" /* yacc.c:1646  */
+#line 1098 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = sawparens(newNULLLIST()); }
 
     break;
 
   case 198:
-#line 1100 "perly.y" /* yacc.c:1646  */
+#line 1100 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 199:
-#line 1102 "perly.y" /* yacc.c:1646  */
+#line 1102 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 200:
-#line 1104 "perly.y" /* yacc.c:1646  */
+#line 1104 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 201:
-#line 1106 "perly.y" /* yacc.c:1646  */
+#line 1106 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 202:
-#line 1108 "perly.y" /* yacc.c:1646  */
+#line 1108 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_AV2ARYLEN, 0, ref((ps[0].val.opval), OP_AV2ARYLEN));}
 
     break;
 
   case 203:
-#line 1110 "perly.y" /* yacc.c:1646  */
+#line 1110 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 204:
-#line 1112 "perly.y" /* yacc.c:1646  */
+#line 1112 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_prepend_elem(OP_ASLICE,
 				newOP(OP_PUSHMARK, 0),
 				    newLISTOP(OP_ASLICE, 0,
@@ -1591,7 +1591,7 @@ case 2:
     break;
 
   case 205:
-#line 1122 "perly.y" /* yacc.c:1646  */
+#line 1122 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_prepend_elem(OP_KVASLICE,
 				newOP(OP_PUSHMARK, 0),
 				    newLISTOP(OP_KVASLICE, 0,
@@ -1605,7 +1605,7 @@ case 2:
     break;
 
   case 206:
-#line 1132 "perly.y" /* yacc.c:1646  */
+#line 1132 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_prepend_elem(OP_HSLICE,
 				newOP(OP_PUSHMARK, 0),
 				    newLISTOP(OP_HSLICE, 0,
@@ -1619,7 +1619,7 @@ case 2:
     break;
 
   case 207:
-#line 1142 "perly.y" /* yacc.c:1646  */
+#line 1142 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_prepend_elem(OP_KVHSLICE,
 				newOP(OP_PUSHMARK, 0),
 				    newLISTOP(OP_KVHSLICE, 0,
@@ -1633,26 +1633,26 @@ case 2:
     break;
 
   case 208:
-#line 1152 "perly.y" /* yacc.c:1646  */
+#line 1152 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 209:
-#line 1154 "perly.y" /* yacc.c:1646  */
+#line 1154 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, 0, scalar((ps[0].val.opval))); }
 
     break;
 
   case 210:
-#line 1156 "perly.y" /* yacc.c:1646  */
+#line 1156 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED, scalar((ps[-2].val.opval)));
 			}
 
     break;
 
   case 211:
-#line 1159 "perly.y" /* yacc.c:1646  */
+#line 1159 "perly.y" /* yacc.c:1648  */
     {
 			  (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED,
 				op_append_elem(OP_LIST, (ps[-1].val.opval), scalar((ps[-3].val.opval))));
@@ -1661,7 +1661,7 @@ case 2:
     break;
 
   case 212:
-#line 1164 "perly.y" /* yacc.c:1646  */
+#line 1164 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED,
 			    op_append_elem(OP_LIST, (ps[0].val.opval), scalar((ps[-1].val.opval))));
 			}
@@ -1669,130 +1669,130 @@ case 2:
     break;
 
   case 213:
-#line 1168 "perly.y" /* yacc.c:1646  */
+#line 1168 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newSVREF((ps[-3].val.opval)); }
 
     break;
 
   case 214:
-#line 1170 "perly.y" /* yacc.c:1646  */
+#line 1170 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newAVREF((ps[-3].val.opval)); }
 
     break;
 
   case 215:
-#line 1172 "perly.y" /* yacc.c:1646  */
+#line 1172 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newHVREF((ps[-3].val.opval)); }
 
     break;
 
   case 216:
-#line 1174 "perly.y" /* yacc.c:1646  */
+#line 1174 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, 0,
 				       scalar(newCVREF((ps[-1].val.ival),(ps[-3].val.opval)))); }
 
     break;
 
   case 217:
-#line 1177 "perly.y" /* yacc.c:1646  */
+#line 1177 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newGVREF(0,(ps[-3].val.opval)); }
 
     break;
 
   case 218:
-#line 1179 "perly.y" /* yacc.c:1646  */
+#line 1179 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newOP((ps[0].val.ival), OPf_SPECIAL);
 			    PL_hints |= HINT_BLOCK_SCOPE; }
 
     break;
 
   case 219:
-#line 1182 "perly.y" /* yacc.c:1646  */
+#line 1182 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newLOOPEX((ps[-1].val.ival),(ps[0].val.opval)); }
 
     break;
 
   case 220:
-#line 1184 "perly.y" /* yacc.c:1646  */
+#line 1184 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_NOT, 0, scalar((ps[0].val.opval))); }
 
     break;
 
   case 221:
-#line 1186 "perly.y" /* yacc.c:1646  */
+#line 1186 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newOP((ps[0].val.ival), 0); }
 
     break;
 
   case 222:
-#line 1188 "perly.y" /* yacc.c:1646  */
+#line 1188 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP((ps[-1].val.ival), 0, (ps[0].val.opval)); }
 
     break;
 
   case 223:
-#line 1190 "perly.y" /* yacc.c:1646  */
+#line 1190 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP((ps[-1].val.ival), 0, (ps[0].val.opval)); }
 
     break;
 
   case 224:
-#line 1192 "perly.y" /* yacc.c:1646  */
+#line 1192 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newOP(OP_REQUIRE, (ps[0].val.ival) ? OPf_SPECIAL : 0); }
 
     break;
 
   case 225:
-#line 1194 "perly.y" /* yacc.c:1646  */
+#line 1194 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_REQUIRE, (ps[-1].val.ival) ? OPf_SPECIAL : 0, (ps[0].val.opval)); }
 
     break;
 
   case 226:
-#line 1196 "perly.y" /* yacc.c:1646  */
+#line 1196 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED, scalar((ps[0].val.opval))); }
 
     break;
 
   case 227:
-#line 1198 "perly.y" /* yacc.c:1646  */
+#line 1198 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED,
 			    op_append_elem(OP_LIST, (ps[0].val.opval), scalar((ps[-1].val.opval)))); }
 
     break;
 
   case 228:
-#line 1201 "perly.y" /* yacc.c:1646  */
+#line 1201 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newOP((ps[0].val.ival), 0); }
 
     break;
 
   case 229:
-#line 1203 "perly.y" /* yacc.c:1646  */
+#line 1203 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newOP((ps[-2].val.ival), 0);}
 
     break;
 
   case 230:
-#line 1205 "perly.y" /* yacc.c:1646  */
+#line 1205 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 231:
-#line 1207 "perly.y" /* yacc.c:1646  */
+#line 1207 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[-2].val.opval); }
 
     break;
 
   case 232:
-#line 1209 "perly.y" /* yacc.c:1646  */
+#line 1209 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_ENTERSUB, OPf_STACKED, scalar((ps[0].val.opval))); }
 
     break;
 
   case 233:
-#line 1211 "perly.y" /* yacc.c:1646  */
+#line 1211 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = ((ps[-2].val.ival) == OP_NOT)
                           ? newUNOP((ps[-2].val.ival), 0, newSVOP(OP_CONST, 0, newSViv(0)))
                           : newOP((ps[-2].val.ival), OPf_SPECIAL); }
@@ -1800,13 +1800,13 @@ case 2:
     break;
 
   case 234:
-#line 1215 "perly.y" /* yacc.c:1646  */
+#line 1215 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP((ps[-3].val.ival), 0, (ps[-1].val.opval)); }
 
     break;
 
   case 235:
-#line 1217 "perly.y" /* yacc.c:1646  */
+#line 1217 "perly.y" /* yacc.c:1648  */
     {
 			    if (   (ps[0].val.opval)->op_type != OP_TRANS
 			        && (ps[0].val.opval)->op_type != OP_TRANSR
@@ -1821,115 +1821,115 @@ case 2:
     break;
 
   case 236:
-#line 1228 "perly.y" /* yacc.c:1646  */
+#line 1228 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = pmruntime((ps[-5].val.opval), (ps[-2].val.opval), (ps[-1].val.opval), 1, (ps[-4].val.ival)); }
 
     break;
 
   case 240:
-#line 1236 "perly.y" /* yacc.c:1646  */
+#line 1236 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = my_attrs((ps[-1].val.opval),(ps[0].val.opval)); }
 
     break;
 
   case 241:
-#line 1238 "perly.y" /* yacc.c:1646  */
+#line 1238 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = localize((ps[0].val.opval),1); }
 
     break;
 
   case 242:
-#line 1240 "perly.y" /* yacc.c:1646  */
+#line 1240 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newUNOP(OP_REFGEN, 0, my_attrs((ps[-1].val.opval),(ps[0].val.opval))); }
 
     break;
 
   case 243:
-#line 1245 "perly.y" /* yacc.c:1646  */
+#line 1245 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = sawparens((ps[-1].val.opval)); }
 
     break;
 
   case 244:
-#line 1247 "perly.y" /* yacc.c:1646  */
+#line 1247 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = sawparens(newNULLLIST()); }
 
     break;
 
   case 245:
-#line 1250 "perly.y" /* yacc.c:1646  */
+#line 1250 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 246:
-#line 1252 "perly.y" /* yacc.c:1646  */
+#line 1252 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 247:
-#line 1254 "perly.y" /* yacc.c:1646  */
+#line 1254 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 248:
-#line 1259 "perly.y" /* yacc.c:1646  */
+#line 1259 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 249:
-#line 1261 "perly.y" /* yacc.c:1646  */
+#line 1261 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 250:
-#line 1265 "perly.y" /* yacc.c:1646  */
+#line 1265 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 251:
-#line 1267 "perly.y" /* yacc.c:1646  */
+#line 1267 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 252:
-#line 1271 "perly.y" /* yacc.c:1646  */
+#line 1271 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = NULL; }
 
     break;
 
   case 253:
-#line 1273 "perly.y" /* yacc.c:1646  */
+#line 1273 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
 
   case 254:
-#line 1279 "perly.y" /* yacc.c:1646  */
+#line 1279 "perly.y" /* yacc.c:1648  */
     { parser->in_my = 0; (yyval.opval) = my((ps[0].val.opval)); }
 
     break;
 
   case 262:
-#line 1296 "perly.y" /* yacc.c:1646  */
+#line 1296 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newCVREF((ps[-1].val.ival),(ps[0].val.opval)); }
 
     break;
 
   case 263:
-#line 1300 "perly.y" /* yacc.c:1646  */
+#line 1300 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newSVREF((ps[0].val.opval)); }
 
     break;
 
   case 264:
-#line 1304 "perly.y" /* yacc.c:1646  */
+#line 1304 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newAVREF((ps[0].val.opval));
 			  if ((yyval.opval)) (yyval.opval)->op_private |= (ps[-1].val.ival);
 			}
@@ -1937,7 +1937,7 @@ case 2:
     break;
 
   case 265:
-#line 1310 "perly.y" /* yacc.c:1646  */
+#line 1310 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newHVREF((ps[0].val.opval));
 			  if ((yyval.opval)) (yyval.opval)->op_private |= (ps[-1].val.ival);
 			}
@@ -1945,61 +1945,61 @@ case 2:
     break;
 
   case 266:
-#line 1316 "perly.y" /* yacc.c:1646  */
+#line 1316 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newAVREF((ps[0].val.opval)); }
 
     break;
 
   case 267:
-#line 1318 "perly.y" /* yacc.c:1646  */
+#line 1318 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newAVREF((ps[-3].val.opval)); }
 
     break;
 
   case 268:
-#line 1322 "perly.y" /* yacc.c:1646  */
+#line 1322 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newGVREF(0,(ps[0].val.opval)); }
 
     break;
 
   case 270:
-#line 1327 "perly.y" /* yacc.c:1646  */
+#line 1327 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newAVREF((ps[-2].val.opval)); }
 
     break;
 
   case 272:
-#line 1332 "perly.y" /* yacc.c:1646  */
+#line 1332 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newHVREF((ps[-2].val.opval)); }
 
     break;
 
   case 274:
-#line 1337 "perly.y" /* yacc.c:1646  */
+#line 1337 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = newGVREF(0,(ps[-2].val.opval)); }
 
     break;
 
   case 275:
-#line 1342 "perly.y" /* yacc.c:1646  */
+#line 1342 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = scalar((ps[0].val.opval)); }
 
     break;
 
   case 276:
-#line 1344 "perly.y" /* yacc.c:1646  */
+#line 1344 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = scalar((ps[0].val.opval)); }
 
     break;
 
   case 277:
-#line 1346 "perly.y" /* yacc.c:1646  */
+#line 1346 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = op_scope((ps[0].val.opval)); }
 
     break;
 
   case 278:
-#line 1349 "perly.y" /* yacc.c:1646  */
+#line 1349 "perly.y" /* yacc.c:1648  */
     { (yyval.opval) = (ps[0].val.opval); }
 
     break;
@@ -2010,6 +2010,6 @@ case 2:
     
 
 /* Generated from:
- * ac323bb6f3cacf4eb8eda6403a58c41f30954d07a4a6d6f7f00de81577b2c79a perly.y
- * b6fae5748f9bef6db4740aa5e122b84ac5181852d42474d0ecad621fa4253306 regen_perly.pl
+ * 13adfb5833c2eeeb2f9729dda89ce1490f78709a0a5ae658b0dbc922704f06cb perly.y
+ * c94484dbdeb64039efd375d7aa522bc82300a56849280b2bfcff6631ee96fe93 regen_perly.pl
  * ex: set ro: */
diff --git a/perly.h b/perly.h
index 7727e7b..9579dcf 100644
--- a/perly.h
+++ b/perly.h
@@ -7,11 +7,11 @@
 #define PERL_BISON_VERSION  30000
 
 #ifdef PERL_CORE
-/* A Bison parser, made by GNU Bison 3.0.2.  */
+/* A Bison parser, made by GNU Bison 3.0.5.  */
 
 /* Bison interface for Yacc-like parsers in C
 
-   Copyright (C) 1984, 1989-1990, 2000-2013 Free Software Foundation, Inc.
+   Copyright (C) 1984, 1989-1990, 2000-2015, 2018 Free Software Foundation, Inc.
 
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -162,7 +162,7 @@ S_is_opval_token(int type) {
 #endif /* PERL_IN_TOKE_C */
 #endif /* PERL_CORE */
 #if ! defined YYSTYPE && ! defined YYSTYPE_IS_DECLARED
-typedef union YYSTYPE YYSTYPE;
+
 union YYSTYPE
 {
 
@@ -173,6 +173,8 @@ union YYSTYPE
     GV *gvval;
 
 };
+
+typedef union YYSTYPE YYSTYPE;
 # define YYSTYPE_IS_TRIVIAL 1
 # define YYSTYPE_IS_DECLARED 1
 #endif
@@ -183,6 +185,6 @@ int yyparse (void);
 
 
 /* Generated from:
- * ac323bb6f3cacf4eb8eda6403a58c41f30954d07a4a6d6f7f00de81577b2c79a perly.y
- * b6fae5748f9bef6db4740aa5e122b84ac5181852d42474d0ecad621fa4253306 regen_perly.pl
+ * 13adfb5833c2eeeb2f9729dda89ce1490f78709a0a5ae658b0dbc922704f06cb perly.y
+ * c94484dbdeb64039efd375d7aa522bc82300a56849280b2bfcff6631ee96fe93 regen_perly.pl
  * ex: set ro: */
diff --git a/perly.tab b/perly.tab
index d918aa6..d19378f 100644
--- a/perly.tab
+++ b/perly.tab
@@ -1130,6 +1130,6 @@ static const toketypes yy_type_tab[] =
 };
 
 /* Generated from:
- * ac323bb6f3cacf4eb8eda6403a58c41f30954d07a4a6d6f7f00de81577b2c79a perly.y
- * b6fae5748f9bef6db4740aa5e122b84ac5181852d42474d0ecad621fa4253306 regen_perly.pl
+ * 13adfb5833c2eeeb2f9729dda89ce1490f78709a0a5ae658b0dbc922704f06cb perly.y
+ * c94484dbdeb64039efd375d7aa522bc82300a56849280b2bfcff6631ee96fe93 regen_perly.pl
  * ex: set ro: */
diff --git a/pod/perldelta.pod b/pod/perldelta.pod
index ad87d40..4570f6e 100644
--- a/pod/perldelta.pod
+++ b/pod/perldelta.pod
@@ -44,7 +44,59 @@ L</Reporting Bugs> below.
 
 =item *
 
-L<Module::CoreList> has been upgraded from version 5.20180622 to 5.20181129_28.
+L<B> has been upgraded from version 1.74 to 1.75.
+
+=item *
+
+L<B::Concise> has been upgraded from version 1.003 to 1.004.
+
+=item *
+
+L<B::Deparse> has been upgraded from version 1.48 to 1.49.
+
+=item *
+
+L<bignum> has been upgraded from version 0.50 to 0.51.
+
+=item *
+
+L<bytes> has been upgraded from version 1.06 to 1.07.
+
+=item *
+
+L<CPAN> has been upgraded from version 2.20 to 2.21.
+
+=item *
+
+L<Devel::Peek> has been upgraded from version 1.27 to 1.28.
+
+=item *
+
+L<feature> has been upgraded from version 1.53 to 1.54.
+
+=item *
+
+L<File::Copy> has been upgraded from version 2.33 to 2.34.
+
+=item *
+
+L<File::Glob> has been upgraded from version 1.31 to 1.32.
+
+=item *
+
+L<Module::CoreList> has been upgraded from version 5.20180920 to 5.20181020.
+
+=item *
+
+L<sigtrap> has been upgraded from version 1.08 to 1.09.
+
+=item *
+
+L<Unicode::UCD> has been upgraded from version 0.71 to 0.72.
+
+=item *
+
+L<vars> has been upgraded from version 1.04 to 1.05.
 
 =back
 
diff --git a/dist/Module-CoreList/lib/Module/CoreList.pm b/dist/Module-CoreList/lib/Module/CoreList.pm
index ad87d40..4570f6e 100644
--- a/dist/Module-CoreList/lib/Module/CoreList.pm
+++ b/dist/Module-CoreList/lib/Module/CoreList.pm
@@ -15815,14 +15815,20 @@ for my $version ( sort { $a <=> $b } keys %released ) {
         }
     },
     5.028001 => {
-        delta_from => 5.028,
+        delta_from => 5.028000,
         changed => {
             'B::Op_private'         => '5.028001',
             'Config'                => '5.028001',
+            'ExtUtils::CBuilder::Platform::msys'=> '0.280230',
             'Module::CoreList'      => '5.20181129_28',
             'Module::CoreList::Utils'=> '5.20181129_28',
+            'Win32'                 => '0.52_01',
         },
         removed => {
+            'Pod::Functions'        => 1,
+            'Storable'              => 1,
+            'Storable::Limit'       => 1,
+            'unicore::Name'         => 1,
         }
     },
 );
@@ -17393,8 +17399,8 @@ sub is_core
     'IPC::Semaphore'        => undef,
     'IPC::SharedMem'        => undef,
     'IPC::SysV'             => undef,
-    'JSON::PP'              => undef,
-    'JSON::PP::Boolean'     => undef,
+    'JSON::PP'              => 'https://github.com/makamaka/JSON-PP/issues',
+    'JSON::PP::Boolean'     => 'https://github.com/makamaka/JSON-PP/issues',
     'List::Util'            => 'https://rt.cpan.org/Public/Dist/Display.html?Name=Scalar-List-Utils',
     'List::Util::XS'        => 'https://rt.cpan.org/Public/Dist/Display.html?Name=Scalar-List-Utils',
     'Locale::Codes'         => undef,
diff --git a/sv.c b/sv.c
index 7dd492a..ea7adfd 100644
--- a/sv.c
+++ b/sv.c
@@ -874,6 +874,7 @@ PL_body_roots[HE_SVSLOT=SVt_NULL] is filled by S_more_he, but the
 bodies_by_type[SVt_NULL] slot is not used, as the table is not
 available in hv.c.
 
+=cut
 */
 
 struct body_details {
diff --git a/t/re/regexp.t b/t/re/regexp.t
index 037d7b7..d420722 100755
--- a/t/re/regexp.t
+++ b/t/re/regexp.t
@@ -145,6 +145,8 @@ foreach (@tests) {
         die "Bad test definition on line $test: $_\n";
     }
     $reason = '' unless defined $reason;
+	$expect = '' unless defined $expect;
+	$repl = '' unless defined $repl;
     my $input = join(':',$pat,$subject,$result,$repl,$expect);
 
     # the double '' below keeps simple syntax highlighters from going crazy
diff --git a/t/porting/customized.dat b/t/porting/customized.dat
index 70722d5..f4c9365 100644
--- a/t/porting/customized.dat
+++ b/t/porting/customized.dat
@@ -1,28 +1,28 @@
 # Regenerate this file using:
 #     cd t
 #     ./perl -I../lib porting/customized.t --regen
-Devel::PPPort dist/Devel-PPPort/parts/embed.fnc e030719d9c6921810554a8e2d398543348b4878c
+Devel::PPPort dist/Devel-PPPort/parts/embed.fnc 475daf779f99308ea3c8474debf92ca527bbd23a
 Digest cpan/Digest/Digest.pm 43f7f544cb11842b2f55c73e28930da50774e081
 Encode cpan/Encode/t/truncated_utf8.t 36e6780c20f1d22ed7c97b1a388a63629617451f
 Locale::Maketext::Simple cpan/Locale-Maketext-Simple/lib/Locale/Maketext/Simple.pm 57ed38905791a17c150210cd6f42ead22a7707b6
-Math::Complex cpan/Math-Complex/lib/Math/Complex.pm 198ea6c6c584f5ea79a0fd7e9d411d0878f3b2af
+Math::Complex cpan/Math-Complex/lib/Math/Complex.pm c4c206d1a273d9b90c57346b5eaf6a27b4cb1740
 Math::Complex cpan/Math-Complex/t/Complex.t 4f307ed6fc59f1e5fb0e6b11103fc631b6bdb335
 Math::Complex cpan/Math-Complex/t/Trig.t 2682526e23a161d54732c2a66393fe4a234d1865
-Memoize cpan/Memoize/Memoize.pm 902092ff91cdec9c7b4bd06202eb179e1ce26ca2
-NEXT cpan/NEXT/lib/NEXT.pm 2c83d03ee361816e53ccb931137d314ab878d19f
+Memoize cpan/Memoize/Memoize.pm 61ed1a0be11f1e56f016473780a802358af077f5
+NEXT cpan/NEXT/lib/NEXT.pm 24d34be494751dbc0d89760435d7f29846016062
 NEXT cpan/NEXT/t/next.t 66fd60eb0f75b6f3eea95d1dee745f9a7a348146
 Net::Ping dist/Net-Ping/t/000_load.t deff5dc2ca54dae28cb19d3631427db127279ac2
-Net::Ping dist/Net-Ping/t/001_new.t 90c9d63509b3efc8941449fbd1ca8b807fa42040
+Net::Ping dist/Net-Ping/t/001_new.t d4fe6bdc9aafb42c3566bdd8d2288173ce207a36
 Net::Ping dist/Net-Ping/t/500_ping_icmp.t a003daa5eaf215e58234786bb1fbfbebf669bf44
 Pod::Checker cpan/Pod-Checker/t/pod/contains_bad_pod.xr 73538fd80dfe6e19ad561fe034009b44460208f6
 Pod::Checker cpan/Pod-Checker/t/pod/selfcheck.t 8ce3cfd38e4b9bcf5bc7fe7f2a14195e49aed7d8
-Pod::Checker cpan/Pod-Checker/t/pod/testcmp.pl a0cd5c8eca775c7753f4464eee96fa916e3d8a16
+Pod::Checker cpan/Pod-Checker/t/pod/testcmp.pl 56c3afd23c8b065bfa2b677e205658fa32d282b6
 Pod::Checker cpan/Pod-Checker/t/pod/testpchk.pl b2072c7f4379fd050e15424175d7cac5facf5b3b
-Pod::Perldoc cpan/Pod-Perldoc/lib/Pod/Perldoc.pm 582be34c077c9ff44d99914724a0cc2140bcd48c
+Pod::Perldoc cpan/Pod-Perldoc/lib/Pod/Perldoc.pm 9747d48aad3db0b067401879ecb19536a0e1cb66
 autodie cpan/autodie/lib/autodie/exception.pm b99e4e35a9ed36de94d54437888822ced4936207
-autodie cpan/autodie/lib/autodie/hints.pm e1998fec61fb4e82fe46585bd82c73200be6f262
+autodie cpan/autodie/lib/autodie/hints.pm f2b0a287e680356c73386ac3e0e74c959b6cbfbb
 autodie cpan/autodie/t/exceptions.t ad315a208f875e06b0964012ce8d65daa438c036
-autodie cpan/autodie/t/lib/Hints_pod_examples.pm 6944c218e9754b3613c8d0c90a5ae8aceccb5c99
+autodie cpan/autodie/t/lib/Hints_pod_examples.pm a1aa9ee3bf6075e3159a9a194647aaf947c76170
 autodie cpan/autodie/t/mkdir.t 9e70d2282a3cc7d76a78bf8144fccba20fb37dac
 experimental cpan/experimental/t/basic.t a073ea03ccc98dec496569f3648ab01a5fe1c7a0
 perlfaq cpan/perlfaq/lib/perlfaq5.pod bcc1b6af3b6dff3973643acf8d5e741463374123
diff --git a/charclass_invlists.h b/charclass_invlists.h
index 8155e07..a6003df 100644
--- a/charclass_invlists.h
+++ b/charclass_invlists.h
@@ -374605,7 +374605,7 @@ static const U8 WB_table[24][24] = {
 #endif	/* defined(PERL_IN_REGEXEC_C) */
 
 /* Generated from:
- * 4d64b650346ff9ddbcd3986f0da13f3be2379d64c574a120ca7f534d8d17934c lib/Unicode/UCD.pm
+ * aac0a5761f2de82c5e768484e2d0ba5fedf635190ef700ae77bcb12a3f09fbd4 lib/Unicode/UCD.pm
  * ff4404ec64f308bdf7714c50f9fdf0d1d0bf3c34db4d0a67e58ef0c6f88e818f lib/unicore/ArabicShaping.txt
  * 292171a0a1c13d7e581e8781eb4cdf248243b1ab267354a63c7a14429dea2740 lib/unicore/BidiBrackets.txt
  * 8f2695cc42989a79a715ab0d2892bd0c998759180cfdfb998674447f48231940 lib/unicore/BidiMirroring.txt
diff --git a/regcharclass.h b/regcharclass.h
index dc2e382..5f4e2ac 100644
--- a/regcharclass.h
+++ b/regcharclass.h
@@ -1854,7 +1854,7 @@
 #endif /* PERL_REGCHARCLASS_H_ */
 
 /* Generated from:
- * 4d64b650346ff9ddbcd3986f0da13f3be2379d64c574a120ca7f534d8d17934c lib/Unicode/UCD.pm
+ * aac0a5761f2de82c5e768484e2d0ba5fedf635190ef700ae77bcb12a3f09fbd4 lib/Unicode/UCD.pm
  * ff4404ec64f308bdf7714c50f9fdf0d1d0bf3c34db4d0a67e58ef0c6f88e818f lib/unicore/ArabicShaping.txt
  * 292171a0a1c13d7e581e8781eb4cdf248243b1ab267354a63c7a14429dea2740 lib/unicore/BidiBrackets.txt
  * 8f2695cc42989a79a715ab0d2892bd0c998759180cfdfb998674447f48231940 lib/unicore/BidiMirroring.txt
@@ -1901,6 +1901,6 @@
  * ba5b13123afb59ea004c14fcbac80b5ff428c5b8f1b84bf8602b4ffb7da0407f lib/unicore/mktables
  * 21653d2744fdd071f9ef138c805393901bb9547cf3e777ebf50215a191f986ea lib/unicore/version
  * 913d2f93f3cb6cdf1664db888bf840bc4eb074eef824e082fceda24a9445e60c regen/charset_translations.pl
- * 9ea6338945a7d70e5ea4b31ac7856c0b521df96be002e94b4b3b7d31debbf3ab regen/regcharclass.pl
+ * 7a7f186a7e9785844ffebf584c49ed06d202d2e2331a28aeb798964b85decbda regen/regcharclass.pl
  * 393f8d882713a3ba227351ad0f00ea4839fda74fcf77dcd1cdf31519925adba5 regen/regcharclass_multi_char_folds.pl
  * ex: set ro: */
