diff -Naur perl-5.18.0-orig/configpm perl-5.18.0/configpm
--- perl-5.18.0-orig/configpm	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/configpm	2013-06-15 18:12:28.215820300 +0000
@@ -150,8 +150,8 @@
 my $heavy_txt;
 
 my $from = $^O eq 'VMS' ? 'PERLSHR image' : 'binary (from libperl)';
-my $env_cygwin = $^O eq 'cygwin'
-    ? 'push @env, "CYGWIN=\"$ENV{CYGWIN}\"" if $ENV{CYGWIN};' . "\n" : "";
+my $env_cygwin = ($^O eq 'cygwin' || $^O eq 'msys')
+    ? 'push @env, "MSYS=\"$ENV{MSYS}\"" if $ENV{MSYS};' . "\n" : "";
 $heavy_txt .= sprintf <<'ENDOFBEG', $^O, $^O, $from, $^O, $env_cygwin;
 # This file was created by configpm when Perl was built. Any changes
 # made to this file will be lost the next time perl is built.
diff -Naur perl-5.18.0-orig/Configure perl-5.18.0/Configure
--- perl-5.18.0-orig/Configure	2013-06-15 18:57:13.539062500 +0000
+++ perl-5.18.0/Configure	2013-06-15 18:13:19.434570300 +0000
@@ -100,6 +100,7 @@
 	elif test -n "$DJGPP"; then
 		case "X${MACHTYPE:-nonesuchmach}" in
 		*cygwin) ;;
+		*msys) ;;
 		*) p_=\; ;;
 		esac
 	fi
@@ -3261,6 +3262,9 @@
 		cygwin*) osname=cygwin
 			osvers="$3"
 			;;
+		msys*) osname=msys
+			osvers="$3"
+			;;
 		*dc.osx) osname=dcosx
 			osvers="$3"
 			;;
@@ -8180,7 +8184,7 @@
 	;;
 *)	case "$useshrplib" in
 	'')	case "$osname" in
-		svr4*|nonstopux|dgux|dynixptx|esix|powerux|haiku|cygwin*)
+		svr4*|nonstopux|dgux|dynixptx|esix|powerux|haiku|cygwin*|msys*)
 			dflt=y
 			also='Building a shared libperl is required for dynamic loading to work on your system.'
 			;;
@@ -8264,6 +8268,13 @@
 			majonly=cygperl5.$majonly.$so
 			dflt=$majmin
 			;;
+		msys*) # ld links now against the dll directly
+			majmin="msys-perl5_${patchlevel}_${subversion}.${so}"
+			majonly=`echo $patchlevel $subversion |
+				$awk '{printf "%03d%03d", $1, $2}'`
+			majonly=msys-perl5.$majonly.$so
+			dflt=$majmin
+			;;
 		*)	# Try to guess based on whether libc has major.minor.
 			case "$libc" in
 			*libc.$so.[0-9]*.[0-9]*) dflt=$majmin ;;
@@ -8361,6 +8372,9 @@
 	cygwin)
 		# cygwin needs only ldlibpth
 		;;
+	msys)
+		# cygwin needs only ldlibpth
+		;;
 	*)
 		tmp_shrpenv="env LD_RUN_PATH=$shrpdir"
 		;;
@@ -21955,7 +21969,7 @@
 BSD_4_3 BSD_4_4 BSD_NET2 BSD_TIME BSD_TYPES BSDCOMPAT bsdi
 bull c cadmus clipper CMU COFF COMPILER_VERSION
 concurrent convex cpu cray CRAY CRAYMPP ctix CX_UX
-CYGWIN DECC DGUX DGUX_SOURCE DJGPP dmert DOLPHIN DPX2 DSO
+CYGWIN MSYS DECC DGUX DGUX_SOURCE DJGPP dmert DOLPHIN DPX2 DSO
 Dynix DynixPTX ELF encore EPI EXTENSIONS FAVOR_BSD
 FILE_OFFSET_BITS FreeBSD GCC_NEW_VARARGS gcos gcx gimpel
 GLIBC GLIBC_MINOR
@@ -22560,6 +22574,7 @@
 	Win32*)
 		case "$osname" in
 		cygwin) avail_ext="$avail_ext $xxx" ;;
+		msys) avail_ext="$avail_ext $xxx" ;;
 		esac
 		;;
 	XS/APItest|xs/apitest)
diff -Naur perl-5.18.0-orig/cpan/Archive-Extract/t/01_Archive-Extract.t perl-5.18.0/cpan/Archive-Extract/t/01_Archive-Extract.t
--- perl-5.18.0-orig/cpan/Archive-Extract/t/01_Archive-Extract.t	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/Archive-Extract/t/01_Archive-Extract.t	2013-06-15 18:06:59.160156300 +0000
@@ -8,7 +8,7 @@
 use lib qw[../lib];
 
 use constant IS_WIN32   => $^O eq 'MSWin32' ? 1 : 0;
-use constant IS_CYGWIN  => $^O eq 'cygwin'  ? 1 : 0;
+use constant IS_CYGWIN  => ($^O eq 'cygwin' or $^O eq 'msys')  ? 1 : 0;
 use constant IS_VMS     => $^O eq 'VMS'     ? 1 : 0;
 
 use Cwd                         qw[cwd];
diff -Naur perl-5.18.0-orig/cpan/Archive-Tar/t/02_methods.t perl-5.18.0/cpan/Archive-Tar/t/02_methods.t
--- perl-5.18.0-orig/cpan/Archive-Tar/t/02_methods.t	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/Archive-Tar/t/02_methods.t	2013-06-15 18:06:59.160156300 +0000
@@ -60,7 +60,7 @@
 my $LONG_FILE = qq[directory/really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-really-long-directory-name/myfile];
 
 ### wintendo can't deal with too long paths, so we might have to skip tests ###
-my $TOO_LONG    =   ($^O eq 'MSWin32' or $^O eq 'cygwin' or $^O eq 'VMS')
+my $TOO_LONG    =   ($^O eq 'MSWin32' or $^O eq 'cygwin' or $^O eq 'msys' or $^O eq 'VMS')
                     && length( cwd(). $LONG_FILE ) > 247;
 
 if(!$TOO_LONG) {
diff -Naur perl-5.18.0-orig/cpan/CGI/lib/CGI.pm perl-5.18.0/cpan/CGI/lib/CGI.pm
--- perl-5.18.0-orig/cpan/CGI/lib/CGI.pm	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CGI/lib/CGI.pm	2013-06-15 18:06:59.175781300 +0000
@@ -155,6 +155,8 @@
     $OS = 'EPOC';
 } elsif ($OS =~ /^cygwin/i) {
     $OS = 'CYGWIN';
+} elsif ($OS =~ /^msys/i) {
+    $OS = 'MSYS';
 } elsif ($OS =~ /^NetWare/i) {
     $OS = 'NETWARE';
 } else {
@@ -162,7 +164,7 @@
 }
 
 # Some OS logic.  Binary mode enabled on DOS, NT and VMS
-$needs_binmode = $OS=~/^(WINDOWS|DOS|OS2|MSWin|CYGWIN|NETWARE)/;
+$needs_binmode = $OS=~/^(WINDOWS|DOS|OS2|MSWin|CYGWIN|MSYS|NETWARE)/;
 
 # This is the default class for the CGI object to use when all else fails.
 $DefaultClass = 'CGI' unless defined $CGI::DefaultClass;
@@ -173,7 +175,7 @@
 # The path separator is a slash, backslash or semicolon, depending
 # on the paltform.
 $SL = {
-     UNIX    => '/',  OS2 => '\\', EPOC      => '/', CYGWIN => '/', NETWARE => '/',
+     UNIX    => '/',  OS2 => '\\', EPOC      => '/', CYGWIN => '/', MSYS => '/', NETWARE => '/',
      WINDOWS => '\\', DOS => '\\', MACINTOSH => ':', VMS    => '/'
     }->{$OS};
 
diff -Naur perl-5.18.0-orig/cpan/Compress-Raw-Zlib/Makefile.PL perl-5.18.0/cpan/Compress-Raw-Zlib/Makefile.PL
--- perl-5.18.0-orig/cpan/Compress-Raw-Zlib/Makefile.PL	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/Compress-Raw-Zlib/Makefile.PL	2013-06-15 18:06:59.191406300 +0000
@@ -23,7 +23,7 @@
 # ExtUtils::Install.
 
 # Don't ask if MM_USE_DEFAULT is set -- enables perl core building on cygwin
-if ($^O =~ /cygwin/i and $ExtUtils::Install::VERSION < 1.39 
+if (($^O =~ /cygwin/i or $^O =~ /msys/i) and $ExtUtils::Install::VERSION < 1.39
         and not ($ENV{PERL_MM_USE_DEFAULT} or $ENV{PERL_CORE}))
 {
     print <<EOM ;
diff -Naur perl-5.18.0-orig/cpan/CPAN-Meta/t/data/META-1_4.yml perl-5.18.0/cpan/CPAN-Meta/t/data/META-1_4.yml
--- perl-5.18.0-orig/cpan/CPAN-Meta/t/data/META-1_4.yml	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CPAN-Meta/t/data/META-1_4.yml	2013-06-15 18:06:59.191406300 +0000
@@ -77,6 +77,9 @@
   Module::Build::Platform::cygwin:
     file: lib/Module/Build/Platform/cygwin.pm
     version: 0.36_08
+  Module::Build::Platform::msys:
+    file: lib/Module/Build/Platform/cygwin.pm
+    version: 0.36_08
   Module::Build::Platform::darwin:
     file: lib/Module/Build/Platform/darwin.pm
     version: 0.36_08
diff -Naur perl-5.18.0-orig/cpan/CPAN-Meta/t/data/restrictive-1_4.yml perl-5.18.0/cpan/CPAN-Meta/t/data/restrictive-1_4.yml
--- perl-5.18.0-orig/cpan/CPAN-Meta/t/data/restrictive-1_4.yml	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CPAN-Meta/t/data/restrictive-1_4.yml	2013-06-15 18:06:59.191406300 +0000
@@ -77,6 +77,9 @@
   Module::Build::Platform::cygwin:
     file: lib/Module/Build/Platform/cygwin.pm
     version: 0.36_08
+  Module::Build::Platform::msys:
+    file: lib/Module/Build/Platform/cygwin.pm
+    version: 0.36_08
   Module::Build::Platform::darwin:
     file: lib/Module/Build/Platform/darwin.pm
     version: 0.36_08
diff -Naur perl-5.18.0-orig/cpan/CPAN-Meta/t/data/unicode.yml perl-5.18.0/cpan/CPAN-Meta/t/data/unicode.yml
--- perl-5.18.0-orig/cpan/CPAN-Meta/t/data/unicode.yml	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CPAN-Meta/t/data/unicode.yml	2013-06-15 18:36:47.273437500 +0000
@@ -77,6 +77,9 @@
   Module::Build::Platform::cygwin:
     file: lib/Module/Build/Platform/cygwin.pm
     version: 0.36_08
+  Module::Build::Platform::msys:
+    file: lib/Module/Build/Platform/cygwin.pm
+    version: 0.36_08
   Module::Build::Platform::darwin:
     file: lib/Module/Build/Platform/darwin.pm
     version: 0.36_08
diff -Naur perl-5.18.0-orig/cpan/CPAN-Meta/t/data-bad/META-1_4.yml perl-5.18.0/cpan/CPAN-Meta/t/data-bad/META-1_4.yml
--- perl-5.18.0-orig/cpan/CPAN-Meta/t/data-bad/META-1_4.yml	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CPAN-Meta/t/data-bad/META-1_4.yml	2013-06-15 18:06:59.191406300 +0000
@@ -76,6 +76,9 @@
   Module::Build::Platform::cygwin:
     file: lib/Module/Build/Platform/cygwin.pm
     version: 0.36_08
+  Module::Build::Platform::msys:
+    file: lib/Module/Build/Platform/cygwin.pm
+    version: 0.36_08
   Module::Build::Platform::darwin:
     file: lib/Module/Build/Platform/darwin.pm
     version: 0.36_08
diff -Naur perl-5.18.0-orig/cpan/CPAN-Meta/t/data-fail/META-1_4.yml perl-5.18.0/cpan/CPAN-Meta/t/data-fail/META-1_4.yml
--- perl-5.18.0-orig/cpan/CPAN-Meta/t/data-fail/META-1_4.yml	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CPAN-Meta/t/data-fail/META-1_4.yml	2013-06-15 18:06:59.207031300 +0000
@@ -77,6 +77,9 @@
   Module::Build::Platform::cygwin:
     file: lib/Module/Build/Platform/cygwin.pm
     version: 0.36_08
+  Module::Build::Platform::msys:
+    file: lib/Module/Build/Platform/cygwin.pm
+    version: 0.36_08
   Module::Build::Platform::darwin:
     file: lib/Module/Build/Platform/darwin.pm
     version: 0.36_08
diff -Naur perl-5.18.0-orig/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants/Report.pm perl-5.18.0/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants/Report.pm
--- perl-5.18.0-orig/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants/Report.pm	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants/Report.pm	2013-06-15 18:06:59.207031300 +0000
@@ -27,11 +27,12 @@
     Be          => 'beos',
     BeOS        => 'beos',
     Cygwin      => 'cygwin',
+    Msys        => 'msys',
     Darwin      => 'darwin',
     EBCDIC      => 'os390|os400|posix-bc|vmesa',
     HPUX        => 'hpux',
     Linux       => 'linux',
-    MSDOS       => 'dos|os2|MSWin32|cygwin',
+    MSDOS       => 'dos|os2|MSWin32|cygwin|msys',
     'bin\\d*Mac'=> 'MacOS|darwin', # binMac, bin56Mac, bin58Mac...
     Mac         => 'MacOS|darwin',
     MacPerl     => 'MacOS',
@@ -49,8 +50,8 @@
                    'svr4|sco_sv|unicos|unicosmk|solaris|sunos',
     VMS         => 'VMS',
     VOS         => 'VOS',
-    Win32       => 'MSWin32|cygwin',
-    Win32API    => 'MSWin32|cygwin',
+    Win32       => 'MSWin32|cygwin|msys',
+    Win32API    => 'MSWin32|cygwin|msys',
 );
 
 use constant GRADE_FAIL     => 'fail';
diff -Naur perl-5.18.0-orig/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm perl-5.18.0/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm
--- perl-5.18.0-orig/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm	2013-06-15 18:06:59.207031300 +0000
@@ -48,7 +48,7 @@
 
 use constant ON_WIN32       => $^O eq 'MSWin32';
 use constant ON_NETWARE     => $^O eq 'NetWare';
-use constant ON_CYGWIN      => $^O eq 'cygwin';
+use constant ON_CYGWIN      => ($^O eq 'cygwin' or $^O eq 'msys');
 use constant ON_VMS         => $^O eq 'VMS';
 
 use constant DOT_CPANPLUS   => ON_VMS ? '_cpanplus' : '.cpanplus';
diff -Naur perl-5.18.0-orig/cpan/CPANPLUS/lib/CPANPLUS/Internals/Report.pm perl-5.18.0/cpan/CPANPLUS/lib/CPANPLUS/Internals/Report.pm
--- perl-5.18.0-orig/cpan/CPANPLUS/lib/CPANPLUS/Internals/Report.pm	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/CPANPLUS/lib/CPANPLUS/Internals/Report.pm	2013-06-15 18:06:59.222656300 +0000
@@ -136,7 +136,7 @@
           {
             'grade' => 'FAIL',
             'dist' => 'CPANPLUS-0.042',
-            'platform' => 'cygwin-multi-64int',
+            'platform' => 'msys-multi-64int',
             'details' => 'http://nntp.x.perl.org/group/perl.cpan.testers/99371'
             ...
           },
diff -Naur perl-5.18.0-orig/cpan/DB_File/t/db-btree.t perl-5.18.0/cpan/DB_File/t/db-btree.t
--- perl-5.18.0-orig/cpan/DB_File/t/db-btree.t	2013-05-18 01:28:39.000000000 +0000
+++ perl-5.18.0/cpan/DB_File/t/db-btree.t	2013-06-15 18:06:59.222656300 +0000
@@ -105,7 +105,7 @@
 {
     my $data = shift ;
     $data =~ s#\r\n#\n#g 
-        if $^O eq 'cygwin' ;
+        if ($^O eq 'cygwin' or $^O eq 'msys') ;
 
     return $data ;
 }
@@ -175,7 +175,7 @@
 my ($dev,$ino,$mode,$nlink,$uid,$gid,$rdev,$size,$atime,$mtime,$ctime,
    $blksize,$blocks) = stat($Dfile);
 
-my %noMode = map { $_, 1} qw( amigaos MSWin32 NetWare cygwin ) ;
+my %noMode = map { $_, 1} qw( amigaos MSWin32 NetWare cygwin msys ) ;
 
 ok(18, ($mode & 0777) == (($^O eq 'os2' || $^O eq 'MacOS') ? 0666 : 0640)
    || $noMode{$^O} );
diff -Naur perl-5.18.0-orig/cpan/DB_File/t/db-hash.t perl-5.18.0/cpan/DB_File/t/db-hash.t
--- perl-5.18.0-orig/cpan/DB_File/t/db-hash.t	2013-05-18 01:28:39.000000000 +0000
+++ perl-5.18.0/cpan/DB_File/t/db-hash.t	2013-06-15 18:06:59.222656300 +0000
@@ -69,7 +69,7 @@
 {
     my $data = shift ;
     $data =~ s#\r\n#\n#g 
-        if $^O eq 'cygwin' ;
+        if ($^O eq 'cygwin' or $^O eq 'msys') ;
     return $data ;
 }
 
@@ -137,7 +137,7 @@
 my ($dev,$ino,$mode,$nlink,$uid,$gid,$rdev,$size,$atime,$mtime,$ctime,
    $blksize,$blocks) = stat($Dfile);
 
-my %noMode = map { $_, 1} qw( amigaos MSWin32 NetWare cygwin ) ;
+my %noMode = map { $_, 1} qw( amigaos MSWin32 NetWare cygwin msys ) ;
 
 ok(16, ($mode & 0777) == (($^O eq 'os2' || $^O eq 'MacOS') ? 0666 : 0640) ||
    $noMode{$^O} );
diff -Naur perl-5.18.0-orig/cpan/DB_File/t/db-recno.t perl-5.18.0/cpan/DB_File/t/db-recno.t
--- perl-5.18.0-orig/cpan/DB_File/t/db-recno.t	2013-05-18 01:28:39.000000000 +0000
+++ perl-5.18.0/cpan/DB_File/t/db-recno.t	2013-06-15 18:06:59.238281300 +0000
@@ -125,7 +125,7 @@
 
 sub normalise
 {
-    return unless $^O eq 'cygwin' ;
+    return unless ($^O eq 'cygwin' || $^O eq 'msys') ;
     foreach (@_)
       { s#\r\n#\n#g }     
 }
@@ -197,7 +197,7 @@
 my @h ;
 ok(17, $X = tie @h, 'DB_File', $Dfile, O_RDWR|O_CREAT, 0640, $DB_RECNO ) ;
 
-my %noMode = map { $_, 1} qw( amigaos MSWin32 NetWare cygwin ) ;
+my %noMode = map { $_, 1} qw( amigaos MSWin32 NetWare cygwin msys ) ;
 
 ok(18, ((stat($Dfile))[2] & 0777) == (($^O eq 'os2' || $^O eq 'MacOS') ? 0666 : 0640)
 	||  $noMode{$^O} );
diff -Naur perl-5.18.0-orig/cpan/Digest-SHA/shasum perl-5.18.0/cpan/Digest-SHA/shasum
--- perl-5.18.0-orig/cpan/Digest-SHA/shasum	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/Digest-SHA/shasum	2013-06-15 18:06:59.238281300 +0000
@@ -184,7 +184,7 @@
 	## explicitly overriden by command line "--text" or
 	## "--portable" options.
 
-my $isDOSish = ($^O =~ /^(MSWin\d\d|os2|dos|mint|cygwin)$/);
+my $isDOSish = ($^O =~ /^(MSWin\d\d|os2|dos|mint|cygwin|msys)$/);
 if ($isDOSish) { $binary = 1 unless $text || $portable }
 
 my $modesym = $binary ? '*' : ($portable ? '?' : ($BITS ? '^' : ' '));
diff -Naur perl-5.18.0-orig/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM.pm perl-5.18.0/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM.pm
--- perl-5.18.0-orig/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM.pm	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM.pm	2013-06-15 18:06:59.238281300 +0000
@@ -57,7 +57,7 @@
     _is_win95() ? $Is{Win95} = 1 : $Is{Win32} = 1;
 }
 $Is{UWIN}   = $^O =~ /^uwin(-nt)?$/;
-$Is{Cygwin} = $^O eq 'cygwin';
+$Is{Cygwin} = ($^O eq 'cygwin' or $^O eq 'msys');
 $Is{NW5}    = $Config{osname} eq 'NetWare';  # intentional
 $Is{BeOS}   = ($^O =~ /beos/i or $^O eq 'haiku');
 $Is{DOS}    = $^O eq 'dos';
diff -Naur perl-5.18.0-orig/cpan/ExtUtils-MakeMaker/t/MM_Cygwin.t perl-5.18.0/cpan/ExtUtils-MakeMaker/t/MM_Cygwin.t
--- perl-5.18.0-orig/cpan/ExtUtils-MakeMaker/t/MM_Cygwin.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/ExtUtils-MakeMaker/t/MM_Cygwin.t	2013-06-15 18:06:59.253906300 +0000
@@ -9,7 +9,7 @@
 use Test::More;
 
 BEGIN {
-	if ($^O =~ /cygwin/i) {
+	if ($^O =~ /cygwin|msys/i) {
 		plan tests => 14;
 	} else {
 		plan skip_all => "This is not cygwin";
diff -Naur perl-5.18.0-orig/cpan/ExtUtils-MakeMaker/t/MM_Unix.t perl-5.18.0/cpan/ExtUtils-MakeMaker/t/MM_Unix.t
--- perl-5.18.0-orig/cpan/ExtUtils-MakeMaker/t/MM_Unix.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/ExtUtils-MakeMaker/t/MM_Unix.t	2013-06-15 18:06:59.253906300 +0000
@@ -8,7 +8,7 @@
 BEGIN { 
     use Test::More; 
 
-    if( $^O =~ /^VMS|os2|MacOS|MSWin32|cygwin|beos|netware$/i ) {
+    if( $^O =~ /^VMS|os2|MacOS|MSWin32|cygwin|msys|beos|netware$/i ) {
         plan skip_all => 'Non-Unix platform';
     }
     else {
diff -Naur perl-5.18.0-orig/cpan/File-Temp/lib/File/Temp.pm perl-5.18.0/cpan/File-Temp/lib/File/Temp.pm
--- perl-5.18.0-orig/cpan/File-Temp/lib/File/Temp.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/File-Temp/lib/File/Temp.pm	2013-06-15 18:14:54.062500000 +0000
@@ -828,7 +828,7 @@
   return 1 if $level == STANDARD;
 
   # Currently, the systems that can do HIGH or MEDIUM are identical
-  if ( $^O eq 'MSWin32' || $^O eq 'os2' || $^O eq 'cygwin' || $^O eq 'dos' || $^O eq 'MacOS' || $^O eq 'mpeix') {
+  if ( $^O eq 'MSWin32' || $^O eq 'os2' || $^O eq 'cygwin' || $^O eq 'msys' || $^O eq 'dos' || $^O eq 'MacOS' || $^O eq 'mpeix') {
     return 0;
   } else {
     return 1;
@@ -2051,7 +2051,7 @@
     # On NFS the link count may still be 1 but we can't know that
     # we are on NFS.  Since we can't be sure, we'll defer it
 
-    return 1 if $fh[3] == 0 || $^O eq 'cygwin';
+    return 1 if $fh[3] == 0 || $^O eq 'cygwin' || $^O eq 'msys';
   }
   # fall-through if we can't unlink now
   _deferred_unlink($fh, $path, 0);
diff -Naur perl-5.18.0-orig/cpan/IO-Compress/t/cz-14gzopen.t perl-5.18.0/cpan/IO-Compress/t/cz-14gzopen.t
--- perl-5.18.0-orig/cpan/IO-Compress/t/cz-14gzopen.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/IO-Compress/t/cz-14gzopen.t	2013-06-15 18:38:48.914062500 +0000
@@ -490,7 +490,7 @@
     SKIP:
     {
         skip "Cannot create non-writable file", 3 
-            if $^O eq 'cygwin';
+            if ($^O eq 'cygwin' or $^O eq 'msys');
 
         my $lex = new LexFile my $name ;
         writeFile($name, "abc");
@@ -514,7 +514,7 @@
     {
         my $lex = new LexFile my $name ;
         skip "Cannot create non-readable file", 3 
-            if $^O eq 'cygwin';
+            if ($^O eq 'cygwin' or $^O eq 'msys');
 
         writeFile($name, "abc");
         chmod 0222, $name ;
diff -Naur perl-5.18.0-orig/cpan/IPC-SysV/t/ipcsysv.t perl-5.18.0/cpan/IPC-SysV/t/ipcsysv.t
--- perl-5.18.0-orig/cpan/IPC-SysV/t/ipcsysv.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/IPC-SysV/t/ipcsysv.t	2013-06-15 18:07:54.982421900 +0000
@@ -45,18 +45,18 @@
   {
     return if $did_diag++;
 
-    if ($^O eq 'cygwin') {
+    if ($^O eq 'cygwin' or $^O eq 'msys') {
       diag(<<EOM);
 
 It may be that the cygserver service isn't running.
 
 EOM
 
-      diag(<<EOM) unless exists $ENV{CYGWIN} && $ENV{CYGWIN} eq 'server';
+      diag(<<EOM) unless exists $ENV{MSYS} && $ENV{MSYS} eq 'server';
 You also may have to set the CYGWIN environment variable
 to 'server' before running the test suite:
 
-  export CYGWIN=server
+  export MSYS=server
 
 EOM
     }
diff -Naur perl-5.18.0-orig/cpan/libnet/Net/Domain.pm perl-5.18.0/cpan/libnet/Net/Domain.pm
--- perl-5.18.0-orig/cpan/libnet/Net/Domain.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/libnet/Net/Domain.pm	2013-06-15 18:07:54.982421900 +0000
@@ -169,7 +169,7 @@
     }
 
     chop($dom = `domainname 2>/dev/null`)
-      unless (defined $dom || $^O =~ /^(?:cygwin|MSWin32)/);
+      unless (defined $dom || $^O =~ /^(?:cygwin|msys|MSWin32)/);
 
     if (defined $dom) {
       my @h = ();
diff -Naur perl-5.18.0-orig/cpan/libnet/Net/Netrc.pm perl-5.18.0/cpan/libnet/Net/Netrc.pm
--- perl-5.18.0-orig/cpan/libnet/Net/Netrc.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/libnet/Net/Netrc.pm	2013-06-15 18:07:54.982421900 +0000
@@ -43,7 +43,8 @@
   unless ($^O eq 'os2'
     || $^O eq 'MSWin32'
     || $^O eq 'MacOS'
-    || $^O =~ /^cygwin/)
+    || $^O =~ /^cygwin/
+    || $^O =~ /^msys/)
   {
     my @stat = stat($file);
 
diff -Naur perl-5.18.0-orig/cpan/libnet/t/netrc.t perl-5.18.0/cpan/libnet/t/netrc.t
--- perl-5.18.0-orig/cpan/libnet/t/netrc.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/libnet/t/netrc.t	2013-06-15 18:07:54.998046900 +0000
@@ -47,7 +47,7 @@
 
 SKIP: {
 	skip('incompatible stat() handling for OS', 4), next SKIP 
-		if ($^O =~ /os2|win32|macos|cygwin/i or $] < 5.005);
+		if ($^O =~ /os2|win32|macos|cygwin|msys/i or $] < 5.005);
 	
 	my $warn;
 	local $SIG{__WARN__} = sub {
diff -Naur perl-5.18.0-orig/cpan/Module-Build/t/destinations.t perl-5.18.0/cpan/Module-Build/t/destinations.t
--- perl-5.18.0-orig/cpan/Module-Build/t/destinations.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Module-Build/t/destinations.t	2013-06-15 18:07:54.998046900 +0000
@@ -300,11 +300,11 @@
 sub have_same_ending {
   my ($dir1, $dir2, $message) = @_;
 
-  $dir1 =~ s{/$}{} if $^O eq 'cygwin'; # remove any trailing slash
+  $dir1 =~ s{/$}{} if ($^O eq 'cygwin' or $^O eq 'msys'); # remove any trailing slash
   my (undef, $dirs1, undef) = splitpath $dir1;
   my @dir1 = splitdir $dirs1;
 
-  $dir2 =~ s{/$}{} if $^O eq 'cygwin'; # remove any trailing slash
+  $dir2 =~ s{/$}{} if ($^O eq 'cygwin' or $^O eq 'msys'); # remove any trailing slash
   my (undef, $dirs2, undef) = splitpath $dir2;
   my @dir2 = splitdir $dirs2;
 
diff -Naur perl-5.18.0-orig/cpan/Perl-OSType/lib/Perl/OSType.pm perl-5.18.0/cpan/Perl-OSType/lib/Perl/OSType.pm
--- perl-5.18.0-orig/cpan/Perl-OSType/lib/Perl/OSType.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Perl-OSType/lib/Perl/OSType.pm	2013-06-15 18:07:54.998046900 +0000
@@ -44,6 +44,7 @@
   solaris     Unix
   sunos       Unix
   cygwin      Unix
+  msys        Unix
   os2         Unix
   interix     Unix
   gnu         Unix
diff -Naur perl-5.18.0-orig/cpan/Pod-Perldoc/lib/Pod/Perldoc/BaseTo.pm perl-5.18.0/cpan/Pod-Perldoc/lib/Pod/Perldoc/BaseTo.pm
--- perl-5.18.0-orig/cpan/Pod-Perldoc/lib/Pod/Perldoc/BaseTo.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Pod-Perldoc/lib/Pod/Perldoc/BaseTo.pm	2013-06-15 18:40:37.039062500 +0000
@@ -29,7 +29,7 @@
  *is_mswin32 = $^O eq 'MSWin32'  ? \&TRUE : \&FALSE unless defined &is_mswin32;
  *is_dos     = $^O eq 'dos'      ? \&TRUE : \&FALSE unless defined &is_dos;
  *is_os2     = $^O eq 'os2'      ? \&TRUE : \&FALSE unless defined &is_os2;
- *is_cygwin  = $^O eq 'cygwin'   ? \&TRUE : \&FALSE unless defined &is_cygwin;
+ *is_cygwin  = ($^O eq 'cygwin' || $^O eq 'msys')   ? \&TRUE : \&FALSE unless defined &is_cygwin;
  *is_linux   = $^O eq 'linux'    ? \&TRUE : \&FALSE unless defined &is_linux;
  *is_hpux    = $^O =~ m/hpux/    ? \&TRUE : \&FALSE unless defined &is_hpux;
  *is_openbsd = $^O =~ m/openbsd/ ? \&TRUE : \&FALSE unless defined &is_openbsd;
diff -Naur perl-5.18.0-orig/cpan/Pod-Perldoc/lib/Pod/Perldoc.pm perl-5.18.0/cpan/Pod-Perldoc/lib/Pod/Perldoc.pm
--- perl-5.18.0-orig/cpan/Pod-Perldoc/lib/Pod/Perldoc.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Pod-Perldoc/lib/Pod/Perldoc.pm	2013-06-15 18:11:18.036132800 +0000
@@ -66,7 +66,7 @@
  *is_mswin32 = $^O eq 'MSWin32' ? \&TRUE : \&FALSE unless defined &is_mswin32;
  *is_dos     = $^O eq 'dos'     ? \&TRUE : \&FALSE unless defined &is_dos;
  *is_os2     = $^O eq 'os2'     ? \&TRUE : \&FALSE unless defined &is_os2;
- *is_cygwin  = $^O eq 'cygwin'  ? \&TRUE : \&FALSE unless defined &is_cygwin;
+ *is_cygwin  = ($^O eq 'cygwin' || $^O eq 'msys')  ? \&TRUE : \&FALSE unless defined &is_cygwin;
  *is_linux   = $^O eq 'linux'   ? \&TRUE : \&FALSE unless defined &is_linux;
  *is_hpux    = $^O =~ m/hpux/   ? \&TRUE : \&FALSE unless defined &is_hpux;
 }
diff -Naur perl-5.18.0-orig/cpan/Sys-Syslog/t/syslog.t perl-5.18.0/cpan/Sys-Syslog/t/syslog.t
--- perl-5.18.0-orig/cpan/Sys-Syslog/t/syslog.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Sys-Syslog/t/syslog.t	2013-06-15 18:07:55.357421900 +0000
@@ -17,7 +17,7 @@
 $^W = 0 if $] < 5.006;
 
 my $is_Win32  = $^O =~ /win32/i;
-my $is_Cygwin = $^O =~ /cygwin/i;
+my $is_Cygwin = $^O =~ /cygwin|msys/i;
 
 # if testing in core, check that the module is at least available
 if ($ENV{PERL_CORE}) {
diff -Naur perl-5.18.0-orig/cpan/Sys-Syslog/win32/compile.pl perl-5.18.0/cpan/Sys-Syslog/win32/compile.pl
--- perl-5.18.0-orig/cpan/Sys-Syslog/win32/compile.pl	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Sys-Syslog/win32/compile.pl	2013-06-15 18:07:55.357421900 +0000
@@ -127,7 +127,7 @@
     ),
 );    
 
-my $is_Cygwin = $^O =~ /Cygwin/i;
+my $is_Cygwin = $^O =~ /Cygwin|Msys/i;
 my $is_Win32  = $^O =~ /Win32/i;
 
 my %const = (
diff -Naur perl-5.18.0-orig/cpan/Sys-Syslog/win32/Win32.pm perl-5.18.0/cpan/Sys-Syslog/win32/Win32.pm
--- perl-5.18.0-orig/cpan/Sys-Syslog/win32/Win32.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Sys-Syslog/win32/Win32.pm	2013-06-15 18:07:55.373046900 +0000
@@ -39,7 +39,7 @@
     ),
 );    
 
-my $is_Cygwin = $^O =~ /Cygwin/i;
+my $is_Cygwin = $^O =~ /Cygwin|Msys/i;
 my $is_Win32  = $^O =~ /Win32/i;
 
 my %const = (
diff -Naur perl-5.18.0-orig/cpan/Time-HiRes/Makefile.PL perl-5.18.0/cpan/Time-HiRes/Makefile.PL
--- perl-5.18.0-orig/cpan/Time-HiRes/Makefile.PL	2013-05-05 19:32:18.000000000 +0000
+++ perl-5.18.0/cpan/Time-HiRes/Makefile.PL	2013-06-15 18:07:55.373046900 +0000
@@ -21,7 +21,7 @@
 
 # TBD: Can we just use $Config(exe_ext) here instead of this complex
 #      expression?
-my $ld_exeext = ($^O eq 'cygwin' ||
+my $ld_exeext = ($^O eq 'cygwin' || $^O eq 'msys' ||
                  $^O eq 'os2' && $Config{ldflags} =~ /-Zexe\b/) ? '.exe' :
                 (($^O eq 'vos') ? $Config{exe_ext} : '');
 
@@ -717,7 +717,7 @@
 
     my $has_w32api_windows_h;
 
-    if ($^O eq 'cygwin') {
+    if ($^O eq 'cygwin' or $^O eq 'msys') {
         print "Looking for <w32api/windows.h>... ";
         if (has_include('w32api/windows.h')) {
 	    $has_w32api_windows_h++;
diff -Naur perl-5.18.0-orig/cpan/Time-HiRes/t/stat.t perl-5.18.0/cpan/Time-HiRes/t/stat.t
--- perl-5.18.0-orig/cpan/Time-HiRes/t/stat.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Time-HiRes/t/stat.t	2013-06-15 18:41:37.007812500 +0000
@@ -6,7 +6,7 @@
 	require Test::More;
 	Test::More::plan(skip_all => "no hi-res stat");
     }
-    if($^O =~ /\A(?:cygwin|MSWin)/) {
+    if($^O =~ /\A(?:cygwin|msys|MSWin)/) {
 	require Test::More;
 	Test::More::plan(skip_all =>
 		"$^O file timestamps not reliable enough for stat test");
diff -Naur perl-5.18.0-orig/cpan/Win32/Makefile.PL perl-5.18.0/cpan/Win32/Makefile.PL
--- perl-5.18.0-orig/cpan/Win32/Makefile.PL	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Win32/Makefile.PL	2013-06-15 18:17:32.865234400 +0000
@@ -3,7 +3,7 @@
 use warnings;
 use ExtUtils::MakeMaker;
 
-unless ($^O eq "MSWin32" || $^O eq "cygwin") {
+unless ($^O eq "MSWin32" || $^O eq "cygwin" || $^O eq "msys") {
     die "OS unsupported\n";
 }
 
@@ -13,6 +13,6 @@
     INSTALLDIRS   => ($] >= 5.008004 && $] < 5.012 ? 'perl' : 'site'),
 );
 $param{NO_META} = 1 if eval "$ExtUtils::MakeMaker::VERSION" >= 6.10_03;
-$param{LIBS} = ['-L/lib/w32api -lole32 -lversion'] if $^O eq "cygwin";
+$param{LIBS} = ['-L/lib/w32api -lole32 -lversion'] if ($^O eq "cygwin" or $^O eq "msys");
 
 WriteMakefile(%param);
diff -Naur perl-5.18.0-orig/cpan/Win32/t/GetCurrentThreadId.t perl-5.18.0/cpan/Win32/t/GetCurrentThreadId.t
--- perl-5.18.0-orig/cpan/Win32/t/GetCurrentThreadId.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Win32/t/GetCurrentThreadId.t	2013-06-15 18:18:30.749023500 +0000
@@ -10,7 +10,7 @@
 
 my $pid = $$+0; # make sure we don't copy any magic to $pid
 
-if ($^O eq "cygwin") {
+if ($^O eq "cygwin" or $^O eq "msys") {
     skip(!defined &Cygwin::pid_to_winpid,
 	 Cygwin::pid_to_winpid($pid),
 	 Win32::GetCurrentProcessId());
diff -Naur perl-5.18.0-orig/cpan/Win32/t/Unicode.t perl-5.18.0/cpan/Win32/t/Unicode.t
--- perl-5.18.0-orig/cpan/Win32/t/Unicode.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Win32/t/Unicode.t	2013-06-15 18:20:17.358398500 +0000
@@ -52,7 +52,7 @@
     next if /^\./;
     # On Cygwin 1.7 readdir() returns the utf8 representation of the
     # filename but doesn't turn on the SvUTF8 bit
-    Encode::_utf8_on($_) if $^O eq "cygwin" && $Config{osvers} !~ /^1.5/;
+    Encode::_utf8_on($_) if ($^O eq "cygwin" or $^O eq "msys") && $Config{osvers} !~ /^1.5/;
     ok($file, Win32::GetLongPathName("$dir\\$_"));
 }
 closedir($dh);
@@ -79,7 +79,7 @@
 
 # cwd() on Cygwin returns a mapped path that we need to translate
 # back to a Windows path. Invoking `cygpath` on $subdir doesn't work.
-if ($^O eq "cygwin") {
+if ($^O eq "cygwin" or $^O eq "msys") {
     $subdir = Cygwin::posix_to_win_path($subdir, 1);
 }
 $subdir =~ s,/,\\,g;
diff -Naur perl-5.18.0-orig/cpan/Win32API-File/File.pm perl-5.18.0/cpan/Win32API-File/File.pm
--- perl-5.18.0-orig/cpan/Win32API-File/File.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Win32API-File/File.pm	2013-06-15 18:09:05.833984400 +0000
@@ -738,7 +738,7 @@
 
 	return $self->_fileno() if defined $self->_fileno();
 
-	return -1 if $^O eq 'cygwin';
+	return -1 if ($^O eq 'cygwin' or $^O eq 'msys');
 
 # FIXME: We don't always open the handle, better to query the handle or to set
 # the right access info at TIEHANDLE time.
diff -Naur perl-5.18.0-orig/cpan/Win32API-File/Makefile.PL perl-5.18.0/cpan/Win32API-File/Makefile.PL
--- perl-5.18.0-orig/cpan/Win32API-File/Makefile.PL	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Win32API-File/Makefile.PL	2013-06-15 18:09:06.146484400 +0000
@@ -5,7 +5,7 @@
 use strict;
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
-unless ($^O eq "MSWin32" || $^O eq "cygwin" || $^O eq "interix") { #not tested on Interix
+unless ($^O eq "MSWin32" || $^O eq "cygwin" || $^O eq "msys" || $^O eq "interix") { #not tested on Interix
     die "OS unsupported\n";
 }
 
diff -Naur perl-5.18.0-orig/cpan/Win32API-File/t/file.t perl-5.18.0/cpan/Win32API-File/t/file.t
--- perl-5.18.0-orig/cpan/Win32API-File/t/file.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/cpan/Win32API-File/t/file.t	2013-06-15 18:09:06.162109400 +0000
@@ -151,7 +151,7 @@
 $ok or print "# ",fileLastError(),"\n";
 print $ok ? "" : "not ", "ok ", ++$test, "\n";	# ok 19
 
-SetFilePointer($h1, 0, [], FILE_BEGIN) if $^O eq 'cygwin';
+SetFilePointer($h1, 0, [], FILE_BEGIN) if ($^O eq 'cygwin' or $^O eq 'msys');
 
 $ok= ReadFile( $h1, $text, 0, [], [] );
 $ok or print "# ",fileLastError(),"\n";
@@ -166,7 +166,7 @@
 print $ok ? "" : "not ", "ok ", ++$test, "\n";	# ok 21
 
 $skip = "";
-if ($^O eq 'cygwin') {
+if ($^O eq 'cygwin' or $^O eq 'msys') {
     $ok = 1;
     $skip = " # skip cygwin can delete open files";
 }
@@ -247,7 +247,7 @@
 $ok or print "# ",fileLastError(),"\n";
 print $ok ? "" : "not ", "ok ", ++$test, "\n";	# ok 35
 
-if ($^O eq 'cygwin') {
+if ($^O eq 'cygwin' or $^O eq 'msys') {
     chmod( 0200 | 07777 & (stat("Moved.cp"))[2], "Moved.cp" );
 }
 else {
@@ -343,7 +343,7 @@
 print $ok ? "" : "not ", "ok ", ++$test, "\n";	# ok 49
 
 my $path = $ENV{WINDIR};
-$ok= -f $let.substr($path,$^O eq 'cygwin'?2:3)."/win.ini";
+$ok= -f $let.substr($path,($^O eq 'cygwin' || $^O eq 'msys')?2:3)."/win.ini";
 $ok or print "# ",$let.substr($path,3)."/win.ini ",fileLastError(),"\n";
 print $ok ? "" : "not ", "ok ", ++$test, "\n";	# ok 50
 
diff -Naur perl-5.18.0-orig/Cross/Makefile-cross-SH perl-5.18.0/Cross/Makefile-cross-SH
--- perl-5.18.0-orig/Cross/Makefile-cross-SH	2013-05-01 02:52:54.000000000 +0000
+++ perl-5.18.0/Cross/Makefile-cross-SH	2013-06-15 18:09:06.162109400 +0000
@@ -61,7 +61,7 @@
 				${revision}.${patchlevel}.${subversion} \
 			     -install_name \$(shrpdir)/\$@"
 		;;
-	cygwin*)
+	cygwin* | msys*)
 		linklibperl="-lperl"
 		;;
 	sunos*)
@@ -469,7 +469,7 @@
 
 !NO!SUBS!
 case "$osname" in
-cygwin)
+cygwin | msys)
 	;; # Let cygwin/Makefile.SHs do its work.
 *)
 	$spitshell >>$Makefile <<'!NO!SUBS!'
@@ -961,7 +961,7 @@
 	-rm -f perl.third lib*.so.perl.third perl.3log t/perl.third t/perl.3log
 	-rm -f perl.pixie lib*.so.perl.pixie lib*.so.Addrs
 	-rm -f perl.Addrs perl.Counts t/perl.Addrs t/perl.Counts *perl.xok
-	-rm -f cygwin.c libperl*.def libperl*.dll cygperl*.dll *.exe.stackdump
+	-rm -f cygwin.c libperl*.def libperl*.dll msys-perl*.dll *.exe.stackdump
 	-rm -f perl$(EXE_EXT) miniperl$(EXE_EXT) $(LIBPERL) libperl.* microperl
 	-rm -f config.over
 
diff -Naur perl-5.18.0-orig/dist/Cwd/Cwd.pm perl-5.18.0/dist/Cwd/Cwd.pm
--- perl-5.18.0-orig/dist/Cwd/Cwd.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Cwd/Cwd.pm	2013-06-15 18:09:06.162109400 +0000
@@ -306,6 +306,15 @@
     realpath		=> 'fast_abs_path',
    },
 
+   msys =>
+   {
+    getcwd		=> 'cwd',
+    fastgetcwd		=> 'cwd',
+    fastcwd		=> 'cwd',
+    abs_path		=> 'fast_abs_path',
+    realpath		=> 'fast_abs_path',
+   },
+
    epoc =>
    {
     cwd			=> '_epoc_cwd',
@@ -395,7 +404,7 @@
     }
 }
 
-if ($^O eq 'cygwin') {
+if ($^O eq 'cygwin' or $^O eq 'msys') {
   # We need to make sure cwd() is called with no args, because it's
   # got an arg-less prototype and will die if args are present.
   local $^W = 0;
diff -Naur perl-5.18.0-orig/dist/Cwd/lib/File/Spec/Cygwin.pm perl-5.18.0/dist/Cwd/lib/File/Spec/Cygwin.pm
--- perl-5.18.0-orig/dist/Cwd/lib/File/Spec/Cygwin.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Cwd/lib/File/Spec/Cygwin.pm	2013-06-15 18:09:06.177734400 +0000
@@ -113,7 +113,7 @@
 =cut
 
 sub case_tolerant {
-  return 1 unless $^O eq 'cygwin'
+  return 1 unless ($^O eq 'cygwin' || $^O eq 'msys')
     and defined &Cygwin::mount_flags;
 
   my $drive = shift;
diff -Naur perl-5.18.0-orig/dist/Cwd/lib/File/Spec/Unix.pm perl-5.18.0/dist/Cwd/lib/File/Spec/Unix.pm
--- perl-5.18.0-orig/dist/Cwd/lib/File/Spec/Unix.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Cwd/lib/File/Spec/Unix.pm	2013-06-15 18:43:48.007812500 +0000
@@ -57,7 +57,7 @@
       $node = $1;
     }
     # This used to be
-    # $path =~ s|/+|/|g unless ($^O eq 'cygwin');
+    # $path =~ s|/+|/|g unless ($^O eq 'cygwin' or $^O eq 'msys');
     # but that made tests 29, 30, 35, 46, and 213 (as of #13272) to fail
     # (Mainly because trailing "" directories didn't get stripped).
     # Why would cygwin avoid collapsing multiple slashes into one? --jhi
diff -Naur perl-5.18.0-orig/dist/Cwd/lib/File/Spec.pm perl-5.18.0/dist/Cwd/lib/File/Spec.pm
--- perl-5.18.0-orig/dist/Cwd/lib/File/Spec.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Cwd/lib/File/Spec.pm	2013-06-15 18:09:06.177734400 +0000
@@ -14,7 +14,8 @@
 	      NetWare => 'Win32', # Yes, File::Spec::Win32 works on NetWare.
 	      symbian => 'Win32', # Yes, File::Spec::Win32 works on symbian.
 	      dos     => 'OS2',   # Yes, File::Spec::OS2 works on DJGPP.
-	      cygwin  => 'Cygwin');
+	      cygwin  => 'Cygwin',
+          msys    => 'Cygwin');
 
 
 my $module = $module{$^O} || 'Unix';
diff -Naur perl-5.18.0-orig/dist/Cwd/t/crossplatform.t perl-5.18.0/dist/Cwd/t/crossplatform.t
--- perl-5.18.0-orig/dist/Cwd/t/crossplatform.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Cwd/t/crossplatform.t	2013-06-15 18:09:06.177734400 +0000
@@ -6,7 +6,7 @@
 use Test::More;
 local $|=1;
 
-my @platforms = qw(Cygwin Epoc Mac OS2 Unix VMS Win32);
+my @platforms = qw(Cygwin Msys Epoc Mac OS2 Unix VMS Win32);
 my $tests_per_platform = 10;
 
 my $vms_unix_rpt = 0;
diff -Naur perl-5.18.0-orig/dist/Cwd/t/cwd.t perl-5.18.0/dist/Cwd/t/cwd.t
--- perl-5.18.0-orig/dist/Cwd/t/cwd.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Cwd/t/cwd.t	2013-06-15 18:09:06.193359400 +0000
@@ -39,7 +39,7 @@
 my $tests = 31;
 # _perl_abs_path() currently only works when the directory separator
 # is '/', so don't test it when it won't work.
-my $EXTRA_ABSPATH_TESTS = ($Config{prefix} =~ m/\//) && $^O ne 'cygwin';
+my $EXTRA_ABSPATH_TESTS = ($Config{prefix} =~ m/\//) && ($^O ne 'cygwin' || $^O ne 'msys');
 $tests += 4 if $EXTRA_ABSPATH_TESTS;
 plan tests => $tests;
 
diff -Naur perl-5.18.0-orig/dist/Cwd/t/Spec.t perl-5.18.0/dist/Cwd/t/Spec.t
--- perl-5.18.0-orig/dist/Cwd/t/Spec.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Cwd/t/Spec.t	2013-06-15 18:09:06.193359400 +0000
@@ -38,7 +38,7 @@
    $INC{"VMS/Filespec.pm"} = 1 ;
 }
 
-foreach (qw(Unix Win32 VMS OS2 Mac Epoc Cygwin)) {
+foreach (qw(Unix Win32 VMS OS2 Mac Epoc Cygwin Msys)) {
     require_ok("File::Spec::$_");
 }
 
diff -Naur perl-5.18.0-orig/dist/ExtUtils-Command/t/eu_command.t perl-5.18.0/dist/ExtUtils-Command/t/eu_command.t
--- perl-5.18.0-orig/dist/ExtUtils-Command/t/eu_command.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/ExtUtils-Command/t/eu_command.t	2013-06-15 18:48:04.007812500 +0000
@@ -95,7 +95,7 @@
     SKIP: {
         if ($^O eq 'amigaos' || $^O eq 'os2' || $^O eq 'MSWin32' ||
             $^O eq 'NetWare' || $^O eq 'dos' || $^O eq 'cygwin'  ||
-            $^O eq 'MacOS'
+            $^O eq 'msys'    || $^O eq 'MacOS'
            ) {
             skip( "different file permission semantics on $^O", 3);
         }
@@ -135,7 +135,7 @@
     SKIP: {
         if ($^O eq 'amigaos' || $^O eq 'os2' || $^O eq 'MSWin32' ||
             $^O eq 'NetWare' || $^O eq 'dos' || $^O eq 'cygwin'  ||
-            $^O eq 'MacOS'
+            $^O eq 'msys'    || $^O eq 'MacOS'
            ) {
             skip( "different file permission semantics on $^O", 5);
         }
diff -Naur perl-5.18.0-orig/dist/ExtUtils-Install/lib/ExtUtils/Install.pm perl-5.18.0/dist/ExtUtils-Install/lib/ExtUtils/Install.pm
--- perl-5.18.0-orig/dist/ExtUtils-Install/lib/ExtUtils/Install.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/ExtUtils-Install/lib/ExtUtils/Install.pm	2013-06-15 18:09:06.208984400 +0000
@@ -98,7 +98,7 @@
 my $Is_VMS     = $^O eq 'VMS';
 my $Is_MacPerl = $^O eq 'MacOS';
 my $Is_Win32   = $^O eq 'MSWin32';
-my $Is_cygwin  = $^O eq 'cygwin';
+my $Is_cygwin  = ($^O eq 'cygwin' || $^O eq 'msys');
 my $CanMoveAtBoot = ($Is_Win32 || $Is_cygwin);
 
 # *note* CanMoveAtBoot is only incidentally the same condition as below
diff -Naur perl-5.18.0-orig/dist/IO/t/io_xs.t perl-5.18.0/dist/IO/t/io_xs.t
--- perl-5.18.0-orig/dist/IO/t/io_xs.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/IO/t/io_xs.t	2013-06-15 18:48:56.757812500 +0000
@@ -47,6 +47,8 @@
       and skip "fsync() documented to fail on non-writable handles on AIX", 1;
     $^O eq 'cygwin'
       and skip "fsync() on cygwin uses FlushFileBuffers which requires a writable handle", 1;
+    $^O eq 'msys'
+      and skip "fsync() on msys uses FlushFileBuffers which requires a writable handle", 1;
     open my $fh, "<", "t/io_xs.t"
        or skip "Cannot open t/io_xs.t read-only: $!", 1;
     ok($fh->sync, "sync to a read only handle")
diff -Naur perl-5.18.0-orig/dist/Net-Ping/lib/Net/Ping.pm perl-5.18.0/dist/Net-Ping/lib/Net/Ping.pm
--- perl-5.18.0-orig/dist/Net-Ping/lib/Net/Ping.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Net-Ping/lib/Net/Ping.pm	2013-06-15 18:09:06.208984400 +0000
@@ -154,7 +154,7 @@
   }
   elsif ($self->{"proto"} eq "icmp")
   {
-    croak("icmp ping requires root privilege") if ($> and $^O ne 'VMS' and $^O ne 'cygwin');
+    croak("icmp ping requires root privilege") if ($> and $^O ne 'VMS' and ($^O ne 'cygwin' or $^O ne 'msys'));
     $self->{"proto_num"} = (getprotobyname('icmp'))[2] ||
       croak("Can't get icmp protocol by name");
     $self->{"pid"} = $$ & 0xffff;           # Save lower 16 bits of pid
@@ -660,7 +660,7 @@
             # This should set $! to the correct error.
             my $char;
             sysread($self->{"fh"},$char,1);
-            $! = ECONNREFUSED if ($! == EAGAIN && $^O =~ /cygwin/i);
+            $! = ECONNREFUSED if ($! == EAGAIN && $^O =~ /cygwin|msys/i);
 
             $ret = 1 if (!$self->{"econnrefused"}
                          && $! == ECONNREFUSED);
@@ -1250,7 +1250,7 @@
             $self->{"bad"}->{$entry->[0]} = $!;
             if (!$self->{"econnrefused"} &&
                 (($! == ECONNREFUSED) ||
-                 ($! == EAGAIN && $^O =~ /cygwin/i))) {
+                 ($! == EAGAIN && $^O =~ /cygwin|msys/i))) {
               # "Connection refused" means reachable
               # Good, continue
             } else {
diff -Naur perl-5.18.0-orig/dist/Net-Ping/t/110_icmp_inst.t perl-5.18.0/dist/Net-Ping/t/110_icmp_inst.t
--- perl-5.18.0-orig/dist/Net-Ping/t/110_icmp_inst.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Net-Ping/t/110_icmp_inst.t	2013-06-15 18:23:36.546875000 +0000
@@ -15,8 +15,8 @@
 
 SKIP: {
   skip "icmp ping requires root privileges.", 1
-    if ($> and $^O ne 'VMS' and $^O ne 'cygwin')
-      or (($^O eq 'MSWin32' or $^O eq 'cygwin')
+    if ($> and $^O ne 'VMS' and $^O ne 'cygwin' and $^O ne 'msys')
+      or (($^O eq 'MSWin32' or $^O eq 'cygwin' or $^O eq 'msys')
 	  and !IsAdminUser())
 	or ($^O eq 'VMS'
 	    and (`write sys\$output f\$privilege("SYSPRV")` =~ m/FALSE/));
@@ -25,7 +25,7 @@
 }
 
 sub IsAdminUser {
-  return unless $^O eq 'MSWin32' or $^O eq 'cygwin';
+  return unless $^O eq 'MSWin32' or $^O eq 'cygwin' or $^O eq 'msys';
   return unless eval { require Win32 };
   return unless defined &Win32::IsAdminUser;
   return Win32::IsAdminUser();
diff -Naur perl-5.18.0-orig/dist/Net-Ping/t/500_ping_icmp.t perl-5.18.0/dist/Net-Ping/t/500_ping_icmp.t
--- perl-5.18.0-orig/dist/Net-Ping/t/500_ping_icmp.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Net-Ping/t/500_ping_icmp.t	2013-06-15 18:23:15.984375000 +0000
@@ -15,8 +15,8 @@
 
 SKIP: {
   skip "icmp ping requires root privileges.", 1
-    if ($> and $^O ne 'VMS' and $^O ne 'cygwin')
-      or (($^O eq 'MSWin32' or $^O eq 'cygwin')
+    if ($> and $^O ne 'VMS' and $^O ne 'cygwin' and $^O ne 'msys')
+      or (($^O eq 'MSWin32' or $^O eq 'cygwin' or $^O eq 'msys')
 	  and !IsAdminUser())
 	or ($^O eq 'VMS'
 	    and (`write sys\$output f\$privilege("SYSPRV")` =~ m/FALSE/));
@@ -25,7 +25,7 @@
 }
 
 sub IsAdminUser {
-  return unless $^O eq 'MSWin32' or $^O eq "cygwin";
+  return unless $^O eq 'MSWin32' or $^O eq "cygwin" or $^O ne "msys";
   return unless eval { require Win32 };
   return unless defined &Win32::IsAdminUser;
   return Win32::IsAdminUser();
diff -Naur perl-5.18.0-orig/dist/Net-Ping/t/510_ping_udp.t perl-5.18.0/dist/Net-Ping/t/510_ping_udp.t
--- perl-5.18.0-orig/dist/Net-Ping/t/510_ping_udp.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Net-Ping/t/510_ping_udp.t	2013-06-15 18:49:43.617187500 +0000
@@ -3,7 +3,7 @@
 use strict;
 
 sub isWindowsVista {
-   return unless $^O eq 'MSWin32' or $^O eq "cygwin";
+   return unless $^O eq 'MSWin32' or $^O eq "cygwin" or $^O eq "msys";
    return unless eval { require Win32 };
    return unless defined &Win32::GetOSVersion();
 
diff -Naur perl-5.18.0-orig/dist/Net-Ping/t/520_icmp_ttl.t perl-5.18.0/dist/Net-Ping/t/520_icmp_ttl.t
--- perl-5.18.0-orig/dist/Net-Ping/t/520_icmp_ttl.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/dist/Net-Ping/t/520_icmp_ttl.t	2013-06-15 18:50:22.960937500 +0000
@@ -13,8 +13,8 @@
 
 SKIP: {
   skip "icmp ping requires root privileges.", 1
-    if ($> and $^O ne 'VMS' and $^O ne 'cygwin')
-      or (($^O eq 'MSWin32' or $^O eq 'cygwin')
+    if ($> and $^O ne 'VMS' and $^O ne 'cygwin' and $^O ne 'msys')
+      or (($^O eq 'MSWin32' or $^O eq 'cygwin' or $^O eq 'msys')
     and !IsAdminUser())
   or ($^O eq 'VMS'
       and (`write sys\$output f\$privilege("SYSPRV")` =~ m/FALSE/));
@@ -40,7 +40,7 @@
 }
 
 sub IsAdminUser {
-  return unless $^O eq 'MSWin32' or $^O eq "cygwin";
+  return unless $^O eq 'MSWin32' or $^O eq "cygwin" or $^O eq "msys";;
   return unless eval { require Win32 };
   return unless defined &Win32::IsAdminUser;
   return Win32::IsAdminUser();
diff -Naur perl-5.18.0-orig/ext/DynaLoader/DynaLoader_pm.PL perl-5.18.0/ext/DynaLoader/DynaLoader_pm.PL
--- perl-5.18.0-orig/ext/DynaLoader/DynaLoader_pm.PL	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/ext/DynaLoader/DynaLoader_pm.PL	2013-06-15 18:10:32.926757800 +0000
@@ -440,6 +440,9 @@
 	    <<$^O-eq-cygwin>>
             push(@names,"cyg$_.$dl_so")  unless m:/:;
 	    <</$^O-eq-cygwin>>
+	    <<$^O-eq-msys>>
+            push(@names,"msys-$_.$dl_so")  unless m:/:;
+	    <</$^O-eq-msys>>
             push(@names,"lib$_.$dl_so")  unless m:/:;
             push(@names,"$_.a")          if !m/\.a$/ and $dlsrc eq "dl_dld.xs";
             push(@names, $_);
diff -Naur perl-5.18.0-orig/ext/DynaLoader/t/DynaLoader.t perl-5.18.0/ext/DynaLoader/t/DynaLoader.t
--- perl-5.18.0-orig/ext/DynaLoader/t/DynaLoader.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/ext/DynaLoader/t/DynaLoader.t	2013-06-15 18:10:32.942382800 +0000
@@ -112,7 +112,7 @@
     # (not at least by that name) that the dl_findfile()
     # could find.
     skip "dl_findfile test not appropriate on $^O", 1
-	if $^O =~ /(win32|vms|openbsd|cygwin)/i;
+	if $^O =~ /(win32|vms|openbsd|cygwin|msys)/i;
     # Play safe and only try this test if this system
     # looks pretty much Unix-like.
     skip "dl_findfile test not appropriate on $^O", 1
diff -Naur perl-5.18.0-orig/ext/File-Glob/t/basic.t perl-5.18.0/ext/File-Glob/t/basic.t
--- perl-5.18.0-orig/ext/File-Glob/t/basic.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/ext/File-Glob/t/basic.t	2013-06-15 18:25:13.816406300 +0000
@@ -131,7 +131,7 @@
 # should return an empty list, and set ERROR
 SKIP: {
     skip $^O, 2 if $^O eq 'MSWin32' or $^O eq 'NetWare'
-	or $^O eq 'os2' or $^O eq 'VMS' or $^O eq 'cygwin';
+	or $^O eq 'os2' or $^O eq 'VMS' or $^O eq 'cygwin' or $^O eq 'msys';
     skip "AFS", 2 if Cwd::cwd() =~ m#^$Config{'afsroot'}#s;
     skip "running as root", 2 if not $>;
 
diff -Naur perl-5.18.0-orig/ext/POSIX/t/sysconf.t perl-5.18.0/ext/POSIX/t/sysconf.t
--- perl-5.18.0-orig/ext/POSIX/t/sysconf.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/ext/POSIX/t/sysconf.t	2013-06-15 18:10:32.942382800 +0000
@@ -155,7 +155,7 @@
 }
 
 SKIP: {
-    if($^O eq 'cygwin') {
+    if($^O eq 'cygwin' or $^O eq 'msys') {
         pop @sys_consts;
         skip("No _SC_TZNAME_MAX on Cygwin", 1);
     }
diff -Naur perl-5.18.0-orig/ext/POSIX/t/time.t perl-5.18.0/ext/POSIX/t/time.t
--- perl-5.18.0-orig/ext/POSIX/t/time.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/ext/POSIX/t/time.t	2013-06-15 18:10:32.958007800 +0000
@@ -17,7 +17,7 @@
     # actually do anything.  Cygwin works in some places, but not others.  The
     # other Win32's below are guesses.
     skip "No tzset()", 2
-       if $^O eq "MacOS" || $^O eq "VMS" || $^O eq "cygwin" || $^O eq "djgpp" ||
+       if $^O eq "MacOS" || $^O eq "VMS" || $^O eq "cygwin" || $^O eq "msys" || $^O eq "djgpp" ||
           $^O eq "MSWin32" || $^O eq "dos" || $^O eq "interix";
     tzset();
     my @tzname = tzname();
diff -Naur perl-5.18.0-orig/ext/Win32CORE/t/win32core.t perl-5.18.0/ext/Win32CORE/t/win32core.t
--- perl-5.18.0-orig/ext/Win32CORE/t/win32core.t	2013-02-26 21:28:27.000000000 +0000
+++ perl-5.18.0/ext/Win32CORE/t/win32core.t	2013-06-15 18:10:32.958007800 +0000
@@ -20,7 +20,7 @@
 # [perl #42925] - Loading Win32::GetLastError() via the forwarder function
 # should not affect the last error being retrieved
 $^E = 42;
-is(Win32::GetLastError(), $^O eq 'cygwin' ? 0 : 42, 'GetLastError() works on the first call');
+is(Win32::GetLastError(), ($^O eq 'cygwin' || $^O eq 'msys') ? 0 : 42, 'GetLastError() works on the first call');
 
 # Now all Win32::* functions should be loaded
 ok(defined &Win32::ExpandEnvironmentStrings);
diff -Naur perl-5.18.0-orig/haiku/Haiku/Makefile.PL perl-5.18.0/haiku/Haiku/Makefile.PL
--- perl-5.18.0-orig/haiku/Haiku/Makefile.PL	2013-02-26 21:28:27.000000000 +0000
+++ perl-5.18.0/haiku/Haiku/Makefile.PL	2013-06-15 18:10:32.958007800 +0000
@@ -6,7 +6,7 @@
 }
 
 #my @libs;
-#push @libs, '-L/lib/w32api -lole32 -lversion' if $^O eq "cygwin";
+#push @libs, '-L/lib/w32api -lole32 -lversion' if ($^O eq "cygwin" or $^O eq "msys");
 
 WriteMakefile(
     NAME          => 'Haiku',
diff -Naur perl-5.18.0-orig/hints/cygwin.sh perl-5.18.0/hints/cygwin.sh
--- perl-5.18.0-orig/hints/cygwin.sh	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/hints/cygwin.sh	2013-06-15 18:29:08.455078200 +0000
@@ -31,6 +31,7 @@
 man3ext='3pm'
 test -z "$use64bitint" && use64bitint='define'
 test -z "$useithreads" && useithreads='define'
+test -z "$usemymalloc" && usemymalloc='undef'
 ccflags="$ccflags -DPERL_USE_SAFE_PUTENV -U__STRICT_ANSI__"
 # - otherwise i686-cygwin
 archname='cygwin'
diff -Naur perl-5.18.0-orig/hints/msys.sh perl-5.18.0/hints/msys.sh
--- perl-5.18.0-orig/hints/msys.sh	1970-01-01 00:00:00.000000000 +0000
+++ perl-5.18.0/hints/msys.sh	2013-06-15 18:28:56.626953200 +0000
@@ -0,0 +1,83 @@
+#! /bin/sh
+# msys.sh - hints for building perl using the MSYS environment for Win32
+#
+
+# not otherwise settable
+exe_ext='.exe'
+firstmakefile='GNUmakefile'
+case "$ldlibpthname" in
+'') ldlibpthname=PATH ;;
+esac
+archobjs='cygwin.o'
+
+# mandatory (overrides incorrect defaults)
+test -z "$cc" && cc='gcc'
+if test -z "$plibpth"
+then
+    plibpth=`gcc -print-file-name=libc.a`
+    plibpth=`dirname $plibpth`
+    plibpth=`cd $plibpth && pwd`
+fi
+so='dll'
+# - eliminate -lc, implied by gcc and a symlink to libmsys-2.0.a
+libswanted=`echo " $libswanted " | sed -e 's/ c / /g'`
+# - eliminate -lm, symlink to libmsys-2.0.a
+libswanted=`echo " $libswanted " | sed -e 's/ m / /g'`
+# - eliminate -lutil, symbols are all in libmsys-2.0.a
+libswanted=`echo " $libswanted " | sed -e 's/ util / /g'`
+# - add libgdbm_compat $libswanted
+libswanted="$libswanted gdbm_compat"
+test -z "$optimize" && optimize='-O3'
+man3ext='3pm'
+test -z "$use64bitint" && use64bitint='define'
+test -z "$usethreads" && usethreads='define'
+test -z "$usemymalloc" && usemymalloc='undef'
+ccflags="$ccflags -DPERL_USE_SAFE_PUTENV -U__STRICT_ANSI__"
+# - otherwise i686-msys
+archname='msys'
+
+# dynamic loading
+# - otherwise -fpic
+cccdlflags=' '
+lddlflags=' --shared'
+test -z "$ld" && ld='g++'
+
+case "$osvers" in
+    # Configure gets these wrong if the IPC server isn't yet running:
+    # only use for 1.5.7 and onwards
+    [2-9]*|1.[6-9]*|1.[1-5][0-9]*|1.5.[7-9]*|1.5.[1-6][0-9]*)
+        d_semctl_semid_ds='define'
+        d_semctl_semun='define'
+        ;;
+esac
+
+case "$osvers" in
+    [2-9]*|1.[6-9]*)
+        # IPv6 only since 1.7
+        d_inetntop='define'
+        d_inetpton='define'
+        ;;
+    *)
+        # IPv6 not implemented before cygwin-1.7
+        d_inetntop='undef'
+        d_inetpton='undef'
+esac
+
+# compile Win32CORE "module" as static. try to avoid the space.
+if test -z "$static_ext"; then
+  static_ext="Win32CORE"
+else
+  static_ext="$static_ext Win32CORE"
+fi
+
+# Win9x problem with non-blocking read from a closed pipe
+d_eofnblk='define'
+
+# suppress auto-import warnings
+ldflags="$ldflags -Wl,--enable-auto-import -Wl,--export-all-symbols -Wl,--enable-auto-image-base"
+lddlflags="$lddlflags $ldflags"
+
+# strip exe's and dll's, better do it afterwards
+#ldflags="$ldflags -s"
+#ccdlflags="$ccdlflags -s"
+#lddlflags="$lddlflags -s"
diff -Naur perl-5.18.0-orig/install_lib.pl perl-5.18.0/install_lib.pl
--- perl-5.18.0-orig/install_lib.pl	2013-04-01 20:28:20.000000000 +0000
+++ perl-5.18.0/install_lib.pl	2013-06-15 18:10:32.973632800 +0000
@@ -44,7 +44,7 @@
 $Is_VMS = $^O eq 'VMS';
 $Is_W32 = $^O eq 'MSWin32';
 $Is_OS2 = $^O eq 'os2';
-$Is_Cygwin = $^O eq 'cygwin';
+$Is_Cygwin = ($^O eq 'cygwin' || $^O eq 'msys');
 $Is_Darwin = $^O eq 'darwin';
 $Is_NetWare = $Config{osname} eq 'NetWare';
 
diff -Naur perl-5.18.0-orig/installman perl-5.18.0/installman
--- perl-5.18.0-orig/installman	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/installman	2013-06-15 18:31:02.026367200 +0000
@@ -143,7 +143,7 @@
 	    next;
 	}
 
-	if ($^O eq 'os2' || $^O eq 'amigaos' || $^O eq 'uwin' || $^O eq 'cygwin') {
+	if ($^O eq 'os2' || $^O eq 'amigaos' || $^O eq 'uwin' || $^O eq 'cygwin' || $^O eq 'msys') {
             $manpage =~ s#::#.#g;
 	}
         my $tmp = "${mandir}/${manpage}.tmp";
diff -Naur perl-5.18.0-orig/lib/AnyDBM_File.t perl-5.18.0/lib/AnyDBM_File.t
--- perl-5.18.0-orig/lib/AnyDBM_File.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/lib/AnyDBM_File.t	2013-06-15 18:26:19.565429700 +0000
@@ -15,7 +15,8 @@
 $Is_Dosish = ($^O eq 'amigaos' || $^O eq 'MSWin32' ||
 	      $^O eq 'NetWare' || $^O eq 'dos' ||
 	      $^O eq 'os2' ||
-	      $^O eq 'cygwin');
+	      $^O eq 'cygwin' ||
+	      $^O eq 'msys');
 
 my $filename = "Any_dbmx$$";
 unlink <"$filename*">;
diff -Naur perl-5.18.0-orig/lib/ExtUtils/t/Embed.t perl-5.18.0/lib/ExtUtils/t/Embed.t
--- perl-5.18.0-orig/lib/ExtUtils/t/Embed.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/lib/ExtUtils/t/Embed.t	2013-06-15 18:10:32.989257800 +0000
@@ -98,7 +98,7 @@
         s!-bE:(\S+)!-bE:$perl_exp!;
     }
    }
-   elsif ($^O eq 'cygwin') { # Cygwin needs no special treatment like below
+   elsif ($^O eq 'cygwin' or $^O eq 'msys') { # Cygwin needs no special treatment like below
        ;
    }
    elsif ($Config{'libperl'} !~ /\Alibperl\./) {
@@ -141,7 +141,7 @@
 unlink("$exe.manifest") if $cl and $Config{'ccversion'} =~ /^(\d+)/ and $1 >= 14;
 unlink("$exe$Config{exe_ext}") if $skip_exe;
 unlink("embed_test.map","embed_test.lis") if $^O eq 'VMS';
-unlink(glob("./*.dll")) if $^O eq 'cygwin';
+unlink(glob("./*.dll")) if ($^O eq 'cygwin' or $^O eq 'msys');
 unlink($testlib)	       if $libperl_copied;
 
 # gcc -g -I.. -L../ -o perl_test perl_test.c -lperl `../perl -I../lib -MExtUtils::Embed -I../ -e ccflags -e ldopts`
diff -Naur perl-5.18.0-orig/lib/File/Compare.t perl-5.18.0/lib/File/Compare.t
--- perl-5.18.0-orig/lib/File/Compare.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/lib/File/Compare.t	2013-06-15 18:26:59.770507800 +0000
@@ -111,7 +111,7 @@
   print "ok 12 # file/file [$donetests[1]]\n";
   print "not " unless $donetests[2] == 0;
   print "ok 13 # ";
-  print "TODO" if $^O eq "cygwin"; # spaces after filename silently trunc'd
+  print "TODO" if ($^O eq "cygwin" or $^O eq "msys"); # spaces after filename silently trunc'd
   print "TODO" if $^O eq "vos"; # spaces after filename silently trunc'd
   print " file/fileCR [$donetests[2]]\n";
 }
diff -Naur perl-5.18.0-orig/lib/File/Copy.t perl-5.18.0/lib/File/Copy.t
--- perl-5.18.0-orig/lib/File/Copy.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/lib/File/Copy.t	2013-06-15 18:10:33.004882800 +0000
@@ -173,7 +173,7 @@
 
   SKIP: {
     skip "Testing hard links", 3 
-         if !$Config{d_link} or $^O eq 'MSWin32' or $^O eq 'cygwin';
+         if !$Config{d_link} or $^O eq 'MSWin32' or $^O eq 'cygwin' or $^O eq 'msys';
 
     open(F, ">file-$$") or die $!;
     print F "dummy content\n";
@@ -365,7 +365,7 @@
             is (__$perm5, __$c_perm1, "Permission bits set correctly");
             TODO: {
                 local $TODO = 'Permission bits inconsistent under cygwin'
-                   if $^O eq 'cygwin';
+                   if ($^O eq 'cygwin' or $^O eq 'msys');
                 is (__$perm3, __$c_perm3, "Permission bits not modified");
                 is (__$perm6, __$c_perm3, "Permission bits not modified");
             }
diff -Naur perl-5.18.0-orig/lib/File/Find/t/find.t perl-5.18.0/lib/File/Find/t/find.t
--- perl-5.18.0-orig/lib/File/Find/t/find.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/lib/File/Find/t/find.t	2013-06-15 18:10:33.004882800 +0000
@@ -48,7 +48,7 @@
 
 BEGIN {
     use File::Spec;
-    if ($^O eq 'MSWin32' || $^O eq 'cygwin' || $^O eq 'VMS')
+    if ($^O eq 'MSWin32' || $^O eq 'cygwin' || $^O eq 'msys' || $^O eq 'VMS')
      {
       # This is a hack - at present File::Find does not produce native names on 
       # Win32 or VMS, so force File::Spec to use Unix names.
diff -Naur perl-5.18.0-orig/lib/File/Find.pm perl-5.18.0/lib/File/Find.pm
--- perl-5.18.0-orig/lib/File/Find.pm	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/lib/File/Find.pm	2013-06-15 18:27:27.754882800 +0000
@@ -1104,7 +1104,7 @@
 
 $File::Find::dont_use_nlink = 1
     if $^O eq 'os2' || $^O eq 'dos' || $^O eq 'amigaos' || $Is_Win32 ||
-       $^O eq 'interix' || $^O eq 'cygwin' || $^O eq 'qnx' || $^O eq 'nto';
+       $^O eq 'interix' || $^O eq 'cygwin' || $^O eq 'msys' || $^O eq 'qnx' || $^O eq 'nto';
 
 # Set dont_use_nlink in your hint file if your system's stat doesn't
 # report the number of links in a directory as an indication
diff -Naur perl-5.18.0-orig/lib/Net/hostent.t perl-5.18.0/lib/Net/hostent.t
--- perl-5.18.0-orig/lib/Net/hostent.t	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/lib/Net/hostent.t	2013-06-15 18:10:33.020507800 +0000
@@ -57,7 +57,7 @@
 
 SKIP: {
     skip "Windows will return the machine name instead of 'localhost'", 2
-      if $^O eq 'MSWin32' or $^O eq 'NetWare' or $^O eq 'cygwin';
+      if $^O eq 'MSWin32' or $^O eq 'NetWare' or $^O eq 'cygwin' or $^O eq 'msys';
 
     print "# name = " . $h->name . ", aliases = " . join (",", @{$h->aliases}) . "\n";
 
diff -Naur perl-5.18.0-orig/lib/perl5db.pl perl-5.18.0/lib/perl5db.pl
--- perl-5.18.0-orig/lib/perl5db.pl	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/lib/perl5db.pl	2013-06-15 18:32:29.250000000 +0000
@@ -1516,7 +1516,7 @@
 
 =cut
 
-    if ( $^O eq 'cygwin' ) {
+    if ( $^O eq 'cygwin' or $^O eq 'msys' ) {
 
         # /dev/tty is binary. use stdin for textmode
         undef $console;
@@ -8974,6 +8974,7 @@
     modlib
     mod
     modstyle
+    msys
     netware
     newmod
     number
diff -Naur perl-5.18.0-orig/lib/User/grent.t perl-5.18.0/lib/User/grent.t
--- perl-5.18.0-orig/lib/User/grent.t	2013-02-26 21:28:27.000000000 +0000
+++ perl-5.18.0/lib/User/grent.t	2013-06-15 18:10:33.051757800 +0000
@@ -18,7 +18,7 @@
 }
 
 BEGIN {
-    our $gid = $^O ne 'cygwin' ? 0 : 18;
+    our $gid = ($^O ne 'cygwin' || $^O ne 'msys') ? 0 : 18;
     our @grent = getgrgid $gid; # This is the function getgrgid.
     unless (@grent) { plan skip_all => "no gid 0"; }
 }
diff -Naur perl-5.18.0-orig/lib/User/pwent.t perl-5.18.0/lib/User/pwent.t
--- perl-5.18.0-orig/lib/User/pwent.t	2013-02-26 21:28:27.000000000 +0000
+++ perl-5.18.0/lib/User/pwent.t	2013-06-15 18:10:33.051757800 +0000
@@ -21,7 +21,7 @@
     # On VMS getpwuid(0) may return [$gid,0] UIC info (which may not exist).
     # It is better to use the $< uid for testing on VMS instead.
     if ( $^O eq 'VMS' ) { $uid = $< ; }
-    if ( $^O eq 'cygwin' ) { $uid = 500 ; }
+    if ( $^O eq 'cygwin' or $^O eq 'msys' ) { $uid = 500 ; }
     our @pwent = getpwuid $uid; # This is the function getpwuid.
     unless (@pwent) { print "1..0 # Skip: no uid $uid\n"; exit 0 }
 }
@@ -35,7 +35,7 @@
 my $pwent = getpwuid $uid; # This is the OO getpwuid.
 
 my $uid_expect = $uid;
-if ( $^O eq 'cygwin' ) {
+if ( $^O eq 'cygwin' or $^O eq 'msys' ) {
     print "not " unless (   $pwent->uid == $uid_expect
                          || $pwent->uid == 500         );  # go figure
 }
diff -Naur perl-5.18.0-orig/makedepend.SH perl-5.18.0/makedepend.SH
--- perl-5.18.0-orig/makedepend.SH	2013-05-01 02:52:55.000000000 +0000
+++ perl-5.18.0/makedepend.SH	2013-06-15 18:10:33.067382800 +0000
@@ -118,6 +118,7 @@
     uwin)     uwinfix="-e s,\\\\\\\\,/,g -e s,\\([a-zA-Z]\\):/,/\\1/,g" ;;
     os2)      uwinfix="-e s,\\\\\\\\,/,g" ;;
     cygwin)   uwinfix="-e s,\\\\\\\\,/,g" ;;
+    msys)     uwinfix="-e s,\\\\\\\\,/,g" ;;
     posix-bc) uwinfix="-e s/\\*POSIX(\\(.*\\))/\\1/" ;;
     vos)      uwinfix="-e s/\#/\\\#/" ;;
     *)        uwinfix="" ;;
diff -Naur perl-5.18.0-orig/Makefile.SH perl-5.18.0/Makefile.SH
--- perl-5.18.0-orig/Makefile.SH	2013-06-15 18:57:13.898437500 +0000
+++ perl-5.18.0/Makefile.SH	2013-06-15 18:10:33.067382800 +0000
@@ -70,8 +70,8 @@
 				${revision}.${patchlevel}.${subversion} \
 			     -install_name \$(shrpdir)/\$@"
 		;;
-	cygwin*)
-		shrpldflags="$shrpldflags -Wl,--out-implib=libperl.dll.a"
+	cygwin*|msys*)
+		shrpldflags="$shrpldflags -Wl,--out-implib=libperl.dll.a -Wl,--image-base,0x52000000"
 		linklibperl="-L. -lperl"
 		;;
 	sunos*)
@@ -750,7 +750,7 @@
 
 !NO!SUBS!
 	;;
-cygwin)
+cygwin|msys)
 	$spitshell >>$Makefile <<'!NO!SUBS!'
 cygwin.c: cygwin/cygwin.c
 	$(LNS) cygwin/cygwin.c
@@ -1275,7 +1275,7 @@
 	-rm -f perl.third lib*.so.perl.third perl.3log t/perl.third t/perl.3log
 	-rm -f perl.pixie lib*.so.perl.pixie lib*.so.Addrs
 	-rm -f perl.Addrs perl.Counts t/perl.Addrs t/perl.Counts *perl.xok
-	-rm -f cygwin.c libperl*.def libperl*.dll cygperl*.dll *.exe.stackdump
+	-rm -f cygwin.c libperl*.def libperl*.dll cygperl*.dll msys-perl*.dll *.exe.stackdump
 	-rm -f $(PERL_EXE) $(MINIPERL_EXE) $(LIBPERL) libperl.* microperl
 	-rm -f config.arch config.over $(DTRACE_H) runtests
 
diff -Naur perl-5.18.0-orig/regen/regen_lib.pl perl-5.18.0/regen/regen_lib.pl
--- perl-5.18.0-orig/regen/regen_lib.pl	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/regen/regen_lib.pl	2013-06-15 18:10:33.067382800 +0000
@@ -7,7 +7,7 @@
 
 # Common functions needed by the regen scripts
 
-$Needs_Write = $^O eq 'cygwin' || $^O eq 'os2' || $^O eq 'MSWin32';
+$Needs_Write = $^O eq 'cygwin' || $^O eq 'msys' || $^O eq 'os2' || $^O eq 'MSWin32';
 
 $Verbose = 0;
 @ARGV = grep { not($_ eq '-q' and $Verbose = -1) }
diff -Naur perl-5.18.0-orig/symbian/sanity.pl perl-5.18.0/symbian/sanity.pl
--- perl-5.18.0-orig/symbian/sanity.pl	2013-02-26 21:28:27.000000000 +0000
+++ perl-5.18.0/symbian/sanity.pl	2013-06-15 18:10:33.067382800 +0000
@@ -2,7 +2,7 @@
 
 if (exists $ENV{'!C:'}) {
   print "You are running this under Cygwin, aren't you? (found '!C' in %ENV)\n";
-  print "Are you perhaps using Cygwin Perl? (\$^O is '$^O')\n" if $^O =~ /cygwin/;
+  print "Are you perhaps using Cygwin Perl? (\$^O is '$^O')\n" if $^O =~ /cygwin|msys/;
   print "I'm sorry but only cmd.exe with e.g. the ActivePerl will work.\n";
   exit(1);
 }
diff -Naur perl-5.18.0-orig/t/io/eintr.t perl-5.18.0/t/io/eintr.t
--- perl-5.18.0-orig/t/io/eintr.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/io/eintr.t	2013-06-15 18:30:09.874023500 +0000
@@ -50,7 +50,7 @@
 # platforms
 
 my ($osmajmin) = $Config{osvers} =~ /^(\d+\.\d+)/;
-if ($^O eq 'VMS' || $^O eq 'MSWin32' || $^O eq 'cygwin' || $^O =~ /freebsd/ || $^O eq 'midnightbsd' ||
+if ($^O eq 'VMS' || $^O eq 'MSWin32' || $^O eq 'cygwin' || $^O eq 'msys' || $^O =~ /freebsd/ || $^O eq 'midnightbsd' ||
      ($^O eq 'solaris' && $Config{osvers} eq '2.8') || $^O eq 'nto' ||
      ($^O eq 'darwin' && $osmajmin < 9) ||
     ((int($]*1000) & 1) == 0)
diff -Naur perl-5.18.0-orig/t/io/fs.t perl-5.18.0/t/io/fs.t
--- perl-5.18.0-orig/t/io/fs.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/io/fs.t	2013-06-15 18:10:33.083007800 +0000
@@ -26,7 +26,7 @@
     !($^O eq 'MSWin32' || $^O eq 'NetWare' ||
       $^O eq 'dos'     || $^O eq 'os2'     ||
       $^O eq 'cygwin'  || $^O eq 'amigaos' ||
-	  $wd =~ m#$Config{afsroot}/#
+	  $^O eq 'msys'    || $wd =~ m#$Config{afsroot}/#
      );
 
 if (defined &Win32::IsWinNT && Win32::IsWinNT()) {
@@ -44,7 +44,7 @@
 $needs_fh_reopen = 1 if (defined &Win32::IsWin95 && Win32::IsWin95());
 
 my $skip_mode_checks =
-    $^O eq 'cygwin' && $ENV{CYGWIN} !~ /ntsec/;
+    $^O eq 'msys' && $ENV{MSYS} !~ /ntsec/;
 
 plan tests => 52;
 
@@ -106,7 +106,7 @@
         skip "hard links not that hard in $^O", 1 if $^O eq 'amigaos';
 	skip "no mode checks", 1 if $skip_mode_checks;
 
-#      if ($^O eq 'cygwin') { # new files on cygwin get rwx instead of rw-
+#      if ($^O eq 'cygwin' or $^O eq 'msys') { # new files on cygwin get rwx instead of rw-
 #          is($mode & 0777, 0777, "mode of triply-linked file");
 #      } else {
             is(sprintf("0%o", $mode & 0777), 
@@ -421,7 +421,7 @@
 # check if rename() can be used to just change case of filename
 SKIP: {
     skip "Works in Cygwin only if check_case is set to relaxed", 1
-      if ($ENV{'CYGWIN'} && ($ENV{'CYGWIN'} =~ /check_case:(?:adjust|strict)/));
+      if ($ENV{'MSYS'} && ($ENV{'MSYS'} =~ /check_case:(?:adjust|strict)/));
 
     chdir "./$tmpdir";
     open(FH,'>x') || die "Can't create x";
diff -Naur perl-5.18.0-orig/t/io/inplace.t perl-5.18.0/t/io/inplace.t
--- perl-5.18.0-orig/t/io/inplace.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/io/inplace.t	2013-06-15 18:30:30.452148500 +0000
@@ -37,7 +37,7 @@
 {
     # based on code, dosish systems can't do no-backup inplace
     # edits
-    $^O =~ /^(MSWin32|cygwin|uwin|dos|os2)$/
+    $^O =~ /^(MSWin32|cygwin|msys|uwin|dos|os2)$/
 	and skip("Can't inplace edit without backups on $^O", 4);
     
     our @ifiles = ( tempfile(), tempfile(), tempfile() );
diff -Naur perl-5.18.0-orig/t/io/tell.t perl-5.18.0/t/io/tell.t
--- perl-5.18.0-orig/t/io/tell.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/io/tell.t	2013-06-15 18:10:33.098632800 +0000
@@ -11,7 +11,7 @@
 $TST = 'TST';
 
 $Is_Dosish = ($^O eq 'MSWin32' or $^O eq 'NetWare' or $^O eq 'dos' or
-              $^O eq 'os2' or $^O eq 'cygwin' or
+              $^O eq 'os2' or $^O eq 'cygwin' or $^O eq 'msys' or
               $^O =~ /^uwin/);
 
 open($TST, 'harness') || (die "Can't open harness");
@@ -153,7 +153,7 @@
 # This test makes a questionable assumption that the file pointer will
 # be at eof after opening a file but before seeking, reading, or writing.
 # Only known failure is on cygwin.
-my $todo = $^O eq "cygwin" && &PerlIO::get_layers($tst) eq 'stdio'
+my $todo = ($^O eq "cygwin" || $^O eq "msys") && &PerlIO::get_layers($tst) eq 'stdio'
     && ' # TODO: file pointer not at eof';
 
 if (tell($tst) == 6)
diff -Naur perl-5.18.0-orig/t/lib/cygwin.t perl-5.18.0/t/lib/cygwin.t
--- perl-5.18.0-orig/t/lib/cygwin.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/lib/cygwin.t	2013-06-15 18:10:33.098632800 +0000
@@ -4,7 +4,7 @@
     chdir 't' if -d 't';
     @INC = ('../lib');
     require './test.pl';
-    skip_all('cygwin specific test') unless $^O eq 'cygwin';
+    skip_all('cygwin specific test') unless ($^O eq 'cygwin' || $^O eq 'msys');
 }
 
 plan(tests => 16);
diff -Naur perl-5.18.0-orig/t/lib/dbmt_common.pl perl-5.18.0/t/lib/dbmt_common.pl
--- perl-5.18.0-orig/t/lib/dbmt_common.pl	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/lib/dbmt_common.pl	2013-06-15 18:29:43.645507800 +0000
@@ -43,7 +43,7 @@
 }
 SKIP: {
     skip "different file permission semantics on $^O", 1
-	if $^O eq 'amigaos' || $^O eq 'os2' || $^O eq 'MSWin32' || $^O eq 'NetWare' || $^O eq 'dos' || $^O eq 'cygwin' || $^O eq 'vos';
+	if $^O eq 'amigaos' || $^O eq 'os2' || $^O eq 'MSWin32' || $^O eq 'NetWare' || $^O eq 'dos' || $^O eq 'cygwin' || $^O eq 'msys' || $^O eq 'vos';
     my ($dev,$ino,$mode,$nlink,$uid,$gid,$rdev,$size,$atime,$mtime,$ctime,
 	$blksize,$blocks) = stat($Dfile);
     is($mode & 0777, 0640);
diff -Naur perl-5.18.0-orig/t/op/groups.t perl-5.18.0/t/op/groups.t
--- perl-5.18.0-orig/t/op/groups.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/op/groups.t	2013-06-15 18:10:33.114257800 +0000
@@ -284,7 +284,7 @@
 
 # Test if this is Cygwin
 sub cygwin_nt {
-    return $Config::Config{myuname} =~ /^cygwin_nt/i;
+    return $Config::Config{myuname} =~ /^msys_nt/i;
 }
 
 # Get perl's supplementary groups and the number of times each gid
diff -Naur perl-5.18.0-orig/t/op/magic.t perl-5.18.0/t/op/magic.t
--- perl-5.18.0-orig/t/op/magic.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/op/magic.t	2013-06-15 18:31:37.197265700 +0000
@@ -54,7 +54,7 @@
 $Is_VMS      = $^O eq 'VMS';
 $Is_Dos      = $^O eq 'dos';
 $Is_os2      = $^O eq 'os2';
-$Is_Cygwin   = $^O eq 'cygwin';
+$Is_Cygwin   = ($^O eq 'cygwin' || $^O eq 'msys');
 
 $PERL = $ENV{PERL}
     || ($Is_NetWare           ? 'perl'   :
diff -Naur perl-5.18.0-orig/t/op/require_errors.t perl-5.18.0/t/op/require_errors.t
--- perl-5.18.0-orig/t/op/require_errors.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/op/require_errors.t	2013-06-15 18:54:38.367187500 +0000
@@ -72,7 +72,7 @@
     push @INC, '../lib';
     require Cwd;
     require File::Spec::Functions;
-    if ($^O eq 'cygwin') {
+    if ($^O eq 'cygwin' or $^O eq 'msys') {
         require Win32;
     }
 
@@ -81,7 +81,7 @@
     my $olduid = $>;
     eval { $> = 1; };
     skip "Can't test permissions meaningfully if you're superuser", 2
-        if ($^O eq 'cygwin' ? Win32::IsAdminUser() : $> == 0);
+        if (($^O eq 'cygwin' or $^O eq 'msys') ? Win32::IsAdminUser() : $> == 0);
 
     local @INC = ".";
     eval "use $module";
diff -Naur perl-5.18.0-orig/t/op/sigdispatch.t perl-5.18.0/t/op/sigdispatch.t
--- perl-5.18.0-orig/t/op/sigdispatch.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/op/sigdispatch.t	2013-06-15 18:55:33.210937500 +0000
@@ -43,7 +43,7 @@
     skip('We can\'t test blocking without sigprocmask', 17)
 	if is_miniperl() || !$Config{d_sigprocmask};
     skip('This doesn\'t work on $^O threaded builds RT#88814', 17)
-        if $^O =~ /openbsd|cygwin/ && $Config{useithreads};
+        if $^O =~ /openbsd|cygwin|msys/ && $Config{useithreads};
 
     require POSIX;
     my $pending = POSIX::SigSet->new();
@@ -57,7 +57,7 @@
     kill 'SIGUSR1', $$;
     is $gotit, 0, 'Haven\'t received third signal yet';
 
-    diag "2nd sigpending crashes on cygwin" if $^O eq 'cygwin';
+    diag "2nd sigpending crashes on cygwin" if ($^O eq 'cygwin' or $^O eq 'msys');
     is POSIX::sigpending($pending), '0 but true', 'sigpending';
     is $pending->ismember(&POSIX::SIGUSR1), 1, 'SIGUSR1 is pending';
     
@@ -112,7 +112,7 @@
 
 SKIP: {
     skip("alarm cannot interrupt blocking system calls on $^O", 2)
-	if $^O =~ /MSWin32|cygwin|VMS/;
+	if $^O =~ /MSWin32|cygwin|msys|VMS/;
     # RT #88774
     # make sure the signal handler's called in an eval block *before*
     # the eval is popped
diff -Naur perl-5.18.0-orig/t/op/stat.t perl-5.18.0/t/op/stat.t
--- perl-5.18.0-orig/t/op/stat.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/op/stat.t	2013-06-15 18:10:33.129882800 +0000
@@ -28,7 +28,7 @@
 $ENV{LANGUAGE} = 'C';		# Ditto in GNU.
 
 $Is_Amiga   = $^O eq 'amigaos';
-$Is_Cygwin  = $^O eq 'cygwin';
+$Is_Cygwin  = ($^O eq 'cygwin' || $^O eq 'msys');
 $Is_Darwin  = $^O eq 'darwin';
 $Is_Dos     = $^O eq 'dos';
 $Is_MSWin32 = $^O eq 'MSWin32';
@@ -108,7 +108,7 @@
     SKIP: {
         skip "No link count", 1 if $Config{dont_use_nlink};
         skip "Cygwin9X fakes hard links by copying", 1
-          if $Config{myuname} =~ /^cygwin_(?:9\d|me)\b/i;
+          if $Config{myuname} =~ /^msys_(?:9\d|me)\b/i;
 
         is($nlink, 2,     'Link count on hard linked file' );
     }
diff -Naur perl-5.18.0-orig/t/op/taint.t perl-5.18.0/t/op/taint.t
--- perl-5.18.0-orig/t/op/taint.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/op/taint.t	2013-06-15 18:10:33.129882800 +0000
@@ -46,7 +46,7 @@
 my $Is_MSWin32  = $^O eq 'MSWin32';
 my $Is_NetWare  = $^O eq 'NetWare';
 my $Is_Dos      = $^O eq 'dos';
-my $Is_Cygwin   = $^O eq 'cygwin';
+my $Is_Cygwin   = ($^O eq 'cygwin' || $^O eq 'msys');
 my $Is_OpenBSD  = $^O eq 'openbsd';
 my $Is_MirBSD   = $^O eq 'mirbsd';
 my $Invoke_Perl = $Is_VMS      ? 'MCR Sys$Disk:[]Perl.exe' :
diff -Naur perl-5.18.0-orig/t/op/time.t perl-5.18.0/t/op/time.t
--- perl-5.18.0-orig/t/op/time.t	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/op/time.t	2013-06-15 18:10:33.145507800 +0000
@@ -50,7 +50,7 @@
 SKIP: {
     # This conditional of "No tzset()" is stolen from ext/POSIX/t/time.t
     skip "No tzset()", 1
-        if $^O eq "VMS" || $^O eq "cygwin" ||
+        if $^O eq "VMS" || $^O eq "cygwin" || $^O eq "msys" ||
            $^O eq "djgpp" || $^O eq "MSWin32" || $^O eq "dos" ||
            $^O eq "interix";
 
diff -Naur perl-5.18.0-orig/t/porting/extrefs.t perl-5.18.0/t/porting/extrefs.t
--- perl-5.18.0-orig/t/porting/extrefs.t	2013-05-14 00:00:45.000000000 +0000
+++ perl-5.18.0/t/porting/extrefs.t	2013-06-15 18:56:14.601562500 +0000
@@ -43,7 +43,7 @@
 sub try_compile_and_link {
     my ($c, %args) = @_;
 
-    my $ld_exeext = ($^O eq 'cygwin' || $^O eq 'MSWin32' ||
+    my $ld_exeext = ($^O eq 'cygwin' || $^O eq 'MSWin32' || $^O eq 'msys' ||
                  $^O eq 'os2' && $Config{ldflags} =~ /-Zexe\b/) ? '.exe' :
                 (($^O eq 'vos') ? $Config{exe_ext} : '');
 
diff -Naur perl-5.18.0-orig/t/test.pl perl-5.18.0/t/test.pl
--- perl-5.18.0-orig/t/test.pl	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/t/test.pl	2013-06-15 18:10:33.145507800 +0000
@@ -562,7 +562,7 @@
 my $is_mswin    = $^O eq 'MSWin32';
 my $is_netware  = $^O eq 'NetWare';
 my $is_vms      = $^O eq 'VMS';
-my $is_cygwin   = $^O eq 'cygwin';
+my $is_cygwin   = ($^O eq 'cygwin' || $^O eq 'msys');
 
 sub _quote_args {
     my ($runperl, $args) = @_;
diff -Naur perl-5.18.0-orig/utils/perlbug.PL perl-5.18.0/utils/perlbug.PL
--- perl-5.18.0-orig/utils/perlbug.PL	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/utils/perlbug.PL	2013-06-15 18:10:33.161132800 +0000
@@ -710,7 +710,7 @@
     my @env =
         qw(PATH LD_LIBRARY_PATH LANG PERL_BADLANG SHELL HOME LOGDIR LANGUAGE);
     push @env, $Config{ldlibpthname} if $Config{ldlibpthname} ne '';
-    push @env, grep /^(?:PERL|LC_|LANG|CYGWIN)/, keys %ENV;
+    push @env, grep /^(?:PERL|LC_|LANG|MSYS)/, keys %ENV;
     my %env;
     @env{@env} = @env;
     for my $env (sort keys %env) {
diff -Naur perl-5.18.0-orig/win32/FindExt.pm perl-5.18.0/win32/FindExt.pm
--- perl-5.18.0-orig/win32/FindExt.pm	2013-05-01 02:52:56.000000000 +0000
+++ perl-5.18.0/win32/FindExt.pm	2013-06-15 18:10:33.161132800 +0000
@@ -22,7 +22,7 @@
     push @no, "ODBM_File"
       unless ($config->{i_dbm} || $config->{i_rpcsvcdbm}) && !$config->{d_cplusplus};
     push @no, "VMS.*" unless $^O eq "VMS";
-    push @no, "Win32.*" unless $^O eq "MSWin32" || $^O eq "cygwin";
+    push @no, "Win32.*" unless $^O eq "MSWin32" || $^O eq "cygwin" || $^O eq "msys";
 
     $no = join('|', @no);
     $no = qr/^(?:$no)$/i;
