--- perl-5.18.2/nostdio.h.orig	2014-02-02 20:14:10.012000000 +0400
+++ perl-5.18.2/nostdio.h	2014-02-02 20:18:56.053600000 +0400
@@ -17,8 +17,8 @@
    It is completely pointless as we have already included it ourselves.
 */
 
-#if !defined(_STDIO_H) && !defined(FILE) && !defined(_STDIO_INCLUDED) && !defined(__STDIO_LOADED)
-#define _STDIO_H
+#if !defined(_STDIO_H_) && !defined(FILE) && !defined(_STDIO_INCLUDED) && !defined(__STDIO_LOADED)
+#define _STDIO_H_
 #define _STDIO_INCLUDED
 #define __STDIO_LOADED
 struct _FILE;
