--- perl5181/cpan/Pod-Parser/lib/Pod/Find.pm.orig	2013-08-20 12:19:20.081054600 +0400
+++ perl5181/cpan/Pod-Parser/lib/Pod/Find.pm	2013-08-20 12:19:13.549804600 +0400
@@ -472,13 +472,13 @@
         if $options{'-verbose'};
       next Dir;
     }
-    # for some strange reason the path on MacOS/darwin/cygwin is
+    # for some strange reason the path on MacOS/darwin/cygwin/MSYS is
     # 'pods' not 'pod'
     # this could be the case also for other systems that
     # have a case-tolerant file system, but File::Spec
     # does not recognize 'darwin' yet. And cygwin also has "pods",
     # but is not case tolerant. Oh well...
-    if((File::Spec->case_tolerant || $^O =~ /macos|darwin|cygwin/i)
+    if((File::Spec->case_tolerant || $^O =~ /macos|darwin|cygwin|msys/i)
      && -d File::Spec->catdir($dir,'pods')) {
       $dir = File::Spec->catdir($dir,'pods');
       redo Dir;
