diff --git a/perl-5.26.0/Makefile.SH.orig b/perl-5.26.0/Makefile.SH
index 51d3c8e..ac82b4a 100755
--- a/perl-5.26.0/Makefile.SH.orig
+++ b/perl-5.26.0/Makefile.SH
@@ -71,7 +71,7 @@ true)
 			     -install_name \$(shrpdir)/\$@"
 		;;
 	cygwin*)
-		shrpldflags="$shrpldflags -Wl,--out-implib=libperl.dll.a -Wl,--image-base,0x52000000"
+		shrpldflags="$shrpldflags -Wl,--out-implib=libperl.dll.a"
 		linklibperl="-L. -lperl"
 		;;
 	sunos*)
