--- perl-5.28.0/Configure.orig	2018-06-28 08:43:31.289358200 +0300
+++ perl-5.28.0/Configure	2018-06-28 09:15:58.239396500 +0300
@@ -22971,26 +22987,6 @@
 	fi
 fi
 
-: add -D_FORTIFY_SOURCE if feasible and not already there
-case "$gccversion" in
-[4567].*)	case "$optimize$ccflags" in
-	*-O*)	case "$ccflags$cppsymbols" in
-		*_FORTIFY_SOURCE=*) # Don't add it again.
-			echo "You seem to have -D_FORTIFY_SOURCE already, not adding it." >&4
-			;;
-		*)	echo "Adding -D_FORTIFY_SOURCE=2 to ccflags..." >&4
-			ccflags="$ccflags -D_FORTIFY_SOURCE=2"
-			;;
-		esac
-		;;
-	*)	echo "You have gcc 4.* or later but not optimizing, not adding -D_FORTIFY_SOURCE." >&4
-		;;
-	esac
-	;;
-*)	echo "You seem not to have gcc 4.* or later, not adding -D_FORTIFY_SOURCE." >&4
-	;;
-esac
-
 : script used to emit important warnings
 cat >warn <<EOS
 $startsh
