diff --git a/regen/lib_cleanup.pl b/regen/lib_cleanup.pl
index 5e40b40..c917cc0 100644
--- a/regen/lib_cleanup.pl
+++ b/regen/lib_cleanup.pl
@@ -159,7 +159,7 @@ foreach ('win32/Makefile', 'win32/makefile.mk') {
 }
 
 # This must come last as it can exit early:
-if ($TAP && !-d '.git' && !-f 'lib/.gitignore') {
+if ($TAP && !-d '.git' || !-f 'lib/.gitignore') {
     print "ok # skip not being run from a git checkout, hence no lib/.gitignore\n";
     exit 0;
 }
