--- perl-5.24.1/utils/Makefile.PL	Sat Feb 18 19:06:04 2017
+++ perl-5.24.1/utils/Makefile.PL	Tue Jan 17 08:34:48 2017
@@ -13,7 +13,7 @@
 # attempting to move the work from them to the extension directories and
 # ExtUtils::MakeMaker.
 
-require 'regen/regen_lib.pl';
+require './regen/regen_lib.pl';
 
 my $target = 'utils/Makefile';
 print "Extracting $target (with variable substitutions)\n";
--- perl-5.24.1/regen/ebcdic.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/ebcdic.pl	Sat Jan 14 13:52:29 2017
@@ -4,8 +4,8 @@
 
 BEGIN { unshift @INC, '.' }
 
-require 'regen/regen_lib.pl';
-require 'regen/charset_translations.pl';
+require './regen/regen_lib.pl';
+require './regen/charset_translations.pl';
 
 # Generates the EBCDIC translation tables that were formerly hard-coded into
 # utfebcdic.h
--- perl-5.24.1/regen/embed.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/embed.pl	Sat Jan 14 13:52:29 2017
@@ -26,8 +26,8 @@
 
 BEGIN {
     # Get function prototypes
-    require 'regen/regen_lib.pl';
-    require 'regen/embed_lib.pl';
+    require './regen/regen_lib.pl';
+    require './regen/embed_lib.pl';
 }
 
 my $SPLINT = 0; # Turn true for experimental splint support http://www.splint.org
--- perl-5.24.1/regen/feature.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/feature.pl	Sat Jan 14 13:52:29 2017
@@ -11,7 +11,7 @@
 # This script is normally invoked from regen.pl.
 
 BEGIN {
-    require 'regen/regen_lib.pl';
+    require './regen/regen_lib.pl';
     push @INC, './lib';
 }
 use strict ;
--- perl-5.24.1/regen/keywords.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/keywords.pl	Sat Jan 14 13:52:29 2017
@@ -11,7 +11,7 @@
 use strict;
 use Devel::Tokenizer::C 0.05;
 
-require 'regen/regen_lib.pl';
+require './regen/regen_lib.pl';
 
 my $h = open_new('keywords.h', '>',
 		 { by => 'regen/keywords.pl', from => 'its data',
--- perl-5.24.1/regen/lib_cleanup.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/lib_cleanup.pl	Sat Jan 14 13:52:29 2017
@@ -1,7 +1,7 @@
 #!perl -w
 use strict;
-require 'regen/regen_lib.pl';
-require 'Porting/pod_lib.pl';
+require './regen/regen_lib.pl';
+require './Porting/pod_lib.pl';
 use vars qw($TAP $Verbose);
 
 # For processing later
--- perl-5.24.1/regen/miniperlmain.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/miniperlmain.pl	Sat Jan 14 13:52:29 2017
@@ -4,7 +4,7 @@
 
 BEGIN {
     # Get function prototypes
-    require 'regen/regen_lib.pl';
+    require './regen/regen_lib.pl';
     unshift @INC, 'ext/ExtUtils-Miniperl/lib';
 }
 
--- perl-5.24.1/regen/mk_invlists.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/mk_invlists.pl	Sat Jan 14 13:52:29 2017
@@ -8,8 +8,8 @@
                     prop_invlist
                     prop_invmap search_invlist
                    );
-require 'regen/regen_lib.pl';
-require 'regen/charset_translations.pl';
+require './regen/regen_lib.pl';
+require './regen/charset_translations.pl';
 
 # This program outputs charclass_invlists.h, which contains various inversion
 # lists in the form of C arrays that are to be used as-is for inversion lists.
--- perl-5.24.1/regen/mk_PL_charclass.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/mk_PL_charclass.pl	Sat Jan 14 13:52:29 2017
@@ -2,8 +2,8 @@
 use v5.15.8;
 use strict;
 use warnings;
-require 'regen/regen_lib.pl';
-require 'regen/charset_translations.pl';
+require './regen/regen_lib.pl';
+require './regen/charset_translations.pl';
 
 # This program outputs l1_charclass_tab.h, which defines the guts of the
 # PL_charclass table.  Each line is a bit map of properties that the Unicode
--- perl-5.24.1/regen/opcode.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/opcode.pl	Sat Jan 14 13:52:29 2017
@@ -20,7 +20,7 @@
 
 BEGIN {
     # Get function prototypes
-    require 'regen/regen_lib.pl';
+    require './regen/regen_lib.pl';
 }
 
 my $oc = open_new('opcode.h', '>',
@@ -915,7 +915,7 @@
 # which define what bits in op_private have what meanings for each op.
 # It populates %LABELS, %DEFINES, %FLAGS, %BITFIELDS.
 
-require 'regen/op_private';
+require './regen/op_private';
 
 #use Data::Dumper;
 #print Dumper \%LABELS, \%DEFINES, \%FLAGS, \%BITFIELDS;
--- perl-5.24.1/regen/overload.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/overload.pl	Sat Jan 14 13:52:29 2017
@@ -16,7 +16,7 @@
 
 BEGIN {
     # Get function prototypes
-    require 'regen/regen_lib.pl';
+    require './regen/regen_lib.pl';
 }
 
 use strict;
--- perl-5.24.1/regen/reentr.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/reentr.pl	Sat Jan 14 13:52:29 2017
@@ -18,7 +18,7 @@
 
 BEGIN {
     # Get function prototypes
-    require 'regen/regen_lib.pl';
+    require './regen/regen_lib.pl';
 }
 
 use strict;
--- perl-5.24.1/regen/regcharclass.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/regcharclass.pl	Sat Jan 14 13:52:29 2017
@@ -11,8 +11,8 @@
 sub DEBUG () { 0 }
 $|=1 if DEBUG;
 
-require 'regen/regen_lib.pl';
-require 'regen/charset_translations.pl';
+require './regen/regen_lib.pl';
+require './regen/charset_translations.pl';
 require "regen/regcharclass_multi_char_folds.pl";
 
 =head1 NAME
--- perl-5.24.1/regen/regcomp.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/regcomp.pl	Sat Jan 14 13:52:29 2017
@@ -20,7 +20,7 @@
 
 BEGIN {
     # Get function prototypes
-    require 'regen/regen_lib.pl';
+    require './regen/regen_lib.pl';
 }
 use strict;
 
--- perl-5.24.1/regen/uconfig_h.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/uconfig_h.pl	Sat Jan 14 13:52:29 2017
@@ -10,7 +10,7 @@
 
 use strict;
 use Config;
-require 'regen/regen_lib.pl';
+require './regen/regen_lib.pl';
 
 my ($uconfig_h, $uconfig_h_new, $config_h_sh)
     = ('uconfig.h', 'uconfig.h-new', 'config_h.SH');
--- perl-5.24.1/regen/unicode_constants.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/unicode_constants.pl	Sat Jan 14 13:52:29 2017
@@ -1,8 +1,8 @@
 use v5.16.0;
 use strict;
 use warnings;
-require 'regen/regen_lib.pl';
-require 'regen/charset_translations.pl';
+require './regen/regen_lib.pl';
+require './regen/charset_translations.pl';
 use Unicode::UCD;
 use charnames qw(:loose);
 
--- perl-5.24.1/regen/warnings.pl	Sat Feb 18 19:05:54 2017
+++ perl-5.24.1/regen/warnings.pl	Mon Jan 23 21:59:53 2017
@@ -19,7 +19,7 @@
 $VERSION = '1.36';
 
 BEGIN {
-    require 'regen/regen_lib.pl';
+    require './regen/regen_lib.pl';
     push @INC, './lib';
 }
 use strict ;
--- perl-5.24.1/Porting/checkcfgvar.pl	Sat Feb 18 19:05:51 2017
+++ perl-5.24.1/Porting/checkcfgvar.pl	Sat Jan 14 13:51:21 2017
@@ -36,7 +36,7 @@
 
 $default and $default =~ s/^'(.*)'$/$1/; # Will be quoted on generation
 
-require 'regen/regen_lib.pl' if $opt_r;
+require './regen/regen_lib.pl' if $opt_r;
 
 my $MASTER_CFG = "config_h.SH";
 # Inclusive bounds on the main part of the file, $section == 1 below:
--- perl-5.24.1/Porting/new-perldelta.pl	Sat Feb 18 19:05:52 2017
+++ perl-5.24.1/Porting/new-perldelta.pl	Sat Jan 14 13:51:21 2017
@@ -4,7 +4,7 @@
 # This needs to be able to run from a clean checkout, hence assume only system
 # perl, which may be too old to have autodie
 
-require 'Porting/pod_lib.pl';
+require './Porting/pod_lib.pl';
 
 my $state = get_pod_metadata(1);
 my (undef, $old_major, $old_minor) = @{$state->{delta_version}};
--- perl-5.24.1/pod/buildtoc	Sat Feb 18 19:05:47 2017
+++ perl-5.24.1/pod/buildtoc	Sat Jan 14 13:52:26 2017
@@ -14,7 +14,7 @@
 BEGIN {
   my $Top = File::Spec->catdir($FindBin::Bin, File::Spec->updir);
   chdir $Top or die "Can't chdir to $Top: $!";
-  require 'Porting/pod_lib.pl';
+  require './Porting/pod_lib.pl';
 }
 
 die "$0: Usage: $0 [--quiet]\n"
--- perl-5.24.1/pod/perlmodlib.PL	Sat Feb 18 19:05:49 2017
+++ perl-5.24.1/pod/perlmodlib.PL	Sat Jan 14 13:52:28 2017
@@ -13,7 +13,7 @@
     chdir $workdir
         or die "Couldn't chdir to '$workdir': $!";
 }
-require 'regen/regen_lib.pl';
+require './regen/regen_lib.pl';
 
 # MANIFEST itself is Unix style filenames, so we have to assume that Unix style
 # filenames will work.
--- perl-5.24.1/h2pl/cbreak.pl	Sat Feb 18 19:05:35 2017
+++ perl-5.24.1/h2pl/cbreak.pl	Sat Jan 14 13:52:22 2017
@@ -12,7 +12,7 @@
     local($on) = @_;
 
     require 'sizeof.ph';
-    require 'sys/ioctl.ph';
+    require './sys/ioctl.ph';
 
     ioctl(STDIN,&TIOCGETP,$sgttyb)
         || die "Can't ioctl TIOCGETP: $!";
--- perl-5.24.1/h2pl/cbreak2.pl	Sat Feb 18 19:05:35 2017
+++ perl-5.24.1/h2pl/cbreak2.pl	Sat Jan 14 13:52:22 2017
@@ -11,7 +11,7 @@
 sub set_cbreak {
     local($on) = @_;
 
-    require 'sys/ioctl.pl';
+    require './sys/ioctl.pl';
 
     ioctl(STDIN,$TIOCGETP,$sgttyb)
         || die "Can't ioctl TIOCGETP: $!";
--- perl-5.24.1/autodoc.pl	Mon Mar 27 09:23:26 2017
+++ perl-5.24.1/autodoc.pl	Sat Jan 14 14:51:22 2017
@@ -29,8 +29,8 @@
     chdir $workdir
         or die "Couldn't chdir to '$workdir': $!";
 }
-require 'regen/regen_lib.pl';
-require 'regen/embed_lib.pl';
+require './regen/regen_lib.pl';
+require './regen/embed_lib.pl';
 
 #
 # See database of global and static function prototypes in embed.fnc
--- perl-5.24.1/regen_perly.pl	Mon Mar 27 09:24:48 2017
+++ perl-5.24.1/regen_perly.pl	Sat Jan 14 14:52:29 2017
@@ -35,7 +35,7 @@
 use warnings;
 use strict;
 
-BEGIN { require 'regen/regen_lib.pl'; }
+BEGIN { require './regen/regen_lib.pl'; }
 
 my $bison = 'bison';
 
--- perl-5.24.1/Porting/corelist.pl	Mon Mar 27 10:42:11 2017
+++ perl-5.24.1/Porting/corelist.pl	Mon Mar 20 18:08:38 2017
@@ -21,8 +21,8 @@
 use HTTP::Tiny;
 use IO::Uncompress::Gunzip;
 
-my $corelist_file = 'dist/Module-CoreList/lib/Module/CoreList.pm';
-my $utils_file = 'dist/Module-CoreList/lib/Module/CoreList/Utils.pm';
+my $corelist_file = './dist/Module-CoreList/lib/Module/CoreList.pm';
+my $utils_file = './dist/Module-CoreList/lib/Module/CoreList/Utils.pm';
 
 my %lines;
 my %module_to_file;
--- perl-5.24.1/cpan/Devel-PPPort/devel/mktodo.pl	Mon Mar 27 11:46:27 2017
+++ perl-5.24.1/cpan/Devel-PPPort/devel/mktodo.pl	Sat Jan 14 14:51:28 2017
@@ -22,7 +22,7 @@
 use Config;
 use Time::HiRes qw( gettimeofday tv_interval );
 
-require 'devel/devtools.pl';
+require './devel/devtools.pl';
 
 our %opt = (
   debug   => 0,
--- perl-5.24.1/utils/perlivp.PL	Mon Mar 27 12:22:48 2017
+++ perl-5.24.1/utils/perlivp.PL	Sat Jan 14 14:52:37 2017
@@ -231,7 +231,7 @@
         next if $_ eq 'XS/APItest';
         next if $_ eq 'XS/Typemap';
            # VMS$ perl  -e "eval ""require \""Devel/DProf.pm\"";"" print $@"
-           # \NT> perl  -e "eval \"require 'Devel/DProf.pm'\"; print $@"
+           # \NT> perl  -e "eval \"require './Devel/DProf.pm'\"; print $@"
            # DProf: run perl with -d to use DProf.
            # Compilation failed in require at (eval 1) line 1.
         eval " require \"$_.pm\"; ";
--- perl-5.24.1/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Cygwin.pm	Mon Mar 27 11:46:36 2017
+++ perl-5.24.1/cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Cygwin.pm	Sat Jan 14 14:51:40 2017
@@ -86,7 +86,7 @@
     if ($Config{useshrplib} eq 'true') {
         my $libperl = '$(PERL_INC)' .'/'. "$Config{libperl}";
         if( $] >= 5.006002 ) {
-            $libperl =~ s/a$/dll.a/;
+            $libperl =~ s/(dll\.)?a$/dll.a/;
         }
         $self->{PERL_ARCHIVE} = $libperl;
     } else {
--- perl-5.24.1/Makefile.SH	Mon Mar 27 13:29:44 2017
+++ perl-5.24.1/Makefile.SH	Wed Mar 22 10:30:42 2017
@@ -189,6 +189,7 @@
 dynamic_ext_re="lib/auto/re/re.$dlext"
 extra_dep='
 ext/Pod-Functions/pm_to_blib: cpan/Pod-Simple/pm_to_blib cpan/Pod-Escapes/pm_to_blib pod/perlfunc.pod
+cpan/IO-Compress/pm_to_blib: dist/lib/pm_to_blib
 '
 for f in $dynamic_ext; do
     : the dependency named here will never exist
