--- Doc/Makefile.in.orig	2014-08-01 16:50:26.000000000 +0900
+++ Doc/Makefile.in	2014-08-01 16:51:22.000000000 +0900
@@ -319,7 +319,7 @@
 	    $(INSTALL_DATA) help/* $(DESTDIR)$(runhelpdir); \
 	    while read from to; do \
 	        rm -f $(DESTDIR)$(runhelpdir)/$$to || : ; \
-	        $(LN_S) $$from $(DESTDIR)$(runhelpdir)/$$to; \
+	        $(LN_S) $(sdir)/help/$$from $(DESTDIR)$(runhelpdir)/$$to; \
 	    done < help.txt; \
 	fi
 .PHONY: install.runhelp
