--- zsh-5.0.7/Src/zsh_system.h.old	2014-12-28 03:55:04.657231300 +0900
+++ zsh-5.0.7/Src/zsh_system.h	2014-12-28 04:07:32.618012200 +0900
@@ -784,6 +784,13 @@
 # define IS_DIRSEP(c) ((c) == '/')
 #endif
 
+#ifdef __MSYS__
+/* MSYS2 Hack! MSYS2 symlink() cannot create a symbolic link to a non existing files */
+# ifdef HAVE_SYMLINK
+#  undef HAVE_SYMLINK
+# endif
+#endif
+
 #if defined(__GNUC__) && (!defined(__APPLE__) || defined(__clang__))
 /* Does the OS X port of gcc still gag on __attribute__? */
 #define UNUSED(x) x __attribute__((__unused__))
