diff --git a/configure b/configure-patched
index 99430fb..c041bf5 100755
--- a/configure
+++ b/configure-patched
@@ -10588,7 +10588,7 @@ $as_echo_n "checking for /dev/fd filesystem... " >&6; }
 if ${zsh_cv_sys_path_dev_fd+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  if test "$host_os" = cygwin; then
+  if test "$host_os" = cygwin || test "$host_os" = msys; then
 zsh_cv_sys_path_dev_fd=no
 else
 for zsh_cv_sys_path_dev_fd in /proc/self/fd /dev/fd no; do
@@ -11138,7 +11138,7 @@ $as_echo_n "checking if named FIFOs work... " >&6; }
 if ${zsh_cv_sys_fifo+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  if test "$host_os" = cygwin; then
+  if test "$host_os" = cygwin || test "$host_os" = msys; then
 zsh_cv_sys_fifo=yes
 else
 if test "$cross_compiling" = yes; then :
@@ -11752,7 +11752,7 @@ if test "x$aixdynamic" = xyes; then
   zsh_cv_sys_dynamic_strip_exe="${zsh_cv_sys_dynamic_strip_exe=yes}"
   zsh_cv_sys_dynamic_strip_lib="${zsh_cv_sys_dynamic_strip_lib=yes}"
   zsh_cv_shared_environ="${zsh_cv_shared_environ=yes}"
-elif test "$host_os" = cygwin; then
+elif test "$host_os" = cygwin || test "$host_os" = msys; then
   DL_EXT="${DL_EXT=dll}"
 ##DLLD="${DLLD=dllwrap}"
   DLLD="${DLLD=$CC}"
@@ -12934,7 +12934,7 @@ $as_echo "$zsh_cv_sys_dynamic_strip_lib" >&6; }
   if $strip_libldflags && test "$zsh_cv_sys_dynamic_strip_lib" = yes; then
     LIBLDFLAGS="$LIBLDFLAGS -s"
   fi
-  if test "$host_os" = cygwin; then
+  if test "$host_os" = cygwin || test "$host_os" = msys; then
     INSTLIB="install.cygwin-lib"
     UNINSTLIB="uninstall.cygwin-lib"
   fi
@@ -12968,7 +12968,7 @@ fi
 
 
 
-if test "$host_os" = cygwin; then
+if test "$host_os" = cygwin || test "$host_os" = msys; then
   EXTRAZSHOBJS="$EXTRAZSHOBJS zsh.res.o"
 fi
 
