diff --git Src/exec.c Src/exec.c
index c1181c5eb..4313124e6 100644
--- Src/exec.c
+++ Src/exec.c
@@ -707,7 +707,8 @@ search_defpath(char *cmd, char *pbuf, int plen)
 		    continue;
 		strucpy(&s, ps);
 	    }
-	    *s++ = '/';
+	    if (s == pbuf || s[-1] != '/')
+		*s++ = '/';
 	    if ((s - pbuf) + strlen(cmd) >= plen)
 		continue;
 	    strucpy(&s, cmd);
@@ -841,7 +842,8 @@ execute(LinkList args, int flags, int defpath)
 		    } else if (**pp != '/') {
 			z = buf;
 			strucpy(&z, *pp);
-			*z++ = '/';
+			if (z[-1] != '/')
+			    *z++ = '/';
 			strcpy(z, arg0);
 			ee = zexecve(buf, argv, newenvp);
 			if (isgooderr(ee, *pp))
@@ -866,7 +868,8 @@ execute(LinkList args, int flags, int defpath)
 	    } else {
 		z = buf;
 		strucpy(&z, *pp);
-		*z++ = '/';
+		if (z[-1] != '/')
+		    *z++ = '/';
 		strcpy(z, arg0);
 		ee = zexecve(buf, argv, newenvp);
 		if (isgooderr(ee, *pp))
@@ -935,7 +938,8 @@ findcmd(char *arg0, int docopy, int default_path)
 		    z = buf;
 		    if (**pp) {
 			strucpy(&z, *pp);
-			*z++ = '/';
+			if (z[-1] != '/')
+			    *z++ = '/';
 		    }
 		    strcpy(z, arg0);
 		    RET_IF_COM(buf);
@@ -950,7 +954,8 @@ findcmd(char *arg0, int docopy, int default_path)
 	z = buf;
 	if (**pp) {
 	    strucpy(&z, *pp);
-	    *z++ = '/';
+	    if (z[-1] != '/')
+		*z++ = '/';
 	}
 	strcpy(z, arg0);
 	RET_IF_COM(buf);
@@ -1025,7 +1030,8 @@ hashcmd(char *arg0, char **pp)
 	if (**pp == '/') {
 	    s = buf;
 	    struncpy(&s, *pp, PATH_MAX);
-	    *s++ = '/';
+	    if (s[-1] != '/')
+		*s++ = '/';
 	    if ((s - buf) + strlen(arg0) >= PATH_MAX)
 		continue;
 	    strcpy(s, arg0);
