--- emacs-29.3/src/Makefile.in.orig	2024-01-06 13:56:31.000000000 +0100
+++ emacs-29.3/src/Makefile.in	2024-06-11 07:41:07.067826400 +0200
@@ -613,9 +613,6 @@
 emacs$(EXEEXT): temacs$(EXEEXT) \
                 lisp.mk $(etc)/DOC $(lisp) \
                 $(lispsource)/international/charprop.el ${charsets}
-ifeq ($(SYSTEM_TYPE),cygwin)
-	find ${top_builddir} -name '*.eln' | rebase -v -O -T -
-endif
 ifeq ($(DUMPING),unexec)
 	LC_ALL=C $(RUN_TEMACS) -batch $(BUILD_DETAILS) -l loadup --temacs=dump
   ifneq ($(PAXCTL_dumped),)
