--- emacs-31.1/src/Makefile.in.orig	2026-08-24 17:17:52.787999000 +0200
+++ emacs-31.1/src/Makefile.in	2026-08-24 17:20:18.274448500 +0200
@@ -630,9 +630,6 @@
 emacs$(EXEEXT): temacs$(EXEEXT) \
                 $(abs_top_builddir)/src/lisp.mk $(etc)/DOC $(lisp) \
                 $(lispsource)/international/charprop.el ${charsets}
-ifeq ($(SYSTEM_TYPE),cygwin)
-	find ${top_builddir} -name '*.eln' | rebase -v -O -T -
-endif
 	rm -f $@ && cp -f temacs$(EXEEXT) $@
 
 ## On Haiku, also produce a binary named Emacs with the appropriate
