--- profiler.c	2015-04-02 15:23:06.000000000 +0800
+++ profiler_ricky.c	2015-11-21 08:49:42.605795300 +0800
@@ -218,6 +218,12 @@ static EMACS_INT current_sampling_interv
 
 /* Signal handler for sampling profiler.  */
 
+/* timer_getoverrun is not implemented on Cygwin, but the following
+   seems to be good enough for profiling. */
+#ifdef CYGWIN
+#define timer_getoverrun(x) 0
+#endif
+
 static void
 handle_profiler_signal (int signal)
 {
