--- tcsh-6.18.01/config.sub.orig	2011-12-28 06:20:03.000000000 +0900
+++ tcsh-6.18.01/config.sub	2014-08-15 16:05:17.453684900 +0900
@@ -1294,7 +1294,7 @@
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
-	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
+	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
--- tcsh-6.18.01/config.guess.orig	2011-12-28 06:20:03.000000000 +0900
+++ tcsh-6.18.01/config.guess	2014-08-15 16:12:29.397342600 +0900
@@ -801,6 +801,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    *:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
--- tcsh-6.18.01/config.rpath.orig	2008-08-30 16:56:27.000000000 +0900
+++ tcsh-6.18.01/config.rpath	2014-08-15 16:24:49.628258600 +0900
@@ -112,7 +112,7 @@
 hardcode_minus_L=no
 
 case "$host_os" in
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     # FIXME: the MSVC++ port hasn't been tested in a loooong time
     # When not using gcc, we currently assume that we are using
     # Microsoft Visual C++.
@@ -152,7 +152,7 @@
         ld_shlibs=no
       fi
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
@@ -278,7 +278,7 @@
       ;;
     bsdi[45]*)
       ;;
-    cygwin* | mingw* | pw32*)
+    cygwin* | msys* | mingw* | pw32*)
       # When not using gcc, we currently assume that we are using
       # Microsoft Visual C++.
       # hardcode_libdir_flag_spec is actually meaningless, as there is
@@ -459,7 +459,7 @@
     ;;
   bsdi[45]*)
     ;;
-  cygwin* | mingw* | pw32*)
+  cygwin* | msys* | mingw* | pw32*)
     shrext=.dll
     ;;
   darwin* | rhapsody*)
--- tcsh-6.18.01/configure.in.orig	2012-02-14 01:33:50.000000000 +0900
+++ tcsh-6.18.01/configure.in	2014-08-15 16:28:40.326872700 +0900
@@ -239,6 +239,11 @@
     tcsh_config_file=cray
   ;;
 
+  ## Minimal System
+  *-msys)
+    tcsh_config_file=msys
+  ;;
+
   ## Red Hat Cygwin
   *-cygwin)
     tcsh_config_file=cygwin
