From 3a6a36176055dc74aab536807a26429ba7dd76ab Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Thu, 15 Jan 2026 07:32:29 +0800
Subject: [PATCH] Add IMPORTED_IMPLIB for windows and cygwin

---
 meson.build | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 561b41c..a0cd4ee 100644
--- a/meson.build
+++ b/meson.build
@@ -82,8 +82,17 @@ if get_option('default_library') == 'shared' or get_option('default_library') ==
    cmakeconf.set('MESON_SHARED_TARGET', '''
 add_library(jsoncpp_lib IMPORTED SHARED)
 set_target_properties(jsoncpp_lib PROPERTIES
-  IMPORTED_LOCATION "''' + join_paths('${PACKAGE_PREFIX_DIR}', get_option('libdir'), shared_name) + '''"
-  INTERFACE_INCLUDE_DIRECTORIES "''' + join_paths('${PACKAGE_PREFIX_DIR}', get_option('includedir')) + '")')
+  IMPORTED_LOCATION "@0@"
+  INTERFACE_INCLUDE_DIRECTORIES "@1@"
+)
+set_target_properties(jsoncpp_lib PROPERTIES
+  IMPORTED_IMPLIB "@2@"
+)
+'''.format(
+  join_paths('${PACKAGE_PREFIX_DIR}', get_option('bindir'), shared_name),
+  join_paths('${PACKAGE_PREFIX_DIR}', get_option('includedir')),
+  join_paths('${PACKAGE_PREFIX_DIR}', get_option('libdir'), 'lib' + jsoncpp_lib.name() + '.dll.a'),
+))
 endif
 if get_option('default_library') == 'static' or get_option('default_library') == 'both'
    cmakeconf.set('MESON_STATIC_TARGET', '''
-- 
2.52.0.windows.1

