--- jsoncpp-1.6.2/src/lib_json/json_writer.cpp.old	2015-04-11 20:45:33.000000000 +0100
+++ jsoncpp-1.6.2/src/lib_json/json_writer.cpp	2015-06-05 00:14:09.610468200 +0100
@@ -29,7 +29,7 @@
 
 #if defined(_MSC_VER) && _MSC_VER < 1500 // VC++ 8.0 and below
 #define snprintf _snprintf
-#elif __cplusplus >= 201103L
+#elif __cplusplus >= 201103L && !defined(__CYGWIN__)
 #define snprintf std::snprintf
 #endif
 
