--- origsrc/doxygen-1.7.4/tmake/lib/linux-g++/tmake.conf	2008-12-06 06:16:20.000000000 -0700
+++ src/doxygen-1.7.4/tmake/lib/linux-g++/tmake.conf	2011-07-26 15:48:50.130200000 -0600
@@ -42,7 +42,7 @@ TMAKE_LFLAGS_DEBUG	= -g
 TMAKE_LFLAGS_SHLIB	= -shared
 TMAKE_LFLAGS_SONAME	= -Wl,-soname,
 
-TMAKE_LIBS		=
+TMAKE_LIBS		= -liconv
 TMAKE_LIBS_X11		= -lXext -lX11 -lm
 TMAKE_LIBS_X11SM	= -lICE -lSM
 TMAKE_LIBS_QT		= -lqt
