--- doxygen-1.8.8/tmake/lib/linux-g++/tmake.conf.orig	2014-11-07 21:22:37.755309800 +0000
+++ doxygen-1.8.8/tmake/lib/linux-g++/tmake.conf	2014-11-07 21:22:42.518582300 +0000
@@ -12,7 +12,7 @@
 TMAKE_CFLAGS_WARN_ON	= -Wall -W 
 TMAKE_CFLAGS_WARN_OFF	=
 TMAKE_CFLAGS_RELEASE	= -O2
-TMAKE_CFLAGS_DEBUG	= -g
+TMAKE_CFLAGS_DEBUG	= -g -Wa,-mbig-obj
 TMAKE_CFLAGS_SHLIB	= -fPIC
 TMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 
