--- doxygen-1.8.7/src/libdoxygen.t.in.orig	2014-05-13 09:28:23.279400000 +0400
+++ doxygen-1.8.7/src/libdoxygen.t.in	2014-05-13 09:31:13.475400000 +0400
@@ -105,20 +105,20 @@
 #$ GenerateLex("constexp",0);
 
 #$ GenerateDep("\$(GENERATED_SRC)/ce_parse.cpp","constexp.y");
-	$(YACC) -l -p constexpYY constexp.y -o \$(GENERATED_SRC)/ce_parse.cpp 
+	$(YACC) -l -p constexpYY constexp.y -o $(GENERATED_SRC)/ce_parse.cpp 
 
 #$ GenerateDep("\$(GENERATED_SRC)/ce_parse.h","constexp.y");
-	$(YACC) -l -d -p ce_parsexpYY constexp.y -o \$(GENERATED_SRC)/ce_parse.c 
+	$(YACC) -l -d -p ce_parsexpYY constexp.y -o $(GENERATED_SRC)/ce_parse.c 
 	-rm $(GENERATED_SRC)/ce_parse.c
 
 #$ GenerateDep("\$(GENERATED_SRC)/vhdlscanner.cpp","vhdlscanner.l","\$(GENERATED_SRC)/vhdlparser.h");
 #$ GenerateLex("vhdlscanner",1);
 
 #$ GenerateDep("\$(GENERATED_SRC)/vhdlparser.cpp","vhdlparser.y");
-	$(YACC) -l -p vhdlscannerYY vhdlparser.y -o \$(GENERATED_SRC)/vhdlparser.cpp 	
+	$(YACC) -l -p vhdlscannerYY vhdlparser.y -o $(GENERATED_SRC)/vhdlparser.cpp 	
 
 #$ GenerateDep("\$(GENERATED_SRC)/vhdlparser.h","vhdlparser.y");
-	$(YACC) -l -d -p vhdlscannerYY vhdlparser.y -o \$(GENERATED_SRC)/vhdlparser.c 
+	$(YACC) -l -d -p vhdlscannerYY vhdlparser.y -o $(GENERATED_SRC)/vhdlparser.c 
 	-rm $(GENERATED_SRC)/vhdlparser.c	
 	
 TO_C_CMD=$(PYTHON) to_c_cmd.py < $< > $@
