--- namcap-3.3.1/Namcap/rules/__init__.py.orig	2022-06-23 11:21:01.000000000 +0200
+++ namcap-3.3.1/Namcap/rules/__init__.py	2022-07-02 23:33:10.803349000 +0200
@@ -23,7 +23,7 @@
 # Tarball rules
 from . import (
   anyelf,
-  elffiles,
+  #elffiles,
   emptydir,
   externalhooks,
   fhs,
@@ -36,7 +36,7 @@
   javafiles,
   kdeprograms,
   libtool,
-  licensepkg,
+  #licensepkg,
   lotsofdocs,
   missingbackups,
   pathdepends,
@@ -44,11 +44,11 @@
   permissions,
   py_mtime,
   pydepends,
-  rpath,
-  runpath,
+  #rpath,
+  #runpath,
   scrollkeeper,
   shebangdepends,
-  sodepends,
+  #sodepends,
   symlink,
   systemdlocation,
   unusedsodepends
