--- namcap-3.2.10/Namcap/rules/__init__.py.orig	2019-12-15 19:37:54.000000000 +0100
+++ namcap-3.2.10/Namcap/rules/__init__.py	2020-05-29 08:05:59.885381000 +0200
@@ -23,7 +23,7 @@
 # Tarball rules
 from . import (
   anyelf,
-  elffiles,
+  #elffiles,
   emptydir,
   externalhooks,
   fhs,
@@ -36,17 +36,17 @@
   javafiles,
   kdeprograms,
   libtool,
-  licensepkg,
+  #licensepkg,
   lotsofdocs,
   missingbackups,
   pathdepends,
   perllocal,
   permissions,
   py_mtime,
-  rpath,
+  #rpath,
   scrollkeeper,
   shebangdepends,
-  sodepends,
+  #sodepends,
   symlink,
   systemdlocation,
   unusedsodepends
--- namcap-3.2.10/setup.py.orig	2019-12-15 19:37:54.000000000 +0100
+++ namcap-3.2.10/setup.py	2020-05-29 07:50:19.848393400 +0200
@@ -16,7 +16,7 @@
 
 	py_modules=["namcap"],
 	packages = find_packages(),
-	scripts=["namcap", 'parsepkgbuild'],
+	scripts=["namcap.sh", 'parsepkgbuild'],
 	test_suite = "Namcap.tests",
 	data_files=DATAFILES)
 