--- namcap-3.3.1/setup.py.orig	2026-09-04 04:10:41.246735200 +0000
+++ namcap-3.3.1/setup.py	2026-09-04 04:14:00.268733500 +0000
@@ -4,9 +4,9 @@
 import Namcap.version
 
 DATAFILES = [
-    ("/usr/share/man/man1", ["namcap.1"]),
-    ("/usr/share/namcap", ["namcap-tags", "parsepkgbuild.sh"]),
-    ("/usr/share/doc/namcap", ["README.md", "AUTHORS"]),
+    ("share/man/man1", ["namcap.1"]),
+    ("share/namcap", ["namcap-tags", "parsepkgbuild.sh"]),
+    ("share/doc/namcap", ["README.md", "AUTHORS"]),
 ]
 
 setup(
