--- namcap-3.3.1/setup.py.orig	2022-06-23 11:21:01.000000000 +0200
+++ namcap-3.3.1/setup.py	2022-12-18 17:19:16.630188300 +0100
@@ -2,9 +2,9 @@
 from setuptools import setup, find_packages
 import Namcap.version
 
-DATAFILES = [('/usr/share/man/man1', ['namcap.1']),
-		('/usr/share/namcap', ['namcap-tags', 'parsepkgbuild.sh']),
-		('/usr/share/doc/namcap',['README','AUTHORS','TODO'])]
+DATAFILES = [('share/man/man1', ['namcap.1']),
+		('share/namcap', ['namcap-tags', 'parsepkgbuild.sh']),
+		('share/doc/namcap',['README','AUTHORS','TODO'])]
 
 setup(name="namcap",
 	version=Namcap.version.get_version(),
