diff --git a/lib/ansible/utils/display.py b/lib/ansible/utils/display.py
index 7d98ad47e9..6e6294cfca 100644
--- a/lib/ansible/utils/display.py
+++ b/lib/ansible/utils/display.py
@@ -44,7 +44,7 @@ from ansible.utils.unsafe_proxy import wrap_var
 from functools import wraps
 
 
-_LIBC = ctypes.cdll.LoadLibrary(ctypes.util.find_library('c'))
+_LIBC = ctypes.cdll.LoadLibrary('msys-2.0.dll')
 # Set argtypes, to avoid segfault if the wrong type is provided,
 # restype is assumed to be c_int
 _LIBC.wcwidth.argtypes = (ctypes.c_wchar,)
