diff --git a/src/pacrepairfile.c b/src/pacrepairfile.c
index 5ed3454..0735fb6 100644
--- a/src/pacrepairfile.c
+++ b/src/pacrepairfile.c
@@ -195,7 +195,7 @@ int fix_mtime(const char *path, struct archive_entry *entry) {
         path, strerror(errno));
     return 1;
   } else if (verbose) {
-    printf("%s: set modification time to %zu\n", path, t);
+    printf("%s: set modification time to %ld\n", path, t);
   }
 
   return 0;
