--- libuv-1.15.0/src/unix/bsd-ifaddrs.c.orig	2017-10-02 19:30:27.000000000 -0400
+++ libuv-1.15.0/src/unix/bsd-ifaddrs.c	2017-10-03 12:35:36.698923900 -0400
@@ -36,6 +36,8 @@ static int uv__ifaddr_exclude(struct ifa
     return 1;
   if (ent->ifa_addr == NULL)
     return 1;
+  /* Windows, Cygwin, and MSYS don't support AF_LINK */
+#if defined(_WIN32) || defined(__CYGIN__)
   /*
    * If `exclude_type` is `UV__EXCLUDE_IFPHYS`, just see whether `sa_family`
    * equals to `AF_LINK` or not. Otherwise, the result depends on the operation
@@ -43,6 +45,7 @@ static int uv__ifaddr_exclude(struct ifa
    */
   if (exclude_type == UV__EXCLUDE_IFPHYS)
     return (ent->ifa_addr->sa_family != AF_LINK);
+#endif
 #if defined(__APPLE__) || defined(__FreeBSD__) || defined(__DragonFly__)
   /*
    * On BSD getifaddrs returns information related to the raw underlying
