--- cmake-3.17.3/Modules/FindTclsh.cmake.orig	2020-05-28 12:57:26.000000000 +0200
+++ cmake-3.17.3/Modules/FindTclsh.cmake	2020-06-21 13:01:38.095169400 +0200
@@ -16,14 +16,8 @@
   TCLSH_FOUND = TRUE if tclsh has been found
   TCL_TCLSH = the path to the tclsh executable
 
-In cygwin, look for the cygwin version first.  Don't look for it later
-to avoid finding the cygwin version on a Win32 build.
 #]=======================================================================]
 
-if(CYGWIN)
-  find_program(TCL_TCLSH NAMES cygtclsh83 cygtclsh80)
-endif()
-
 get_filename_component(TK_WISH_PATH "${TK_WISH}" PATH)
 get_filename_component(TK_WISH_PATH_PARENT "${TK_WISH_PATH}" PATH)
 string(REGEX REPLACE
