--- cmake-3.17.3/Source/cmInstallCommand.cxx.orig	2020-05-28 12:57:26.000000000 +0200
+++ cmake-3.17.3/Source/cmInstallCommand.cxx	2020-06-20 15:35:16.893105600 +0200
@@ -1158,7 +1158,7 @@ bool HandleDirectoryMode(std::vector<std
     } else if (doing == DoingRegex) {
       literal_args += " REGEX \"";
 // Match rules are case-insensitive on some platforms.
-#if defined(_WIN32) || defined(__APPLE__) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__APPLE__) 
       std::string regex = cmSystemTools::LowerCase(args[i]);
 #else
       std::string regex = args[i];
--- cmake-3.17.3/Source/kwsys/Glob.cxx.orig	2020-05-28 12:57:26.000000000 +0200
+++ cmake-3.17.3/Source/kwsys/Glob.cxx	2020-06-20 15:36:21.952845700 +0200
@@ -27,7 +27,7 @@
 #include <stdio.h>
 #include <string.h>
 namespace KWSYS_NAMESPACE {
-#if defined(_WIN32) || defined(__APPLE__) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__APPLE__) 
 // On Windows and Apple, no difference between lower and upper case
 #  define KWSYS_GLOB_CASE_INDEPENDENT
 #endif
