diff -aur cmake-3.5.2/Source/cmFileCommand.cxx.orig cmake-3.5.2/Source/cmFileCommand.cxx
--- cmake-3.5.2/Source/cmFileCommand.cxx.orig	2016-05-22 16:38:04.545005800 -0400
+++ cmake-3.5.2/Source/cmFileCommand.cxx	2016-05-22 16:38:04.806538600 -0400
@@ -1227,7 +1227,7 @@
   MatchProperties CollectMatchProperties(const char* file)
     {
     // Match rules are case-insensitive on some platforms.
-#if defined(_WIN32) || defined(__APPLE__) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__APPLE__)
     std::string lower = cmSystemTools::LowerCase(file);
     const char* file_to_match = lower.c_str();
 #else
diff -aur cmake-3.5.2/Source/cmInstallCommand.cxx.orig cmake-3.5.2/Source/cmInstallCommand.cxx > patch
--- cmake-3.5.2/Source/cmInstallCommand.cxx.orig	2016-05-22 16:38:04.417989000 -0400
+++ cmake-3.5.2/Source/cmInstallCommand.cxx	2016-05-22 16:38:04.812039200 -0400
@@ -1182,7 +1182,7 @@
       {
       literal_args += " REGEX \"";
     // Match rules are case-insensitive on some platforms.
-#if defined(_WIN32) || defined(__APPLE__) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__APPLE__)
       std::string regex = cmSystemTools::LowerCase(args[i]);
 #else
       std::string regex = args[i];
diff -aur cmake-3.5.2/Source/kwsys/Glob.cxx.orig cmake-3.5.2/Source/kwsys/Glob.cxx > patch
--- cmake-3.5.2/Source/kwsys/Glob.cxx.orig	2016-05-22 16:38:04.789536600 -0400
+++ cmake-3.5.2/Source/kwsys/Glob.cxx	2016-05-22 16:38:04.817040300 -0400
@@ -37,7 +37,7 @@
 #include <string.h>
 namespace KWSYS_NAMESPACE
 {
-#if defined(_WIN32) || defined(__APPLE__) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__APPLE__)
 // On Windows and apple, no difference between lower and upper case
 # define KWSYS_GLOB_CASE_INDEPENDENT
 #endif
