diff -aur cmake-3.5.2/CMakeLists.txt.orig-2 cmake-3.5.2/CMakeLists.txt
--- cmake-3.5.2/CMakeLists.txt.orig-2	2016-05-22 15:54:50.164818300 -0400
+++ cmake-3.5.2/CMakeLists.txt	2016-05-22 15:54:50.186949300 -0400
@@ -454,7 +454,7 @@
 
   #---------------------------------------------------------------------
   # Use curses?
-  if (UNIX)
+  if (UNIX AND NOT MSYS)
     # there is a bug in the Syllable libraries which makes linking ccmake fail, Alex
     if(NOT CMAKE_SYSTEM_NAME MATCHES syllable)
       set(CURSES_NEED_NCURSES TRUE)
