--- cmake/src/cmake-3.2.3/CMakeLists.txt.orig	2015-06-05 00:33:15.424175400 +0100
+++ cmake-3.2.3/CMakeLists.txt	2015-06-05 00:27:44.021827700 +0100
@@ -412,7 +412,7 @@
 
   #---------------------------------------------------------------------
   # Use curses?
-  if (UNIX)
+  if (UNIX AND NOT MSYS)
     # there is a bug in the Syllable libraries which makes linking ccmake fail, Alex
     if(NOT CMAKE_SYSTEM_NAME MATCHES syllable)
       set(CURSES_NEED_NCURSES TRUE)
