diff -aur cmake-3.5.2/Modules/Platform/Windows-GNU.cmake.orig cmake-3.5.2/Modules/Platform/Windows-GNU.cmake
--- cmake-3.5.2/Modules/Platform/Windows-GNU.cmake.orig	2016-04-15 09:41:21.000000000 -0400
+++ cmake-3.5.2/Modules/Platform/Windows-GNU.cmake	2016-05-22 16:14:39.524241600 -0400
@@ -77,9 +77,11 @@
   set(CMAKE_${lang}_LINK_OPTIONS_NO_PIE "")
   set(CMAKE_SHARED_LIBRARY_${lang}_FLAGS "")
 
-  set(CMAKE_${lang}_USE_RESPONSE_FILE_FOR_OBJECTS ${__WINDOWS_GNU_LD_RESPONSE})
-  set(CMAKE_${lang}_USE_RESPONSE_FILE_FOR_LIBRARIES ${__WINDOWS_GNU_LD_RESPONSE})
-  set(CMAKE_${lang}_USE_RESPONSE_FILE_FOR_INCLUDES 1)
+  if(NOT CMAKE_GENERATOR STREQUAL "MSYS Makefiles")
+    set(CMAKE_${lang}_USE_RESPONSE_FILE_FOR_OBJECTS ${__WINDOWS_GNU_LD_RESPONSE})
+    set(CMAKE_${lang}_USE_RESPONSE_FILE_FOR_LIBRARIES ${__WINDOWS_GNU_LD_RESPONSE})
+    set(CMAKE_${lang}_USE_RESPONSE_FILE_FOR_INCLUDES 1)
+  endif()
 
   # We prefer "@" for response files but it is not supported by gcc 3.
   execute_process(COMMAND ${CMAKE_${lang}_COMPILER} --version OUTPUT_VARIABLE _ver ERROR_VARIABLE _ver)
