auth_des.c also has numerous undefined symbols

--- origsrc/libtirpc-0.2.4/src/auth_des.c	2013-12-09 14:59:51.000000000 -0600
+++ src/libtirpc-0.2.4/src/auth_des.c	2014-06-09 18:54:34.209620700 -0500
@@ -31,6 +31,7 @@
 /*
  * auth_des.c, client-side implementation of DES authentication
  */
+#ifndef __CYGWIN__
 #include <pthread.h>
 #include <reentrant.h>
 #include <err.h>
@@ -500,3 +501,5 @@ authdes_ops(void)
 	mutex_unlock(&authdes_ops_lock);
 	return (&ops);
 }
+
+#endif /* !__CYGWIN__ */
--- origsrc/libtirpc-0.2.4/src/auth_time.c	2013-12-09 14:59:51.000000000 -0600
+++ src/libtirpc-0.2.4/src/auth_time.c	2014-06-09 18:55:28.051457700 -0500
@@ -25,6 +25,7 @@
  *	needed to deal with TCP connections.
  */
 
+#ifndef __CYGWIN__
 #include <sys/cdefs.h>
 #include <stdio.h>
 #include <syslog.h>
@@ -491,3 +492,5 @@ error:
 
 	return (time_valid);
 }
+
+#endif /* !__CYGWIN__ */
--- origsrc/libtirpc-0.2.4/src/rpc_soc.c	2014-06-09 17:08:42.515057900 -0500
+++ src/libtirpc-0.2.4/src/rpc_soc.c	2014-06-09 17:08:43.126135500 -0500
@@ -520,6 +520,7 @@ clnt_broadcast(prog, vers, proc, xargs,
 	    (resultproc_t) rpc_wrap_bcast, "udp");
 }
 
+#ifndef __CYGWIN__
 /*
  * Create the client des authentication object. Obsoleted by
  * authdes_seccreate().
@@ -551,6 +552,7 @@ fallback:
 	dummy = authdes_seccreate(servername, window, NULL, ckey);
 	return (dummy);
 }
+#endif
 
 /*
  * Create a client handle for a unix connection. Obsoleted by clnt_vc_create()
