--- origsrc/libtirpc-0.2.4/src/clnt_bcast.c	2013-12-09 14:59:51.000000000 -0600
+++ src/libtirpc-0.2.4/src/clnt_bcast.c	2014-06-09 19:01:16.077151400 -0500
@@ -71,8 +71,12 @@
 #define	INITTIME 4000	/* Time to wait initially */
 #define	WAITTIME 8000	/* Maximum time to wait */
 
+#ifndef POLLRDNORM
 # define POLLRDNORM     0x040           /* Normal data may be read.  */
+#endif
+#ifndef POLLRDBAND
 # define POLLRDBAND     0x080           /* Priority data may be read.  */
+#endif
 
 
 
@@ -539,7 +543,7 @@ rpc_broadcast_exp(prog, vers, proc, xarg
 		try_again:
 			inlen = recvfrom(fdlist[i].fd, inbuf, fdlist[i].dsize,
 			    0, (struct sockaddr *)(void *)&fdlist[i].raddr,
-			    &fdlist[i].asize);
+			    (socklen_t *)&fdlist[i].asize);
 			if (inlen < 0) {
 				if (errno == EINTR)
 					goto try_again;
