--- origsrc/libtirpc-0.3.2/tirpc/rpc/rpcent.h	2015-06-16 10:35:08.000000000 -0500
+++ src/libtirpc-0.3.2/tirpc/rpc/rpcent.h	2015-10-13 16:51:15.548500000 -0500
@@ -49,7 +49,7 @@ extern "C" {
 
 /* These are defined in /usr/include/rpc/netdb.h, unless we are using
    the C library without RPC support. */
-#if defined(__UCLIBC__) && !defined(__UCLIBC_HAS_RPC__) || !defined(__GLIBC__)
+#if (defined(__UCLIBC__) && !defined(__UCLIBC_HAS_RPC__)) || (!defined(__GLIBC__) && !defined(__CYGWIN__))
 struct rpcent {
 	char	*r_name;	/* name of server for this rpc program */
 	char	**r_aliases;	/* alias list */
