--- origsrc/libtirpc-0.3.2/configure.ac	2015-10-13 13:06:41.829054900 -0500
+++ src/libtirpc-0.3.2/configure.ac	2015-10-13 13:09:15.308044200 -0500
@@ -66,19 +66,24 @@ AC_ARG_ENABLE(symvers,
       [],[enable_symvers=yes])
 AM_CONDITIONAL(SYMVERS, test "x$enable_symvers" = xyes)
 
-AC_CANONICAL_BUILD
+AC_CANONICAL_HOST
 # Check for which host we are on and setup a few things
 # specifically based on the host
-case $build_os in
+case $host_os in
      linux*)
          # Do something specific for linux
          LDFLAG_NOUNDEFINED="-Wl,--no-undefined"
-         AC_SUBST(LDFLAG_NOUNDEFINED)
+         ;;
+     cygwin*)
+         # Do something specific for cygwin
+         LDFLAG_NOUNDEFINED="-no-undefined"
          ;;
      *)
          #Default Case
+         LDFLAG_NOUNDEFINED=
          ;;
 esac
+AC_SUBST(LDFLAG_NOUNDEFINED)
 
 
 AC_CONFIG_HEADERS([config.h])
