--- origsrc/libtirpc-1.0.2/Makefile.am	2017-07-05 10:02:23.000000000 -0500
+++ src/libtirpc-1.0.2/Makefile.am	2018-02-06 21:58:49.569565500 -0600
@@ -1,50 +1,6 @@
-SUBDIRS = src man doc
+SUBDIRS = tirpc src man doc
 ACLOCAL_AMFLAGS = -I m4
 
-noinst_HEADERS	       = tirpc/reentrant.h \
-			 tirpc/getpeereid.h \
-			 tirpc/libc_private.h \
-			 tirpc/un-namespace.h
-
-nobase_include_HEADERS = tirpc/netconfig.h \
-			 tirpc/rpcsvc/crypt.x \
-			 tirpc/rpcsvc/crypt.h \
-			 tirpc/rpc/xdr.h \
-			 tirpc/rpc/types.h \
-			 tirpc/rpc/svc_soc.h \
-			 tirpc/rpc/svc.h \
-			 tirpc/rpc/svc_dg.h \
-			 tirpc/rpc/svc_auth.h \
-			 tirpc/rpc/svc_mt.h \
-			 tirpc/rpc/rpc_msg.h \
-			 tirpc/rpc/rpc.h \
-			 tirpc/rpc/rpcent.h \
-			 tirpc/rpc/rpc_com.h \
-			 tirpc/rpc/rpcb_prot.x \
-			 tirpc/rpc/rpcb_prot.h \
-			 tirpc/rpc/rpcb_clnt.h \
-			 tirpc/rpc/raw.h \
-			 tirpc/rpc/pmap_rmt.h \
-			 tirpc/rpc/pmap_prot.h \
-			 tirpc/rpc/pmap_clnt.h \
-			 tirpc/rpc/nettype.h \
-			 tirpc/rpc/key_prot.h \
-			 tirpc/rpc/des.h \
-			 tirpc/rpc/des_crypt.h \
-			 tirpc/rpc/clnt_stat.h \
-			 tirpc/rpc/clnt_soc.h \
-			 tirpc/rpc/clnt.h \
-			 tirpc/rpc/auth_unix.h \
-			 tirpc/rpc/auth_des.h \
-			 tirpc/rpc/auth.h
-
-if GSS
-nobase_include_HEADERS += \
-			 tirpc/rpc/rpcsec_gss.h \
-			 tirpc/rpc/auth_gss.h \
-			 tirpc/rpc/svc_auth_gss.h
-endif
-
 pkgconfigdir=$(libdir)/pkgconfig
 pkgconfig_DATA = libtirpc.pc
 
--- origsrc/libtirpc-1.0.2/configure.ac	2018-02-06 21:45:15.467634500 -0600
+++ src/libtirpc-1.0.2/configure.ac	2018-02-06 21:59:51.687928500 -0600
@@ -96,3 +96,3 @@ AC_CHECK_HEADERS([arpa/inet.h fcntl.h li
 
-AC_CONFIG_FILES([Makefile src/Makefile man/Makefile doc/Makefile])
+AC_CONFIG_FILES([Makefile tirpc/Makefile src/Makefile man/Makefile doc/Makefile])
 AC_OUTPUT(libtirpc.pc)
--- origsrc/libtirpc-1.0.2/libtirpc.pc.in	2017-07-05 10:02:23.000000000 -0500
+++ src/libtirpc-1.0.2/libtirpc.pc.in	2018-02-06 21:53:20.587043700 -0600
@@ -9,4 +9,4 @@ Requires:
 Version: @PACKAGE_VERSION@
 Libs: -L@libdir@ -ltirpc
 Libs.private: -lpthread
-Cflags: -I@includedir@/tirpc
+Cflags: -I@includedir@
--- origsrc/libtirpc-1.0.2/tirpc/Makefile.am	1969-12-31 18:00:00.000000000 -0600
+++ src/libtirpc-1.0.2/tirpc/Makefile.am	2018-02-06 21:59:10.087938200 -0600
@@ -0,0 +1,43 @@
+noinst_HEADERS	       = reentrant.h \
+			 getpeereid.h \
+			 libc_private.h \
+			 un-namespace.h
+
+nobase_include_HEADERS = netconfig.h \
+			 rpcsvc/crypt.x \
+			 rpcsvc/crypt.h \
+			 rpc/xdr.h \
+			 rpc/types.h \
+			 rpc/svc_soc.h \
+			 rpc/svc.h \
+			 rpc/svc_dg.h \
+			 rpc/svc_auth.h \
+			 rpc/svc_mt.h \
+			 rpc/rpc_msg.h \
+			 rpc/rpc.h \
+			 rpc/rpcent.h \
+			 rpc/rpc_com.h \
+			 rpc/rpcb_prot.x \
+			 rpc/rpcb_prot.h \
+			 rpc/rpcb_clnt.h \
+			 rpc/raw.h \
+			 rpc/pmap_rmt.h \
+			 rpc/pmap_prot.h \
+			 rpc/pmap_clnt.h \
+			 rpc/nettype.h \
+			 rpc/key_prot.h \
+			 rpc/des.h \
+			 rpc/des_crypt.h \
+			 rpc/clnt_stat.h \
+			 rpc/clnt_soc.h \
+			 rpc/clnt.h \
+			 rpc/auth_unix.h \
+			 rpc/auth_des.h \
+			 rpc/auth.h
+
+if GSS
+nobase_include_HEADERS += \
+			 rpc/rpcsec_gss.h \
+			 rpc/auth_gss.h \
+			 rpc/svc_auth_gss.h
+endif
