--- origsrc/libtirpc-1.3.2/tirpc/rpc/types.h	2022-01-26 21:44:56.018912000 +0100
+++ src/libtirpc-1.3.2/tirpc/rpc/types.h	2022-01-26 21:55:20.751367400 +0100
@@ -71,7 +71,7 @@ typedef   int32_t rpc_inline_t;
 #define mem_free(ptr, bsize)	free(ptr)
 
 
-#if defined __APPLE_CC__ || defined __FreeBSD__ || !defined (__GLIBC__)
+#if defined __APPLE_CC__ || defined __FreeBSD__ || defined __CYGWIN__ || !defined (__GLIBC__)
 # define __u_char_defined
 # define __daddr_t_defined
 #endif
