--- origsrc/libtirpc-1.3.2/tirpc/rpc/rpcent.h	2021-05-10 19:59:15.000000000 +0200
+++ src/libtirpc-1.3.2/tirpc/rpc/rpcent.h	2022-01-26 21:48:15.398905000 +0100
@@ -50,7 +50,7 @@ extern "C" {
 
 /* These are defined in /usr/include/rpc/netdb.h, unless we are using
    the C library without RPC support. */
-#if defined(__UCLIBC__) && !defined(__UCLIBC_HAS_RPC__) || !defined(__GLIBC__)
+#if defined(__UCLIBC__) && !defined(__UCLIBC_HAS_RPC__) || !defined(__GLIBC__) && !defined(__CYGWIN__)
 struct rpcent {
 	char	*r_name;	/* name of server for this rpc program */
 	char	**r_aliases;	/* alias list */
