--- origsrc/libtirpc-1.3.7/configure.ac	2025-10-09 19:40:02.245683000 +0200
+++ configure.ac	2025-10-09 19:53:42.506526900 +0200
@@ -202,16 +202,13 @@ AC_CONFIG_HEADERS([config.h])
 LT_INIT
 AC_HEADER_DIRENT
 AC_PREFIX_DEFAULT(/usr)
-AC_CHECK_HEADERS([arpa/inet.h fcntl.h libintl.h limits.h locale.h
-netdb.h netinet/in.h stddef.h stdint.h stdlib.h string.h sys/ioctl.h
-sys/param.h sys/socket.h sys/time.h syslog.h unistd.h features.h
-gssapi/gssapi_ext.h endian.h machine/endian.h])
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h libintl.h limits.h locale.h netdb.h netinet/in.h stddef.h stdint.h stdlib.h string.h sys/ioctl.h sys/param.h sys/socket.h sys/time.h syslog.h unistd.h features.h gssapi/gssapi_ext.h endian.h machine/endian.h])
 AX_PTHREAD
-AC_CHECK_FUNCS([getpeereid getrpcbyname getrpcbynumber setrpcent endrpcent getrpcent])
+AC_CHECK_FUNCS([bindresvport getpeereid getrpcbyname getrpcbynumber setrpcent endrpcent getrpcent])
 
-AC_CHECK_TYPES(struct rpcent,,, [
+AC_CHECK_TYPES([struct rpcent],,, [
       #include <netdb.h>])
-AC_CONFIG_FILES([Makefile src/Makefile man/Makefile doc/Makefile])
+AC_CONFIG_FILES([Makefile tirpc/Makefile src/Makefile man/Makefile doc/Makefile])
 AC_CONFIG_FILES([src/libtirpc.map])
 AC_CONFIG_FILES([libtirpc.pc])
 AC_OUTPUT
--- origsrc/libtirpc-1.3.7/src/bindresvport.c	2025-08-25 17:10:30.000000000 +0200
+++ src/bindresvport.c	2025-10-09 19:48:57.770617800 +0200
@@ -33,6 +33,10 @@
  * Portions Copyright(C) 1996, Jason Downs.  All rights reserved.
  */
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/socket.h>
 
@@ -51,6 +55,7 @@
 #include <string.h>
 #include <reentrant.h>
 
+#if !HAVE_BINDRESVPORT
 extern pthread_mutex_t port_lock;
 
 /*
@@ -334,3 +339,4 @@ bindresvport_sa(sd, sa)
 	return (error);
 }
 #endif
+#endif /* !HAVE_BINDRESVPORT */
