diff -Naur libtirpc-0.2.4.orig/config.guess libtirpc-0.2.4/config.guess
--- libtirpc-0.2.4.orig/config.guess	2013-12-09 22:02:33.000000000 +0100
+++ libtirpc-0.2.4/config.guess	2014-10-15 04:57:52.087890600 +0200
@@ -869,6 +869,9 @@
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2`echo ${UNAME_RELEASE}|sed -e 's/[^.]*//'`
 	exit ;;
