--- fish-3.7.1/cmake/Tests.cmake.orig	2025-03-30 10:38:59.107492700 +0200
+++ fish-3.7.1/cmake/Tests.cmake	2025-03-30 10:39:21.414080300 +0200
@@ -41,15 +41,6 @@
   USES_TERMINAL
 )
 
-# If CMP0037 is available, also make an alias "test" target.
-# Note that this policy may not be available, in which case definining such a target silently fails.
-cmake_policy(PUSH)
-if(POLICY CMP0037)
-  cmake_policy(SET CMP0037 OLD)
-  add_custom_target(test DEPENDS fish_run_tests)
-endif()
-cmake_policy(POP)
-
 # Build the low-level tests code
 add_executable(fish_tests EXCLUDE_FROM_ALL
                src/fish_tests.cpp)
