--- popt-1.18.ORIG/src/poptconfig.c	2020-04-16 12:32:54.000000000 +0200
+++ popt-1.18/src/poptconfig.c	2020-12-31 11:33:48.634910900 +0100
@@ -21,7 +21,7 @@
 #if defined(HAVE_GLOB_H)
 #include <glob.h>
 
-#if !defined(__GLIBC__)
+#if !defined(__GLIBC__) && !defined(__CYGWIN__)
 /* Return nonzero if PATTERN contains any metacharacters.
    Metacharacters can be quoted with backslashes if QUOTE is nonzero.  */
 static int
