From d4fa2c2fd3db741da6690cc68a461dab54abfb11 Mon Sep 17 00:00:00 2001
From: nsf <no.smile.face@gmail.com>
Date: Mon, 26 Sep 2016 02:24:23 +0500
Subject: [PATCH] Add _XOPEN_SOURCE define for python module. Fixes #89.

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 0a8ba7e..db68b15 100644
--- a/setup.py
+++ b/setup.py
@@ -19,5 +19,5 @@
     url = 'http://code.google.com/p/termbox/',
     license = 'MIT',
     cmdclass = {'build_ext': build_ext},
-    ext_modules = [Extension('termbox', sourcefiles, extra_compile_args=["-Wno-error=declaration-after-statement"])],
+    ext_modules = [Extension('termbox', sourcefiles, extra_compile_args=["-D_XOPEN_SOURCE", "-Wno-error=declaration-after-statement"])],
 )
