--- bash-4.1-orig/builtins/read.def	2013-03-01 03:29:56.563290600 -0600
+++ bash-4.1/builtins/read.def	2013-03-01 03:34:52.687228000 -0600
@@ -466,7 +466,7 @@ read_builtin (list)
     }
 
 #if defined (__CYGWIN__) && defined (O_TEXT)
-  setmode (0, O_TEXT);
+  _setmode (0, O_TEXT);
 #endif
 
   ps2 = 0;
--- bash-4.1-orig/general.c	2009-11-28 20:44:46.000000000 -0600
+++ bash-4.1/general.c	2013-03-01 03:34:33.366122900 -0600
@@ -43,6 +43,10 @@
 
 #include <tilde/tilde.h>
 
+#ifdef __CYGWIN__
+#include <sys/cygwin.h>
+#endif
+
 #if !defined (errno)
 extern int errno;
 #endif /* !errno */
@@ -601,7 +605,7 @@ make_absolute (string, dot_path)
     {
       char pathbuf[PATH_MAX + 1];
 
-      cygwin_conv_to_full_posix_path (string, pathbuf);
+      cygwin_conv_path (CCP_WIN_A_TO_POSIX|CCP_ABSOLUTE, string, pathbuf, PATH_MAX+1);
       result = savestring (pathbuf);
     }
 #else
