From 65a5f605c0aa706e7385954af298dbfa3ea6c649 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Sat, 13 Sep 2014 21:32:08 +0100
Subject: [PATCH 1/4] Added Support for MSYS2

---
 configure | 23 ++++++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 7df110d..702ef9e 100755
--- a/configure
+++ b/configure
@@ -79,8 +79,29 @@ case $(uname -s) in
         # msys-1.0.dll, but no gcc or stdc++ DLL.
         UNIX_LDFLAGS_STATIC_LIBSTDCXX=
         ;;
+    MSYS*|MINGW*)
+        echo 'uname -s identifies a Msys2 environment.'
+        IS_MSYS=1
+        case $(uname -m) in
+            i686)
+                echo 'uname -m identifies a i686 environment.'
+                UNIX_GPP=i686-pc-msys-g++ /usr/bin/i686-pc-msys-g++
+                MINGW_GPP="i686-w64-mingw32-g++ /mingw32/bin/i686-w64-mingw32-g++"
+                ;;
+            x86_64)
+                echo 'uname -m identifies a x86_64 environment.'
+                UNIX_GPP="x86_64-pc-msys-g++ /usr/bin/x86_64-pc-msys-g++"
+                MINGW_GPP="x86_64-w64-mingw32-g++ /mingw64/bin/x86_64-w64-mingw32-g++"
+                ;;
+            *)
+                echo 'Error: uname -m did not match either i686 or x86_64.'
+                exit 1
+                ;;
+        esac
+        UNIX_LDFLAGS_STATIC_LIBSTDCXX="-static-libgcc -static-libstdc++ "
+        ;;
     *)
-        echo 'Error: uname -s did not match either CYGWIN* or MINGW*.'
+        echo 'Error: uname -s did not match either CYGWIN* MSYS* or MINGW*.'
         exit 1
         ;;
 esac
-- 
2.1.0

