From 808631e64e8256a686bf6d67621b8b32e1532c8f Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Sat, 13 Sep 2014 22:25:49 +0100
Subject: [PATCH 2/4] remove mingw.org case

---
 configure | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/configure b/configure
index 702ef9e..53dfbae 100755
--- a/configure
+++ b/configure
@@ -70,15 +70,6 @@ case $(uname -s) in
 	    esac
         UNIX_LDFLAGS_STATIC_LIBSTDCXX="-static-libgcc -static-libstdc++"
         ;;
-    MINGW*)
-        echo 'uname -s identifies a MSYS environment.'
-        IS_MSYS=1
-        UNIX_GPP=i686-pc-msys-g++
-        MINGW_GPP=mingw32-g++
-        # The MSYS compiler does not recognize -static-libstdc++.  There is a
-        # msys-1.0.dll, but no gcc or stdc++ DLL.
-        UNIX_LDFLAGS_STATIC_LIBSTDCXX=
-        ;;
     MSYS*|MINGW*)
         echo 'uname -s identifies a Msys2 environment.'
         IS_MSYS=1
-- 
2.1.0

