From 2f9c59cf632f0b01b81757d354162d2570f1d5c4 Mon Sep 17 00:00:00 2001
From: David Macek <david.macek.0@gmail.com>
Date: Thu, 8 Oct 2015 21:38:59 +0200
Subject: [PATCH] Add support for conversion of all argv path args

---
 unix-adapter/main.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/unix-adapter/main.cc b/unix-adapter/main.cc
index 7e75d28..6c34fd9 100644
--- a/unix-adapter/main.cc
+++ b/unix-adapter/main.cc
@@ -377,7 +377,7 @@ int main(int argc, char *argv[])
         std::vector<std::string> argVector;
         argVector.push_back(convertPosixPathToWin(argv[1]));
         for (int i = 2; i < argc; ++i)
-            argVector.push_back(argv[i]);
+            argVector.push_back(convertPosixPathToWin(argv[i]));
         std::string cmdLine = argvToCommandLine(argVector);
         wchar_t *cmdLineW = heapMbsToWcs(cmdLine.c_str());
         int ret = winpty_start_process(winpty,
-- 
2.5.2

