From bee434ceb77bac5801caf13412f4041507cb51e9 Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Mon, 15 Sep 2014 16:14:08 +0100
Subject: [PATCH 3/3] rename console.exe to winpty.exe for mintty unicode supp

---
 Makefile              | 2 +-
 unix-adapter/Makefile | 2 +-
 2 files changed, 2 insertion(+), 2 deletion(-)

diff --git a/Makefile b/Makefile
index a89aee1..1d8d89b 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@
 
 install : all
 	mkdir -p $(PREFIX)/bin
-	install -m 755 -p -s build/console.exe $(PREFIX)/bin
+	install -m 755 -p -s build/winpty.exe $(PREFIX)/bin
 	install -m 755 -p -s build/winpty.dll $(PREFIX)/bin
 	install -m 755 -p -s build/winpty-agent.exe $(PREFIX)/bin
 	install -m 755 -p -s build/winpty-debugserver.exe $(PREFIX)/bin
diff --git a/unix-adapter/Makefile b/unix-adapter/Makefile
index a89aee1..1d8d89b 100644
--- a/unix-adapter/Makefile
+++ b/unix-adapter/Makefile
@@ -21,7 +21,7 @@
 include ../config.mk
 include ../config-unix.mk
 
-PROGRAM = ../build/console.exe
+PROGRAM = ../build/winpty.exe
 OBJECTS = main.o Shared.o
 CXXFLAGS += -I../include
 LDFLAGS += $(LDFLAGS_STATIC_LIBSTDCXX) ../build/winpty.dll
-- 
2.1.0

