From 95824f5edd6941bfa7b8a582a1998b70d29e090f Mon Sep 17 00:00:00 2001
From: martell <martellmalone@gmail.com>
Date: Sat, 13 Sep 2014 22:37:30 +0100
Subject: [PATCH 3/4] Fix path convert for MSYS2 (hacky)

---
 unix-adapter/main.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/unix-adapter/main.cc b/unix-adapter/main.cc
index bd9281f..dcceabd 100644
--- a/unix-adapter/main.cc
+++ b/unix-adapter/main.cc
@@ -217,6 +217,13 @@ static void setFdNonBlock(int fd)
     fcntl(fd, F_SETFL, status | O_NONBLOCK);
 }
 
+extern "C" int
+cygwin_conv_to_win32_path (const char path, char win32_path)
+{
+return cygwin_conv_path (CCP_POSIX_TO_WIN_A | CCP_RELATIVE, path, win32_path,
+MAX_PATH);
+}
+
 // Convert the path to a Win32 path if it is a POSIX path, and convert slashes
 // to backslashes.
 static std::string convertPosixPathToWin(const std::string &path)
-- 
2.1.0

