--- unix-tree-2.0.4/Makefile.orig	2022-09-06 15:31:06.000000000 +0200
+++ unix-tree-2.0.4/Makefile	2022-09-10 13:59:54.433331000 +0200
@@ -54,9 +54,9 @@
 #MANDIR=${prefix}/share/man
 
 # Uncomment for Cygwin:
-#CFLAGS+=-O2 -Wall -fomit-frame-pointer
-#LDFLAGS+=-s
-#TREE_DEST=tree.exe
+CFLAGS+=-O2 -Wall -fomit-frame-pointer
+LDFLAGS+=-s
+TREE_DEST=tree.exe
 
 # Uncomment for OS X:
 # It is not allowed to install to /usr/bin on OS X any longer (SIP):
