--- tree-1.7.0/Makefile.old	2014-04-19 02:35:39.000000000 +0900
+++ tree-1.7.0/Makefile	2014-09-14 02:24:32.800993500 +0900
@@ -47,10 +47,10 @@
 #MANDIR=${prefix}/share/man/man1
 
 # Uncomment for Cygwin:
-#CFLAGS=-O2 -Wall -fomit-frame-pointer -DCYGWIN
-#LDFLAGS=-s
-#TREE_DEST=tree.exe
-#OBJS+=strverscmp.o
+CFLAGS=-O2 -Wall -fomit-frame-pointer -DCYGWIN
+LDFLAGS=-s
+TREE_DEST=tree.exe
+OBJS+=strverscmp.o
 
 # Uncomment for OS X:
 #CC=cc
