--- unix-tree-2.2.1/Makefile.orig	2025-01-16 08:44:15.661894300 +0100
+++ unix-tree-2.2.1/Makefile	2025-01-16 08:46:11.052690100 +0100
@@ -60,10 +60,10 @@
 #MANDIR=${prefix}/share/man
 
 # Uncomment for Cygwin:
-#CFLAGS?=-O2
-#CFLAGS+=-Wall -fomit-frame-pointer
-#LDFLAGS+=-s
-#TREE_DEST=tree.exe
+CFLAGS?=-O2
+CFLAGS+=-Wall -fomit-frame-pointer
+LDFLAGS+=-s
+TREE_DEST=tree.exe
 
 # Uncomment for MacOS:
 # It is not allowed to install to /usr/bin on MacOS any longer (SIP):
