From 35535856b9457cbc6d999bd4ac1a3ef695b3f69e Mon Sep 17 00:00:00 2001
From: Corinna Vinschen <vinschen@redhat.com>
Date: Thu, 10 Feb 2022 18:19:29 +0100
Subject: [PATCH 1/6] compat code for fido_dev_is_winhello()

Signed-off-by: Corinna Vinschen <vinschen@redhat.com>
---
 configure.ac | 1 +
 sk-usbhid.c  | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/configure.ac b/configure.ac
index 17fb1e609fbd..473dc38e374a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3235,6 +3235,7 @@ if test "x$enable_sk" = "xyes" -a "x$enable_sk_internal" = "xyes" ; then
 		fido_dev_get_touch_begin \
 		fido_dev_get_touch_status \
 		fido_dev_supports_cred_prot \
+		fido_dev_is_winhello \
 	])
 	LIBS="$saved_LIBS"
 	AC_CHECK_HEADER([fido.h], [],
diff --git a/sk-usbhid.c b/sk-usbhid.c
index 2d36ac337ffa..721076c7f4a4 100644
--- a/sk-usbhid.c
+++ b/sk-usbhid.c
@@ -381,6 +381,14 @@ fido_assert_set_clientdata(fido_assert_t *assert, const u_char *ptr, size_t len)
 }
 #endif /* HAVE_FIDO_ASSERT_SET_CLIENTDATA */
 
+#ifndef HAVE_FIDO_DEV_IS_WINHELLO
+static bool
+fido_dev_is_winhello(const fido_dev_t *)
+{
+	return false;
+}
+#endif /* HAVE_FIDO_DEV_IS_WINHELLO */
+
 /* Check if the specified key handle exists on a given sk. */
 static int
 sk_try(const struct sk_usbhid *sk, const char *application,
-- 
2.35.1

