diff -Naur openssh-6.3p1-orig/config.guess openssh-6.3p1/config.guess
--- openssh-6.3p1-orig/config.guess	2013-04-18 05:26:26.000000000 +0400
+++ openssh-6.3p1/config.guess	2013-10-20 21:05:15.482000000 +0400
@@ -851,6 +851,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur openssh-6.3p1-orig/configure openssh-6.3p1/configure
--- openssh-6.3p1-orig/configure	2013-09-13 10:20:40.000000000 +0400
+++ openssh-6.3p1/configure	2013-10-20 21:04:13.523400000 +0400
@@ -6686,7 +6686,7 @@
 $as_echo "#define DISABLE_WTMP 1" >>confdefs.h
 
 	;;
-*-*-cygwin*)
+*-*-cygwin* | *-*-msys*)
 	check_for_libcrypt_later=1
 	LIBS="$LIBS /usr/lib/textreadmode.o"
 
diff -Naur openssh-6.3p1-orig/configure.ac openssh-6.3p1/configure.ac
--- openssh-6.3p1-orig/configure.ac	2013-08-04 15:48:41.000000000 +0400
+++ openssh-6.3p1/configure.ac	2013-10-20 21:04:13.570200000 +0400
@@ -497,7 +497,7 @@
 	AC_DEFINE([DISABLE_UTMP], [1], [Define if you don't want to use utmp])
 	AC_DEFINE([DISABLE_WTMP], [1], [Define if you don't want to use wtmp])
 	;;
-*-*-cygwin*)
+*-*-cygwin* | *-*-msys*)
 	check_for_libcrypt_later=1
 	LIBS="$LIBS /usr/lib/textreadmode.o"
 	AC_DEFINE([HAVE_CYGWIN], [1], [Define if you are on Cygwin])
diff -Naur openssh-6.3p1-orig/contrib/cygwin/Makefile openssh-6.3p1/contrib/cygwin/Makefile
--- openssh-6.3p1-orig/contrib/cygwin/Makefile	2011-10-25 02:38:00.000000000 +0400
+++ openssh-6.3p1/contrib/cygwin/Makefile	2013-10-20 21:04:13.570200000 +0400
@@ -7,7 +7,7 @@
 mandir=$(datadir)/man
 docdir=$(datadir)/doc
 sshdocdir=$(docdir)/openssh
-cygdocdir=$(docdir)/Cygwin
+cygdocdir=$(docdir)/MSYS
 sysconfdir=/etc
 defaultsdir=$(sysconfdir)/defaults/etc
 inetdefdir=$(defaultsdir)/inetd.d
diff -Naur openssh-6.3p1-orig/openbsd-compat/bsd-cygwin_util.c openssh-6.3p1/openbsd-compat/bsd-cygwin_util.c
--- openssh-6.3p1-orig/openbsd-compat/bsd-cygwin_util.c	2013-06-02 02:07:32.000000000 +0400
+++ openssh-6.3p1/openbsd-compat/bsd-cygwin_util.c	2013-10-20 21:04:13.570200000 +0400
@@ -67,7 +67,7 @@
 	{ NL("ALLUSERSPROFILE=") },
 	{ NL("COMPUTERNAME=") },
 	{ NL("COMSPEC=") },
-	{ NL("CYGWIN=") },
+	{ NL("MSYS=") },
 	{ NL("OS=") },
 	{ NL("PATH=") },
 	{ NL("PATHEXT=") },
diff -Naur openssh-6.3p1-orig/regress/agent-ptrace.sh openssh-6.3p1/regress/agent-ptrace.sh
--- openssh-6.3p1-orig/regress/agent-ptrace.sh	2011-01-17 03:20:20.000000000 +0300
+++ openssh-6.3p1/regress/agent-ptrace.sh	2013-10-20 21:04:13.570200000 +0400
@@ -5,7 +5,7 @@
 
 if have_prog uname ; then
 	case `uname` in
-	AIX|CYGWIN*|OSF1)
+	AIX|CYGWIN*|MSYS*|OSF1)
 		echo "skipped (not supported on this platform)"
 		exit 0
 		;;
diff -Naur openssh-6.3p1-orig/regress/reexec.sh openssh-6.3p1/regress/reexec.sh
--- openssh-6.3p1-orig/regress/reexec.sh	2013-05-17 14:41:07.000000000 +0400
+++ openssh-6.3p1/regress/reexec.sh	2013-10-20 21:04:13.570200000 +0400
@@ -45,7 +45,7 @@
 cp $OBJ/sshd_config.orig $OBJ/sshd_config
 
 # cygwin can't fork a deleted binary
-if [ "$os" != "cygwin" ]; then
+if [ "$os" != "cygwin" || "$os" != "msys" ]; then
 
 verbose "test reexec fallback"
 
diff -Naur openssh-6.3p1-orig/regress/sftp-cmds.sh openssh-6.3p1/regress/sftp-cmds.sh
--- openssh-6.3p1-orig/regress/sftp-cmds.sh	2013-07-25 06:11:20.000000000 +0400
+++ openssh-6.3p1/regress/sftp-cmds.sh	2013-10-20 21:04:13.585800000 +0400
@@ -77,7 +77,7 @@
 	|| fail "get failed"
 cmp $DATA ${COPY} || fail "corrupted copy after get"
 
-if [ "$os" != "cygwin" ]; then
+if [ "$os" != "cygwin" || "$os" != "msys" ]; then
 rm -f ${QUOTECOPY}
 cp $DATA ${QUOTECOPY}
 verbose "$tid: get filename with quotes"
@@ -136,7 +136,7 @@
 	${SFTP} -D ${SFTPSERVER} >/dev/null 2>&1 || fail "put failed"
 cmp $DATA ${COPY} || fail "corrupted copy after put"
 
-if [ "$os" != "cygwin" ]; then
+if [ "$os" != "cygwin" || "$os" != "msys" ]; then
 rm -f ${QUOTECOPY}
 verbose "$tid: put filename with quotes"
 echo "put $DATA \"$QUOTECOPY_ARG\"" | \
diff -Naur openssh-6.3p1-orig/regress/test-exec.sh openssh-6.3p1/regress/test-exec.sh
--- openssh-6.3p1-orig/regress/test-exec.sh	2013-08-08 11:02:12.000000000 +0400
+++ openssh-6.3p1/regress/test-exec.sh	2013-10-20 21:04:13.585800000 +0400
@@ -16,9 +16,16 @@
 	os=cygwin
 	TEST_SSH_IPV6=no
 	;;
+MSYS_NT-5.0)
+	os=cygwin
+	TEST_SSH_IPV6=no
+	;;
 CYGWIN*)
 	os=cygwin
 	;;
+MSYS*)
+	os=cygwin
+	;;
 esac
 
 if [ ! -z "$TEST_SSH_PORT" ]; then
