From fce067196f6587d01d204bad74fa5b7420ee6a16 Mon Sep 17 00:00:00 2001
From: Evgeny Karpov <eukarpov@gmail.com>
Date: Wed, 6 Dec 2023 22:26:03 +0100
Subject: [PATCH 5/8] Fix reading double from a string on aarch64 (#4)

---
 mingw-w64-crt/gdtoa/strtodnrp.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/mingw-w64-crt/gdtoa/strtodnrp.c b/mingw-w64-crt/gdtoa/strtodnrp.c
index 9304866bf..38b75ec6e 100644
--- a/mingw-w64-crt/gdtoa/strtodnrp.c
+++ b/mingw-w64-crt/gdtoa/strtodnrp.c
@@ -90,6 +90,8 @@ __mingw_strtod (const char * __restrict__ src, char ** __restrict__ endptr)
  * long double functions as aliases to __strtod. */
 
 long double __cdecl
-__mingw_strtold (const char * __restrict__ src, char ** __restrict__ endptr)
-  __attribute__((alias("__strtod")));
+__mingw_strtold (const char * __restrict__ src, char ** __restrict__ endptr) 
+{
+	return __strtod(src, endptr);
+}
 #endif
