--- gcc/config/i386/cygwin.h	2013-01-10 14:38:27.000000000 -0600
+++ gcc/config/i386/cygwin.h	2013-03-17 01:31:17.078372600 -0500
@@ -137,5 +137,5 @@
 #define LIBGCC_SONAME "cyggcc_s" LIBGCC_EH_EXTN "-1.dll"
 
 /* We should find a way to not have to update this manually.  */
-#define LIBGCJ_SONAME "cyggcj" /*LIBGCC_EH_EXTN*/ "-13.dll"
+#define LIBGCJ_SONAME "cyggcj" /*LIBGCC_EH_EXTN*/ "-14.dll"
 
--- gcc/config/i386/mingw32.h	2013-01-10 14:38:27.000000000 -0600
+++ gcc/config/i386/mingw32.h	2013-03-17 01:31:17.078372600 -0500
@@ -245,4 +245,4 @@
 #define LIBGCC_SONAME "libgcc_s" LIBGCC_EH_EXTN "-1.dll"
 
 /* We should find a way to not have to update this manually.  */
-#define LIBGCJ_SONAME "libgcj" /*LIBGCC_EH_EXTN*/ "-13.dll"
+#define LIBGCJ_SONAME "libgcj" /*LIBGCC_EH_EXTN*/ "-14.dll"
