--- curl-7.74.0/libcurl.pc.in.orig	2020-12-15 00:25:37.098392800 +0100
+++ curl-7.74.0/libcurl.pc.in	2020-12-15 00:25:44.117463900 +0100
@@ -34,6 +34,7 @@
 URL: https://curl.se/
 Description: Library to transfer files with ftp, http, etc.
 Version: @CURLVERSION@
+Requires.private: libidn2 libbrotlidec
 Libs: -L${libdir} -lcurl @LIBCURL_NO_SHARED@
 Libs.private: @LIBCURL_LIBS@
 Cflags: -I${includedir} @CPPFLAG_CURL_STATICLIB@
