--- curl-7.80.0/configure.ac.orig	2021-11-09 08:35:38.000000000 +0100
+++ curl-7.80.0/configure.ac	2021-11-15 22:56:06.438790700 +0100
@@ -1682,12 +1682,12 @@
 if test x"$want_gss" = xyes; then
   AC_MSG_RESULT(yes)
 
-  CURL_CHECK_PKGCONFIG(mit-krb5-gssapi)
+  CURL_CHECK_PKGCONFIG(krb5-gssapi)
   if test -z "$GSSAPI_INCS"; then
      if test -n "$host_alias" -a -f "$GSSAPI_ROOT/bin/$host_alias-krb5-config"; then
         GSSAPI_INCS=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --cflags gssapi`
      elif test "$PKGCONFIG" != "no" ; then
-        GSSAPI_INCS=`$PKGCONFIG --cflags mit-krb5-gssapi`
+        GSSAPI_INCS=`$PKGCONFIG --cflags krb5-gssapi`
      elif test "$GSSAPI_ROOT" != "yes"; then
         GSSAPI_INCS="-I$GSSAPI_ROOT/include"
      fi
@@ -1772,15 +1772,16 @@
         LIBS="-lgssapi_krb5 -lresolv $LIBS"
         ;;
      *)
-        CURL_CHECK_PKGCONFIG(mit-krb5-gssapi)
+        CURL_CHECK_PKGCONFIG(krb5-gssapi)
         if test -n "$host_alias" -a -f "$GSSAPI_ROOT/bin/$host_alias-krb5-config"; then
            dnl krb5-config doesn't have --libs-only-L or similar, put everything
            dnl into LIBS
-           gss_libs=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --libs gssapi`
+           gss_libs=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --deps --libs gssapi`
            LIBS="$gss_libs $LIBS"
         elif test "$PKGCONFIG" != "no" ; then
-           gss_libs=`$PKGCONFIG --libs mit-krb5-gssapi`
-           LIBS="$gss_libs $LIBS"
+           gss_libs=`$PKGCONFIG --static --libs krb5-gssapi`
+           # XXX: sqlite3 is missing due to https://github.com/msys2/MSYS2-packages/issues/2701
+           LIBS="$gss_libs -lsqlite3 $LIBS"
         else
            case $host in
            *-hp-hpux*)
