diff -Naur curl-7.31.0-orig/compile curl-7.31.0/compile
--- curl-7.31.0-orig/compile	2013-06-20 01:43:19.000000000 +0400
+++ curl-7.31.0/compile	2013-08-09 23:00:52.279296900 +0400
@@ -53,7 +53,7 @@
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN* | MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/* | msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -Naur curl-7.31.0-orig/config.guess curl-7.31.0/config.guess
--- curl-7.31.0-orig/config.guess	2013-06-20 01:43:19.000000000 +0400
+++ curl-7.31.0/config.guess	2013-08-09 23:00:52.294921900 +0400
@@ -866,6 +866,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
diff -Naur curl-7.31.0-orig/m4/xc-lt-iface.m4 curl-7.31.0/m4/xc-lt-iface.m4
--- curl-7.31.0-orig/m4/xc-lt-iface.m4	2013-06-22 02:29:04.000000000 +0400
+++ curl-7.31.0/m4/xc-lt-iface.m4	2013-08-09 23:00:52.529296900 +0400
@@ -263,7 +263,7 @@
   xc_lt_shlib_use_no_undefined='yes'
 fi
 case $host_os in @%:@ (
-  cygwin* | mingw* | pw32* | cegcc* | os2* | aix*)
+  cygwin* | msys* | mingw* | pw32* | cegcc* | os2* | aix*)
     xc_lt_shlib_use_no_undefined='yes'
     ;;
 esac
