diff -Naur curl-7.31.0-orig/config.guess curl-7.31.0/config.guess
--- curl-7.31.0-orig/config.guess	2013-06-20 01:43:19.000000000 +0400
+++ curl-7.31.0/config.guess	2013-08-09 23:00:52.294921900 +0400
@@ -866,6 +866,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2`echo ${UNAME_RELEASE}|sed -e 's/[^.]*//'`
 	exit ;;
diff -Naur curl-7.31.0-orig/m4/xc-lt-iface.m4 curl-7.31.0/m4/xc-lt-iface.m4
--- curl-7.31.0-orig/m4/xc-lt-iface.m4	2013-06-22 02:29:04.000000000 +0400
+++ curl-7.31.0/m4/xc-lt-iface.m4	2013-08-09 23:00:52.529296900 +0400
@@ -263,7 +263,7 @@
   xc_lt_shlib_use_no_undefined='yes'
 fi
 case $host_os in @%:@ (
-  cygwin* | mingw* | pw32* | cegcc* | os2* | aix*)
+  cygwin* | msys* | mingw* | pw32* | cegcc* | os2* | aix*)
     xc_lt_shlib_use_no_undefined='yes'
     ;;
 esac
--- curl-7.55.1/m4/curl-compilers.m4.orig	2017-09-13 18:35:58.451704400 -0400
+++ curl-7.55.1/m4/curl-compilers.m4	2017-09-13 18:39:22.721632800 -0400
@@ -904,7 +904,7 @@ AC_DEFUN([CURL_SET_COMPILER_WARNING_OPTS
           dnl Only clang 3.2 or later
           if test "$compiler_num" -ge "302"; then
             case $host_os in
-            cygwin* | mingw*)
+            cygwin* | msys* | mingw*)
               dnl skip missing-variable-declarations warnings for cygwin and
               dnl mingw because the libtool wrapper executable causes them
               ;;
