--- curl-7.60.0/configure.ac.orig	2018-07-06 15:08:41.153181400 +0300
+++ curl-7.60.0/configure.ac	2018-07-06 15:08:45.489989000 +0300
@@ -1389,12 +1389,12 @@
         if test -n "$host_alias" -a -f "$GSSAPI_ROOT/bin/$host_alias-krb5-config"; then
            dnl krb5-config doesn't have --libs-only-L or similar, put everything
            dnl into LIBS
-           gss_libs=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --libs gssapi`
+           gss_libs=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --deps --libs gssapi`
            LIBS="$gss_libs $LIBS"
         elif test -f "$KRB5CONFIG"; then
            dnl krb5-config doesn't have --libs-only-L or similar, put everything
            dnl into LIBS
-           gss_libs=`$KRB5CONFIG --libs gssapi`
+           gss_libs=`$KRB5CONFIG --deps --libs gssapi`
            LIBS="$gss_libs $LIBS"
         else
            case $host in
