--- curl-7.82.0/configure.ac.orig	2022-04-03 13:21:15.513746700 +0200
+++ curl-7.82.0/configure.ac	2022-04-03 13:27:03.712294600 +0200
@@ -1704,15 +1704,15 @@
   AC_MSG_RESULT(yes)
 
   if test $GSSAPI_ROOT != "/usr"; then
-    CURL_CHECK_PKGCONFIG(mit-krb5-gssapi, $GSSAPI_ROOT/lib/pkgconfig)
+    CURL_CHECK_PKGCONFIG(krb5-gssapi, $GSSAPI_ROOT/lib/pkgconfig)
   else
-    CURL_CHECK_PKGCONFIG(mit-krb5-gssapi)
+    CURL_CHECK_PKGCONFIG(krb5-gssapi)
   fi
   if test -z "$GSSAPI_INCS"; then
      if test -n "$host_alias" -a -f "$GSSAPI_ROOT/bin/$host_alias-krb5-config"; then
-        GSSAPI_INCS=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --cflags gssapi`
+        GSSAPI_INCS=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --deps --libs --cflags gssapi`
      elif test "$PKGCONFIG" != "no" ; then
-        GSSAPI_INCS=`$PKGCONFIG --cflags mit-krb5-gssapi`
+        GSSAPI_INCS=`$PKGCONFIG --cflags krb5-gssapi`
      elif test -f "$KRB5CONFIG"; then
         GSSAPI_INCS=`$KRB5CONFIG --cflags gssapi`
      elif test "$GSSAPI_ROOT" != "yes"; then
@@ -1800,17 +1800,17 @@
         ;;
      *)
         if test $GSSAPI_ROOT != "/usr"; then
-          CURL_CHECK_PKGCONFIG(mit-krb5-gssapi, $GSSAPI_ROOT/lib/pkgconfig)
+          CURL_CHECK_PKGCONFIG(krb5-gssapi, $GSSAPI_ROOT/lib/pkgconfig)
         else
-          CURL_CHECK_PKGCONFIG(mit-krb5-gssapi)
+          CURL_CHECK_PKGCONFIG(krb5-gssapi)
         fi
         if test -n "$host_alias" -a -f "$GSSAPI_ROOT/bin/$host_alias-krb5-config"; then
            dnl krb5-config doesn't have --libs-only-L or similar, put everything
            dnl into LIBS
-           gss_libs=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --libs gssapi`
+           gss_libs=`$GSSAPI_ROOT/bin/$host_alias-krb5-config --deps --libs gssapi`
            LIBS="$gss_libs $LIBS"
         elif test "$PKGCONFIG" != "no" ; then
-           gss_libs=`$PKGCONFIG --libs mit-krb5-gssapi`
+           gss_libs=`$PKGCONFIG --static --libs krb5-gssapi`
            LIBS="$gss_libs $LIBS"
         elif test -f "$KRB5CONFIG"; then
            dnl krb5-config doesn't have --libs-only-L or similar, put everything
