--- a/src/python_interpreter/mod.rs.orig	2025-11-02 16:41:44.189964600 +0100
+++ a/src/python_interpreter/mod.rs	2025-11-02 16:42:33.331858400 +0100
@@ -292,8 +292,8 @@
     if bridge != &BridgeModel::Cffi
         && target.get_python_os() != message.system
         && !target.cross_compiling()
-        && !(target.get_python_os() == "cygwin"
-            && message.system.to_lowercase().starts_with("cygwin"))
+        && !((target.get_python_os() == "cygwin" && message.system.to_lowercase().starts_with("cygwin"))
+            || (target.get_python_os() == "cygwin" && message.system.to_lowercase().starts_with("msys")))
     {
         bail!(
             "platform.system() in python, {}, and the rust target, {:?}, don't match ಠ_ಠ",
