--- a/src/lib.rs
+++ b/src/lib.rs
@@ -87,6 +87,7 @@ cfg_if! {
         target_os = "freebsd",
         target_os = "hurd",
         target_os = "illumos",
+        target_os = "cygwin",
         // Check for target_arch = "arm" to only include the 3DS. Does not
         // include the Nintendo Switch (which is target_arch = "aarch64").
         all(target_os = "horizon", target_arch = "arm"),
diff --git a/src/util_libc.rs b/src/util_libc.rs
index 80003013..24c53c0c 100644
--- a/src/util_libc.rs
+++ b/src/util_libc.rs
@@ -2,7 +2,7 @@ use crate::Error;
 use core::mem::MaybeUninit;
 
 cfg_if! {
-    if #[cfg(any(target_os = "netbsd", target_os = "openbsd", target_os = "android"))] {
+    if #[cfg(any(target_os = "netbsd", target_os = "openbsd", target_os = "android", target_os = "cygwin"))] {
         use libc::__errno as errno_location;
     } else if #[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "hurd", target_os = "redox", target_os = "dragonfly"))] {
         use libc::__errno_location as errno_location;
