From c0ae0e050f995631bc04e99de7d6b046f6c52d59 Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgad.shaneh@audiocodes.com>
Date: Mon, 12 Apr 2021 18:49:46 +0300
Subject: [PATCH 3/3] Disable man symlinks

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9df65cf0..f8fd3495 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2702,7 +2702,7 @@ if(NOT MSVC)
     if(MINGW)
         find_program(LINK_EXECUTABLE ln)
     endif(MINGW)
-    if(UNIX OR (MINGW AND LINK_EXECUTABLE))
+    if(FALSE)
         set(MAN1 "")
         foreach(MANPAGE ${MAN1_NOEXPAND})
             set(MAN1 ${MAN1} ${CMAKE_CURRENT_SOURCE_DIR}/${MANPAGE})
@@ -2752,7 +2752,7 @@ if(NOT MSVC)
             set(MANMISC ${MANMISC} ${CMAKE_CURRENT_BINARY_DIR}/${MANPAGE})
         endforeach(TEMPLATE_MANPAGE)
         install(FILES ${MANMISC} DESTINATION ${CMAKE_INSTALL_MANDIR}/man${MAN_MISC_INFO})
-    endif(UNIX OR (MINGW AND LINK_EXECUTABLE))
+    endif(FALSE)
 endif(NOT MSVC)
 
 # uninstall target
-- 
2.31.0.windows.1

