--- origsrc/zlib-1.2.8/zconf.h	2013-04-28 18:57:11.000000000 -0400
+++ src/zlib-1.2.8/zconf.h	2013-05-09 16:40:53.393931700 -0400
@@ -408,11 +408,11 @@ typedef uLong FAR uLongf;
    typedef unsigned long z_crc_t;
 #endif
 
-#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */
+#if 1                   /* HAVE_UNISTD_H may be set to #if 1 by ./configure */
 #  define Z_HAVE_UNISTD_H
 #endif
 
-#ifdef HAVE_STDARG_H    /* may be set to #if 1 by ./configure */
+#if 1                   /* HAVE_STDARG_H may be set to #if 1 by ./configure */
 #  define Z_HAVE_STDARG_H
 #endif
 
