--- origsrc/zlib-1.2.13/configure	2022-10-10 20:11:12.000000000 +0200
+++ src/zlib-1.2.13/configure	2022-10-15 19:47:10.544034900 +0200
@@ -232,6 +232,7 @@ if test "$gcc" -eq 1 && ($cc -c $test.c)
         LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,${SRCDIR}zlib.map"}
         LDCONFIG="ldconfig -m" ;;
   CYGWIN* | Cygwin* | cygwin* | *-cygwin* | OS/2*)
+        LDSHARED=${LDSHARED-"$cc -shared"}
         EXE='.exe' ;;
   MINGW* | mingw* | *-mingw*)
         rm -f $test.[co] $test $test$shared_ext
