diff -rupN old/build/autoconf/compile new/build/autoconf/compile
--- old/build/autoconf/compile	2017-04-23 16:39:23.355759000 -0500
+++ new/build/autoconf/compile	2017-04-23 16:41:14.016632500 -0500
@@ -53,7 +53,7 @@ func_file_conv ()
 	  MINGW*)
 	    file_conv=mingw
 	    ;;
-	  CYGWIN*)
+	  CYGWIN*|MSYS*)
 	    file_conv=cygwin
 	    ;;
 	  *)
@@ -67,7 +67,7 @@ func_file_conv ()
 	mingw/*)
 	  file=`cmd //C echo "$file " | sed -e 's/"\(.*\) " *$/\1/'`
 	  ;;
-	cygwin/*)
+	cygwin/*|msys/*)
 	  file=`cygpath -m "$file" || echo "$file"`
 	  ;;
 	wine/*)
diff -rupN old/build/autoconf/config.guess new/build/autoconf/config.guess
--- old/build/autoconf/config.guess	2017-04-23 16:39:23.359756400 -0500
+++ new/build/autoconf/config.guess	2017-04-23 16:43:56.289990900 -0500
@@ -891,6 +891,9 @@ EOF
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     prep*:SunOS:5.*:*)
 	echo powerpcle-unknown-solaris2"`echo "$UNAME_RELEASE"|sed -e 's/[^.]*//'`"
 	exit ;;
