--- origsrc/scons-2.3.0/engine/SCons/Platform/cygwin.py	2013-03-03 06:48:39.000000000 -0800
+++ src/scons-2.3.0/engine/SCons/Platform/cygwin.py	2013-11-03 11:05:30.850047400 -0800
@@ -38,6 +38,9 @@ from SCons.Platform import TempFileMunge
 def generate(env):
     posix.generate(env)
 
+    # /bin/gcc doesn't work on Cygwin because gcc attempts to
+    # find libexec as /bin/../libexec. So prefer /usr/bin instead.
+    env['ENV']['PATH']    = '/usr/local/bin:/opt/bin:/usr/bin:/bin'
     env['PROGPREFIX']  = ''
     env['PROGSUFFIX']  = '.exe'
     env['SHLIBPREFIX'] = ''
