--- origsrc/gnome-doc-utils-0.18.0/bootstrap.make	2009-09-07 19:38:59.000000000 -0500
+++ src/gnome-doc-utils-0.18.0/bootstrap.make	2009-10-26 23:19:25.335732900 -0500
@@ -1,4 +1,4 @@
-_xml2po = PYTHONPATH="$(shell pwd)/$(top_builddir)/xml2po:$(shell pwd)/$(top_srcdir)/xml2po:$(PYTHONPATH)" "$(shell pwd)/$(top_builddir)/xml2po/xml2po/xml2po"
+_xml2po = PYTHONPATH="$(abs_top_builddir)/xml2po:$(abs_top_srcdir)/xml2po:$(PYTHONPATH)" "$(abs_top_builddir)/xml2po/xml2po/xml2po"
 
 _db2html = $(top_srcdir)/xslt/docbook/html/db2html.xsl
 _db2omf  = $(top_srcdir)/xslt/docbook/omf/db2omf.xsl
