diff -Naur gnome-doc-utils-0.20.10-orig/config.guess gnome-doc-utils-0.20.10/config.guess
--- gnome-doc-utils-0.20.10-orig/config.guess	2012-03-26 22:23:18.000000000 +0400
+++ gnome-doc-utils-0.20.10/config.guess	2014-07-23 15:34:04.129800000 +0400
@@ -800,6 +800,9 @@
     i*:CYGWIN*:*)
 	echo ${UNAME_MACHINE}-pc-cygwin
 	exit ;;
+    i*:MSYS*:*)
+	echo ${UNAME_MACHINE}-pc-msys
+	exit ;;
     *:MINGW*:*)
 	echo ${UNAME_MACHINE}-pc-mingw32
 	exit ;;
@@ -837,7 +840,10 @@
     i*:UWIN*:*)
 	echo ${UNAME_MACHINE}-pc-uwin
 	exit ;;
-    amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
+    amd64:CYGWIN*:*:* | x86_64:*:*:*)
+	echo x86_64-unknown-cygwin
+	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
     p*:CYGWIN*:*)
diff -Naur gnome-doc-utils-0.20.10-orig/config.sub gnome-doc-utils-0.20.10/config.sub
--- gnome-doc-utils-0.20.10-orig/config.sub	2012-03-26 22:23:18.000000000 +0400
+++ gnome-doc-utils-0.20.10/config.sub	2014-07-23 15:31:40.609800000 +0400
@@ -1293,7 +1293,7 @@
 	      | -ptx* | -coff* | -ecoff* | -winnt* | -domain* | -vsta* \
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
-	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
+	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
 	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
