diff -durN neon-0.30.2.orig/config.guess neon-0.30.2/config.guess
--- neon-0.30.2.orig/config.guess	2016-09-30 17:33:33.000000000 +0800
+++ neon-0.30.2/config.guess	2017-05-20 10:55:55.421875000 +0800
@@ -866,6 +866,9 @@
     amd64:CYGWIN*:*:* | x86_64:CYGWIN*:*:*)
 	echo x86_64-unknown-cygwin
 	exit ;;
+    amd64:MSYS*:*:* | x86_64:MSYS*:*:*)
+	echo x86_64-unknown-msys
+	exit ;;
     p*:CYGWIN*:*)
 	echo powerpcle-unknown-cygwin
 	exit ;;
